// @apexfintechsolutions/ascend-sdk/sdk/scheduletransfers.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleTransfers = void 0;
const scheduleTransfersCancelAchDepositSchedule_js_1 = require("../funcs/scheduleTransfersCancelAchDepositSchedule.js");
const scheduleTransfersCancelAchWithdrawalSchedule_js_1 = require("../funcs/scheduleTransfersCancelAchWithdrawalSchedule.js");
const scheduleTransfersCancelWireWithdrawalSchedule_js_1 = require("../funcs/scheduleTransfersCancelWireWithdrawalSchedule.js");
const scheduleTransfersCreateAchDepositSchedule_js_1 = require("../funcs/scheduleTransfersCreateAchDepositSchedule.js");
const scheduleTransfersCreateAchWithdrawalSchedule_js_1 = require("../funcs/scheduleTransfersCreateAchWithdrawalSchedule.js");
const scheduleTransfersCreateWireWithdrawalSchedule_js_1 = require("../funcs/scheduleTransfersCreateWireWithdrawalSchedule.js");
const scheduleTransfersGetAchDepositSchedule_js_1 = require("../funcs/scheduleTransfersGetAchDepositSchedule.js");
const scheduleTransfersGetAchWithdrawalSchedule_js_1 = require("../funcs/scheduleTransfersGetAchWithdrawalSchedule.js");
const scheduleTransfersGetWireWithdrawalSchedule_js_1 = require("../funcs/scheduleTransfersGetWireWithdrawalSchedule.js");
const scheduleTransfersListAchDepositSchedules_js_1 = require("../funcs/scheduleTransfersListAchDepositSchedules.js");
const scheduleTransfersListAchWithdrawalSchedules_js_1 = require("../funcs/scheduleTransfersListAchWithdrawalSchedules.js");
const scheduleTransfersListScheduleSummaries_js_1 = require("../funcs/scheduleTransfersListScheduleSummaries.js");
const scheduleTransfersListWireWithdrawalSchedules_js_1 = require("../funcs/scheduleTransfersListWireWithdrawalSchedules.js");
const scheduleTransfersUpdateAchDepositSchedule_js_1 = require("../funcs/scheduleTransfersUpdateAchDepositSchedule.js");
const scheduleTransfersUpdateAchWithdrawalSchedule_js_1 = require("../funcs/scheduleTransfersUpdateAchWithdrawalSchedule.js");
const scheduleTransfersUpdateWireWithdrawalSchedule_js_1 = require("../funcs/scheduleTransfersUpdateWireWithdrawalSchedule.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
class ScheduleTransfers extends sdks_js_1.ClientSDK {
    /**
     * List Schedule Summaries
     *
     * @remarks
     * Lists transfer schedule summaries that match the filter in the request
     */
    async listScheduleSummaries(filter, pageSize, pageToken, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersListScheduleSummaries_js_1.scheduleTransfersListScheduleSummaries)(this, filter, pageSize, pageToken, options));
    }
    /**
     * Create ACH Deposit Schedule
     *
     * @remarks
     * Creates an ACH deposit transfer schedule
     */
    async createAchDepositSchedule(achDepositScheduleCreate, accountId, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersCreateAchDepositSchedule_js_1.scheduleTransfersCreateAchDepositSchedule)(this, achDepositScheduleCreate, accountId, options));
    }
    /**
     * List ACH Deposit Schedules
     *
     * @remarks
     * Return a list of ACH deposit schedules for the specified account and filter params
     */
    async listAchDepositSchedules(accountId, filter, pageSize, pageToken, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersListAchDepositSchedules_js_1.scheduleTransfersListAchDepositSchedules)(this, accountId, filter, pageSize, pageToken, options));
    }
    /**
     * Get ACH Deposit Schedule
     *
     * @remarks
     * Gets an ACH deposit transfer schedule
     */
    async getAchDepositSchedule(accountId, achDepositScheduleId, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersGetAchDepositSchedule_js_1.scheduleTransfersGetAchDepositSchedule)(this, accountId, achDepositScheduleId, options));
    }
    /**
     * Update ACH Deposit Schedules
     *
     * @remarks
     * Updates the amount of an ACH deposit transfer schedule
     */
    async updateAchDepositSchedule(achDepositScheduleUpdate, accountId, achDepositScheduleId, updateMask, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersUpdateAchDepositSchedule_js_1.scheduleTransfersUpdateAchDepositSchedule)(this, achDepositScheduleUpdate, accountId, achDepositScheduleId, updateMask, options));
    }
    /**
     * Cancel ACH Deposit Schedule
     *
     * @remarks
     * Cancels an ACH deposit transfer schedule
     */
    async cancelAchDepositSchedule(cancelAchDepositScheduleRequestCreate, accountId, achDepositScheduleId, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersCancelAchDepositSchedule_js_1.scheduleTransfersCancelAchDepositSchedule)(this, cancelAchDepositScheduleRequestCreate, accountId, achDepositScheduleId, options));
    }
    /**
     * Create ACH Withdrawal Schedule
     *
     * @remarks
     * Creates an ACH withdrawal transfer schedule
     */
    async createAchWithdrawalSchedule(achWithdrawalScheduleCreate, accountId, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersCreateAchWithdrawalSchedule_js_1.scheduleTransfersCreateAchWithdrawalSchedule)(this, achWithdrawalScheduleCreate, accountId, options));
    }
    /**
     * List ACH Withdrawal Schedules
     *
     * @remarks
     * Return a list of ACH withdrawal schedules for the specified account and filter params
     */
    async listAchWithdrawalSchedules(accountId, filter, pageSize, pageToken, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersListAchWithdrawalSchedules_js_1.scheduleTransfersListAchWithdrawalSchedules)(this, accountId, filter, pageSize, pageToken, options));
    }
    /**
     * Get ACH Withdrawal Schedule
     *
     * @remarks
     * Gets an ACH withdrawal transfer schedule
     */
    async getAchWithdrawalSchedule(accountId, achWithdrawalScheduleId, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersGetAchWithdrawalSchedule_js_1.scheduleTransfersGetAchWithdrawalSchedule)(this, accountId, achWithdrawalScheduleId, options));
    }
    /**
     * Update ACH Withdrawal Schedule
     *
     * @remarks
     * Updates the amount of an ACH withdrawal transfer schedule
     */
    async updateAchWithdrawalSchedule(achWithdrawalScheduleUpdate, accountId, achWithdrawalScheduleId, updateMask, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersUpdateAchWithdrawalSchedule_js_1.scheduleTransfersUpdateAchWithdrawalSchedule)(this, achWithdrawalScheduleUpdate, accountId, achWithdrawalScheduleId, updateMask, options));
    }
    /**
     * Cancel ACH Withdrawal Schedule
     *
     * @remarks
     * Cancels an ACH withdrawal transfer schedule
     */
    async cancelAchWithdrawalSchedule(cancelAchWithdrawalScheduleRequestCreate, accountId, achWithdrawalScheduleId, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersCancelAchWithdrawalSchedule_js_1.scheduleTransfersCancelAchWithdrawalSchedule)(this, cancelAchWithdrawalScheduleRequestCreate, accountId, achWithdrawalScheduleId, options));
    }
    /**
     * Create Wire Withdrawal Schedule
     *
     * @remarks
     * Creates a Wire withdrawal transfer schedule
     */
    async createWireWithdrawalSchedule(wireWithdrawalScheduleCreate, accountId, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersCreateWireWithdrawalSchedule_js_1.scheduleTransfersCreateWireWithdrawalSchedule)(this, wireWithdrawalScheduleCreate, accountId, options));
    }
    /**
     * List Wire Withdrawal Schedules
     *
     * @remarks
     * Return a list of Wire withdrawal schedules for the specified account and filter params
     */
    async listWireWithdrawalSchedules(accountId, filter, pageSize, pageToken, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersListWireWithdrawalSchedules_js_1.scheduleTransfersListWireWithdrawalSchedules)(this, accountId, filter, pageSize, pageToken, options));
    }
    /**
     * Get Wire Withdrawal Schedule
     *
     * @remarks
     * Gets a Wire withdrawal transfer schedule
     */
    async getWireWithdrawalSchedule(accountId, wireWithdrawalScheduleId, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersGetWireWithdrawalSchedule_js_1.scheduleTransfersGetWireWithdrawalSchedule)(this, accountId, wireWithdrawalScheduleId, options));
    }
    /**
     * Update Wire Withdrawal Schedule
     *
     * @remarks
     * Updates the amount of a Wire withdrawal transfer schedule
     */
    async updateWireWithdrawalSchedule(wireWithdrawalScheduleUpdate, accountId, wireWithdrawalScheduleId, updateMask, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersUpdateWireWithdrawalSchedule_js_1.scheduleTransfersUpdateWireWithdrawalSchedule)(this, wireWithdrawalScheduleUpdate, accountId, wireWithdrawalScheduleId, updateMask, options));
    }
    /**
     * Cancel Wire Withdrawal Schedule
     *
     * @remarks
     * Cancels a Wire withdrawal transfer schedule
     */
    async cancelWireWithdrawalSchedule(cancelWireWithdrawalScheduleRequestCreate, accountId, wireWithdrawalScheduleId, options) {
        return (0, fp_js_1.unwrapAsync)((0, scheduleTransfersCancelWireWithdrawalSchedule_js_1.scheduleTransfersCancelWireWithdrawalSchedule)(this, cancelWireWithdrawalScheduleRequestCreate, accountId, wireWithdrawalScheduleId, options));
    }
}
exports.ScheduleTransfers = ScheduleTransfers;
