// @apexfintechsolutions/ascend-sdk/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Apexascend = void 0;
const sdks_js_1 = require("../lib/sdks.js");
const accountcreation_js_1 = require("./accountcreation.js");
const accountmanagement_js_1 = require("./accountmanagement.js");
const accounttransfers_js_1 = require("./accounttransfers.js");
const achtransfers_js_1 = require("./achtransfers.js");
const assets_js_1 = require("./assets.js");
const authentication_js_1 = require("./authentication.js");
const bankrelationships_js_1 = require("./bankrelationships.js");
const basketorders_js_1 = require("./basketorders.js");
const cashbalances_js_1 = require("./cashbalances.js");
const createorder_js_1 = require("./createorder.js");
const dataretrieval_js_1 = require("./dataretrieval.js");
const enrollmentsandagreements_js_1 = require("./enrollmentsandagreements.js");
const feesandcredits_js_1 = require("./feesandcredits.js");
const fixedincomepricing_js_1 = require("./fixedincomepricing.js");
const instantcashtransferict_js_1 = require("./instantcashtransferict.js");
const investigations_js_1 = require("./investigations.js");
const investordocs_js_1 = require("./investordocs.js");
const journals_js_1 = require("./journals.js");
const ledger_js_1 = require("./ledger.js");
const margins_js_1 = require("./margins.js");
const personmanagement_js_1 = require("./personmanagement.js");
const reader_js_1 = require("./reader.js");
const retirements_js_1 = require("./retirements.js");
const scheduletransfers_js_1 = require("./scheduletransfers.js");
const subscriber_js_1 = require("./subscriber.js");
const testsimulation_js_1 = require("./testsimulation.js");
const tradeallocation_js_1 = require("./tradeallocation.js");
const tradebooking_js_1 = require("./tradebooking.js");
const wires_js_1 = require("./wires.js");
class Apexascend extends sdks_js_1.ClientSDK {
    get authentication() {
        var _a;
        return ((_a = this._authentication) !== null && _a !== void 0 ? _a : (this._authentication = new authentication_js_1.Authentication(this._options)));
    }
    get reader() {
        var _a;
        return ((_a = this._reader) !== null && _a !== void 0 ? _a : (this._reader = new reader_js_1.Reader(this._options)));
    }
    get subscriber() {
        var _a;
        return ((_a = this._subscriber) !== null && _a !== void 0 ? _a : (this._subscriber = new subscriber_js_1.Subscriber(this._options)));
    }
    get personManagement() {
        var _a;
        return ((_a = this._personManagement) !== null && _a !== void 0 ? _a : (this._personManagement = new personmanagement_js_1.PersonManagement(this._options)));
    }
    get accountCreation() {
        var _a;
        return ((_a = this._accountCreation) !== null && _a !== void 0 ? _a : (this._accountCreation = new accountcreation_js_1.AccountCreation(this._options)));
    }
    get accountManagement() {
        var _a;
        return ((_a = this._accountManagement) !== null && _a !== void 0 ? _a : (this._accountManagement = new accountmanagement_js_1.AccountManagement(this._options)));
    }
    get enrollmentsAndAgreements() {
        var _a;
        return ((_a = this._enrollmentsAndAgreements) !== null && _a !== void 0 ? _a : (this._enrollmentsAndAgreements = new enrollmentsandagreements_js_1.EnrollmentsAndAgreements(this._options)));
    }
    get investigations() {
        var _a;
        return ((_a = this._investigations) !== null && _a !== void 0 ? _a : (this._investigations = new investigations_js_1.Investigations(this._options)));
    }
    get bankRelationships() {
        var _a;
        return ((_a = this._bankRelationships) !== null && _a !== void 0 ? _a : (this._bankRelationships = new bankrelationships_js_1.BankRelationships(this._options)));
    }
    get achTransfers() {
        var _a;
        return ((_a = this._achTransfers) !== null && _a !== void 0 ? _a : (this._achTransfers = new achtransfers_js_1.ACHTransfers(this._options)));
    }
    get instantCashTransferICT() {
        var _a;
        return ((_a = this._instantCashTransferICT) !== null && _a !== void 0 ? _a : (this._instantCashTransferICT = new instantcashtransferict_js_1.InstantCashTransferICT(this._options)));
    }
    get retirements() {
        var _a;
        return ((_a = this._retirements) !== null && _a !== void 0 ? _a : (this._retirements = new retirements_js_1.Retirements(this._options)));
    }
    get journals() {
        var _a;
        return ((_a = this._journals) !== null && _a !== void 0 ? _a : (this._journals = new journals_js_1.Journals(this._options)));
    }
    get scheduleTransfers() {
        var _a;
        return ((_a = this._scheduleTransfers) !== null && _a !== void 0 ? _a : (this._scheduleTransfers = new scheduletransfers_js_1.ScheduleTransfers(this._options)));
    }
    get wires() {
        var _a;
        return ((_a = this._wires) !== null && _a !== void 0 ? _a : (this._wires = new wires_js_1.Wires(this._options)));
    }
    get cashBalances() {
        var _a;
        return ((_a = this._cashBalances) !== null && _a !== void 0 ? _a : (this._cashBalances = new cashbalances_js_1.CashBalances(this._options)));
    }
    get feesAndCredits() {
        var _a;
        return ((_a = this._feesAndCredits) !== null && _a !== void 0 ? _a : (this._feesAndCredits = new feesandcredits_js_1.FeesAndCredits(this._options)));
    }
    get testSimulation() {
        var _a;
        return ((_a = this._testSimulation) !== null && _a !== void 0 ? _a : (this._testSimulation = new testsimulation_js_1.TestSimulation(this._options)));
    }
    get accountTransfers() {
        var _a;
        return ((_a = this._accountTransfers) !== null && _a !== void 0 ? _a : (this._accountTransfers = new accounttransfers_js_1.AccountTransfers(this._options)));
    }
    get createOrder() {
        var _a;
        return ((_a = this._createOrder) !== null && _a !== void 0 ? _a : (this._createOrder = new createorder_js_1.CreateOrder(this._options)));
    }
    get fixedIncomePricing() {
        var _a;
        return ((_a = this._fixedIncomePricing) !== null && _a !== void 0 ? _a : (this._fixedIncomePricing = new fixedincomepricing_js_1.FixedIncomePricing(this._options)));
    }
    get basketOrders() {
        var _a;
        return ((_a = this._basketOrders) !== null && _a !== void 0 ? _a : (this._basketOrders = new basketorders_js_1.BasketOrders(this._options)));
    }
    get tradeBooking() {
        var _a;
        return ((_a = this._tradeBooking) !== null && _a !== void 0 ? _a : (this._tradeBooking = new tradebooking_js_1.TradeBooking(this._options)));
    }
    get tradeAllocation() {
        var _a;
        return ((_a = this._tradeAllocation) !== null && _a !== void 0 ? _a : (this._tradeAllocation = new tradeallocation_js_1.TradeAllocation(this._options)));
    }
    get assets() {
        var _a;
        return ((_a = this._assets) !== null && _a !== void 0 ? _a : (this._assets = new assets_js_1.Assets(this._options)));
    }
    get ledger() {
        var _a;
        return ((_a = this._ledger) !== null && _a !== void 0 ? _a : (this._ledger = new ledger_js_1.Ledger(this._options)));
    }
    get margins() {
        var _a;
        return ((_a = this._margins) !== null && _a !== void 0 ? _a : (this._margins = new margins_js_1.Margins(this._options)));
    }
    get investorDocs() {
        var _a;
        return ((_a = this._investorDocs) !== null && _a !== void 0 ? _a : (this._investorDocs = new investordocs_js_1.InvestorDocs(this._options)));
    }
    get dataRetrieval() {
        var _a;
        return ((_a = this._dataRetrieval) !== null && _a !== void 0 ? _a : (this._dataRetrieval = new dataretrieval_js_1.DataRetrieval(this._options)));
    }
}
exports.Apexascend = Apexascend;
