// @apexfintechsolutions/ascend-sdk/src/funcs/authenticationListSigningKeys.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { ApexascendCore } from "../core.js";
import { encodeFormQuery } from "../lib/encodings.js";
import * as M from "../lib/matchers.js";
import { safeParse } from "../lib/schemas.js";
import { RequestOptions } from "../lib/sdks.js";
import { resolveSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import {
  ConnectionError,
  InvalidRequestError,
  RequestAbortedError,
  RequestTimeoutError,
  UnexpectedClientError,
} from "../models/errors/httpclienterrors.js";
import * as errors from "../models/errors/index.js";
import { SDKError } from "../models/errors/sdkerror.js";
import { SDKValidationError } from "../models/errors/sdkvalidationerror.js";
import * as operations from "../models/operations/index.js";
import { Result } from "../types/fp.js";

/**
 * List Signing Keys
 *
 * @remarks
 * Gets the public signing keys used to verify JSON Web Tokens generated by this service.
 */
export async function authenticationListSigningKeys(
  client: ApexascendCore,
  security: operations.AuthenticationListSigningKeysSecurity,
  pageSize?: number | undefined,
  pageToken?: string | undefined,
  options?: RequestOptions,
): Promise<
  Result<
    operations.AuthenticationListSigningKeysResponse,
    | errors.Status
    | SDKError
    | SDKValidationError
    | UnexpectedClientError
    | InvalidRequestError
    | RequestAbortedError
    | RequestTimeoutError
    | ConnectionError
  >
> {
  const input: operations.AuthenticationListSigningKeysRequest = {
    pageSize: pageSize,
    pageToken: pageToken,
  };

  const parsed = safeParse(
    input,
    (value) =>
      operations.AuthenticationListSigningKeysRequest$outboundSchema.parse(
        value,
      ),
    "Input validation failed",
  );
  if (!parsed.ok) {
    return parsed;
  }
  const payload = parsed.value;
  const body = null;

  const path = pathToFunc("/iam/v1/keys")();

  const query = encodeFormQuery({
    "page_size": payload.page_size,
    "page_token": payload.page_token,
  });

  const headers = new Headers({
    Accept: "application/json",
  });

  const requestSecurity = resolveSecurity(
    [
      {
        fieldName: "x-api-key",
        type: "apiKey:header",
        value: security?.apiKeyAuth,
      },
    ],
  );
  const context = {
    operationID: "Authentication_ListSigningKeys",
    oAuth2Scopes: [],
    securitySource: security,
  };

  const requestRes = client._createRequest(context, {
    security: requestSecurity,
    method: "GET",
    path: path,
    headers: headers,
    query: query,
    body: body,
    timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
  }, options);
  if (!requestRes.ok) {
    return requestRes;
  }
  const req = requestRes.value;

  const doResult = await client._do(req, {
    context,
    errorCodes: ["4XX", "500", "5XX"],
    retryConfig: options?.retries
      || client._options.retryConfig,
    retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
  });
  if (!doResult.ok) {
    return doResult;
  }
  const response = doResult.value;

  const responseFields = {
    HttpMeta: { Response: response, Request: req },
  };

  const [result] = await M.match<
    operations.AuthenticationListSigningKeysResponse,
    | errors.Status
    | SDKError
    | SDKValidationError
    | UnexpectedClientError
    | InvalidRequestError
    | RequestAbortedError
    | RequestTimeoutError
    | ConnectionError
  >(
    M.json(
      200,
      operations.AuthenticationListSigningKeysResponse$inboundSchema,
      { key: "ListSigningKeysResponse" },
    ),
    M.jsonErr(500, errors.Status$inboundSchema),
    M.fail(["4XX", "5XX"]),
    M.json(
      "default",
      operations.AuthenticationListSigningKeysResponse$inboundSchema,
      { key: "Status" },
    ),
  )(response, req, { extraFields: responseFields });
  if (!result.ok) {
    return result;
  }

  return result;
}
