// @apexfintechsolutions/ascend-sdk/src/funcs/fixedIncomePricingRetrieveFixedIncomeMarks.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { ApexascendCore } from "../core.js";
import { encodeJSON, encodeSimple } from "../lib/encodings.js";
import * as M from "../lib/matchers.js";
import { safeParse } from "../lib/schemas.js";
import { RequestOptions } from "../lib/sdks.js";
import { extractSecurity, resolveGlobalSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import * as components from "../models/components/index.js";
import {
  ConnectionError,
  InvalidRequestError,
  RequestAbortedError,
  RequestTimeoutError,
  UnexpectedClientError,
} from "../models/errors/httpclienterrors.js";
import * as errors from "../models/errors/index.js";
import { SDKError } from "../models/errors/sdkerror.js";
import { SDKValidationError } from "../models/errors/sdkvalidationerror.js";
import * as operations from "../models/operations/index.js";
import { Result } from "../types/fp.js";

/**
 * Retrieve Fixed Income Marks
 *
 * @remarks
 * Returns marks for a specified set of Fixed Income assets (up to 100 per request)
 */
export async function fixedIncomePricingRetrieveFixedIncomeMarks(
  client: ApexascendCore,
  retrieveFixedIncomeMarksRequestCreate:
    components.RetrieveFixedIncomeMarksRequestCreate,
  correspondentId: string,
  options?: RequestOptions,
): Promise<
  Result<
    operations.OrderPriceServiceRetrieveFixedIncomeMarksResponse,
    | errors.Status
    | SDKError
    | SDKValidationError
    | UnexpectedClientError
    | InvalidRequestError
    | RequestAbortedError
    | RequestTimeoutError
    | ConnectionError
  >
> {
  const input: operations.OrderPriceServiceRetrieveFixedIncomeMarksRequest = {
    retrieveFixedIncomeMarksRequestCreate:
      retrieveFixedIncomeMarksRequestCreate,
    correspondentId: correspondentId,
  };

  const parsed = safeParse(
    input,
    (value) =>
      operations.OrderPriceServiceRetrieveFixedIncomeMarksRequest$outboundSchema
        .parse(value),
    "Input validation failed",
  );
  if (!parsed.ok) {
    return parsed;
  }
  const payload = parsed.value;
  const body = encodeJSON(
    "body",
    payload.RetrieveFixedIncomeMarksRequestCreate,
    { explode: true },
  );

  const pathParams = {
    correspondent_id: encodeSimple(
      "correspondent_id",
      payload.correspondent_id,
      { explode: false, charEncoding: "percent" },
    ),
  };

  const path = pathToFunc(
    "/trading/v1/correspondents/{correspondent_id}/prices:retrieveFixedIncomeMarks",
  )(pathParams);

  const headers = new Headers({
    "Content-Type": "application/json",
    Accept: "application/json",
  });

  const securityInput = await extractSecurity(client._options.security);
  const context = {
    operationID: "OrderPriceService_RetrieveFixedIncomeMarks",
    oAuth2Scopes: [],
    securitySource: client._options.security,
  };
  const requestSecurity = resolveGlobalSecurity(securityInput);

  const requestRes = client._createRequest(context, {
    security: requestSecurity,
    method: "POST",
    path: path,
    headers: headers,
    body: body,
    timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
  }, options);
  if (!requestRes.ok) {
    return requestRes;
  }
  const req = requestRes.value;

  const doResult = await client._do(req, {
    context,
    errorCodes: ["400", "401", "403", "4XX", "500", "503", "5XX"],
    retryConfig: options?.retries
      || client._options.retryConfig,
    retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
  });
  if (!doResult.ok) {
    return doResult;
  }
  const response = doResult.value;

  const responseFields = {
    HttpMeta: { Response: response, Request: req },
  };

  const [result] = await M.match<
    operations.OrderPriceServiceRetrieveFixedIncomeMarksResponse,
    | errors.Status
    | SDKError
    | SDKValidationError
    | UnexpectedClientError
    | InvalidRequestError
    | RequestAbortedError
    | RequestTimeoutError
    | ConnectionError
  >(
    M.json(
      200,
      operations
        .OrderPriceServiceRetrieveFixedIncomeMarksResponse$inboundSchema,
      { key: "RetrieveFixedIncomeMarksResponse" },
    ),
    M.jsonErr([400, 401, 403, 500, 503], errors.Status$inboundSchema),
    M.fail(["4XX", "5XX"]),
    M.json(
      "default",
      operations
        .OrderPriceServiceRetrieveFixedIncomeMarksResponse$inboundSchema,
      { key: "Status" },
    ),
  )(response, req, { extraFields: responseFields });
  if (!result.ok) {
    return result;
  }

  return result;
}
