// @apexfintechsolutions/ascend-sdk/src/funcs/scheduleTransfersUpdateAchDepositSchedule.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import { ApexascendCore } from "../core.js";
import { encodeFormQuery, encodeJSON, encodeSimple } from "../lib/encodings.js";
import * as M from "../lib/matchers.js";
import { safeParse } from "../lib/schemas.js";
import { RequestOptions } from "../lib/sdks.js";
import { extractSecurity, resolveGlobalSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import * as components from "../models/components/index.js";
import {
  ConnectionError,
  InvalidRequestError,
  RequestAbortedError,
  RequestTimeoutError,
  UnexpectedClientError,
} from "../models/errors/httpclienterrors.js";
import * as errors from "../models/errors/index.js";
import { SDKError } from "../models/errors/sdkerror.js";
import { SDKValidationError } from "../models/errors/sdkvalidationerror.js";
import * as operations from "../models/operations/index.js";
import { Result } from "../types/fp.js";

/**
 * Update ACH Deposit Schedules
 *
 * @remarks
 * Updates the amount of an ACH deposit transfer schedule
 */
export async function scheduleTransfersUpdateAchDepositSchedule(
  client: ApexascendCore,
  achDepositScheduleUpdate: components.AchDepositScheduleUpdate,
  accountId: string,
  achDepositScheduleId: string,
  updateMask?: string | undefined,
  options?: RequestOptions,
): Promise<
  Result<
    operations.AchDepositSchedulesUpdateAchDepositScheduleResponse,
    | errors.Status
    | SDKError
    | SDKValidationError
    | UnexpectedClientError
    | InvalidRequestError
    | RequestAbortedError
    | RequestTimeoutError
    | ConnectionError
  >
> {
  const input: operations.AchDepositSchedulesUpdateAchDepositScheduleRequest = {
    achDepositScheduleUpdate: achDepositScheduleUpdate,
    accountId: accountId,
    achDepositScheduleId: achDepositScheduleId,
    updateMask: updateMask,
  };

  const parsed = safeParse(
    input,
    (value) =>
      operations
        .AchDepositSchedulesUpdateAchDepositScheduleRequest$outboundSchema
        .parse(value),
    "Input validation failed",
  );
  if (!parsed.ok) {
    return parsed;
  }
  const payload = parsed.value;
  const body = encodeJSON("body", payload.AchDepositScheduleUpdate, {
    explode: true,
  });

  const pathParams = {
    account_id: encodeSimple("account_id", payload.account_id, {
      explode: false,
      charEncoding: "percent",
    }),
    achDepositSchedule_id: encodeSimple(
      "achDepositSchedule_id",
      payload.achDepositSchedule_id,
      { explode: false, charEncoding: "percent" },
    ),
  };

  const path = pathToFunc(
    "/transfers/v1/accounts/{account_id}/achDepositSchedules/{achDepositSchedule_id}",
  )(pathParams);

  const query = encodeFormQuery({
    "update_mask": payload.update_mask,
  });

  const headers = new Headers({
    "Content-Type": "application/json",
    Accept: "application/json",
  });

  const securityInput = await extractSecurity(client._options.security);
  const context = {
    operationID: "AchDepositSchedules_UpdateAchDepositSchedule",
    oAuth2Scopes: [],
    securitySource: client._options.security,
  };
  const requestSecurity = resolveGlobalSecurity(securityInput);

  const requestRes = client._createRequest(context, {
    security: requestSecurity,
    method: "PATCH",
    path: path,
    headers: headers,
    query: query,
    body: body,
    timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
  }, options);
  if (!requestRes.ok) {
    return requestRes;
  }
  const req = requestRes.value;

  const doResult = await client._do(req, {
    context,
    errorCodes: ["400", "403", "404", "4XX", "5XX"],
    retryConfig: options?.retries
      || client._options.retryConfig,
    retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
  });
  if (!doResult.ok) {
    return doResult;
  }
  const response = doResult.value;

  const responseFields = {
    HttpMeta: { Response: response, Request: req },
  };

  const [result] = await M.match<
    operations.AchDepositSchedulesUpdateAchDepositScheduleResponse,
    | errors.Status
    | SDKError
    | SDKValidationError
    | UnexpectedClientError
    | InvalidRequestError
    | RequestAbortedError
    | RequestTimeoutError
    | ConnectionError
  >(
    M.json(
      200,
      operations
        .AchDepositSchedulesUpdateAchDepositScheduleResponse$inboundSchema,
      { key: "AchDepositSchedule" },
    ),
    M.jsonErr([400, 403, 404], errors.Status$inboundSchema),
    M.fail(["4XX", "5XX"]),
    M.json(
      "default",
      operations
        .AchDepositSchedulesUpdateAchDepositScheduleResponse$inboundSchema,
      { key: "Status" },
    ),
  )(response, req, { extraFields: responseFields });
  if (!result.ok) {
    return result;
  }

  return result;
}
