// @apexfintechsolutions/ascend-sdk/src/models/components/accounttaxprofileupdate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";

/**
 * A method of determining the cost basis of an asset that has been sold or disposed of, by identifying which specific lot of the asset was sold and using the cost of that lot to calculate the cost basis; this method is commonly used for tax purposes to determine the amount of reportable capital gains or losses By default, this is set to `COST_BASIS_LOT_DISPOSAL_MIN_TAX_TERM`
 */
export enum AccountTaxProfileUpdateCostBasisLotDisposalMethod {
  CostBasisLotDisposalMethodUnspecified =
    "COST_BASIS_LOT_DISPOSAL_METHOD_UNSPECIFIED",
  CostBasisLotDisposalFifo = "COST_BASIS_LOT_DISPOSAL_FIFO",
  CostBasisLotDisposalLifo = "COST_BASIS_LOT_DISPOSAL_LIFO",
  CostBasisLotDisposalHighCost = "COST_BASIS_LOT_DISPOSAL_HIGH_COST",
  CostBasisLotDisposalHighCostLongTerm =
    "COST_BASIS_LOT_DISPOSAL_HIGH_COST_LONG_TERM",
  CostBasisLotDisposalHighCostShortTerm =
    "COST_BASIS_LOT_DISPOSAL_HIGH_COST_SHORT_TERM",
  CostBasisLotDisposalLowCost = "COST_BASIS_LOT_DISPOSAL_LOW_COST",
  CostBasisLotDisposalLowCostLongTerm =
    "COST_BASIS_LOT_DISPOSAL_LOW_COST_LONG_TERM",
  CostBasisLotDisposalLowCostShortTerm =
    "COST_BASIS_LOT_DISPOSAL_LOW_COST_SHORT_TERM",
  CostBasisLotDisposalMinTaxTerm = "COST_BASIS_LOT_DISPOSAL_MIN_TAX_TERM",
}
/**
 * A method of determining the cost basis of an asset that has been sold or disposed of, by identifying which specific lot of the asset was sold and using the cost of that lot to calculate the cost basis; this method is commonly used for tax purposes to determine the amount of reportable capital gains or losses By default, this is set to `COST_BASIS_LOT_DISPOSAL_MIN_TAX_TERM`
 */
export type AccountTaxProfileUpdateCostBasisLotDisposalMethodOpen = OpenEnum<
  typeof AccountTaxProfileUpdateCostBasisLotDisposalMethod
>;

/**
 * The account tax profile.
 */
export type AccountTaxProfileUpdate = {
  /**
   * A method of determining the cost basis of an asset that has been sold or disposed of, by identifying which specific lot of the asset was sold and using the cost of that lot to calculate the cost basis; this method is commonly used for tax purposes to determine the amount of reportable capital gains or losses By default, this is set to `COST_BASIS_LOT_DISPOSAL_MIN_TAX_TERM`
   */
  costBasisLotDisposalMethod?:
    | AccountTaxProfileUpdateCostBasisLotDisposalMethodOpen
    | undefined;
  /**
   * Indicates if the account is eligible to mark-to-market their securities and commodities holdings; Named after the related section of the IRS tax code
   */
  section475Election?: boolean | undefined;
};

/** @internal */
export const AccountTaxProfileUpdateCostBasisLotDisposalMethod$inboundSchema:
  z.ZodType<
    AccountTaxProfileUpdateCostBasisLotDisposalMethodOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(AccountTaxProfileUpdateCostBasisLotDisposalMethod),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const AccountTaxProfileUpdateCostBasisLotDisposalMethod$outboundSchema:
  z.ZodType<
    AccountTaxProfileUpdateCostBasisLotDisposalMethodOpen,
    z.ZodTypeDef,
    AccountTaxProfileUpdateCostBasisLotDisposalMethodOpen
  > = z.union([
    z.nativeEnum(AccountTaxProfileUpdateCostBasisLotDisposalMethod),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AccountTaxProfileUpdateCostBasisLotDisposalMethod$ {
  /** @deprecated use `AccountTaxProfileUpdateCostBasisLotDisposalMethod$inboundSchema` instead. */
  export const inboundSchema =
    AccountTaxProfileUpdateCostBasisLotDisposalMethod$inboundSchema;
  /** @deprecated use `AccountTaxProfileUpdateCostBasisLotDisposalMethod$outboundSchema` instead. */
  export const outboundSchema =
    AccountTaxProfileUpdateCostBasisLotDisposalMethod$outboundSchema;
}

/** @internal */
export const AccountTaxProfileUpdate$inboundSchema: z.ZodType<
  AccountTaxProfileUpdate,
  z.ZodTypeDef,
  unknown
> = z.object({
  cost_basis_lot_disposal_method:
    AccountTaxProfileUpdateCostBasisLotDisposalMethod$inboundSchema.optional(),
  section_475_election: z.boolean().optional(),
}).transform((v) => {
  return remap$(v, {
    "cost_basis_lot_disposal_method": "costBasisLotDisposalMethod",
    "section_475_election": "section475Election",
  });
});

/** @internal */
export type AccountTaxProfileUpdate$Outbound = {
  cost_basis_lot_disposal_method?: string | undefined;
  section_475_election?: boolean | undefined;
};

/** @internal */
export const AccountTaxProfileUpdate$outboundSchema: z.ZodType<
  AccountTaxProfileUpdate$Outbound,
  z.ZodTypeDef,
  AccountTaxProfileUpdate
> = z.object({
  costBasisLotDisposalMethod:
    AccountTaxProfileUpdateCostBasisLotDisposalMethod$outboundSchema.optional(),
  section475Election: z.boolean().optional(),
}).transform((v) => {
  return remap$(v, {
    costBasisLotDisposalMethod: "cost_basis_lot_disposal_method",
    section475Election: "section_475_election",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AccountTaxProfileUpdate$ {
  /** @deprecated use `AccountTaxProfileUpdate$inboundSchema` instead. */
  export const inboundSchema = AccountTaxProfileUpdate$inboundSchema;
  /** @deprecated use `AccountTaxProfileUpdate$outboundSchema` instead. */
  export const outboundSchema = AccountTaxProfileUpdate$outboundSchema;
  /** @deprecated use `AccountTaxProfileUpdate$Outbound` instead. */
  export type Outbound = AccountTaxProfileUpdate$Outbound;
}
