// @apexfintechsolutions/ascend-sdk/src/models/components/assignlargetraderrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";

/**
 * The request to assign a Large Trader ID to a Legal Natural Person/Legal Entity.
 */
export type AssignLargeTraderRequestCreate = {
  largeTraderId: string;
};

/** @internal */
export const AssignLargeTraderRequestCreate$inboundSchema: z.ZodType<
  AssignLargeTraderRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  large_trader_id: z.string(),
}).transform((v) => {
  return remap$(v, {
    "large_trader_id": "largeTraderId",
  });
});

/** @internal */
export type AssignLargeTraderRequestCreate$Outbound = {
  large_trader_id: string;
};

/** @internal */
export const AssignLargeTraderRequestCreate$outboundSchema: z.ZodType<
  AssignLargeTraderRequestCreate$Outbound,
  z.ZodTypeDef,
  AssignLargeTraderRequestCreate
> = z.object({
  largeTraderId: z.string(),
}).transform((v) => {
  return remap$(v, {
    largeTraderId: "large_trader_id",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AssignLargeTraderRequestCreate$ {
  /** @deprecated use `AssignLargeTraderRequestCreate$inboundSchema` instead. */
  export const inboundSchema = AssignLargeTraderRequestCreate$inboundSchema;
  /** @deprecated use `AssignLargeTraderRequestCreate$outboundSchema` instead. */
  export const outboundSchema = AssignLargeTraderRequestCreate$outboundSchema;
  /** @deprecated use `AssignLargeTraderRequestCreate$Outbound` instead. */
  export type Outbound = AssignLargeTraderRequestCreate$Outbound;
}
