// @apexfintechsolutions/ascend-sdk/src/models/components/basketorder.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";
import {
  ExecutedPrice,
  ExecutedPrice$inboundSchema,
  ExecutedPrice$Outbound,
  ExecutedPrice$outboundSchema,
} from "./executedprice.js";

/**
 * The type of the asset in this order
 */
export enum BasketOrderAssetType {
  AssetTypeUnspecified = "ASSET_TYPE_UNSPECIFIED",
  Equity = "EQUITY",
  MutualFund = "MUTUAL_FUND",
}
/**
 * The type of the asset in this order
 */
export type BasketOrderAssetTypeOpen = OpenEnum<typeof BasketOrderAssetType>;

/**
 * The product of order quantity & price, summed across all fills, reported in the currency specified in the order. (This will be rounded to 2 decimal places for USD currencies). Will be absent if an order has no fill information.
 */
export type BasketOrderCumulativeNotionalValue = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/**
 * The summed quantity value across all fills in this order, up to a maximum of 5 decimal places. Will be absent if an order has no fill information.
 */
export type BasketOrderFilledQuantity = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/**
 * The identifier type of the asset being ordered. For Equities: only SYMBOL is supported For Mutual Funds: only SYMBOL and CUSIP are supported
 */
export enum BasketOrderIdentifierType {
  Symbol = "SYMBOL",
  Cusip = "CUSIP",
  Isin = "ISIN",
}
/**
 * The identifier type of the asset being ordered. For Equities: only SYMBOL is supported For Mutual Funds: only SYMBOL and CUSIP are supported
 */
export type BasketOrderIdentifierTypeOpen = OpenEnum<
  typeof BasketOrderIdentifierType
>;

/**
 * Notional quantity of the order, measured in USD. Maximum 2 decimal place precision. Either a quantity or notional_value MUST be specified (but not both). For Equities: currently not supported yet For Mutual Funds: Only supported for BUY orders. The order will be transacted at the full notional amount specified.
 */
export type BasketOrderNotionalValue = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/**
 * When an order has the REJECTED status, this will be populated with a system code describing the rejection.
 */
export enum BasketOrderOrderRejectedReason {
  OrderRejectReasonUnspecified = "ORDER_REJECT_REASON_UNSPECIFIED",
  BrokerOption = "BROKER_OPTION",
  UnknownSecurity = "UNKNOWN_SECURITY",
  ExchangeClosed = "EXCHANGE_CLOSED",
  TooLateToEnter = "TOO_LATE_TO_ENTER",
  UnknownOrder = "UNKNOWN_ORDER",
  DuplicateOrder = "DUPLICATE_ORDER",
  StaleOrder = "STALE_ORDER",
  BelowNotionalMinimum = "BELOW_NOTIONAL_MINIMUM",
  AccountNotEntitled = "ACCOUNT_NOT_ENTITLED",
  SystemError = "SYSTEM_ERROR",
  BlockingCorporateAction = "BLOCKING_CORPORATE_ACTION",
  UnavailablePriceQuote = "UNAVAILABLE_PRICE_QUOTE",
  ExecutionMisconfiguredClient = "EXECUTION_MISCONFIGURED_CLIENT",
  FractionalQuantityNotAllowedForSecurity =
    "FRACTIONAL_QUANTITY_NOT_ALLOWED_FOR_SECURITY",
  OnlyFractionalSellOrWholeOrdersAllowedForSecurity =
    "ONLY_FRACTIONAL_SELL_OR_WHOLE_ORDERS_ALLOWED_FOR_SECURITY",
  SymbolNotTradeable = "SYMBOL_NOT_TRADEABLE",
  AboveNotionalMaximum = "ABOVE_NOTIONAL_MAXIMUM",
  AboveShareMaximum = "ABOVE_SHARE_MAXIMUM",
  MaxSellQuantityRequired = "MAX_SELL_QUANTITY_REQUIRED",
  MaxSellQuantityProhibited = "MAX_SELL_QUANTITY_PROHIBITED",
  StockTradesDisabled = "STOCK_TRADES_DISABLED",
  AssetNotSetUpToTrade = "ASSET_NOT_SET_UP_TO_TRADE",
  AnotherBasketOrderForAccountHasFailedRiskChecks =
    "ANOTHER_BASKET_ORDER_FOR_ACCOUNT_HAS_FAILED_RISK_CHECKS",
  InsufficientPosition = "INSUFFICIENT_POSITION",
  FailedBuyingPower = "FAILED_BUYING_POWER",
}
/**
 * When an order has the REJECTED status, this will be populated with a system code describing the rejection.
 */
export type BasketOrderOrderRejectedReasonOpen = OpenEnum<
  typeof BasketOrderOrderRejectedReason
>;

/**
 * The processing status of the order
 */
export enum BasketOrderOrderStatus {
  OrderStatusUnspecified = "ORDER_STATUS_UNSPECIFIED",
  PendingNew = "PENDING_NEW",
  New = "NEW",
  PartiallyFilled = "PARTIALLY_FILLED",
  Filled = "FILLED",
  Rejected = "REJECTED",
}
/**
 * The processing status of the order
 */
export type BasketOrderOrderStatusOpen = OpenEnum<
  typeof BasketOrderOrderStatus
>;

/**
 * The execution type of this order.
 */
export enum BasketOrderOrderType {
  OrderTypeUnspecified = "ORDER_TYPE_UNSPECIFIED",
  Market = "MARKET",
}
/**
 * The execution type of this order.
 */
export type BasketOrderOrderTypeOpen = OpenEnum<typeof BasketOrderOrderType>;

/**
 * Numeric quantity of the order. Either a quantity or notional_value MUST be specified (but not both). For Equities: Represents the number of shares, must be greater than zero and may not exceed 5 decimal places. For Mutual Funds: Only supported for SELL orders. Represents the number of shares, up to a maximum of 3 decimal places.
 */
export type BasketOrderQuantity = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/**
 * The side of this order.
 */
export enum BasketOrderSide {
  SideUnspecified = "SIDE_UNSPECIFIED",
  Buy = "BUY",
  Sell = "SELL",
}
/**
 * The side of this order.
 */
export type BasketOrderSideOpen = OpenEnum<typeof BasketOrderSide>;

/**
 * Must be the value "DAY". Regulatory requirements dictate that the system capture the intended time_in_force, which is why this a mandatory field.
 */
export enum BasketOrderTimeInForce {
  TimeInForceUnspecified = "TIME_IN_FORCE_UNSPECIFIED",
  Day = "DAY",
}
/**
 * Must be the value "DAY". Regulatory requirements dictate that the system capture the intended time_in_force, which is why this a mandatory field.
 */
export type BasketOrderTimeInForceOpen = OpenEnum<
  typeof BasketOrderTimeInForce
>;

/**
 * The message describing an order that has been added to a basket
 */
export type BasketOrder = {
  /**
   * The identifier of the account transacting this order
   */
  accountId?: string | undefined;
  /**
   * Apex Asset ID for this asset. This will not be returned in the initial CreateOrder response and will be available after an order completes validation. If the provided identifier does not match any Apex asset available for trading, an OrderRejectReason of "UNKNOWN_SECURITY" will be returned and the asset_id will not be set.
   */
  assetId?: string | undefined;
  /**
   * The type of the asset in this order
   */
  assetType?: BasketOrderAssetTypeOpen | undefined;
  /**
   * The average prices, as weighted averages, across all executions in this order. Will be absent if an order has no executions.
   *
   * @remarks
   *
   *  When asset_type = EQUITY, there will be at most one value present, with a type of PRICE_PER_UNIT. This will have up to 4 decimal places for USD amounts less than $1, and a maximum of two for larger USD amounts.
   */
  averagePrices?: Array<ExecutedPrice> | undefined;
  /**
   * System generated unique id for the basket order.
   */
  basketOrderId?: string | undefined;
  /**
   * User-supplied unique order ID. Cannot be more than 40 characters long.
   */
  clientOrderId?: string | undefined;
  /**
   * Time the order request was received by the client. Must be in the past, and must be less than 24 hours old.
   */
  clientOrderReceivedTime?: Date | null | undefined;
  /**
   * Time of the order creation
   */
  createTime?: Date | null | undefined;
  /**
   * The product of order quantity & price, summed across all fills, reported in the currency specified in the order. (This will be rounded to 2 decimal places for USD currencies). Will be absent if an order has no fill information.
   */
  cumulativeNotionalValue?:
    | BasketOrderCumulativeNotionalValue
    | null
    | undefined;
  /**
   * Defaults to "USD". Only "USD" is supported. Full list of currency codes is defined at: https://en.wikipedia.org/wiki/ISO_4217
   */
  currencyCode?: string | undefined;
  /**
   * The summed quantity value across all fills in this order, up to a maximum of 5 decimal places. Will be absent if an order has no fill information.
   */
  filledQuantity?: BasketOrderFilledQuantity | null | undefined;
  /**
   * Identifier of the asset (of the type specified in `identifier_type`).
   */
  identifier?: string | undefined;
  /**
   * The identifier type of the asset being ordered. For Equities: only SYMBOL is supported For Mutual Funds: only SYMBOL and CUSIP are supported
   */
  identifierType?: BasketOrderIdentifierTypeOpen | undefined;
  /**
   * Time of the last order update
   */
  lastUpdateTime?: Date | null | undefined;
  /**
   * System generated name of the basket order.
   */
  name?: string | undefined;
  /**
   * Notional quantity of the order, measured in USD. Maximum 2 decimal place precision. Either a quantity or notional_value MUST be specified (but not both). For Equities: currently not supported yet For Mutual Funds: Only supported for BUY orders. The order will be transacted at the full notional amount specified.
   */
  notionalValue?: BasketOrderNotionalValue | null | undefined;
  /**
   * When an order has the REJECTED status, this will be populated with a system code describing the rejection.
   */
  orderRejectedReason?: BasketOrderOrderRejectedReasonOpen | undefined;
  /**
   * The processing status of the order
   */
  orderStatus?: BasketOrderOrderStatusOpen | undefined;
  /**
   * The execution type of this order.
   */
  orderType?: BasketOrderOrderTypeOpen | undefined;
  /**
   * Numeric quantity of the order. Either a quantity or notional_value MUST be specified (but not both). For Equities: Represents the number of shares, must be greater than zero and may not exceed 5 decimal places. For Mutual Funds: Only supported for SELL orders. Represents the number of shares, up to a maximum of 3 decimal places.
   */
  quantity?: BasketOrderQuantity | null | undefined;
  /**
   * The side of this order.
   */
  side?: BasketOrderSideOpen | undefined;
  /**
   * Must be the value "DAY". Regulatory requirements dictate that the system capture the intended time_in_force, which is why this a mandatory field.
   */
  timeInForce?: BasketOrderTimeInForceOpen | undefined;
};

/** @internal */
export const BasketOrderAssetType$inboundSchema: z.ZodType<
  BasketOrderAssetTypeOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(BasketOrderAssetType),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const BasketOrderAssetType$outboundSchema: z.ZodType<
  BasketOrderAssetTypeOpen,
  z.ZodTypeDef,
  BasketOrderAssetTypeOpen
> = z.union([
  z.nativeEnum(BasketOrderAssetType),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrderAssetType$ {
  /** @deprecated use `BasketOrderAssetType$inboundSchema` instead. */
  export const inboundSchema = BasketOrderAssetType$inboundSchema;
  /** @deprecated use `BasketOrderAssetType$outboundSchema` instead. */
  export const outboundSchema = BasketOrderAssetType$outboundSchema;
}

/** @internal */
export const BasketOrderCumulativeNotionalValue$inboundSchema: z.ZodType<
  BasketOrderCumulativeNotionalValue,
  z.ZodTypeDef,
  unknown
> = z.object({
  value: z.string().optional(),
});

/** @internal */
export type BasketOrderCumulativeNotionalValue$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const BasketOrderCumulativeNotionalValue$outboundSchema: z.ZodType<
  BasketOrderCumulativeNotionalValue$Outbound,
  z.ZodTypeDef,
  BasketOrderCumulativeNotionalValue
> = z.object({
  value: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrderCumulativeNotionalValue$ {
  /** @deprecated use `BasketOrderCumulativeNotionalValue$inboundSchema` instead. */
  export const inboundSchema = BasketOrderCumulativeNotionalValue$inboundSchema;
  /** @deprecated use `BasketOrderCumulativeNotionalValue$outboundSchema` instead. */
  export const outboundSchema =
    BasketOrderCumulativeNotionalValue$outboundSchema;
  /** @deprecated use `BasketOrderCumulativeNotionalValue$Outbound` instead. */
  export type Outbound = BasketOrderCumulativeNotionalValue$Outbound;
}

/** @internal */
export const BasketOrderFilledQuantity$inboundSchema: z.ZodType<
  BasketOrderFilledQuantity,
  z.ZodTypeDef,
  unknown
> = z.object({
  value: z.string().optional(),
});

/** @internal */
export type BasketOrderFilledQuantity$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const BasketOrderFilledQuantity$outboundSchema: z.ZodType<
  BasketOrderFilledQuantity$Outbound,
  z.ZodTypeDef,
  BasketOrderFilledQuantity
> = z.object({
  value: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrderFilledQuantity$ {
  /** @deprecated use `BasketOrderFilledQuantity$inboundSchema` instead. */
  export const inboundSchema = BasketOrderFilledQuantity$inboundSchema;
  /** @deprecated use `BasketOrderFilledQuantity$outboundSchema` instead. */
  export const outboundSchema = BasketOrderFilledQuantity$outboundSchema;
  /** @deprecated use `BasketOrderFilledQuantity$Outbound` instead. */
  export type Outbound = BasketOrderFilledQuantity$Outbound;
}

/** @internal */
export const BasketOrderIdentifierType$inboundSchema: z.ZodType<
  BasketOrderIdentifierTypeOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(BasketOrderIdentifierType),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const BasketOrderIdentifierType$outboundSchema: z.ZodType<
  BasketOrderIdentifierTypeOpen,
  z.ZodTypeDef,
  BasketOrderIdentifierTypeOpen
> = z.union([
  z.nativeEnum(BasketOrderIdentifierType),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrderIdentifierType$ {
  /** @deprecated use `BasketOrderIdentifierType$inboundSchema` instead. */
  export const inboundSchema = BasketOrderIdentifierType$inboundSchema;
  /** @deprecated use `BasketOrderIdentifierType$outboundSchema` instead. */
  export const outboundSchema = BasketOrderIdentifierType$outboundSchema;
}

/** @internal */
export const BasketOrderNotionalValue$inboundSchema: z.ZodType<
  BasketOrderNotionalValue,
  z.ZodTypeDef,
  unknown
> = z.object({
  value: z.string().optional(),
});

/** @internal */
export type BasketOrderNotionalValue$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const BasketOrderNotionalValue$outboundSchema: z.ZodType<
  BasketOrderNotionalValue$Outbound,
  z.ZodTypeDef,
  BasketOrderNotionalValue
> = z.object({
  value: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrderNotionalValue$ {
  /** @deprecated use `BasketOrderNotionalValue$inboundSchema` instead. */
  export const inboundSchema = BasketOrderNotionalValue$inboundSchema;
  /** @deprecated use `BasketOrderNotionalValue$outboundSchema` instead. */
  export const outboundSchema = BasketOrderNotionalValue$outboundSchema;
  /** @deprecated use `BasketOrderNotionalValue$Outbound` instead. */
  export type Outbound = BasketOrderNotionalValue$Outbound;
}

/** @internal */
export const BasketOrderOrderRejectedReason$inboundSchema: z.ZodType<
  BasketOrderOrderRejectedReasonOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(BasketOrderOrderRejectedReason),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const BasketOrderOrderRejectedReason$outboundSchema: z.ZodType<
  BasketOrderOrderRejectedReasonOpen,
  z.ZodTypeDef,
  BasketOrderOrderRejectedReasonOpen
> = z.union([
  z.nativeEnum(BasketOrderOrderRejectedReason),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrderOrderRejectedReason$ {
  /** @deprecated use `BasketOrderOrderRejectedReason$inboundSchema` instead. */
  export const inboundSchema = BasketOrderOrderRejectedReason$inboundSchema;
  /** @deprecated use `BasketOrderOrderRejectedReason$outboundSchema` instead. */
  export const outboundSchema = BasketOrderOrderRejectedReason$outboundSchema;
}

/** @internal */
export const BasketOrderOrderStatus$inboundSchema: z.ZodType<
  BasketOrderOrderStatusOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(BasketOrderOrderStatus),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const BasketOrderOrderStatus$outboundSchema: z.ZodType<
  BasketOrderOrderStatusOpen,
  z.ZodTypeDef,
  BasketOrderOrderStatusOpen
> = z.union([
  z.nativeEnum(BasketOrderOrderStatus),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrderOrderStatus$ {
  /** @deprecated use `BasketOrderOrderStatus$inboundSchema` instead. */
  export const inboundSchema = BasketOrderOrderStatus$inboundSchema;
  /** @deprecated use `BasketOrderOrderStatus$outboundSchema` instead. */
  export const outboundSchema = BasketOrderOrderStatus$outboundSchema;
}

/** @internal */
export const BasketOrderOrderType$inboundSchema: z.ZodType<
  BasketOrderOrderTypeOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(BasketOrderOrderType),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const BasketOrderOrderType$outboundSchema: z.ZodType<
  BasketOrderOrderTypeOpen,
  z.ZodTypeDef,
  BasketOrderOrderTypeOpen
> = z.union([
  z.nativeEnum(BasketOrderOrderType),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrderOrderType$ {
  /** @deprecated use `BasketOrderOrderType$inboundSchema` instead. */
  export const inboundSchema = BasketOrderOrderType$inboundSchema;
  /** @deprecated use `BasketOrderOrderType$outboundSchema` instead. */
  export const outboundSchema = BasketOrderOrderType$outboundSchema;
}

/** @internal */
export const BasketOrderQuantity$inboundSchema: z.ZodType<
  BasketOrderQuantity,
  z.ZodTypeDef,
  unknown
> = z.object({
  value: z.string().optional(),
});

/** @internal */
export type BasketOrderQuantity$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const BasketOrderQuantity$outboundSchema: z.ZodType<
  BasketOrderQuantity$Outbound,
  z.ZodTypeDef,
  BasketOrderQuantity
> = z.object({
  value: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrderQuantity$ {
  /** @deprecated use `BasketOrderQuantity$inboundSchema` instead. */
  export const inboundSchema = BasketOrderQuantity$inboundSchema;
  /** @deprecated use `BasketOrderQuantity$outboundSchema` instead. */
  export const outboundSchema = BasketOrderQuantity$outboundSchema;
  /** @deprecated use `BasketOrderQuantity$Outbound` instead. */
  export type Outbound = BasketOrderQuantity$Outbound;
}

/** @internal */
export const BasketOrderSide$inboundSchema: z.ZodType<
  BasketOrderSideOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(BasketOrderSide),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const BasketOrderSide$outboundSchema: z.ZodType<
  BasketOrderSideOpen,
  z.ZodTypeDef,
  BasketOrderSideOpen
> = z.union([
  z.nativeEnum(BasketOrderSide),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrderSide$ {
  /** @deprecated use `BasketOrderSide$inboundSchema` instead. */
  export const inboundSchema = BasketOrderSide$inboundSchema;
  /** @deprecated use `BasketOrderSide$outboundSchema` instead. */
  export const outboundSchema = BasketOrderSide$outboundSchema;
}

/** @internal */
export const BasketOrderTimeInForce$inboundSchema: z.ZodType<
  BasketOrderTimeInForceOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(BasketOrderTimeInForce),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const BasketOrderTimeInForce$outboundSchema: z.ZodType<
  BasketOrderTimeInForceOpen,
  z.ZodTypeDef,
  BasketOrderTimeInForceOpen
> = z.union([
  z.nativeEnum(BasketOrderTimeInForce),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrderTimeInForce$ {
  /** @deprecated use `BasketOrderTimeInForce$inboundSchema` instead. */
  export const inboundSchema = BasketOrderTimeInForce$inboundSchema;
  /** @deprecated use `BasketOrderTimeInForce$outboundSchema` instead. */
  export const outboundSchema = BasketOrderTimeInForce$outboundSchema;
}

/** @internal */
export const BasketOrder$inboundSchema: z.ZodType<
  BasketOrder,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string().optional(),
  asset_id: z.string().optional(),
  asset_type: BasketOrderAssetType$inboundSchema.optional(),
  average_prices: z.array(ExecutedPrice$inboundSchema).optional(),
  basket_order_id: z.string().optional(),
  client_order_id: z.string().optional(),
  client_order_received_time: z.nullable(
    z.string().datetime({ offset: true }).transform(v => new Date(v)),
  ).optional(),
  create_time: z.nullable(
    z.string().datetime({ offset: true }).transform(v => new Date(v)),
  ).optional(),
  cumulative_notional_value: z.nullable(
    z.lazy(() => BasketOrderCumulativeNotionalValue$inboundSchema),
  ).optional(),
  currency_code: z.string().optional(),
  filled_quantity: z.nullable(
    z.lazy(() => BasketOrderFilledQuantity$inboundSchema),
  ).optional(),
  identifier: z.string().optional(),
  identifier_type: BasketOrderIdentifierType$inboundSchema.optional(),
  last_update_time: z.nullable(
    z.string().datetime({ offset: true }).transform(v => new Date(v)),
  ).optional(),
  name: z.string().optional(),
  notional_value: z.nullable(
    z.lazy(() => BasketOrderNotionalValue$inboundSchema),
  ).optional(),
  order_rejected_reason: BasketOrderOrderRejectedReason$inboundSchema
    .optional(),
  order_status: BasketOrderOrderStatus$inboundSchema.optional(),
  order_type: BasketOrderOrderType$inboundSchema.optional(),
  quantity: z.nullable(z.lazy(() => BasketOrderQuantity$inboundSchema))
    .optional(),
  side: BasketOrderSide$inboundSchema.optional(),
  time_in_force: BasketOrderTimeInForce$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "asset_id": "assetId",
    "asset_type": "assetType",
    "average_prices": "averagePrices",
    "basket_order_id": "basketOrderId",
    "client_order_id": "clientOrderId",
    "client_order_received_time": "clientOrderReceivedTime",
    "create_time": "createTime",
    "cumulative_notional_value": "cumulativeNotionalValue",
    "currency_code": "currencyCode",
    "filled_quantity": "filledQuantity",
    "identifier_type": "identifierType",
    "last_update_time": "lastUpdateTime",
    "notional_value": "notionalValue",
    "order_rejected_reason": "orderRejectedReason",
    "order_status": "orderStatus",
    "order_type": "orderType",
    "time_in_force": "timeInForce",
  });
});

/** @internal */
export type BasketOrder$Outbound = {
  account_id?: string | undefined;
  asset_id?: string | undefined;
  asset_type?: string | undefined;
  average_prices?: Array<ExecutedPrice$Outbound> | undefined;
  basket_order_id?: string | undefined;
  client_order_id?: string | undefined;
  client_order_received_time?: string | null | undefined;
  create_time?: string | null | undefined;
  cumulative_notional_value?:
    | BasketOrderCumulativeNotionalValue$Outbound
    | null
    | undefined;
  currency_code?: string | undefined;
  filled_quantity?: BasketOrderFilledQuantity$Outbound | null | undefined;
  identifier?: string | undefined;
  identifier_type?: string | undefined;
  last_update_time?: string | null | undefined;
  name?: string | undefined;
  notional_value?: BasketOrderNotionalValue$Outbound | null | undefined;
  order_rejected_reason?: string | undefined;
  order_status?: string | undefined;
  order_type?: string | undefined;
  quantity?: BasketOrderQuantity$Outbound | null | undefined;
  side?: string | undefined;
  time_in_force?: string | undefined;
};

/** @internal */
export const BasketOrder$outboundSchema: z.ZodType<
  BasketOrder$Outbound,
  z.ZodTypeDef,
  BasketOrder
> = z.object({
  accountId: z.string().optional(),
  assetId: z.string().optional(),
  assetType: BasketOrderAssetType$outboundSchema.optional(),
  averagePrices: z.array(ExecutedPrice$outboundSchema).optional(),
  basketOrderId: z.string().optional(),
  clientOrderId: z.string().optional(),
  clientOrderReceivedTime: z.nullable(z.date().transform(v => v.toISOString()))
    .optional(),
  createTime: z.nullable(z.date().transform(v => v.toISOString())).optional(),
  cumulativeNotionalValue: z.nullable(
    z.lazy(() => BasketOrderCumulativeNotionalValue$outboundSchema),
  ).optional(),
  currencyCode: z.string().optional(),
  filledQuantity: z.nullable(
    z.lazy(() => BasketOrderFilledQuantity$outboundSchema),
  ).optional(),
  identifier: z.string().optional(),
  identifierType: BasketOrderIdentifierType$outboundSchema.optional(),
  lastUpdateTime: z.nullable(z.date().transform(v => v.toISOString()))
    .optional(),
  name: z.string().optional(),
  notionalValue: z.nullable(
    z.lazy(() => BasketOrderNotionalValue$outboundSchema),
  ).optional(),
  orderRejectedReason: BasketOrderOrderRejectedReason$outboundSchema.optional(),
  orderStatus: BasketOrderOrderStatus$outboundSchema.optional(),
  orderType: BasketOrderOrderType$outboundSchema.optional(),
  quantity: z.nullable(z.lazy(() => BasketOrderQuantity$outboundSchema))
    .optional(),
  side: BasketOrderSide$outboundSchema.optional(),
  timeInForce: BasketOrderTimeInForce$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    assetId: "asset_id",
    assetType: "asset_type",
    averagePrices: "average_prices",
    basketOrderId: "basket_order_id",
    clientOrderId: "client_order_id",
    clientOrderReceivedTime: "client_order_received_time",
    createTime: "create_time",
    cumulativeNotionalValue: "cumulative_notional_value",
    currencyCode: "currency_code",
    filledQuantity: "filled_quantity",
    identifierType: "identifier_type",
    lastUpdateTime: "last_update_time",
    notionalValue: "notional_value",
    orderRejectedReason: "order_rejected_reason",
    orderStatus: "order_status",
    orderType: "order_type",
    timeInForce: "time_in_force",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrder$ {
  /** @deprecated use `BasketOrder$inboundSchema` instead. */
  export const inboundSchema = BasketOrder$inboundSchema;
  /** @deprecated use `BasketOrder$outboundSchema` instead. */
  export const outboundSchema = BasketOrder$outboundSchema;
  /** @deprecated use `BasketOrder$Outbound` instead. */
  export type Outbound = BasketOrder$Outbound;
}
