// @apexfintechsolutions/ascend-sdk/src/models/components/batchcreateuploadlinksrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  CreateUploadLinkRequestCreate,
  CreateUploadLinkRequestCreate$inboundSchema,
  CreateUploadLinkRequestCreate$Outbound,
  CreateUploadLinkRequestCreate$outboundSchema,
} from "./createuploadlinkrequestcreate.js";

/**
 * List of documents to get upload signed links for.
 */
export type BatchCreateUploadLinksRequestCreate = {
  /**
   * List of documents to get upload signed links for
   */
  createDocumentUploadLinkRequest?:
    | Array<CreateUploadLinkRequestCreate>
    | undefined;
};

/** @internal */
export const BatchCreateUploadLinksRequestCreate$inboundSchema: z.ZodType<
  BatchCreateUploadLinksRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  create_document_upload_link_request: z.array(
    CreateUploadLinkRequestCreate$inboundSchema,
  ).optional(),
}).transform((v) => {
  return remap$(v, {
    "create_document_upload_link_request": "createDocumentUploadLinkRequest",
  });
});

/** @internal */
export type BatchCreateUploadLinksRequestCreate$Outbound = {
  create_document_upload_link_request?:
    | Array<CreateUploadLinkRequestCreate$Outbound>
    | undefined;
};

/** @internal */
export const BatchCreateUploadLinksRequestCreate$outboundSchema: z.ZodType<
  BatchCreateUploadLinksRequestCreate$Outbound,
  z.ZodTypeDef,
  BatchCreateUploadLinksRequestCreate
> = z.object({
  createDocumentUploadLinkRequest: z.array(
    CreateUploadLinkRequestCreate$outboundSchema,
  ).optional(),
}).transform((v) => {
  return remap$(v, {
    createDocumentUploadLinkRequest: "create_document_upload_link_request",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BatchCreateUploadLinksRequestCreate$ {
  /** @deprecated use `BatchCreateUploadLinksRequestCreate$inboundSchema` instead. */
  export const inboundSchema =
    BatchCreateUploadLinksRequestCreate$inboundSchema;
  /** @deprecated use `BatchCreateUploadLinksRequestCreate$outboundSchema` instead. */
  export const outboundSchema =
    BatchCreateUploadLinksRequestCreate$outboundSchema;
  /** @deprecated use `BatchCreateUploadLinksRequestCreate$Outbound` instead. */
  export type Outbound = BatchCreateUploadLinksRequestCreate$Outbound;
}
