// @apexfintechsolutions/ascend-sdk/src/models/components/beneficiaryenrollmentmetadatacreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  BeneficiaryCreate,
  BeneficiaryCreate$inboundSchema,
  BeneficiaryCreate$Outbound,
  BeneficiaryCreate$outboundSchema,
} from "./beneficiarycreate.js";

/**
 * Enrollment metadata for the BENEFICIARY_DESIGNATION enrollment type.
 */
export type BeneficiaryEnrollmentMetadataCreate = {
  /**
   * Contingent Beneficiary list is optional, with a maximum of five contingent beneficiaries.
   */
  contingentBeneficiaries?: Array<BeneficiaryCreate> | undefined;
  /**
   * At least one primary beneficiary must be provided, with a maximum of five primary beneficiaries.
   */
  primaryBeneficiaries: Array<BeneficiaryCreate>;
};

/** @internal */
export const BeneficiaryEnrollmentMetadataCreate$inboundSchema: z.ZodType<
  BeneficiaryEnrollmentMetadataCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  contingent_beneficiaries: z.array(BeneficiaryCreate$inboundSchema).optional(),
  primary_beneficiaries: z.array(BeneficiaryCreate$inboundSchema),
}).transform((v) => {
  return remap$(v, {
    "contingent_beneficiaries": "contingentBeneficiaries",
    "primary_beneficiaries": "primaryBeneficiaries",
  });
});

/** @internal */
export type BeneficiaryEnrollmentMetadataCreate$Outbound = {
  contingent_beneficiaries?: Array<BeneficiaryCreate$Outbound> | undefined;
  primary_beneficiaries: Array<BeneficiaryCreate$Outbound>;
};

/** @internal */
export const BeneficiaryEnrollmentMetadataCreate$outboundSchema: z.ZodType<
  BeneficiaryEnrollmentMetadataCreate$Outbound,
  z.ZodTypeDef,
  BeneficiaryEnrollmentMetadataCreate
> = z.object({
  contingentBeneficiaries: z.array(BeneficiaryCreate$outboundSchema).optional(),
  primaryBeneficiaries: z.array(BeneficiaryCreate$outboundSchema),
}).transform((v) => {
  return remap$(v, {
    contingentBeneficiaries: "contingent_beneficiaries",
    primaryBeneficiaries: "primary_beneficiaries",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BeneficiaryEnrollmentMetadataCreate$ {
  /** @deprecated use `BeneficiaryEnrollmentMetadataCreate$inboundSchema` instead. */
  export const inboundSchema =
    BeneficiaryEnrollmentMetadataCreate$inboundSchema;
  /** @deprecated use `BeneficiaryEnrollmentMetadataCreate$outboundSchema` instead. */
  export const outboundSchema =
    BeneficiaryEnrollmentMetadataCreate$outboundSchema;
  /** @deprecated use `BeneficiaryEnrollmentMetadataCreate$Outbound` instead. */
  export type Outbound = BeneficiaryEnrollmentMetadataCreate$Outbound;
}
