// @apexfintechsolutions/ascend-sdk/src/models/components/calculatecashbalanceresponsetransfersummary.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";

/**
 * The amount of the transfer in USD. The value will be positive for deposits and negative for withdrawals.
 */
export type CalculateCashBalanceResponseTransferSummaryAmount = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/**
 * The mechanism of the transfer.
 */
export enum CalculateCashBalanceResponseTransferSummaryMechanism {
  MechanismUnspecified = "MECHANISM_UNSPECIFIED",
  Acat = "ACAT",
  Ach = "ACH",
  CashJournal = "CASH_JOURNAL",
  Check = "CHECK",
  Credit = "CREDIT",
  Fee = "FEE",
  Ict = "ICT",
  Paypal = "PAYPAL",
  Rtp = "RTP",
  Tpj = "TPJ",
  Wire = "WIRE",
  ExternalAch = "EXTERNAL_ACH",
}
/**
 * The mechanism of the transfer.
 */
export type CalculateCashBalanceResponseTransferSummaryMechanismOpen = OpenEnum<
  typeof CalculateCashBalanceResponseTransferSummaryMechanism
>;

/**
 * A summary of a transfer.
 */
export type CalculateCashBalanceResponseTransferSummary = {
  /**
   * The amount of the transfer in USD. The value will be positive for deposits and negative for withdrawals.
   */
  amount?: CalculateCashBalanceResponseTransferSummaryAmount | null | undefined;
  /**
   * The identifier of the transfer. The value is the last part of the transfer resource name.
   */
  id?: string | undefined;
  /**
   * The mechanism of the transfer.
   */
  mechanism?:
    | CalculateCashBalanceResponseTransferSummaryMechanismOpen
    | undefined;
};

/** @internal */
export const CalculateCashBalanceResponseTransferSummaryAmount$inboundSchema:
  z.ZodType<
    CalculateCashBalanceResponseTransferSummaryAmount,
    z.ZodTypeDef,
    unknown
  > = z.object({
    value: z.string().optional(),
  });

/** @internal */
export type CalculateCashBalanceResponseTransferSummaryAmount$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const CalculateCashBalanceResponseTransferSummaryAmount$outboundSchema:
  z.ZodType<
    CalculateCashBalanceResponseTransferSummaryAmount$Outbound,
    z.ZodTypeDef,
    CalculateCashBalanceResponseTransferSummaryAmount
  > = z.object({
    value: z.string().optional(),
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CalculateCashBalanceResponseTransferSummaryAmount$ {
  /** @deprecated use `CalculateCashBalanceResponseTransferSummaryAmount$inboundSchema` instead. */
  export const inboundSchema =
    CalculateCashBalanceResponseTransferSummaryAmount$inboundSchema;
  /** @deprecated use `CalculateCashBalanceResponseTransferSummaryAmount$outboundSchema` instead. */
  export const outboundSchema =
    CalculateCashBalanceResponseTransferSummaryAmount$outboundSchema;
  /** @deprecated use `CalculateCashBalanceResponseTransferSummaryAmount$Outbound` instead. */
  export type Outbound =
    CalculateCashBalanceResponseTransferSummaryAmount$Outbound;
}

/** @internal */
export const CalculateCashBalanceResponseTransferSummaryMechanism$inboundSchema:
  z.ZodType<
    CalculateCashBalanceResponseTransferSummaryMechanismOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(CalculateCashBalanceResponseTransferSummaryMechanism),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const CalculateCashBalanceResponseTransferSummaryMechanism$outboundSchema:
  z.ZodType<
    CalculateCashBalanceResponseTransferSummaryMechanismOpen,
    z.ZodTypeDef,
    CalculateCashBalanceResponseTransferSummaryMechanismOpen
  > = z.union([
    z.nativeEnum(CalculateCashBalanceResponseTransferSummaryMechanism),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CalculateCashBalanceResponseTransferSummaryMechanism$ {
  /** @deprecated use `CalculateCashBalanceResponseTransferSummaryMechanism$inboundSchema` instead. */
  export const inboundSchema =
    CalculateCashBalanceResponseTransferSummaryMechanism$inboundSchema;
  /** @deprecated use `CalculateCashBalanceResponseTransferSummaryMechanism$outboundSchema` instead. */
  export const outboundSchema =
    CalculateCashBalanceResponseTransferSummaryMechanism$outboundSchema;
}

/** @internal */
export const CalculateCashBalanceResponseTransferSummary$inboundSchema:
  z.ZodType<
    CalculateCashBalanceResponseTransferSummary,
    z.ZodTypeDef,
    unknown
  > = z.object({
    amount: z.nullable(
      z.lazy(() =>
        CalculateCashBalanceResponseTransferSummaryAmount$inboundSchema
      ),
    ).optional(),
    id: z.string().optional(),
    mechanism:
      CalculateCashBalanceResponseTransferSummaryMechanism$inboundSchema
        .optional(),
  });

/** @internal */
export type CalculateCashBalanceResponseTransferSummary$Outbound = {
  amount?:
    | CalculateCashBalanceResponseTransferSummaryAmount$Outbound
    | null
    | undefined;
  id?: string | undefined;
  mechanism?: string | undefined;
};

/** @internal */
export const CalculateCashBalanceResponseTransferSummary$outboundSchema:
  z.ZodType<
    CalculateCashBalanceResponseTransferSummary$Outbound,
    z.ZodTypeDef,
    CalculateCashBalanceResponseTransferSummary
  > = z.object({
    amount: z.nullable(
      z.lazy(() =>
        CalculateCashBalanceResponseTransferSummaryAmount$outboundSchema
      ),
    ).optional(),
    id: z.string().optional(),
    mechanism:
      CalculateCashBalanceResponseTransferSummaryMechanism$outboundSchema
        .optional(),
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CalculateCashBalanceResponseTransferSummary$ {
  /** @deprecated use `CalculateCashBalanceResponseTransferSummary$inboundSchema` instead. */
  export const inboundSchema =
    CalculateCashBalanceResponseTransferSummary$inboundSchema;
  /** @deprecated use `CalculateCashBalanceResponseTransferSummary$outboundSchema` instead. */
  export const outboundSchema =
    CalculateCashBalanceResponseTransferSummary$outboundSchema;
  /** @deprecated use `CalculateCashBalanceResponseTransferSummary$Outbound` instead. */
  export type Outbound = CalculateCashBalanceResponseTransferSummary$Outbound;
}
