// @apexfintechsolutions/ascend-sdk/src/models/components/cancelictdepositrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

/**
 * Request to cancel an existing ICT deposit
 */
export type CancelIctDepositRequestCreate = {
  /**
   * Full name of the ICT deposit resource, which contains account id and ICT deposit id
   */
  name: string;
  /**
   * Reason why the ICT deposit is being canceled
   */
  reason?: string | undefined;
};

/** @internal */
export const CancelIctDepositRequestCreate$inboundSchema: z.ZodType<
  CancelIctDepositRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
  reason: z.string().optional(),
});

/** @internal */
export type CancelIctDepositRequestCreate$Outbound = {
  name: string;
  reason?: string | undefined;
};

/** @internal */
export const CancelIctDepositRequestCreate$outboundSchema: z.ZodType<
  CancelIctDepositRequestCreate$Outbound,
  z.ZodTypeDef,
  CancelIctDepositRequestCreate
> = z.object({
  name: z.string(),
  reason: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CancelIctDepositRequestCreate$ {
  /** @deprecated use `CancelIctDepositRequestCreate$inboundSchema` instead. */
  export const inboundSchema = CancelIctDepositRequestCreate$inboundSchema;
  /** @deprecated use `CancelIctDepositRequestCreate$outboundSchema` instead. */
  export const outboundSchema = CancelIctDepositRequestCreate$outboundSchema;
  /** @deprecated use `CancelIctDepositRequestCreate$Outbound` instead. */
  export type Outbound = CancelIctDepositRequestCreate$Outbound;
}
