// @apexfintechsolutions/ascend-sdk/src/models/components/cancelorderrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

/**
 * The message to request cancellation of an existing order
 */
export type CancelOrderRequestCreate = {
  /**
   * Format: accounts/{account_id}/orders/{order_id}
   */
  name: string;
};

/** @internal */
export const CancelOrderRequestCreate$inboundSchema: z.ZodType<
  CancelOrderRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
});

/** @internal */
export type CancelOrderRequestCreate$Outbound = {
  name: string;
};

/** @internal */
export const CancelOrderRequestCreate$outboundSchema: z.ZodType<
  CancelOrderRequestCreate$Outbound,
  z.ZodTypeDef,
  CancelOrderRequestCreate
> = z.object({
  name: z.string(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CancelOrderRequestCreate$ {
  /** @deprecated use `CancelOrderRequestCreate$inboundSchema` instead. */
  export const inboundSchema = CancelOrderRequestCreate$inboundSchema;
  /** @deprecated use `CancelOrderRequestCreate$outboundSchema` instead. */
  export const outboundSchema = CancelOrderRequestCreate$outboundSchema;
  /** @deprecated use `CancelOrderRequestCreate$Outbound` instead. */
  export type Outbound = CancelOrderRequestCreate$Outbound;
}
