// @apexfintechsolutions/ascend-sdk/src/models/components/cancelcashjournalrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

/**
 * Request to cancel an existing cash journal
 */
export type CancelCashJournalRequestCreate = {
  /**
   * The name of the cash journal to cancel
   */
  name: string;
  /**
   * The reason for canceling the cash journal Maximum of 100 characters are supported
   */
  reason?: string | undefined;
};

/** @internal */
export const CancelCashJournalRequestCreate$inboundSchema: z.ZodType<
  CancelCashJournalRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
  reason: z.string().optional(),
});

/** @internal */
export type CancelCashJournalRequestCreate$Outbound = {
  name: string;
  reason?: string | undefined;
};

/** @internal */
export const CancelCashJournalRequestCreate$outboundSchema: z.ZodType<
  CancelCashJournalRequestCreate$Outbound,
  z.ZodTypeDef,
  CancelCashJournalRequestCreate
> = z.object({
  name: z.string(),
  reason: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CancelCashJournalRequestCreate$ {
  /** @deprecated use `CancelCashJournalRequestCreate$inboundSchema` instead. */
  export const inboundSchema = CancelCashJournalRequestCreate$inboundSchema;
  /** @deprecated use `CancelCashJournalRequestCreate$outboundSchema` instead. */
  export const outboundSchema = CancelCashJournalRequestCreate$outboundSchema;
  /** @deprecated use `CancelCashJournalRequestCreate$Outbound` instead. */
  export type Outbound = CancelCashJournalRequestCreate$Outbound;
}
