// @apexfintechsolutions/ascend-sdk/src/models/components/canceltraderesponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";
import {
  Lot,
  Lot$inboundSchema,
  Lot$Outbound,
  Lot$outboundSchema,
} from "./lot.js";
import {
  YieldRecord,
  YieldRecord$inboundSchema,
  YieldRecord$Outbound,
  YieldRecord$outboundSchema,
} from "./yieldrecord.js";

/**
 * Used to calculate broadridge blotter code
 */
export enum CancelTradeResponseBrokerCapacity {
  CapacityUnspecified = "CAPACITY_UNSPECIFIED",
  Agency = "AGENCY",
  Principal = "PRINCIPAL",
  Mixed = "MIXED",
}
/**
 * Used to calculate broadridge blotter code
 */
export type CancelTradeResponseBrokerCapacityOpen = OpenEnum<
  typeof CancelTradeResponseBrokerCapacity
>;

/**
 * The price for the instrument that is prevailing in the market.
 */
export type CancelTradeResponsePrevailingMarketPrice = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/**
 * Total monetary value of the price_adjustment
 */
export type CancelTradeResponsePriceAdjustmentAmount = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/**
 * The percent at which the price was adjusted. Expressed as a number from 0.00-100 (rounded to 2 decimals)
 */
export type CancelTradeResponsePriceAdjustmentPercent = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/**
 * The type of price adjustment being applied by the broker to the net price of the security
 */
export enum CancelTradeResponsePriceAdjustmentType {
  PriceAdjustmentTypeUnspecified = "PRICE_ADJUSTMENT_TYPE_UNSPECIFIED",
  Markup = "MARKUP",
  Markdown = "MARKDOWN",
  SalesLoad = "SALES_LOAD",
}
/**
 * The type of price adjustment being applied by the broker to the net price of the security
 */
export type CancelTradeResponsePriceAdjustmentTypeOpen = OpenEnum<
  typeof CancelTradeResponsePriceAdjustmentType
>;

/**
 * Information about any price adjustments applied to the security
 */
export type CancelTradeResponsePriceAdjustmentRecord = {
  /**
   * Total monetary value of the price_adjustment
   */
  priceAdjustmentAmount?:
    | CancelTradeResponsePriceAdjustmentAmount
    | null
    | undefined;
  /**
   * The percent at which the price was adjusted. Expressed as a number from 0.00-100 (rounded to 2 decimals)
   */
  priceAdjustmentPercent?:
    | CancelTradeResponsePriceAdjustmentPercent
    | null
    | undefined;
  /**
   * The type of price adjustment being applied by the broker to the net price of the security
   */
  priceAdjustmentType?: CancelTradeResponsePriceAdjustmentTypeOpen | undefined;
};

/**
 * The trade that was canceled.
 */
export type CancelTradeResponseTrade = {
  /**
   * Free form text field containing additional information for a trade
   */
  additionalInstructions?: Array<string> | undefined;
  /**
   * Max Length 100 characters. Alternate order id from the street used for FRAC trades
   */
  alternateOrderId?: string | undefined;
  /**
   * uuid assigned by the Booking API if a trade belongs to an allocation
   */
  bookingApiTradeAllocationId?: string | undefined;
  /**
   * uuid assigned by the Booking API to all trades
   */
  bookingApiTradeId?: string | undefined;
  /**
   * Executing broker of the trade
   */
  broker?: string | undefined;
  /**
   * Used to calculate broadridge blotter code
   */
  brokerCapacity?: CancelTradeResponseBrokerCapacityOpen | undefined;
  /**
   * Free form text submitted by the client for internal purposes
   */
  clientMemos?: Array<string> | undefined;
  /**
   * 32 characters. The client order ID from the order submitted
   */
  clientOrderId?: string | undefined;
  /**
   * MIC code for the exchange
   */
  exchange?: string | undefined;
  /**
   * Execution id from the street
   */
  executionId?: string | undefined;
  /**
   * Indicates whether Apex is the clearing broker for this trade. When false, indicates Apex is the clearing broker
   */
  executionOnly?: boolean | undefined;
  /**
   * Max Length 100 characters. External system id provided by a client
   */
  externalId?: string | undefined;
  /**
   * The confirmation number associated with a mutual fund trade
   */
  fundConfirmationNumber?: string | undefined;
  /**
   * Max length 100 characters. Order id generated by trading-gateway (Trade-Ex) to uniquely identify all orders in their system. Used as the client_order_id on new order singles sent downstream of the trading-gateway
   */
  gatewayClientOrderId?: string | undefined;
  /**
   * If set to true, indicates the trade should be omitted from client billing
   */
  internalError?: boolean | undefined;
  /**
   * Set on penny-for-the-lot trades
   */
  isWriteoff?: boolean | undefined;
  /**
   * Repeated record containing information about the tax lots, if specified
   */
  lots?: Array<Lot> | undefined;
  /**
   * Max Length 100 characters. Internally generated order id that is returned to client on exec reports
   */
  orderId?: string | undefined;
  /**
   * The price for the instrument that is prevailing in the market.
   */
  prevailingMarketPrice?:
    | CancelTradeResponsePrevailingMarketPrice
    | null
    | undefined;
  /**
   * Information about any price adjustments applied to the security
   */
  priceAdjustmentRecord?:
    | CancelTradeResponsePriceAdjustmentRecord
    | null
    | undefined;
  /**
   * Used to calculate broadridge blotter code
   */
  route?: string | undefined;
  /**
   * The special instructions for a trade
   */
  specialInstructions?: Array<string> | undefined;
  /**
   * Name of the issuer of a security and additional descriptive information about the particular issue
   */
  symbolDescription?: string | undefined;
  /**
   * Indicates the trade was executed in a security that is not currently listed. When-issued securities are bought and sold before they are officially issued, allowing investors to speculate on their future value
   */
  whenIssued?: boolean | undefined;
  /**
   * The yields associated with a fixed income trade. only valid if the SecurityType is FIXED_INCOME.
   */
  yieldRecords?: Array<YieldRecord> | undefined;
};

/**
 * A response for the cancel trade method.
 */
export type CancelTradeResponse = {
  /**
   * The trade that was canceled.
   */
  trade?: CancelTradeResponseTrade | null | undefined;
};

/** @internal */
export const CancelTradeResponseBrokerCapacity$inboundSchema: z.ZodType<
  CancelTradeResponseBrokerCapacityOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(CancelTradeResponseBrokerCapacity),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const CancelTradeResponseBrokerCapacity$outboundSchema: z.ZodType<
  CancelTradeResponseBrokerCapacityOpen,
  z.ZodTypeDef,
  CancelTradeResponseBrokerCapacityOpen
> = z.union([
  z.nativeEnum(CancelTradeResponseBrokerCapacity),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CancelTradeResponseBrokerCapacity$ {
  /** @deprecated use `CancelTradeResponseBrokerCapacity$inboundSchema` instead. */
  export const inboundSchema = CancelTradeResponseBrokerCapacity$inboundSchema;
  /** @deprecated use `CancelTradeResponseBrokerCapacity$outboundSchema` instead. */
  export const outboundSchema =
    CancelTradeResponseBrokerCapacity$outboundSchema;
}

/** @internal */
export const CancelTradeResponsePrevailingMarketPrice$inboundSchema: z.ZodType<
  CancelTradeResponsePrevailingMarketPrice,
  z.ZodTypeDef,
  unknown
> = z.object({
  value: z.string().optional(),
});

/** @internal */
export type CancelTradeResponsePrevailingMarketPrice$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const CancelTradeResponsePrevailingMarketPrice$outboundSchema: z.ZodType<
  CancelTradeResponsePrevailingMarketPrice$Outbound,
  z.ZodTypeDef,
  CancelTradeResponsePrevailingMarketPrice
> = z.object({
  value: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CancelTradeResponsePrevailingMarketPrice$ {
  /** @deprecated use `CancelTradeResponsePrevailingMarketPrice$inboundSchema` instead. */
  export const inboundSchema =
    CancelTradeResponsePrevailingMarketPrice$inboundSchema;
  /** @deprecated use `CancelTradeResponsePrevailingMarketPrice$outboundSchema` instead. */
  export const outboundSchema =
    CancelTradeResponsePrevailingMarketPrice$outboundSchema;
  /** @deprecated use `CancelTradeResponsePrevailingMarketPrice$Outbound` instead. */
  export type Outbound = CancelTradeResponsePrevailingMarketPrice$Outbound;
}

/** @internal */
export const CancelTradeResponsePriceAdjustmentAmount$inboundSchema: z.ZodType<
  CancelTradeResponsePriceAdjustmentAmount,
  z.ZodTypeDef,
  unknown
> = z.object({
  value: z.string().optional(),
});

/** @internal */
export type CancelTradeResponsePriceAdjustmentAmount$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const CancelTradeResponsePriceAdjustmentAmount$outboundSchema: z.ZodType<
  CancelTradeResponsePriceAdjustmentAmount$Outbound,
  z.ZodTypeDef,
  CancelTradeResponsePriceAdjustmentAmount
> = z.object({
  value: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CancelTradeResponsePriceAdjustmentAmount$ {
  /** @deprecated use `CancelTradeResponsePriceAdjustmentAmount$inboundSchema` instead. */
  export const inboundSchema =
    CancelTradeResponsePriceAdjustmentAmount$inboundSchema;
  /** @deprecated use `CancelTradeResponsePriceAdjustmentAmount$outboundSchema` instead. */
  export const outboundSchema =
    CancelTradeResponsePriceAdjustmentAmount$outboundSchema;
  /** @deprecated use `CancelTradeResponsePriceAdjustmentAmount$Outbound` instead. */
  export type Outbound = CancelTradeResponsePriceAdjustmentAmount$Outbound;
}

/** @internal */
export const CancelTradeResponsePriceAdjustmentPercent$inboundSchema: z.ZodType<
  CancelTradeResponsePriceAdjustmentPercent,
  z.ZodTypeDef,
  unknown
> = z.object({
  value: z.string().optional(),
});

/** @internal */
export type CancelTradeResponsePriceAdjustmentPercent$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const CancelTradeResponsePriceAdjustmentPercent$outboundSchema:
  z.ZodType<
    CancelTradeResponsePriceAdjustmentPercent$Outbound,
    z.ZodTypeDef,
    CancelTradeResponsePriceAdjustmentPercent
  > = z.object({
    value: z.string().optional(),
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CancelTradeResponsePriceAdjustmentPercent$ {
  /** @deprecated use `CancelTradeResponsePriceAdjustmentPercent$inboundSchema` instead. */
  export const inboundSchema =
    CancelTradeResponsePriceAdjustmentPercent$inboundSchema;
  /** @deprecated use `CancelTradeResponsePriceAdjustmentPercent$outboundSchema` instead. */
  export const outboundSchema =
    CancelTradeResponsePriceAdjustmentPercent$outboundSchema;
  /** @deprecated use `CancelTradeResponsePriceAdjustmentPercent$Outbound` instead. */
  export type Outbound = CancelTradeResponsePriceAdjustmentPercent$Outbound;
}

/** @internal */
export const CancelTradeResponsePriceAdjustmentType$inboundSchema: z.ZodType<
  CancelTradeResponsePriceAdjustmentTypeOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(CancelTradeResponsePriceAdjustmentType),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const CancelTradeResponsePriceAdjustmentType$outboundSchema: z.ZodType<
  CancelTradeResponsePriceAdjustmentTypeOpen,
  z.ZodTypeDef,
  CancelTradeResponsePriceAdjustmentTypeOpen
> = z.union([
  z.nativeEnum(CancelTradeResponsePriceAdjustmentType),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CancelTradeResponsePriceAdjustmentType$ {
  /** @deprecated use `CancelTradeResponsePriceAdjustmentType$inboundSchema` instead. */
  export const inboundSchema =
    CancelTradeResponsePriceAdjustmentType$inboundSchema;
  /** @deprecated use `CancelTradeResponsePriceAdjustmentType$outboundSchema` instead. */
  export const outboundSchema =
    CancelTradeResponsePriceAdjustmentType$outboundSchema;
}

/** @internal */
export const CancelTradeResponsePriceAdjustmentRecord$inboundSchema: z.ZodType<
  CancelTradeResponsePriceAdjustmentRecord,
  z.ZodTypeDef,
  unknown
> = z.object({
  price_adjustment_amount: z.nullable(
    z.lazy(() => CancelTradeResponsePriceAdjustmentAmount$inboundSchema),
  ).optional(),
  price_adjustment_percent: z.nullable(
    z.lazy(() => CancelTradeResponsePriceAdjustmentPercent$inboundSchema),
  ).optional(),
  price_adjustment_type: CancelTradeResponsePriceAdjustmentType$inboundSchema
    .optional(),
}).transform((v) => {
  return remap$(v, {
    "price_adjustment_amount": "priceAdjustmentAmount",
    "price_adjustment_percent": "priceAdjustmentPercent",
    "price_adjustment_type": "priceAdjustmentType",
  });
});

/** @internal */
export type CancelTradeResponsePriceAdjustmentRecord$Outbound = {
  price_adjustment_amount?:
    | CancelTradeResponsePriceAdjustmentAmount$Outbound
    | null
    | undefined;
  price_adjustment_percent?:
    | CancelTradeResponsePriceAdjustmentPercent$Outbound
    | null
    | undefined;
  price_adjustment_type?: string | undefined;
};

/** @internal */
export const CancelTradeResponsePriceAdjustmentRecord$outboundSchema: z.ZodType<
  CancelTradeResponsePriceAdjustmentRecord$Outbound,
  z.ZodTypeDef,
  CancelTradeResponsePriceAdjustmentRecord
> = z.object({
  priceAdjustmentAmount: z.nullable(
    z.lazy(() => CancelTradeResponsePriceAdjustmentAmount$outboundSchema),
  ).optional(),
  priceAdjustmentPercent: z.nullable(
    z.lazy(() => CancelTradeResponsePriceAdjustmentPercent$outboundSchema),
  ).optional(),
  priceAdjustmentType: CancelTradeResponsePriceAdjustmentType$outboundSchema
    .optional(),
}).transform((v) => {
  return remap$(v, {
    priceAdjustmentAmount: "price_adjustment_amount",
    priceAdjustmentPercent: "price_adjustment_percent",
    priceAdjustmentType: "price_adjustment_type",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CancelTradeResponsePriceAdjustmentRecord$ {
  /** @deprecated use `CancelTradeResponsePriceAdjustmentRecord$inboundSchema` instead. */
  export const inboundSchema =
    CancelTradeResponsePriceAdjustmentRecord$inboundSchema;
  /** @deprecated use `CancelTradeResponsePriceAdjustmentRecord$outboundSchema` instead. */
  export const outboundSchema =
    CancelTradeResponsePriceAdjustmentRecord$outboundSchema;
  /** @deprecated use `CancelTradeResponsePriceAdjustmentRecord$Outbound` instead. */
  export type Outbound = CancelTradeResponsePriceAdjustmentRecord$Outbound;
}

/** @internal */
export const CancelTradeResponseTrade$inboundSchema: z.ZodType<
  CancelTradeResponseTrade,
  z.ZodTypeDef,
  unknown
> = z.object({
  additional_instructions: z.array(z.string()).optional(),
  alternate_order_id: z.string().optional(),
  booking_api_trade_allocation_id: z.string().optional(),
  booking_api_trade_id: z.string().optional(),
  broker: z.string().optional(),
  broker_capacity: CancelTradeResponseBrokerCapacity$inboundSchema.optional(),
  client_memos: z.array(z.string()).optional(),
  client_order_id: z.string().optional(),
  exchange: z.string().optional(),
  execution_id: z.string().optional(),
  execution_only: z.boolean().optional(),
  external_id: z.string().optional(),
  fund_confirmation_number: z.string().optional(),
  gateway_client_order_id: z.string().optional(),
  internal_error: z.boolean().optional(),
  is_writeoff: z.boolean().optional(),
  lots: z.array(Lot$inboundSchema).optional(),
  order_id: z.string().optional(),
  prevailing_market_price: z.nullable(
    z.lazy(() => CancelTradeResponsePrevailingMarketPrice$inboundSchema),
  ).optional(),
  price_adjustment_record: z.nullable(
    z.lazy(() => CancelTradeResponsePriceAdjustmentRecord$inboundSchema),
  ).optional(),
  route: z.string().optional(),
  special_instructions: z.array(z.string()).optional(),
  symbol_description: z.string().optional(),
  when_issued: z.boolean().optional(),
  yield_records: z.array(YieldRecord$inboundSchema).optional(),
}).transform((v) => {
  return remap$(v, {
    "additional_instructions": "additionalInstructions",
    "alternate_order_id": "alternateOrderId",
    "booking_api_trade_allocation_id": "bookingApiTradeAllocationId",
    "booking_api_trade_id": "bookingApiTradeId",
    "broker_capacity": "brokerCapacity",
    "client_memos": "clientMemos",
    "client_order_id": "clientOrderId",
    "execution_id": "executionId",
    "execution_only": "executionOnly",
    "external_id": "externalId",
    "fund_confirmation_number": "fundConfirmationNumber",
    "gateway_client_order_id": "gatewayClientOrderId",
    "internal_error": "internalError",
    "is_writeoff": "isWriteoff",
    "order_id": "orderId",
    "prevailing_market_price": "prevailingMarketPrice",
    "price_adjustment_record": "priceAdjustmentRecord",
    "special_instructions": "specialInstructions",
    "symbol_description": "symbolDescription",
    "when_issued": "whenIssued",
    "yield_records": "yieldRecords",
  });
});

/** @internal */
export type CancelTradeResponseTrade$Outbound = {
  additional_instructions?: Array<string> | undefined;
  alternate_order_id?: string | undefined;
  booking_api_trade_allocation_id?: string | undefined;
  booking_api_trade_id?: string | undefined;
  broker?: string | undefined;
  broker_capacity?: string | undefined;
  client_memos?: Array<string> | undefined;
  client_order_id?: string | undefined;
  exchange?: string | undefined;
  execution_id?: string | undefined;
  execution_only?: boolean | undefined;
  external_id?: string | undefined;
  fund_confirmation_number?: string | undefined;
  gateway_client_order_id?: string | undefined;
  internal_error?: boolean | undefined;
  is_writeoff?: boolean | undefined;
  lots?: Array<Lot$Outbound> | undefined;
  order_id?: string | undefined;
  prevailing_market_price?:
    | CancelTradeResponsePrevailingMarketPrice$Outbound
    | null
    | undefined;
  price_adjustment_record?:
    | CancelTradeResponsePriceAdjustmentRecord$Outbound
    | null
    | undefined;
  route?: string | undefined;
  special_instructions?: Array<string> | undefined;
  symbol_description?: string | undefined;
  when_issued?: boolean | undefined;
  yield_records?: Array<YieldRecord$Outbound> | undefined;
};

/** @internal */
export const CancelTradeResponseTrade$outboundSchema: z.ZodType<
  CancelTradeResponseTrade$Outbound,
  z.ZodTypeDef,
  CancelTradeResponseTrade
> = z.object({
  additionalInstructions: z.array(z.string()).optional(),
  alternateOrderId: z.string().optional(),
  bookingApiTradeAllocationId: z.string().optional(),
  bookingApiTradeId: z.string().optional(),
  broker: z.string().optional(),
  brokerCapacity: CancelTradeResponseBrokerCapacity$outboundSchema.optional(),
  clientMemos: z.array(z.string()).optional(),
  clientOrderId: z.string().optional(),
  exchange: z.string().optional(),
  executionId: z.string().optional(),
  executionOnly: z.boolean().optional(),
  externalId: z.string().optional(),
  fundConfirmationNumber: z.string().optional(),
  gatewayClientOrderId: z.string().optional(),
  internalError: z.boolean().optional(),
  isWriteoff: z.boolean().optional(),
  lots: z.array(Lot$outboundSchema).optional(),
  orderId: z.string().optional(),
  prevailingMarketPrice: z.nullable(
    z.lazy(() => CancelTradeResponsePrevailingMarketPrice$outboundSchema),
  ).optional(),
  priceAdjustmentRecord: z.nullable(
    z.lazy(() => CancelTradeResponsePriceAdjustmentRecord$outboundSchema),
  ).optional(),
  route: z.string().optional(),
  specialInstructions: z.array(z.string()).optional(),
  symbolDescription: z.string().optional(),
  whenIssued: z.boolean().optional(),
  yieldRecords: z.array(YieldRecord$outboundSchema).optional(),
}).transform((v) => {
  return remap$(v, {
    additionalInstructions: "additional_instructions",
    alternateOrderId: "alternate_order_id",
    bookingApiTradeAllocationId: "booking_api_trade_allocation_id",
    bookingApiTradeId: "booking_api_trade_id",
    brokerCapacity: "broker_capacity",
    clientMemos: "client_memos",
    clientOrderId: "client_order_id",
    executionId: "execution_id",
    executionOnly: "execution_only",
    externalId: "external_id",
    fundConfirmationNumber: "fund_confirmation_number",
    gatewayClientOrderId: "gateway_client_order_id",
    internalError: "internal_error",
    isWriteoff: "is_writeoff",
    orderId: "order_id",
    prevailingMarketPrice: "prevailing_market_price",
    priceAdjustmentRecord: "price_adjustment_record",
    specialInstructions: "special_instructions",
    symbolDescription: "symbol_description",
    whenIssued: "when_issued",
    yieldRecords: "yield_records",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CancelTradeResponseTrade$ {
  /** @deprecated use `CancelTradeResponseTrade$inboundSchema` instead. */
  export const inboundSchema = CancelTradeResponseTrade$inboundSchema;
  /** @deprecated use `CancelTradeResponseTrade$outboundSchema` instead. */
  export const outboundSchema = CancelTradeResponseTrade$outboundSchema;
  /** @deprecated use `CancelTradeResponseTrade$Outbound` instead. */
  export type Outbound = CancelTradeResponseTrade$Outbound;
}

/** @internal */
export const CancelTradeResponse$inboundSchema: z.ZodType<
  CancelTradeResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  trade: z.nullable(z.lazy(() => CancelTradeResponseTrade$inboundSchema))
    .optional(),
});

/** @internal */
export type CancelTradeResponse$Outbound = {
  trade?: CancelTradeResponseTrade$Outbound | null | undefined;
};

/** @internal */
export const CancelTradeResponse$outboundSchema: z.ZodType<
  CancelTradeResponse$Outbound,
  z.ZodTypeDef,
  CancelTradeResponse
> = z.object({
  trade: z.nullable(z.lazy(() => CancelTradeResponseTrade$outboundSchema))
    .optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CancelTradeResponse$ {
  /** @deprecated use `CancelTradeResponse$inboundSchema` instead. */
  export const inboundSchema = CancelTradeResponse$inboundSchema;
  /** @deprecated use `CancelTradeResponse$outboundSchema` instead. */
  export const outboundSchema = CancelTradeResponse$outboundSchema;
  /** @deprecated use `CancelTradeResponse$Outbound` instead. */
  export type Outbound = CancelTradeResponse$Outbound;
}
