// @apexfintechsolutions/ascend-sdk/src/models/components/completetraderequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import {
  FeeCreate,
  FeeCreate$inboundSchema,
  FeeCreate$Outbound,
  FeeCreate$outboundSchema,
} from "./feecreate.js";

/**
 * A request for completing a trade.
 */
export type CompleteTradeRequestCreate = {
  /**
   * Client calculated fees to use while completing an existing trade.
   */
  fees?: Array<FeeCreate> | undefined;
  /**
   * The name of the trade to complete.
   */
  name: string;
};

/** @internal */
export const CompleteTradeRequestCreate$inboundSchema: z.ZodType<
  CompleteTradeRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  fees: z.array(FeeCreate$inboundSchema).optional(),
  name: z.string(),
});

/** @internal */
export type CompleteTradeRequestCreate$Outbound = {
  fees?: Array<FeeCreate$Outbound> | undefined;
  name: string;
};

/** @internal */
export const CompleteTradeRequestCreate$outboundSchema: z.ZodType<
  CompleteTradeRequestCreate$Outbound,
  z.ZodTypeDef,
  CompleteTradeRequestCreate
> = z.object({
  fees: z.array(FeeCreate$outboundSchema).optional(),
  name: z.string(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CompleteTradeRequestCreate$ {
  /** @deprecated use `CompleteTradeRequestCreate$inboundSchema` instead. */
  export const inboundSchema = CompleteTradeRequestCreate$inboundSchema;
  /** @deprecated use `CompleteTradeRequestCreate$outboundSchema` instead. */
  export const outboundSchema = CompleteTradeRequestCreate$outboundSchema;
  /** @deprecated use `CompleteTradeRequestCreate$Outbound` instead. */
  export type Outbound = CompleteTradeRequestCreate$Outbound;
}
