// @apexfintechsolutions/ascend-sdk/src/models/components/decimalupdate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

/**
 * A representation of a decimal value, such as 2.5. Clients may convert values into language-native decimal formats, such as Java's [BigDecimal][] or Python's [decimal.Decimal][].
 *
 * @remarks
 *
 *  [BigDecimal]:
 *  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html
 *  [decimal.Decimal]: https://docs.python.org/3/library/decimal.html
 */
export type DecimalUpdate = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/** @internal */
export const DecimalUpdate$inboundSchema: z.ZodType<
  DecimalUpdate,
  z.ZodTypeDef,
  unknown
> = z.object({
  value: z.string().optional(),
});

/** @internal */
export type DecimalUpdate$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const DecimalUpdate$outboundSchema: z.ZodType<
  DecimalUpdate$Outbound,
  z.ZodTypeDef,
  DecimalUpdate
> = z.object({
  value: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace DecimalUpdate$ {
  /** @deprecated use `DecimalUpdate$inboundSchema` instead. */
  export const inboundSchema = DecimalUpdate$inboundSchema;
  /** @deprecated use `DecimalUpdate$outboundSchema` instead. */
  export const outboundSchema = DecimalUpdate$outboundSchema;
  /** @deprecated use `DecimalUpdate$Outbound` instead. */
  export type Outbound = DecimalUpdate$Outbound;
}
