// @apexfintechsolutions/ascend-sdk/src/models/components/depositscheduledetailsupdate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import {
  DecimalUpdate,
  DecimalUpdate$inboundSchema,
  DecimalUpdate$Outbound,
  DecimalUpdate$outboundSchema,
} from "./decimalupdate.js";

/**
 * Details of deposit schedule transfers
 */
export type DepositScheduleDetailsUpdate = {
  /**
   * A representation of a decimal value, such as 2.5. Clients may convert values into language-native decimal formats, such as Java's [BigDecimal][] or Python's [decimal.Decimal][].
   *
   * @remarks
   *
   *  [BigDecimal]:
   *  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html
   *  [decimal.Decimal]: https://docs.python.org/3/library/decimal.html
   */
  amount?: DecimalUpdate | undefined;
};

/** @internal */
export const DepositScheduleDetailsUpdate$inboundSchema: z.ZodType<
  DepositScheduleDetailsUpdate,
  z.ZodTypeDef,
  unknown
> = z.object({
  amount: DecimalUpdate$inboundSchema.optional(),
});

/** @internal */
export type DepositScheduleDetailsUpdate$Outbound = {
  amount?: DecimalUpdate$Outbound | undefined;
};

/** @internal */
export const DepositScheduleDetailsUpdate$outboundSchema: z.ZodType<
  DepositScheduleDetailsUpdate$Outbound,
  z.ZodTypeDef,
  DepositScheduleDetailsUpdate
> = z.object({
  amount: DecimalUpdate$outboundSchema.optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace DepositScheduleDetailsUpdate$ {
  /** @deprecated use `DepositScheduleDetailsUpdate$inboundSchema` instead. */
  export const inboundSchema = DepositScheduleDetailsUpdate$inboundSchema;
  /** @deprecated use `DepositScheduleDetailsUpdate$outboundSchema` instead. */
  export const outboundSchema = DepositScheduleDetailsUpdate$outboundSchema;
  /** @deprecated use `DepositScheduleDetailsUpdate$Outbound` instead. */
  export type Outbound = DepositScheduleDetailsUpdate$Outbound;
}
