// @apexfintechsolutions/ascend-sdk/src/models/components/endrestrictionrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

/**
 * The request for ending a Restriction on an Account.
 */
export type EndRestrictionRequestCreate = {
  /**
   * The reason to end the restriction.
   */
  reason: string;
};

/** @internal */
export const EndRestrictionRequestCreate$inboundSchema: z.ZodType<
  EndRestrictionRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  reason: z.string(),
});

/** @internal */
export type EndRestrictionRequestCreate$Outbound = {
  reason: string;
};

/** @internal */
export const EndRestrictionRequestCreate$outboundSchema: z.ZodType<
  EndRestrictionRequestCreate$Outbound,
  z.ZodTypeDef,
  EndRestrictionRequestCreate
> = z.object({
  reason: z.string(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace EndRestrictionRequestCreate$ {
  /** @deprecated use `EndRestrictionRequestCreate$inboundSchema` instead. */
  export const inboundSchema = EndRestrictionRequestCreate$inboundSchema;
  /** @deprecated use `EndRestrictionRequestCreate$outboundSchema` instead. */
  export const outboundSchema = EndRestrictionRequestCreate$outboundSchema;
  /** @deprecated use `EndRestrictionRequestCreate$Outbound` instead. */
  export type Outbound = EndRestrictionRequestCreate$Outbound;
}
