// @apexfintechsolutions/ascend-sdk/src/models/components/enrollmentcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";
import {
  BeneficiaryEnrollmentMetadataCreate,
  BeneficiaryEnrollmentMetadataCreate$inboundSchema,
  BeneficiaryEnrollmentMetadataCreate$Outbound,
  BeneficiaryEnrollmentMetadataCreate$outboundSchema,
} from "./beneficiaryenrollmentmetadatacreate.js";
import {
  CorporationEnrollmentMetadataCreate,
  CorporationEnrollmentMetadataCreate$inboundSchema,
  CorporationEnrollmentMetadataCreate$Outbound,
  CorporationEnrollmentMetadataCreate$outboundSchema,
} from "./corporationenrollmentmetadatacreate.js";
import {
  CustodialEnrollmentMetadataCreate,
  CustodialEnrollmentMetadataCreate$inboundSchema,
  CustodialEnrollmentMetadataCreate$Outbound,
  CustodialEnrollmentMetadataCreate$outboundSchema,
} from "./custodialenrollmentmetadatacreate.js";
import {
  EstateEnrollmentMetadataCreate,
  EstateEnrollmentMetadataCreate$inboundSchema,
  EstateEnrollmentMetadataCreate$Outbound,
  EstateEnrollmentMetadataCreate$outboundSchema,
} from "./estateenrollmentmetadatacreate.js";
import {
  ForeignIndividualAccountEnrollmentMetadataCreate,
  ForeignIndividualAccountEnrollmentMetadataCreate$inboundSchema,
  ForeignIndividualAccountEnrollmentMetadataCreate$Outbound,
  ForeignIndividualAccountEnrollmentMetadataCreate$outboundSchema,
} from "./foreignindividualaccountenrollmentmetadatacreate.js";
import {
  ForeignJointAccountEnrollmentMetadataCreate,
  ForeignJointAccountEnrollmentMetadataCreate$inboundSchema,
  ForeignJointAccountEnrollmentMetadataCreate$Outbound,
  ForeignJointAccountEnrollmentMetadataCreate$outboundSchema,
} from "./foreignjointaccountenrollmentmetadatacreate.js";
import {
  FPSLEnrollmentMetaDataCreate,
  FPSLEnrollmentMetaDataCreate$inboundSchema,
  FPSLEnrollmentMetaDataCreate$Outbound,
  FPSLEnrollmentMetaDataCreate$outboundSchema,
} from "./fpslenrollmentmetadatacreate.js";
import {
  IndividualEnrollmentMetadataCreate,
  IndividualEnrollmentMetadataCreate$inboundSchema,
  IndividualEnrollmentMetadataCreate$Outbound,
  IndividualEnrollmentMetadataCreate$outboundSchema,
} from "./individualenrollmentmetadatacreate.js";
import {
  IRABeneficiaryEnrollmentMetadataCreate,
  IRABeneficiaryEnrollmentMetadataCreate$inboundSchema,
  IRABeneficiaryEnrollmentMetadataCreate$Outbound,
  IRABeneficiaryEnrollmentMetadataCreate$outboundSchema,
} from "./irabeneficiaryenrollmentmetadatacreate.js";
import {
  IRARolloverEnrollmentMetadataCreate,
  IRARolloverEnrollmentMetadataCreate$inboundSchema,
  IRARolloverEnrollmentMetadataCreate$Outbound,
  IRARolloverEnrollmentMetadataCreate$outboundSchema,
} from "./irarolloverenrollmentmetadatacreate.js";
import {
  IRARothEnrollmentMetadataCreate,
  IRARothEnrollmentMetadataCreate$inboundSchema,
  IRARothEnrollmentMetadataCreate$Outbound,
  IRARothEnrollmentMetadataCreate$outboundSchema,
} from "./irarothenrollmentmetadatacreate.js";
import {
  IRASEPEnrollmentMetadataCreate,
  IRASEPEnrollmentMetadataCreate$inboundSchema,
  IRASEPEnrollmentMetadataCreate$Outbound,
  IRASEPEnrollmentMetadataCreate$outboundSchema,
} from "./irasepenrollmentmetadatacreate.js";
import {
  IRASimpleEnrollmentMetadataCreate,
  IRASimpleEnrollmentMetadataCreate$inboundSchema,
  IRASimpleEnrollmentMetadataCreate$Outbound,
  IRASimpleEnrollmentMetadataCreate$outboundSchema,
} from "./irasimpleenrollmentmetadatacreate.js";
import {
  IRATraditionalEnrollmentMetadataCreate,
  IRATraditionalEnrollmentMetadataCreate$inboundSchema,
  IRATraditionalEnrollmentMetadataCreate$Outbound,
  IRATraditionalEnrollmentMetadataCreate$outboundSchema,
} from "./iratraditionalenrollmentmetadatacreate.js";
import {
  JointCommunityPropertyEnrollmentMetadataCreate,
  JointCommunityPropertyEnrollmentMetadataCreate$inboundSchema,
  JointCommunityPropertyEnrollmentMetadataCreate$Outbound,
  JointCommunityPropertyEnrollmentMetadataCreate$outboundSchema,
} from "./jointcommunitypropertyenrollmentmetadatacreate.js";
import {
  JointTenantsByEntiretyEnrollmentMetadataCreate,
  JointTenantsByEntiretyEnrollmentMetadataCreate$inboundSchema,
  JointTenantsByEntiretyEnrollmentMetadataCreate$Outbound,
  JointTenantsByEntiretyEnrollmentMetadataCreate$outboundSchema,
} from "./jointtenantsbyentiretyenrollmentmetadatacreate.js";
import {
  JointTenantsInCommonEnrollmentMetadataCreate,
  JointTenantsInCommonEnrollmentMetadataCreate$inboundSchema,
  JointTenantsInCommonEnrollmentMetadataCreate$Outbound,
  JointTenantsInCommonEnrollmentMetadataCreate$outboundSchema,
} from "./jointtenantsincommonenrollmentmetadatacreate.js";
import {
  JointWithRightsOfSurvivorshipEnrollmentMetadataCreate,
  JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$inboundSchema,
  JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$Outbound,
  JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$outboundSchema,
} from "./jointwithrightsofsurvivorshipenrollmentmetadatacreate.js";
import {
  LLCEnrollmentMetadataCreate,
  LLCEnrollmentMetadataCreate$inboundSchema,
  LLCEnrollmentMetadataCreate$Outbound,
  LLCEnrollmentMetadataCreate$outboundSchema,
} from "./llcenrollmentmetadatacreate.js";
import {
  OperatingEnrollmentMetadataCreate,
  OperatingEnrollmentMetadataCreate$inboundSchema,
  OperatingEnrollmentMetadataCreate$Outbound,
  OperatingEnrollmentMetadataCreate$outboundSchema,
} from "./operatingenrollmentmetadatacreate.js";
import {
  OrdersOptionsTradingEnrollmentMetadataCreate,
  OrdersOptionsTradingEnrollmentMetadataCreate$inboundSchema,
  OrdersOptionsTradingEnrollmentMetadataCreate$Outbound,
  OrdersOptionsTradingEnrollmentMetadataCreate$outboundSchema,
} from "./ordersoptionstradingenrollmentmetadatacreate.js";
import {
  TrustEnrollmentMetadataCreate,
  TrustEnrollmentMetadataCreate$inboundSchema,
  TrustEnrollmentMetadataCreate$Outbound,
  TrustEnrollmentMetadataCreate$outboundSchema,
} from "./trustenrollmentmetadatacreate.js";
import {
  VirtualAccountNumberEnrollmentMetadataCreate,
  VirtualAccountNumberEnrollmentMetadataCreate$inboundSchema,
  VirtualAccountNumberEnrollmentMetadataCreate$Outbound,
  VirtualAccountNumberEnrollmentMetadataCreate$outboundSchema,
} from "./virtualaccountnumberenrollmentmetadatacreate.js";

/**
 * The consent method for the enrollment. Defaults to ESIGNATURE.
 */
export enum EnrollmentCreateConsentMethod {
  ConsentMethodUnspecified = "CONSENT_METHOD_UNSPECIFIED",
  Esignature = "ESIGNATURE",
  WetSignature = "WET_SIGNATURE",
  NegativeConsentConversion = "NEGATIVE_CONSENT_CONVERSION",
  InternalConversion = "INTERNAL_CONVERSION",
}
/**
 * The consent method for the enrollment. Defaults to ESIGNATURE.
 */
export type EnrollmentCreateConsentMethodOpen = OpenEnum<
  typeof EnrollmentCreateConsentMethod
>;

/**
 * Describes the name of the enrollment; Expressed as an enum
 */
export enum EnrollmentCreateType {
  EnrollmentTypeUnspecified = "ENROLLMENT_TYPE_UNSPECIFIED",
  RegistrationIndividual = "REGISTRATION_INDIVIDUAL",
  LendingFullyPaidStockLoan = "LENDING_FULLY_PAID_STOCK_LOAN",
  BeneficiaryDesignation = "BENEFICIARY_DESIGNATION",
  RegistrationJointWros = "REGISTRATION_JOINT_WROS",
  RegistrationJointTic = "REGISTRATION_JOINT_TIC",
  RegistrationJointTbe = "REGISTRATION_JOINT_TBE",
  RegistrationJointCp = "REGISTRATION_JOINT_CP",
  RegistrationEstate = "REGISTRATION_ESTATE",
  RegistrationIraTraditional = "REGISTRATION_IRA_TRADITIONAL",
  RegistrationIraSimple = "REGISTRATION_IRA_SIMPLE",
  RegistrationIraSep = "REGISTRATION_IRA_SEP",
  RegistrationIraRoth = "REGISTRATION_IRA_ROTH",
  RegistrationIraRollover = "REGISTRATION_IRA_ROLLOVER",
  RegistrationTrust = "REGISTRATION_TRUST",
  RegistrationCorporation = "REGISTRATION_CORPORATION",
  RegistrationLlc = "REGISTRATION_LLC",
  CashFdicCashSweep = "CASH_FDIC_CASH_SWEEP",
  RetirementBeneficiaryDesignation = "RETIREMENT_BENEFICIARY_DESIGNATION",
  DividendReinvestmentPlan = "DIVIDEND_REINVESTMENT_PLAN",
  RegistrationIraBeneficiaryTraditional =
    "REGISTRATION_IRA_BENEFICIARY_TRADITIONAL",
  RegistrationIraBeneficiaryRoth = "REGISTRATION_IRA_BENEFICIARY_ROTH",
  RegistrationIndividualForeign = "REGISTRATION_INDIVIDUAL_FOREIGN",
  RegistrationCustodial = "REGISTRATION_CUSTODIAL",
  VirtualAccountNumber = "VIRTUAL_ACCOUNT_NUMBER",
}
/**
 * Describes the name of the enrollment; Expressed as an enum
 */
export type EnrollmentCreateTypeOpen = OpenEnum<typeof EnrollmentCreateType>;

/**
 * An Enrollment represents programs the account may enroll in.
 */
export type EnrollmentCreate = {
  /**
   * Enrollment metadata for the BENEFICIARY_DESIGNATION enrollment type.
   */
  beneficiaryEnrollmentMetadata?:
    | BeneficiaryEnrollmentMetadataCreate
    | undefined;
  /**
   * The consent method for the enrollment. Defaults to ESIGNATURE.
   */
  consentMethod?: EnrollmentCreateConsentMethodOpen | undefined;
  corporationEnrollmentMetadata?:
    | CorporationEnrollmentMetadataCreate
    | undefined;
  /**
   * Enrollment metadata for the CUSTODIAL enrollment type
   */
  custodialEnrollmentMetadata?: CustodialEnrollmentMetadataCreate | undefined;
  /**
   * Enrollment metadata for estate enrollments
   */
  estateEnrollmentMetadata?: EstateEnrollmentMetadataCreate | undefined;
  foreignIndividualAccountEnrollmentMetadata?:
    | ForeignIndividualAccountEnrollmentMetadataCreate
    | undefined;
  /**
   * Enrollment metadata for the FOREIGN_JOINT_WROS enrollment type
   */
  foreignJointAccountEnrollmentMetadata?:
    | ForeignJointAccountEnrollmentMetadataCreate
    | undefined;
  /**
   * Percentages for FPSL Enrollment, must equal 100
   */
  fpslEnrollmentMetadata?: FPSLEnrollmentMetaDataCreate | undefined;
  /**
   * Enrollment metadata for Individual accounts enrollment type
   */
  individualEnrollmentMetadata?: IndividualEnrollmentMetadataCreate | undefined;
  /**
   * Enrollment metadata for beneficiary IRA accounts enrollment type
   */
  iraBeneficiaryEnrollmentMetadata?:
    | IRABeneficiaryEnrollmentMetadataCreate
    | undefined;
  /**
   * Enrollment metadata for Rollover IRA accounts enrollment type
   */
  iraRolloverEnrollmentMetadata?:
    | IRARolloverEnrollmentMetadataCreate
    | undefined;
  /**
   * Enrollment metadata for ROTH IRA accounts enrollment type
   */
  iraRothEnrollmentMetadata?: IRARothEnrollmentMetadataCreate | undefined;
  /**
   * Enrollment metadata for IRA SEP_IRA accounts enrollment type
   */
  iraSepEnrollmentMetadata?: IRASEPEnrollmentMetadataCreate | undefined;
  /**
   * Enrollment metadata for Simple IRA accounts enrollment type
   */
  iraSimpleEnrollmentMetadata?: IRASimpleEnrollmentMetadataCreate | undefined;
  /**
   * Enrollment metadata for Traditional IRA accounts enrollment type
   */
  iraTraditionalEnrollmentMetadata?:
    | IRATraditionalEnrollmentMetadataCreate
    | undefined;
  /**
   * Enrollment metadata for the Joint Community Property Registration enrollment type
   */
  jointCommunityPropertyEnrollmentMetadata?:
    | JointCommunityPropertyEnrollmentMetadataCreate
    | undefined;
  /**
   * Enrollment metadata for the Joint Tenants by Entirety Registration enrollment type
   */
  jointTenantsByEntiretyEnrollmentMetadata?:
    | JointTenantsByEntiretyEnrollmentMetadataCreate
    | undefined;
  /**
   * Enrollment metadata for the Joint Tenants In Common enrollment type
   */
  jointTenantsInCommonEnrollmentMetadata?:
    | JointTenantsInCommonEnrollmentMetadataCreate
    | undefined;
  /**
   * Enrollment metadata for the With Right of Survivorship enrollment type
   */
  jointWithRightsOfSurvivorshipEnrollmentMetadata?:
    | JointWithRightsOfSurvivorshipEnrollmentMetadataCreate
    | undefined;
  llcEnrollmentMetadata?: LLCEnrollmentMetadataCreate | undefined;
  /**
   * Enrollment metadata for the REGISTRATION_OPERATING enrollment type.
   */
  operatingEnrollmentMetadata?: OperatingEnrollmentMetadataCreate | undefined;
  /**
   * Enrollment metadata for the ORDERS_OPTIONS_TRADING enrollment type
   */
  ordersOptionsTradingEnrollmentMetadata?:
    | OrdersOptionsTradingEnrollmentMetadataCreate
    | undefined;
  /**
   * The ULID is associated with the approver of a given enrollment. The approver you create will contain the CRD Number issued to the person by FINRA. As an RIA, you should use the ULID associated with Apex's approver.
   */
  principalApproverId: string;
  trustEnrollmentMetadata?: TrustEnrollmentMetadataCreate | undefined;
  /**
   * Describes the name of the enrollment; Expressed as an enum
   */
  type: EnrollmentCreateTypeOpen;
  /**
   * Enrollment metadata for the VIRTUAL_ACCOUNT_NUMBER enrollment type
   */
  virtualAccountNumberEnrollmentMetadata?:
    | VirtualAccountNumberEnrollmentMetadataCreate
    | undefined;
};

/** @internal */
export const EnrollmentCreateConsentMethod$inboundSchema: z.ZodType<
  EnrollmentCreateConsentMethodOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(EnrollmentCreateConsentMethod),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const EnrollmentCreateConsentMethod$outboundSchema: z.ZodType<
  EnrollmentCreateConsentMethodOpen,
  z.ZodTypeDef,
  EnrollmentCreateConsentMethodOpen
> = z.union([
  z.nativeEnum(EnrollmentCreateConsentMethod),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace EnrollmentCreateConsentMethod$ {
  /** @deprecated use `EnrollmentCreateConsentMethod$inboundSchema` instead. */
  export const inboundSchema = EnrollmentCreateConsentMethod$inboundSchema;
  /** @deprecated use `EnrollmentCreateConsentMethod$outboundSchema` instead. */
  export const outboundSchema = EnrollmentCreateConsentMethod$outboundSchema;
}

/** @internal */
export const EnrollmentCreateType$inboundSchema: z.ZodType<
  EnrollmentCreateTypeOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(EnrollmentCreateType),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const EnrollmentCreateType$outboundSchema: z.ZodType<
  EnrollmentCreateTypeOpen,
  z.ZodTypeDef,
  EnrollmentCreateTypeOpen
> = z.union([
  z.nativeEnum(EnrollmentCreateType),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace EnrollmentCreateType$ {
  /** @deprecated use `EnrollmentCreateType$inboundSchema` instead. */
  export const inboundSchema = EnrollmentCreateType$inboundSchema;
  /** @deprecated use `EnrollmentCreateType$outboundSchema` instead. */
  export const outboundSchema = EnrollmentCreateType$outboundSchema;
}

/** @internal */
export const EnrollmentCreate$inboundSchema: z.ZodType<
  EnrollmentCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  beneficiary_enrollment_metadata:
    BeneficiaryEnrollmentMetadataCreate$inboundSchema.optional(),
  consent_method: EnrollmentCreateConsentMethod$inboundSchema.optional(),
  corporation_enrollment_metadata:
    CorporationEnrollmentMetadataCreate$inboundSchema.optional(),
  custodial_enrollment_metadata: CustodialEnrollmentMetadataCreate$inboundSchema
    .optional(),
  estate_enrollment_metadata: EstateEnrollmentMetadataCreate$inboundSchema
    .optional(),
  foreign_individual_account_enrollment_metadata:
    ForeignIndividualAccountEnrollmentMetadataCreate$inboundSchema.optional(),
  foreign_joint_account_enrollment_metadata:
    ForeignJointAccountEnrollmentMetadataCreate$inboundSchema.optional(),
  fpsl_enrollment_metadata: FPSLEnrollmentMetaDataCreate$inboundSchema
    .optional(),
  individual_enrollment_metadata:
    IndividualEnrollmentMetadataCreate$inboundSchema.optional(),
  ira_beneficiary_enrollment_metadata:
    IRABeneficiaryEnrollmentMetadataCreate$inboundSchema.optional(),
  ira_rollover_enrollment_metadata:
    IRARolloverEnrollmentMetadataCreate$inboundSchema.optional(),
  ira_roth_enrollment_metadata: IRARothEnrollmentMetadataCreate$inboundSchema
    .optional(),
  ira_sep_enrollment_metadata: IRASEPEnrollmentMetadataCreate$inboundSchema
    .optional(),
  ira_simple_enrollment_metadata:
    IRASimpleEnrollmentMetadataCreate$inboundSchema.optional(),
  ira_traditional_enrollment_metadata:
    IRATraditionalEnrollmentMetadataCreate$inboundSchema.optional(),
  joint_community_property_enrollment_metadata:
    JointCommunityPropertyEnrollmentMetadataCreate$inboundSchema.optional(),
  joint_tenants_by_entirety_enrollment_metadata:
    JointTenantsByEntiretyEnrollmentMetadataCreate$inboundSchema.optional(),
  joint_tenants_in_common_enrollment_metadata:
    JointTenantsInCommonEnrollmentMetadataCreate$inboundSchema.optional(),
  joint_with_rights_of_survivorship_enrollment_metadata:
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$inboundSchema
      .optional(),
  llc_enrollment_metadata: LLCEnrollmentMetadataCreate$inboundSchema.optional(),
  operating_enrollment_metadata: OperatingEnrollmentMetadataCreate$inboundSchema
    .optional(),
  orders_options_trading_enrollment_metadata:
    OrdersOptionsTradingEnrollmentMetadataCreate$inboundSchema.optional(),
  principal_approver_id: z.string(),
  trust_enrollment_metadata: TrustEnrollmentMetadataCreate$inboundSchema
    .optional(),
  type: EnrollmentCreateType$inboundSchema,
  virtual_account_number_enrollment_metadata:
    VirtualAccountNumberEnrollmentMetadataCreate$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "beneficiary_enrollment_metadata": "beneficiaryEnrollmentMetadata",
    "consent_method": "consentMethod",
    "corporation_enrollment_metadata": "corporationEnrollmentMetadata",
    "custodial_enrollment_metadata": "custodialEnrollmentMetadata",
    "estate_enrollment_metadata": "estateEnrollmentMetadata",
    "foreign_individual_account_enrollment_metadata":
      "foreignIndividualAccountEnrollmentMetadata",
    "foreign_joint_account_enrollment_metadata":
      "foreignJointAccountEnrollmentMetadata",
    "fpsl_enrollment_metadata": "fpslEnrollmentMetadata",
    "individual_enrollment_metadata": "individualEnrollmentMetadata",
    "ira_beneficiary_enrollment_metadata": "iraBeneficiaryEnrollmentMetadata",
    "ira_rollover_enrollment_metadata": "iraRolloverEnrollmentMetadata",
    "ira_roth_enrollment_metadata": "iraRothEnrollmentMetadata",
    "ira_sep_enrollment_metadata": "iraSepEnrollmentMetadata",
    "ira_simple_enrollment_metadata": "iraSimpleEnrollmentMetadata",
    "ira_traditional_enrollment_metadata": "iraTraditionalEnrollmentMetadata",
    "joint_community_property_enrollment_metadata":
      "jointCommunityPropertyEnrollmentMetadata",
    "joint_tenants_by_entirety_enrollment_metadata":
      "jointTenantsByEntiretyEnrollmentMetadata",
    "joint_tenants_in_common_enrollment_metadata":
      "jointTenantsInCommonEnrollmentMetadata",
    "joint_with_rights_of_survivorship_enrollment_metadata":
      "jointWithRightsOfSurvivorshipEnrollmentMetadata",
    "llc_enrollment_metadata": "llcEnrollmentMetadata",
    "operating_enrollment_metadata": "operatingEnrollmentMetadata",
    "orders_options_trading_enrollment_metadata":
      "ordersOptionsTradingEnrollmentMetadata",
    "principal_approver_id": "principalApproverId",
    "trust_enrollment_metadata": "trustEnrollmentMetadata",
    "virtual_account_number_enrollment_metadata":
      "virtualAccountNumberEnrollmentMetadata",
  });
});

/** @internal */
export type EnrollmentCreate$Outbound = {
  beneficiary_enrollment_metadata?:
    | BeneficiaryEnrollmentMetadataCreate$Outbound
    | undefined;
  consent_method?: string | undefined;
  corporation_enrollment_metadata?:
    | CorporationEnrollmentMetadataCreate$Outbound
    | undefined;
  custodial_enrollment_metadata?:
    | CustodialEnrollmentMetadataCreate$Outbound
    | undefined;
  estate_enrollment_metadata?:
    | EstateEnrollmentMetadataCreate$Outbound
    | undefined;
  foreign_individual_account_enrollment_metadata?:
    | ForeignIndividualAccountEnrollmentMetadataCreate$Outbound
    | undefined;
  foreign_joint_account_enrollment_metadata?:
    | ForeignJointAccountEnrollmentMetadataCreate$Outbound
    | undefined;
  fpsl_enrollment_metadata?: FPSLEnrollmentMetaDataCreate$Outbound | undefined;
  individual_enrollment_metadata?:
    | IndividualEnrollmentMetadataCreate$Outbound
    | undefined;
  ira_beneficiary_enrollment_metadata?:
    | IRABeneficiaryEnrollmentMetadataCreate$Outbound
    | undefined;
  ira_rollover_enrollment_metadata?:
    | IRARolloverEnrollmentMetadataCreate$Outbound
    | undefined;
  ira_roth_enrollment_metadata?:
    | IRARothEnrollmentMetadataCreate$Outbound
    | undefined;
  ira_sep_enrollment_metadata?:
    | IRASEPEnrollmentMetadataCreate$Outbound
    | undefined;
  ira_simple_enrollment_metadata?:
    | IRASimpleEnrollmentMetadataCreate$Outbound
    | undefined;
  ira_traditional_enrollment_metadata?:
    | IRATraditionalEnrollmentMetadataCreate$Outbound
    | undefined;
  joint_community_property_enrollment_metadata?:
    | JointCommunityPropertyEnrollmentMetadataCreate$Outbound
    | undefined;
  joint_tenants_by_entirety_enrollment_metadata?:
    | JointTenantsByEntiretyEnrollmentMetadataCreate$Outbound
    | undefined;
  joint_tenants_in_common_enrollment_metadata?:
    | JointTenantsInCommonEnrollmentMetadataCreate$Outbound
    | undefined;
  joint_with_rights_of_survivorship_enrollment_metadata?:
    | JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$Outbound
    | undefined;
  llc_enrollment_metadata?: LLCEnrollmentMetadataCreate$Outbound | undefined;
  operating_enrollment_metadata?:
    | OperatingEnrollmentMetadataCreate$Outbound
    | undefined;
  orders_options_trading_enrollment_metadata?:
    | OrdersOptionsTradingEnrollmentMetadataCreate$Outbound
    | undefined;
  principal_approver_id: string;
  trust_enrollment_metadata?:
    | TrustEnrollmentMetadataCreate$Outbound
    | undefined;
  type: string;
  virtual_account_number_enrollment_metadata?:
    | VirtualAccountNumberEnrollmentMetadataCreate$Outbound
    | undefined;
};

/** @internal */
export const EnrollmentCreate$outboundSchema: z.ZodType<
  EnrollmentCreate$Outbound,
  z.ZodTypeDef,
  EnrollmentCreate
> = z.object({
  beneficiaryEnrollmentMetadata:
    BeneficiaryEnrollmentMetadataCreate$outboundSchema.optional(),
  consentMethod: EnrollmentCreateConsentMethod$outboundSchema.optional(),
  corporationEnrollmentMetadata:
    CorporationEnrollmentMetadataCreate$outboundSchema.optional(),
  custodialEnrollmentMetadata: CustodialEnrollmentMetadataCreate$outboundSchema
    .optional(),
  estateEnrollmentMetadata: EstateEnrollmentMetadataCreate$outboundSchema
    .optional(),
  foreignIndividualAccountEnrollmentMetadata:
    ForeignIndividualAccountEnrollmentMetadataCreate$outboundSchema.optional(),
  foreignJointAccountEnrollmentMetadata:
    ForeignJointAccountEnrollmentMetadataCreate$outboundSchema.optional(),
  fpslEnrollmentMetadata: FPSLEnrollmentMetaDataCreate$outboundSchema
    .optional(),
  individualEnrollmentMetadata:
    IndividualEnrollmentMetadataCreate$outboundSchema.optional(),
  iraBeneficiaryEnrollmentMetadata:
    IRABeneficiaryEnrollmentMetadataCreate$outboundSchema.optional(),
  iraRolloverEnrollmentMetadata:
    IRARolloverEnrollmentMetadataCreate$outboundSchema.optional(),
  iraRothEnrollmentMetadata: IRARothEnrollmentMetadataCreate$outboundSchema
    .optional(),
  iraSepEnrollmentMetadata: IRASEPEnrollmentMetadataCreate$outboundSchema
    .optional(),
  iraSimpleEnrollmentMetadata: IRASimpleEnrollmentMetadataCreate$outboundSchema
    .optional(),
  iraTraditionalEnrollmentMetadata:
    IRATraditionalEnrollmentMetadataCreate$outboundSchema.optional(),
  jointCommunityPropertyEnrollmentMetadata:
    JointCommunityPropertyEnrollmentMetadataCreate$outboundSchema.optional(),
  jointTenantsByEntiretyEnrollmentMetadata:
    JointTenantsByEntiretyEnrollmentMetadataCreate$outboundSchema.optional(),
  jointTenantsInCommonEnrollmentMetadata:
    JointTenantsInCommonEnrollmentMetadataCreate$outboundSchema.optional(),
  jointWithRightsOfSurvivorshipEnrollmentMetadata:
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$outboundSchema
      .optional(),
  llcEnrollmentMetadata: LLCEnrollmentMetadataCreate$outboundSchema.optional(),
  operatingEnrollmentMetadata: OperatingEnrollmentMetadataCreate$outboundSchema
    .optional(),
  ordersOptionsTradingEnrollmentMetadata:
    OrdersOptionsTradingEnrollmentMetadataCreate$outboundSchema.optional(),
  principalApproverId: z.string(),
  trustEnrollmentMetadata: TrustEnrollmentMetadataCreate$outboundSchema
    .optional(),
  type: EnrollmentCreateType$outboundSchema,
  virtualAccountNumberEnrollmentMetadata:
    VirtualAccountNumberEnrollmentMetadataCreate$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    beneficiaryEnrollmentMetadata: "beneficiary_enrollment_metadata",
    consentMethod: "consent_method",
    corporationEnrollmentMetadata: "corporation_enrollment_metadata",
    custodialEnrollmentMetadata: "custodial_enrollment_metadata",
    estateEnrollmentMetadata: "estate_enrollment_metadata",
    foreignIndividualAccountEnrollmentMetadata:
      "foreign_individual_account_enrollment_metadata",
    foreignJointAccountEnrollmentMetadata:
      "foreign_joint_account_enrollment_metadata",
    fpslEnrollmentMetadata: "fpsl_enrollment_metadata",
    individualEnrollmentMetadata: "individual_enrollment_metadata",
    iraBeneficiaryEnrollmentMetadata: "ira_beneficiary_enrollment_metadata",
    iraRolloverEnrollmentMetadata: "ira_rollover_enrollment_metadata",
    iraRothEnrollmentMetadata: "ira_roth_enrollment_metadata",
    iraSepEnrollmentMetadata: "ira_sep_enrollment_metadata",
    iraSimpleEnrollmentMetadata: "ira_simple_enrollment_metadata",
    iraTraditionalEnrollmentMetadata: "ira_traditional_enrollment_metadata",
    jointCommunityPropertyEnrollmentMetadata:
      "joint_community_property_enrollment_metadata",
    jointTenantsByEntiretyEnrollmentMetadata:
      "joint_tenants_by_entirety_enrollment_metadata",
    jointTenantsInCommonEnrollmentMetadata:
      "joint_tenants_in_common_enrollment_metadata",
    jointWithRightsOfSurvivorshipEnrollmentMetadata:
      "joint_with_rights_of_survivorship_enrollment_metadata",
    llcEnrollmentMetadata: "llc_enrollment_metadata",
    operatingEnrollmentMetadata: "operating_enrollment_metadata",
    ordersOptionsTradingEnrollmentMetadata:
      "orders_options_trading_enrollment_metadata",
    principalApproverId: "principal_approver_id",
    trustEnrollmentMetadata: "trust_enrollment_metadata",
    virtualAccountNumberEnrollmentMetadata:
      "virtual_account_number_enrollment_metadata",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace EnrollmentCreate$ {
  /** @deprecated use `EnrollmentCreate$inboundSchema` instead. */
  export const inboundSchema = EnrollmentCreate$inboundSchema;
  /** @deprecated use `EnrollmentCreate$outboundSchema` instead. */
  export const outboundSchema = EnrollmentCreate$outboundSchema;
  /** @deprecated use `EnrollmentCreate$Outbound` instead. */
  export type Outbound = EnrollmentCreate$Outbound;
}
