// @apexfintechsolutions/ascend-sdk/src/models/components/feecreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";
import {
  DecimalCreate,
  DecimalCreate$inboundSchema,
  DecimalCreate$Outbound,
  DecimalCreate$outboundSchema,
} from "./decimalcreate.js";

/**
 * The type of fee
 */
export enum FeeCreateType {
  FeeTypeUnspecified = "FEE_TYPE_UNSPECIFIED",
  ClientClearing = "CLIENT_CLEARING",
  Liquidity = "LIQUIDITY",
  TradeActivity = "TRADE_ACTIVITY",
  FinancialTransactionTax = "FINANCIAL_TRANSACTION_TAX",
  IndexOptionFee = "INDEX_OPTION_FEE",
  SecFee = "SEC_FEE",
  OptionsRegulatory = "OPTIONS_REGULATORY",
  GeneralPurposeFee = "GENERAL_PURPOSE_FEE",
  BrokerFee = "BROKER_FEE",
  ContractFee = "CONTRACT_FEE",
  OccFee = "OCC_FEE",
}
/**
 * The type of fee
 */
export type FeeCreateTypeOpen = OpenEnum<typeof FeeCreateType>;

/**
 * Fee message includes both fee type as well as the fee amount
 */
export type FeeCreate = {
  /**
   * A representation of a decimal value, such as 2.5. Clients may convert values into language-native decimal formats, such as Java's [BigDecimal][] or Python's [decimal.Decimal][].
   *
   * @remarks
   *
   *  [BigDecimal]:
   *  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html
   *  [decimal.Decimal]: https://docs.python.org/3/library/decimal.html
   */
  amount?: DecimalCreate | undefined;
  /**
   * Indicates whether to explicitly suppress this fee type. If the trade would normally calculate fees (e.g., for TRADE_ACTIVITY), the client can add a fee with this boolean value set to true, and the Booking Service will not calculate or assess that fee on the trade.
   */
  suppressFee?: boolean | undefined;
  /**
   * The type of fee
   */
  type?: FeeCreateTypeOpen | undefined;
};

/** @internal */
export const FeeCreateType$inboundSchema: z.ZodType<
  FeeCreateTypeOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(FeeCreateType),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const FeeCreateType$outboundSchema: z.ZodType<
  FeeCreateTypeOpen,
  z.ZodTypeDef,
  FeeCreateTypeOpen
> = z.union([
  z.nativeEnum(FeeCreateType),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace FeeCreateType$ {
  /** @deprecated use `FeeCreateType$inboundSchema` instead. */
  export const inboundSchema = FeeCreateType$inboundSchema;
  /** @deprecated use `FeeCreateType$outboundSchema` instead. */
  export const outboundSchema = FeeCreateType$outboundSchema;
}

/** @internal */
export const FeeCreate$inboundSchema: z.ZodType<
  FeeCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  amount: DecimalCreate$inboundSchema.optional(),
  suppress_fee: z.boolean().optional(),
  type: FeeCreateType$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "suppress_fee": "suppressFee",
  });
});

/** @internal */
export type FeeCreate$Outbound = {
  amount?: DecimalCreate$Outbound | undefined;
  suppress_fee?: boolean | undefined;
  type?: string | undefined;
};

/** @internal */
export const FeeCreate$outboundSchema: z.ZodType<
  FeeCreate$Outbound,
  z.ZodTypeDef,
  FeeCreate
> = z.object({
  amount: DecimalCreate$outboundSchema.optional(),
  suppressFee: z.boolean().optional(),
  type: FeeCreateType$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    suppressFee: "suppress_fee",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace FeeCreate$ {
  /** @deprecated use `FeeCreate$inboundSchema` instead. */
  export const inboundSchema = FeeCreate$inboundSchema;
  /** @deprecated use `FeeCreate$outboundSchema` instead. */
  export const outboundSchema = FeeCreate$outboundSchema;
  /** @deprecated use `FeeCreate$Outbound` instead. */
  export type Outbound = FeeCreate$Outbound;
}
