// @apexfintechsolutions/ascend-sdk/src/models/components/forceapproveictwithdrawalrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

/**
 * Request to simulate the approval of an ICT withdrawal
 */
export type ForceApproveIctWithdrawalRequestCreate = {
  /**
   * Full name of the ICT withdrawal resource to force a rejection on. Contains account id and ICT withdrawal id.
   */
  name: string;
};

/** @internal */
export const ForceApproveIctWithdrawalRequestCreate$inboundSchema: z.ZodType<
  ForceApproveIctWithdrawalRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
});

/** @internal */
export type ForceApproveIctWithdrawalRequestCreate$Outbound = {
  name: string;
};

/** @internal */
export const ForceApproveIctWithdrawalRequestCreate$outboundSchema: z.ZodType<
  ForceApproveIctWithdrawalRequestCreate$Outbound,
  z.ZodTypeDef,
  ForceApproveIctWithdrawalRequestCreate
> = z.object({
  name: z.string(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForceApproveIctWithdrawalRequestCreate$ {
  /** @deprecated use `ForceApproveIctWithdrawalRequestCreate$inboundSchema` instead. */
  export const inboundSchema =
    ForceApproveIctWithdrawalRequestCreate$inboundSchema;
  /** @deprecated use `ForceApproveIctWithdrawalRequestCreate$outboundSchema` instead. */
  export const outboundSchema =
    ForceApproveIctWithdrawalRequestCreate$outboundSchema;
  /** @deprecated use `ForceApproveIctWithdrawalRequestCreate$Outbound` instead. */
  export type Outbound = ForceApproveIctWithdrawalRequestCreate$Outbound;
}
