// @apexfintechsolutions/ascend-sdk/src/models/components/forcerejectcashjournalrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

/**
 * Request to force reject a pending cash journal
 */
export type ForceRejectCashJournalRequestCreate = {
  /**
   * The name of the cash journal to force reject
   */
  name: string;
  /**
   * The optional reason for force rejecting a cash journal
   */
  reason?: string | undefined;
};

/** @internal */
export const ForceRejectCashJournalRequestCreate$inboundSchema: z.ZodType<
  ForceRejectCashJournalRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
  reason: z.string().optional(),
});

/** @internal */
export type ForceRejectCashJournalRequestCreate$Outbound = {
  name: string;
  reason?: string | undefined;
};

/** @internal */
export const ForceRejectCashJournalRequestCreate$outboundSchema: z.ZodType<
  ForceRejectCashJournalRequestCreate$Outbound,
  z.ZodTypeDef,
  ForceRejectCashJournalRequestCreate
> = z.object({
  name: z.string(),
  reason: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForceRejectCashJournalRequestCreate$ {
  /** @deprecated use `ForceRejectCashJournalRequestCreate$inboundSchema` instead. */
  export const inboundSchema =
    ForceRejectCashJournalRequestCreate$inboundSchema;
  /** @deprecated use `ForceRejectCashJournalRequestCreate$outboundSchema` instead. */
  export const outboundSchema =
    ForceRejectCashJournalRequestCreate$outboundSchema;
  /** @deprecated use `ForceRejectCashJournalRequestCreate$Outbound` instead. */
  export type Outbound = ForceRejectCashJournalRequestCreate$Outbound;
}
