// @apexfintechsolutions/ascend-sdk/src/models/components/forcereturnachwithdrawalrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  NachaReturnCreate,
  NachaReturnCreate$inboundSchema,
  NachaReturnCreate$Outbound,
  NachaReturnCreate$outboundSchema,
} from "./nachareturncreate.js";

/**
 * Request to force a Nacha return on a completed ACH withdrawal. FOR TESTING ONLY!
 */
export type ForceReturnAchWithdrawalRequestCreate = {
  /**
   * A return on an ACH transfer from the Nacha network.
   */
  nachaReturn: NachaReturnCreate;
  /**
   * The name of the ACH withdrawal to force a Nacha return on.
   */
  name: string;
};

/** @internal */
export const ForceReturnAchWithdrawalRequestCreate$inboundSchema: z.ZodType<
  ForceReturnAchWithdrawalRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  nacha_return: NachaReturnCreate$inboundSchema,
  name: z.string(),
}).transform((v) => {
  return remap$(v, {
    "nacha_return": "nachaReturn",
  });
});

/** @internal */
export type ForceReturnAchWithdrawalRequestCreate$Outbound = {
  nacha_return: NachaReturnCreate$Outbound;
  name: string;
};

/** @internal */
export const ForceReturnAchWithdrawalRequestCreate$outboundSchema: z.ZodType<
  ForceReturnAchWithdrawalRequestCreate$Outbound,
  z.ZodTypeDef,
  ForceReturnAchWithdrawalRequestCreate
> = z.object({
  nachaReturn: NachaReturnCreate$outboundSchema,
  name: z.string(),
}).transform((v) => {
  return remap$(v, {
    nachaReturn: "nacha_return",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForceReturnAchWithdrawalRequestCreate$ {
  /** @deprecated use `ForceReturnAchWithdrawalRequestCreate$inboundSchema` instead. */
  export const inboundSchema =
    ForceReturnAchWithdrawalRequestCreate$inboundSchema;
  /** @deprecated use `ForceReturnAchWithdrawalRequestCreate$outboundSchema` instead. */
  export const outboundSchema =
    ForceReturnAchWithdrawalRequestCreate$outboundSchema;
  /** @deprecated use `ForceReturnAchWithdrawalRequestCreate$Outbound` instead. */
  export type Outbound = ForceReturnAchWithdrawalRequestCreate$Outbound;
}
