// @apexfintechsolutions/ascend-sdk/src/models/components/forcerejectictdepositrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

/**
 * Request to simulate the rejection of an ICT deposit
 */
export type ForceRejectIctDepositRequestCreate = {
  /**
   * Full name of the ICT deposit resource to force a rejection on. Contains account id and ICT deposit id.
   */
  name: string;
  /**
   * Reason why the ICT deposit is being rejected
   */
  reason?: string | undefined;
};

/** @internal */
export const ForceRejectIctDepositRequestCreate$inboundSchema: z.ZodType<
  ForceRejectIctDepositRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  name: z.string(),
  reason: z.string().optional(),
});

/** @internal */
export type ForceRejectIctDepositRequestCreate$Outbound = {
  name: string;
  reason?: string | undefined;
};

/** @internal */
export const ForceRejectIctDepositRequestCreate$outboundSchema: z.ZodType<
  ForceRejectIctDepositRequestCreate$Outbound,
  z.ZodTypeDef,
  ForceRejectIctDepositRequestCreate
> = z.object({
  name: z.string(),
  reason: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForceRejectIctDepositRequestCreate$ {
  /** @deprecated use `ForceRejectIctDepositRequestCreate$inboundSchema` instead. */
  export const inboundSchema = ForceRejectIctDepositRequestCreate$inboundSchema;
  /** @deprecated use `ForceRejectIctDepositRequestCreate$outboundSchema` instead. */
  export const outboundSchema =
    ForceRejectIctDepositRequestCreate$outboundSchema;
  /** @deprecated use `ForceRejectIctDepositRequestCreate$Outbound` instead. */
  export type Outbound = ForceRejectIctDepositRequestCreate$Outbound;
}
