// @apexfintechsolutions/ascend-sdk/src/models/components/forcenocachdepositrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  NachaNocCreate,
  NachaNocCreate$inboundSchema,
  NachaNocCreate$Outbound,
  NachaNocCreate$outboundSchema,
} from "./nachanoccreate.js";

/**
 * Request to force a Nacha notice of change (NOC) on a completed ACH deposit. FOR TESTING ONLY!
 */
export type ForceNocAchDepositRequestCreate = {
  /**
   * A notice of change (NOC) on an ACH transfer from the Nacha network.
   */
  nachaNoc: NachaNocCreate;
  /**
   * The name of the ACH deposit to force a Nacha NOC on.
   */
  name: string;
};

/** @internal */
export const ForceNocAchDepositRequestCreate$inboundSchema: z.ZodType<
  ForceNocAchDepositRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  nacha_noc: NachaNocCreate$inboundSchema,
  name: z.string(),
}).transform((v) => {
  return remap$(v, {
    "nacha_noc": "nachaNoc",
  });
});

/** @internal */
export type ForceNocAchDepositRequestCreate$Outbound = {
  nacha_noc: NachaNocCreate$Outbound;
  name: string;
};

/** @internal */
export const ForceNocAchDepositRequestCreate$outboundSchema: z.ZodType<
  ForceNocAchDepositRequestCreate$Outbound,
  z.ZodTypeDef,
  ForceNocAchDepositRequestCreate
> = z.object({
  nachaNoc: NachaNocCreate$outboundSchema,
  name: z.string(),
}).transform((v) => {
  return remap$(v, {
    nachaNoc: "nacha_noc",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForceNocAchDepositRequestCreate$ {
  /** @deprecated use `ForceNocAchDepositRequestCreate$inboundSchema` instead. */
  export const inboundSchema = ForceNocAchDepositRequestCreate$inboundSchema;
  /** @deprecated use `ForceNocAchDepositRequestCreate$outboundSchema` instead. */
  export const outboundSchema = ForceNocAchDepositRequestCreate$outboundSchema;
  /** @deprecated use `ForceNocAchDepositRequestCreate$Outbound` instead. */
  export type Outbound = ForceNocAchDepositRequestCreate$Outbound;
}
