// @apexfintechsolutions/ascend-sdk/src/models/components/foreignbondtradingdetailcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  DecimalCreate,
  DecimalCreate$inboundSchema,
  DecimalCreate$Outbound,
  DecimalCreate$outboundSchema,
} from "./decimalcreate.js";

/**
 * Foreign bond trading detail
 */
export type ForeignBondTradingDetailCreate = {
  /**
   * A representation of a decimal value, such as 2.5. Clients may convert values into language-native decimal formats, such as Java's [BigDecimal][] or Python's [decimal.Decimal][].
   *
   * @remarks
   *
   *  [BigDecimal]:
   *  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html
   *  [decimal.Decimal]: https://docs.python.org/3/library/decimal.html
   */
  percentage: DecimalCreate;
  /**
   * The region where the foreign bond trading activity is taking place. Must be a two-character CLDR code.
   */
  regionCode: string;
};

/** @internal */
export const ForeignBondTradingDetailCreate$inboundSchema: z.ZodType<
  ForeignBondTradingDetailCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  percentage: DecimalCreate$inboundSchema,
  region_code: z.string(),
}).transform((v) => {
  return remap$(v, {
    "region_code": "regionCode",
  });
});

/** @internal */
export type ForeignBondTradingDetailCreate$Outbound = {
  percentage: DecimalCreate$Outbound;
  region_code: string;
};

/** @internal */
export const ForeignBondTradingDetailCreate$outboundSchema: z.ZodType<
  ForeignBondTradingDetailCreate$Outbound,
  z.ZodTypeDef,
  ForeignBondTradingDetailCreate
> = z.object({
  percentage: DecimalCreate$outboundSchema,
  regionCode: z.string(),
}).transform((v) => {
  return remap$(v, {
    regionCode: "region_code",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForeignBondTradingDetailCreate$ {
  /** @deprecated use `ForeignBondTradingDetailCreate$inboundSchema` instead. */
  export const inboundSchema = ForeignBondTradingDetailCreate$inboundSchema;
  /** @deprecated use `ForeignBondTradingDetailCreate$outboundSchema` instead. */
  export const outboundSchema = ForeignBondTradingDetailCreate$outboundSchema;
  /** @deprecated use `ForeignBondTradingDetailCreate$Outbound` instead. */
  export type Outbound = ForeignBondTradingDetailCreate$Outbound;
}
