// @apexfintechsolutions/ascend-sdk/src/models/components/foreignindividualaccountenrollmentmetadatacreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";
import {
  ForeignNaturalPersonAccountEnrollmentMetadataCreate,
  ForeignNaturalPersonAccountEnrollmentMetadataCreate$inboundSchema,
  ForeignNaturalPersonAccountEnrollmentMetadataCreate$Outbound,
  ForeignNaturalPersonAccountEnrollmentMetadataCreate$outboundSchema,
} from "./foreignnaturalpersonaccountenrollmentmetadatacreate.js";

/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export enum ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan {
  AutoEnrollDividendReinvestmentUnspecified =
    "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED",
  DividendReinvestmentEnroll = "DIVIDEND_REINVESTMENT_ENROLL",
  DividendReinvestmentDecline = "DIVIDEND_REINVESTMENT_DECLINE",
}
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export type ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlanOpen =
  OpenEnum<
    typeof ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan
  >;

/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export enum ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep {
  AutoEnrollFdicCashSweepUnspecified =
    "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED",
  FdicCashSweepEnroll = "FDIC_CASH_SWEEP_ENROLL",
  FdicCashSweepDecline = "FDIC_CASH_SWEEP_DECLINE",
}
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export type ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweepOpen =
  OpenEnum<
    typeof ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep
  >;

export type ForeignIndividualAccountEnrollmentMetadataCreate = {
  /**
   * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
   */
  dividendReinvestmentPlan?:
    | ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlanOpen
    | undefined;
  /**
   * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
   */
  fdicCashSweep?:
    | ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweepOpen
    | undefined;
  /**
   * Enrollment metadata for Accounts that have a foreign Legal Natural Person owner.
   */
  foreignNaturalPersonAccountEnrollmentMetadata:
    ForeignNaturalPersonAccountEnrollmentMetadataCreate;
};

/** @internal */
export const ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema:
  z.ZodType<
    ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(
        ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan,
      ),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema:
  z.ZodType<
    ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlanOpen
  > = z.union([
    z.nativeEnum(
      ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan,
    ),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$ {
  /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
  export const inboundSchema =
    ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
  /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
  export const outboundSchema =
    ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
}

/** @internal */
export const ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema:
  z.ZodType<
    ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(
        ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep,
      ),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema:
  z.ZodType<
    ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweepOpen
  > = z.union([
    z.nativeEnum(ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$ {
  /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
  export const inboundSchema =
    ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
  /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
  export const outboundSchema =
    ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
}

/** @internal */
export const ForeignIndividualAccountEnrollmentMetadataCreate$inboundSchema:
  z.ZodType<
    ForeignIndividualAccountEnrollmentMetadataCreate,
    z.ZodTypeDef,
    unknown
  > = z.object({
    dividend_reinvestment_plan:
      ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep:
      ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
    foreign_natural_person_account_enrollment_metadata:
      ForeignNaturalPersonAccountEnrollmentMetadataCreate$inboundSchema,
  }).transform((v) => {
    return remap$(v, {
      "dividend_reinvestment_plan": "dividendReinvestmentPlan",
      "fdic_cash_sweep": "fdicCashSweep",
      "foreign_natural_person_account_enrollment_metadata":
        "foreignNaturalPersonAccountEnrollmentMetadata",
    });
  });

/** @internal */
export type ForeignIndividualAccountEnrollmentMetadataCreate$Outbound = {
  dividend_reinvestment_plan?: string | undefined;
  fdic_cash_sweep?: string | undefined;
  foreign_natural_person_account_enrollment_metadata:
    ForeignNaturalPersonAccountEnrollmentMetadataCreate$Outbound;
};

/** @internal */
export const ForeignIndividualAccountEnrollmentMetadataCreate$outboundSchema:
  z.ZodType<
    ForeignIndividualAccountEnrollmentMetadataCreate$Outbound,
    z.ZodTypeDef,
    ForeignIndividualAccountEnrollmentMetadataCreate
  > = z.object({
    dividendReinvestmentPlan:
      ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep:
      ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
    foreignNaturalPersonAccountEnrollmentMetadata:
      ForeignNaturalPersonAccountEnrollmentMetadataCreate$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      dividendReinvestmentPlan: "dividend_reinvestment_plan",
      fdicCashSweep: "fdic_cash_sweep",
      foreignNaturalPersonAccountEnrollmentMetadata:
        "foreign_natural_person_account_enrollment_metadata",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ForeignIndividualAccountEnrollmentMetadataCreate$ {
  /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreate$inboundSchema` instead. */
  export const inboundSchema =
    ForeignIndividualAccountEnrollmentMetadataCreate$inboundSchema;
  /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreate$outboundSchema` instead. */
  export const outboundSchema =
    ForeignIndividualAccountEnrollmentMetadataCreate$outboundSchema;
  /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreate$Outbound` instead. */
  export type Outbound =
    ForeignIndividualAccountEnrollmentMetadataCreate$Outbound;
}
