// @apexfintechsolutions/ascend-sdk/src/models/components/fpslenrollmentmetadatacreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

/**
 * Percentages for FPSL Enrollment, must equal 100
 */
export type FPSLEnrollmentMetaDataCreate = {};

/** @internal */
export const FPSLEnrollmentMetaDataCreate$inboundSchema: z.ZodType<
  FPSLEnrollmentMetaDataCreate,
  z.ZodTypeDef,
  unknown
> = z.object({});

/** @internal */
export type FPSLEnrollmentMetaDataCreate$Outbound = {};

/** @internal */
export const FPSLEnrollmentMetaDataCreate$outboundSchema: z.ZodType<
  FPSLEnrollmentMetaDataCreate$Outbound,
  z.ZodTypeDef,
  FPSLEnrollmentMetaDataCreate
> = z.object({});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace FPSLEnrollmentMetaDataCreate$ {
  /** @deprecated use `FPSLEnrollmentMetaDataCreate$inboundSchema` instead. */
  export const inboundSchema = FPSLEnrollmentMetaDataCreate$inboundSchema;
  /** @deprecated use `FPSLEnrollmentMetaDataCreate$outboundSchema` instead. */
  export const outboundSchema = FPSLEnrollmentMetaDataCreate$outboundSchema;
  /** @deprecated use `FPSLEnrollmentMetaDataCreate$Outbound` instead. */
  export type Outbound = FPSLEnrollmentMetaDataCreate$Outbound;
}
