// @apexfintechsolutions/ascend-sdk/src/models/components/individualenrollmentmetadatacreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";

/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export enum IndividualEnrollmentMetadataCreateDividendReinvestmentPlan {
  AutoEnrollDividendReinvestmentUnspecified =
    "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED",
  DividendReinvestmentEnroll = "DIVIDEND_REINVESTMENT_ENROLL",
  DividendReinvestmentDecline = "DIVIDEND_REINVESTMENT_DECLINE",
}
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export type IndividualEnrollmentMetadataCreateDividendReinvestmentPlanOpen =
  OpenEnum<typeof IndividualEnrollmentMetadataCreateDividendReinvestmentPlan>;

/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export enum IndividualEnrollmentMetadataCreateFdicCashSweep {
  AutoEnrollFdicCashSweepUnspecified =
    "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED",
  FdicCashSweepEnroll = "FDIC_CASH_SWEEP_ENROLL",
  FdicCashSweepDecline = "FDIC_CASH_SWEEP_DECLINE",
}
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export type IndividualEnrollmentMetadataCreateFdicCashSweepOpen = OpenEnum<
  typeof IndividualEnrollmentMetadataCreateFdicCashSweep
>;

/**
 * Enrollment metadata for Individual accounts enrollment type
 */
export type IndividualEnrollmentMetadataCreate = {
  /**
   * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
   */
  dividendReinvestmentPlan?:
    | IndividualEnrollmentMetadataCreateDividendReinvestmentPlanOpen
    | undefined;
  /**
   * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
   */
  fdicCashSweep?:
    | IndividualEnrollmentMetadataCreateFdicCashSweepOpen
    | undefined;
};

/** @internal */
export const IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema:
  z.ZodType<
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(IndividualEnrollmentMetadataCreateDividendReinvestmentPlan),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema:
  z.ZodType<
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlanOpen
  > = z.union([
    z.nativeEnum(IndividualEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$ {
  /** @deprecated use `IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
  export const inboundSchema =
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
  /** @deprecated use `IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
  export const outboundSchema =
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
}

/** @internal */
export const IndividualEnrollmentMetadataCreateFdicCashSweep$inboundSchema:
  z.ZodType<
    IndividualEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(IndividualEnrollmentMetadataCreateFdicCashSweep),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const IndividualEnrollmentMetadataCreateFdicCashSweep$outboundSchema:
  z.ZodType<
    IndividualEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    IndividualEnrollmentMetadataCreateFdicCashSweepOpen
  > = z.union([
    z.nativeEnum(IndividualEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IndividualEnrollmentMetadataCreateFdicCashSweep$ {
  /** @deprecated use `IndividualEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
  export const inboundSchema =
    IndividualEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
  /** @deprecated use `IndividualEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
  export const outboundSchema =
    IndividualEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
}

/** @internal */
export const IndividualEnrollmentMetadataCreate$inboundSchema: z.ZodType<
  IndividualEnrollmentMetadataCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  dividend_reinvestment_plan:
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
      .optional(),
  fdic_cash_sweep: IndividualEnrollmentMetadataCreateFdicCashSweep$inboundSchema
    .optional(),
}).transform((v) => {
  return remap$(v, {
    "dividend_reinvestment_plan": "dividendReinvestmentPlan",
    "fdic_cash_sweep": "fdicCashSweep",
  });
});

/** @internal */
export type IndividualEnrollmentMetadataCreate$Outbound = {
  dividend_reinvestment_plan?: string | undefined;
  fdic_cash_sweep?: string | undefined;
};

/** @internal */
export const IndividualEnrollmentMetadataCreate$outboundSchema: z.ZodType<
  IndividualEnrollmentMetadataCreate$Outbound,
  z.ZodTypeDef,
  IndividualEnrollmentMetadataCreate
> = z.object({
  dividendReinvestmentPlan:
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
      .optional(),
  fdicCashSweep: IndividualEnrollmentMetadataCreateFdicCashSweep$outboundSchema
    .optional(),
}).transform((v) => {
  return remap$(v, {
    dividendReinvestmentPlan: "dividend_reinvestment_plan",
    fdicCashSweep: "fdic_cash_sweep",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IndividualEnrollmentMetadataCreate$ {
  /** @deprecated use `IndividualEnrollmentMetadataCreate$inboundSchema` instead. */
  export const inboundSchema = IndividualEnrollmentMetadataCreate$inboundSchema;
  /** @deprecated use `IndividualEnrollmentMetadataCreate$outboundSchema` instead. */
  export const outboundSchema =
    IndividualEnrollmentMetadataCreate$outboundSchema;
  /** @deprecated use `IndividualEnrollmentMetadataCreate$Outbound` instead. */
  export type Outbound = IndividualEnrollmentMetadataCreate$Outbound;
}
