// @apexfintechsolutions/ascend-sdk/src/models/components/investmentvehicleexperiencecreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";

/**
 * Investment vehicle experience
 */
export type InvestmentVehicleExperienceCreate = {
  /**
   * The number of average annual trades of commodities
   */
  commoditiesAverageAnnualTradeCount: number;
  /**
   * The number of average annual trades of corporate bonds
   */
  corporateBondsAverageAnnualTradeCount: number;
  /**
   * The number of average annual trades of funds
   */
  fundsAverageAnnualTradeCount: number;
  /**
   * The number of average annual trades of government bonds
   */
  governmentBondsAverageAnnualTradeCount: number;
  /**
   * The number of average annual trades in margin accounts
   */
  marginAverageAnnualTradeCount: number;
  /**
   * The number of average annual trades of municipal bonds
   */
  municipalBondsAverageAnnualTradeCount: number;
  /**
   * The number of average annual trades of options
   */
  optionsAverageAnnualTradeCount: number;
  /**
   * The number of average annual trades of stocks
   */
  stocksAverageAnnualTradeCount: number;
  /**
   * The number of average annual trades in tax sheltered accounts
   */
  taxSheltersAverageAnnualTradeCount: number;
};

/** @internal */
export const InvestmentVehicleExperienceCreate$inboundSchema: z.ZodType<
  InvestmentVehicleExperienceCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  commodities_average_annual_trade_count: z.number().int(),
  corporate_bonds_average_annual_trade_count: z.number().int(),
  funds_average_annual_trade_count: z.number().int(),
  government_bonds_average_annual_trade_count: z.number().int(),
  margin_average_annual_trade_count: z.number().int(),
  municipal_bonds_average_annual_trade_count: z.number().int(),
  options_average_annual_trade_count: z.number().int(),
  stocks_average_annual_trade_count: z.number().int(),
  tax_shelters_average_annual_trade_count: z.number().int(),
}).transform((v) => {
  return remap$(v, {
    "commodities_average_annual_trade_count":
      "commoditiesAverageAnnualTradeCount",
    "corporate_bonds_average_annual_trade_count":
      "corporateBondsAverageAnnualTradeCount",
    "funds_average_annual_trade_count": "fundsAverageAnnualTradeCount",
    "government_bonds_average_annual_trade_count":
      "governmentBondsAverageAnnualTradeCount",
    "margin_average_annual_trade_count": "marginAverageAnnualTradeCount",
    "municipal_bonds_average_annual_trade_count":
      "municipalBondsAverageAnnualTradeCount",
    "options_average_annual_trade_count": "optionsAverageAnnualTradeCount",
    "stocks_average_annual_trade_count": "stocksAverageAnnualTradeCount",
    "tax_shelters_average_annual_trade_count":
      "taxSheltersAverageAnnualTradeCount",
  });
});

/** @internal */
export type InvestmentVehicleExperienceCreate$Outbound = {
  commodities_average_annual_trade_count: number;
  corporate_bonds_average_annual_trade_count: number;
  funds_average_annual_trade_count: number;
  government_bonds_average_annual_trade_count: number;
  margin_average_annual_trade_count: number;
  municipal_bonds_average_annual_trade_count: number;
  options_average_annual_trade_count: number;
  stocks_average_annual_trade_count: number;
  tax_shelters_average_annual_trade_count: number;
};

/** @internal */
export const InvestmentVehicleExperienceCreate$outboundSchema: z.ZodType<
  InvestmentVehicleExperienceCreate$Outbound,
  z.ZodTypeDef,
  InvestmentVehicleExperienceCreate
> = z.object({
  commoditiesAverageAnnualTradeCount: z.number().int(),
  corporateBondsAverageAnnualTradeCount: z.number().int(),
  fundsAverageAnnualTradeCount: z.number().int(),
  governmentBondsAverageAnnualTradeCount: z.number().int(),
  marginAverageAnnualTradeCount: z.number().int(),
  municipalBondsAverageAnnualTradeCount: z.number().int(),
  optionsAverageAnnualTradeCount: z.number().int(),
  stocksAverageAnnualTradeCount: z.number().int(),
  taxSheltersAverageAnnualTradeCount: z.number().int(),
}).transform((v) => {
  return remap$(v, {
    commoditiesAverageAnnualTradeCount:
      "commodities_average_annual_trade_count",
    corporateBondsAverageAnnualTradeCount:
      "corporate_bonds_average_annual_trade_count",
    fundsAverageAnnualTradeCount: "funds_average_annual_trade_count",
    governmentBondsAverageAnnualTradeCount:
      "government_bonds_average_annual_trade_count",
    marginAverageAnnualTradeCount: "margin_average_annual_trade_count",
    municipalBondsAverageAnnualTradeCount:
      "municipal_bonds_average_annual_trade_count",
    optionsAverageAnnualTradeCount: "options_average_annual_trade_count",
    stocksAverageAnnualTradeCount: "stocks_average_annual_trade_count",
    taxSheltersAverageAnnualTradeCount:
      "tax_shelters_average_annual_trade_count",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace InvestmentVehicleExperienceCreate$ {
  /** @deprecated use `InvestmentVehicleExperienceCreate$inboundSchema` instead. */
  export const inboundSchema = InvestmentVehicleExperienceCreate$inboundSchema;
  /** @deprecated use `InvestmentVehicleExperienceCreate$outboundSchema` instead. */
  export const outboundSchema =
    InvestmentVehicleExperienceCreate$outboundSchema;
  /** @deprecated use `InvestmentVehicleExperienceCreate$Outbound` instead. */
  export type Outbound = InvestmentVehicleExperienceCreate$Outbound;
}
