// @apexfintechsolutions/ascend-sdk/src/models/components/irarolloverenrollmentmetadatacreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";

/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export enum IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan {
  AutoEnrollDividendReinvestmentUnspecified =
    "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED",
  DividendReinvestmentEnroll = "DIVIDEND_REINVESTMENT_ENROLL",
  DividendReinvestmentDecline = "DIVIDEND_REINVESTMENT_DECLINE",
}
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export type IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlanOpen =
  OpenEnum<typeof IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan>;

/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export enum IRARolloverEnrollmentMetadataCreateFdicCashSweep {
  AutoEnrollFdicCashSweepUnspecified =
    "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED",
  FdicCashSweepEnroll = "FDIC_CASH_SWEEP_ENROLL",
  FdicCashSweepDecline = "FDIC_CASH_SWEEP_DECLINE",
}
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export type IRARolloverEnrollmentMetadataCreateFdicCashSweepOpen = OpenEnum<
  typeof IRARolloverEnrollmentMetadataCreateFdicCashSweep
>;

/**
 * Enrollment metadata for Rollover IRA accounts enrollment type
 */
export type IRARolloverEnrollmentMetadataCreate = {
  /**
   * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
   */
  dividendReinvestmentPlan?:
    | IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlanOpen
    | undefined;
  /**
   * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
   */
  fdicCashSweep?:
    | IRARolloverEnrollmentMetadataCreateFdicCashSweepOpen
    | undefined;
};

/** @internal */
export const IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema:
  z.ZodType<
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema:
  z.ZodType<
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlanOpen
  > = z.union([
    z.nativeEnum(IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$ {
  /** @deprecated use `IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
  export const inboundSchema =
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
  /** @deprecated use `IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
  export const outboundSchema =
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
}

/** @internal */
export const IRARolloverEnrollmentMetadataCreateFdicCashSweep$inboundSchema:
  z.ZodType<
    IRARolloverEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(IRARolloverEnrollmentMetadataCreateFdicCashSweep),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const IRARolloverEnrollmentMetadataCreateFdicCashSweep$outboundSchema:
  z.ZodType<
    IRARolloverEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    IRARolloverEnrollmentMetadataCreateFdicCashSweepOpen
  > = z.union([
    z.nativeEnum(IRARolloverEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IRARolloverEnrollmentMetadataCreateFdicCashSweep$ {
  /** @deprecated use `IRARolloverEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
  export const inboundSchema =
    IRARolloverEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
  /** @deprecated use `IRARolloverEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
  export const outboundSchema =
    IRARolloverEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
}

/** @internal */
export const IRARolloverEnrollmentMetadataCreate$inboundSchema: z.ZodType<
  IRARolloverEnrollmentMetadataCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  dividend_reinvestment_plan:
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
      .optional(),
  fdic_cash_sweep:
    IRARolloverEnrollmentMetadataCreateFdicCashSweep$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "dividend_reinvestment_plan": "dividendReinvestmentPlan",
    "fdic_cash_sweep": "fdicCashSweep",
  });
});

/** @internal */
export type IRARolloverEnrollmentMetadataCreate$Outbound = {
  dividend_reinvestment_plan?: string | undefined;
  fdic_cash_sweep?: string | undefined;
};

/** @internal */
export const IRARolloverEnrollmentMetadataCreate$outboundSchema: z.ZodType<
  IRARolloverEnrollmentMetadataCreate$Outbound,
  z.ZodTypeDef,
  IRARolloverEnrollmentMetadataCreate
> = z.object({
  dividendReinvestmentPlan:
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
      .optional(),
  fdicCashSweep: IRARolloverEnrollmentMetadataCreateFdicCashSweep$outboundSchema
    .optional(),
}).transform((v) => {
  return remap$(v, {
    dividendReinvestmentPlan: "dividend_reinvestment_plan",
    fdicCashSweep: "fdic_cash_sweep",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IRARolloverEnrollmentMetadataCreate$ {
  /** @deprecated use `IRARolloverEnrollmentMetadataCreate$inboundSchema` instead. */
  export const inboundSchema =
    IRARolloverEnrollmentMetadataCreate$inboundSchema;
  /** @deprecated use `IRARolloverEnrollmentMetadataCreate$outboundSchema` instead. */
  export const outboundSchema =
    IRARolloverEnrollmentMetadataCreate$outboundSchema;
  /** @deprecated use `IRARolloverEnrollmentMetadataCreate$Outbound` instead. */
  export type Outbound = IRARolloverEnrollmentMetadataCreate$Outbound;
}
