// @apexfintechsolutions/ascend-sdk/src/models/components/iratraditionalenrollmentmetadatacreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";

/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export enum IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan {
  AutoEnrollDividendReinvestmentUnspecified =
    "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED",
  DividendReinvestmentEnroll = "DIVIDEND_REINVESTMENT_ENROLL",
  DividendReinvestmentDecline = "DIVIDEND_REINVESTMENT_DECLINE",
}
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export type IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlanOpen =
  OpenEnum<
    typeof IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan
  >;

/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export enum IRATraditionalEnrollmentMetadataCreateFdicCashSweep {
  AutoEnrollFdicCashSweepUnspecified =
    "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED",
  FdicCashSweepEnroll = "FDIC_CASH_SWEEP_ENROLL",
  FdicCashSweepDecline = "FDIC_CASH_SWEEP_DECLINE",
}
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export type IRATraditionalEnrollmentMetadataCreateFdicCashSweepOpen = OpenEnum<
  typeof IRATraditionalEnrollmentMetadataCreateFdicCashSweep
>;

/**
 * Enrollment metadata for Traditional IRA accounts enrollment type
 */
export type IRATraditionalEnrollmentMetadataCreate = {
  /**
   * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
   */
  dividendReinvestmentPlan?:
    | IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlanOpen
    | undefined;
  /**
   * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
   */
  fdicCashSweep?:
    | IRATraditionalEnrollmentMetadataCreateFdicCashSweepOpen
    | undefined;
};

/** @internal */
export const IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema:
  z.ZodType<
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(
        IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan,
      ),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema:
  z.ZodType<
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlanOpen
  > = z.union([
    z.nativeEnum(
      IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan,
    ),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$ {
  /** @deprecated use `IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
  export const inboundSchema =
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
  /** @deprecated use `IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
  export const outboundSchema =
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
}

/** @internal */
export const IRATraditionalEnrollmentMetadataCreateFdicCashSweep$inboundSchema:
  z.ZodType<
    IRATraditionalEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(IRATraditionalEnrollmentMetadataCreateFdicCashSweep),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const IRATraditionalEnrollmentMetadataCreateFdicCashSweep$outboundSchema:
  z.ZodType<
    IRATraditionalEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    IRATraditionalEnrollmentMetadataCreateFdicCashSweepOpen
  > = z.union([
    z.nativeEnum(IRATraditionalEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IRATraditionalEnrollmentMetadataCreateFdicCashSweep$ {
  /** @deprecated use `IRATraditionalEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
  export const inboundSchema =
    IRATraditionalEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
  /** @deprecated use `IRATraditionalEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
  export const outboundSchema =
    IRATraditionalEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
}

/** @internal */
export const IRATraditionalEnrollmentMetadataCreate$inboundSchema: z.ZodType<
  IRATraditionalEnrollmentMetadataCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  dividend_reinvestment_plan:
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
      .optional(),
  fdic_cash_sweep:
    IRATraditionalEnrollmentMetadataCreateFdicCashSweep$inboundSchema
      .optional(),
}).transform((v) => {
  return remap$(v, {
    "dividend_reinvestment_plan": "dividendReinvestmentPlan",
    "fdic_cash_sweep": "fdicCashSweep",
  });
});

/** @internal */
export type IRATraditionalEnrollmentMetadataCreate$Outbound = {
  dividend_reinvestment_plan?: string | undefined;
  fdic_cash_sweep?: string | undefined;
};

/** @internal */
export const IRATraditionalEnrollmentMetadataCreate$outboundSchema: z.ZodType<
  IRATraditionalEnrollmentMetadataCreate$Outbound,
  z.ZodTypeDef,
  IRATraditionalEnrollmentMetadataCreate
> = z.object({
  dividendReinvestmentPlan:
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
      .optional(),
  fdicCashSweep:
    IRATraditionalEnrollmentMetadataCreateFdicCashSweep$outboundSchema
      .optional(),
}).transform((v) => {
  return remap$(v, {
    dividendReinvestmentPlan: "dividend_reinvestment_plan",
    fdicCashSweep: "fdic_cash_sweep",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IRATraditionalEnrollmentMetadataCreate$ {
  /** @deprecated use `IRATraditionalEnrollmentMetadataCreate$inboundSchema` instead. */
  export const inboundSchema =
    IRATraditionalEnrollmentMetadataCreate$inboundSchema;
  /** @deprecated use `IRATraditionalEnrollmentMetadataCreate$outboundSchema` instead. */
  export const outboundSchema =
    IRATraditionalEnrollmentMetadataCreate$outboundSchema;
  /** @deprecated use `IRATraditionalEnrollmentMetadataCreate$Outbound` instead. */
  export type Outbound = IRATraditionalEnrollmentMetadataCreate$Outbound;
}
