// @apexfintechsolutions/ascend-sdk/src/models/components/jointtenantsincommonenrollmentmetadatacreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";

/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export enum JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan {
  AutoEnrollDividendReinvestmentUnspecified =
    "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED",
  DividendReinvestmentEnroll = "DIVIDEND_REINVESTMENT_ENROLL",
  DividendReinvestmentDecline = "DIVIDEND_REINVESTMENT_DECLINE",
}
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export type JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlanOpen =
  OpenEnum<
    typeof JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan
  >;

/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export enum JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep {
  AutoEnrollFdicCashSweepUnspecified =
    "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED",
  FdicCashSweepEnroll = "FDIC_CASH_SWEEP_ENROLL",
  FdicCashSweepDecline = "FDIC_CASH_SWEEP_DECLINE",
}
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export type JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweepOpen =
  OpenEnum<typeof JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep>;

/**
 * Enrollment metadata for the Joint Tenants In Common enrollment type
 */
export type JointTenantsInCommonEnrollmentMetadataCreate = {
  /**
   * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
   */
  dividendReinvestmentPlan?:
    | JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlanOpen
    | undefined;
  /**
   * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
   */
  fdicCashSweep?:
    | JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweepOpen
    | undefined;
};

/** @internal */
export const JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema:
  z.ZodType<
    JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(
        JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan,
      ),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema:
  z.ZodType<
    JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlanOpen
  > = z.union([
    z.nativeEnum(
      JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan,
    ),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$ {
  /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
  export const inboundSchema =
    JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
  /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
  export const outboundSchema =
    JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
}

/** @internal */
export const JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$inboundSchema:
  z.ZodType<
    JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$outboundSchema:
  z.ZodType<
    JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweepOpen
  > = z.union([
    z.nativeEnum(JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$ {
  /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
  export const inboundSchema =
    JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
  /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
  export const outboundSchema =
    JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
}

/** @internal */
export const JointTenantsInCommonEnrollmentMetadataCreate$inboundSchema:
  z.ZodType<
    JointTenantsInCommonEnrollmentMetadataCreate,
    z.ZodTypeDef,
    unknown
  > = z.object({
    dividend_reinvestment_plan:
      JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep:
      JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
  }).transform((v) => {
    return remap$(v, {
      "dividend_reinvestment_plan": "dividendReinvestmentPlan",
      "fdic_cash_sweep": "fdicCashSweep",
    });
  });

/** @internal */
export type JointTenantsInCommonEnrollmentMetadataCreate$Outbound = {
  dividend_reinvestment_plan?: string | undefined;
  fdic_cash_sweep?: string | undefined;
};

/** @internal */
export const JointTenantsInCommonEnrollmentMetadataCreate$outboundSchema:
  z.ZodType<
    JointTenantsInCommonEnrollmentMetadataCreate$Outbound,
    z.ZodTypeDef,
    JointTenantsInCommonEnrollmentMetadataCreate
  > = z.object({
    dividendReinvestmentPlan:
      JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep:
      JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
  }).transform((v) => {
    return remap$(v, {
      dividendReinvestmentPlan: "dividend_reinvestment_plan",
      fdicCashSweep: "fdic_cash_sweep",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace JointTenantsInCommonEnrollmentMetadataCreate$ {
  /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreate$inboundSchema` instead. */
  export const inboundSchema =
    JointTenantsInCommonEnrollmentMetadataCreate$inboundSchema;
  /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreate$outboundSchema` instead. */
  export const outboundSchema =
    JointTenantsInCommonEnrollmentMetadataCreate$outboundSchema;
  /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreate$Outbound` instead. */
  export type Outbound = JointTenantsInCommonEnrollmentMetadataCreate$Outbound;
}
