// @apexfintechsolutions/ascend-sdk/src/models/components/jointwithrightsofsurvivorshipenrollmentmetadatacreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";

/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export enum JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan {
  AutoEnrollDividendReinvestmentUnspecified =
    "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED",
  DividendReinvestmentEnroll = "DIVIDEND_REINVESTMENT_ENROLL",
  DividendReinvestmentDecline = "DIVIDEND_REINVESTMENT_DECLINE",
}
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export type JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlanOpen =
  OpenEnum<
    typeof JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan
  >;

/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export enum JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep {
  AutoEnrollFdicCashSweepUnspecified =
    "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED",
  FdicCashSweepEnroll = "FDIC_CASH_SWEEP_ENROLL",
  FdicCashSweepDecline = "FDIC_CASH_SWEEP_DECLINE",
}
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export type JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweepOpen =
  OpenEnum<
    typeof JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep
  >;

/**
 * Enrollment metadata for the With Right of Survivorship enrollment type
 */
export type JointWithRightsOfSurvivorshipEnrollmentMetadataCreate = {
  /**
   * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
   */
  dividendReinvestmentPlan?:
    | JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlanOpen
    | undefined;
  /**
   * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
   */
  fdicCashSweep?:
    | JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweepOpen
    | undefined;
};

/** @internal */
export const JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema:
  z.ZodType<
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(
        JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan,
      ),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema:
  z.ZodType<
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlanOpen
  > = z.union([
    z.nativeEnum(
      JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan,
    ),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$ {
  /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
  export const inboundSchema =
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
  /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
  export const outboundSchema =
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
}

/** @internal */
export const JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$inboundSchema:
  z.ZodType<
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(
        JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep,
      ),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$outboundSchema:
  z.ZodType<
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweepOpen,
    z.ZodTypeDef,
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweepOpen
  > = z.union([
    z.nativeEnum(
      JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep,
    ),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$ {
  /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
  export const inboundSchema =
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
  /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
  export const outboundSchema =
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
}

/** @internal */
export const JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$inboundSchema:
  z.ZodType<
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreate,
    z.ZodTypeDef,
    unknown
  > = z.object({
    dividend_reinvestment_plan:
      JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep:
      JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
  }).transform((v) => {
    return remap$(v, {
      "dividend_reinvestment_plan": "dividendReinvestmentPlan",
      "fdic_cash_sweep": "fdicCashSweep",
    });
  });

/** @internal */
export type JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$Outbound = {
  dividend_reinvestment_plan?: string | undefined;
  fdic_cash_sweep?: string | undefined;
};

/** @internal */
export const JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$outboundSchema:
  z.ZodType<
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$Outbound,
    z.ZodTypeDef,
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreate
  > = z.object({
    dividendReinvestmentPlan:
      JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep:
      JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
  }).transform((v) => {
    return remap$(v, {
      dividendReinvestmentPlan: "dividend_reinvestment_plan",
      fdicCashSweep: "fdic_cash_sweep",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$ {
  /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$inboundSchema` instead. */
  export const inboundSchema =
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$inboundSchema;
  /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$outboundSchema` instead. */
  export const outboundSchema =
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$outboundSchema;
  /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$Outbound` instead. */
  export type Outbound =
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$Outbound;
}
