// @apexfintechsolutions/ascend-sdk/src/models/components/limitpricecreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";
import {
  DecimalCreate,
  DecimalCreate$inboundSchema,
  DecimalCreate$Outbound,
  DecimalCreate$outboundSchema,
} from "./decimalcreate.js";

/**
 * The type of this price, which must be PRICE_PER_UNIT for equity orders, or PERCENTAGE_OF_PAR for fixed income orders.
 */
export enum LimitPriceCreateType {
  LimitPriceTypeUnspecified = "LIMIT_PRICE_TYPE_UNSPECIFIED",
  PricePerUnit = "PRICE_PER_UNIT",
  PercentageOfPar = "PERCENTAGE_OF_PAR",
}
/**
 * The type of this price, which must be PRICE_PER_UNIT for equity orders, or PERCENTAGE_OF_PAR for fixed income orders.
 */
export type LimitPriceCreateTypeOpen = OpenEnum<typeof LimitPriceCreateType>;

/**
 * A limit price definition
 */
export type LimitPriceCreate = {
  /**
   * A representation of a decimal value, such as 2.5. Clients may convert values into language-native decimal formats, such as Java's [BigDecimal][] or Python's [decimal.Decimal][].
   *
   * @remarks
   *
   *  [BigDecimal]:
   *  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html
   *  [decimal.Decimal]: https://docs.python.org/3/library/decimal.html
   */
  price: DecimalCreate;
  /**
   * The type of this price, which must be PRICE_PER_UNIT for equity orders, or PERCENTAGE_OF_PAR for fixed income orders.
   */
  type: LimitPriceCreateTypeOpen;
};

/** @internal */
export const LimitPriceCreateType$inboundSchema: z.ZodType<
  LimitPriceCreateTypeOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(LimitPriceCreateType),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const LimitPriceCreateType$outboundSchema: z.ZodType<
  LimitPriceCreateTypeOpen,
  z.ZodTypeDef,
  LimitPriceCreateTypeOpen
> = z.union([
  z.nativeEnum(LimitPriceCreateType),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace LimitPriceCreateType$ {
  /** @deprecated use `LimitPriceCreateType$inboundSchema` instead. */
  export const inboundSchema = LimitPriceCreateType$inboundSchema;
  /** @deprecated use `LimitPriceCreateType$outboundSchema` instead. */
  export const outboundSchema = LimitPriceCreateType$outboundSchema;
}

/** @internal */
export const LimitPriceCreate$inboundSchema: z.ZodType<
  LimitPriceCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  price: DecimalCreate$inboundSchema,
  type: LimitPriceCreateType$inboundSchema,
});

/** @internal */
export type LimitPriceCreate$Outbound = {
  price: DecimalCreate$Outbound;
  type: string;
};

/** @internal */
export const LimitPriceCreate$outboundSchema: z.ZodType<
  LimitPriceCreate$Outbound,
  z.ZodTypeDef,
  LimitPriceCreate
> = z.object({
  price: DecimalCreate$outboundSchema,
  type: LimitPriceCreateType$outboundSchema,
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace LimitPriceCreate$ {
  /** @deprecated use `LimitPriceCreate$inboundSchema` instead. */
  export const inboundSchema = LimitPriceCreate$inboundSchema;
  /** @deprecated use `LimitPriceCreate$outboundSchema` instead. */
  export const outboundSchema = LimitPriceCreate$outboundSchema;
  /** @deprecated use `LimitPriceCreate$Outbound` instead. */
  export type Outbound = LimitPriceCreate$Outbound;
}
