// @apexfintechsolutions/ascend-sdk/src/models/components/linkdocumentsrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";

/**
 * Custom request - adds identity verification documentIds to investigation request
 */
export type LinkDocumentsRequestCreate = {
  /**
   * Document ids to be added to identity verification result
   */
  identityVerificationDocumentIds: Array<string>;
};

/** @internal */
export const LinkDocumentsRequestCreate$inboundSchema: z.ZodType<
  LinkDocumentsRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  identity_verification_document_ids: z.array(z.string()),
}).transform((v) => {
  return remap$(v, {
    "identity_verification_document_ids": "identityVerificationDocumentIds",
  });
});

/** @internal */
export type LinkDocumentsRequestCreate$Outbound = {
  identity_verification_document_ids: Array<string>;
};

/** @internal */
export const LinkDocumentsRequestCreate$outboundSchema: z.ZodType<
  LinkDocumentsRequestCreate$Outbound,
  z.ZodTypeDef,
  LinkDocumentsRequestCreate
> = z.object({
  identityVerificationDocumentIds: z.array(z.string()),
}).transform((v) => {
  return remap$(v, {
    identityVerificationDocumentIds: "identity_verification_document_ids",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace LinkDocumentsRequestCreate$ {
  /** @deprecated use `LinkDocumentsRequestCreate$inboundSchema` instead. */
  export const inboundSchema = LinkDocumentsRequestCreate$inboundSchema;
  /** @deprecated use `LinkDocumentsRequestCreate$outboundSchema` instead. */
  export const outboundSchema = LinkDocumentsRequestCreate$outboundSchema;
  /** @deprecated use `LinkDocumentsRequestCreate$Outbound` instead. */
  export type Outbound = LinkDocumentsRequestCreate$Outbound;
}
