// @apexfintechsolutions/ascend-sdk/src/models/components/listwirewithdrawalschedulesresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  WireWithdrawalSchedule,
  WireWithdrawalSchedule$inboundSchema,
  WireWithdrawalSchedule$Outbound,
  WireWithdrawalSchedule$outboundSchema,
} from "./wirewithdrawalschedule.js";

/**
 * A paged response containing a list of Wire withdrawal transfer schedules
 */
export type ListWireWithdrawalSchedulesResponse = {
  /**
   * The next page token
   */
  nextPageToken?: string | undefined;
  /**
   * The list of transfer schedules
   */
  wireWithdrawalSchedules?: Array<WireWithdrawalSchedule> | undefined;
};

/** @internal */
export const ListWireWithdrawalSchedulesResponse$inboundSchema: z.ZodType<
  ListWireWithdrawalSchedulesResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  next_page_token: z.string().optional(),
  wire_withdrawal_schedules: z.array(WireWithdrawalSchedule$inboundSchema)
    .optional(),
}).transform((v) => {
  return remap$(v, {
    "next_page_token": "nextPageToken",
    "wire_withdrawal_schedules": "wireWithdrawalSchedules",
  });
});

/** @internal */
export type ListWireWithdrawalSchedulesResponse$Outbound = {
  next_page_token?: string | undefined;
  wire_withdrawal_schedules?:
    | Array<WireWithdrawalSchedule$Outbound>
    | undefined;
};

/** @internal */
export const ListWireWithdrawalSchedulesResponse$outboundSchema: z.ZodType<
  ListWireWithdrawalSchedulesResponse$Outbound,
  z.ZodTypeDef,
  ListWireWithdrawalSchedulesResponse
> = z.object({
  nextPageToken: z.string().optional(),
  wireWithdrawalSchedules: z.array(WireWithdrawalSchedule$outboundSchema)
    .optional(),
}).transform((v) => {
  return remap$(v, {
    nextPageToken: "next_page_token",
    wireWithdrawalSchedules: "wire_withdrawal_schedules",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListWireWithdrawalSchedulesResponse$ {
  /** @deprecated use `ListWireWithdrawalSchedulesResponse$inboundSchema` instead. */
  export const inboundSchema =
    ListWireWithdrawalSchedulesResponse$inboundSchema;
  /** @deprecated use `ListWireWithdrawalSchedulesResponse$outboundSchema` instead. */
  export const outboundSchema =
    ListWireWithdrawalSchedulesResponse$outboundSchema;
  /** @deprecated use `ListWireWithdrawalSchedulesResponse$Outbound` instead. */
  export type Outbound = ListWireWithdrawalSchedulesResponse$Outbound;
}
