// @apexfintechsolutions/ascend-sdk/src/models/components/listdistributionsummariesresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  DistributionSummary,
  DistributionSummary$inboundSchema,
  DistributionSummary$Outbound,
  DistributionSummary$outboundSchema,
} from "./distributionsummary.js";

/**
 * Response object for aggregated distribution retirement summaries
 */
export type ListDistributionSummariesResponse = {
  /**
   * List of distribution amount(s) and rollover amount(s), with their corresponding tax year(s)
   */
  distributionSummaries?: Array<DistributionSummary> | undefined;
  /**
   * Can be sent as `page_token` in the request to retrieve the next page If this field is omitted, there are no subsequent pages
   */
  nextPageToken?: string | undefined;
};

/** @internal */
export const ListDistributionSummariesResponse$inboundSchema: z.ZodType<
  ListDistributionSummariesResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  distribution_summaries: z.array(DistributionSummary$inboundSchema).optional(),
  next_page_token: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "distribution_summaries": "distributionSummaries",
    "next_page_token": "nextPageToken",
  });
});

/** @internal */
export type ListDistributionSummariesResponse$Outbound = {
  distribution_summaries?: Array<DistributionSummary$Outbound> | undefined;
  next_page_token?: string | undefined;
};

/** @internal */
export const ListDistributionSummariesResponse$outboundSchema: z.ZodType<
  ListDistributionSummariesResponse$Outbound,
  z.ZodTypeDef,
  ListDistributionSummariesResponse
> = z.object({
  distributionSummaries: z.array(DistributionSummary$outboundSchema).optional(),
  nextPageToken: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    distributionSummaries: "distribution_summaries",
    nextPageToken: "next_page_token",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListDistributionSummariesResponse$ {
  /** @deprecated use `ListDistributionSummariesResponse$inboundSchema` instead. */
  export const inboundSchema = ListDistributionSummariesResponse$inboundSchema;
  /** @deprecated use `ListDistributionSummariesResponse$outboundSchema` instead. */
  export const outboundSchema =
    ListDistributionSummariesResponse$outboundSchema;
  /** @deprecated use `ListDistributionSummariesResponse$Outbound` instead. */
  export type Outbound = ListDistributionSummariesResponse$Outbound;
}
