// @apexfintechsolutions/ascend-sdk/src/models/components/listaccountsresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  Account,
  Account$inboundSchema,
  Account$Outbound,
  Account$outboundSchema,
} from "./account.js";

/**
 * The response to list Accounts.
 */
export type ListAccountsResponse = {
  /**
   * The list of available accounts
   */
  accounts?: Array<Account> | undefined;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page. If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string | undefined;
  /**
   * The total number of accounts in the response. This is an estimate and may be more than the number of accounts in this response if the user is not authorized to view all accounts.
   */
  totalSize?: string | undefined;
};

/** @internal */
export const ListAccountsResponse$inboundSchema: z.ZodType<
  ListAccountsResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  accounts: z.array(Account$inboundSchema).optional(),
  next_page_token: z.string().optional(),
  total_size: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "next_page_token": "nextPageToken",
    "total_size": "totalSize",
  });
});

/** @internal */
export type ListAccountsResponse$Outbound = {
  accounts?: Array<Account$Outbound> | undefined;
  next_page_token?: string | undefined;
  total_size?: string | undefined;
};

/** @internal */
export const ListAccountsResponse$outboundSchema: z.ZodType<
  ListAccountsResponse$Outbound,
  z.ZodTypeDef,
  ListAccountsResponse
> = z.object({
  accounts: z.array(Account$outboundSchema).optional(),
  nextPageToken: z.string().optional(),
  totalSize: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    nextPageToken: "next_page_token",
    totalSize: "total_size",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListAccountsResponse$ {
  /** @deprecated use `ListAccountsResponse$inboundSchema` instead. */
  export const inboundSchema = ListAccountsResponse$inboundSchema;
  /** @deprecated use `ListAccountsResponse$outboundSchema` instead. */
  export const outboundSchema = ListAccountsResponse$outboundSchema;
  /** @deprecated use `ListAccountsResponse$Outbound` instead. */
  export type Outbound = ListAccountsResponse$Outbound;
}
