// @apexfintechsolutions/ascend-sdk/src/models/components/listavailableenrollmentsresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  AvailableEnrollment,
  AvailableEnrollment$inboundSchema,
  AvailableEnrollment$Outbound,
  AvailableEnrollment$outboundSchema,
} from "./availableenrollment.js";

/**
 * The response to list Enrollments available for an Account.
 */
export type ListAvailableEnrollmentsResponse = {
  /**
   * The available enrollments for a response
   */
  availableEnrollments?: Array<AvailableEnrollment> | undefined;
};

/** @internal */
export const ListAvailableEnrollmentsResponse$inboundSchema: z.ZodType<
  ListAvailableEnrollmentsResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  available_enrollments: z.array(AvailableEnrollment$inboundSchema).optional(),
}).transform((v) => {
  return remap$(v, {
    "available_enrollments": "availableEnrollments",
  });
});

/** @internal */
export type ListAvailableEnrollmentsResponse$Outbound = {
  available_enrollments?: Array<AvailableEnrollment$Outbound> | undefined;
};

/** @internal */
export const ListAvailableEnrollmentsResponse$outboundSchema: z.ZodType<
  ListAvailableEnrollmentsResponse$Outbound,
  z.ZodTypeDef,
  ListAvailableEnrollmentsResponse
> = z.object({
  availableEnrollments: z.array(AvailableEnrollment$outboundSchema).optional(),
}).transform((v) => {
  return remap$(v, {
    availableEnrollments: "available_enrollments",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListAvailableEnrollmentsResponse$ {
  /** @deprecated use `ListAvailableEnrollmentsResponse$inboundSchema` instead. */
  export const inboundSchema = ListAvailableEnrollmentsResponse$inboundSchema;
  /** @deprecated use `ListAvailableEnrollmentsResponse$outboundSchema` instead. */
  export const outboundSchema = ListAvailableEnrollmentsResponse$outboundSchema;
  /** @deprecated use `ListAvailableEnrollmentsResponse$Outbound` instead. */
  export type Outbound = ListAvailableEnrollmentsResponse$Outbound;
}
