// @apexfintechsolutions/ascend-sdk/src/models/components/listentitlementsresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  Entitlement,
  Entitlement$inboundSchema,
  Entitlement$Outbound,
  Entitlement$outboundSchema,
} from "./entitlement.js";

/**
 * The response to list Entitlements on an Account.
 */
export type ListEntitlementsResponse = {
  /**
   * The list of Entitlements on an Account
   */
  entitlements?: Array<Entitlement> | undefined;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page. If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string | undefined;
};

/** @internal */
export const ListEntitlementsResponse$inboundSchema: z.ZodType<
  ListEntitlementsResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  entitlements: z.array(Entitlement$inboundSchema).optional(),
  next_page_token: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "next_page_token": "nextPageToken",
  });
});

/** @internal */
export type ListEntitlementsResponse$Outbound = {
  entitlements?: Array<Entitlement$Outbound> | undefined;
  next_page_token?: string | undefined;
};

/** @internal */
export const ListEntitlementsResponse$outboundSchema: z.ZodType<
  ListEntitlementsResponse$Outbound,
  z.ZodTypeDef,
  ListEntitlementsResponse
> = z.object({
  entitlements: z.array(Entitlement$outboundSchema).optional(),
  nextPageToken: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    nextPageToken: "next_page_token",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListEntitlementsResponse$ {
  /** @deprecated use `ListEntitlementsResponse$inboundSchema` instead. */
  export const inboundSchema = ListEntitlementsResponse$inboundSchema;
  /** @deprecated use `ListEntitlementsResponse$outboundSchema` instead. */
  export const outboundSchema = ListEntitlementsResponse$outboundSchema;
  /** @deprecated use `ListEntitlementsResponse$Outbound` instead. */
  export type Outbound = ListEntitlementsResponse$Outbound;
}
