// @apexfintechsolutions/ascend-sdk/src/models/components/listbankrelationshipsresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  BankRelationship,
  BankRelationship$inboundSchema,
  BankRelationship$Outbound,
  BankRelationship$outboundSchema,
} from "./bankrelationship.js";

/**
 * Returns bank relationships for a parent account.
 */
export type ListBankRelationshipsResponse = {
  /**
   * The bank relationships from the specified account.
   */
  bankRelationships?: Array<BankRelationship> | undefined;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page. If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string | undefined;
};

/** @internal */
export const ListBankRelationshipsResponse$inboundSchema: z.ZodType<
  ListBankRelationshipsResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  bank_relationships: z.array(BankRelationship$inboundSchema).optional(),
  next_page_token: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "bank_relationships": "bankRelationships",
    "next_page_token": "nextPageToken",
  });
});

/** @internal */
export type ListBankRelationshipsResponse$Outbound = {
  bank_relationships?: Array<BankRelationship$Outbound> | undefined;
  next_page_token?: string | undefined;
};

/** @internal */
export const ListBankRelationshipsResponse$outboundSchema: z.ZodType<
  ListBankRelationshipsResponse$Outbound,
  z.ZodTypeDef,
  ListBankRelationshipsResponse
> = z.object({
  bankRelationships: z.array(BankRelationship$outboundSchema).optional(),
  nextPageToken: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    bankRelationships: "bank_relationships",
    nextPageToken: "next_page_token",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListBankRelationshipsResponse$ {
  /** @deprecated use `ListBankRelationshipsResponse$inboundSchema` instead. */
  export const inboundSchema = ListBankRelationshipsResponse$inboundSchema;
  /** @deprecated use `ListBankRelationshipsResponse$outboundSchema` instead. */
  export const outboundSchema = ListBankRelationshipsResponse$outboundSchema;
  /** @deprecated use `ListBankRelationshipsResponse$Outbound` instead. */
  export type Outbound = ListBankRelationshipsResponse$Outbound;
}
