// @apexfintechsolutions/ascend-sdk/src/models/components/listpushsubscriptiondeliveriesresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  PushSubscriptionDelivery,
  PushSubscriptionDelivery$inboundSchema,
  PushSubscriptionDelivery$Outbound,
  PushSubscriptionDelivery$outboundSchema,
} from "./pushsubscriptiondelivery.js";

/**
 * A response to a list push subscription deliveries method
 */
export type ListPushSubscriptionDeliveriesResponse = {
  /**
   * Page token used for pagination; Supplying a page token returns the next page of results
   */
  nextPageToken?: string | undefined;
  /**
   * The returned collection of deliveries
   */
  pushSubscriptionDeliveries?: Array<PushSubscriptionDelivery> | undefined;
};

/** @internal */
export const ListPushSubscriptionDeliveriesResponse$inboundSchema: z.ZodType<
  ListPushSubscriptionDeliveriesResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  next_page_token: z.string().optional(),
  push_subscription_deliveries: z.array(PushSubscriptionDelivery$inboundSchema)
    .optional(),
}).transform((v) => {
  return remap$(v, {
    "next_page_token": "nextPageToken",
    "push_subscription_deliveries": "pushSubscriptionDeliveries",
  });
});

/** @internal */
export type ListPushSubscriptionDeliveriesResponse$Outbound = {
  next_page_token?: string | undefined;
  push_subscription_deliveries?:
    | Array<PushSubscriptionDelivery$Outbound>
    | undefined;
};

/** @internal */
export const ListPushSubscriptionDeliveriesResponse$outboundSchema: z.ZodType<
  ListPushSubscriptionDeliveriesResponse$Outbound,
  z.ZodTypeDef,
  ListPushSubscriptionDeliveriesResponse
> = z.object({
  nextPageToken: z.string().optional(),
  pushSubscriptionDeliveries: z.array(PushSubscriptionDelivery$outboundSchema)
    .optional(),
}).transform((v) => {
  return remap$(v, {
    nextPageToken: "next_page_token",
    pushSubscriptionDeliveries: "push_subscription_deliveries",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListPushSubscriptionDeliveriesResponse$ {
  /** @deprecated use `ListPushSubscriptionDeliveriesResponse$inboundSchema` instead. */
  export const inboundSchema =
    ListPushSubscriptionDeliveriesResponse$inboundSchema;
  /** @deprecated use `ListPushSubscriptionDeliveriesResponse$outboundSchema` instead. */
  export const outboundSchema =
    ListPushSubscriptionDeliveriesResponse$outboundSchema;
  /** @deprecated use `ListPushSubscriptionDeliveriesResponse$Outbound` instead. */
  export type Outbound = ListPushSubscriptionDeliveriesResponse$Outbound;
}
