// @apexfintechsolutions/ascend-sdk/src/models/components/listenrollmentsresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  Enrollment,
  Enrollment$inboundSchema,
  Enrollment$Outbound,
  Enrollment$outboundSchema,
} from "./enrollment.js";

/**
 * The response to list Enrollments on an Account.
 */
export type ListEnrollmentsResponse = {
  /**
   * The list of available enrollments
   */
  enrollments?: Array<Enrollment> | undefined;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page. If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string | undefined;
};

/** @internal */
export const ListEnrollmentsResponse$inboundSchema: z.ZodType<
  ListEnrollmentsResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  enrollments: z.array(Enrollment$inboundSchema).optional(),
  next_page_token: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "next_page_token": "nextPageToken",
  });
});

/** @internal */
export type ListEnrollmentsResponse$Outbound = {
  enrollments?: Array<Enrollment$Outbound> | undefined;
  next_page_token?: string | undefined;
};

/** @internal */
export const ListEnrollmentsResponse$outboundSchema: z.ZodType<
  ListEnrollmentsResponse$Outbound,
  z.ZodTypeDef,
  ListEnrollmentsResponse
> = z.object({
  enrollments: z.array(Enrollment$outboundSchema).optional(),
  nextPageToken: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    nextPageToken: "next_page_token",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListEnrollmentsResponse$ {
  /** @deprecated use `ListEnrollmentsResponse$inboundSchema` instead. */
  export const inboundSchema = ListEnrollmentsResponse$inboundSchema;
  /** @deprecated use `ListEnrollmentsResponse$outboundSchema` instead. */
  export const outboundSchema = ListEnrollmentsResponse$outboundSchema;
  /** @deprecated use `ListEnrollmentsResponse$Outbound` instead. */
  export type Outbound = ListEnrollmentsResponse$Outbound;
}
