// @apexfintechsolutions/ascend-sdk/src/models/components/listschedulesummariesresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  TransferScheduleSummary,
  TransferScheduleSummary$inboundSchema,
  TransferScheduleSummary$Outbound,
  TransferScheduleSummary$outboundSchema,
} from "./transferschedulesummary.js";

/**
 * Response to a search transfer schedules request
 */
export type ListScheduleSummariesResponse = {
  /**
   * The next page token
   */
  nextPageToken?: string | undefined;
  /**
   * The list of transfer schedule details
   */
  scheduleSummaries?: Array<TransferScheduleSummary> | undefined;
};

/** @internal */
export const ListScheduleSummariesResponse$inboundSchema: z.ZodType<
  ListScheduleSummariesResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  next_page_token: z.string().optional(),
  schedule_summaries: z.array(TransferScheduleSummary$inboundSchema).optional(),
}).transform((v) => {
  return remap$(v, {
    "next_page_token": "nextPageToken",
    "schedule_summaries": "scheduleSummaries",
  });
});

/** @internal */
export type ListScheduleSummariesResponse$Outbound = {
  next_page_token?: string | undefined;
  schedule_summaries?: Array<TransferScheduleSummary$Outbound> | undefined;
};

/** @internal */
export const ListScheduleSummariesResponse$outboundSchema: z.ZodType<
  ListScheduleSummariesResponse$Outbound,
  z.ZodTypeDef,
  ListScheduleSummariesResponse
> = z.object({
  nextPageToken: z.string().optional(),
  scheduleSummaries: z.array(TransferScheduleSummary$outboundSchema).optional(),
}).transform((v) => {
  return remap$(v, {
    nextPageToken: "next_page_token",
    scheduleSummaries: "schedule_summaries",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListScheduleSummariesResponse$ {
  /** @deprecated use `ListScheduleSummariesResponse$inboundSchema` instead. */
  export const inboundSchema = ListScheduleSummariesResponse$inboundSchema;
  /** @deprecated use `ListScheduleSummariesResponse$outboundSchema` instead. */
  export const outboundSchema = ListScheduleSummariesResponse$outboundSchema;
  /** @deprecated use `ListScheduleSummariesResponse$Outbound` instead. */
  export type Outbound = ListScheduleSummariesResponse$Outbound;
}
