// @apexfintechsolutions/ascend-sdk/src/models/components/listavailablerestrictionsresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  AvailableRestriction,
  AvailableRestriction$inboundSchema,
  AvailableRestriction$Outbound,
  AvailableRestriction$outboundSchema,
} from "./availablerestriction.js";

/**
 * The response to list available Restrictions for an Account.
 */
export type ListAvailableRestrictionsResponse = {
  /**
   * The available restrictions on an account
   */
  availableRestrictions?: Array<AvailableRestriction> | undefined;
};

/** @internal */
export const ListAvailableRestrictionsResponse$inboundSchema: z.ZodType<
  ListAvailableRestrictionsResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  available_restrictions: z.array(AvailableRestriction$inboundSchema)
    .optional(),
}).transform((v) => {
  return remap$(v, {
    "available_restrictions": "availableRestrictions",
  });
});

/** @internal */
export type ListAvailableRestrictionsResponse$Outbound = {
  available_restrictions?: Array<AvailableRestriction$Outbound> | undefined;
};

/** @internal */
export const ListAvailableRestrictionsResponse$outboundSchema: z.ZodType<
  ListAvailableRestrictionsResponse$Outbound,
  z.ZodTypeDef,
  ListAvailableRestrictionsResponse
> = z.object({
  availableRestrictions: z.array(AvailableRestriction$outboundSchema)
    .optional(),
}).transform((v) => {
  return remap$(v, {
    availableRestrictions: "available_restrictions",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace ListAvailableRestrictionsResponse$ {
  /** @deprecated use `ListAvailableRestrictionsResponse$inboundSchema` instead. */
  export const inboundSchema = ListAvailableRestrictionsResponse$inboundSchema;
  /** @deprecated use `ListAvailableRestrictionsResponse$outboundSchema` instead. */
  export const outboundSchema =
    ListAvailableRestrictionsResponse$outboundSchema;
  /** @deprecated use `ListAvailableRestrictionsResponse$Outbound` instead. */
  export type Outbound = ListAvailableRestrictionsResponse$Outbound;
}
