// @apexfintechsolutions/ascend-sdk/src/models/components/llcenrollmentmetadatacreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";
import {
  EddAccountEnrollmentMetadataCreate,
  EddAccountEnrollmentMetadataCreate$inboundSchema,
  EddAccountEnrollmentMetadataCreate$Outbound,
  EddAccountEnrollmentMetadataCreate$outboundSchema,
} from "./eddaccountenrollmentmetadatacreate.js";

/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export enum LLCEnrollmentMetadataCreateDividendReinvestmentPlan {
  AutoEnrollDividendReinvestmentUnspecified =
    "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED",
  DividendReinvestmentEnroll = "DIVIDEND_REINVESTMENT_ENROLL",
  DividendReinvestmentDecline = "DIVIDEND_REINVESTMENT_DECLINE",
}
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
export type LLCEnrollmentMetadataCreateDividendReinvestmentPlanOpen = OpenEnum<
  typeof LLCEnrollmentMetadataCreateDividendReinvestmentPlan
>;

/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export enum LLCEnrollmentMetadataCreateFdicCashSweep {
  AutoEnrollFdicCashSweepUnspecified =
    "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED",
  FdicCashSweepEnroll = "FDIC_CASH_SWEEP_ENROLL",
  FdicCashSweepDecline = "FDIC_CASH_SWEEP_DECLINE",
}
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
export type LLCEnrollmentMetadataCreateFdicCashSweepOpen = OpenEnum<
  typeof LLCEnrollmentMetadataCreateFdicCashSweep
>;

export type LLCEnrollmentMetadataCreate = {
  /**
   * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
   */
  dividendReinvestmentPlan?:
    | LLCEnrollmentMetadataCreateDividendReinvestmentPlanOpen
    | undefined;
  /**
   * Enrollment metadata for Entity Accounts
   */
  eddAccountEnrollmentMetadata?: EddAccountEnrollmentMetadataCreate | undefined;
  /**
   * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
   */
  fdicCashSweep?: LLCEnrollmentMetadataCreateFdicCashSweepOpen | undefined;
};

/** @internal */
export const LLCEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema:
  z.ZodType<
    LLCEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(LLCEnrollmentMetadataCreateDividendReinvestmentPlan),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const LLCEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema:
  z.ZodType<
    LLCEnrollmentMetadataCreateDividendReinvestmentPlanOpen,
    z.ZodTypeDef,
    LLCEnrollmentMetadataCreateDividendReinvestmentPlanOpen
  > = z.union([
    z.nativeEnum(LLCEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace LLCEnrollmentMetadataCreateDividendReinvestmentPlan$ {
  /** @deprecated use `LLCEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
  export const inboundSchema =
    LLCEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
  /** @deprecated use `LLCEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
  export const outboundSchema =
    LLCEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
}

/** @internal */
export const LLCEnrollmentMetadataCreateFdicCashSweep$inboundSchema: z.ZodType<
  LLCEnrollmentMetadataCreateFdicCashSweepOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(LLCEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const LLCEnrollmentMetadataCreateFdicCashSweep$outboundSchema: z.ZodType<
  LLCEnrollmentMetadataCreateFdicCashSweepOpen,
  z.ZodTypeDef,
  LLCEnrollmentMetadataCreateFdicCashSweepOpen
> = z.union([
  z.nativeEnum(LLCEnrollmentMetadataCreateFdicCashSweep),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace LLCEnrollmentMetadataCreateFdicCashSweep$ {
  /** @deprecated use `LLCEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
  export const inboundSchema =
    LLCEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
  /** @deprecated use `LLCEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
  export const outboundSchema =
    LLCEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
}

/** @internal */
export const LLCEnrollmentMetadataCreate$inboundSchema: z.ZodType<
  LLCEnrollmentMetadataCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  dividend_reinvestment_plan:
    LLCEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
      .optional(),
  edd_account_enrollment_metadata:
    EddAccountEnrollmentMetadataCreate$inboundSchema.optional(),
  fdic_cash_sweep: LLCEnrollmentMetadataCreateFdicCashSweep$inboundSchema
    .optional(),
}).transform((v) => {
  return remap$(v, {
    "dividend_reinvestment_plan": "dividendReinvestmentPlan",
    "edd_account_enrollment_metadata": "eddAccountEnrollmentMetadata",
    "fdic_cash_sweep": "fdicCashSweep",
  });
});

/** @internal */
export type LLCEnrollmentMetadataCreate$Outbound = {
  dividend_reinvestment_plan?: string | undefined;
  edd_account_enrollment_metadata?:
    | EddAccountEnrollmentMetadataCreate$Outbound
    | undefined;
  fdic_cash_sweep?: string | undefined;
};

/** @internal */
export const LLCEnrollmentMetadataCreate$outboundSchema: z.ZodType<
  LLCEnrollmentMetadataCreate$Outbound,
  z.ZodTypeDef,
  LLCEnrollmentMetadataCreate
> = z.object({
  dividendReinvestmentPlan:
    LLCEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
      .optional(),
  eddAccountEnrollmentMetadata:
    EddAccountEnrollmentMetadataCreate$outboundSchema.optional(),
  fdicCashSweep: LLCEnrollmentMetadataCreateFdicCashSweep$outboundSchema
    .optional(),
}).transform((v) => {
  return remap$(v, {
    dividendReinvestmentPlan: "dividend_reinvestment_plan",
    eddAccountEnrollmentMetadata: "edd_account_enrollment_metadata",
    fdicCashSweep: "fdic_cash_sweep",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace LLCEnrollmentMetadataCreate$ {
  /** @deprecated use `LLCEnrollmentMetadataCreate$inboundSchema` instead. */
  export const inboundSchema = LLCEnrollmentMetadataCreate$inboundSchema;
  /** @deprecated use `LLCEnrollmentMetadataCreate$outboundSchema` instead. */
  export const outboundSchema = LLCEnrollmentMetadataCreate$outboundSchema;
  /** @deprecated use `LLCEnrollmentMetadataCreate$Outbound` instead. */
  export type Outbound = LLCEnrollmentMetadataCreate$Outbound;
}
