// @apexfintechsolutions/ascend-sdk/src/models/components/lowpricedsecuritiescreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  DecimalCreate,
  DecimalCreate$inboundSchema,
  DecimalCreate$Outbound,
  DecimalCreate$outboundSchema,
} from "./decimalcreate.js";

/**
 * Low priced securities
 */
export type LowPricedSecuritiesCreate = {
  /**
   * The account anticipates trading in securities trading for less than $5 per share and are typically traded over-the-counter (OTC) or through pink sheets
   */
  lowPricedSecurities: boolean;
  /**
   * A representation of a decimal value, such as 2.5. Clients may convert values into language-native decimal formats, such as Java's [BigDecimal][] or Python's [decimal.Decimal][].
   *
   * @remarks
   *
   *  [BigDecimal]:
   *  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html
   *  [decimal.Decimal]: https://docs.python.org/3/library/decimal.html
   */
  lowPricedSecuritiesPercentage?: DecimalCreate | undefined;
};

/** @internal */
export const LowPricedSecuritiesCreate$inboundSchema: z.ZodType<
  LowPricedSecuritiesCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  low_priced_securities: z.boolean(),
  low_priced_securities_percentage: DecimalCreate$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "low_priced_securities": "lowPricedSecurities",
    "low_priced_securities_percentage": "lowPricedSecuritiesPercentage",
  });
});

/** @internal */
export type LowPricedSecuritiesCreate$Outbound = {
  low_priced_securities: boolean;
  low_priced_securities_percentage?: DecimalCreate$Outbound | undefined;
};

/** @internal */
export const LowPricedSecuritiesCreate$outboundSchema: z.ZodType<
  LowPricedSecuritiesCreate$Outbound,
  z.ZodTypeDef,
  LowPricedSecuritiesCreate
> = z.object({
  lowPricedSecurities: z.boolean(),
  lowPricedSecuritiesPercentage: DecimalCreate$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    lowPricedSecurities: "low_priced_securities",
    lowPricedSecuritiesPercentage: "low_priced_securities_percentage",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace LowPricedSecuritiesCreate$ {
  /** @deprecated use `LowPricedSecuritiesCreate$inboundSchema` instead. */
  export const inboundSchema = LowPricedSecuritiesCreate$inboundSchema;
  /** @deprecated use `LowPricedSecuritiesCreate$outboundSchema` instead. */
  export const outboundSchema = LowPricedSecuritiesCreate$outboundSchema;
  /** @deprecated use `LowPricedSecuritiesCreate$Outbound` instead. */
  export type Outbound = LowPricedSecuritiesCreate$Outbound;
}
