// @apexfintechsolutions/ascend-sdk/src/models/components/microdepositamountscreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import {
  DecimalCreate,
  DecimalCreate$inboundSchema,
  DecimalCreate$Outbound,
  DecimalCreate$outboundSchema,
} from "./decimalcreate.js";

/**
 * The micro deposits sent to verify a pending bank relationship. FOR TESTING ONLY!
 */
export type MicroDepositAmountsCreate = {
  /**
   * A representation of a decimal value, such as 2.5. Clients may convert values into language-native decimal formats, such as Java's [BigDecimal][] or Python's [decimal.Decimal][].
   *
   * @remarks
   *
   *  [BigDecimal]:
   *  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html
   *  [decimal.Decimal]: https://docs.python.org/3/library/decimal.html
   */
  amount1: DecimalCreate;
  /**
   * A representation of a decimal value, such as 2.5. Clients may convert values into language-native decimal formats, such as Java's [BigDecimal][] or Python's [decimal.Decimal][].
   *
   * @remarks
   *
   *  [BigDecimal]:
   *  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html
   *  [decimal.Decimal]: https://docs.python.org/3/library/decimal.html
   */
  amount2: DecimalCreate;
};

/** @internal */
export const MicroDepositAmountsCreate$inboundSchema: z.ZodType<
  MicroDepositAmountsCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  amount1: DecimalCreate$inboundSchema,
  amount2: DecimalCreate$inboundSchema,
});

/** @internal */
export type MicroDepositAmountsCreate$Outbound = {
  amount1: DecimalCreate$Outbound;
  amount2: DecimalCreate$Outbound;
};

/** @internal */
export const MicroDepositAmountsCreate$outboundSchema: z.ZodType<
  MicroDepositAmountsCreate$Outbound,
  z.ZodTypeDef,
  MicroDepositAmountsCreate
> = z.object({
  amount1: DecimalCreate$outboundSchema,
  amount2: DecimalCreate$outboundSchema,
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace MicroDepositAmountsCreate$ {
  /** @deprecated use `MicroDepositAmountsCreate$inboundSchema` instead. */
  export const inboundSchema = MicroDepositAmountsCreate$inboundSchema;
  /** @deprecated use `MicroDepositAmountsCreate$outboundSchema` instead. */
  export const outboundSchema = MicroDepositAmountsCreate$outboundSchema;
  /** @deprecated use `MicroDepositAmountsCreate$Outbound` instead. */
  export type Outbound = MicroDepositAmountsCreate$Outbound;
}
