// @apexfintechsolutions/ascend-sdk/src/models/components/negativenewscreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";

/**
 * Negative News detail.
 */
export type NegativeNewsCreate = {
  /**
   * Indicates whether there is negative news against related parties
   */
  negativeNewsAgainstRelatedParties: boolean;
  /**
   * Description of the negative news against related parties
   */
  negativeNewsAgainstRelatedPartiesDescription?: string | undefined;
};

/** @internal */
export const NegativeNewsCreate$inboundSchema: z.ZodType<
  NegativeNewsCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  negative_news_against_related_parties: z.boolean(),
  negative_news_against_related_parties_description: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "negative_news_against_related_parties":
      "negativeNewsAgainstRelatedParties",
    "negative_news_against_related_parties_description":
      "negativeNewsAgainstRelatedPartiesDescription",
  });
});

/** @internal */
export type NegativeNewsCreate$Outbound = {
  negative_news_against_related_parties: boolean;
  negative_news_against_related_parties_description?: string | undefined;
};

/** @internal */
export const NegativeNewsCreate$outboundSchema: z.ZodType<
  NegativeNewsCreate$Outbound,
  z.ZodTypeDef,
  NegativeNewsCreate
> = z.object({
  negativeNewsAgainstRelatedParties: z.boolean(),
  negativeNewsAgainstRelatedPartiesDescription: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    negativeNewsAgainstRelatedParties: "negative_news_against_related_parties",
    negativeNewsAgainstRelatedPartiesDescription:
      "negative_news_against_related_parties_description",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace NegativeNewsCreate$ {
  /** @deprecated use `NegativeNewsCreate$inboundSchema` instead. */
  export const inboundSchema = NegativeNewsCreate$inboundSchema;
  /** @deprecated use `NegativeNewsCreate$outboundSchema` instead. */
  export const outboundSchema = NegativeNewsCreate$outboundSchema;
  /** @deprecated use `NegativeNewsCreate$Outbound` instead. */
  export type Outbound = NegativeNewsCreate$Outbound;
}
