// @apexfintechsolutions/ascend-sdk/src/models/components/ordersoptionstradingenrollmentmetadatacreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";
import {
  InvestmentVehicleExperienceCreate,
  InvestmentVehicleExperienceCreate$inboundSchema,
  InvestmentVehicleExperienceCreate$Outbound,
  InvestmentVehicleExperienceCreate$outboundSchema,
} from "./investmentvehicleexperiencecreate.js";

/**
 * Requested options level
 */
export enum OptionsLevel {
  OptionsLevelUnspecified = "OPTIONS_LEVEL_UNSPECIFIED",
  Level1BasicDirectionalTrades = "LEVEL_1_BASIC_DIRECTIONAL_TRADES",
}
/**
 * Requested options level
 */
export type OptionsLevelOpen = OpenEnum<typeof OptionsLevel>;

/**
 * Enrollment metadata for the ORDERS_OPTIONS_TRADING enrollment type
 */
export type OrdersOptionsTradingEnrollmentMetadataCreate = {
  /**
   * Investment vehicle experience
   */
  investmentVehicleExperience: InvestmentVehicleExperienceCreate;
  /**
   * Requested options level
   */
  optionsLevel: OptionsLevelOpen;
  /**
   * Total years of options trading experience
   */
  totalYearsOptionsTradingExperience: number;
};

/** @internal */
export const OptionsLevel$inboundSchema: z.ZodType<
  OptionsLevelOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(OptionsLevel),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const OptionsLevel$outboundSchema: z.ZodType<
  OptionsLevelOpen,
  z.ZodTypeDef,
  OptionsLevelOpen
> = z.union([
  z.nativeEnum(OptionsLevel),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace OptionsLevel$ {
  /** @deprecated use `OptionsLevel$inboundSchema` instead. */
  export const inboundSchema = OptionsLevel$inboundSchema;
  /** @deprecated use `OptionsLevel$outboundSchema` instead. */
  export const outboundSchema = OptionsLevel$outboundSchema;
}

/** @internal */
export const OrdersOptionsTradingEnrollmentMetadataCreate$inboundSchema:
  z.ZodType<
    OrdersOptionsTradingEnrollmentMetadataCreate,
    z.ZodTypeDef,
    unknown
  > = z.object({
    investment_vehicle_experience:
      InvestmentVehicleExperienceCreate$inboundSchema,
    options_level: OptionsLevel$inboundSchema,
    total_years_options_trading_experience: z.number().int(),
  }).transform((v) => {
    return remap$(v, {
      "investment_vehicle_experience": "investmentVehicleExperience",
      "options_level": "optionsLevel",
      "total_years_options_trading_experience":
        "totalYearsOptionsTradingExperience",
    });
  });

/** @internal */
export type OrdersOptionsTradingEnrollmentMetadataCreate$Outbound = {
  investment_vehicle_experience: InvestmentVehicleExperienceCreate$Outbound;
  options_level: string;
  total_years_options_trading_experience: number;
};

/** @internal */
export const OrdersOptionsTradingEnrollmentMetadataCreate$outboundSchema:
  z.ZodType<
    OrdersOptionsTradingEnrollmentMetadataCreate$Outbound,
    z.ZodTypeDef,
    OrdersOptionsTradingEnrollmentMetadataCreate
  > = z.object({
    investmentVehicleExperience:
      InvestmentVehicleExperienceCreate$outboundSchema,
    optionsLevel: OptionsLevel$outboundSchema,
    totalYearsOptionsTradingExperience: z.number().int(),
  }).transform((v) => {
    return remap$(v, {
      investmentVehicleExperience: "investment_vehicle_experience",
      optionsLevel: "options_level",
      totalYearsOptionsTradingExperience:
        "total_years_options_trading_experience",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace OrdersOptionsTradingEnrollmentMetadataCreate$ {
  /** @deprecated use `OrdersOptionsTradingEnrollmentMetadataCreate$inboundSchema` instead. */
  export const inboundSchema =
    OrdersOptionsTradingEnrollmentMetadataCreate$inboundSchema;
  /** @deprecated use `OrdersOptionsTradingEnrollmentMetadataCreate$outboundSchema` instead. */
  export const outboundSchema =
    OrdersOptionsTradingEnrollmentMetadataCreate$outboundSchema;
  /** @deprecated use `OrdersOptionsTradingEnrollmentMetadataCreate$Outbound` instead. */
  export type Outbound = OrdersOptionsTradingEnrollmentMetadataCreate$Outbound;
}
