// @apexfintechsolutions/ascend-sdk/src/models/components/relatedpepdetailscreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  RelatedPepCreate,
  RelatedPepCreate$inboundSchema,
  RelatedPepCreate$Outbound,
  RelatedPepCreate$outboundSchema,
} from "./relatedpepcreate.js";

/**
 * Details surrounding the related politically exposed persons
 */
export type RelatedPepDetailsCreate = {
  /**
   * Indication as to whether or not an account has direct or indirect related politically exposed persons
   */
  directOrIndirectRelatedPeps: boolean;
  /**
   * Related Peps
   */
  relatedPeps?: Array<RelatedPepCreate> | undefined;
};

/** @internal */
export const RelatedPepDetailsCreate$inboundSchema: z.ZodType<
  RelatedPepDetailsCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  direct_or_indirect_related_peps: z.boolean(),
  related_peps: z.array(RelatedPepCreate$inboundSchema).optional(),
}).transform((v) => {
  return remap$(v, {
    "direct_or_indirect_related_peps": "directOrIndirectRelatedPeps",
    "related_peps": "relatedPeps",
  });
});

/** @internal */
export type RelatedPepDetailsCreate$Outbound = {
  direct_or_indirect_related_peps: boolean;
  related_peps?: Array<RelatedPepCreate$Outbound> | undefined;
};

/** @internal */
export const RelatedPepDetailsCreate$outboundSchema: z.ZodType<
  RelatedPepDetailsCreate$Outbound,
  z.ZodTypeDef,
  RelatedPepDetailsCreate
> = z.object({
  directOrIndirectRelatedPeps: z.boolean(),
  relatedPeps: z.array(RelatedPepCreate$outboundSchema).optional(),
}).transform((v) => {
  return remap$(v, {
    directOrIndirectRelatedPeps: "direct_or_indirect_related_peps",
    relatedPeps: "related_peps",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RelatedPepDetailsCreate$ {
  /** @deprecated use `RelatedPepDetailsCreate$inboundSchema` instead. */
  export const inboundSchema = RelatedPepDetailsCreate$inboundSchema;
  /** @deprecated use `RelatedPepDetailsCreate$outboundSchema` instead. */
  export const outboundSchema = RelatedPepDetailsCreate$outboundSchema;
  /** @deprecated use `RelatedPepDetailsCreate$Outbound` instead. */
  export type Outbound = RelatedPepDetailsCreate$Outbound;
}
