// @apexfintechsolutions/ascend-sdk/src/models/components/retirementdistributiontaxwithholdingcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import {
  DecimalCreate,
  DecimalCreate$inboundSchema,
  DecimalCreate$Outbound,
  DecimalCreate$outboundSchema,
} from "./decimalcreate.js";

/**
 * A representation of a tax withholding.
 */
export type RetirementDistributionTaxWithholdingCreate = {
  /**
   * A representation of a decimal value, such as 2.5. Clients may convert values into language-native decimal formats, such as Java's [BigDecimal][] or Python's [decimal.Decimal][].
   *
   * @remarks
   *
   *  [BigDecimal]:
   *  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html
   *  [decimal.Decimal]: https://docs.python.org/3/library/decimal.html
   */
  amount?: DecimalCreate | undefined;
  /**
   * A representation of a decimal value, such as 2.5. Clients may convert values into language-native decimal formats, such as Java's [BigDecimal][] or Python's [decimal.Decimal][].
   *
   * @remarks
   *
   *  [BigDecimal]:
   *  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html
   *  [decimal.Decimal]: https://docs.python.org/3/library/decimal.html
   */
  percentage?: DecimalCreate | undefined;
};

/** @internal */
export const RetirementDistributionTaxWithholdingCreate$inboundSchema:
  z.ZodType<RetirementDistributionTaxWithholdingCreate, z.ZodTypeDef, unknown> =
    z.object({
      amount: DecimalCreate$inboundSchema.optional(),
      percentage: DecimalCreate$inboundSchema.optional(),
    });

/** @internal */
export type RetirementDistributionTaxWithholdingCreate$Outbound = {
  amount?: DecimalCreate$Outbound | undefined;
  percentage?: DecimalCreate$Outbound | undefined;
};

/** @internal */
export const RetirementDistributionTaxWithholdingCreate$outboundSchema:
  z.ZodType<
    RetirementDistributionTaxWithholdingCreate$Outbound,
    z.ZodTypeDef,
    RetirementDistributionTaxWithholdingCreate
  > = z.object({
    amount: DecimalCreate$outboundSchema.optional(),
    percentage: DecimalCreate$outboundSchema.optional(),
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetirementDistributionTaxWithholdingCreate$ {
  /** @deprecated use `RetirementDistributionTaxWithholdingCreate$inboundSchema` instead. */
  export const inboundSchema =
    RetirementDistributionTaxWithholdingCreate$inboundSchema;
  /** @deprecated use `RetirementDistributionTaxWithholdingCreate$outboundSchema` instead. */
  export const outboundSchema =
    RetirementDistributionTaxWithholdingCreate$outboundSchema;
  /** @deprecated use `RetirementDistributionTaxWithholdingCreate$Outbound` instead. */
  export type Outbound = RetirementDistributionTaxWithholdingCreate$Outbound;
}
