// @apexfintechsolutions/ascend-sdk/src/models/components/retrievefixedincomemarksrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate,
  RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$inboundSchema,
  RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$Outbound,
  RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$outboundSchema,
} from "./retrievefixedincomemarksrequestsecurityidentifierscreate.js";

/**
 * Request object for retrieving fixed income marks
 */
export type RetrieveFixedIncomeMarksRequestCreate = {
  /**
   * The parent resource where this price will be sourced under. Format: correspondents/{correspondent_id}
   */
  parent: string;
  /**
   * Identifiers specifying for which assets mark data should be returned. A maximum of 100 identifiers are allowed. At least one identifier must be provided in the request.
   */
  securityIdentifiers: Array<
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate
  >;
};

/** @internal */
export const RetrieveFixedIncomeMarksRequestCreate$inboundSchema: z.ZodType<
  RetrieveFixedIncomeMarksRequestCreate,
  z.ZodTypeDef,
  unknown
> = z.object({
  parent: z.string(),
  security_identifiers: z.array(
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$inboundSchema,
  ),
}).transform((v) => {
  return remap$(v, {
    "security_identifiers": "securityIdentifiers",
  });
});

/** @internal */
export type RetrieveFixedIncomeMarksRequestCreate$Outbound = {
  parent: string;
  security_identifiers: Array<
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$Outbound
  >;
};

/** @internal */
export const RetrieveFixedIncomeMarksRequestCreate$outboundSchema: z.ZodType<
  RetrieveFixedIncomeMarksRequestCreate$Outbound,
  z.ZodTypeDef,
  RetrieveFixedIncomeMarksRequestCreate
> = z.object({
  parent: z.string(),
  securityIdentifiers: z.array(
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$outboundSchema,
  ),
}).transform((v) => {
  return remap$(v, {
    securityIdentifiers: "security_identifiers",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetrieveFixedIncomeMarksRequestCreate$ {
  /** @deprecated use `RetrieveFixedIncomeMarksRequestCreate$inboundSchema` instead. */
  export const inboundSchema =
    RetrieveFixedIncomeMarksRequestCreate$inboundSchema;
  /** @deprecated use `RetrieveFixedIncomeMarksRequestCreate$outboundSchema` instead. */
  export const outboundSchema =
    RetrieveFixedIncomeMarksRequestCreate$outboundSchema;
  /** @deprecated use `RetrieveFixedIncomeMarksRequestCreate$Outbound` instead. */
  export type Outbound = RetrieveFixedIncomeMarksRequestCreate$Outbound;
}
