// @apexfintechsolutions/ascend-sdk/src/models/components/retrievefixedincomemarksrequestsecurityidentifierscreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";

/**
 * The identifier type of the asset being sought
 */
export enum RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType {
  AssetId = "ASSET_ID",
  Cusip = "CUSIP",
  Isin = "ISIN",
}
/**
 * The identifier type of the asset being sought
 */
export type RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierTypeOpen =
  OpenEnum<
    typeof RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType
  >;

/**
 * The identifier for the asset.
 */
export type RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate = {
  /**
   * Identifier of the asset (of the type specified in `identifier_type`).
   */
  identifier: string;
  /**
   * The identifier type of the asset being sought
   */
  identifierType:
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierTypeOpen;
};

/** @internal */
export const RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$inboundSchema:
  z.ZodType<
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierTypeOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(
        RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType,
      ),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$outboundSchema:
  z.ZodType<
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierTypeOpen,
    z.ZodTypeDef,
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierTypeOpen
  > = z.union([
    z.nativeEnum(
      RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType,
    ),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$ {
  /** @deprecated use `RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$inboundSchema` instead. */
  export const inboundSchema =
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$inboundSchema;
  /** @deprecated use `RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$outboundSchema` instead. */
  export const outboundSchema =
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$outboundSchema;
}

/** @internal */
export const RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$inboundSchema:
  z.ZodType<
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate,
    z.ZodTypeDef,
    unknown
  > = z.object({
    identifier: z.string(),
    identifier_type:
      RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$inboundSchema,
  }).transform((v) => {
    return remap$(v, {
      "identifier_type": "identifierType",
    });
  });

/** @internal */
export type RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$Outbound =
  {
    identifier: string;
    identifier_type: string;
  };

/** @internal */
export const RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$outboundSchema:
  z.ZodType<
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$Outbound,
    z.ZodTypeDef,
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate
  > = z.object({
    identifier: z.string(),
    identifierType:
      RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      identifierType: "identifier_type",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$ {
  /** @deprecated use `RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$inboundSchema` instead. */
  export const inboundSchema =
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$inboundSchema;
  /** @deprecated use `RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$outboundSchema` instead. */
  export const outboundSchema =
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$outboundSchema;
  /** @deprecated use `RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$Outbound` instead. */
  export type Outbound =
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$Outbound;
}
