// @apexfintechsolutions/ascend-sdk/src/models/components/retrievefixedincomemarksresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  RetrieveFixedIncomeMarksResponseAssetMark,
  RetrieveFixedIncomeMarksResponseAssetMark$inboundSchema,
  RetrieveFixedIncomeMarksResponseAssetMark$Outbound,
  RetrieveFixedIncomeMarksResponseAssetMark$outboundSchema,
} from "./retrievefixedincomemarksresponseassetmark.js";

/**
 * List of objects with the most recently observed mark per security
 */
export type RetrieveFixedIncomeMarksResponse = {
  /**
   * The list of mark data for each of the requested assets found. This may not be returned in the same order as the identifiers were provided, and will only return mark data for the distinct set of assets matching the requested identifiers.
   */
  assetMarks?: Array<RetrieveFixedIncomeMarksResponseAssetMark> | undefined;
};

/** @internal */
export const RetrieveFixedIncomeMarksResponse$inboundSchema: z.ZodType<
  RetrieveFixedIncomeMarksResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  asset_marks: z.array(RetrieveFixedIncomeMarksResponseAssetMark$inboundSchema)
    .optional(),
}).transform((v) => {
  return remap$(v, {
    "asset_marks": "assetMarks",
  });
});

/** @internal */
export type RetrieveFixedIncomeMarksResponse$Outbound = {
  asset_marks?:
    | Array<RetrieveFixedIncomeMarksResponseAssetMark$Outbound>
    | undefined;
};

/** @internal */
export const RetrieveFixedIncomeMarksResponse$outboundSchema: z.ZodType<
  RetrieveFixedIncomeMarksResponse$Outbound,
  z.ZodTypeDef,
  RetrieveFixedIncomeMarksResponse
> = z.object({
  assetMarks: z.array(RetrieveFixedIncomeMarksResponseAssetMark$outboundSchema)
    .optional(),
}).transform((v) => {
  return remap$(v, {
    assetMarks: "asset_marks",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetrieveFixedIncomeMarksResponse$ {
  /** @deprecated use `RetrieveFixedIncomeMarksResponse$inboundSchema` instead. */
  export const inboundSchema = RetrieveFixedIncomeMarksResponse$inboundSchema;
  /** @deprecated use `RetrieveFixedIncomeMarksResponse$outboundSchema` instead. */
  export const outboundSchema = RetrieveFixedIncomeMarksResponse$outboundSchema;
  /** @deprecated use `RetrieveFixedIncomeMarksResponse$Outbound` instead. */
  export type Outbound = RetrieveFixedIncomeMarksResponse$Outbound;
}
