// @apexfintechsolutions/ascend-sdk/src/models/components/retrievedistributionconstraintsrequestcreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";

/**
 * Cash transfer mechanism to search constraints for
 */
export enum RetrieveDistributionConstraintsRequestCreateMechanism {
  Ach = "ACH",
  Ict = "ICT",
}
/**
 * Cash transfer mechanism to search constraints for
 */
export type RetrieveDistributionConstraintsRequestCreateMechanismOpen =
  OpenEnum<typeof RetrieveDistributionConstraintsRequestCreateMechanism>;

/**
 * Request to retrieve retirement distribution constraints
 */
export type RetrieveDistributionConstraintsRequestCreate = {
  /**
   * Cash transfer mechanism to search constraints for
   */
  mechanism: RetrieveDistributionConstraintsRequestCreateMechanismOpen;
  /**
   * Name of the account being queried, for retirement distribution constraints Format: accounts/{account}
   */
  name: string;
};

/** @internal */
export const RetrieveDistributionConstraintsRequestCreateMechanism$inboundSchema:
  z.ZodType<
    RetrieveDistributionConstraintsRequestCreateMechanismOpen,
    z.ZodTypeDef,
    unknown
  > = z
    .union([
      z.nativeEnum(RetrieveDistributionConstraintsRequestCreateMechanism),
      z.string().transform(catchUnrecognizedEnum),
    ]);

/** @internal */
export const RetrieveDistributionConstraintsRequestCreateMechanism$outboundSchema:
  z.ZodType<
    RetrieveDistributionConstraintsRequestCreateMechanismOpen,
    z.ZodTypeDef,
    RetrieveDistributionConstraintsRequestCreateMechanismOpen
  > = z.union([
    z.nativeEnum(RetrieveDistributionConstraintsRequestCreateMechanism),
    z.string().and(z.custom<Unrecognized<string>>()),
  ]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetrieveDistributionConstraintsRequestCreateMechanism$ {
  /** @deprecated use `RetrieveDistributionConstraintsRequestCreateMechanism$inboundSchema` instead. */
  export const inboundSchema =
    RetrieveDistributionConstraintsRequestCreateMechanism$inboundSchema;
  /** @deprecated use `RetrieveDistributionConstraintsRequestCreateMechanism$outboundSchema` instead. */
  export const outboundSchema =
    RetrieveDistributionConstraintsRequestCreateMechanism$outboundSchema;
}

/** @internal */
export const RetrieveDistributionConstraintsRequestCreate$inboundSchema:
  z.ZodType<
    RetrieveDistributionConstraintsRequestCreate,
    z.ZodTypeDef,
    unknown
  > = z.object({
    mechanism:
      RetrieveDistributionConstraintsRequestCreateMechanism$inboundSchema,
    name: z.string(),
  });

/** @internal */
export type RetrieveDistributionConstraintsRequestCreate$Outbound = {
  mechanism: string;
  name: string;
};

/** @internal */
export const RetrieveDistributionConstraintsRequestCreate$outboundSchema:
  z.ZodType<
    RetrieveDistributionConstraintsRequestCreate$Outbound,
    z.ZodTypeDef,
    RetrieveDistributionConstraintsRequestCreate
  > = z.object({
    mechanism:
      RetrieveDistributionConstraintsRequestCreateMechanism$outboundSchema,
    name: z.string(),
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetrieveDistributionConstraintsRequestCreate$ {
  /** @deprecated use `RetrieveDistributionConstraintsRequestCreate$inboundSchema` instead. */
  export const inboundSchema =
    RetrieveDistributionConstraintsRequestCreate$inboundSchema;
  /** @deprecated use `RetrieveDistributionConstraintsRequestCreate$outboundSchema` instead. */
  export const outboundSchema =
    RetrieveDistributionConstraintsRequestCreate$outboundSchema;
  /** @deprecated use `RetrieveDistributionConstraintsRequestCreate$Outbound` instead. */
  export type Outbound = RetrieveDistributionConstraintsRequestCreate$Outbound;
}
