// @apexfintechsolutions/ascend-sdk/src/models/components/trade.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import {
  catchUnrecognizedEnum,
  OpenEnum,
  Unrecognized,
} from "../../types/enums.js";
import {
  Lot,
  Lot$inboundSchema,
  Lot$Outbound,
  Lot$outboundSchema,
} from "./lot.js";
import {
  YieldRecord,
  YieldRecord$inboundSchema,
  YieldRecord$Outbound,
  YieldRecord$outboundSchema,
} from "./yieldrecord.js";

/**
 * Used to calculate broadridge blotter code
 */
export enum TradeBrokerCapacity {
  CapacityUnspecified = "CAPACITY_UNSPECIFIED",
  Agency = "AGENCY",
  Principal = "PRINCIPAL",
  Mixed = "MIXED",
}
/**
 * Used to calculate broadridge blotter code
 */
export type TradeBrokerCapacityOpen = OpenEnum<typeof TradeBrokerCapacity>;

/**
 * The price for the instrument that is prevailing in the market.
 */
export type TradePrevailingMarketPrice = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/**
 * Total monetary value of the price_adjustment
 */
export type PriceAdjustmentAmount = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/**
 * The percent at which the price was adjusted. Expressed as a number from 0.00-100 (rounded to 2 decimals)
 */
export type PriceAdjustmentPercent = {
  /**
   * The decimal value, as a string; Refer to [Google’s Decimal type protocol buffer](https://github.com/googleapis/googleapis/blob/40203ca1880849480bbff7b8715491060bbccdf1/google/type/decimal.proto#L33) for details
   */
  value?: string | undefined;
};

/**
 * The type of price adjustment being applied by the broker to the net price of the security
 */
export enum TradePriceAdjustmentType {
  PriceAdjustmentTypeUnspecified = "PRICE_ADJUSTMENT_TYPE_UNSPECIFIED",
  Markup = "MARKUP",
  Markdown = "MARKDOWN",
  SalesLoad = "SALES_LOAD",
}
/**
 * The type of price adjustment being applied by the broker to the net price of the security
 */
export type TradePriceAdjustmentTypeOpen = OpenEnum<
  typeof TradePriceAdjustmentType
>;

/**
 * Information about any price adjustments applied to the security
 */
export type PriceAdjustmentRecord = {
  /**
   * Total monetary value of the price_adjustment
   */
  priceAdjustmentAmount?: PriceAdjustmentAmount | null | undefined;
  /**
   * The percent at which the price was adjusted. Expressed as a number from 0.00-100 (rounded to 2 decimals)
   */
  priceAdjustmentPercent?: PriceAdjustmentPercent | null | undefined;
  /**
   * The type of price adjustment being applied by the broker to the net price of the security
   */
  priceAdjustmentType?: TradePriceAdjustmentTypeOpen | undefined;
};

export type Trade = {
  /**
   * Free form text field containing additional information for a trade
   */
  additionalInstructions?: Array<string> | undefined;
  /**
   * Max Length 100 characters. Alternate order id from the street used for FRAC trades
   */
  alternateOrderId?: string | undefined;
  /**
   * uuid assigned by the Booking API if a trade belongs to an allocation
   */
  bookingApiTradeAllocationId?: string | undefined;
  /**
   * uuid assigned by the Booking API to all trades
   */
  bookingApiTradeId?: string | undefined;
  /**
   * Executing broker of the trade
   */
  broker?: string | undefined;
  /**
   * Used to calculate broadridge blotter code
   */
  brokerCapacity?: TradeBrokerCapacityOpen | undefined;
  /**
   * Free form text submitted by the client for internal purposes
   */
  clientMemos?: Array<string> | undefined;
  /**
   * 32 characters. The client order ID from the order submitted
   */
  clientOrderId?: string | undefined;
  /**
   * MIC code for the exchange
   */
  exchange?: string | undefined;
  /**
   * Execution id from the street
   */
  executionId?: string | undefined;
  /**
   * Indicates whether Apex is the clearing broker for this trade. When false, indicates Apex is the clearing broker
   */
  executionOnly?: boolean | undefined;
  /**
   * Max Length 100 characters. External system id provided by a client
   */
  externalId?: string | undefined;
  /**
   * The confirmation number associated with a mutual fund trade
   */
  fundConfirmationNumber?: string | undefined;
  /**
   * Max length 100 characters. Order id generated by trading-gateway (Trade-Ex) to uniquely identify all orders in their system. Used as the client_order_id on new order singles sent downstream of the trading-gateway
   */
  gatewayClientOrderId?: string | undefined;
  /**
   * If set to true, indicates the trade should be omitted from client billing
   */
  internalError?: boolean | undefined;
  /**
   * Set on penny-for-the-lot trades
   */
  isWriteoff?: boolean | undefined;
  /**
   * Repeated record containing information about the tax lots, if specified
   */
  lots?: Array<Lot> | undefined;
  /**
   * Max Length 100 characters. Internally generated order id that is returned to client on exec reports
   */
  orderId?: string | undefined;
  /**
   * The price for the instrument that is prevailing in the market.
   */
  prevailingMarketPrice?: TradePrevailingMarketPrice | null | undefined;
  /**
   * Information about any price adjustments applied to the security
   */
  priceAdjustmentRecord?: PriceAdjustmentRecord | null | undefined;
  /**
   * Used to calculate broadridge blotter code
   */
  route?: string | undefined;
  /**
   * The special instructions for a trade
   */
  specialInstructions?: Array<string> | undefined;
  /**
   * Name of the issuer of a security and additional descriptive information about the particular issue
   */
  symbolDescription?: string | undefined;
  /**
   * Indicates the trade was executed in a security that is not currently listed. When-issued securities are bought and sold before they are officially issued, allowing investors to speculate on their future value
   */
  whenIssued?: boolean | undefined;
  /**
   * The yields associated with a fixed income trade. only valid if the SecurityType is FIXED_INCOME.
   */
  yieldRecords?: Array<YieldRecord> | undefined;
};

/** @internal */
export const TradeBrokerCapacity$inboundSchema: z.ZodType<
  TradeBrokerCapacityOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(TradeBrokerCapacity),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const TradeBrokerCapacity$outboundSchema: z.ZodType<
  TradeBrokerCapacityOpen,
  z.ZodTypeDef,
  TradeBrokerCapacityOpen
> = z.union([
  z.nativeEnum(TradeBrokerCapacity),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TradeBrokerCapacity$ {
  /** @deprecated use `TradeBrokerCapacity$inboundSchema` instead. */
  export const inboundSchema = TradeBrokerCapacity$inboundSchema;
  /** @deprecated use `TradeBrokerCapacity$outboundSchema` instead. */
  export const outboundSchema = TradeBrokerCapacity$outboundSchema;
}

/** @internal */
export const TradePrevailingMarketPrice$inboundSchema: z.ZodType<
  TradePrevailingMarketPrice,
  z.ZodTypeDef,
  unknown
> = z.object({
  value: z.string().optional(),
});

/** @internal */
export type TradePrevailingMarketPrice$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const TradePrevailingMarketPrice$outboundSchema: z.ZodType<
  TradePrevailingMarketPrice$Outbound,
  z.ZodTypeDef,
  TradePrevailingMarketPrice
> = z.object({
  value: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TradePrevailingMarketPrice$ {
  /** @deprecated use `TradePrevailingMarketPrice$inboundSchema` instead. */
  export const inboundSchema = TradePrevailingMarketPrice$inboundSchema;
  /** @deprecated use `TradePrevailingMarketPrice$outboundSchema` instead. */
  export const outboundSchema = TradePrevailingMarketPrice$outboundSchema;
  /** @deprecated use `TradePrevailingMarketPrice$Outbound` instead. */
  export type Outbound = TradePrevailingMarketPrice$Outbound;
}

/** @internal */
export const PriceAdjustmentAmount$inboundSchema: z.ZodType<
  PriceAdjustmentAmount,
  z.ZodTypeDef,
  unknown
> = z.object({
  value: z.string().optional(),
});

/** @internal */
export type PriceAdjustmentAmount$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const PriceAdjustmentAmount$outboundSchema: z.ZodType<
  PriceAdjustmentAmount$Outbound,
  z.ZodTypeDef,
  PriceAdjustmentAmount
> = z.object({
  value: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace PriceAdjustmentAmount$ {
  /** @deprecated use `PriceAdjustmentAmount$inboundSchema` instead. */
  export const inboundSchema = PriceAdjustmentAmount$inboundSchema;
  /** @deprecated use `PriceAdjustmentAmount$outboundSchema` instead. */
  export const outboundSchema = PriceAdjustmentAmount$outboundSchema;
  /** @deprecated use `PriceAdjustmentAmount$Outbound` instead. */
  export type Outbound = PriceAdjustmentAmount$Outbound;
}

/** @internal */
export const PriceAdjustmentPercent$inboundSchema: z.ZodType<
  PriceAdjustmentPercent,
  z.ZodTypeDef,
  unknown
> = z.object({
  value: z.string().optional(),
});

/** @internal */
export type PriceAdjustmentPercent$Outbound = {
  value?: string | undefined;
};

/** @internal */
export const PriceAdjustmentPercent$outboundSchema: z.ZodType<
  PriceAdjustmentPercent$Outbound,
  z.ZodTypeDef,
  PriceAdjustmentPercent
> = z.object({
  value: z.string().optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace PriceAdjustmentPercent$ {
  /** @deprecated use `PriceAdjustmentPercent$inboundSchema` instead. */
  export const inboundSchema = PriceAdjustmentPercent$inboundSchema;
  /** @deprecated use `PriceAdjustmentPercent$outboundSchema` instead. */
  export const outboundSchema = PriceAdjustmentPercent$outboundSchema;
  /** @deprecated use `PriceAdjustmentPercent$Outbound` instead. */
  export type Outbound = PriceAdjustmentPercent$Outbound;
}

/** @internal */
export const TradePriceAdjustmentType$inboundSchema: z.ZodType<
  TradePriceAdjustmentTypeOpen,
  z.ZodTypeDef,
  unknown
> = z
  .union([
    z.nativeEnum(TradePriceAdjustmentType),
    z.string().transform(catchUnrecognizedEnum),
  ]);

/** @internal */
export const TradePriceAdjustmentType$outboundSchema: z.ZodType<
  TradePriceAdjustmentTypeOpen,
  z.ZodTypeDef,
  TradePriceAdjustmentTypeOpen
> = z.union([
  z.nativeEnum(TradePriceAdjustmentType),
  z.string().and(z.custom<Unrecognized<string>>()),
]);

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TradePriceAdjustmentType$ {
  /** @deprecated use `TradePriceAdjustmentType$inboundSchema` instead. */
  export const inboundSchema = TradePriceAdjustmentType$inboundSchema;
  /** @deprecated use `TradePriceAdjustmentType$outboundSchema` instead. */
  export const outboundSchema = TradePriceAdjustmentType$outboundSchema;
}

/** @internal */
export const PriceAdjustmentRecord$inboundSchema: z.ZodType<
  PriceAdjustmentRecord,
  z.ZodTypeDef,
  unknown
> = z.object({
  price_adjustment_amount: z.nullable(
    z.lazy(() => PriceAdjustmentAmount$inboundSchema),
  ).optional(),
  price_adjustment_percent: z.nullable(
    z.lazy(() => PriceAdjustmentPercent$inboundSchema),
  ).optional(),
  price_adjustment_type: TradePriceAdjustmentType$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "price_adjustment_amount": "priceAdjustmentAmount",
    "price_adjustment_percent": "priceAdjustmentPercent",
    "price_adjustment_type": "priceAdjustmentType",
  });
});

/** @internal */
export type PriceAdjustmentRecord$Outbound = {
  price_adjustment_amount?: PriceAdjustmentAmount$Outbound | null | undefined;
  price_adjustment_percent?: PriceAdjustmentPercent$Outbound | null | undefined;
  price_adjustment_type?: string | undefined;
};

/** @internal */
export const PriceAdjustmentRecord$outboundSchema: z.ZodType<
  PriceAdjustmentRecord$Outbound,
  z.ZodTypeDef,
  PriceAdjustmentRecord
> = z.object({
  priceAdjustmentAmount: z.nullable(
    z.lazy(() => PriceAdjustmentAmount$outboundSchema),
  ).optional(),
  priceAdjustmentPercent: z.nullable(
    z.lazy(() => PriceAdjustmentPercent$outboundSchema),
  ).optional(),
  priceAdjustmentType: TradePriceAdjustmentType$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    priceAdjustmentAmount: "price_adjustment_amount",
    priceAdjustmentPercent: "price_adjustment_percent",
    priceAdjustmentType: "price_adjustment_type",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace PriceAdjustmentRecord$ {
  /** @deprecated use `PriceAdjustmentRecord$inboundSchema` instead. */
  export const inboundSchema = PriceAdjustmentRecord$inboundSchema;
  /** @deprecated use `PriceAdjustmentRecord$outboundSchema` instead. */
  export const outboundSchema = PriceAdjustmentRecord$outboundSchema;
  /** @deprecated use `PriceAdjustmentRecord$Outbound` instead. */
  export type Outbound = PriceAdjustmentRecord$Outbound;
}

/** @internal */
export const Trade$inboundSchema: z.ZodType<Trade, z.ZodTypeDef, unknown> = z
  .object({
    additional_instructions: z.array(z.string()).optional(),
    alternate_order_id: z.string().optional(),
    booking_api_trade_allocation_id: z.string().optional(),
    booking_api_trade_id: z.string().optional(),
    broker: z.string().optional(),
    broker_capacity: TradeBrokerCapacity$inboundSchema.optional(),
    client_memos: z.array(z.string()).optional(),
    client_order_id: z.string().optional(),
    exchange: z.string().optional(),
    execution_id: z.string().optional(),
    execution_only: z.boolean().optional(),
    external_id: z.string().optional(),
    fund_confirmation_number: z.string().optional(),
    gateway_client_order_id: z.string().optional(),
    internal_error: z.boolean().optional(),
    is_writeoff: z.boolean().optional(),
    lots: z.array(Lot$inboundSchema).optional(),
    order_id: z.string().optional(),
    prevailing_market_price: z.nullable(
      z.lazy(() => TradePrevailingMarketPrice$inboundSchema),
    ).optional(),
    price_adjustment_record: z.nullable(
      z.lazy(() => PriceAdjustmentRecord$inboundSchema),
    ).optional(),
    route: z.string().optional(),
    special_instructions: z.array(z.string()).optional(),
    symbol_description: z.string().optional(),
    when_issued: z.boolean().optional(),
    yield_records: z.array(YieldRecord$inboundSchema).optional(),
  }).transform((v) => {
    return remap$(v, {
      "additional_instructions": "additionalInstructions",
      "alternate_order_id": "alternateOrderId",
      "booking_api_trade_allocation_id": "bookingApiTradeAllocationId",
      "booking_api_trade_id": "bookingApiTradeId",
      "broker_capacity": "brokerCapacity",
      "client_memos": "clientMemos",
      "client_order_id": "clientOrderId",
      "execution_id": "executionId",
      "execution_only": "executionOnly",
      "external_id": "externalId",
      "fund_confirmation_number": "fundConfirmationNumber",
      "gateway_client_order_id": "gatewayClientOrderId",
      "internal_error": "internalError",
      "is_writeoff": "isWriteoff",
      "order_id": "orderId",
      "prevailing_market_price": "prevailingMarketPrice",
      "price_adjustment_record": "priceAdjustmentRecord",
      "special_instructions": "specialInstructions",
      "symbol_description": "symbolDescription",
      "when_issued": "whenIssued",
      "yield_records": "yieldRecords",
    });
  });

/** @internal */
export type Trade$Outbound = {
  additional_instructions?: Array<string> | undefined;
  alternate_order_id?: string | undefined;
  booking_api_trade_allocation_id?: string | undefined;
  booking_api_trade_id?: string | undefined;
  broker?: string | undefined;
  broker_capacity?: string | undefined;
  client_memos?: Array<string> | undefined;
  client_order_id?: string | undefined;
  exchange?: string | undefined;
  execution_id?: string | undefined;
  execution_only?: boolean | undefined;
  external_id?: string | undefined;
  fund_confirmation_number?: string | undefined;
  gateway_client_order_id?: string | undefined;
  internal_error?: boolean | undefined;
  is_writeoff?: boolean | undefined;
  lots?: Array<Lot$Outbound> | undefined;
  order_id?: string | undefined;
  prevailing_market_price?:
    | TradePrevailingMarketPrice$Outbound
    | null
    | undefined;
  price_adjustment_record?: PriceAdjustmentRecord$Outbound | null | undefined;
  route?: string | undefined;
  special_instructions?: Array<string> | undefined;
  symbol_description?: string | undefined;
  when_issued?: boolean | undefined;
  yield_records?: Array<YieldRecord$Outbound> | undefined;
};

/** @internal */
export const Trade$outboundSchema: z.ZodType<
  Trade$Outbound,
  z.ZodTypeDef,
  Trade
> = z.object({
  additionalInstructions: z.array(z.string()).optional(),
  alternateOrderId: z.string().optional(),
  bookingApiTradeAllocationId: z.string().optional(),
  bookingApiTradeId: z.string().optional(),
  broker: z.string().optional(),
  brokerCapacity: TradeBrokerCapacity$outboundSchema.optional(),
  clientMemos: z.array(z.string()).optional(),
  clientOrderId: z.string().optional(),
  exchange: z.string().optional(),
  executionId: z.string().optional(),
  executionOnly: z.boolean().optional(),
  externalId: z.string().optional(),
  fundConfirmationNumber: z.string().optional(),
  gatewayClientOrderId: z.string().optional(),
  internalError: z.boolean().optional(),
  isWriteoff: z.boolean().optional(),
  lots: z.array(Lot$outboundSchema).optional(),
  orderId: z.string().optional(),
  prevailingMarketPrice: z.nullable(
    z.lazy(() => TradePrevailingMarketPrice$outboundSchema),
  ).optional(),
  priceAdjustmentRecord: z.nullable(
    z.lazy(() => PriceAdjustmentRecord$outboundSchema),
  ).optional(),
  route: z.string().optional(),
  specialInstructions: z.array(z.string()).optional(),
  symbolDescription: z.string().optional(),
  whenIssued: z.boolean().optional(),
  yieldRecords: z.array(YieldRecord$outboundSchema).optional(),
}).transform((v) => {
  return remap$(v, {
    additionalInstructions: "additional_instructions",
    alternateOrderId: "alternate_order_id",
    bookingApiTradeAllocationId: "booking_api_trade_allocation_id",
    bookingApiTradeId: "booking_api_trade_id",
    brokerCapacity: "broker_capacity",
    clientMemos: "client_memos",
    clientOrderId: "client_order_id",
    executionId: "execution_id",
    executionOnly: "execution_only",
    externalId: "external_id",
    fundConfirmationNumber: "fund_confirmation_number",
    gatewayClientOrderId: "gateway_client_order_id",
    internalError: "internal_error",
    isWriteoff: "is_writeoff",
    orderId: "order_id",
    prevailingMarketPrice: "prevailing_market_price",
    priceAdjustmentRecord: "price_adjustment_record",
    specialInstructions: "special_instructions",
    symbolDescription: "symbol_description",
    whenIssued: "when_issued",
    yieldRecords: "yield_records",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace Trade$ {
  /** @deprecated use `Trade$inboundSchema` instead. */
  export const inboundSchema = Trade$inboundSchema;
  /** @deprecated use `Trade$outboundSchema` instead. */
  export const outboundSchema = Trade$outboundSchema;
  /** @deprecated use `Trade$Outbound` instead. */
  export type Outbound = Trade$Outbound;
}
