// @apexfintechsolutions/ascend-sdk/src/models/components/virtualaccountnumberenrollmentmetadatacreate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";

/**
 * Enrollment metadata for the VIRTUAL_ACCOUNT_NUMBER enrollment type
 */
export type VirtualAccountNumberEnrollmentMetadataCreate = {};

/** @internal */
export const VirtualAccountNumberEnrollmentMetadataCreate$inboundSchema:
  z.ZodType<
    VirtualAccountNumberEnrollmentMetadataCreate,
    z.ZodTypeDef,
    unknown
  > = z.object({});

/** @internal */
export type VirtualAccountNumberEnrollmentMetadataCreate$Outbound = {};

/** @internal */
export const VirtualAccountNumberEnrollmentMetadataCreate$outboundSchema:
  z.ZodType<
    VirtualAccountNumberEnrollmentMetadataCreate$Outbound,
    z.ZodTypeDef,
    VirtualAccountNumberEnrollmentMetadataCreate
  > = z.object({});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace VirtualAccountNumberEnrollmentMetadataCreate$ {
  /** @deprecated use `VirtualAccountNumberEnrollmentMetadataCreate$inboundSchema` instead. */
  export const inboundSchema =
    VirtualAccountNumberEnrollmentMetadataCreate$inboundSchema;
  /** @deprecated use `VirtualAccountNumberEnrollmentMetadataCreate$outboundSchema` instead. */
  export const outboundSchema =
    VirtualAccountNumberEnrollmentMetadataCreate$outboundSchema;
  /** @deprecated use `VirtualAccountNumberEnrollmentMetadataCreate$Outbound` instead. */
  export type Outbound = VirtualAccountNumberEnrollmentMetadataCreate$Outbound;
}
