// @apexfintechsolutions/ascend-sdk/src/models/components/withdrawalscheduledetailsupdate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import {
  DecimalUpdate,
  DecimalUpdate$inboundSchema,
  DecimalUpdate$Outbound,
  DecimalUpdate$outboundSchema,
} from "./decimalupdate.js";

/**
 * Details of withdrawal schedule transfers
 */
export type WithdrawalScheduleDetailsUpdate = {
  /**
   * A representation of a decimal value, such as 2.5. Clients may convert values into language-native decimal formats, such as Java's [BigDecimal][] or Python's [decimal.Decimal][].
   *
   * @remarks
   *
   *  [BigDecimal]:
   *  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/math/BigDecimal.html
   *  [decimal.Decimal]: https://docs.python.org/3/library/decimal.html
   */
  amount?: DecimalUpdate | undefined;
};

/** @internal */
export const WithdrawalScheduleDetailsUpdate$inboundSchema: z.ZodType<
  WithdrawalScheduleDetailsUpdate,
  z.ZodTypeDef,
  unknown
> = z.object({
  amount: DecimalUpdate$inboundSchema.optional(),
});

/** @internal */
export type WithdrawalScheduleDetailsUpdate$Outbound = {
  amount?: DecimalUpdate$Outbound | undefined;
};

/** @internal */
export const WithdrawalScheduleDetailsUpdate$outboundSchema: z.ZodType<
  WithdrawalScheduleDetailsUpdate$Outbound,
  z.ZodTypeDef,
  WithdrawalScheduleDetailsUpdate
> = z.object({
  amount: DecimalUpdate$outboundSchema.optional(),
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace WithdrawalScheduleDetailsUpdate$ {
  /** @deprecated use `WithdrawalScheduleDetailsUpdate$inboundSchema` instead. */
  export const inboundSchema = WithdrawalScheduleDetailsUpdate$inboundSchema;
  /** @deprecated use `WithdrawalScheduleDetailsUpdate$outboundSchema` instead. */
  export const outboundSchema = WithdrawalScheduleDetailsUpdate$outboundSchema;
  /** @deprecated use `WithdrawalScheduleDetailsUpdate$Outbound` instead. */
  export type Outbound = WithdrawalScheduleDetailsUpdate$Outbound;
}
