// @apexfintechsolutions/ascend-sdk/models/components/accountrequestupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountRequestUpdate$ = exports.AccountRequestUpdate$outboundSchema = exports.AccountRequestUpdate$inboundSchema = exports.AccountRequestUpdateCatAccountHolderType$ = exports.AccountRequestUpdateCatAccountHolderType$outboundSchema = exports.AccountRequestUpdateCatAccountHolderType$inboundSchema = exports.AccountRequestUpdateCatAccountHolderType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const accounttaxprofileupdate_js_1 = require("./accounttaxprofileupdate.js");
const identifierupdate_js_1 = require("./identifierupdate.js");
const interestedpartyupdate_js_1 = require("./interestedpartyupdate.js");
const investmentprofileupdate_js_1 = require("./investmentprofileupdate.js");
const partyrequestupdate_js_1 = require("./partyrequestupdate.js");
const trustedcontactupdate_js_1 = require("./trustedcontactupdate.js");
/**
 * The FINRA CAT classification for the Account Holder; Is set automatically based on attributes of the owners and account type
 */
var AccountRequestUpdateCatAccountHolderType;
(function (AccountRequestUpdateCatAccountHolderType) {
    AccountRequestUpdateCatAccountHolderType["CatAccountHolderTypeUnspecified"] = "CAT_ACCOUNT_HOLDER_TYPE_UNSPECIFIED";
    AccountRequestUpdateCatAccountHolderType["AInstitutionalCustomer"] = "A_INSTITUTIONAL_CUSTOMER";
    AccountRequestUpdateCatAccountHolderType["EEmployeeAccount"] = "E_EMPLOYEE_ACCOUNT";
    AccountRequestUpdateCatAccountHolderType["FForeign"] = "F_FOREIGN";
    AccountRequestUpdateCatAccountHolderType["IIndividual"] = "I_INDIVIDUAL";
    AccountRequestUpdateCatAccountHolderType["OMarketMaking"] = "O_MARKET_MAKING";
    AccountRequestUpdateCatAccountHolderType["VAgencyAveragePriceAccount"] = "V_AGENCY_AVERAGE_PRICE_ACCOUNT";
    AccountRequestUpdateCatAccountHolderType["POtherProprietary"] = "P_OTHER_PROPRIETARY";
    AccountRequestUpdateCatAccountHolderType["XErrorAccount"] = "X_ERROR_ACCOUNT";
})(AccountRequestUpdateCatAccountHolderType || (exports.AccountRequestUpdateCatAccountHolderType = AccountRequestUpdateCatAccountHolderType = {}));
/** @internal */
exports.AccountRequestUpdateCatAccountHolderType$inboundSchema = z
    .union([
    z.nativeEnum(AccountRequestUpdateCatAccountHolderType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.AccountRequestUpdateCatAccountHolderType$outboundSchema = z.union([
    z.nativeEnum(AccountRequestUpdateCatAccountHolderType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountRequestUpdateCatAccountHolderType$;
(function (AccountRequestUpdateCatAccountHolderType$) {
    /** @deprecated use `AccountRequestUpdateCatAccountHolderType$inboundSchema` instead. */
    AccountRequestUpdateCatAccountHolderType$.inboundSchema = exports.AccountRequestUpdateCatAccountHolderType$inboundSchema;
    /** @deprecated use `AccountRequestUpdateCatAccountHolderType$outboundSchema` instead. */
    AccountRequestUpdateCatAccountHolderType$.outboundSchema = exports.AccountRequestUpdateCatAccountHolderType$outboundSchema;
})(AccountRequestUpdateCatAccountHolderType$ || (exports.AccountRequestUpdateCatAccountHolderType$ = AccountRequestUpdateCatAccountHolderType$ = {}));
/** @internal */
exports.AccountRequestUpdate$inboundSchema = z.object({
    accepts_issuer_direct_communication: z.boolean().optional(),
    advised: z.boolean().optional(),
    cat_account_holder_type: exports.AccountRequestUpdateCatAccountHolderType$inboundSchema.optional(),
    identifiers: z.array(identifierupdate_js_1.IdentifierUpdate$inboundSchema).optional(),
    interested_parties: z.array(interestedpartyupdate_js_1.InterestedPartyUpdate$inboundSchema).optional(),
    investment_profile: investmentprofileupdate_js_1.InvestmentProfileUpdate$inboundSchema.optional(),
    managed: z.boolean().optional(),
    parties: z.array(partyrequestupdate_js_1.PartyRequestUpdate$inboundSchema).optional(),
    primary_registered_rep_id: z.string().optional(),
    tax_profile: accounttaxprofileupdate_js_1.AccountTaxProfileUpdate$inboundSchema.optional(),
    trusted_contacts: z.array(trustedcontactupdate_js_1.TrustedContactUpdate$inboundSchema).optional(),
    wrap_fee_billed: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "accepts_issuer_direct_communication": "acceptsIssuerDirectCommunication",
        "cat_account_holder_type": "catAccountHolderType",
        "interested_parties": "interestedParties",
        "investment_profile": "investmentProfile",
        "primary_registered_rep_id": "primaryRegisteredRepId",
        "tax_profile": "taxProfile",
        "trusted_contacts": "trustedContacts",
        "wrap_fee_billed": "wrapFeeBilled",
    });
});
/** @internal */
exports.AccountRequestUpdate$outboundSchema = z.object({
    acceptsIssuerDirectCommunication: z.boolean().optional(),
    advised: z.boolean().optional(),
    catAccountHolderType: exports.AccountRequestUpdateCatAccountHolderType$outboundSchema
        .optional(),
    identifiers: z.array(identifierupdate_js_1.IdentifierUpdate$outboundSchema).optional(),
    interestedParties: z.array(interestedpartyupdate_js_1.InterestedPartyUpdate$outboundSchema).optional(),
    investmentProfile: investmentprofileupdate_js_1.InvestmentProfileUpdate$outboundSchema.optional(),
    managed: z.boolean().optional(),
    parties: z.array(partyrequestupdate_js_1.PartyRequestUpdate$outboundSchema).optional(),
    primaryRegisteredRepId: z.string().optional(),
    taxProfile: accounttaxprofileupdate_js_1.AccountTaxProfileUpdate$outboundSchema.optional(),
    trustedContacts: z.array(trustedcontactupdate_js_1.TrustedContactUpdate$outboundSchema).optional(),
    wrapFeeBilled: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        acceptsIssuerDirectCommunication: "accepts_issuer_direct_communication",
        catAccountHolderType: "cat_account_holder_type",
        interestedParties: "interested_parties",
        investmentProfile: "investment_profile",
        primaryRegisteredRepId: "primary_registered_rep_id",
        taxProfile: "tax_profile",
        trustedContacts: "trusted_contacts",
        wrapFeeBilled: "wrap_fee_billed",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountRequestUpdate$;
(function (AccountRequestUpdate$) {
    /** @deprecated use `AccountRequestUpdate$inboundSchema` instead. */
    AccountRequestUpdate$.inboundSchema = exports.AccountRequestUpdate$inboundSchema;
    /** @deprecated use `AccountRequestUpdate$outboundSchema` instead. */
    AccountRequestUpdate$.outboundSchema = exports.AccountRequestUpdate$outboundSchema;
})(AccountRequestUpdate$ || (exports.AccountRequestUpdate$ = AccountRequestUpdate$ = {}));
