// @apexfintechsolutions/ascend-sdk/models/components/accountdocumentuploadrequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountDocumentUploadRequestCreate$ = exports.AccountDocumentUploadRequestCreate$outboundSchema = exports.AccountDocumentUploadRequestCreate$inboundSchema = exports.DocumentType$ = exports.DocumentType$outboundSchema = exports.DocumentType$inboundSchema = exports.DocumentType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Describes the contents of a document and how it is used; Required for all account documents
 */
var DocumentType;
(function (DocumentType) {
    DocumentType["AccountDocumentTypeUnspecified"] = "ACCOUNT_DOCUMENT_TYPE_UNSPECIFIED";
    DocumentType["CertificateOfAppointment"] = "CERTIFICATE_OF_APPOINTMENT";
    DocumentType["FdicSweepProgramAgreement"] = "FDIC_SWEEP_PROGRAM_AGREEMENT";
    DocumentType["FpslMasterServicesLoanAgreement"] = "FPSL_MASTER_SERVICES_LOAN_AGREEMENT";
    DocumentType["TransferOnDeathBeneficiaryDesignation"] = "TRANSFER_ON_DEATH_BENEFICIARY_DESIGNATION";
    DocumentType["ChangeOfAddress"] = "CHANGE_OF_ADDRESS";
    DocumentType["NewAccountForm"] = "NEW_ACCOUNT_FORM";
    DocumentType["Letter3210"] = "LETTER_3210";
    DocumentType["IraApplForm"] = "IRA_APPL_FORM";
    DocumentType["RiaIraApplForm"] = "RIA_IRA_APPL_FORM";
    DocumentType["IraDepositForm"] = "IRA_DEPOSIT_FORM";
    DocumentType["RiaIraDepositForm"] = "RIA_IRA_DEPOSIT_FORM";
    DocumentType["IraDistForm"] = "IRA_DIST_FORM";
    DocumentType["RiaIraDistForm"] = "RIA_IRA_DIST_FORM";
    DocumentType["RothApplForm"] = "ROTH_APPL_FORM";
    DocumentType["RiaRothApplForm"] = "RIA_ROTH_APPL_FORM";
    DocumentType["RothDistForm"] = "ROTH_DIST_FORM";
    DocumentType["RiaRothDistForm"] = "RIA_ROTH_DIST_FORM";
    DocumentType["SimpleContTransm"] = "SIMPLE_CONT_TRANSM";
    DocumentType["EsaApplForm"] = "ESA_APPL_FORM";
    DocumentType["EsaWthdrStatement"] = "ESA_WTHDR_STATEMENT";
    DocumentType["DirConvForm"] = "DIR_CONV_FORM";
    DocumentType["IraBeneForm"] = "IRA_BENE_FORM";
    DocumentType["IraJrnlForm"] = "IRA_JRNL_FORM";
    DocumentType["SepEmplPlanDoc"] = "SEP_EMPL_PLAN_DOC";
    DocumentType["SimpleAdoptAgreement"] = "SIMPLE_ADOPT_AGREEMENT";
    DocumentType["SimpleApplForm"] = "SIMPLE_APPL_FORM";
    DocumentType["DeathCertification"] = "DEATH_CERTIFICATION";
    DocumentType["AchAgreement"] = "ACH_AGREEMENT";
    DocumentType["ApexSelectApplication"] = "APEX_SELECT_APPLICATION";
    DocumentType["AsaDirectDeposit"] = "ASA_DIRECT_DEPOSIT";
    DocumentType["BankingMisc"] = "BANKING_MISC";
    DocumentType["AccTransferForm"] = "ACC_TRANSFER_FORM";
    DocumentType["AddressVerificationNotice"] = "ADDRESS_VERIFICATION_NOTICE";
    DocumentType["AffidavitOfDomicile"] = "AFFIDAVIT_OF_DOMICILE";
    DocumentType["ApexSimpleDirectDeposit"] = "APEX_SIMPLE_DIRECT_DEPOSIT";
    DocumentType["TransferForm"] = "TRANSFER_FORM";
    DocumentType["JournalRequestForm"] = "JOURNAL_REQUEST_FORM";
    DocumentType["Check"] = "CHECK";
    DocumentType["CheckRequestForm"] = "CHECK_REQUEST_FORM";
    DocumentType["CorpAgreementCash"] = "CORP_AGREEMENT_CASH";
    DocumentType["CorpResolution"] = "CORP_RESOLUTION";
    DocumentType["CourtDocument"] = "COURT_DOCUMENT";
    DocumentType["CourtOrder"] = "COURT_ORDER";
    DocumentType["DepositSecuritiesRequestForm"] = "DEPOSIT_SECURITIES_REQUEST_FORM";
    DocumentType["DigitalSignature"] = "DIGITAL_SIGNATURE";
    DocumentType["DirectConversionRequest"] = "DIRECT_CONVERSION_REQUEST";
    DocumentType["DistributionForm"] = "DISTRIBUTION_FORM";
    DocumentType["DocumentIrs"] = "DOCUMENT_IRS";
    DocumentType["DomesticWire"] = "DOMESTIC_WIRE";
    DocumentType["DwacRequestForm"] = "DWAC_REQUEST_FORM";
    DocumentType["Dwre"] = "DWRE";
    DocumentType["EntityDueDiligenceQuestionnaire"] = "ENTITY_DUE_DILIGENCE_QUESTIONNAIRE";
    DocumentType["EntityUpdateForm"] = "ENTITY_UPDATE_FORM";
    DocumentType["EscheatForm"] = "ESCHEAT_FORM";
    DocumentType["EscheatmentDueDiligenceLetter"] = "ESCHEATMENT_DUE_DILIGENCE_LETTER";
    DocumentType["ESignature"] = "E_SIGNATURE";
    DocumentType["ForeignAccountDueDiligenceQuestionnaire"] = "FOREIGN_ACCOUNT_DUE_DILIGENCE_QUESTIONNAIRE";
    DocumentType["ForeignBankCertificationDisclosure"] = "FOREIGN_BANK_CERTIFICATION_DISCLOSURE";
    DocumentType["ForeignEntityDeclarationForm"] = "FOREIGN_ENTITY_DECLARATION_FORM";
    DocumentType["ForeignFinInstDueDiligenceQuestionnaire"] = "FOREIGN_FIN_INST_DUE_DILIGENCE_QUESTIONNAIRE";
    DocumentType["Form144"] = "FORM_144";
    DocumentType["FreeDeliveryForm"] = "FREE_DELIVERY_FORM";
    DocumentType["FullTradingAuthorization"] = "FULL_TRADING_AUTHORIZATION";
    DocumentType["InternationalWireRequestForm"] = "INTERNATIONAL_WIRE_REQUEST_FORM";
    DocumentType["IrrevocableStockBondPower"] = "IRREVOCABLE_STOCK_BOND_POWER";
    DocumentType["JointAccAgreementEntirety"] = "JOINT_ACC_AGREEMENT_ENTIRETY";
    DocumentType["JointAccAgreementRightsOfSurvivorship"] = "JOINT_ACC_AGREEMENT_RIGHTS_OF_SURVIVORSHIP";
    DocumentType["JointAccAgreementCommunityProperty"] = "JOINT_ACC_AGREEMENT_COMMUNITY_PROPERTY";
    DocumentType["JointAccAgreementTenantsInCommon"] = "JOINT_ACC_AGREEMENT_TENANTS_IN_COMMON";
    DocumentType["JointTefraW8"] = "JOINT_TEFRA_W8";
    DocumentType["LegalOpinion"] = "LEGAL_OPINION";
    DocumentType["LetterOfAuthority"] = "LETTER_OF_AUTHORITY";
    DocumentType["LettersOfTestamentaryCertAppointment"] = "LETTERS_OF_TESTAMENTARY_CERT_APPOINTMENT";
    DocumentType["LimitedLiabilityCompanyAgreement"] = "LIMITED_LIABILITY_COMPANY_AGREEMENT";
    DocumentType["LimitedPurposeMargin"] = "LIMITED_PURPOSE_MARGIN";
    DocumentType["MailAuthorization"] = "MAIL_AUTHORIZATION";
    DocumentType["Msla"] = "MSLA";
    DocumentType["MiscGeneral"] = "MISC_GENERAL";
    DocumentType["NewAccountApplicationEntity"] = "NEW_ACCOUNT_APPLICATION_ENTITY";
    DocumentType["NonCorporateAssociation"] = "NON_CORPORATE_ASSOCIATION";
    DocumentType["PartnershipAgreement"] = "PARTNERSHIP_AGREEMENT";
    DocumentType["PatternDayTraderForm"] = "PATTERN_DAY_TRADER_FORM";
    DocumentType["PowerOfAttorneyDurable"] = "POWER_OF_ATTORNEY_DURABLE";
    DocumentType["PowerOfAttorneyGeneral"] = "POWER_OF_ATTORNEY_GENERAL";
    DocumentType["RegisteredAdvisoryAuthorityForm"] = "REGISTERED_ADVISORY_AUTHORITY_FORM";
    DocumentType["Rmd"] = "RMD";
    DocumentType["RiskDisclosure"] = "RISK_DISCLOSURE";
    DocumentType["Rule145SellersLetter"] = "RULE_145_SELLERS_LETTER";
    DocumentType["Rule701Letter"] = "RULE_701_LETTER";
    DocumentType["SameNameLoa"] = "SAME_NAME_LOA";
    DocumentType["SecuritiesBasedLendingAgreement"] = "SECURITIES_BASED_LENDING_AGREEMENT";
    DocumentType["SepContributionAgreement"] = "SEP_CONTRIBUTION_AGREEMENT";
    DocumentType["SepEmployerPlanDocumentation"] = "SEP_EMPLOYER_PLAN_DOCUMENTATION";
    DocumentType["SignatureCard"] = "SIGNATURE_CARD";
    DocumentType["SimpleForm5304"] = "SIMPLE_FORM_5304";
    DocumentType["SimplifiedEmployeePension"] = "SIMPLIFIED_EMPLOYEE_PENSION";
    DocumentType["CertificateSoleProprietorship"] = "CERTIFICATE_SOLE_PROPRIETORSHIP";
    DocumentType["TransferOnDeath"] = "TRANSFER_ON_DEATH";
    DocumentType["TrustAgreement"] = "TRUST_AGREEMENT";
    DocumentType["TaxW8Ben"] = "TAX_W8_BEN";
    DocumentType["TaxW8BenJointAccountHolder"] = "TAX_W8_BEN_JOINT_ACCOUNT_HOLDER";
    DocumentType["TaxW8Eci"] = "TAX_W8_ECI";
    DocumentType["TaxW8Exp"] = "TAX_W8_EXP";
    DocumentType["TaxW8Imy"] = "TAX_W8_IMY";
    DocumentType["TaxW9"] = "TAX_W9";
    DocumentType["WillCopy"] = "WILL_COPY";
    DocumentType["WireRequestForm"] = "WIRE_REQUEST_FORM";
    DocumentType["WorthlessSecurities"] = "WORTHLESS_SECURITIES";
    DocumentType["ChangeOfAddressRequest"] = "CHANGE_OF_ADDRESS_REQUEST";
})(DocumentType || (exports.DocumentType = DocumentType = {}));
/** @internal */
exports.DocumentType$inboundSchema = z
    .union([
    z.nativeEnum(DocumentType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.DocumentType$outboundSchema = z.union([
    z.nativeEnum(DocumentType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DocumentType$;
(function (DocumentType$) {
    /** @deprecated use `DocumentType$inboundSchema` instead. */
    DocumentType$.inboundSchema = exports.DocumentType$inboundSchema;
    /** @deprecated use `DocumentType$outboundSchema` instead. */
    DocumentType$.outboundSchema = exports.DocumentType$outboundSchema;
})(DocumentType$ || (exports.DocumentType$ = DocumentType$ = {}));
/** @internal */
exports.AccountDocumentUploadRequestCreate$inboundSchema = z.object({
    account_id: z.string(),
    correspondent_id: z.string(),
    document_type: exports.DocumentType$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "correspondent_id": "correspondentId",
        "document_type": "documentType",
    });
});
/** @internal */
exports.AccountDocumentUploadRequestCreate$outboundSchema = z.object({
    accountId: z.string(),
    correspondentId: z.string(),
    documentType: exports.DocumentType$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        correspondentId: "correspondent_id",
        documentType: "document_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountDocumentUploadRequestCreate$;
(function (AccountDocumentUploadRequestCreate$) {
    /** @deprecated use `AccountDocumentUploadRequestCreate$inboundSchema` instead. */
    AccountDocumentUploadRequestCreate$.inboundSchema = exports.AccountDocumentUploadRequestCreate$inboundSchema;
    /** @deprecated use `AccountDocumentUploadRequestCreate$outboundSchema` instead. */
    AccountDocumentUploadRequestCreate$.outboundSchema = exports.AccountDocumentUploadRequestCreate$outboundSchema;
})(AccountDocumentUploadRequestCreate$ || (exports.AccountDocumentUploadRequestCreate$ = AccountDocumentUploadRequestCreate$ = {}));
