// @apexfintechsolutions/ascend-sdk/models/components/achdepositschedule.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchDepositSchedule$ = exports.AchDepositSchedule$outboundSchema = exports.AchDepositSchedule$inboundSchema = exports.ScheduleDetails$ = exports.ScheduleDetails$outboundSchema = exports.ScheduleDetails$inboundSchema = exports.AchDepositScheduleScheduleProperties$ = exports.AchDepositScheduleScheduleProperties$outboundSchema = exports.AchDepositScheduleScheduleProperties$inboundSchema = exports.AchDepositScheduleTimeUnit$ = exports.AchDepositScheduleTimeUnit$outboundSchema = exports.AchDepositScheduleTimeUnit$inboundSchema = exports.AchDepositScheduleState$ = exports.AchDepositScheduleState$outboundSchema = exports.AchDepositScheduleState$inboundSchema = exports.AchDepositScheduleStartDate$ = exports.AchDepositScheduleStartDate$outboundSchema = exports.AchDepositScheduleStartDate$inboundSchema = exports.AchDepositScheduleAmount$ = exports.AchDepositScheduleAmount$outboundSchema = exports.AchDepositScheduleAmount$inboundSchema = exports.AchDepositScheduleRetirementContribution$ = exports.AchDepositScheduleRetirementContribution$outboundSchema = exports.AchDepositScheduleRetirementContribution$inboundSchema = exports.AchDepositScheduleRetirementContributionType$ = exports.AchDepositScheduleRetirementContributionType$outboundSchema = exports.AchDepositScheduleRetirementContributionType$inboundSchema = exports.AchDepositScheduleTemporalTaxYear$ = exports.AchDepositScheduleTemporalTaxYear$outboundSchema = exports.AchDepositScheduleTemporalTaxYear$inboundSchema = exports.AchDepositScheduleIraContribution$ = exports.AchDepositScheduleIraContribution$outboundSchema = exports.AchDepositScheduleIraContribution$inboundSchema = exports.AchDepositScheduleType$ = exports.AchDepositScheduleType$outboundSchema = exports.AchDepositScheduleType$inboundSchema = exports.AchDepositScheduleTimeUnit = exports.AchDepositScheduleState = exports.AchDepositScheduleRetirementContributionType = exports.AchDepositScheduleTemporalTaxYear = exports.AchDepositScheduleType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * The type of retirement contribution.
 */
var AchDepositScheduleType;
(function (AchDepositScheduleType) {
    AchDepositScheduleType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    AchDepositScheduleType["Regular"] = "REGULAR";
    AchDepositScheduleType["Employee"] = "EMPLOYEE";
    AchDepositScheduleType["Employer"] = "EMPLOYER";
    AchDepositScheduleType["Recharacterization"] = "RECHARACTERIZATION";
    AchDepositScheduleType["Rollover60Day"] = "ROLLOVER_60_DAY";
    AchDepositScheduleType["RolloverDirect"] = "ROLLOVER_DIRECT";
    AchDepositScheduleType["Transfer"] = "TRANSFER";
    AchDepositScheduleType["TrusteeFee"] = "TRUSTEE_FEE";
    AchDepositScheduleType["Conversion"] = "CONVERSION";
    AchDepositScheduleType["Repayment"] = "REPAYMENT";
    AchDepositScheduleType["ContributionNonReportable"] = "CONTRIBUTION_NON_REPORTABLE";
})(AchDepositScheduleType || (exports.AchDepositScheduleType = AchDepositScheduleType = {}));
/**
 * A temporal tax year value. This will always evaluate to a year based on the date the transfer was initiated.
 */
var AchDepositScheduleTemporalTaxYear;
(function (AchDepositScheduleTemporalTaxYear) {
    AchDepositScheduleTemporalTaxYear["TemporalTaxYearUnspecified"] = "TEMPORAL_TAX_YEAR_UNSPECIFIED";
    AchDepositScheduleTemporalTaxYear["CurrentCalendarYear"] = "CURRENT_CALENDAR_YEAR";
    AchDepositScheduleTemporalTaxYear["MinimumTaxYear"] = "MINIMUM_TAX_YEAR";
})(AchDepositScheduleTemporalTaxYear || (exports.AchDepositScheduleTemporalTaxYear = AchDepositScheduleTemporalTaxYear = {}));
/**
 * The type of retirement contribution.
 */
var AchDepositScheduleRetirementContributionType;
(function (AchDepositScheduleRetirementContributionType) {
    AchDepositScheduleRetirementContributionType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    AchDepositScheduleRetirementContributionType["Regular"] = "REGULAR";
    AchDepositScheduleRetirementContributionType["Employee"] = "EMPLOYEE";
    AchDepositScheduleRetirementContributionType["Employer"] = "EMPLOYER";
    AchDepositScheduleRetirementContributionType["Recharacterization"] = "RECHARACTERIZATION";
    AchDepositScheduleRetirementContributionType["Rollover60Day"] = "ROLLOVER_60_DAY";
    AchDepositScheduleRetirementContributionType["RolloverDirect"] = "ROLLOVER_DIRECT";
    AchDepositScheduleRetirementContributionType["Transfer"] = "TRANSFER";
    AchDepositScheduleRetirementContributionType["TrusteeFee"] = "TRUSTEE_FEE";
    AchDepositScheduleRetirementContributionType["Conversion"] = "CONVERSION";
    AchDepositScheduleRetirementContributionType["Repayment"] = "REPAYMENT";
    AchDepositScheduleRetirementContributionType["ContributionNonReportable"] = "CONTRIBUTION_NON_REPORTABLE";
})(AchDepositScheduleRetirementContributionType || (exports.AchDepositScheduleRetirementContributionType = AchDepositScheduleRetirementContributionType = {}));
/**
 * The state of the represented schedule
 */
var AchDepositScheduleState;
(function (AchDepositScheduleState) {
    AchDepositScheduleState["Active"] = "ACTIVE";
    AchDepositScheduleState["Canceled"] = "CANCELED";
    AchDepositScheduleState["Completed"] = "COMPLETED";
})(AchDepositScheduleState || (exports.AchDepositScheduleState = AchDepositScheduleState = {}));
/**
 * The time unit used to calculate the interval between transfers. The time period between transfers in a scheduled series is the unit of time times the multiplier
 */
var AchDepositScheduleTimeUnit;
(function (AchDepositScheduleTimeUnit) {
    AchDepositScheduleTimeUnit["Day"] = "DAY";
    AchDepositScheduleTimeUnit["Week"] = "WEEK";
    AchDepositScheduleTimeUnit["Month"] = "MONTH";
})(AchDepositScheduleTimeUnit || (exports.AchDepositScheduleTimeUnit = AchDepositScheduleTimeUnit = {}));
/** @internal */
exports.AchDepositScheduleType$inboundSchema = z
    .union([
    z.nativeEnum(AchDepositScheduleType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.AchDepositScheduleType$outboundSchema = z.union([
    z.nativeEnum(AchDepositScheduleType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleType$;
(function (AchDepositScheduleType$) {
    /** @deprecated use `AchDepositScheduleType$inboundSchema` instead. */
    AchDepositScheduleType$.inboundSchema = exports.AchDepositScheduleType$inboundSchema;
    /** @deprecated use `AchDepositScheduleType$outboundSchema` instead. */
    AchDepositScheduleType$.outboundSchema = exports.AchDepositScheduleType$outboundSchema;
})(AchDepositScheduleType$ || (exports.AchDepositScheduleType$ = AchDepositScheduleType$ = {}));
/** @internal */
exports.AchDepositScheduleIraContribution$inboundSchema = z.object({
    tax_year: z.number().int().optional(),
    type: exports.AchDepositScheduleType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "tax_year": "taxYear",
    });
});
/** @internal */
exports.AchDepositScheduleIraContribution$outboundSchema = z.object({
    taxYear: z.number().int().optional(),
    type: exports.AchDepositScheduleType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        taxYear: "tax_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleIraContribution$;
(function (AchDepositScheduleIraContribution$) {
    /** @deprecated use `AchDepositScheduleIraContribution$inboundSchema` instead. */
    AchDepositScheduleIraContribution$.inboundSchema = exports.AchDepositScheduleIraContribution$inboundSchema;
    /** @deprecated use `AchDepositScheduleIraContribution$outboundSchema` instead. */
    AchDepositScheduleIraContribution$.outboundSchema = exports.AchDepositScheduleIraContribution$outboundSchema;
})(AchDepositScheduleIraContribution$ || (exports.AchDepositScheduleIraContribution$ = AchDepositScheduleIraContribution$ = {}));
/** @internal */
exports.AchDepositScheduleTemporalTaxYear$inboundSchema = z
    .union([
    z.nativeEnum(AchDepositScheduleTemporalTaxYear),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.AchDepositScheduleTemporalTaxYear$outboundSchema = z.union([
    z.nativeEnum(AchDepositScheduleTemporalTaxYear),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleTemporalTaxYear$;
(function (AchDepositScheduleTemporalTaxYear$) {
    /** @deprecated use `AchDepositScheduleTemporalTaxYear$inboundSchema` instead. */
    AchDepositScheduleTemporalTaxYear$.inboundSchema = exports.AchDepositScheduleTemporalTaxYear$inboundSchema;
    /** @deprecated use `AchDepositScheduleTemporalTaxYear$outboundSchema` instead. */
    AchDepositScheduleTemporalTaxYear$.outboundSchema = exports.AchDepositScheduleTemporalTaxYear$outboundSchema;
})(AchDepositScheduleTemporalTaxYear$ || (exports.AchDepositScheduleTemporalTaxYear$ = AchDepositScheduleTemporalTaxYear$ = {}));
/** @internal */
exports.AchDepositScheduleRetirementContributionType$inboundSchema = z
    .union([
    z.nativeEnum(AchDepositScheduleRetirementContributionType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.AchDepositScheduleRetirementContributionType$outboundSchema = z.union([
    z.nativeEnum(AchDepositScheduleRetirementContributionType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleRetirementContributionType$;
(function (AchDepositScheduleRetirementContributionType$) {
    /** @deprecated use `AchDepositScheduleRetirementContributionType$inboundSchema` instead. */
    AchDepositScheduleRetirementContributionType$.inboundSchema = exports.AchDepositScheduleRetirementContributionType$inboundSchema;
    /** @deprecated use `AchDepositScheduleRetirementContributionType$outboundSchema` instead. */
    AchDepositScheduleRetirementContributionType$.outboundSchema = exports.AchDepositScheduleRetirementContributionType$outboundSchema;
})(AchDepositScheduleRetirementContributionType$ || (exports.AchDepositScheduleRetirementContributionType$ = AchDepositScheduleRetirementContributionType$ = {}));
/** @internal */
exports.AchDepositScheduleRetirementContribution$inboundSchema = z.object({
    tax_year: z.number().int().optional(),
    temporal_tax_year: exports.AchDepositScheduleTemporalTaxYear$inboundSchema.optional(),
    type: exports.AchDepositScheduleRetirementContributionType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "tax_year": "taxYear",
        "temporal_tax_year": "temporalTaxYear",
    });
});
/** @internal */
exports.AchDepositScheduleRetirementContribution$outboundSchema = z.object({
    taxYear: z.number().int().optional(),
    temporalTaxYear: exports.AchDepositScheduleTemporalTaxYear$outboundSchema.optional(),
    type: exports.AchDepositScheduleRetirementContributionType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        taxYear: "tax_year",
        temporalTaxYear: "temporal_tax_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleRetirementContribution$;
(function (AchDepositScheduleRetirementContribution$) {
    /** @deprecated use `AchDepositScheduleRetirementContribution$inboundSchema` instead. */
    AchDepositScheduleRetirementContribution$.inboundSchema = exports.AchDepositScheduleRetirementContribution$inboundSchema;
    /** @deprecated use `AchDepositScheduleRetirementContribution$outboundSchema` instead. */
    AchDepositScheduleRetirementContribution$.outboundSchema = exports.AchDepositScheduleRetirementContribution$outboundSchema;
})(AchDepositScheduleRetirementContribution$ || (exports.AchDepositScheduleRetirementContribution$ = AchDepositScheduleRetirementContribution$ = {}));
/** @internal */
exports.AchDepositScheduleAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.AchDepositScheduleAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleAmount$;
(function (AchDepositScheduleAmount$) {
    /** @deprecated use `AchDepositScheduleAmount$inboundSchema` instead. */
    AchDepositScheduleAmount$.inboundSchema = exports.AchDepositScheduleAmount$inboundSchema;
    /** @deprecated use `AchDepositScheduleAmount$outboundSchema` instead. */
    AchDepositScheduleAmount$.outboundSchema = exports.AchDepositScheduleAmount$outboundSchema;
})(AchDepositScheduleAmount$ || (exports.AchDepositScheduleAmount$ = AchDepositScheduleAmount$ = {}));
/** @internal */
exports.AchDepositScheduleStartDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.AchDepositScheduleStartDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleStartDate$;
(function (AchDepositScheduleStartDate$) {
    /** @deprecated use `AchDepositScheduleStartDate$inboundSchema` instead. */
    AchDepositScheduleStartDate$.inboundSchema = exports.AchDepositScheduleStartDate$inboundSchema;
    /** @deprecated use `AchDepositScheduleStartDate$outboundSchema` instead. */
    AchDepositScheduleStartDate$.outboundSchema = exports.AchDepositScheduleStartDate$outboundSchema;
})(AchDepositScheduleStartDate$ || (exports.AchDepositScheduleStartDate$ = AchDepositScheduleStartDate$ = {}));
/** @internal */
exports.AchDepositScheduleState$inboundSchema = z
    .union([
    z.nativeEnum(AchDepositScheduleState),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.AchDepositScheduleState$outboundSchema = z.union([
    z.nativeEnum(AchDepositScheduleState),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleState$;
(function (AchDepositScheduleState$) {
    /** @deprecated use `AchDepositScheduleState$inboundSchema` instead. */
    AchDepositScheduleState$.inboundSchema = exports.AchDepositScheduleState$inboundSchema;
    /** @deprecated use `AchDepositScheduleState$outboundSchema` instead. */
    AchDepositScheduleState$.outboundSchema = exports.AchDepositScheduleState$outboundSchema;
})(AchDepositScheduleState$ || (exports.AchDepositScheduleState$ = AchDepositScheduleState$ = {}));
/** @internal */
exports.AchDepositScheduleTimeUnit$inboundSchema = z
    .union([
    z.nativeEnum(AchDepositScheduleTimeUnit),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.AchDepositScheduleTimeUnit$outboundSchema = z.union([
    z.nativeEnum(AchDepositScheduleTimeUnit),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleTimeUnit$;
(function (AchDepositScheduleTimeUnit$) {
    /** @deprecated use `AchDepositScheduleTimeUnit$inboundSchema` instead. */
    AchDepositScheduleTimeUnit$.inboundSchema = exports.AchDepositScheduleTimeUnit$inboundSchema;
    /** @deprecated use `AchDepositScheduleTimeUnit$outboundSchema` instead. */
    AchDepositScheduleTimeUnit$.outboundSchema = exports.AchDepositScheduleTimeUnit$outboundSchema;
})(AchDepositScheduleTimeUnit$ || (exports.AchDepositScheduleTimeUnit$ = AchDepositScheduleTimeUnit$ = {}));
/** @internal */
exports.AchDepositScheduleScheduleProperties$inboundSchema = z.object({
    occurrences: z.number().int().optional(),
    start_date: z.nullable(z.lazy(() => exports.AchDepositScheduleStartDate$inboundSchema)).optional(),
    state: exports.AchDepositScheduleState$inboundSchema.optional(),
    time_unit: exports.AchDepositScheduleTimeUnit$inboundSchema.optional(),
    unit_multiplier: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "start_date": "startDate",
        "time_unit": "timeUnit",
        "unit_multiplier": "unitMultiplier",
    });
});
/** @internal */
exports.AchDepositScheduleScheduleProperties$outboundSchema = z.object({
    occurrences: z.number().int().optional(),
    startDate: z.nullable(z.lazy(() => exports.AchDepositScheduleStartDate$outboundSchema)).optional(),
    state: exports.AchDepositScheduleState$outboundSchema.optional(),
    timeUnit: exports.AchDepositScheduleTimeUnit$outboundSchema.optional(),
    unitMultiplier: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        startDate: "start_date",
        timeUnit: "time_unit",
        unitMultiplier: "unit_multiplier",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleScheduleProperties$;
(function (AchDepositScheduleScheduleProperties$) {
    /** @deprecated use `AchDepositScheduleScheduleProperties$inboundSchema` instead. */
    AchDepositScheduleScheduleProperties$.inboundSchema = exports.AchDepositScheduleScheduleProperties$inboundSchema;
    /** @deprecated use `AchDepositScheduleScheduleProperties$outboundSchema` instead. */
    AchDepositScheduleScheduleProperties$.outboundSchema = exports.AchDepositScheduleScheduleProperties$outboundSchema;
})(AchDepositScheduleScheduleProperties$ || (exports.AchDepositScheduleScheduleProperties$ = AchDepositScheduleScheduleProperties$ = {}));
/** @internal */
exports.ScheduleDetails$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.AchDepositScheduleAmount$inboundSchema))
        .optional(),
    client_schedule_id: z.string().optional(),
    schedule_properties: z.nullable(z.lazy(() => exports.AchDepositScheduleScheduleProperties$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_schedule_id": "clientScheduleId",
        "schedule_properties": "scheduleProperties",
    });
});
/** @internal */
exports.ScheduleDetails$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.AchDepositScheduleAmount$outboundSchema))
        .optional(),
    clientScheduleId: z.string().optional(),
    scheduleProperties: z.nullable(z.lazy(() => exports.AchDepositScheduleScheduleProperties$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientScheduleId: "client_schedule_id",
        scheduleProperties: "schedule_properties",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ScheduleDetails$;
(function (ScheduleDetails$) {
    /** @deprecated use `ScheduleDetails$inboundSchema` instead. */
    ScheduleDetails$.inboundSchema = exports.ScheduleDetails$inboundSchema;
    /** @deprecated use `ScheduleDetails$outboundSchema` instead. */
    ScheduleDetails$.outboundSchema = exports.ScheduleDetails$outboundSchema;
})(ScheduleDetails$ || (exports.ScheduleDetails$ = ScheduleDetails$ = {}));
/** @internal */
exports.AchDepositSchedule$inboundSchema = z.object({
    bank_relationship: z.string().optional(),
    ira_contribution: z.nullable(z.lazy(() => exports.AchDepositScheduleIraContribution$inboundSchema)).optional(),
    name: z.string().optional(),
    retirement_contribution: z.nullable(z.lazy(() => exports.AchDepositScheduleRetirementContribution$inboundSchema)).optional(),
    schedule_details: z.nullable(z.lazy(() => exports.ScheduleDetails$inboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "bank_relationship": "bankRelationship",
        "ira_contribution": "iraContribution",
        "retirement_contribution": "retirementContribution",
        "schedule_details": "scheduleDetails",
    });
});
/** @internal */
exports.AchDepositSchedule$outboundSchema = z.object({
    bankRelationship: z.string().optional(),
    iraContribution: z.nullable(z.lazy(() => exports.AchDepositScheduleIraContribution$outboundSchema)).optional(),
    name: z.string().optional(),
    retirementContribution: z.nullable(z.lazy(() => exports.AchDepositScheduleRetirementContribution$outboundSchema)).optional(),
    scheduleDetails: z.nullable(z.lazy(() => exports.ScheduleDetails$outboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        bankRelationship: "bank_relationship",
        iraContribution: "ira_contribution",
        retirementContribution: "retirement_contribution",
        scheduleDetails: "schedule_details",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositSchedule$;
(function (AchDepositSchedule$) {
    /** @deprecated use `AchDepositSchedule$inboundSchema` instead. */
    AchDepositSchedule$.inboundSchema = exports.AchDepositSchedule$inboundSchema;
    /** @deprecated use `AchDepositSchedule$outboundSchema` instead. */
    AchDepositSchedule$.outboundSchema = exports.AchDepositSchedule$outboundSchema;
})(AchDepositSchedule$ || (exports.AchDepositSchedule$ = AchDepositSchedule$ = {}));
