// @apexfintechsolutions/ascend-sdk/models/components/achdepositcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchDepositCreate$ = exports.AchDepositCreate$outboundSchema = exports.AchDepositCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
const retirementcontributioncreate_js_1 = require("./retirementcontributioncreate.js");
/** @internal */
exports.AchDepositCreate$inboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$inboundSchema,
    bank_relationship: z.string(),
    client_transfer_id: z.string(),
    memo: z.string().optional(),
    retirement_contribution: retirementcontributioncreate_js_1.RetirementContributionCreate$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "bank_relationship": "bankRelationship",
        "client_transfer_id": "clientTransferId",
        "retirement_contribution": "retirementContribution",
    });
});
/** @internal */
exports.AchDepositCreate$outboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$outboundSchema,
    bankRelationship: z.string(),
    clientTransferId: z.string(),
    memo: z.string().optional(),
    retirementContribution: retirementcontributioncreate_js_1.RetirementContributionCreate$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        bankRelationship: "bank_relationship",
        clientTransferId: "client_transfer_id",
        retirementContribution: "retirement_contribution",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositCreate$;
(function (AchDepositCreate$) {
    /** @deprecated use `AchDepositCreate$inboundSchema` instead. */
    AchDepositCreate$.inboundSchema = exports.AchDepositCreate$inboundSchema;
    /** @deprecated use `AchDepositCreate$outboundSchema` instead. */
    AchDepositCreate$.outboundSchema = exports.AchDepositCreate$outboundSchema;
})(AchDepositCreate$ || (exports.AchDepositCreate$ = AchDepositCreate$ = {}));
