// @apexfintechsolutions/ascend-sdk/models/components/achdepositschedulecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchDepositScheduleCreate$ = exports.AchDepositScheduleCreate$outboundSchema = exports.AchDepositScheduleCreate$inboundSchema = exports.IraContribution$ = exports.IraContribution$outboundSchema = exports.IraContribution$inboundSchema = exports.AchDepositScheduleCreateType$ = exports.AchDepositScheduleCreateType$outboundSchema = exports.AchDepositScheduleCreateType$inboundSchema = exports.AchDepositScheduleCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const depositscheduledetailscreate_js_1 = require("./depositscheduledetailscreate.js");
const scheduledretirementcontributioncreate_js_1 = require("./scheduledretirementcontributioncreate.js");
/**
 * The type of retirement contribution.
 */
var AchDepositScheduleCreateType;
(function (AchDepositScheduleCreateType) {
    AchDepositScheduleCreateType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    AchDepositScheduleCreateType["Regular"] = "REGULAR";
    AchDepositScheduleCreateType["Employee"] = "EMPLOYEE";
    AchDepositScheduleCreateType["Employer"] = "EMPLOYER";
    AchDepositScheduleCreateType["Recharacterization"] = "RECHARACTERIZATION";
    AchDepositScheduleCreateType["Rollover60Day"] = "ROLLOVER_60_DAY";
    AchDepositScheduleCreateType["RolloverDirect"] = "ROLLOVER_DIRECT";
    AchDepositScheduleCreateType["Transfer"] = "TRANSFER";
    AchDepositScheduleCreateType["TrusteeFee"] = "TRUSTEE_FEE";
    AchDepositScheduleCreateType["Conversion"] = "CONVERSION";
    AchDepositScheduleCreateType["Repayment"] = "REPAYMENT";
    AchDepositScheduleCreateType["ContributionNonReportable"] = "CONTRIBUTION_NON_REPORTABLE";
})(AchDepositScheduleCreateType || (exports.AchDepositScheduleCreateType = AchDepositScheduleCreateType = {}));
/** @internal */
exports.AchDepositScheduleCreateType$inboundSchema = z
    .union([
    z.nativeEnum(AchDepositScheduleCreateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.AchDepositScheduleCreateType$outboundSchema = z.union([
    z.nativeEnum(AchDepositScheduleCreateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleCreateType$;
(function (AchDepositScheduleCreateType$) {
    /** @deprecated use `AchDepositScheduleCreateType$inboundSchema` instead. */
    AchDepositScheduleCreateType$.inboundSchema = exports.AchDepositScheduleCreateType$inboundSchema;
    /** @deprecated use `AchDepositScheduleCreateType$outboundSchema` instead. */
    AchDepositScheduleCreateType$.outboundSchema = exports.AchDepositScheduleCreateType$outboundSchema;
})(AchDepositScheduleCreateType$ || (exports.AchDepositScheduleCreateType$ = AchDepositScheduleCreateType$ = {}));
/** @internal */
exports.IraContribution$inboundSchema = z.object({
    tax_year: z.number().int(),
    type: exports.AchDepositScheduleCreateType$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "tax_year": "taxYear",
    });
});
/** @internal */
exports.IraContribution$outboundSchema = z.object({
    taxYear: z.number().int(),
    type: exports.AchDepositScheduleCreateType$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        taxYear: "tax_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IraContribution$;
(function (IraContribution$) {
    /** @deprecated use `IraContribution$inboundSchema` instead. */
    IraContribution$.inboundSchema = exports.IraContribution$inboundSchema;
    /** @deprecated use `IraContribution$outboundSchema` instead. */
    IraContribution$.outboundSchema = exports.IraContribution$outboundSchema;
})(IraContribution$ || (exports.IraContribution$ = IraContribution$ = {}));
/** @internal */
exports.AchDepositScheduleCreate$inboundSchema = z.object({
    bank_relationship: z.string(),
    ira_contribution: z.lazy(() => exports.IraContribution$inboundSchema).optional(),
    retirement_contribution: scheduledretirementcontributioncreate_js_1.ScheduledRetirementContributionCreate$inboundSchema
        .optional(),
    schedule_details: depositscheduledetailscreate_js_1.DepositScheduleDetailsCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "bank_relationship": "bankRelationship",
        "ira_contribution": "iraContribution",
        "retirement_contribution": "retirementContribution",
        "schedule_details": "scheduleDetails",
    });
});
/** @internal */
exports.AchDepositScheduleCreate$outboundSchema = z.object({
    bankRelationship: z.string(),
    iraContribution: z.lazy(() => exports.IraContribution$outboundSchema).optional(),
    retirementContribution: scheduledretirementcontributioncreate_js_1.ScheduledRetirementContributionCreate$outboundSchema
        .optional(),
    scheduleDetails: depositscheduledetailscreate_js_1.DepositScheduleDetailsCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        bankRelationship: "bank_relationship",
        iraContribution: "ira_contribution",
        retirementContribution: "retirement_contribution",
        scheduleDetails: "schedule_details",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositScheduleCreate$;
(function (AchDepositScheduleCreate$) {
    /** @deprecated use `AchDepositScheduleCreate$inboundSchema` instead. */
    AchDepositScheduleCreate$.inboundSchema = exports.AchDepositScheduleCreate$inboundSchema;
    /** @deprecated use `AchDepositScheduleCreate$outboundSchema` instead. */
    AchDepositScheduleCreate$.outboundSchema = exports.AchDepositScheduleCreate$outboundSchema;
})(AchDepositScheduleCreate$ || (exports.AchDepositScheduleCreate$ = AchDepositScheduleCreate$ = {}));
