// @apexfintechsolutions/ascend-sdk/models/components/achwithdrawalschedulecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchWithdrawalScheduleCreate$ = exports.AchWithdrawalScheduleCreate$outboundSchema = exports.AchWithdrawalScheduleCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const retirementdistributioncreate_js_1 = require("./retirementdistributioncreate.js");
const withdrawalscheduledetailscreate_js_1 = require("./withdrawalscheduledetailscreate.js");
/** @internal */
exports.AchWithdrawalScheduleCreate$inboundSchema = z.object({
    bank_relationship: z.string(),
    ira_distribution: retirementdistributioncreate_js_1.RetirementDistributionCreate$inboundSchema.optional(),
    schedule_details: withdrawalscheduledetailscreate_js_1.WithdrawalScheduleDetailsCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "bank_relationship": "bankRelationship",
        "ira_distribution": "iraDistribution",
        "schedule_details": "scheduleDetails",
    });
});
/** @internal */
exports.AchWithdrawalScheduleCreate$outboundSchema = z.object({
    bankRelationship: z.string(),
    iraDistribution: retirementdistributioncreate_js_1.RetirementDistributionCreate$outboundSchema.optional(),
    scheduleDetails: withdrawalscheduledetailscreate_js_1.WithdrawalScheduleDetailsCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        bankRelationship: "bank_relationship",
        iraDistribution: "ira_distribution",
        scheduleDetails: "schedule_details",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleCreate$;
(function (AchWithdrawalScheduleCreate$) {
    /** @deprecated use `AchWithdrawalScheduleCreate$inboundSchema` instead. */
    AchWithdrawalScheduleCreate$.inboundSchema = exports.AchWithdrawalScheduleCreate$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleCreate$outboundSchema` instead. */
    AchWithdrawalScheduleCreate$.outboundSchema = exports.AchWithdrawalScheduleCreate$outboundSchema;
})(AchWithdrawalScheduleCreate$ || (exports.AchWithdrawalScheduleCreate$ = AchWithdrawalScheduleCreate$ = {}));
