// @apexfintechsolutions/ascend-sdk/models/components/achwithdrawalschedule.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchWithdrawalSchedule$ = exports.AchWithdrawalSchedule$outboundSchema = exports.AchWithdrawalSchedule$inboundSchema = exports.AchWithdrawalScheduleScheduleDetails$ = exports.AchWithdrawalScheduleScheduleDetails$outboundSchema = exports.AchWithdrawalScheduleScheduleDetails$inboundSchema = exports.AchWithdrawalScheduleScheduleProperties$ = exports.AchWithdrawalScheduleScheduleProperties$outboundSchema = exports.AchWithdrawalScheduleScheduleProperties$inboundSchema = exports.AchWithdrawalScheduleTimeUnit$ = exports.AchWithdrawalScheduleTimeUnit$outboundSchema = exports.AchWithdrawalScheduleTimeUnit$inboundSchema = exports.AchWithdrawalScheduleState$ = exports.AchWithdrawalScheduleState$outboundSchema = exports.AchWithdrawalScheduleState$inboundSchema = exports.AchWithdrawalScheduleStartDate$ = exports.AchWithdrawalScheduleStartDate$outboundSchema = exports.AchWithdrawalScheduleStartDate$inboundSchema = exports.AchWithdrawalScheduleAmount$ = exports.AchWithdrawalScheduleAmount$outboundSchema = exports.AchWithdrawalScheduleAmount$inboundSchema = exports.IraDistribution$ = exports.IraDistribution$outboundSchema = exports.IraDistribution$inboundSchema = exports.AchWithdrawalScheduleType$ = exports.AchWithdrawalScheduleType$outboundSchema = exports.AchWithdrawalScheduleType$inboundSchema = exports.AchWithdrawalScheduleStateTaxWithholding$ = exports.AchWithdrawalScheduleStateTaxWithholding$outboundSchema = exports.AchWithdrawalScheduleStateTaxWithholding$inboundSchema = exports.AchWithdrawalScheduleIraDistributionPercentage$ = exports.AchWithdrawalScheduleIraDistributionPercentage$outboundSchema = exports.AchWithdrawalScheduleIraDistributionPercentage$inboundSchema = exports.AchWithdrawalScheduleIraDistributionAmount$ = exports.AchWithdrawalScheduleIraDistributionAmount$outboundSchema = exports.AchWithdrawalScheduleIraDistributionAmount$inboundSchema = exports.AchWithdrawalScheduleFederalTaxWithholding$ = exports.AchWithdrawalScheduleFederalTaxWithholding$outboundSchema = exports.AchWithdrawalScheduleFederalTaxWithholding$inboundSchema = exports.AchWithdrawalSchedulePercentage$ = exports.AchWithdrawalSchedulePercentage$outboundSchema = exports.AchWithdrawalSchedulePercentage$inboundSchema = exports.AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$ = exports.AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$outboundSchema = exports.AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$inboundSchema = exports.AchWithdrawalScheduleTimeUnit = exports.AchWithdrawalScheduleState = exports.AchWithdrawalScheduleType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * The type of retirement distribution.
 */
var AchWithdrawalScheduleType;
(function (AchWithdrawalScheduleType) {
    AchWithdrawalScheduleType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    AchWithdrawalScheduleType["Normal"] = "NORMAL";
    AchWithdrawalScheduleType["Disability"] = "DISABILITY";
    AchWithdrawalScheduleType["Sosepp"] = "SOSEPP";
    AchWithdrawalScheduleType["Premature"] = "PREMATURE";
    AchWithdrawalScheduleType["Death"] = "DEATH";
    AchWithdrawalScheduleType["ExcessContributionRemovalBeforeTaxDeadline"] = "EXCESS_CONTRIBUTION_REMOVAL_BEFORE_TAX_DEADLINE";
    AchWithdrawalScheduleType["ExcessContributionRemovalAfterTaxDeadline"] = "EXCESS_CONTRIBUTION_REMOVAL_AFTER_TAX_DEADLINE";
    AchWithdrawalScheduleType["RolloverToQualifiedPlan"] = "ROLLOVER_TO_QUALIFIED_PLAN";
    AchWithdrawalScheduleType["RolloverToIra"] = "ROLLOVER_TO_IRA";
    AchWithdrawalScheduleType["DistributionTransfer"] = "DISTRIBUTION_TRANSFER";
    AchWithdrawalScheduleType["RecharacterizationPriorYear"] = "RECHARACTERIZATION_PRIOR_YEAR";
    AchWithdrawalScheduleType["RecharacterizationCurrentYear"] = "RECHARACTERIZATION_CURRENT_YEAR";
    AchWithdrawalScheduleType["DistributionConversion"] = "DISTRIBUTION_CONVERSION";
    AchWithdrawalScheduleType["ManagementFee"] = "MANAGEMENT_FEE";
    AchWithdrawalScheduleType["PlanLoan401K"] = "PLAN_LOAN_401K";
    AchWithdrawalScheduleType["PrematureSimpleIraLessThan2Years"] = "PREMATURE_SIMPLE_IRA_LESS_THAN_2_YEARS";
    AchWithdrawalScheduleType["NormalRothIraGreaterThan5Years"] = "NORMAL_ROTH_IRA_GREATER_THAN_5_YEARS";
    AchWithdrawalScheduleType["NetIncomeAttributable"] = "NET_INCOME_ATTRIBUTABLE";
    AchWithdrawalScheduleType["Revocation"] = "REVOCATION";
    AchWithdrawalScheduleType["NonReportable"] = "NON_REPORTABLE";
})(AchWithdrawalScheduleType || (exports.AchWithdrawalScheduleType = AchWithdrawalScheduleType = {}));
/**
 * The state of the represented schedule
 */
var AchWithdrawalScheduleState;
(function (AchWithdrawalScheduleState) {
    AchWithdrawalScheduleState["Active"] = "ACTIVE";
    AchWithdrawalScheduleState["Canceled"] = "CANCELED";
    AchWithdrawalScheduleState["Completed"] = "COMPLETED";
})(AchWithdrawalScheduleState || (exports.AchWithdrawalScheduleState = AchWithdrawalScheduleState = {}));
/**
 * The time unit used to calculate the interval between transfers. The time period between transfers in a scheduled series is the unit of time times the multiplier
 */
var AchWithdrawalScheduleTimeUnit;
(function (AchWithdrawalScheduleTimeUnit) {
    AchWithdrawalScheduleTimeUnit["Day"] = "DAY";
    AchWithdrawalScheduleTimeUnit["Week"] = "WEEK";
    AchWithdrawalScheduleTimeUnit["Month"] = "MONTH";
})(AchWithdrawalScheduleTimeUnit || (exports.AchWithdrawalScheduleTimeUnit = AchWithdrawalScheduleTimeUnit = {}));
/** @internal */
exports.AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$;
(function (AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$) {
    /** @deprecated use `AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$inboundSchema` instead. */
    AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$.inboundSchema = exports.AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$outboundSchema` instead. */
    AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$.outboundSchema = exports.AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$outboundSchema;
})(AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$ || (exports.AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$ = AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$ = {}));
/** @internal */
exports.AchWithdrawalSchedulePercentage$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.AchWithdrawalSchedulePercentage$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalSchedulePercentage$;
(function (AchWithdrawalSchedulePercentage$) {
    /** @deprecated use `AchWithdrawalSchedulePercentage$inboundSchema` instead. */
    AchWithdrawalSchedulePercentage$.inboundSchema = exports.AchWithdrawalSchedulePercentage$inboundSchema;
    /** @deprecated use `AchWithdrawalSchedulePercentage$outboundSchema` instead. */
    AchWithdrawalSchedulePercentage$.outboundSchema = exports.AchWithdrawalSchedulePercentage$outboundSchema;
})(AchWithdrawalSchedulePercentage$ || (exports.AchWithdrawalSchedulePercentage$ = AchWithdrawalSchedulePercentage$ = {}));
/** @internal */
exports.AchWithdrawalScheduleFederalTaxWithholding$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$inboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.AchWithdrawalSchedulePercentage$inboundSchema)).optional(),
});
/** @internal */
exports.AchWithdrawalScheduleFederalTaxWithholding$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleIraDistributionFederalTaxWithholdingAmount$outboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.AchWithdrawalSchedulePercentage$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleFederalTaxWithholding$;
(function (AchWithdrawalScheduleFederalTaxWithholding$) {
    /** @deprecated use `AchWithdrawalScheduleFederalTaxWithholding$inboundSchema` instead. */
    AchWithdrawalScheduleFederalTaxWithholding$.inboundSchema = exports.AchWithdrawalScheduleFederalTaxWithholding$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleFederalTaxWithholding$outboundSchema` instead. */
    AchWithdrawalScheduleFederalTaxWithholding$.outboundSchema = exports.AchWithdrawalScheduleFederalTaxWithholding$outboundSchema;
})(AchWithdrawalScheduleFederalTaxWithholding$ || (exports.AchWithdrawalScheduleFederalTaxWithholding$ = AchWithdrawalScheduleFederalTaxWithholding$ = {}));
/** @internal */
exports.AchWithdrawalScheduleIraDistributionAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.AchWithdrawalScheduleIraDistributionAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleIraDistributionAmount$;
(function (AchWithdrawalScheduleIraDistributionAmount$) {
    /** @deprecated use `AchWithdrawalScheduleIraDistributionAmount$inboundSchema` instead. */
    AchWithdrawalScheduleIraDistributionAmount$.inboundSchema = exports.AchWithdrawalScheduleIraDistributionAmount$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleIraDistributionAmount$outboundSchema` instead. */
    AchWithdrawalScheduleIraDistributionAmount$.outboundSchema = exports.AchWithdrawalScheduleIraDistributionAmount$outboundSchema;
})(AchWithdrawalScheduleIraDistributionAmount$ || (exports.AchWithdrawalScheduleIraDistributionAmount$ = AchWithdrawalScheduleIraDistributionAmount$ = {}));
/** @internal */
exports.AchWithdrawalScheduleIraDistributionPercentage$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.AchWithdrawalScheduleIraDistributionPercentage$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleIraDistributionPercentage$;
(function (AchWithdrawalScheduleIraDistributionPercentage$) {
    /** @deprecated use `AchWithdrawalScheduleIraDistributionPercentage$inboundSchema` instead. */
    AchWithdrawalScheduleIraDistributionPercentage$.inboundSchema = exports.AchWithdrawalScheduleIraDistributionPercentage$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleIraDistributionPercentage$outboundSchema` instead. */
    AchWithdrawalScheduleIraDistributionPercentage$.outboundSchema = exports.AchWithdrawalScheduleIraDistributionPercentage$outboundSchema;
})(AchWithdrawalScheduleIraDistributionPercentage$ || (exports.AchWithdrawalScheduleIraDistributionPercentage$ = AchWithdrawalScheduleIraDistributionPercentage$ = {}));
/** @internal */
exports.AchWithdrawalScheduleStateTaxWithholding$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleIraDistributionAmount$inboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleIraDistributionPercentage$inboundSchema)).optional(),
});
/** @internal */
exports.AchWithdrawalScheduleStateTaxWithholding$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleIraDistributionAmount$outboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleIraDistributionPercentage$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleStateTaxWithholding$;
(function (AchWithdrawalScheduleStateTaxWithholding$) {
    /** @deprecated use `AchWithdrawalScheduleStateTaxWithholding$inboundSchema` instead. */
    AchWithdrawalScheduleStateTaxWithholding$.inboundSchema = exports.AchWithdrawalScheduleStateTaxWithholding$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleStateTaxWithholding$outboundSchema` instead. */
    AchWithdrawalScheduleStateTaxWithholding$.outboundSchema = exports.AchWithdrawalScheduleStateTaxWithholding$outboundSchema;
})(AchWithdrawalScheduleStateTaxWithholding$ || (exports.AchWithdrawalScheduleStateTaxWithholding$ = AchWithdrawalScheduleStateTaxWithholding$ = {}));
/** @internal */
exports.AchWithdrawalScheduleType$inboundSchema = z
    .union([
    z.nativeEnum(AchWithdrawalScheduleType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.AchWithdrawalScheduleType$outboundSchema = z.union([
    z.nativeEnum(AchWithdrawalScheduleType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleType$;
(function (AchWithdrawalScheduleType$) {
    /** @deprecated use `AchWithdrawalScheduleType$inboundSchema` instead. */
    AchWithdrawalScheduleType$.inboundSchema = exports.AchWithdrawalScheduleType$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleType$outboundSchema` instead. */
    AchWithdrawalScheduleType$.outboundSchema = exports.AchWithdrawalScheduleType$outboundSchema;
})(AchWithdrawalScheduleType$ || (exports.AchWithdrawalScheduleType$ = AchWithdrawalScheduleType$ = {}));
/** @internal */
exports.IraDistribution$inboundSchema = z.object({
    federal_tax_withholding: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleFederalTaxWithholding$inboundSchema)).optional(),
    receiving_institution: z.string().optional(),
    state_tax_withholding: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleStateTaxWithholding$inboundSchema)).optional(),
    state_withholding_waiver: z.boolean().optional(),
    tax_year: z.number().int().optional(),
    type: exports.AchWithdrawalScheduleType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "federal_tax_withholding": "federalTaxWithholding",
        "receiving_institution": "receivingInstitution",
        "state_tax_withholding": "stateTaxWithholding",
        "state_withholding_waiver": "stateWithholdingWaiver",
        "tax_year": "taxYear",
    });
});
/** @internal */
exports.IraDistribution$outboundSchema = z.object({
    federalTaxWithholding: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleFederalTaxWithholding$outboundSchema)).optional(),
    receivingInstitution: z.string().optional(),
    stateTaxWithholding: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleStateTaxWithholding$outboundSchema)).optional(),
    stateWithholdingWaiver: z.boolean().optional(),
    taxYear: z.number().int().optional(),
    type: exports.AchWithdrawalScheduleType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        federalTaxWithholding: "federal_tax_withholding",
        receivingInstitution: "receiving_institution",
        stateTaxWithholding: "state_tax_withholding",
        stateWithholdingWaiver: "state_withholding_waiver",
        taxYear: "tax_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IraDistribution$;
(function (IraDistribution$) {
    /** @deprecated use `IraDistribution$inboundSchema` instead. */
    IraDistribution$.inboundSchema = exports.IraDistribution$inboundSchema;
    /** @deprecated use `IraDistribution$outboundSchema` instead. */
    IraDistribution$.outboundSchema = exports.IraDistribution$outboundSchema;
})(IraDistribution$ || (exports.IraDistribution$ = IraDistribution$ = {}));
/** @internal */
exports.AchWithdrawalScheduleAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.AchWithdrawalScheduleAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleAmount$;
(function (AchWithdrawalScheduleAmount$) {
    /** @deprecated use `AchWithdrawalScheduleAmount$inboundSchema` instead. */
    AchWithdrawalScheduleAmount$.inboundSchema = exports.AchWithdrawalScheduleAmount$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleAmount$outboundSchema` instead. */
    AchWithdrawalScheduleAmount$.outboundSchema = exports.AchWithdrawalScheduleAmount$outboundSchema;
})(AchWithdrawalScheduleAmount$ || (exports.AchWithdrawalScheduleAmount$ = AchWithdrawalScheduleAmount$ = {}));
/** @internal */
exports.AchWithdrawalScheduleStartDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.AchWithdrawalScheduleStartDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleStartDate$;
(function (AchWithdrawalScheduleStartDate$) {
    /** @deprecated use `AchWithdrawalScheduleStartDate$inboundSchema` instead. */
    AchWithdrawalScheduleStartDate$.inboundSchema = exports.AchWithdrawalScheduleStartDate$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleStartDate$outboundSchema` instead. */
    AchWithdrawalScheduleStartDate$.outboundSchema = exports.AchWithdrawalScheduleStartDate$outboundSchema;
})(AchWithdrawalScheduleStartDate$ || (exports.AchWithdrawalScheduleStartDate$ = AchWithdrawalScheduleStartDate$ = {}));
/** @internal */
exports.AchWithdrawalScheduleState$inboundSchema = z
    .union([
    z.nativeEnum(AchWithdrawalScheduleState),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.AchWithdrawalScheduleState$outboundSchema = z.union([
    z.nativeEnum(AchWithdrawalScheduleState),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleState$;
(function (AchWithdrawalScheduleState$) {
    /** @deprecated use `AchWithdrawalScheduleState$inboundSchema` instead. */
    AchWithdrawalScheduleState$.inboundSchema = exports.AchWithdrawalScheduleState$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleState$outboundSchema` instead. */
    AchWithdrawalScheduleState$.outboundSchema = exports.AchWithdrawalScheduleState$outboundSchema;
})(AchWithdrawalScheduleState$ || (exports.AchWithdrawalScheduleState$ = AchWithdrawalScheduleState$ = {}));
/** @internal */
exports.AchWithdrawalScheduleTimeUnit$inboundSchema = z
    .union([
    z.nativeEnum(AchWithdrawalScheduleTimeUnit),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.AchWithdrawalScheduleTimeUnit$outboundSchema = z.union([
    z.nativeEnum(AchWithdrawalScheduleTimeUnit),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleTimeUnit$;
(function (AchWithdrawalScheduleTimeUnit$) {
    /** @deprecated use `AchWithdrawalScheduleTimeUnit$inboundSchema` instead. */
    AchWithdrawalScheduleTimeUnit$.inboundSchema = exports.AchWithdrawalScheduleTimeUnit$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleTimeUnit$outboundSchema` instead. */
    AchWithdrawalScheduleTimeUnit$.outboundSchema = exports.AchWithdrawalScheduleTimeUnit$outboundSchema;
})(AchWithdrawalScheduleTimeUnit$ || (exports.AchWithdrawalScheduleTimeUnit$ = AchWithdrawalScheduleTimeUnit$ = {}));
/** @internal */
exports.AchWithdrawalScheduleScheduleProperties$inboundSchema = z.object({
    occurrences: z.number().int().optional(),
    start_date: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleStartDate$inboundSchema)).optional(),
    state: exports.AchWithdrawalScheduleState$inboundSchema.optional(),
    time_unit: exports.AchWithdrawalScheduleTimeUnit$inboundSchema.optional(),
    unit_multiplier: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "start_date": "startDate",
        "time_unit": "timeUnit",
        "unit_multiplier": "unitMultiplier",
    });
});
/** @internal */
exports.AchWithdrawalScheduleScheduleProperties$outboundSchema = z.object({
    occurrences: z.number().int().optional(),
    startDate: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleStartDate$outboundSchema)).optional(),
    state: exports.AchWithdrawalScheduleState$outboundSchema.optional(),
    timeUnit: exports.AchWithdrawalScheduleTimeUnit$outboundSchema.optional(),
    unitMultiplier: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        startDate: "start_date",
        timeUnit: "time_unit",
        unitMultiplier: "unit_multiplier",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleScheduleProperties$;
(function (AchWithdrawalScheduleScheduleProperties$) {
    /** @deprecated use `AchWithdrawalScheduleScheduleProperties$inboundSchema` instead. */
    AchWithdrawalScheduleScheduleProperties$.inboundSchema = exports.AchWithdrawalScheduleScheduleProperties$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleScheduleProperties$outboundSchema` instead. */
    AchWithdrawalScheduleScheduleProperties$.outboundSchema = exports.AchWithdrawalScheduleScheduleProperties$outboundSchema;
})(AchWithdrawalScheduleScheduleProperties$ || (exports.AchWithdrawalScheduleScheduleProperties$ = AchWithdrawalScheduleScheduleProperties$ = {}));
/** @internal */
exports.AchWithdrawalScheduleScheduleDetails$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleAmount$inboundSchema))
        .optional(),
    client_schedule_id: z.string().optional(),
    full_disbursement: z.boolean().optional(),
    schedule_properties: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleScheduleProperties$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_schedule_id": "clientScheduleId",
        "full_disbursement": "fullDisbursement",
        "schedule_properties": "scheduleProperties",
    });
});
/** @internal */
exports.AchWithdrawalScheduleScheduleDetails$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleAmount$outboundSchema))
        .optional(),
    clientScheduleId: z.string().optional(),
    fullDisbursement: z.boolean().optional(),
    scheduleProperties: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleScheduleProperties$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientScheduleId: "client_schedule_id",
        fullDisbursement: "full_disbursement",
        scheduleProperties: "schedule_properties",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalScheduleScheduleDetails$;
(function (AchWithdrawalScheduleScheduleDetails$) {
    /** @deprecated use `AchWithdrawalScheduleScheduleDetails$inboundSchema` instead. */
    AchWithdrawalScheduleScheduleDetails$.inboundSchema = exports.AchWithdrawalScheduleScheduleDetails$inboundSchema;
    /** @deprecated use `AchWithdrawalScheduleScheduleDetails$outboundSchema` instead. */
    AchWithdrawalScheduleScheduleDetails$.outboundSchema = exports.AchWithdrawalScheduleScheduleDetails$outboundSchema;
})(AchWithdrawalScheduleScheduleDetails$ || (exports.AchWithdrawalScheduleScheduleDetails$ = AchWithdrawalScheduleScheduleDetails$ = {}));
/** @internal */
exports.AchWithdrawalSchedule$inboundSchema = z.object({
    bank_relationship: z.string().optional(),
    ira_distribution: z.nullable(z.lazy(() => exports.IraDistribution$inboundSchema))
        .optional(),
    name: z.string().optional(),
    schedule_details: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleScheduleDetails$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "bank_relationship": "bankRelationship",
        "ira_distribution": "iraDistribution",
        "schedule_details": "scheduleDetails",
    });
});
/** @internal */
exports.AchWithdrawalSchedule$outboundSchema = z.object({
    bankRelationship: z.string().optional(),
    iraDistribution: z.nullable(z.lazy(() => exports.IraDistribution$outboundSchema))
        .optional(),
    name: z.string().optional(),
    scheduleDetails: z.nullable(z.lazy(() => exports.AchWithdrawalScheduleScheduleDetails$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        bankRelationship: "bank_relationship",
        iraDistribution: "ira_distribution",
        scheduleDetails: "schedule_details",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalSchedule$;
(function (AchWithdrawalSchedule$) {
    /** @deprecated use `AchWithdrawalSchedule$inboundSchema` instead. */
    AchWithdrawalSchedule$.inboundSchema = exports.AchWithdrawalSchedule$inboundSchema;
    /** @deprecated use `AchWithdrawalSchedule$outboundSchema` instead. */
    AchWithdrawalSchedule$.outboundSchema = exports.AchWithdrawalSchedule$outboundSchema;
})(AchWithdrawalSchedule$ || (exports.AchWithdrawalSchedule$ = AchWithdrawalSchedule$ = {}));
