// @apexfintechsolutions/ascend-sdk/models/components/assetcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssetCreate$ = exports.AssetCreate$outboundSchema = exports.AssetCreate$inboundSchema = exports.AssetCreateType$ = exports.AssetCreateType$outboundSchema = exports.AssetCreateType$inboundSchema = exports.AssetCreateType = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
const positioncreate_js_1 = require("./positioncreate.js");
/**
 * The asset identifier type
 */
var AssetCreateType;
(function (AssetCreateType) {
    AssetCreateType["IdentifierTypeUnspecified"] = "IDENTIFIER_TYPE_UNSPECIFIED";
    AssetCreateType["CurrencyCode"] = "CURRENCY_CODE";
    AssetCreateType["Cusip"] = "CUSIP";
    AssetCreateType["Symbol"] = "SYMBOL";
    AssetCreateType["Isin"] = "ISIN";
    AssetCreateType["AssetId"] = "ASSET_ID";
})(AssetCreateType || (exports.AssetCreateType = AssetCreateType = {}));
/** @internal */
exports.AssetCreateType$inboundSchema = z
    .union([
    z.nativeEnum(AssetCreateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.AssetCreateType$outboundSchema = z.union([
    z.nativeEnum(AssetCreateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AssetCreateType$;
(function (AssetCreateType$) {
    /** @deprecated use `AssetCreateType$inboundSchema` instead. */
    AssetCreateType$.inboundSchema = exports.AssetCreateType$inboundSchema;
    /** @deprecated use `AssetCreateType$outboundSchema` instead. */
    AssetCreateType$.outboundSchema = exports.AssetCreateType$outboundSchema;
})(AssetCreateType$ || (exports.AssetCreateType$ = AssetCreateType$ = {}));
/** @internal */
exports.AssetCreate$inboundSchema = z.object({
    identifier: z.string(),
    position: positioncreate_js_1.PositionCreate$inboundSchema,
    type: exports.AssetCreateType$inboundSchema,
});
/** @internal */
exports.AssetCreate$outboundSchema = z.object({
    identifier: z.string(),
    position: positioncreate_js_1.PositionCreate$outboundSchema,
    type: exports.AssetCreateType$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AssetCreate$;
(function (AssetCreate$) {
    /** @deprecated use `AssetCreate$inboundSchema` instead. */
    AssetCreate$.inboundSchema = exports.AssetCreate$inboundSchema;
    /** @deprecated use `AssetCreate$outboundSchema` instead. */
    AssetCreate$.outboundSchema = exports.AssetCreate$outboundSchema;
})(AssetCreate$ || (exports.AssetCreate$ = AssetCreate$ = {}));
