// @apexfintechsolutions/ascend-sdk/models/components/bankaccountupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankAccountUpdate$ = exports.BankAccountUpdate$outboundSchema = exports.BankAccountUpdate$inboundSchema = exports.BankAccountUpdateType$ = exports.BankAccountUpdateType$outboundSchema = exports.BankAccountUpdateType$inboundSchema = exports.BankAccountUpdateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * The bank account type.
 */
var BankAccountUpdateType;
(function (BankAccountUpdateType) {
    BankAccountUpdateType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    BankAccountUpdateType["Checking"] = "CHECKING";
    BankAccountUpdateType["Savings"] = "SAVINGS";
})(BankAccountUpdateType || (exports.BankAccountUpdateType = BankAccountUpdateType = {}));
/** @internal */
exports.BankAccountUpdateType$inboundSchema = z
    .union([
    z.nativeEnum(BankAccountUpdateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.BankAccountUpdateType$outboundSchema = z.union([
    z.nativeEnum(BankAccountUpdateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BankAccountUpdateType$;
(function (BankAccountUpdateType$) {
    /** @deprecated use `BankAccountUpdateType$inboundSchema` instead. */
    BankAccountUpdateType$.inboundSchema = exports.BankAccountUpdateType$inboundSchema;
    /** @deprecated use `BankAccountUpdateType$outboundSchema` instead. */
    BankAccountUpdateType$.outboundSchema = exports.BankAccountUpdateType$outboundSchema;
})(BankAccountUpdateType$ || (exports.BankAccountUpdateType$ = BankAccountUpdateType$ = {}));
/** @internal */
exports.BankAccountUpdate$inboundSchema = z.object({
    account_number: z.string().optional(),
    owner: z.string().optional(),
    routing_number: z.string().optional(),
    type: exports.BankAccountUpdateType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_number": "accountNumber",
        "routing_number": "routingNumber",
    });
});
/** @internal */
exports.BankAccountUpdate$outboundSchema = z.object({
    accountNumber: z.string().optional(),
    owner: z.string().optional(),
    routingNumber: z.string().optional(),
    type: exports.BankAccountUpdateType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountNumber: "account_number",
        routingNumber: "routing_number",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BankAccountUpdate$;
(function (BankAccountUpdate$) {
    /** @deprecated use `BankAccountUpdate$inboundSchema` instead. */
    BankAccountUpdate$.inboundSchema = exports.BankAccountUpdate$inboundSchema;
    /** @deprecated use `BankAccountUpdate$outboundSchema` instead. */
    BankAccountUpdate$.outboundSchema = exports.BankAccountUpdate$outboundSchema;
})(BankAccountUpdate$ || (exports.BankAccountUpdate$ = BankAccountUpdate$ = {}));
