// @apexfintechsolutions/ascend-sdk/models/components/bankaccountcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankAccountCreate$ = exports.BankAccountCreate$outboundSchema = exports.BankAccountCreate$inboundSchema = exports.BankAccountCreateType$ = exports.BankAccountCreateType$outboundSchema = exports.BankAccountCreateType$inboundSchema = exports.BankAccountCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * The bank account type.
 */
var BankAccountCreateType;
(function (BankAccountCreateType) {
    BankAccountCreateType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    BankAccountCreateType["Checking"] = "CHECKING";
    BankAccountCreateType["Savings"] = "SAVINGS";
})(BankAccountCreateType || (exports.BankAccountCreateType = BankAccountCreateType = {}));
/** @internal */
exports.BankAccountCreateType$inboundSchema = z
    .union([
    z.nativeEnum(BankAccountCreateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.BankAccountCreateType$outboundSchema = z.union([
    z.nativeEnum(BankAccountCreateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BankAccountCreateType$;
(function (BankAccountCreateType$) {
    /** @deprecated use `BankAccountCreateType$inboundSchema` instead. */
    BankAccountCreateType$.inboundSchema = exports.BankAccountCreateType$inboundSchema;
    /** @deprecated use `BankAccountCreateType$outboundSchema` instead. */
    BankAccountCreateType$.outboundSchema = exports.BankAccountCreateType$outboundSchema;
})(BankAccountCreateType$ || (exports.BankAccountCreateType$ = BankAccountCreateType$ = {}));
/** @internal */
exports.BankAccountCreate$inboundSchema = z.object({
    account_number: z.string(),
    owner: z.string(),
    routing_number: z.string(),
    type: exports.BankAccountCreateType$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_number": "accountNumber",
        "routing_number": "routingNumber",
    });
});
/** @internal */
exports.BankAccountCreate$outboundSchema = z.object({
    accountNumber: z.string(),
    owner: z.string(),
    routingNumber: z.string(),
    type: exports.BankAccountCreateType$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountNumber: "account_number",
        routingNumber: "routing_number",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BankAccountCreate$;
(function (BankAccountCreate$) {
    /** @deprecated use `BankAccountCreate$inboundSchema` instead. */
    BankAccountCreate$.inboundSchema = exports.BankAccountCreate$inboundSchema;
    /** @deprecated use `BankAccountCreate$outboundSchema` instead. */
    BankAccountCreate$.outboundSchema = exports.BankAccountCreate$outboundSchema;
})(BankAccountCreate$ || (exports.BankAccountCreate$ = BankAccountCreate$ = {}));
