// @apexfintechsolutions/ascend-sdk/models/components/bidaskprice.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BidAskPrice$ = exports.BidAskPrice$outboundSchema = exports.BidAskPrice$inboundSchema = exports.BidAskPriceType$ = exports.BidAskPriceType$outboundSchema = exports.BidAskPriceType$inboundSchema = exports.BidAskPricePrice$ = exports.BidAskPricePrice$outboundSchema = exports.BidAskPricePrice$inboundSchema = exports.BidAskPriceType = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
/**
 * The calculation type of this price
 */
var BidAskPriceType;
(function (BidAskPriceType) {
    BidAskPriceType["PercentageOfPar"] = "PERCENTAGE_OF_PAR";
    BidAskPriceType["YieldToWorst"] = "YIELD_TO_WORST";
    BidAskPriceType["YieldToMaturity"] = "YIELD_TO_MATURITY";
})(BidAskPriceType || (exports.BidAskPriceType = BidAskPriceType = {}));
/** @internal */
exports.BidAskPricePrice$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.BidAskPricePrice$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BidAskPricePrice$;
(function (BidAskPricePrice$) {
    /** @deprecated use `BidAskPricePrice$inboundSchema` instead. */
    BidAskPricePrice$.inboundSchema = exports.BidAskPricePrice$inboundSchema;
    /** @deprecated use `BidAskPricePrice$outboundSchema` instead. */
    BidAskPricePrice$.outboundSchema = exports.BidAskPricePrice$outboundSchema;
})(BidAskPricePrice$ || (exports.BidAskPricePrice$ = BidAskPricePrice$ = {}));
/** @internal */
exports.BidAskPriceType$inboundSchema = z
    .union([
    z.nativeEnum(BidAskPriceType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.BidAskPriceType$outboundSchema = z.union([
    z.nativeEnum(BidAskPriceType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BidAskPriceType$;
(function (BidAskPriceType$) {
    /** @deprecated use `BidAskPriceType$inboundSchema` instead. */
    BidAskPriceType$.inboundSchema = exports.BidAskPriceType$inboundSchema;
    /** @deprecated use `BidAskPriceType$outboundSchema` instead. */
    BidAskPriceType$.outboundSchema = exports.BidAskPriceType$outboundSchema;
})(BidAskPriceType$ || (exports.BidAskPriceType$ = BidAskPriceType$ = {}));
/** @internal */
exports.BidAskPrice$inboundSchema = z.object({
    price: z.nullable(z.lazy(() => exports.BidAskPricePrice$inboundSchema)).optional(),
    type: exports.BidAskPriceType$inboundSchema.optional(),
});
/** @internal */
exports.BidAskPrice$outboundSchema = z.object({
    price: z.nullable(z.lazy(() => exports.BidAskPricePrice$outboundSchema)).optional(),
    type: exports.BidAskPriceType$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BidAskPrice$;
(function (BidAskPrice$) {
    /** @deprecated use `BidAskPrice$inboundSchema` instead. */
    BidAskPrice$.inboundSchema = exports.BidAskPrice$inboundSchema;
    /** @deprecated use `BidAskPrice$outboundSchema` instead. */
    BidAskPrice$.outboundSchema = exports.BidAskPrice$outboundSchema;
})(BidAskPrice$ || (exports.BidAskPrice$ = BidAskPrice$ = {}));
