// @apexfintechsolutions/ascend-sdk/models/components/bondyield.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BondYield$ = exports.BondYield$outboundSchema = exports.BondYield$inboundSchema = exports.BondYieldYieldType$ = exports.BondYieldYieldType$outboundSchema = exports.BondYieldYieldType$inboundSchema = exports.Percent$ = exports.Percent$outboundSchema = exports.Percent$inboundSchema = exports.BondYieldYieldType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * The type of yield.
 */
var BondYieldYieldType;
(function (BondYieldYieldType) {
    BondYieldYieldType["YieldTypeUnspecified"] = "YIELD_TYPE_UNSPECIFIED";
    BondYieldYieldType["YieldToCall"] = "YIELD_TO_CALL";
    BondYieldYieldType["YieldToMaturity"] = "YIELD_TO_MATURITY";
    BondYieldYieldType["YieldToPut"] = "YIELD_TO_PUT";
    BondYieldYieldType["YieldToWorst"] = "YIELD_TO_WORST";
})(BondYieldYieldType || (exports.BondYieldYieldType = BondYieldYieldType = {}));
/** @internal */
exports.Percent$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.Percent$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Percent$;
(function (Percent$) {
    /** @deprecated use `Percent$inboundSchema` instead. */
    Percent$.inboundSchema = exports.Percent$inboundSchema;
    /** @deprecated use `Percent$outboundSchema` instead. */
    Percent$.outboundSchema = exports.Percent$outboundSchema;
})(Percent$ || (exports.Percent$ = Percent$ = {}));
/** @internal */
exports.BondYieldYieldType$inboundSchema = z
    .union([
    z.nativeEnum(BondYieldYieldType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.BondYieldYieldType$outboundSchema = z.union([
    z.nativeEnum(BondYieldYieldType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BondYieldYieldType$;
(function (BondYieldYieldType$) {
    /** @deprecated use `BondYieldYieldType$inboundSchema` instead. */
    BondYieldYieldType$.inboundSchema = exports.BondYieldYieldType$inboundSchema;
    /** @deprecated use `BondYieldYieldType$outboundSchema` instead. */
    BondYieldYieldType$.outboundSchema = exports.BondYieldYieldType$outboundSchema;
})(BondYieldYieldType$ || (exports.BondYieldYieldType$ = BondYieldYieldType$ = {}));
/** @internal */
exports.BondYield$inboundSchema = z.object({
    percent: z.nullable(z.lazy(() => exports.Percent$inboundSchema)).optional(),
    yield_type: exports.BondYieldYieldType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "yield_type": "yieldType",
    });
});
/** @internal */
exports.BondYield$outboundSchema = z.object({
    percent: z.nullable(z.lazy(() => exports.Percent$outboundSchema)).optional(),
    yieldType: exports.BondYieldYieldType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        yieldType: "yield_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BondYield$;
(function (BondYield$) {
    /** @deprecated use `BondYield$inboundSchema` instead. */
    BondYield$.inboundSchema = exports.BondYield$inboundSchema;
    /** @deprecated use `BondYield$outboundSchema` instead. */
    BondYield$.outboundSchema = exports.BondYield$outboundSchema;
})(BondYield$ || (exports.BondYield$ = BondYield$ = {}));
