// @apexfintechsolutions/ascend-sdk/models/components/bondyieldcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BondYieldCreate$ = exports.BondYieldCreate$outboundSchema = exports.BondYieldCreate$inboundSchema = exports.YieldType$ = exports.YieldType$outboundSchema = exports.YieldType$inboundSchema = exports.YieldType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
/**
 * The type of yield.
 */
var YieldType;
(function (YieldType) {
    YieldType["YieldTypeUnspecified"] = "YIELD_TYPE_UNSPECIFIED";
    YieldType["YieldToCall"] = "YIELD_TO_CALL";
    YieldType["YieldToMaturity"] = "YIELD_TO_MATURITY";
    YieldType["YieldToPut"] = "YIELD_TO_PUT";
    YieldType["YieldToWorst"] = "YIELD_TO_WORST";
})(YieldType || (exports.YieldType = YieldType = {}));
/** @internal */
exports.YieldType$inboundSchema = z
    .union([
    z.nativeEnum(YieldType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.YieldType$outboundSchema = z.union([
    z.nativeEnum(YieldType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var YieldType$;
(function (YieldType$) {
    /** @deprecated use `YieldType$inboundSchema` instead. */
    YieldType$.inboundSchema = exports.YieldType$inboundSchema;
    /** @deprecated use `YieldType$outboundSchema` instead. */
    YieldType$.outboundSchema = exports.YieldType$outboundSchema;
})(YieldType$ || (exports.YieldType$ = YieldType$ = {}));
/** @internal */
exports.BondYieldCreate$inboundSchema = z.object({
    percent: decimalcreate_js_1.DecimalCreate$inboundSchema,
    yield_type: exports.YieldType$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "yield_type": "yieldType",
    });
});
/** @internal */
exports.BondYieldCreate$outboundSchema = z.object({
    percent: decimalcreate_js_1.DecimalCreate$outboundSchema,
    yieldType: exports.YieldType$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        yieldType: "yield_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BondYieldCreate$;
(function (BondYieldCreate$) {
    /** @deprecated use `BondYieldCreate$inboundSchema` instead. */
    BondYieldCreate$.inboundSchema = exports.BondYieldCreate$inboundSchema;
    /** @deprecated use `BondYieldCreate$outboundSchema` instead. */
    BondYieldCreate$.outboundSchema = exports.BondYieldCreate$outboundSchema;
})(BondYieldCreate$ || (exports.BondYieldCreate$ = BondYieldCreate$ = {}));
