// @apexfintechsolutions/ascend-sdk/models/components/calculatecashbalanceresponsetransfersummary.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CalculateCashBalanceResponseTransferSummary$ = exports.CalculateCashBalanceResponseTransferSummary$outboundSchema = exports.CalculateCashBalanceResponseTransferSummary$inboundSchema = exports.CalculateCashBalanceResponseTransferSummaryMechanism$ = exports.CalculateCashBalanceResponseTransferSummaryMechanism$outboundSchema = exports.CalculateCashBalanceResponseTransferSummaryMechanism$inboundSchema = exports.CalculateCashBalanceResponseTransferSummaryAmount$ = exports.CalculateCashBalanceResponseTransferSummaryAmount$outboundSchema = exports.CalculateCashBalanceResponseTransferSummaryAmount$inboundSchema = exports.CalculateCashBalanceResponseTransferSummaryMechanism = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
/**
 * The mechanism of the transfer.
 */
var CalculateCashBalanceResponseTransferSummaryMechanism;
(function (CalculateCashBalanceResponseTransferSummaryMechanism) {
    CalculateCashBalanceResponseTransferSummaryMechanism["MechanismUnspecified"] = "MECHANISM_UNSPECIFIED";
    CalculateCashBalanceResponseTransferSummaryMechanism["Acat"] = "ACAT";
    CalculateCashBalanceResponseTransferSummaryMechanism["Ach"] = "ACH";
    CalculateCashBalanceResponseTransferSummaryMechanism["CashJournal"] = "CASH_JOURNAL";
    CalculateCashBalanceResponseTransferSummaryMechanism["Check"] = "CHECK";
    CalculateCashBalanceResponseTransferSummaryMechanism["Credit"] = "CREDIT";
    CalculateCashBalanceResponseTransferSummaryMechanism["Fee"] = "FEE";
    CalculateCashBalanceResponseTransferSummaryMechanism["Ict"] = "ICT";
    CalculateCashBalanceResponseTransferSummaryMechanism["Paypal"] = "PAYPAL";
    CalculateCashBalanceResponseTransferSummaryMechanism["Rtp"] = "RTP";
    CalculateCashBalanceResponseTransferSummaryMechanism["Tpj"] = "TPJ";
    CalculateCashBalanceResponseTransferSummaryMechanism["Wire"] = "WIRE";
    CalculateCashBalanceResponseTransferSummaryMechanism["ExternalAch"] = "EXTERNAL_ACH";
})(CalculateCashBalanceResponseTransferSummaryMechanism || (exports.CalculateCashBalanceResponseTransferSummaryMechanism = CalculateCashBalanceResponseTransferSummaryMechanism = {}));
/** @internal */
exports.CalculateCashBalanceResponseTransferSummaryAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.CalculateCashBalanceResponseTransferSummaryAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CalculateCashBalanceResponseTransferSummaryAmount$;
(function (CalculateCashBalanceResponseTransferSummaryAmount$) {
    /** @deprecated use `CalculateCashBalanceResponseTransferSummaryAmount$inboundSchema` instead. */
    CalculateCashBalanceResponseTransferSummaryAmount$.inboundSchema = exports.CalculateCashBalanceResponseTransferSummaryAmount$inboundSchema;
    /** @deprecated use `CalculateCashBalanceResponseTransferSummaryAmount$outboundSchema` instead. */
    CalculateCashBalanceResponseTransferSummaryAmount$.outboundSchema = exports.CalculateCashBalanceResponseTransferSummaryAmount$outboundSchema;
})(CalculateCashBalanceResponseTransferSummaryAmount$ || (exports.CalculateCashBalanceResponseTransferSummaryAmount$ = CalculateCashBalanceResponseTransferSummaryAmount$ = {}));
/** @internal */
exports.CalculateCashBalanceResponseTransferSummaryMechanism$inboundSchema = z
    .union([
    z.nativeEnum(CalculateCashBalanceResponseTransferSummaryMechanism),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CalculateCashBalanceResponseTransferSummaryMechanism$outboundSchema = z.union([
    z.nativeEnum(CalculateCashBalanceResponseTransferSummaryMechanism),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CalculateCashBalanceResponseTransferSummaryMechanism$;
(function (CalculateCashBalanceResponseTransferSummaryMechanism$) {
    /** @deprecated use `CalculateCashBalanceResponseTransferSummaryMechanism$inboundSchema` instead. */
    CalculateCashBalanceResponseTransferSummaryMechanism$.inboundSchema = exports.CalculateCashBalanceResponseTransferSummaryMechanism$inboundSchema;
    /** @deprecated use `CalculateCashBalanceResponseTransferSummaryMechanism$outboundSchema` instead. */
    CalculateCashBalanceResponseTransferSummaryMechanism$.outboundSchema = exports.CalculateCashBalanceResponseTransferSummaryMechanism$outboundSchema;
})(CalculateCashBalanceResponseTransferSummaryMechanism$ || (exports.CalculateCashBalanceResponseTransferSummaryMechanism$ = CalculateCashBalanceResponseTransferSummaryMechanism$ = {}));
/** @internal */
exports.CalculateCashBalanceResponseTransferSummary$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.CalculateCashBalanceResponseTransferSummaryAmount$inboundSchema)).optional(),
    id: z.string().optional(),
    mechanism: exports.CalculateCashBalanceResponseTransferSummaryMechanism$inboundSchema
        .optional(),
});
/** @internal */
exports.CalculateCashBalanceResponseTransferSummary$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.CalculateCashBalanceResponseTransferSummaryAmount$outboundSchema)).optional(),
    id: z.string().optional(),
    mechanism: exports.CalculateCashBalanceResponseTransferSummaryMechanism$outboundSchema
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CalculateCashBalanceResponseTransferSummary$;
(function (CalculateCashBalanceResponseTransferSummary$) {
    /** @deprecated use `CalculateCashBalanceResponseTransferSummary$inboundSchema` instead. */
    CalculateCashBalanceResponseTransferSummary$.inboundSchema = exports.CalculateCashBalanceResponseTransferSummary$inboundSchema;
    /** @deprecated use `CalculateCashBalanceResponseTransferSummary$outboundSchema` instead. */
    CalculateCashBalanceResponseTransferSummary$.outboundSchema = exports.CalculateCashBalanceResponseTransferSummary$outboundSchema;
})(CalculateCashBalanceResponseTransferSummary$ || (exports.CalculateCashBalanceResponseTransferSummary$ = CalculateCashBalanceResponseTransferSummary$ = {}));
