// @apexfintechsolutions/ascend-sdk/models/components/calculatecashbalanceresponsetradesummary.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CalculateCashBalanceResponseTradeSummary$ = exports.CalculateCashBalanceResponseTradeSummary$outboundSchema = exports.CalculateCashBalanceResponseTradeSummary$inboundSchema = exports.NetAmount$ = exports.NetAmount$outboundSchema = exports.NetAmount$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.NetAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.NetAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NetAmount$;
(function (NetAmount$) {
    /** @deprecated use `NetAmount$inboundSchema` instead. */
    NetAmount$.inboundSchema = exports.NetAmount$inboundSchema;
    /** @deprecated use `NetAmount$outboundSchema` instead. */
    NetAmount$.outboundSchema = exports.NetAmount$outboundSchema;
})(NetAmount$ || (exports.NetAmount$ = NetAmount$ = {}));
/** @internal */
exports.CalculateCashBalanceResponseTradeSummary$inboundSchema = z.object({
    activity: z.string().optional(),
    asset: z.string().optional(),
    net_amount: z.nullable(z.lazy(() => exports.NetAmount$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "net_amount": "netAmount",
    });
});
/** @internal */
exports.CalculateCashBalanceResponseTradeSummary$outboundSchema = z.object({
    activity: z.string().optional(),
    asset: z.string().optional(),
    netAmount: z.nullable(z.lazy(() => exports.NetAmount$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        netAmount: "net_amount",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CalculateCashBalanceResponseTradeSummary$;
(function (CalculateCashBalanceResponseTradeSummary$) {
    /** @deprecated use `CalculateCashBalanceResponseTradeSummary$inboundSchema` instead. */
    CalculateCashBalanceResponseTradeSummary$.inboundSchema = exports.CalculateCashBalanceResponseTradeSummary$inboundSchema;
    /** @deprecated use `CalculateCashBalanceResponseTradeSummary$outboundSchema` instead. */
    CalculateCashBalanceResponseTradeSummary$.outboundSchema = exports.CalculateCashBalanceResponseTradeSummary$outboundSchema;
})(CalculateCashBalanceResponseTradeSummary$ || (exports.CalculateCashBalanceResponseTradeSummary$ = CalculateCashBalanceResponseTradeSummary$ = {}));
