// @apexfintechsolutions/ascend-sdk/models/components/calculatecashbalanceresponseopenordersummary.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CalculateCashBalanceResponseOpenOrderSummary$ = exports.CalculateCashBalanceResponseOpenOrderSummary$outboundSchema = exports.CalculateCashBalanceResponseOpenOrderSummary$inboundSchema = exports.ExpectedNotionalCeilingAmount$ = exports.ExpectedNotionalCeilingAmount$outboundSchema = exports.ExpectedNotionalCeilingAmount$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.ExpectedNotionalCeilingAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.ExpectedNotionalCeilingAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ExpectedNotionalCeilingAmount$;
(function (ExpectedNotionalCeilingAmount$) {
    /** @deprecated use `ExpectedNotionalCeilingAmount$inboundSchema` instead. */
    ExpectedNotionalCeilingAmount$.inboundSchema = exports.ExpectedNotionalCeilingAmount$inboundSchema;
    /** @deprecated use `ExpectedNotionalCeilingAmount$outboundSchema` instead. */
    ExpectedNotionalCeilingAmount$.outboundSchema = exports.ExpectedNotionalCeilingAmount$outboundSchema;
})(ExpectedNotionalCeilingAmount$ || (exports.ExpectedNotionalCeilingAmount$ = ExpectedNotionalCeilingAmount$ = {}));
/** @internal */
exports.CalculateCashBalanceResponseOpenOrderSummary$inboundSchema = z.object({
    asset: z.string().optional(),
    expected_notional_ceiling_amount: z.nullable(z.lazy(() => exports.ExpectedNotionalCeilingAmount$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "expected_notional_ceiling_amount": "expectedNotionalCeilingAmount",
    });
});
/** @internal */
exports.CalculateCashBalanceResponseOpenOrderSummary$outboundSchema = z.object({
    asset: z.string().optional(),
    expectedNotionalCeilingAmount: z.nullable(z.lazy(() => exports.ExpectedNotionalCeilingAmount$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        expectedNotionalCeilingAmount: "expected_notional_ceiling_amount",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CalculateCashBalanceResponseOpenOrderSummary$;
(function (CalculateCashBalanceResponseOpenOrderSummary$) {
    /** @deprecated use `CalculateCashBalanceResponseOpenOrderSummary$inboundSchema` instead. */
    CalculateCashBalanceResponseOpenOrderSummary$.inboundSchema = exports.CalculateCashBalanceResponseOpenOrderSummary$inboundSchema;
    /** @deprecated use `CalculateCashBalanceResponseOpenOrderSummary$outboundSchema` instead. */
    CalculateCashBalanceResponseOpenOrderSummary$.outboundSchema = exports.CalculateCashBalanceResponseOpenOrderSummary$outboundSchema;
})(CalculateCashBalanceResponseOpenOrderSummary$ || (exports.CalculateCashBalanceResponseOpenOrderSummary$ = CalculateCashBalanceResponseOpenOrderSummary$ = {}));
