// @apexfintechsolutions/ascend-sdk/models/components/cashjournal.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CashJournal$ = exports.CashJournal$outboundSchema = exports.CashJournal$inboundSchema = exports.CashJournalState$ = exports.CashJournalState$outboundSchema = exports.CashJournalState$inboundSchema = exports.CashJournalStateState$ = exports.CashJournalStateState$outboundSchema = exports.CashJournalStateState$inboundSchema = exports.CashJournalRetirementDistribution$ = exports.CashJournalRetirementDistribution$outboundSchema = exports.CashJournalRetirementDistribution$inboundSchema = exports.CashJournalRetirementDistributionType$ = exports.CashJournalRetirementDistributionType$outboundSchema = exports.CashJournalRetirementDistributionType$inboundSchema = exports.CashJournalStateTaxWithholding$ = exports.CashJournalStateTaxWithholding$outboundSchema = exports.CashJournalStateTaxWithholding$inboundSchema = exports.CashJournalRetirementDistributionPercentage$ = exports.CashJournalRetirementDistributionPercentage$outboundSchema = exports.CashJournalRetirementDistributionPercentage$inboundSchema = exports.CashJournalRetirementDistributionStateTaxWithholdingAmount$ = exports.CashJournalRetirementDistributionStateTaxWithholdingAmount$outboundSchema = exports.CashJournalRetirementDistributionStateTaxWithholdingAmount$inboundSchema = exports.CashJournalFederalTaxWithholding$ = exports.CashJournalFederalTaxWithholding$outboundSchema = exports.CashJournalFederalTaxWithholding$inboundSchema = exports.CashJournalPercentage$ = exports.CashJournalPercentage$outboundSchema = exports.CashJournalPercentage$inboundSchema = exports.CashJournalRetirementDistributionAmount$ = exports.CashJournalRetirementDistributionAmount$outboundSchema = exports.CashJournalRetirementDistributionAmount$inboundSchema = exports.CashJournalRetirementContribution$ = exports.CashJournalRetirementContribution$outboundSchema = exports.CashJournalRetirementContribution$inboundSchema = exports.CashJournalType$ = exports.CashJournalType$outboundSchema = exports.CashJournalType$inboundSchema = exports.PartyType$ = exports.PartyType$outboundSchema = exports.PartyType$inboundSchema = exports.CashJournalAmount$ = exports.CashJournalAmount$outboundSchema = exports.CashJournalAmount$inboundSchema = exports.CashJournalStateState = exports.CashJournalRetirementDistributionType = exports.CashJournalType = exports.PartyType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Whether a cash journal is first party or third party Determined asynchronously when the transfer is processing, and will be set by the time the transfer is posted
 */
var PartyType;
(function (PartyType) {
    PartyType["PartyTypeUnspecified"] = "PARTY_TYPE_UNSPECIFIED";
    PartyType["FirstParty"] = "FIRST_PARTY";
    PartyType["ThirdParty"] = "THIRD_PARTY";
})(PartyType || (exports.PartyType = PartyType = {}));
/**
 * The type of retirement contribution.
 */
var CashJournalType;
(function (CashJournalType) {
    CashJournalType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    CashJournalType["Regular"] = "REGULAR";
    CashJournalType["Employee"] = "EMPLOYEE";
    CashJournalType["Employer"] = "EMPLOYER";
    CashJournalType["Recharacterization"] = "RECHARACTERIZATION";
    CashJournalType["Rollover60Day"] = "ROLLOVER_60_DAY";
    CashJournalType["RolloverDirect"] = "ROLLOVER_DIRECT";
    CashJournalType["Transfer"] = "TRANSFER";
    CashJournalType["TrusteeFee"] = "TRUSTEE_FEE";
    CashJournalType["Conversion"] = "CONVERSION";
    CashJournalType["Repayment"] = "REPAYMENT";
    CashJournalType["ContributionNonReportable"] = "CONTRIBUTION_NON_REPORTABLE";
})(CashJournalType || (exports.CashJournalType = CashJournalType = {}));
/**
 * The type of retirement distribution.
 */
var CashJournalRetirementDistributionType;
(function (CashJournalRetirementDistributionType) {
    CashJournalRetirementDistributionType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    CashJournalRetirementDistributionType["Normal"] = "NORMAL";
    CashJournalRetirementDistributionType["Disability"] = "DISABILITY";
    CashJournalRetirementDistributionType["Sosepp"] = "SOSEPP";
    CashJournalRetirementDistributionType["Premature"] = "PREMATURE";
    CashJournalRetirementDistributionType["Death"] = "DEATH";
    CashJournalRetirementDistributionType["ExcessContributionRemovalBeforeTaxDeadline"] = "EXCESS_CONTRIBUTION_REMOVAL_BEFORE_TAX_DEADLINE";
    CashJournalRetirementDistributionType["ExcessContributionRemovalAfterTaxDeadline"] = "EXCESS_CONTRIBUTION_REMOVAL_AFTER_TAX_DEADLINE";
    CashJournalRetirementDistributionType["RolloverToQualifiedPlan"] = "ROLLOVER_TO_QUALIFIED_PLAN";
    CashJournalRetirementDistributionType["RolloverToIra"] = "ROLLOVER_TO_IRA";
    CashJournalRetirementDistributionType["DistributionTransfer"] = "DISTRIBUTION_TRANSFER";
    CashJournalRetirementDistributionType["RecharacterizationPriorYear"] = "RECHARACTERIZATION_PRIOR_YEAR";
    CashJournalRetirementDistributionType["RecharacterizationCurrentYear"] = "RECHARACTERIZATION_CURRENT_YEAR";
    CashJournalRetirementDistributionType["DistributionConversion"] = "DISTRIBUTION_CONVERSION";
    CashJournalRetirementDistributionType["ManagementFee"] = "MANAGEMENT_FEE";
    CashJournalRetirementDistributionType["PlanLoan401K"] = "PLAN_LOAN_401K";
    CashJournalRetirementDistributionType["PrematureSimpleIraLessThan2Years"] = "PREMATURE_SIMPLE_IRA_LESS_THAN_2_YEARS";
    CashJournalRetirementDistributionType["NormalRothIraGreaterThan5Years"] = "NORMAL_ROTH_IRA_GREATER_THAN_5_YEARS";
    CashJournalRetirementDistributionType["NetIncomeAttributable"] = "NET_INCOME_ATTRIBUTABLE";
    CashJournalRetirementDistributionType["Revocation"] = "REVOCATION";
    CashJournalRetirementDistributionType["NonReportable"] = "NON_REPORTABLE";
})(CashJournalRetirementDistributionType || (exports.CashJournalRetirementDistributionType = CashJournalRetirementDistributionType = {}));
/**
 * The high level state of a transfer, one of:
 *
 * @remarks
 * - `PROCESSING` - The transfer is being processed and will be posted if successful.
 * - `PENDING_REVIEW` - The transfer is pending review and will continue processing if approved.
 * - `POSTED` - The transfer has been posted to the ledger and will be completed at the end of the processing window if not canceled first.
 * - `COMPLETED` - The transfer has been batched and completed.
 * - `REJECTED` - The transfer was rejected.
 * - `CANCELED` - The transfer was canceled.
 * - `RETURNED` - The transfer was returned.
 * - `POSTPONED` - The transfer is postponed and will resume processing during the next processing window.
 */
var CashJournalStateState;
(function (CashJournalStateState) {
    CashJournalStateState["StateUnspecified"] = "STATE_UNSPECIFIED";
    CashJournalStateState["Processing"] = "PROCESSING";
    CashJournalStateState["PendingReview"] = "PENDING_REVIEW";
    CashJournalStateState["Posted"] = "POSTED";
    CashJournalStateState["Completed"] = "COMPLETED";
    CashJournalStateState["Rejected"] = "REJECTED";
    CashJournalStateState["Canceled"] = "CANCELED";
    CashJournalStateState["Returned"] = "RETURNED";
    CashJournalStateState["Postponed"] = "POSTPONED";
})(CashJournalStateState || (exports.CashJournalStateState = CashJournalStateState = {}));
/** @internal */
exports.CashJournalAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.CashJournalAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalAmount$;
(function (CashJournalAmount$) {
    /** @deprecated use `CashJournalAmount$inboundSchema` instead. */
    CashJournalAmount$.inboundSchema = exports.CashJournalAmount$inboundSchema;
    /** @deprecated use `CashJournalAmount$outboundSchema` instead. */
    CashJournalAmount$.outboundSchema = exports.CashJournalAmount$outboundSchema;
})(CashJournalAmount$ || (exports.CashJournalAmount$ = CashJournalAmount$ = {}));
/** @internal */
exports.PartyType$inboundSchema = z
    .union([
    z.nativeEnum(PartyType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.PartyType$outboundSchema = z.union([
    z.nativeEnum(PartyType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PartyType$;
(function (PartyType$) {
    /** @deprecated use `PartyType$inboundSchema` instead. */
    PartyType$.inboundSchema = exports.PartyType$inboundSchema;
    /** @deprecated use `PartyType$outboundSchema` instead. */
    PartyType$.outboundSchema = exports.PartyType$outboundSchema;
})(PartyType$ || (exports.PartyType$ = PartyType$ = {}));
/** @internal */
exports.CashJournalType$inboundSchema = z
    .union([
    z.nativeEnum(CashJournalType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CashJournalType$outboundSchema = z.union([
    z.nativeEnum(CashJournalType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalType$;
(function (CashJournalType$) {
    /** @deprecated use `CashJournalType$inboundSchema` instead. */
    CashJournalType$.inboundSchema = exports.CashJournalType$inboundSchema;
    /** @deprecated use `CashJournalType$outboundSchema` instead. */
    CashJournalType$.outboundSchema = exports.CashJournalType$outboundSchema;
})(CashJournalType$ || (exports.CashJournalType$ = CashJournalType$ = {}));
/** @internal */
exports.CashJournalRetirementContribution$inboundSchema = z.object({
    tax_year: z.number().int().optional(),
    type: exports.CashJournalType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "tax_year": "taxYear",
    });
});
/** @internal */
exports.CashJournalRetirementContribution$outboundSchema = z.object({
    taxYear: z.number().int().optional(),
    type: exports.CashJournalType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        taxYear: "tax_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalRetirementContribution$;
(function (CashJournalRetirementContribution$) {
    /** @deprecated use `CashJournalRetirementContribution$inboundSchema` instead. */
    CashJournalRetirementContribution$.inboundSchema = exports.CashJournalRetirementContribution$inboundSchema;
    /** @deprecated use `CashJournalRetirementContribution$outboundSchema` instead. */
    CashJournalRetirementContribution$.outboundSchema = exports.CashJournalRetirementContribution$outboundSchema;
})(CashJournalRetirementContribution$ || (exports.CashJournalRetirementContribution$ = CashJournalRetirementContribution$ = {}));
/** @internal */
exports.CashJournalRetirementDistributionAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.CashJournalRetirementDistributionAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalRetirementDistributionAmount$;
(function (CashJournalRetirementDistributionAmount$) {
    /** @deprecated use `CashJournalRetirementDistributionAmount$inboundSchema` instead. */
    CashJournalRetirementDistributionAmount$.inboundSchema = exports.CashJournalRetirementDistributionAmount$inboundSchema;
    /** @deprecated use `CashJournalRetirementDistributionAmount$outboundSchema` instead. */
    CashJournalRetirementDistributionAmount$.outboundSchema = exports.CashJournalRetirementDistributionAmount$outboundSchema;
})(CashJournalRetirementDistributionAmount$ || (exports.CashJournalRetirementDistributionAmount$ = CashJournalRetirementDistributionAmount$ = {}));
/** @internal */
exports.CashJournalPercentage$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.CashJournalPercentage$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalPercentage$;
(function (CashJournalPercentage$) {
    /** @deprecated use `CashJournalPercentage$inboundSchema` instead. */
    CashJournalPercentage$.inboundSchema = exports.CashJournalPercentage$inboundSchema;
    /** @deprecated use `CashJournalPercentage$outboundSchema` instead. */
    CashJournalPercentage$.outboundSchema = exports.CashJournalPercentage$outboundSchema;
})(CashJournalPercentage$ || (exports.CashJournalPercentage$ = CashJournalPercentage$ = {}));
/** @internal */
exports.CashJournalFederalTaxWithholding$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.CashJournalRetirementDistributionAmount$inboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.CashJournalPercentage$inboundSchema))
        .optional(),
});
/** @internal */
exports.CashJournalFederalTaxWithholding$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.CashJournalRetirementDistributionAmount$outboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.CashJournalPercentage$outboundSchema))
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalFederalTaxWithholding$;
(function (CashJournalFederalTaxWithholding$) {
    /** @deprecated use `CashJournalFederalTaxWithholding$inboundSchema` instead. */
    CashJournalFederalTaxWithholding$.inboundSchema = exports.CashJournalFederalTaxWithholding$inboundSchema;
    /** @deprecated use `CashJournalFederalTaxWithholding$outboundSchema` instead. */
    CashJournalFederalTaxWithholding$.outboundSchema = exports.CashJournalFederalTaxWithholding$outboundSchema;
})(CashJournalFederalTaxWithholding$ || (exports.CashJournalFederalTaxWithholding$ = CashJournalFederalTaxWithholding$ = {}));
/** @internal */
exports.CashJournalRetirementDistributionStateTaxWithholdingAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.CashJournalRetirementDistributionStateTaxWithholdingAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalRetirementDistributionStateTaxWithholdingAmount$;
(function (CashJournalRetirementDistributionStateTaxWithholdingAmount$) {
    /** @deprecated use `CashJournalRetirementDistributionStateTaxWithholdingAmount$inboundSchema` instead. */
    CashJournalRetirementDistributionStateTaxWithholdingAmount$.inboundSchema = exports.CashJournalRetirementDistributionStateTaxWithholdingAmount$inboundSchema;
    /** @deprecated use `CashJournalRetirementDistributionStateTaxWithholdingAmount$outboundSchema` instead. */
    CashJournalRetirementDistributionStateTaxWithholdingAmount$.outboundSchema = exports.CashJournalRetirementDistributionStateTaxWithholdingAmount$outboundSchema;
})(CashJournalRetirementDistributionStateTaxWithholdingAmount$ || (exports.CashJournalRetirementDistributionStateTaxWithholdingAmount$ = CashJournalRetirementDistributionStateTaxWithholdingAmount$ = {}));
/** @internal */
exports.CashJournalRetirementDistributionPercentage$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.CashJournalRetirementDistributionPercentage$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalRetirementDistributionPercentage$;
(function (CashJournalRetirementDistributionPercentage$) {
    /** @deprecated use `CashJournalRetirementDistributionPercentage$inboundSchema` instead. */
    CashJournalRetirementDistributionPercentage$.inboundSchema = exports.CashJournalRetirementDistributionPercentage$inboundSchema;
    /** @deprecated use `CashJournalRetirementDistributionPercentage$outboundSchema` instead. */
    CashJournalRetirementDistributionPercentage$.outboundSchema = exports.CashJournalRetirementDistributionPercentage$outboundSchema;
})(CashJournalRetirementDistributionPercentage$ || (exports.CashJournalRetirementDistributionPercentage$ = CashJournalRetirementDistributionPercentage$ = {}));
/** @internal */
exports.CashJournalStateTaxWithholding$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.CashJournalRetirementDistributionStateTaxWithholdingAmount$inboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.CashJournalRetirementDistributionPercentage$inboundSchema)).optional(),
});
/** @internal */
exports.CashJournalStateTaxWithholding$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.CashJournalRetirementDistributionStateTaxWithholdingAmount$outboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.CashJournalRetirementDistributionPercentage$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalStateTaxWithholding$;
(function (CashJournalStateTaxWithholding$) {
    /** @deprecated use `CashJournalStateTaxWithholding$inboundSchema` instead. */
    CashJournalStateTaxWithholding$.inboundSchema = exports.CashJournalStateTaxWithholding$inboundSchema;
    /** @deprecated use `CashJournalStateTaxWithholding$outboundSchema` instead. */
    CashJournalStateTaxWithholding$.outboundSchema = exports.CashJournalStateTaxWithholding$outboundSchema;
})(CashJournalStateTaxWithholding$ || (exports.CashJournalStateTaxWithholding$ = CashJournalStateTaxWithholding$ = {}));
/** @internal */
exports.CashJournalRetirementDistributionType$inboundSchema = z
    .union([
    z.nativeEnum(CashJournalRetirementDistributionType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CashJournalRetirementDistributionType$outboundSchema = z.union([
    z.nativeEnum(CashJournalRetirementDistributionType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalRetirementDistributionType$;
(function (CashJournalRetirementDistributionType$) {
    /** @deprecated use `CashJournalRetirementDistributionType$inboundSchema` instead. */
    CashJournalRetirementDistributionType$.inboundSchema = exports.CashJournalRetirementDistributionType$inboundSchema;
    /** @deprecated use `CashJournalRetirementDistributionType$outboundSchema` instead. */
    CashJournalRetirementDistributionType$.outboundSchema = exports.CashJournalRetirementDistributionType$outboundSchema;
})(CashJournalRetirementDistributionType$ || (exports.CashJournalRetirementDistributionType$ = CashJournalRetirementDistributionType$ = {}));
/** @internal */
exports.CashJournalRetirementDistribution$inboundSchema = z.object({
    federal_tax_withholding: z.nullable(z.lazy(() => exports.CashJournalFederalTaxWithholding$inboundSchema)).optional(),
    receiving_institution: z.string().optional(),
    state_tax_withholding: z.nullable(z.lazy(() => exports.CashJournalStateTaxWithholding$inboundSchema)).optional(),
    state_withholding_waiver: z.boolean().optional(),
    tax_year: z.number().int().optional(),
    type: exports.CashJournalRetirementDistributionType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "federal_tax_withholding": "federalTaxWithholding",
        "receiving_institution": "receivingInstitution",
        "state_tax_withholding": "stateTaxWithholding",
        "state_withholding_waiver": "stateWithholdingWaiver",
        "tax_year": "taxYear",
    });
});
/** @internal */
exports.CashJournalRetirementDistribution$outboundSchema = z.object({
    federalTaxWithholding: z.nullable(z.lazy(() => exports.CashJournalFederalTaxWithholding$outboundSchema)).optional(),
    receivingInstitution: z.string().optional(),
    stateTaxWithholding: z.nullable(z.lazy(() => exports.CashJournalStateTaxWithholding$outboundSchema)).optional(),
    stateWithholdingWaiver: z.boolean().optional(),
    taxYear: z.number().int().optional(),
    type: exports.CashJournalRetirementDistributionType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        federalTaxWithholding: "federal_tax_withholding",
        receivingInstitution: "receiving_institution",
        stateTaxWithholding: "state_tax_withholding",
        stateWithholdingWaiver: "state_withholding_waiver",
        taxYear: "tax_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalRetirementDistribution$;
(function (CashJournalRetirementDistribution$) {
    /** @deprecated use `CashJournalRetirementDistribution$inboundSchema` instead. */
    CashJournalRetirementDistribution$.inboundSchema = exports.CashJournalRetirementDistribution$inboundSchema;
    /** @deprecated use `CashJournalRetirementDistribution$outboundSchema` instead. */
    CashJournalRetirementDistribution$.outboundSchema = exports.CashJournalRetirementDistribution$outboundSchema;
})(CashJournalRetirementDistribution$ || (exports.CashJournalRetirementDistribution$ = CashJournalRetirementDistribution$ = {}));
/** @internal */
exports.CashJournalStateState$inboundSchema = z
    .union([
    z.nativeEnum(CashJournalStateState),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CashJournalStateState$outboundSchema = z.union([
    z.nativeEnum(CashJournalStateState),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalStateState$;
(function (CashJournalStateState$) {
    /** @deprecated use `CashJournalStateState$inboundSchema` instead. */
    CashJournalStateState$.inboundSchema = exports.CashJournalStateState$inboundSchema;
    /** @deprecated use `CashJournalStateState$outboundSchema` instead. */
    CashJournalStateState$.outboundSchema = exports.CashJournalStateState$outboundSchema;
})(CashJournalStateState$ || (exports.CashJournalStateState$ = CashJournalStateState$ = {}));
/** @internal */
exports.CashJournalState$inboundSchema = z.object({
    actor: z.string().optional(),
    message: z.string().optional(),
    metadata: z.nullable(z.record(z.any())).optional(),
    state: exports.CashJournalStateState$inboundSchema.optional(),
    update_time: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "update_time": "updateTime",
    });
});
/** @internal */
exports.CashJournalState$outboundSchema = z.object({
    actor: z.string().optional(),
    message: z.string().optional(),
    metadata: z.nullable(z.record(z.any())).optional(),
    state: exports.CashJournalStateState$outboundSchema.optional(),
    updateTime: z.nullable(z.date().transform(v => v.toISOString())).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        updateTime: "update_time",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalState$;
(function (CashJournalState$) {
    /** @deprecated use `CashJournalState$inboundSchema` instead. */
    CashJournalState$.inboundSchema = exports.CashJournalState$inboundSchema;
    /** @deprecated use `CashJournalState$outboundSchema` instead. */
    CashJournalState$.outboundSchema = exports.CashJournalState$outboundSchema;
})(CashJournalState$ || (exports.CashJournalState$ = CashJournalState$ = {}));
/** @internal */
exports.CashJournal$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.CashJournalAmount$inboundSchema)).optional(),
    client_transfer_id: z.string().optional(),
    destination_account: z.string().optional(),
    name: z.string().optional(),
    party_type: exports.PartyType$inboundSchema.optional(),
    retirement_contribution: z.nullable(z.lazy(() => exports.CashJournalRetirementContribution$inboundSchema)).optional(),
    retirement_distribution: z.nullable(z.lazy(() => exports.CashJournalRetirementDistribution$inboundSchema)).optional(),
    source_account: z.string().optional(),
    state: z.nullable(z.lazy(() => exports.CashJournalState$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_transfer_id": "clientTransferId",
        "destination_account": "destinationAccount",
        "party_type": "partyType",
        "retirement_contribution": "retirementContribution",
        "retirement_distribution": "retirementDistribution",
        "source_account": "sourceAccount",
    });
});
/** @internal */
exports.CashJournal$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.CashJournalAmount$outboundSchema)).optional(),
    clientTransferId: z.string().optional(),
    destinationAccount: z.string().optional(),
    name: z.string().optional(),
    partyType: exports.PartyType$outboundSchema.optional(),
    retirementContribution: z.nullable(z.lazy(() => exports.CashJournalRetirementContribution$outboundSchema)).optional(),
    retirementDistribution: z.nullable(z.lazy(() => exports.CashJournalRetirementDistribution$outboundSchema)).optional(),
    sourceAccount: z.string().optional(),
    state: z.nullable(z.lazy(() => exports.CashJournalState$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientTransferId: "client_transfer_id",
        destinationAccount: "destination_account",
        partyType: "party_type",
        retirementContribution: "retirement_contribution",
        retirementDistribution: "retirement_distribution",
        sourceAccount: "source_account",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournal$;
(function (CashJournal$) {
    /** @deprecated use `CashJournal$inboundSchema` instead. */
    CashJournal$.inboundSchema = exports.CashJournal$inboundSchema;
    /** @deprecated use `CashJournal$outboundSchema` instead. */
    CashJournal$.outboundSchema = exports.CashJournal$outboundSchema;
})(CashJournal$ || (exports.CashJournal$ = CashJournal$ = {}));
