// @apexfintechsolutions/ascend-sdk/models/components/cashjournalconstraints.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CashJournalConstraints$ = exports.CashJournalConstraints$outboundSchema = exports.CashJournalConstraints$inboundSchema = exports.CashJournalConstraintsDistributionConstraints$ = exports.CashJournalConstraintsDistributionConstraints$outboundSchema = exports.CashJournalConstraintsDistributionConstraints$inboundSchema = exports.CashJournalConstraintsContributionConstraints$ = exports.CashJournalConstraintsContributionConstraints$outboundSchema = exports.CashJournalConstraintsContributionConstraints$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const contributionconstraintscontributiontypeinfo_js_1 = require("./contributionconstraintscontributiontypeinfo.js");
const distributionconstraintsdistributiontypeinfo_js_1 = require("./distributionconstraintsdistributiontypeinfo.js");
/** @internal */
exports.CashJournalConstraintsContributionConstraints$inboundSchema = z.object({
    only_allow_current_year: z.boolean().optional(),
    valid_types: z.array(contributionconstraintscontributiontypeinfo_js_1.ContributionConstraintsContributionTypeInfo$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "only_allow_current_year": "onlyAllowCurrentYear",
        "valid_types": "validTypes",
    });
});
/** @internal */
exports.CashJournalConstraintsContributionConstraints$outboundSchema = z.object({
    onlyAllowCurrentYear: z.boolean().optional(),
    validTypes: z.array(contributionconstraintscontributiontypeinfo_js_1.ContributionConstraintsContributionTypeInfo$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        onlyAllowCurrentYear: "only_allow_current_year",
        validTypes: "valid_types",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalConstraintsContributionConstraints$;
(function (CashJournalConstraintsContributionConstraints$) {
    /** @deprecated use `CashJournalConstraintsContributionConstraints$inboundSchema` instead. */
    CashJournalConstraintsContributionConstraints$.inboundSchema = exports.CashJournalConstraintsContributionConstraints$inboundSchema;
    /** @deprecated use `CashJournalConstraintsContributionConstraints$outboundSchema` instead. */
    CashJournalConstraintsContributionConstraints$.outboundSchema = exports.CashJournalConstraintsContributionConstraints$outboundSchema;
})(CashJournalConstraintsContributionConstraints$ || (exports.CashJournalConstraintsContributionConstraints$ = CashJournalConstraintsContributionConstraints$ = {}));
/** @internal */
exports.CashJournalConstraintsDistributionConstraints$inboundSchema = z.object({
    full_distribution_allowed: z.boolean().optional(),
    valid_types: z.array(distributionconstraintsdistributiontypeinfo_js_1.DistributionConstraintsDistributionTypeInfo$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "full_distribution_allowed": "fullDistributionAllowed",
        "valid_types": "validTypes",
    });
});
/** @internal */
exports.CashJournalConstraintsDistributionConstraints$outboundSchema = z.object({
    fullDistributionAllowed: z.boolean().optional(),
    validTypes: z.array(distributionconstraintsdistributiontypeinfo_js_1.DistributionConstraintsDistributionTypeInfo$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        fullDistributionAllowed: "full_distribution_allowed",
        validTypes: "valid_types",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalConstraintsDistributionConstraints$;
(function (CashJournalConstraintsDistributionConstraints$) {
    /** @deprecated use `CashJournalConstraintsDistributionConstraints$inboundSchema` instead. */
    CashJournalConstraintsDistributionConstraints$.inboundSchema = exports.CashJournalConstraintsDistributionConstraints$inboundSchema;
    /** @deprecated use `CashJournalConstraintsDistributionConstraints$outboundSchema` instead. */
    CashJournalConstraintsDistributionConstraints$.outboundSchema = exports.CashJournalConstraintsDistributionConstraints$outboundSchema;
})(CashJournalConstraintsDistributionConstraints$ || (exports.CashJournalConstraintsDistributionConstraints$ = CashJournalConstraintsDistributionConstraints$ = {}));
/** @internal */
exports.CashJournalConstraints$inboundSchema = z.object({
    contribution_constraints: z.nullable(z.lazy(() => exports.CashJournalConstraintsContributionConstraints$inboundSchema)).optional(),
    distribution_constraints: z.nullable(z.lazy(() => exports.CashJournalConstraintsDistributionConstraints$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "contribution_constraints": "contributionConstraints",
        "distribution_constraints": "distributionConstraints",
    });
});
/** @internal */
exports.CashJournalConstraints$outboundSchema = z.object({
    contributionConstraints: z.nullable(z.lazy(() => exports.CashJournalConstraintsContributionConstraints$outboundSchema)).optional(),
    distributionConstraints: z.nullable(z.lazy(() => exports.CashJournalConstraintsDistributionConstraints$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        contributionConstraints: "contribution_constraints",
        distributionConstraints: "distribution_constraints",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalConstraints$;
(function (CashJournalConstraints$) {
    /** @deprecated use `CashJournalConstraints$inboundSchema` instead. */
    CashJournalConstraints$.inboundSchema = exports.CashJournalConstraints$inboundSchema;
    /** @deprecated use `CashJournalConstraints$outboundSchema` instead. */
    CashJournalConstraints$.outboundSchema = exports.CashJournalConstraints$outboundSchema;
})(CashJournalConstraints$ || (exports.CashJournalConstraints$ = CashJournalConstraints$ = {}));
