// @apexfintechsolutions/ascend-sdk/models/components/cashjournalcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CashJournalCreate$ = exports.CashJournalCreate$outboundSchema = exports.CashJournalCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
const retirementcontributioncreate_js_1 = require("./retirementcontributioncreate.js");
const retirementdistributioncreate_js_1 = require("./retirementdistributioncreate.js");
/** @internal */
exports.CashJournalCreate$inboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$inboundSchema.optional(),
    client_transfer_id: z.string(),
    destination_account: z.string(),
    full_disbursement: z.boolean().optional(),
    retirement_contribution: retirementcontributioncreate_js_1.RetirementContributionCreate$inboundSchema
        .optional(),
    retirement_distribution: retirementdistributioncreate_js_1.RetirementDistributionCreate$inboundSchema
        .optional(),
    source_account: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_transfer_id": "clientTransferId",
        "destination_account": "destinationAccount",
        "full_disbursement": "fullDisbursement",
        "retirement_contribution": "retirementContribution",
        "retirement_distribution": "retirementDistribution",
        "source_account": "sourceAccount",
    });
});
/** @internal */
exports.CashJournalCreate$outboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$outboundSchema.optional(),
    clientTransferId: z.string(),
    destinationAccount: z.string(),
    fullDisbursement: z.boolean().optional(),
    retirementContribution: retirementcontributioncreate_js_1.RetirementContributionCreate$outboundSchema
        .optional(),
    retirementDistribution: retirementdistributioncreate_js_1.RetirementDistributionCreate$outboundSchema
        .optional(),
    sourceAccount: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientTransferId: "client_transfer_id",
        destinationAccount: "destination_account",
        fullDisbursement: "full_disbursement",
        retirementContribution: "retirement_contribution",
        retirementDistribution: "retirement_distribution",
        sourceAccount: "source_account",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalCreate$;
(function (CashJournalCreate$) {
    /** @deprecated use `CashJournalCreate$inboundSchema` instead. */
    CashJournalCreate$.inboundSchema = exports.CashJournalCreate$inboundSchema;
    /** @deprecated use `CashJournalCreate$outboundSchema` instead. */
    CashJournalCreate$.outboundSchema = exports.CashJournalCreate$outboundSchema;
})(CashJournalCreate$ || (exports.CashJournalCreate$ = CashJournalCreate$ = {}));
