// @apexfintechsolutions/ascend-sdk/models/components/checkpartytyperesponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckPartyTypeResponse$ = exports.CheckPartyTypeResponse$outboundSchema = exports.CheckPartyTypeResponse$inboundSchema = exports.CheckPartyTypeResponsePartyType$ = exports.CheckPartyTypeResponsePartyType$outboundSchema = exports.CheckPartyTypeResponsePartyType$inboundSchema = exports.CheckPartyTypeResponsePartyType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Whether the cash journal is considered first party or third party
 */
var CheckPartyTypeResponsePartyType;
(function (CheckPartyTypeResponsePartyType) {
    CheckPartyTypeResponsePartyType["PartyTypeUnspecified"] = "PARTY_TYPE_UNSPECIFIED";
    CheckPartyTypeResponsePartyType["FirstParty"] = "FIRST_PARTY";
    CheckPartyTypeResponsePartyType["ThirdParty"] = "THIRD_PARTY";
})(CheckPartyTypeResponsePartyType || (exports.CheckPartyTypeResponsePartyType = CheckPartyTypeResponsePartyType = {}));
/** @internal */
exports.CheckPartyTypeResponsePartyType$inboundSchema = z
    .union([
    z.nativeEnum(CheckPartyTypeResponsePartyType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CheckPartyTypeResponsePartyType$outboundSchema = z.union([
    z.nativeEnum(CheckPartyTypeResponsePartyType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckPartyTypeResponsePartyType$;
(function (CheckPartyTypeResponsePartyType$) {
    /** @deprecated use `CheckPartyTypeResponsePartyType$inboundSchema` instead. */
    CheckPartyTypeResponsePartyType$.inboundSchema = exports.CheckPartyTypeResponsePartyType$inboundSchema;
    /** @deprecated use `CheckPartyTypeResponsePartyType$outboundSchema` instead. */
    CheckPartyTypeResponsePartyType$.outboundSchema = exports.CheckPartyTypeResponsePartyType$outboundSchema;
})(CheckPartyTypeResponsePartyType$ || (exports.CheckPartyTypeResponsePartyType$ = CheckPartyTypeResponsePartyType$ = {}));
/** @internal */
exports.CheckPartyTypeResponse$inboundSchema = z.object({
    party_type: exports.CheckPartyTypeResponsePartyType$inboundSchema.optional(),
    reason: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "party_type": "partyType",
    });
});
/** @internal */
exports.CheckPartyTypeResponse$outboundSchema = z.object({
    partyType: exports.CheckPartyTypeResponsePartyType$outboundSchema.optional(),
    reason: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        partyType: "party_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckPartyTypeResponse$;
(function (CheckPartyTypeResponse$) {
    /** @deprecated use `CheckPartyTypeResponse$inboundSchema` instead. */
    CheckPartyTypeResponse$.inboundSchema = exports.CheckPartyTypeResponse$inboundSchema;
    /** @deprecated use `CheckPartyTypeResponse$outboundSchema` instead. */
    CheckPartyTypeResponse$.outboundSchema = exports.CheckPartyTypeResponse$outboundSchema;
})(CheckPartyTypeResponse$ || (exports.CheckPartyTypeResponse$ = CheckPartyTypeResponse$ = {}));
