// @apexfintechsolutions/ascend-sdk/models/components/commission.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Commission$ = exports.Commission$outboundSchema = exports.Commission$inboundSchema = exports.CommissionAmount1$ = exports.CommissionAmount1$outboundSchema = exports.CommissionAmount1$inboundSchema = void 0;
const z = __importStar(require("zod"));
/** @internal */
exports.CommissionAmount1$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.CommissionAmount1$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CommissionAmount1$;
(function (CommissionAmount1$) {
    /** @deprecated use `CommissionAmount1$inboundSchema` instead. */
    CommissionAmount1$.inboundSchema = exports.CommissionAmount1$inboundSchema;
    /** @deprecated use `CommissionAmount1$outboundSchema` instead. */
    CommissionAmount1$.outboundSchema = exports.CommissionAmount1$outboundSchema;
})(CommissionAmount1$ || (exports.CommissionAmount1$ = CommissionAmount1$ = {}));
/** @internal */
exports.Commission$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.CommissionAmount1$inboundSchema)).optional(),
});
/** @internal */
exports.Commission$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.CommissionAmount1$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Commission$;
(function (Commission$) {
    /** @deprecated use `Commission$inboundSchema` instead. */
    Commission$.inboundSchema = exports.Commission$inboundSchema;
    /** @deprecated use `Commission$outboundSchema` instead. */
    Commission$.outboundSchema = exports.Commission$outboundSchema;
})(Commission$ || (exports.Commission$ = Commission$ = {}));
