// @apexfintechsolutions/ascend-sdk/models/components/completetraderesponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompleteTradeResponse$ = exports.CompleteTradeResponse$outboundSchema = exports.CompleteTradeResponse$inboundSchema = exports.CompleteTradeResponseTrade$ = exports.CompleteTradeResponseTrade$outboundSchema = exports.CompleteTradeResponseTrade$inboundSchema = exports.CompleteTradeResponsePriceAdjustmentRecord$ = exports.CompleteTradeResponsePriceAdjustmentRecord$outboundSchema = exports.CompleteTradeResponsePriceAdjustmentRecord$inboundSchema = exports.CompleteTradeResponsePriceAdjustmentType$ = exports.CompleteTradeResponsePriceAdjustmentType$outboundSchema = exports.CompleteTradeResponsePriceAdjustmentType$inboundSchema = exports.CompleteTradeResponsePriceAdjustmentPercent$ = exports.CompleteTradeResponsePriceAdjustmentPercent$outboundSchema = exports.CompleteTradeResponsePriceAdjustmentPercent$inboundSchema = exports.CompleteTradeResponsePriceAdjustmentAmount$ = exports.CompleteTradeResponsePriceAdjustmentAmount$outboundSchema = exports.CompleteTradeResponsePriceAdjustmentAmount$inboundSchema = exports.CompleteTradeResponsePrevailingMarketPrice$ = exports.CompleteTradeResponsePrevailingMarketPrice$outboundSchema = exports.CompleteTradeResponsePrevailingMarketPrice$inboundSchema = exports.CompleteTradeResponseBrokerCapacity$ = exports.CompleteTradeResponseBrokerCapacity$outboundSchema = exports.CompleteTradeResponseBrokerCapacity$inboundSchema = exports.CompleteTradeResponsePriceAdjustmentType = exports.CompleteTradeResponseBrokerCapacity = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const lot_js_1 = require("./lot.js");
const yieldrecord_js_1 = require("./yieldrecord.js");
/**
 * Used to calculate broadridge blotter code
 */
var CompleteTradeResponseBrokerCapacity;
(function (CompleteTradeResponseBrokerCapacity) {
    CompleteTradeResponseBrokerCapacity["CapacityUnspecified"] = "CAPACITY_UNSPECIFIED";
    CompleteTradeResponseBrokerCapacity["Agency"] = "AGENCY";
    CompleteTradeResponseBrokerCapacity["Principal"] = "PRINCIPAL";
    CompleteTradeResponseBrokerCapacity["Mixed"] = "MIXED";
})(CompleteTradeResponseBrokerCapacity || (exports.CompleteTradeResponseBrokerCapacity = CompleteTradeResponseBrokerCapacity = {}));
/**
 * The type of price adjustment being applied by the broker to the net price of the security
 */
var CompleteTradeResponsePriceAdjustmentType;
(function (CompleteTradeResponsePriceAdjustmentType) {
    CompleteTradeResponsePriceAdjustmentType["PriceAdjustmentTypeUnspecified"] = "PRICE_ADJUSTMENT_TYPE_UNSPECIFIED";
    CompleteTradeResponsePriceAdjustmentType["Markup"] = "MARKUP";
    CompleteTradeResponsePriceAdjustmentType["Markdown"] = "MARKDOWN";
    CompleteTradeResponsePriceAdjustmentType["SalesLoad"] = "SALES_LOAD";
})(CompleteTradeResponsePriceAdjustmentType || (exports.CompleteTradeResponsePriceAdjustmentType = CompleteTradeResponsePriceAdjustmentType = {}));
/** @internal */
exports.CompleteTradeResponseBrokerCapacity$inboundSchema = z
    .union([
    z.nativeEnum(CompleteTradeResponseBrokerCapacity),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CompleteTradeResponseBrokerCapacity$outboundSchema = z.union([
    z.nativeEnum(CompleteTradeResponseBrokerCapacity),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CompleteTradeResponseBrokerCapacity$;
(function (CompleteTradeResponseBrokerCapacity$) {
    /** @deprecated use `CompleteTradeResponseBrokerCapacity$inboundSchema` instead. */
    CompleteTradeResponseBrokerCapacity$.inboundSchema = exports.CompleteTradeResponseBrokerCapacity$inboundSchema;
    /** @deprecated use `CompleteTradeResponseBrokerCapacity$outboundSchema` instead. */
    CompleteTradeResponseBrokerCapacity$.outboundSchema = exports.CompleteTradeResponseBrokerCapacity$outboundSchema;
})(CompleteTradeResponseBrokerCapacity$ || (exports.CompleteTradeResponseBrokerCapacity$ = CompleteTradeResponseBrokerCapacity$ = {}));
/** @internal */
exports.CompleteTradeResponsePrevailingMarketPrice$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.CompleteTradeResponsePrevailingMarketPrice$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CompleteTradeResponsePrevailingMarketPrice$;
(function (CompleteTradeResponsePrevailingMarketPrice$) {
    /** @deprecated use `CompleteTradeResponsePrevailingMarketPrice$inboundSchema` instead. */
    CompleteTradeResponsePrevailingMarketPrice$.inboundSchema = exports.CompleteTradeResponsePrevailingMarketPrice$inboundSchema;
    /** @deprecated use `CompleteTradeResponsePrevailingMarketPrice$outboundSchema` instead. */
    CompleteTradeResponsePrevailingMarketPrice$.outboundSchema = exports.CompleteTradeResponsePrevailingMarketPrice$outboundSchema;
})(CompleteTradeResponsePrevailingMarketPrice$ || (exports.CompleteTradeResponsePrevailingMarketPrice$ = CompleteTradeResponsePrevailingMarketPrice$ = {}));
/** @internal */
exports.CompleteTradeResponsePriceAdjustmentAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.CompleteTradeResponsePriceAdjustmentAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CompleteTradeResponsePriceAdjustmentAmount$;
(function (CompleteTradeResponsePriceAdjustmentAmount$) {
    /** @deprecated use `CompleteTradeResponsePriceAdjustmentAmount$inboundSchema` instead. */
    CompleteTradeResponsePriceAdjustmentAmount$.inboundSchema = exports.CompleteTradeResponsePriceAdjustmentAmount$inboundSchema;
    /** @deprecated use `CompleteTradeResponsePriceAdjustmentAmount$outboundSchema` instead. */
    CompleteTradeResponsePriceAdjustmentAmount$.outboundSchema = exports.CompleteTradeResponsePriceAdjustmentAmount$outboundSchema;
})(CompleteTradeResponsePriceAdjustmentAmount$ || (exports.CompleteTradeResponsePriceAdjustmentAmount$ = CompleteTradeResponsePriceAdjustmentAmount$ = {}));
/** @internal */
exports.CompleteTradeResponsePriceAdjustmentPercent$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.CompleteTradeResponsePriceAdjustmentPercent$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CompleteTradeResponsePriceAdjustmentPercent$;
(function (CompleteTradeResponsePriceAdjustmentPercent$) {
    /** @deprecated use `CompleteTradeResponsePriceAdjustmentPercent$inboundSchema` instead. */
    CompleteTradeResponsePriceAdjustmentPercent$.inboundSchema = exports.CompleteTradeResponsePriceAdjustmentPercent$inboundSchema;
    /** @deprecated use `CompleteTradeResponsePriceAdjustmentPercent$outboundSchema` instead. */
    CompleteTradeResponsePriceAdjustmentPercent$.outboundSchema = exports.CompleteTradeResponsePriceAdjustmentPercent$outboundSchema;
})(CompleteTradeResponsePriceAdjustmentPercent$ || (exports.CompleteTradeResponsePriceAdjustmentPercent$ = CompleteTradeResponsePriceAdjustmentPercent$ = {}));
/** @internal */
exports.CompleteTradeResponsePriceAdjustmentType$inboundSchema = z
    .union([
    z.nativeEnum(CompleteTradeResponsePriceAdjustmentType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CompleteTradeResponsePriceAdjustmentType$outboundSchema = z.union([
    z.nativeEnum(CompleteTradeResponsePriceAdjustmentType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CompleteTradeResponsePriceAdjustmentType$;
(function (CompleteTradeResponsePriceAdjustmentType$) {
    /** @deprecated use `CompleteTradeResponsePriceAdjustmentType$inboundSchema` instead. */
    CompleteTradeResponsePriceAdjustmentType$.inboundSchema = exports.CompleteTradeResponsePriceAdjustmentType$inboundSchema;
    /** @deprecated use `CompleteTradeResponsePriceAdjustmentType$outboundSchema` instead. */
    CompleteTradeResponsePriceAdjustmentType$.outboundSchema = exports.CompleteTradeResponsePriceAdjustmentType$outboundSchema;
})(CompleteTradeResponsePriceAdjustmentType$ || (exports.CompleteTradeResponsePriceAdjustmentType$ = CompleteTradeResponsePriceAdjustmentType$ = {}));
/** @internal */
exports.CompleteTradeResponsePriceAdjustmentRecord$inboundSchema = z.object({
    price_adjustment_amount: z.nullable(z.lazy(() => exports.CompleteTradeResponsePriceAdjustmentAmount$inboundSchema)).optional(),
    price_adjustment_percent: z.nullable(z.lazy(() => exports.CompleteTradeResponsePriceAdjustmentPercent$inboundSchema)).optional(),
    price_adjustment_type: exports.CompleteTradeResponsePriceAdjustmentType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "price_adjustment_amount": "priceAdjustmentAmount",
        "price_adjustment_percent": "priceAdjustmentPercent",
        "price_adjustment_type": "priceAdjustmentType",
    });
});
/** @internal */
exports.CompleteTradeResponsePriceAdjustmentRecord$outboundSchema = z.object({
    priceAdjustmentAmount: z.nullable(z.lazy(() => exports.CompleteTradeResponsePriceAdjustmentAmount$outboundSchema)).optional(),
    priceAdjustmentPercent: z.nullable(z.lazy(() => exports.CompleteTradeResponsePriceAdjustmentPercent$outboundSchema)).optional(),
    priceAdjustmentType: exports.CompleteTradeResponsePriceAdjustmentType$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        priceAdjustmentAmount: "price_adjustment_amount",
        priceAdjustmentPercent: "price_adjustment_percent",
        priceAdjustmentType: "price_adjustment_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CompleteTradeResponsePriceAdjustmentRecord$;
(function (CompleteTradeResponsePriceAdjustmentRecord$) {
    /** @deprecated use `CompleteTradeResponsePriceAdjustmentRecord$inboundSchema` instead. */
    CompleteTradeResponsePriceAdjustmentRecord$.inboundSchema = exports.CompleteTradeResponsePriceAdjustmentRecord$inboundSchema;
    /** @deprecated use `CompleteTradeResponsePriceAdjustmentRecord$outboundSchema` instead. */
    CompleteTradeResponsePriceAdjustmentRecord$.outboundSchema = exports.CompleteTradeResponsePriceAdjustmentRecord$outboundSchema;
})(CompleteTradeResponsePriceAdjustmentRecord$ || (exports.CompleteTradeResponsePriceAdjustmentRecord$ = CompleteTradeResponsePriceAdjustmentRecord$ = {}));
/** @internal */
exports.CompleteTradeResponseTrade$inboundSchema = z.object({
    additional_instructions: z.array(z.string()).optional(),
    alternate_order_id: z.string().optional(),
    booking_api_trade_allocation_id: z.string().optional(),
    booking_api_trade_id: z.string().optional(),
    broker: z.string().optional(),
    broker_capacity: exports.CompleteTradeResponseBrokerCapacity$inboundSchema.optional(),
    client_memos: z.array(z.string()).optional(),
    client_order_id: z.string().optional(),
    exchange: z.string().optional(),
    execution_id: z.string().optional(),
    execution_only: z.boolean().optional(),
    external_id: z.string().optional(),
    fund_confirmation_number: z.string().optional(),
    gateway_client_order_id: z.string().optional(),
    internal_error: z.boolean().optional(),
    is_writeoff: z.boolean().optional(),
    lots: z.array(lot_js_1.Lot$inboundSchema).optional(),
    order_id: z.string().optional(),
    prevailing_market_price: z.nullable(z.lazy(() => exports.CompleteTradeResponsePrevailingMarketPrice$inboundSchema)).optional(),
    price_adjustment_record: z.nullable(z.lazy(() => exports.CompleteTradeResponsePriceAdjustmentRecord$inboundSchema)).optional(),
    route: z.string().optional(),
    special_instructions: z.array(z.string()).optional(),
    symbol_description: z.string().optional(),
    when_issued: z.boolean().optional(),
    yield_records: z.array(yieldrecord_js_1.YieldRecord$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "additional_instructions": "additionalInstructions",
        "alternate_order_id": "alternateOrderId",
        "booking_api_trade_allocation_id": "bookingApiTradeAllocationId",
        "booking_api_trade_id": "bookingApiTradeId",
        "broker_capacity": "brokerCapacity",
        "client_memos": "clientMemos",
        "client_order_id": "clientOrderId",
        "execution_id": "executionId",
        "execution_only": "executionOnly",
        "external_id": "externalId",
        "fund_confirmation_number": "fundConfirmationNumber",
        "gateway_client_order_id": "gatewayClientOrderId",
        "internal_error": "internalError",
        "is_writeoff": "isWriteoff",
        "order_id": "orderId",
        "prevailing_market_price": "prevailingMarketPrice",
        "price_adjustment_record": "priceAdjustmentRecord",
        "special_instructions": "specialInstructions",
        "symbol_description": "symbolDescription",
        "when_issued": "whenIssued",
        "yield_records": "yieldRecords",
    });
});
/** @internal */
exports.CompleteTradeResponseTrade$outboundSchema = z.object({
    additionalInstructions: z.array(z.string()).optional(),
    alternateOrderId: z.string().optional(),
    bookingApiTradeAllocationId: z.string().optional(),
    bookingApiTradeId: z.string().optional(),
    broker: z.string().optional(),
    brokerCapacity: exports.CompleteTradeResponseBrokerCapacity$outboundSchema.optional(),
    clientMemos: z.array(z.string()).optional(),
    clientOrderId: z.string().optional(),
    exchange: z.string().optional(),
    executionId: z.string().optional(),
    executionOnly: z.boolean().optional(),
    externalId: z.string().optional(),
    fundConfirmationNumber: z.string().optional(),
    gatewayClientOrderId: z.string().optional(),
    internalError: z.boolean().optional(),
    isWriteoff: z.boolean().optional(),
    lots: z.array(lot_js_1.Lot$outboundSchema).optional(),
    orderId: z.string().optional(),
    prevailingMarketPrice: z.nullable(z.lazy(() => exports.CompleteTradeResponsePrevailingMarketPrice$outboundSchema)).optional(),
    priceAdjustmentRecord: z.nullable(z.lazy(() => exports.CompleteTradeResponsePriceAdjustmentRecord$outboundSchema)).optional(),
    route: z.string().optional(),
    specialInstructions: z.array(z.string()).optional(),
    symbolDescription: z.string().optional(),
    whenIssued: z.boolean().optional(),
    yieldRecords: z.array(yieldrecord_js_1.YieldRecord$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        additionalInstructions: "additional_instructions",
        alternateOrderId: "alternate_order_id",
        bookingApiTradeAllocationId: "booking_api_trade_allocation_id",
        bookingApiTradeId: "booking_api_trade_id",
        brokerCapacity: "broker_capacity",
        clientMemos: "client_memos",
        clientOrderId: "client_order_id",
        executionId: "execution_id",
        executionOnly: "execution_only",
        externalId: "external_id",
        fundConfirmationNumber: "fund_confirmation_number",
        gatewayClientOrderId: "gateway_client_order_id",
        internalError: "internal_error",
        isWriteoff: "is_writeoff",
        orderId: "order_id",
        prevailingMarketPrice: "prevailing_market_price",
        priceAdjustmentRecord: "price_adjustment_record",
        specialInstructions: "special_instructions",
        symbolDescription: "symbol_description",
        whenIssued: "when_issued",
        yieldRecords: "yield_records",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CompleteTradeResponseTrade$;
(function (CompleteTradeResponseTrade$) {
    /** @deprecated use `CompleteTradeResponseTrade$inboundSchema` instead. */
    CompleteTradeResponseTrade$.inboundSchema = exports.CompleteTradeResponseTrade$inboundSchema;
    /** @deprecated use `CompleteTradeResponseTrade$outboundSchema` instead. */
    CompleteTradeResponseTrade$.outboundSchema = exports.CompleteTradeResponseTrade$outboundSchema;
})(CompleteTradeResponseTrade$ || (exports.CompleteTradeResponseTrade$ = CompleteTradeResponseTrade$ = {}));
/** @internal */
exports.CompleteTradeResponse$inboundSchema = z.object({
    trade: z.nullable(z.lazy(() => exports.CompleteTradeResponseTrade$inboundSchema))
        .optional(),
});
/** @internal */
exports.CompleteTradeResponse$outboundSchema = z.object({
    trade: z.nullable(z.lazy(() => exports.CompleteTradeResponseTrade$outboundSchema))
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CompleteTradeResponse$;
(function (CompleteTradeResponse$) {
    /** @deprecated use `CompleteTradeResponse$inboundSchema` instead. */
    CompleteTradeResponse$.inboundSchema = exports.CompleteTradeResponse$inboundSchema;
    /** @deprecated use `CompleteTradeResponse$outboundSchema` instead. */
    CompleteTradeResponse$.outboundSchema = exports.CompleteTradeResponse$outboundSchema;
})(CompleteTradeResponse$ || (exports.CompleteTradeResponse$ = CompleteTradeResponse$ = {}));
