// @apexfintechsolutions/ascend-sdk/models/components/contributionconstraints.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContributionConstraints$ = exports.ContributionConstraints$outboundSchema = exports.ContributionConstraints$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const contributionconstraintscontributiontypeinfo_js_1 = require("./contributionconstraintscontributiontypeinfo.js");
/** @internal */
exports.ContributionConstraints$inboundSchema = z.object({
    only_allow_current_year: z.boolean().optional(),
    valid_types: z.array(contributionconstraintscontributiontypeinfo_js_1.ContributionConstraintsContributionTypeInfo$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "only_allow_current_year": "onlyAllowCurrentYear",
        "valid_types": "validTypes",
    });
});
/** @internal */
exports.ContributionConstraints$outboundSchema = z.object({
    onlyAllowCurrentYear: z.boolean().optional(),
    validTypes: z.array(contributionconstraintscontributiontypeinfo_js_1.ContributionConstraintsContributionTypeInfo$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        onlyAllowCurrentYear: "only_allow_current_year",
        validTypes: "valid_types",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ContributionConstraints$;
(function (ContributionConstraints$) {
    /** @deprecated use `ContributionConstraints$inboundSchema` instead. */
    ContributionConstraints$.inboundSchema = exports.ContributionConstraints$inboundSchema;
    /** @deprecated use `ContributionConstraints$outboundSchema` instead. */
    ContributionConstraints$.outboundSchema = exports.ContributionConstraints$outboundSchema;
})(ContributionConstraints$ || (exports.ContributionConstraints$ = ContributionConstraints$ = {}));
