// @apexfintechsolutions/ascend-sdk/models/components/corporationenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CorporationEnrollmentMetadataCreate$ = exports.CorporationEnrollmentMetadataCreate$outboundSchema = exports.CorporationEnrollmentMetadataCreate$inboundSchema = exports.FdicCashSweep$ = exports.FdicCashSweep$outboundSchema = exports.FdicCashSweep$inboundSchema = exports.DividendReinvestmentPlan$ = exports.DividendReinvestmentPlan$outboundSchema = exports.DividendReinvestmentPlan$inboundSchema = exports.FdicCashSweep = exports.DividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const eddaccountenrollmentmetadatacreate_js_1 = require("./eddaccountenrollmentmetadatacreate.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var DividendReinvestmentPlan;
(function (DividendReinvestmentPlan) {
    DividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    DividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    DividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(DividendReinvestmentPlan || (exports.DividendReinvestmentPlan = DividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var FdicCashSweep;
(function (FdicCashSweep) {
    FdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    FdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    FdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(FdicCashSweep || (exports.FdicCashSweep = FdicCashSweep = {}));
/** @internal */
exports.DividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(DividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.DividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(DividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DividendReinvestmentPlan$;
(function (DividendReinvestmentPlan$) {
    /** @deprecated use `DividendReinvestmentPlan$inboundSchema` instead. */
    DividendReinvestmentPlan$.inboundSchema = exports.DividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `DividendReinvestmentPlan$outboundSchema` instead. */
    DividendReinvestmentPlan$.outboundSchema = exports.DividendReinvestmentPlan$outboundSchema;
})(DividendReinvestmentPlan$ || (exports.DividendReinvestmentPlan$ = DividendReinvestmentPlan$ = {}));
/** @internal */
exports.FdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(FdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.FdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(FdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FdicCashSweep$;
(function (FdicCashSweep$) {
    /** @deprecated use `FdicCashSweep$inboundSchema` instead. */
    FdicCashSweep$.inboundSchema = exports.FdicCashSweep$inboundSchema;
    /** @deprecated use `FdicCashSweep$outboundSchema` instead. */
    FdicCashSweep$.outboundSchema = exports.FdicCashSweep$outboundSchema;
})(FdicCashSweep$ || (exports.FdicCashSweep$ = FdicCashSweep$ = {}));
/** @internal */
exports.CorporationEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.DividendReinvestmentPlan$inboundSchema.optional(),
    edd_account_enrollment_metadata: eddaccountenrollmentmetadatacreate_js_1.EddAccountEnrollmentMetadataCreate$inboundSchema.optional(),
    fdic_cash_sweep: exports.FdicCashSweep$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "edd_account_enrollment_metadata": "eddAccountEnrollmentMetadata",
        "fdic_cash_sweep": "fdicCashSweep",
    });
});
/** @internal */
exports.CorporationEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.DividendReinvestmentPlan$outboundSchema.optional(),
    eddAccountEnrollmentMetadata: eddaccountenrollmentmetadatacreate_js_1.EddAccountEnrollmentMetadataCreate$outboundSchema.optional(),
    fdicCashSweep: exports.FdicCashSweep$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        eddAccountEnrollmentMetadata: "edd_account_enrollment_metadata",
        fdicCashSweep: "fdic_cash_sweep",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CorporationEnrollmentMetadataCreate$;
(function (CorporationEnrollmentMetadataCreate$) {
    /** @deprecated use `CorporationEnrollmentMetadataCreate$inboundSchema` instead. */
    CorporationEnrollmentMetadataCreate$.inboundSchema = exports.CorporationEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `CorporationEnrollmentMetadataCreate$outboundSchema` instead. */
    CorporationEnrollmentMetadataCreate$.outboundSchema = exports.CorporationEnrollmentMetadataCreate$outboundSchema;
})(CorporationEnrollmentMetadataCreate$ || (exports.CorporationEnrollmentMetadataCreate$ = CorporationEnrollmentMetadataCreate$ = {}));
