// @apexfintechsolutions/ascend-sdk/models/components/createuploadlinkrequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUploadLinkRequestCreate$ = exports.CreateUploadLinkRequestCreate$outboundSchema = exports.CreateUploadLinkRequestCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const accountdocumentuploadrequestcreate_js_1 = require("./accountdocumentuploadrequestcreate.js");
const iddocumentuploadrequestcreate_js_1 = require("./iddocumentuploadrequestcreate.js");
/** @internal */
exports.CreateUploadLinkRequestCreate$inboundSchema = z.object({
    account_document_upload_request: accountdocumentuploadrequestcreate_js_1.AccountDocumentUploadRequestCreate$inboundSchema.optional(),
    client_batch_source_id: z.string(),
    id_document_upload_request: iddocumentuploadrequestcreate_js_1.IDDocumentUploadRequestCreate$inboundSchema
        .optional(),
    mime_type: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_document_upload_request": "accountDocumentUploadRequest",
        "client_batch_source_id": "clientBatchSourceId",
        "id_document_upload_request": "idDocumentUploadRequest",
        "mime_type": "mimeType",
    });
});
/** @internal */
exports.CreateUploadLinkRequestCreate$outboundSchema = z.object({
    accountDocumentUploadRequest: accountdocumentuploadrequestcreate_js_1.AccountDocumentUploadRequestCreate$outboundSchema.optional(),
    clientBatchSourceId: z.string(),
    idDocumentUploadRequest: iddocumentuploadrequestcreate_js_1.IDDocumentUploadRequestCreate$outboundSchema
        .optional(),
    mimeType: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountDocumentUploadRequest: "account_document_upload_request",
        clientBatchSourceId: "client_batch_source_id",
        idDocumentUploadRequest: "id_document_upload_request",
        mimeType: "mime_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateUploadLinkRequestCreate$;
(function (CreateUploadLinkRequestCreate$) {
    /** @deprecated use `CreateUploadLinkRequestCreate$inboundSchema` instead. */
    CreateUploadLinkRequestCreate$.inboundSchema = exports.CreateUploadLinkRequestCreate$inboundSchema;
    /** @deprecated use `CreateUploadLinkRequestCreate$outboundSchema` instead. */
    CreateUploadLinkRequestCreate$.outboundSchema = exports.CreateUploadLinkRequestCreate$outboundSchema;
})(CreateUploadLinkRequestCreate$ || (exports.CreateUploadLinkRequestCreate$ = CreateUploadLinkRequestCreate$ = {}));
