// @apexfintechsolutions/ascend-sdk/models/components/customerreferralsourcecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerReferralSourceCreate$ = exports.CustomerReferralSourceCreate$outboundSchema = exports.CustomerReferralSourceCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.CustomerReferralSourceCreate$inboundSchema = z.object({
    name: z.string(),
    relationship_to_applicant: z.string(),
    relationship_years_with_applicant: z.number().int(),
    relationship_years_with_broker: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "relationship_to_applicant": "relationshipToApplicant",
        "relationship_years_with_applicant": "relationshipYearsWithApplicant",
        "relationship_years_with_broker": "relationshipYearsWithBroker",
    });
});
/** @internal */
exports.CustomerReferralSourceCreate$outboundSchema = z.object({
    name: z.string(),
    relationshipToApplicant: z.string(),
    relationshipYearsWithApplicant: z.number().int(),
    relationshipYearsWithBroker: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        relationshipToApplicant: "relationship_to_applicant",
        relationshipYearsWithApplicant: "relationship_years_with_applicant",
        relationshipYearsWithBroker: "relationship_years_with_broker",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerReferralSourceCreate$;
(function (CustomerReferralSourceCreate$) {
    /** @deprecated use `CustomerReferralSourceCreate$inboundSchema` instead. */
    CustomerReferralSourceCreate$.inboundSchema = exports.CustomerReferralSourceCreate$inboundSchema;
    /** @deprecated use `CustomerReferralSourceCreate$outboundSchema` instead. */
    CustomerReferralSourceCreate$.outboundSchema = exports.CustomerReferralSourceCreate$outboundSchema;
})(CustomerReferralSourceCreate$ || (exports.CustomerReferralSourceCreate$ = CustomerReferralSourceCreate$ = {}));
