// @apexfintechsolutions/ascend-sdk/models/components/customerprofileupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerProfileUpdate$ = exports.CustomerProfileUpdate$outboundSchema = exports.CustomerProfileUpdate$inboundSchema = exports.CustomerProfileUpdateTotalNetWorthRangeUsd$ = exports.CustomerProfileUpdateTotalNetWorthRangeUsd$outboundSchema = exports.CustomerProfileUpdateTotalNetWorthRangeUsd$inboundSchema = exports.CustomerProfileUpdateLiquidNetWorthRangeUsd$ = exports.CustomerProfileUpdateLiquidNetWorthRangeUsd$outboundSchema = exports.CustomerProfileUpdateLiquidNetWorthRangeUsd$inboundSchema = exports.CustomerProfileUpdateInvestmentExperience$ = exports.CustomerProfileUpdateInvestmentExperience$outboundSchema = exports.CustomerProfileUpdateInvestmentExperience$inboundSchema = exports.CustomerProfileUpdateAnnualIncomeRangeUsd$ = exports.CustomerProfileUpdateAnnualIncomeRangeUsd$outboundSchema = exports.CustomerProfileUpdateAnnualIncomeRangeUsd$inboundSchema = exports.CustomerProfileUpdateTotalNetWorthRangeUsd = exports.CustomerProfileUpdateLiquidNetWorthRangeUsd = exports.CustomerProfileUpdateInvestmentExperience = exports.CustomerProfileUpdateAnnualIncomeRangeUsd = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Annual income range; the low number is exclusive, the high number is inclusive
 */
var CustomerProfileUpdateAnnualIncomeRangeUsd;
(function (CustomerProfileUpdateAnnualIncomeRangeUsd) {
    CustomerProfileUpdateAnnualIncomeRangeUsd["UsdRangeUnspecified"] = "USD_RANGE_UNSPECIFIED";
    CustomerProfileUpdateAnnualIncomeRangeUsd["Under25K"] = "UNDER_25K";
    CustomerProfileUpdateAnnualIncomeRangeUsd["From25KTo50K"] = "FROM_25K_TO_50K";
    CustomerProfileUpdateAnnualIncomeRangeUsd["From50KTo100K"] = "FROM_50K_TO_100K";
    CustomerProfileUpdateAnnualIncomeRangeUsd["From100KTo200K"] = "FROM_100K_TO_200K";
    CustomerProfileUpdateAnnualIncomeRangeUsd["From200KTo300K"] = "FROM_200K_TO_300K";
    CustomerProfileUpdateAnnualIncomeRangeUsd["From300KTo500K"] = "FROM_300K_TO_500K";
    CustomerProfileUpdateAnnualIncomeRangeUsd["From500KTo1M"] = "FROM_500K_TO_1M";
    CustomerProfileUpdateAnnualIncomeRangeUsd["From1MTo5M"] = "FROM_1M_TO_5M";
    CustomerProfileUpdateAnnualIncomeRangeUsd["Over5M"] = "OVER_5M";
})(CustomerProfileUpdateAnnualIncomeRangeUsd || (exports.CustomerProfileUpdateAnnualIncomeRangeUsd = CustomerProfileUpdateAnnualIncomeRangeUsd = {}));
/**
 * Investment experience.
 */
var CustomerProfileUpdateInvestmentExperience;
(function (CustomerProfileUpdateInvestmentExperience) {
    CustomerProfileUpdateInvestmentExperience["InvestmentExperienceUnspecified"] = "INVESTMENT_EXPERIENCE_UNSPECIFIED";
    CustomerProfileUpdateInvestmentExperience["None"] = "NONE";
    CustomerProfileUpdateInvestmentExperience["Limited"] = "LIMITED";
    CustomerProfileUpdateInvestmentExperience["Good"] = "GOOD";
    CustomerProfileUpdateInvestmentExperience["Extensive"] = "EXTENSIVE";
})(CustomerProfileUpdateInvestmentExperience || (exports.CustomerProfileUpdateInvestmentExperience = CustomerProfileUpdateInvestmentExperience = {}));
/**
 * Liquid net worth range; the low number is exclusive, the high number is inclusive
 */
var CustomerProfileUpdateLiquidNetWorthRangeUsd;
(function (CustomerProfileUpdateLiquidNetWorthRangeUsd) {
    CustomerProfileUpdateLiquidNetWorthRangeUsd["UsdRangeUnspecified"] = "USD_RANGE_UNSPECIFIED";
    CustomerProfileUpdateLiquidNetWorthRangeUsd["Under25K"] = "UNDER_25K";
    CustomerProfileUpdateLiquidNetWorthRangeUsd["From25KTo50K"] = "FROM_25K_TO_50K";
    CustomerProfileUpdateLiquidNetWorthRangeUsd["From50KTo100K"] = "FROM_50K_TO_100K";
    CustomerProfileUpdateLiquidNetWorthRangeUsd["From100KTo200K"] = "FROM_100K_TO_200K";
    CustomerProfileUpdateLiquidNetWorthRangeUsd["From200KTo300K"] = "FROM_200K_TO_300K";
    CustomerProfileUpdateLiquidNetWorthRangeUsd["From300KTo500K"] = "FROM_300K_TO_500K";
    CustomerProfileUpdateLiquidNetWorthRangeUsd["From500KTo1M"] = "FROM_500K_TO_1M";
    CustomerProfileUpdateLiquidNetWorthRangeUsd["From1MTo5M"] = "FROM_1M_TO_5M";
    CustomerProfileUpdateLiquidNetWorthRangeUsd["Over5M"] = "OVER_5M";
})(CustomerProfileUpdateLiquidNetWorthRangeUsd || (exports.CustomerProfileUpdateLiquidNetWorthRangeUsd = CustomerProfileUpdateLiquidNetWorthRangeUsd = {}));
/**
 * Total net worth range; the low number is exclusive, the high number is inclusive
 */
var CustomerProfileUpdateTotalNetWorthRangeUsd;
(function (CustomerProfileUpdateTotalNetWorthRangeUsd) {
    CustomerProfileUpdateTotalNetWorthRangeUsd["UsdRangeUnspecified"] = "USD_RANGE_UNSPECIFIED";
    CustomerProfileUpdateTotalNetWorthRangeUsd["Under25K"] = "UNDER_25K";
    CustomerProfileUpdateTotalNetWorthRangeUsd["From25KTo50K"] = "FROM_25K_TO_50K";
    CustomerProfileUpdateTotalNetWorthRangeUsd["From50KTo100K"] = "FROM_50K_TO_100K";
    CustomerProfileUpdateTotalNetWorthRangeUsd["From100KTo200K"] = "FROM_100K_TO_200K";
    CustomerProfileUpdateTotalNetWorthRangeUsd["From200KTo300K"] = "FROM_200K_TO_300K";
    CustomerProfileUpdateTotalNetWorthRangeUsd["From300KTo500K"] = "FROM_300K_TO_500K";
    CustomerProfileUpdateTotalNetWorthRangeUsd["From500KTo1M"] = "FROM_500K_TO_1M";
    CustomerProfileUpdateTotalNetWorthRangeUsd["From1MTo5M"] = "FROM_1M_TO_5M";
    CustomerProfileUpdateTotalNetWorthRangeUsd["Over5M"] = "OVER_5M";
})(CustomerProfileUpdateTotalNetWorthRangeUsd || (exports.CustomerProfileUpdateTotalNetWorthRangeUsd = CustomerProfileUpdateTotalNetWorthRangeUsd = {}));
/** @internal */
exports.CustomerProfileUpdateAnnualIncomeRangeUsd$inboundSchema = z
    .union([
    z.nativeEnum(CustomerProfileUpdateAnnualIncomeRangeUsd),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CustomerProfileUpdateAnnualIncomeRangeUsd$outboundSchema = z.union([
    z.nativeEnum(CustomerProfileUpdateAnnualIncomeRangeUsd),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerProfileUpdateAnnualIncomeRangeUsd$;
(function (CustomerProfileUpdateAnnualIncomeRangeUsd$) {
    /** @deprecated use `CustomerProfileUpdateAnnualIncomeRangeUsd$inboundSchema` instead. */
    CustomerProfileUpdateAnnualIncomeRangeUsd$.inboundSchema = exports.CustomerProfileUpdateAnnualIncomeRangeUsd$inboundSchema;
    /** @deprecated use `CustomerProfileUpdateAnnualIncomeRangeUsd$outboundSchema` instead. */
    CustomerProfileUpdateAnnualIncomeRangeUsd$.outboundSchema = exports.CustomerProfileUpdateAnnualIncomeRangeUsd$outboundSchema;
})(CustomerProfileUpdateAnnualIncomeRangeUsd$ || (exports.CustomerProfileUpdateAnnualIncomeRangeUsd$ = CustomerProfileUpdateAnnualIncomeRangeUsd$ = {}));
/** @internal */
exports.CustomerProfileUpdateInvestmentExperience$inboundSchema = z
    .union([
    z.nativeEnum(CustomerProfileUpdateInvestmentExperience),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CustomerProfileUpdateInvestmentExperience$outboundSchema = z.union([
    z.nativeEnum(CustomerProfileUpdateInvestmentExperience),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerProfileUpdateInvestmentExperience$;
(function (CustomerProfileUpdateInvestmentExperience$) {
    /** @deprecated use `CustomerProfileUpdateInvestmentExperience$inboundSchema` instead. */
    CustomerProfileUpdateInvestmentExperience$.inboundSchema = exports.CustomerProfileUpdateInvestmentExperience$inboundSchema;
    /** @deprecated use `CustomerProfileUpdateInvestmentExperience$outboundSchema` instead. */
    CustomerProfileUpdateInvestmentExperience$.outboundSchema = exports.CustomerProfileUpdateInvestmentExperience$outboundSchema;
})(CustomerProfileUpdateInvestmentExperience$ || (exports.CustomerProfileUpdateInvestmentExperience$ = CustomerProfileUpdateInvestmentExperience$ = {}));
/** @internal */
exports.CustomerProfileUpdateLiquidNetWorthRangeUsd$inboundSchema = z
    .union([
    z.nativeEnum(CustomerProfileUpdateLiquidNetWorthRangeUsd),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CustomerProfileUpdateLiquidNetWorthRangeUsd$outboundSchema = z.union([
    z.nativeEnum(CustomerProfileUpdateLiquidNetWorthRangeUsd),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerProfileUpdateLiquidNetWorthRangeUsd$;
(function (CustomerProfileUpdateLiquidNetWorthRangeUsd$) {
    /** @deprecated use `CustomerProfileUpdateLiquidNetWorthRangeUsd$inboundSchema` instead. */
    CustomerProfileUpdateLiquidNetWorthRangeUsd$.inboundSchema = exports.CustomerProfileUpdateLiquidNetWorthRangeUsd$inboundSchema;
    /** @deprecated use `CustomerProfileUpdateLiquidNetWorthRangeUsd$outboundSchema` instead. */
    CustomerProfileUpdateLiquidNetWorthRangeUsd$.outboundSchema = exports.CustomerProfileUpdateLiquidNetWorthRangeUsd$outboundSchema;
})(CustomerProfileUpdateLiquidNetWorthRangeUsd$ || (exports.CustomerProfileUpdateLiquidNetWorthRangeUsd$ = CustomerProfileUpdateLiquidNetWorthRangeUsd$ = {}));
/** @internal */
exports.CustomerProfileUpdateTotalNetWorthRangeUsd$inboundSchema = z
    .union([
    z.nativeEnum(CustomerProfileUpdateTotalNetWorthRangeUsd),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CustomerProfileUpdateTotalNetWorthRangeUsd$outboundSchema = z.union([
    z.nativeEnum(CustomerProfileUpdateTotalNetWorthRangeUsd),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerProfileUpdateTotalNetWorthRangeUsd$;
(function (CustomerProfileUpdateTotalNetWorthRangeUsd$) {
    /** @deprecated use `CustomerProfileUpdateTotalNetWorthRangeUsd$inboundSchema` instead. */
    CustomerProfileUpdateTotalNetWorthRangeUsd$.inboundSchema = exports.CustomerProfileUpdateTotalNetWorthRangeUsd$inboundSchema;
    /** @deprecated use `CustomerProfileUpdateTotalNetWorthRangeUsd$outboundSchema` instead. */
    CustomerProfileUpdateTotalNetWorthRangeUsd$.outboundSchema = exports.CustomerProfileUpdateTotalNetWorthRangeUsd$outboundSchema;
})(CustomerProfileUpdateTotalNetWorthRangeUsd$ || (exports.CustomerProfileUpdateTotalNetWorthRangeUsd$ = CustomerProfileUpdateTotalNetWorthRangeUsd$ = {}));
/** @internal */
exports.CustomerProfileUpdate$inboundSchema = z.object({
    annual_income_range_usd: exports.CustomerProfileUpdateAnnualIncomeRangeUsd$inboundSchema.optional(),
    federal_tax_bracket: z.number().optional(),
    investment_experience: exports.CustomerProfileUpdateInvestmentExperience$inboundSchema
        .optional(),
    liquid_net_worth_range_usd: exports.CustomerProfileUpdateLiquidNetWorthRangeUsd$inboundSchema.optional(),
    total_net_worth_range_usd: exports.CustomerProfileUpdateTotalNetWorthRangeUsd$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "annual_income_range_usd": "annualIncomeRangeUsd",
        "federal_tax_bracket": "federalTaxBracket",
        "investment_experience": "investmentExperience",
        "liquid_net_worth_range_usd": "liquidNetWorthRangeUsd",
        "total_net_worth_range_usd": "totalNetWorthRangeUsd",
    });
});
/** @internal */
exports.CustomerProfileUpdate$outboundSchema = z.object({
    annualIncomeRangeUsd: exports.CustomerProfileUpdateAnnualIncomeRangeUsd$outboundSchema
        .optional(),
    federalTaxBracket: z.number().optional(),
    investmentExperience: exports.CustomerProfileUpdateInvestmentExperience$outboundSchema
        .optional(),
    liquidNetWorthRangeUsd: exports.CustomerProfileUpdateLiquidNetWorthRangeUsd$outboundSchema.optional(),
    totalNetWorthRangeUsd: exports.CustomerProfileUpdateTotalNetWorthRangeUsd$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        annualIncomeRangeUsd: "annual_income_range_usd",
        federalTaxBracket: "federal_tax_bracket",
        investmentExperience: "investment_experience",
        liquidNetWorthRangeUsd: "liquid_net_worth_range_usd",
        totalNetWorthRangeUsd: "total_net_worth_range_usd",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerProfileUpdate$;
(function (CustomerProfileUpdate$) {
    /** @deprecated use `CustomerProfileUpdate$inboundSchema` instead. */
    CustomerProfileUpdate$.inboundSchema = exports.CustomerProfileUpdate$inboundSchema;
    /** @deprecated use `CustomerProfileUpdate$outboundSchema` instead. */
    CustomerProfileUpdate$.outboundSchema = exports.CustomerProfileUpdate$outboundSchema;
})(CustomerProfileUpdate$ || (exports.CustomerProfileUpdate$ = CustomerProfileUpdate$ = {}));
