// @apexfintechsolutions/ascend-sdk/models/components/custodialenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustodialEnrollmentMetadataCreate$ = exports.CustodialEnrollmentMetadataCreate$outboundSchema = exports.CustodialEnrollmentMetadataCreate$inboundSchema = exports.CustodialEnrollmentMetadataCreateFdicCashSweep$ = exports.CustodialEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.CustodialEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.CustodialEnrollmentMetadataCreateFdicCashSweep = exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var CustodialEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (CustodialEnrollmentMetadataCreateDividendReinvestmentPlan) {
    CustodialEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    CustodialEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    CustodialEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(CustodialEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan = CustodialEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var CustodialEnrollmentMetadataCreateFdicCashSweep;
(function (CustodialEnrollmentMetadataCreateFdicCashSweep) {
    CustodialEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    CustodialEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    CustodialEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(CustodialEnrollmentMetadataCreateFdicCashSweep || (exports.CustodialEnrollmentMetadataCreateFdicCashSweep = CustodialEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(CustodialEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(CustodialEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$ = CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.CustodialEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(CustodialEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CustodialEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(CustodialEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustodialEnrollmentMetadataCreateFdicCashSweep$;
(function (CustodialEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `CustodialEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    CustodialEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.CustodialEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `CustodialEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    CustodialEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.CustodialEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(CustodialEnrollmentMetadataCreateFdicCashSweep$ || (exports.CustodialEnrollmentMetadataCreateFdicCashSweep$ = CustodialEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.CustodialEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.CustodialEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
    });
});
/** @internal */
exports.CustodialEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.CustodialEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.CustodialEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustodialEnrollmentMetadataCreate$;
(function (CustodialEnrollmentMetadataCreate$) {
    /** @deprecated use `CustodialEnrollmentMetadataCreate$inboundSchema` instead. */
    CustodialEnrollmentMetadataCreate$.inboundSchema = exports.CustodialEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `CustodialEnrollmentMetadataCreate$outboundSchema` instead. */
    CustodialEnrollmentMetadataCreate$.outboundSchema = exports.CustodialEnrollmentMetadataCreate$outboundSchema;
})(CustodialEnrollmentMetadataCreate$ || (exports.CustodialEnrollmentMetadataCreate$ = CustodialEnrollmentMetadataCreate$ = {}));
