// @apexfintechsolutions/ascend-sdk/models/components/deactivateenrollmentrequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeactivateEnrollmentRequestCreate$ = exports.DeactivateEnrollmentRequestCreate$outboundSchema = exports.DeactivateEnrollmentRequestCreate$inboundSchema = exports.DeactivateEnrollmentRequestCreateEnrollmentType$ = exports.DeactivateEnrollmentRequestCreateEnrollmentType$outboundSchema = exports.DeactivateEnrollmentRequestCreateEnrollmentType$inboundSchema = exports.DeactivateEnrollmentRequestCreateEnrollmentType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Describes the name of the enrollment; Expressed as an enum
 */
var DeactivateEnrollmentRequestCreateEnrollmentType;
(function (DeactivateEnrollmentRequestCreateEnrollmentType) {
    DeactivateEnrollmentRequestCreateEnrollmentType["EnrollmentTypeUnspecified"] = "ENROLLMENT_TYPE_UNSPECIFIED";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationIndividual"] = "REGISTRATION_INDIVIDUAL";
    DeactivateEnrollmentRequestCreateEnrollmentType["LendingFullyPaidStockLoan"] = "LENDING_FULLY_PAID_STOCK_LOAN";
    DeactivateEnrollmentRequestCreateEnrollmentType["BeneficiaryDesignation"] = "BENEFICIARY_DESIGNATION";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationJointWros"] = "REGISTRATION_JOINT_WROS";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationJointTic"] = "REGISTRATION_JOINT_TIC";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationJointTbe"] = "REGISTRATION_JOINT_TBE";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationJointCp"] = "REGISTRATION_JOINT_CP";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationEstate"] = "REGISTRATION_ESTATE";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationIraTraditional"] = "REGISTRATION_IRA_TRADITIONAL";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationIraSimple"] = "REGISTRATION_IRA_SIMPLE";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationIraSep"] = "REGISTRATION_IRA_SEP";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationIraRoth"] = "REGISTRATION_IRA_ROTH";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationIraRollover"] = "REGISTRATION_IRA_ROLLOVER";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationTrust"] = "REGISTRATION_TRUST";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationCorporation"] = "REGISTRATION_CORPORATION";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationLlc"] = "REGISTRATION_LLC";
    DeactivateEnrollmentRequestCreateEnrollmentType["CashFdicCashSweep"] = "CASH_FDIC_CASH_SWEEP";
    DeactivateEnrollmentRequestCreateEnrollmentType["RetirementBeneficiaryDesignation"] = "RETIREMENT_BENEFICIARY_DESIGNATION";
    DeactivateEnrollmentRequestCreateEnrollmentType["DividendReinvestmentPlan"] = "DIVIDEND_REINVESTMENT_PLAN";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationIraBeneficiaryTraditional"] = "REGISTRATION_IRA_BENEFICIARY_TRADITIONAL";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationIraBeneficiaryRoth"] = "REGISTRATION_IRA_BENEFICIARY_ROTH";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationIndividualForeign"] = "REGISTRATION_INDIVIDUAL_FOREIGN";
    DeactivateEnrollmentRequestCreateEnrollmentType["RegistrationCustodial"] = "REGISTRATION_CUSTODIAL";
    DeactivateEnrollmentRequestCreateEnrollmentType["VirtualAccountNumber"] = "VIRTUAL_ACCOUNT_NUMBER";
})(DeactivateEnrollmentRequestCreateEnrollmentType || (exports.DeactivateEnrollmentRequestCreateEnrollmentType = DeactivateEnrollmentRequestCreateEnrollmentType = {}));
/** @internal */
exports.DeactivateEnrollmentRequestCreateEnrollmentType$inboundSchema = z
    .union([
    z.nativeEnum(DeactivateEnrollmentRequestCreateEnrollmentType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.DeactivateEnrollmentRequestCreateEnrollmentType$outboundSchema = z.union([
    z.nativeEnum(DeactivateEnrollmentRequestCreateEnrollmentType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeactivateEnrollmentRequestCreateEnrollmentType$;
(function (DeactivateEnrollmentRequestCreateEnrollmentType$) {
    /** @deprecated use `DeactivateEnrollmentRequestCreateEnrollmentType$inboundSchema` instead. */
    DeactivateEnrollmentRequestCreateEnrollmentType$.inboundSchema = exports.DeactivateEnrollmentRequestCreateEnrollmentType$inboundSchema;
    /** @deprecated use `DeactivateEnrollmentRequestCreateEnrollmentType$outboundSchema` instead. */
    DeactivateEnrollmentRequestCreateEnrollmentType$.outboundSchema = exports.DeactivateEnrollmentRequestCreateEnrollmentType$outboundSchema;
})(DeactivateEnrollmentRequestCreateEnrollmentType$ || (exports.DeactivateEnrollmentRequestCreateEnrollmentType$ = DeactivateEnrollmentRequestCreateEnrollmentType$ = {}));
/** @internal */
exports.DeactivateEnrollmentRequestCreate$inboundSchema = z.object({
    enrollment_id: z.string().optional(),
    enrollment_type: exports.DeactivateEnrollmentRequestCreateEnrollmentType$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "enrollment_id": "enrollmentId",
        "enrollment_type": "enrollmentType",
    });
});
/** @internal */
exports.DeactivateEnrollmentRequestCreate$outboundSchema = z.object({
    enrollmentId: z.string().optional(),
    enrollmentType: exports.DeactivateEnrollmentRequestCreateEnrollmentType$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        enrollmentId: "enrollment_id",
        enrollmentType: "enrollment_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeactivateEnrollmentRequestCreate$;
(function (DeactivateEnrollmentRequestCreate$) {
    /** @deprecated use `DeactivateEnrollmentRequestCreate$inboundSchema` instead. */
    DeactivateEnrollmentRequestCreate$.inboundSchema = exports.DeactivateEnrollmentRequestCreate$inboundSchema;
    /** @deprecated use `DeactivateEnrollmentRequestCreate$outboundSchema` instead. */
    DeactivateEnrollmentRequestCreate$.outboundSchema = exports.DeactivateEnrollmentRequestCreate$outboundSchema;
})(DeactivateEnrollmentRequestCreate$ || (exports.DeactivateEnrollmentRequestCreate$ = DeactivateEnrollmentRequestCreate$ = {}));
