// @apexfintechsolutions/ascend-sdk/models/components/depositscheduledetailscreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DepositScheduleDetailsCreate$ = exports.DepositScheduleDetailsCreate$outboundSchema = exports.DepositScheduleDetailsCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
const schedulepropertiescreate_js_1 = require("./schedulepropertiescreate.js");
/** @internal */
exports.DepositScheduleDetailsCreate$inboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$inboundSchema,
    client_schedule_id: z.string(),
    schedule_properties: schedulepropertiescreate_js_1.SchedulePropertiesCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_schedule_id": "clientScheduleId",
        "schedule_properties": "scheduleProperties",
    });
});
/** @internal */
exports.DepositScheduleDetailsCreate$outboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$outboundSchema,
    clientScheduleId: z.string(),
    scheduleProperties: schedulepropertiescreate_js_1.SchedulePropertiesCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientScheduleId: "client_schedule_id",
        scheduleProperties: "schedule_properties",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DepositScheduleDetailsCreate$;
(function (DepositScheduleDetailsCreate$) {
    /** @deprecated use `DepositScheduleDetailsCreate$inboundSchema` instead. */
    DepositScheduleDetailsCreate$.inboundSchema = exports.DepositScheduleDetailsCreate$inboundSchema;
    /** @deprecated use `DepositScheduleDetailsCreate$outboundSchema` instead. */
    DepositScheduleDetailsCreate$.outboundSchema = exports.DepositScheduleDetailsCreate$outboundSchema;
})(DepositScheduleDetailsCreate$ || (exports.DepositScheduleDetailsCreate$ = DepositScheduleDetailsCreate$ = {}));
