// @apexfintechsolutions/ascend-sdk/models/components/descriptiondetail.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescriptionDetail$ = exports.DescriptionDetail$outboundSchema = exports.DescriptionDetail$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.DescriptionDetail$inboundSchema = z.object({
    description_one: z.string().optional(),
    description_three: z.string().optional(),
    description_two: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "description_one": "descriptionOne",
        "description_three": "descriptionThree",
        "description_two": "descriptionTwo",
    });
});
/** @internal */
exports.DescriptionDetail$outboundSchema = z.object({
    descriptionOne: z.string().optional(),
    descriptionThree: z.string().optional(),
    descriptionTwo: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        descriptionOne: "description_one",
        descriptionThree: "description_three",
        descriptionTwo: "description_two",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DescriptionDetail$;
(function (DescriptionDetail$) {
    /** @deprecated use `DescriptionDetail$inboundSchema` instead. */
    DescriptionDetail$.inboundSchema = exports.DescriptionDetail$inboundSchema;
    /** @deprecated use `DescriptionDetail$outboundSchema` instead. */
    DescriptionDetail$.outboundSchema = exports.DescriptionDetail$outboundSchema;
})(DescriptionDetail$ || (exports.DescriptionDetail$ = DescriptionDetail$ = {}));
