// @apexfintechsolutions/ascend-sdk/models/components/eddaccountenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EddAccountEnrollmentMetadataCreate$ = exports.EddAccountEnrollmentMetadataCreate$outboundSchema = exports.EddAccountEnrollmentMetadataCreate$inboundSchema = exports.DeterminedAccountRiskRating$ = exports.DeterminedAccountRiskRating$outboundSchema = exports.DeterminedAccountRiskRating$inboundSchema = exports.DeterminedAccountRiskRating = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const depositedfundscreate_js_1 = require("./depositedfundscreate.js");
const financialprofilecreate_js_1 = require("./financialprofilecreate.js");
const plannedactivitycreate_js_1 = require("./plannedactivitycreate.js");
const relatedpepdetailscreate_js_1 = require("./relatedpepdetailscreate.js");
/**
 * The client determined account risk rating of the entity customer
 */
var DeterminedAccountRiskRating;
(function (DeterminedAccountRiskRating) {
    DeterminedAccountRiskRating["DeterminedAccountRiskRatingUnspecified"] = "DETERMINED_ACCOUNT_RISK_RATING_UNSPECIFIED";
    DeterminedAccountRiskRating["Low"] = "LOW";
    DeterminedAccountRiskRating["Medium"] = "MEDIUM";
    DeterminedAccountRiskRating["High"] = "HIGH";
})(DeterminedAccountRiskRating || (exports.DeterminedAccountRiskRating = DeterminedAccountRiskRating = {}));
/** @internal */
exports.DeterminedAccountRiskRating$inboundSchema = z
    .union([
    z.nativeEnum(DeterminedAccountRiskRating),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.DeterminedAccountRiskRating$outboundSchema = z.union([
    z.nativeEnum(DeterminedAccountRiskRating),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeterminedAccountRiskRating$;
(function (DeterminedAccountRiskRating$) {
    /** @deprecated use `DeterminedAccountRiskRating$inboundSchema` instead. */
    DeterminedAccountRiskRating$.inboundSchema = exports.DeterminedAccountRiskRating$inboundSchema;
    /** @deprecated use `DeterminedAccountRiskRating$outboundSchema` instead. */
    DeterminedAccountRiskRating$.outboundSchema = exports.DeterminedAccountRiskRating$outboundSchema;
})(DeterminedAccountRiskRating$ || (exports.DeterminedAccountRiskRating$ = DeterminedAccountRiskRating$ = {}));
/** @internal */
exports.EddAccountEnrollmentMetadataCreate$inboundSchema = z.object({
    deposited_funds: depositedfundscreate_js_1.DepositedFundsCreate$inboundSchema,
    determined_account_risk_rating: exports.DeterminedAccountRiskRating$inboundSchema,
    financial_profile: financialprofilecreate_js_1.FinancialProfileCreate$inboundSchema,
    planned_activity: plannedactivitycreate_js_1.PlannedActivityCreate$inboundSchema,
    related_pep_details: relatedpepdetailscreate_js_1.RelatedPepDetailsCreate$inboundSchema,
    scope_of_business: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "deposited_funds": "depositedFunds",
        "determined_account_risk_rating": "determinedAccountRiskRating",
        "financial_profile": "financialProfile",
        "planned_activity": "plannedActivity",
        "related_pep_details": "relatedPepDetails",
        "scope_of_business": "scopeOfBusiness",
    });
});
/** @internal */
exports.EddAccountEnrollmentMetadataCreate$outboundSchema = z.object({
    depositedFunds: depositedfundscreate_js_1.DepositedFundsCreate$outboundSchema,
    determinedAccountRiskRating: exports.DeterminedAccountRiskRating$outboundSchema,
    financialProfile: financialprofilecreate_js_1.FinancialProfileCreate$outboundSchema,
    plannedActivity: plannedactivitycreate_js_1.PlannedActivityCreate$outboundSchema,
    relatedPepDetails: relatedpepdetailscreate_js_1.RelatedPepDetailsCreate$outboundSchema,
    scopeOfBusiness: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        depositedFunds: "deposited_funds",
        determinedAccountRiskRating: "determined_account_risk_rating",
        financialProfile: "financial_profile",
        plannedActivity: "planned_activity",
        relatedPepDetails: "related_pep_details",
        scopeOfBusiness: "scope_of_business",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EddAccountEnrollmentMetadataCreate$;
(function (EddAccountEnrollmentMetadataCreate$) {
    /** @deprecated use `EddAccountEnrollmentMetadataCreate$inboundSchema` instead. */
    EddAccountEnrollmentMetadataCreate$.inboundSchema = exports.EddAccountEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `EddAccountEnrollmentMetadataCreate$outboundSchema` instead. */
    EddAccountEnrollmentMetadataCreate$.outboundSchema = exports.EddAccountEnrollmentMetadataCreate$outboundSchema;
})(EddAccountEnrollmentMetadataCreate$ || (exports.EddAccountEnrollmentMetadataCreate$ = EddAccountEnrollmentMetadataCreate$ = {}));
