// @apexfintechsolutions/ascend-sdk/models/components/employmentcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmploymentCreate$ = exports.EmploymentCreate$outboundSchema = exports.EmploymentCreate$inboundSchema = exports.EmploymentStatus$ = exports.EmploymentStatus$outboundSchema = exports.EmploymentStatus$inboundSchema = exports.EmploymentStatus = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const postaladdresscreate_js_1 = require("./postaladdresscreate.js");
/**
 * Classifies in what capacity (or if) the underlying natural person holds a job
 */
var EmploymentStatus;
(function (EmploymentStatus) {
    EmploymentStatus["StatusUnspecified"] = "STATUS_UNSPECIFIED";
    EmploymentStatus["Employed"] = "EMPLOYED";
    EmploymentStatus["SelfEmployed"] = "SELF_EMPLOYED";
    EmploymentStatus["Unemployed"] = "UNEMPLOYED";
    EmploymentStatus["Retired"] = "RETIRED";
    EmploymentStatus["Student"] = "STUDENT";
})(EmploymentStatus || (exports.EmploymentStatus = EmploymentStatus = {}));
/** @internal */
exports.EmploymentStatus$inboundSchema = z
    .union([
    z.nativeEnum(EmploymentStatus),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.EmploymentStatus$outboundSchema = z.union([
    z.nativeEnum(EmploymentStatus),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EmploymentStatus$;
(function (EmploymentStatus$) {
    /** @deprecated use `EmploymentStatus$inboundSchema` instead. */
    EmploymentStatus$.inboundSchema = exports.EmploymentStatus$inboundSchema;
    /** @deprecated use `EmploymentStatus$outboundSchema` instead. */
    EmploymentStatus$.outboundSchema = exports.EmploymentStatus$outboundSchema;
})(EmploymentStatus$ || (exports.EmploymentStatus$ = EmploymentStatus$ = {}));
/** @internal */
exports.EmploymentCreate$inboundSchema = z.object({
    employer: z.string().optional(),
    employer_address: postaladdresscreate_js_1.PostalAddressCreate$inboundSchema.optional(),
    employment_status: exports.EmploymentStatus$inboundSchema,
    occupation: z.string().optional(),
    start_year: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "employer_address": "employerAddress",
        "employment_status": "employmentStatus",
        "start_year": "startYear",
    });
});
/** @internal */
exports.EmploymentCreate$outboundSchema = z.object({
    employer: z.string().optional(),
    employerAddress: postaladdresscreate_js_1.PostalAddressCreate$outboundSchema.optional(),
    employmentStatus: exports.EmploymentStatus$outboundSchema,
    occupation: z.string().optional(),
    startYear: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        employerAddress: "employer_address",
        employmentStatus: "employment_status",
        startYear: "start_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EmploymentCreate$;
(function (EmploymentCreate$) {
    /** @deprecated use `EmploymentCreate$inboundSchema` instead. */
    EmploymentCreate$.inboundSchema = exports.EmploymentCreate$inboundSchema;
    /** @deprecated use `EmploymentCreate$outboundSchema` instead. */
    EmploymentCreate$.outboundSchema = exports.EmploymentCreate$outboundSchema;
})(EmploymentCreate$ || (exports.EmploymentCreate$ = EmploymentCreate$ = {}));
