// @apexfintechsolutions/ascend-sdk/models/components/employmentupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmploymentUpdate$ = exports.EmploymentUpdate$outboundSchema = exports.EmploymentUpdate$inboundSchema = exports.EmploymentUpdateEmploymentStatus$ = exports.EmploymentUpdateEmploymentStatus$outboundSchema = exports.EmploymentUpdateEmploymentStatus$inboundSchema = exports.EmploymentUpdateEmploymentStatus = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const postaladdressupdate_js_1 = require("./postaladdressupdate.js");
/**
 * Classifies in what capacity (or if) the underlying natural person holds a job
 */
var EmploymentUpdateEmploymentStatus;
(function (EmploymentUpdateEmploymentStatus) {
    EmploymentUpdateEmploymentStatus["StatusUnspecified"] = "STATUS_UNSPECIFIED";
    EmploymentUpdateEmploymentStatus["Employed"] = "EMPLOYED";
    EmploymentUpdateEmploymentStatus["SelfEmployed"] = "SELF_EMPLOYED";
    EmploymentUpdateEmploymentStatus["Unemployed"] = "UNEMPLOYED";
    EmploymentUpdateEmploymentStatus["Retired"] = "RETIRED";
    EmploymentUpdateEmploymentStatus["Student"] = "STUDENT";
})(EmploymentUpdateEmploymentStatus || (exports.EmploymentUpdateEmploymentStatus = EmploymentUpdateEmploymentStatus = {}));
/** @internal */
exports.EmploymentUpdateEmploymentStatus$inboundSchema = z
    .union([
    z.nativeEnum(EmploymentUpdateEmploymentStatus),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.EmploymentUpdateEmploymentStatus$outboundSchema = z.union([
    z.nativeEnum(EmploymentUpdateEmploymentStatus),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EmploymentUpdateEmploymentStatus$;
(function (EmploymentUpdateEmploymentStatus$) {
    /** @deprecated use `EmploymentUpdateEmploymentStatus$inboundSchema` instead. */
    EmploymentUpdateEmploymentStatus$.inboundSchema = exports.EmploymentUpdateEmploymentStatus$inboundSchema;
    /** @deprecated use `EmploymentUpdateEmploymentStatus$outboundSchema` instead. */
    EmploymentUpdateEmploymentStatus$.outboundSchema = exports.EmploymentUpdateEmploymentStatus$outboundSchema;
})(EmploymentUpdateEmploymentStatus$ || (exports.EmploymentUpdateEmploymentStatus$ = EmploymentUpdateEmploymentStatus$ = {}));
/** @internal */
exports.EmploymentUpdate$inboundSchema = z.object({
    employer: z.string().optional(),
    employer_address: postaladdressupdate_js_1.PostalAddressUpdate$inboundSchema.optional(),
    employment_status: exports.EmploymentUpdateEmploymentStatus$inboundSchema.optional(),
    occupation: z.string().optional(),
    start_year: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "employer_address": "employerAddress",
        "employment_status": "employmentStatus",
        "start_year": "startYear",
    });
});
/** @internal */
exports.EmploymentUpdate$outboundSchema = z.object({
    employer: z.string().optional(),
    employerAddress: postaladdressupdate_js_1.PostalAddressUpdate$outboundSchema.optional(),
    employmentStatus: exports.EmploymentUpdateEmploymentStatus$outboundSchema.optional(),
    occupation: z.string().optional(),
    startYear: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        employerAddress: "employer_address",
        employmentStatus: "employment_status",
        startYear: "start_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EmploymentUpdate$;
(function (EmploymentUpdate$) {
    /** @deprecated use `EmploymentUpdate$inboundSchema` instead. */
    EmploymentUpdate$.inboundSchema = exports.EmploymentUpdate$inboundSchema;
    /** @deprecated use `EmploymentUpdate$outboundSchema` instead. */
    EmploymentUpdate$.outboundSchema = exports.EmploymentUpdate$outboundSchema;
})(EmploymentUpdate$ || (exports.EmploymentUpdate$ = EmploymentUpdate$ = {}));
