// @apexfintechsolutions/ascend-sdk/models/components/endlargetraderrequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndLargeTraderRequestCreate$ = exports.EndLargeTraderRequestCreate$outboundSchema = exports.EndLargeTraderRequestCreate$inboundSchema = exports.EndReason$ = exports.EndReason$outboundSchema = exports.EndReason$inboundSchema = exports.EndReason = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * The end reason of the LTID.
 */
var EndReason;
(function (EndReason) {
    EndReason["ReportableAccountEventReasonUnspecified"] = "REPORTABLE_ACCOUNT_EVENT_REASON_UNSPECIFIED";
    EndReason["EventReasonCreated"] = "EVENT_REASON_CREATED";
    EndReason["EventReasonCorrection"] = "EVENT_REASON_CORRECTION";
    EndReason["EventReasonEnded"] = "EVENT_REASON_ENDED";
    EndReason["EventReasonReplaced"] = "EVENT_REASON_REPLACED";
    EndReason["EventReasonTransfer"] = "EVENT_REASON_TRANSFER";
    EndReason["EventReasonOther"] = "EVENT_REASON_OTHER";
})(EndReason || (exports.EndReason = EndReason = {}));
/** @internal */
exports.EndReason$inboundSchema = z
    .union([
    z.nativeEnum(EndReason),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.EndReason$outboundSchema = z.union([
    z.nativeEnum(EndReason),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EndReason$;
(function (EndReason$) {
    /** @deprecated use `EndReason$inboundSchema` instead. */
    EndReason$.inboundSchema = exports.EndReason$inboundSchema;
    /** @deprecated use `EndReason$outboundSchema` instead. */
    EndReason$.outboundSchema = exports.EndReason$outboundSchema;
})(EndReason$ || (exports.EndReason$ = EndReason$ = {}));
/** @internal */
exports.EndLargeTraderRequestCreate$inboundSchema = z.object({
    end_reason: exports.EndReason$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "end_reason": "endReason",
    });
});
/** @internal */
exports.EndLargeTraderRequestCreate$outboundSchema = z.object({
    endReason: exports.EndReason$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        endReason: "end_reason",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EndLargeTraderRequestCreate$;
(function (EndLargeTraderRequestCreate$) {
    /** @deprecated use `EndLargeTraderRequestCreate$inboundSchema` instead. */
    EndLargeTraderRequestCreate$.inboundSchema = exports.EndLargeTraderRequestCreate$inboundSchema;
    /** @deprecated use `EndLargeTraderRequestCreate$outboundSchema` instead. */
    EndLargeTraderRequestCreate$.outboundSchema = exports.EndLargeTraderRequestCreate$outboundSchema;
})(EndLargeTraderRequestCreate$ || (exports.EndLargeTraderRequestCreate$ = EndLargeTraderRequestCreate$ = {}));
