// @apexfintechsolutions/ascend-sdk/models/components/entityduediligencecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityDueDiligenceCreate$ = exports.EntityDueDiligenceCreate$outboundSchema = exports.EntityDueDiligenceCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const negativenewscreate_js_1 = require("./negativenewscreate.js");
/** @internal */
exports.EntityDueDiligenceCreate$inboundSchema = z.object({
    entity_issues_bearer_shares: z.boolean(),
    negative_news: negativenewscreate_js_1.NegativeNewsCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "entity_issues_bearer_shares": "entityIssuesBearerShares",
        "negative_news": "negativeNews",
    });
});
/** @internal */
exports.EntityDueDiligenceCreate$outboundSchema = z.object({
    entityIssuesBearerShares: z.boolean(),
    negativeNews: negativenewscreate_js_1.NegativeNewsCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        entityIssuesBearerShares: "entity_issues_bearer_shares",
        negativeNews: "negative_news",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EntityDueDiligenceCreate$;
(function (EntityDueDiligenceCreate$) {
    /** @deprecated use `EntityDueDiligenceCreate$inboundSchema` instead. */
    EntityDueDiligenceCreate$.inboundSchema = exports.EntityDueDiligenceCreate$inboundSchema;
    /** @deprecated use `EntityDueDiligenceCreate$outboundSchema` instead. */
    EntityDueDiligenceCreate$.outboundSchema = exports.EntityDueDiligenceCreate$outboundSchema;
})(EntityDueDiligenceCreate$ || (exports.EntityDueDiligenceCreate$ = EntityDueDiligenceCreate$ = {}));
