// @apexfintechsolutions/ascend-sdk/models/components/estateenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EstateEnrollmentMetadataCreate$ = exports.EstateEnrollmentMetadataCreate$outboundSchema = exports.EstateEnrollmentMetadataCreate$inboundSchema = exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var EstateEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (EstateEnrollmentMetadataCreateDividendReinvestmentPlan) {
    EstateEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    EstateEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    EstateEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(EstateEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan = EstateEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/** @internal */
exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(EstateEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(EstateEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EstateEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (EstateEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `EstateEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    EstateEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `EstateEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    EstateEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(EstateEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan$ = EstateEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.EstateEnrollmentMetadataCreate$inboundSchema = z.object({
    certificate_of_appointment_document_id: z.string().optional(),
    dividend_reinvestment_plan: exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "certificate_of_appointment_document_id": "certificateOfAppointmentDocumentId",
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
    });
});
/** @internal */
exports.EstateEnrollmentMetadataCreate$outboundSchema = z.object({
    certificateOfAppointmentDocumentId: z.string().optional(),
    dividendReinvestmentPlan: exports.EstateEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        certificateOfAppointmentDocumentId: "certificate_of_appointment_document_id",
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EstateEnrollmentMetadataCreate$;
(function (EstateEnrollmentMetadataCreate$) {
    /** @deprecated use `EstateEnrollmentMetadataCreate$inboundSchema` instead. */
    EstateEnrollmentMetadataCreate$.inboundSchema = exports.EstateEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `EstateEnrollmentMetadataCreate$outboundSchema` instead. */
    EstateEnrollmentMetadataCreate$.outboundSchema = exports.EstateEnrollmentMetadataCreate$outboundSchema;
})(EstateEnrollmentMetadataCreate$ || (exports.EstateEnrollmentMetadataCreate$ = EstateEnrollmentMetadataCreate$ = {}));
