// @apexfintechsolutions/ascend-sdk/models/components/eventmessage.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventMessage$ = exports.EventMessage$outboundSchema = exports.EventMessage$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.EventMessage$inboundSchema = z.object({
    account_id: z.string().optional(),
    client_id: z.string().optional(),
    correspondent_id: z.string().optional(),
    data: z.nullable(z.record(z.any())).optional(),
    event_type: z.string().optional(),
    message_id: z.string().optional(),
    name: z.string().optional(),
    partition_key: z.string().optional(),
    publish_time: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "client_id": "clientId",
        "correspondent_id": "correspondentId",
        "event_type": "eventType",
        "message_id": "messageId",
        "partition_key": "partitionKey",
        "publish_time": "publishTime",
    });
});
/** @internal */
exports.EventMessage$outboundSchema = z.object({
    accountId: z.string().optional(),
    clientId: z.string().optional(),
    correspondentId: z.string().optional(),
    data: z.nullable(z.record(z.any())).optional(),
    eventType: z.string().optional(),
    messageId: z.string().optional(),
    name: z.string().optional(),
    partitionKey: z.string().optional(),
    publishTime: z.nullable(z.date().transform(v => v.toISOString())).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        clientId: "client_id",
        correspondentId: "correspondent_id",
        eventType: "event_type",
        messageId: "message_id",
        partitionKey: "partition_key",
        publishTime: "publish_time",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EventMessage$;
(function (EventMessage$) {
    /** @deprecated use `EventMessage$inboundSchema` instead. */
    EventMessage$.inboundSchema = exports.EventMessage$inboundSchema;
    /** @deprecated use `EventMessage$outboundSchema` instead. */
    EventMessage$.outboundSchema = exports.EventMessage$outboundSchema;
})(EventMessage$ || (exports.EventMessage$ = EventMessage$ = {}));
