// @apexfintechsolutions/ascend-sdk/models/components/executedprice.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExecutedPrice$ = exports.ExecutedPrice$outboundSchema = exports.ExecutedPrice$inboundSchema = exports.ExecutedPriceType$ = exports.ExecutedPriceType$outboundSchema = exports.ExecutedPriceType$inboundSchema = exports.Price$ = exports.Price$outboundSchema = exports.Price$inboundSchema = exports.ExecutedPriceType = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
/**
 * The type of this price, which must be PRICE_PER_UNIT for equity orders, or PERCENTAGE_OF_PAR for fixed income orders.
 */
var ExecutedPriceType;
(function (ExecutedPriceType) {
    ExecutedPriceType["PricePerUnit"] = "PRICE_PER_UNIT";
    ExecutedPriceType["PercentageOfPar"] = "PERCENTAGE_OF_PAR";
    ExecutedPriceType["YieldToWorst"] = "YIELD_TO_WORST";
    ExecutedPriceType["YieldToMaturity"] = "YIELD_TO_MATURITY";
})(ExecutedPriceType || (exports.ExecutedPriceType = ExecutedPriceType = {}));
/** @internal */
exports.Price$inboundSchema = z
    .object({
    value: z.string().optional(),
});
/** @internal */
exports.Price$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Price$;
(function (Price$) {
    /** @deprecated use `Price$inboundSchema` instead. */
    Price$.inboundSchema = exports.Price$inboundSchema;
    /** @deprecated use `Price$outboundSchema` instead. */
    Price$.outboundSchema = exports.Price$outboundSchema;
})(Price$ || (exports.Price$ = Price$ = {}));
/** @internal */
exports.ExecutedPriceType$inboundSchema = z
    .union([
    z.nativeEnum(ExecutedPriceType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ExecutedPriceType$outboundSchema = z.union([
    z.nativeEnum(ExecutedPriceType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ExecutedPriceType$;
(function (ExecutedPriceType$) {
    /** @deprecated use `ExecutedPriceType$inboundSchema` instead. */
    ExecutedPriceType$.inboundSchema = exports.ExecutedPriceType$inboundSchema;
    /** @deprecated use `ExecutedPriceType$outboundSchema` instead. */
    ExecutedPriceType$.outboundSchema = exports.ExecutedPriceType$outboundSchema;
})(ExecutedPriceType$ || (exports.ExecutedPriceType$ = ExecutedPriceType$ = {}));
/** @internal */
exports.ExecutedPrice$inboundSchema = z.object({
    price: z.nullable(z.lazy(() => exports.Price$inboundSchema)).optional(),
    type: exports.ExecutedPriceType$inboundSchema.optional(),
});
/** @internal */
exports.ExecutedPrice$outboundSchema = z.object({
    price: z.nullable(z.lazy(() => exports.Price$outboundSchema)).optional(),
    type: exports.ExecutedPriceType$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ExecutedPrice$;
(function (ExecutedPrice$) {
    /** @deprecated use `ExecutedPrice$inboundSchema` instead. */
    ExecutedPrice$.inboundSchema = exports.ExecutedPrice$inboundSchema;
    /** @deprecated use `ExecutedPrice$outboundSchema` instead. */
    ExecutedPrice$.outboundSchema = exports.ExecutedPrice$outboundSchema;
})(ExecutedPrice$ || (exports.ExecutedPrice$ = ExecutedPrice$ = {}));
