// @apexfintechsolutions/ascend-sdk/models/components/feecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeeCreate$ = exports.FeeCreate$outboundSchema = exports.FeeCreate$inboundSchema = exports.FeeCreateType$ = exports.FeeCreateType$outboundSchema = exports.FeeCreateType$inboundSchema = exports.FeeCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
/**
 * The type of fee
 */
var FeeCreateType;
(function (FeeCreateType) {
    FeeCreateType["FeeTypeUnspecified"] = "FEE_TYPE_UNSPECIFIED";
    FeeCreateType["ClientClearing"] = "CLIENT_CLEARING";
    FeeCreateType["Liquidity"] = "LIQUIDITY";
    FeeCreateType["TradeActivity"] = "TRADE_ACTIVITY";
    FeeCreateType["FinancialTransactionTax"] = "FINANCIAL_TRANSACTION_TAX";
    FeeCreateType["IndexOptionFee"] = "INDEX_OPTION_FEE";
    FeeCreateType["SecFee"] = "SEC_FEE";
    FeeCreateType["OptionsRegulatory"] = "OPTIONS_REGULATORY";
    FeeCreateType["GeneralPurposeFee"] = "GENERAL_PURPOSE_FEE";
    FeeCreateType["BrokerFee"] = "BROKER_FEE";
    FeeCreateType["ContractFee"] = "CONTRACT_FEE";
    FeeCreateType["OccFee"] = "OCC_FEE";
})(FeeCreateType || (exports.FeeCreateType = FeeCreateType = {}));
/** @internal */
exports.FeeCreateType$inboundSchema = z
    .union([
    z.nativeEnum(FeeCreateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.FeeCreateType$outboundSchema = z.union([
    z.nativeEnum(FeeCreateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeeCreateType$;
(function (FeeCreateType$) {
    /** @deprecated use `FeeCreateType$inboundSchema` instead. */
    FeeCreateType$.inboundSchema = exports.FeeCreateType$inboundSchema;
    /** @deprecated use `FeeCreateType$outboundSchema` instead. */
    FeeCreateType$.outboundSchema = exports.FeeCreateType$outboundSchema;
})(FeeCreateType$ || (exports.FeeCreateType$ = FeeCreateType$ = {}));
/** @internal */
exports.FeeCreate$inboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$inboundSchema.optional(),
    suppress_fee: z.boolean().optional(),
    type: exports.FeeCreateType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "suppress_fee": "suppressFee",
    });
});
/** @internal */
exports.FeeCreate$outboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$outboundSchema.optional(),
    suppressFee: z.boolean().optional(),
    type: exports.FeeCreateType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        suppressFee: "suppress_fee",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FeeCreate$;
(function (FeeCreate$) {
    /** @deprecated use `FeeCreate$inboundSchema` instead. */
    FeeCreate$.inboundSchema = exports.FeeCreate$inboundSchema;
    /** @deprecated use `FeeCreate$outboundSchema` instead. */
    FeeCreate$.outboundSchema = exports.FeeCreate$outboundSchema;
})(FeeCreate$ || (exports.FeeCreate$ = FeeCreate$ = {}));
