// @apexfintechsolutions/ascend-sdk/models/components/foreignindividualaccountenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ForeignIndividualAccountEnrollmentMetadataCreate$ = exports.ForeignIndividualAccountEnrollmentMetadataCreate$outboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreate$inboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$ = exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep = exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const foreignnaturalpersonaccountenrollmentmetadatacreate_js_1 = require("./foreignnaturalpersonaccountenrollmentmetadatacreate.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan) {
    ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan = ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep;
(function (ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep) {
    ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep || (exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep = ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$ = ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$;
(function (ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$ || (exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$ = ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.ForeignIndividualAccountEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
    foreign_natural_person_account_enrollment_metadata: foreignnaturalpersonaccountenrollmentmetadatacreate_js_1.ForeignNaturalPersonAccountEnrollmentMetadataCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
        "foreign_natural_person_account_enrollment_metadata": "foreignNaturalPersonAccountEnrollmentMetadata",
    });
});
/** @internal */
exports.ForeignIndividualAccountEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.ForeignIndividualAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.ForeignIndividualAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
    foreignNaturalPersonAccountEnrollmentMetadata: foreignnaturalpersonaccountenrollmentmetadatacreate_js_1.ForeignNaturalPersonAccountEnrollmentMetadataCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
        foreignNaturalPersonAccountEnrollmentMetadata: "foreign_natural_person_account_enrollment_metadata",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ForeignIndividualAccountEnrollmentMetadataCreate$;
(function (ForeignIndividualAccountEnrollmentMetadataCreate$) {
    /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreate$inboundSchema` instead. */
    ForeignIndividualAccountEnrollmentMetadataCreate$.inboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `ForeignIndividualAccountEnrollmentMetadataCreate$outboundSchema` instead. */
    ForeignIndividualAccountEnrollmentMetadataCreate$.outboundSchema = exports.ForeignIndividualAccountEnrollmentMetadataCreate$outboundSchema;
})(ForeignIndividualAccountEnrollmentMetadataCreate$ || (exports.ForeignIndividualAccountEnrollmentMetadataCreate$ = ForeignIndividualAccountEnrollmentMetadataCreate$ = {}));
