// @apexfintechsolutions/ascend-sdk/models/components/foreignjointaccountenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ForeignJointAccountEnrollmentMetadataCreate$ = exports.ForeignJointAccountEnrollmentMetadataCreate$outboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreate$inboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$ = exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep = exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const foreignnaturalpersonaccountenrollmentmetadatacreate_js_1 = require("./foreignnaturalpersonaccountenrollmentmetadatacreate.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan) {
    ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan = ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep;
(function (ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep) {
    ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep || (exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep = ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$ = ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$;
(function (ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$ || (exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$ = ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.ForeignJointAccountEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
    foreign_natural_person_account_enrollment_metadata: foreignnaturalpersonaccountenrollmentmetadatacreate_js_1.ForeignNaturalPersonAccountEnrollmentMetadataCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
        "foreign_natural_person_account_enrollment_metadata": "foreignNaturalPersonAccountEnrollmentMetadata",
    });
});
/** @internal */
exports.ForeignJointAccountEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.ForeignJointAccountEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.ForeignJointAccountEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
    foreignNaturalPersonAccountEnrollmentMetadata: foreignnaturalpersonaccountenrollmentmetadatacreate_js_1.ForeignNaturalPersonAccountEnrollmentMetadataCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
        foreignNaturalPersonAccountEnrollmentMetadata: "foreign_natural_person_account_enrollment_metadata",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ForeignJointAccountEnrollmentMetadataCreate$;
(function (ForeignJointAccountEnrollmentMetadataCreate$) {
    /** @deprecated use `ForeignJointAccountEnrollmentMetadataCreate$inboundSchema` instead. */
    ForeignJointAccountEnrollmentMetadataCreate$.inboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `ForeignJointAccountEnrollmentMetadataCreate$outboundSchema` instead. */
    ForeignJointAccountEnrollmentMetadataCreate$.outboundSchema = exports.ForeignJointAccountEnrollmentMetadataCreate$outboundSchema;
})(ForeignJointAccountEnrollmentMetadataCreate$ || (exports.ForeignJointAccountEnrollmentMetadataCreate$ = ForeignJointAccountEnrollmentMetadataCreate$ = {}));
