// @apexfintechsolutions/ascend-sdk/models/components/foreignidentificationupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ForeignIdentificationUpdate$ = exports.ForeignIdentificationUpdate$outboundSchema = exports.ForeignIdentificationUpdate$inboundSchema = exports.ForeignIdentificationUpdateType$ = exports.ForeignIdentificationUpdateType$outboundSchema = exports.ForeignIdentificationUpdateType$inboundSchema = exports.ForeignIdentificationUpdateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const dateupdate_js_1 = require("./dateupdate.js");
/**
 * Identification type
 */
var ForeignIdentificationUpdateType;
(function (ForeignIdentificationUpdateType) {
    ForeignIdentificationUpdateType["IdentificationTypeUnspecified"] = "IDENTIFICATION_TYPE_UNSPECIFIED";
    ForeignIdentificationUpdateType["Passport"] = "PASSPORT";
    ForeignIdentificationUpdateType["NationalId"] = "NATIONAL_ID";
    ForeignIdentificationUpdateType["DriversLicense"] = "DRIVERS_LICENSE";
})(ForeignIdentificationUpdateType || (exports.ForeignIdentificationUpdateType = ForeignIdentificationUpdateType = {}));
/** @internal */
exports.ForeignIdentificationUpdateType$inboundSchema = z
    .union([
    z.nativeEnum(ForeignIdentificationUpdateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ForeignIdentificationUpdateType$outboundSchema = z.union([
    z.nativeEnum(ForeignIdentificationUpdateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ForeignIdentificationUpdateType$;
(function (ForeignIdentificationUpdateType$) {
    /** @deprecated use `ForeignIdentificationUpdateType$inboundSchema` instead. */
    ForeignIdentificationUpdateType$.inboundSchema = exports.ForeignIdentificationUpdateType$inboundSchema;
    /** @deprecated use `ForeignIdentificationUpdateType$outboundSchema` instead. */
    ForeignIdentificationUpdateType$.outboundSchema = exports.ForeignIdentificationUpdateType$outboundSchema;
})(ForeignIdentificationUpdateType$ || (exports.ForeignIdentificationUpdateType$ = ForeignIdentificationUpdateType$ = {}));
/** @internal */
exports.ForeignIdentificationUpdate$inboundSchema = z.object({
    expiration_date: dateupdate_js_1.DateUpdate$inboundSchema.optional(),
    ftin: z.boolean().optional(),
    identification_number: z.string().optional(),
    issue_date: dateupdate_js_1.DateUpdate$inboundSchema.optional(),
    issuing_region_code: z.string().optional(),
    type: exports.ForeignIdentificationUpdateType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "expiration_date": "expirationDate",
        "identification_number": "identificationNumber",
        "issue_date": "issueDate",
        "issuing_region_code": "issuingRegionCode",
    });
});
/** @internal */
exports.ForeignIdentificationUpdate$outboundSchema = z.object({
    expirationDate: dateupdate_js_1.DateUpdate$outboundSchema.optional(),
    ftin: z.boolean().optional(),
    identificationNumber: z.string().optional(),
    issueDate: dateupdate_js_1.DateUpdate$outboundSchema.optional(),
    issuingRegionCode: z.string().optional(),
    type: exports.ForeignIdentificationUpdateType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        expirationDate: "expiration_date",
        identificationNumber: "identification_number",
        issueDate: "issue_date",
        issuingRegionCode: "issuing_region_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ForeignIdentificationUpdate$;
(function (ForeignIdentificationUpdate$) {
    /** @deprecated use `ForeignIdentificationUpdate$inboundSchema` instead. */
    ForeignIdentificationUpdate$.inboundSchema = exports.ForeignIdentificationUpdate$inboundSchema;
    /** @deprecated use `ForeignIdentificationUpdate$outboundSchema` instead. */
    ForeignIdentificationUpdate$.outboundSchema = exports.ForeignIdentificationUpdate$outboundSchema;
})(ForeignIdentificationUpdate$ || (exports.ForeignIdentificationUpdate$ = ForeignIdentificationUpdate$ = {}));
