// @apexfintechsolutions/ascend-sdk/models/components/httpmetadata.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.HTTPMetadata$ = exports.HTTPMetadata$outboundSchema = exports.HTTPMetadata$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.HTTPMetadata$inboundSchema = z.object({
    Response: z.instanceof(Response),
    Request: z.instanceof(Request),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Response": "response",
        "Request": "request",
    });
});
/** @internal */
exports.HTTPMetadata$outboundSchema = z.object({
    response: z.instanceof(Response).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
    request: z.instanceof(Request).transform(() => {
        throw new Error("Response cannot be serialized");
    }),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        response: "Response",
        request: "Request",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var HTTPMetadata$;
(function (HTTPMetadata$) {
    /** @deprecated use `HTTPMetadata$inboundSchema` instead. */
    HTTPMetadata$.inboundSchema = exports.HTTPMetadata$inboundSchema;
    /** @deprecated use `HTTPMetadata$outboundSchema` instead. */
    HTTPMetadata$.outboundSchema = exports.HTTPMetadata$outboundSchema;
})(HTTPMetadata$ || (exports.HTTPMetadata$ = HTTPMetadata$ = {}));
