// @apexfintechsolutions/ascend-sdk/models/components/identifiercreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentifierCreate$ = exports.IdentifierCreate$outboundSchema = exports.IdentifierCreate$inboundSchema = exports.IdentifierCreateType$ = exports.IdentifierCreateType$outboundSchema = exports.IdentifierCreateType$inboundSchema = exports.IdentifierCreateType = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
/**
 * The type of identifier
 */
var IdentifierCreateType;
(function (IdentifierCreateType) {
    IdentifierCreateType["IdentifierTypeUnspecified"] = "IDENTIFIER_TYPE_UNSPECIFIED";
    IdentifierCreateType["OriginatingAccountId"] = "ORIGINATING_ACCOUNT_ID";
    IdentifierCreateType["OriginatingFdid"] = "ORIGINATING_FDID";
    IdentifierCreateType["OriginatingCatReporterCrd"] = "ORIGINATING_CAT_REPORTER_CRD";
    IdentifierCreateType["ClientAccountId"] = "CLIENT_ACCOUNT_ID";
})(IdentifierCreateType || (exports.IdentifierCreateType = IdentifierCreateType = {}));
/** @internal */
exports.IdentifierCreateType$inboundSchema = z
    .union([
    z.nativeEnum(IdentifierCreateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IdentifierCreateType$outboundSchema = z.union([
    z.nativeEnum(IdentifierCreateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IdentifierCreateType$;
(function (IdentifierCreateType$) {
    /** @deprecated use `IdentifierCreateType$inboundSchema` instead. */
    IdentifierCreateType$.inboundSchema = exports.IdentifierCreateType$inboundSchema;
    /** @deprecated use `IdentifierCreateType$outboundSchema` instead. */
    IdentifierCreateType$.outboundSchema = exports.IdentifierCreateType$outboundSchema;
})(IdentifierCreateType$ || (exports.IdentifierCreateType$ = IdentifierCreateType$ = {}));
/** @internal */
exports.IdentifierCreate$inboundSchema = z.object({
    type: exports.IdentifierCreateType$inboundSchema,
    value: z.string(),
});
/** @internal */
exports.IdentifierCreate$outboundSchema = z.object({
    type: exports.IdentifierCreateType$outboundSchema,
    value: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IdentifierCreate$;
(function (IdentifierCreate$) {
    /** @deprecated use `IdentifierCreate$inboundSchema` instead. */
    IdentifierCreate$.inboundSchema = exports.IdentifierCreate$inboundSchema;
    /** @deprecated use `IdentifierCreate$outboundSchema` instead. */
    IdentifierCreate$.outboundSchema = exports.IdentifierCreate$outboundSchema;
})(IdentifierCreate$ || (exports.IdentifierCreate$ = IdentifierCreate$ = {}));
