// @apexfintechsolutions/ascend-sdk/models/components/identifier.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Identifier$ = exports.Identifier$outboundSchema = exports.Identifier$inboundSchema = exports.IdentifierType1$ = exports.IdentifierType1$outboundSchema = exports.IdentifierType1$inboundSchema = exports.IdentifierType1 = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
/**
 * The type of identifier
 */
var IdentifierType1;
(function (IdentifierType1) {
    IdentifierType1["IdentifierTypeUnspecified"] = "IDENTIFIER_TYPE_UNSPECIFIED";
    IdentifierType1["OriginatingAccountId"] = "ORIGINATING_ACCOUNT_ID";
    IdentifierType1["OriginatingFdid"] = "ORIGINATING_FDID";
    IdentifierType1["OriginatingCatReporterCrd"] = "ORIGINATING_CAT_REPORTER_CRD";
    IdentifierType1["ClientAccountId"] = "CLIENT_ACCOUNT_ID";
})(IdentifierType1 || (exports.IdentifierType1 = IdentifierType1 = {}));
/** @internal */
exports.IdentifierType1$inboundSchema = z
    .union([
    z.nativeEnum(IdentifierType1),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IdentifierType1$outboundSchema = z.union([
    z.nativeEnum(IdentifierType1),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IdentifierType1$;
(function (IdentifierType1$) {
    /** @deprecated use `IdentifierType1$inboundSchema` instead. */
    IdentifierType1$.inboundSchema = exports.IdentifierType1$inboundSchema;
    /** @deprecated use `IdentifierType1$outboundSchema` instead. */
    IdentifierType1$.outboundSchema = exports.IdentifierType1$outboundSchema;
})(IdentifierType1$ || (exports.IdentifierType1$ = IdentifierType1$ = {}));
/** @internal */
exports.Identifier$inboundSchema = z.object({
    type: exports.IdentifierType1$inboundSchema.optional(),
    value: z.string().optional(),
});
/** @internal */
exports.Identifier$outboundSchema = z.object({
    type: exports.IdentifierType1$outboundSchema.optional(),
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Identifier$;
(function (Identifier$) {
    /** @deprecated use `Identifier$inboundSchema` instead. */
    Identifier$.inboundSchema = exports.Identifier$inboundSchema;
    /** @deprecated use `Identifier$outboundSchema` instead. */
    Identifier$.outboundSchema = exports.Identifier$outboundSchema;
})(Identifier$ || (exports.Identifier$ = Identifier$ = {}));
