// @apexfintechsolutions/ascend-sdk/models/components/identifierupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentifierUpdate$ = exports.IdentifierUpdate$outboundSchema = exports.IdentifierUpdate$inboundSchema = exports.IdentifierUpdateType$ = exports.IdentifierUpdateType$outboundSchema = exports.IdentifierUpdateType$inboundSchema = exports.IdentifierUpdateType = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
/**
 * The type of identifier
 */
var IdentifierUpdateType;
(function (IdentifierUpdateType) {
    IdentifierUpdateType["IdentifierTypeUnspecified"] = "IDENTIFIER_TYPE_UNSPECIFIED";
    IdentifierUpdateType["OriginatingAccountId"] = "ORIGINATING_ACCOUNT_ID";
    IdentifierUpdateType["OriginatingFdid"] = "ORIGINATING_FDID";
    IdentifierUpdateType["OriginatingCatReporterCrd"] = "ORIGINATING_CAT_REPORTER_CRD";
    IdentifierUpdateType["ClientAccountId"] = "CLIENT_ACCOUNT_ID";
})(IdentifierUpdateType || (exports.IdentifierUpdateType = IdentifierUpdateType = {}));
/** @internal */
exports.IdentifierUpdateType$inboundSchema = z
    .union([
    z.nativeEnum(IdentifierUpdateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IdentifierUpdateType$outboundSchema = z.union([
    z.nativeEnum(IdentifierUpdateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IdentifierUpdateType$;
(function (IdentifierUpdateType$) {
    /** @deprecated use `IdentifierUpdateType$inboundSchema` instead. */
    IdentifierUpdateType$.inboundSchema = exports.IdentifierUpdateType$inboundSchema;
    /** @deprecated use `IdentifierUpdateType$outboundSchema` instead. */
    IdentifierUpdateType$.outboundSchema = exports.IdentifierUpdateType$outboundSchema;
})(IdentifierUpdateType$ || (exports.IdentifierUpdateType$ = IdentifierUpdateType$ = {}));
/** @internal */
exports.IdentifierUpdate$inboundSchema = z.object({
    type: exports.IdentifierUpdateType$inboundSchema.optional(),
});
/** @internal */
exports.IdentifierUpdate$outboundSchema = z.object({
    type: exports.IdentifierUpdateType$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IdentifierUpdate$;
(function (IdentifierUpdate$) {
    /** @deprecated use `IdentifierUpdate$inboundSchema` instead. */
    IdentifierUpdate$.inboundSchema = exports.IdentifierUpdate$inboundSchema;
    /** @deprecated use `IdentifierUpdate$outboundSchema` instead. */
    IdentifierUpdate$.outboundSchema = exports.IdentifierUpdate$outboundSchema;
})(IdentifierUpdate$ || (exports.IdentifierUpdate$ = IdentifierUpdate$ = {}));
