// @apexfintechsolutions/ascend-sdk/models/components/identityverificationresultcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityVerificationResultCreate$ = exports.IdentityVerificationResultCreate$outboundSchema = exports.IdentityVerificationResultCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const datecreate_js_1 = require("./datecreate.js");
/** @internal */
exports.IdentityVerificationResultCreate$inboundSchema = z.object({
    address_verified: z.boolean(),
    birth_date_verified: z.boolean(),
    execution_date: datecreate_js_1.DateCreate$inboundSchema,
    external_case_id: z.string(),
    identity_verification_document_ids: z.array(z.string()).optional(),
    name_verified: z.boolean(),
    raw_vendor_data_document_id: z.string().optional(),
    tax_id_verified: z.boolean(),
    vendor: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "address_verified": "addressVerified",
        "birth_date_verified": "birthDateVerified",
        "execution_date": "executionDate",
        "external_case_id": "externalCaseId",
        "identity_verification_document_ids": "identityVerificationDocumentIds",
        "name_verified": "nameVerified",
        "raw_vendor_data_document_id": "rawVendorDataDocumentId",
        "tax_id_verified": "taxIdVerified",
    });
});
/** @internal */
exports.IdentityVerificationResultCreate$outboundSchema = z.object({
    addressVerified: z.boolean(),
    birthDateVerified: z.boolean(),
    executionDate: datecreate_js_1.DateCreate$outboundSchema,
    externalCaseId: z.string(),
    identityVerificationDocumentIds: z.array(z.string()).optional(),
    nameVerified: z.boolean(),
    rawVendorDataDocumentId: z.string().optional(),
    taxIdVerified: z.boolean(),
    vendor: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addressVerified: "address_verified",
        birthDateVerified: "birth_date_verified",
        executionDate: "execution_date",
        externalCaseId: "external_case_id",
        identityVerificationDocumentIds: "identity_verification_document_ids",
        nameVerified: "name_verified",
        rawVendorDataDocumentId: "raw_vendor_data_document_id",
        taxIdVerified: "tax_id_verified",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IdentityVerificationResultCreate$;
(function (IdentityVerificationResultCreate$) {
    /** @deprecated use `IdentityVerificationResultCreate$inboundSchema` instead. */
    IdentityVerificationResultCreate$.inboundSchema = exports.IdentityVerificationResultCreate$inboundSchema;
    /** @deprecated use `IdentityVerificationResultCreate$outboundSchema` instead. */
    IdentityVerificationResultCreate$.outboundSchema = exports.IdentityVerificationResultCreate$outboundSchema;
})(IdentityVerificationResultCreate$ || (exports.IdentityVerificationResultCreate$ = IdentityVerificationResultCreate$ = {}));
