// @apexfintechsolutions/ascend-sdk/models/components/individualenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndividualEnrollmentMetadataCreate$ = exports.IndividualEnrollmentMetadataCreate$outboundSchema = exports.IndividualEnrollmentMetadataCreate$inboundSchema = exports.IndividualEnrollmentMetadataCreateFdicCashSweep$ = exports.IndividualEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.IndividualEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.IndividualEnrollmentMetadataCreateFdicCashSweep = exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var IndividualEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (IndividualEnrollmentMetadataCreateDividendReinvestmentPlan) {
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(IndividualEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan = IndividualEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var IndividualEnrollmentMetadataCreateFdicCashSweep;
(function (IndividualEnrollmentMetadataCreateFdicCashSweep) {
    IndividualEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    IndividualEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    IndividualEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(IndividualEnrollmentMetadataCreateFdicCashSweep || (exports.IndividualEnrollmentMetadataCreateFdicCashSweep = IndividualEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(IndividualEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(IndividualEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$ = IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.IndividualEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(IndividualEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IndividualEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(IndividualEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IndividualEnrollmentMetadataCreateFdicCashSweep$;
(function (IndividualEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `IndividualEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    IndividualEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.IndividualEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `IndividualEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    IndividualEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.IndividualEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(IndividualEnrollmentMetadataCreateFdicCashSweep$ || (exports.IndividualEnrollmentMetadataCreateFdicCashSweep$ = IndividualEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.IndividualEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.IndividualEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
    });
});
/** @internal */
exports.IndividualEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.IndividualEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.IndividualEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IndividualEnrollmentMetadataCreate$;
(function (IndividualEnrollmentMetadataCreate$) {
    /** @deprecated use `IndividualEnrollmentMetadataCreate$inboundSchema` instead. */
    IndividualEnrollmentMetadataCreate$.inboundSchema = exports.IndividualEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `IndividualEnrollmentMetadataCreate$outboundSchema` instead. */
    IndividualEnrollmentMetadataCreate$.outboundSchema = exports.IndividualEnrollmentMetadataCreate$outboundSchema;
})(IndividualEnrollmentMetadataCreate$ || (exports.IndividualEnrollmentMetadataCreate$ = IndividualEnrollmentMetadataCreate$ = {}));
