// @apexfintechsolutions/ascend-sdk/models/components/interestedpartycreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InterestedPartyCreate$ = exports.InterestedPartyCreate$outboundSchema = exports.InterestedPartyCreate$inboundSchema = exports.TradeConfirmationDeliveryPreference$ = exports.TradeConfirmationDeliveryPreference$outboundSchema = exports.TradeConfirmationDeliveryPreference$inboundSchema = exports.StatementDeliveryPreference$ = exports.StatementDeliveryPreference$outboundSchema = exports.StatementDeliveryPreference$inboundSchema = exports.TradeConfirmationDeliveryPreference = exports.StatementDeliveryPreference = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const postaladdresscreate_js_1 = require("./postaladdresscreate.js");
/**
 * Delivery method instruction for account statements for a given Interested Party; Can be `PHYSICAL`, `SUPPRESS`; Defaults to `PHYSICAL` on party creation
 */
var StatementDeliveryPreference;
(function (StatementDeliveryPreference) {
    StatementDeliveryPreference["Physical"] = "PHYSICAL";
    StatementDeliveryPreference["Suppress"] = "SUPPRESS";
})(StatementDeliveryPreference || (exports.StatementDeliveryPreference = StatementDeliveryPreference = {}));
/**
 * Delivery method instruction for trade confirmations for a given Interested Party record; Can be `PHYSICAL`, `SUPPRESS`; Defaults to `PHYSICAL` on party creation
 */
var TradeConfirmationDeliveryPreference;
(function (TradeConfirmationDeliveryPreference) {
    TradeConfirmationDeliveryPreference["Physical"] = "PHYSICAL";
    TradeConfirmationDeliveryPreference["Suppress"] = "SUPPRESS";
})(TradeConfirmationDeliveryPreference || (exports.TradeConfirmationDeliveryPreference = TradeConfirmationDeliveryPreference = {}));
/** @internal */
exports.StatementDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(StatementDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.StatementDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(StatementDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StatementDeliveryPreference$;
(function (StatementDeliveryPreference$) {
    /** @deprecated use `StatementDeliveryPreference$inboundSchema` instead. */
    StatementDeliveryPreference$.inboundSchema = exports.StatementDeliveryPreference$inboundSchema;
    /** @deprecated use `StatementDeliveryPreference$outboundSchema` instead. */
    StatementDeliveryPreference$.outboundSchema = exports.StatementDeliveryPreference$outboundSchema;
})(StatementDeliveryPreference$ || (exports.StatementDeliveryPreference$ = StatementDeliveryPreference$ = {}));
/** @internal */
exports.TradeConfirmationDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(TradeConfirmationDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TradeConfirmationDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(TradeConfirmationDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TradeConfirmationDeliveryPreference$;
(function (TradeConfirmationDeliveryPreference$) {
    /** @deprecated use `TradeConfirmationDeliveryPreference$inboundSchema` instead. */
    TradeConfirmationDeliveryPreference$.inboundSchema = exports.TradeConfirmationDeliveryPreference$inboundSchema;
    /** @deprecated use `TradeConfirmationDeliveryPreference$outboundSchema` instead. */
    TradeConfirmationDeliveryPreference$.outboundSchema = exports.TradeConfirmationDeliveryPreference$outboundSchema;
})(TradeConfirmationDeliveryPreference$ || (exports.TradeConfirmationDeliveryPreference$ = TradeConfirmationDeliveryPreference$ = {}));
/** @internal */
exports.InterestedPartyCreate$inboundSchema = z.object({
    mailing_address: postaladdresscreate_js_1.PostalAddressCreate$inboundSchema,
    recipient: z.string(),
    statement_delivery_preference: exports.StatementDeliveryPreference$inboundSchema
        .optional(),
    trade_confirmation_delivery_preference: exports.TradeConfirmationDeliveryPreference$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "mailing_address": "mailingAddress",
        "statement_delivery_preference": "statementDeliveryPreference",
        "trade_confirmation_delivery_preference": "tradeConfirmationDeliveryPreference",
    });
});
/** @internal */
exports.InterestedPartyCreate$outboundSchema = z.object({
    mailingAddress: postaladdresscreate_js_1.PostalAddressCreate$outboundSchema,
    recipient: z.string(),
    statementDeliveryPreference: exports.StatementDeliveryPreference$outboundSchema
        .optional(),
    tradeConfirmationDeliveryPreference: exports.TradeConfirmationDeliveryPreference$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        mailingAddress: "mailing_address",
        statementDeliveryPreference: "statement_delivery_preference",
        tradeConfirmationDeliveryPreference: "trade_confirmation_delivery_preference",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InterestedPartyCreate$;
(function (InterestedPartyCreate$) {
    /** @deprecated use `InterestedPartyCreate$inboundSchema` instead. */
    InterestedPartyCreate$.inboundSchema = exports.InterestedPartyCreate$inboundSchema;
    /** @deprecated use `InterestedPartyCreate$outboundSchema` instead. */
    InterestedPartyCreate$.outboundSchema = exports.InterestedPartyCreate$outboundSchema;
})(InterestedPartyCreate$ || (exports.InterestedPartyCreate$ = InterestedPartyCreate$ = {}));
