// @apexfintechsolutions/ascend-sdk/models/components/interestedpartyupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InterestedPartyUpdate$ = exports.InterestedPartyUpdate$outboundSchema = exports.InterestedPartyUpdate$inboundSchema = exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference$ = exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference$outboundSchema = exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference$inboundSchema = exports.InterestedPartyUpdateStatementDeliveryPreference$ = exports.InterestedPartyUpdateStatementDeliveryPreference$outboundSchema = exports.InterestedPartyUpdateStatementDeliveryPreference$inboundSchema = exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference = exports.InterestedPartyUpdateStatementDeliveryPreference = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const postaladdressupdate_js_1 = require("./postaladdressupdate.js");
/**
 * Delivery method instruction for account statements for a given Interested Party; Can be `PHYSICAL`, `SUPPRESS`; Defaults to `PHYSICAL` on party creation
 */
var InterestedPartyUpdateStatementDeliveryPreference;
(function (InterestedPartyUpdateStatementDeliveryPreference) {
    InterestedPartyUpdateStatementDeliveryPreference["Physical"] = "PHYSICAL";
    InterestedPartyUpdateStatementDeliveryPreference["Suppress"] = "SUPPRESS";
})(InterestedPartyUpdateStatementDeliveryPreference || (exports.InterestedPartyUpdateStatementDeliveryPreference = InterestedPartyUpdateStatementDeliveryPreference = {}));
/**
 * Delivery method instruction for trade confirmations for a given Interested Party record; Can be `PHYSICAL`, `SUPPRESS`; Defaults to `PHYSICAL` on party creation
 */
var InterestedPartyUpdateTradeConfirmationDeliveryPreference;
(function (InterestedPartyUpdateTradeConfirmationDeliveryPreference) {
    InterestedPartyUpdateTradeConfirmationDeliveryPreference["Physical"] = "PHYSICAL";
    InterestedPartyUpdateTradeConfirmationDeliveryPreference["Suppress"] = "SUPPRESS";
})(InterestedPartyUpdateTradeConfirmationDeliveryPreference || (exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference = InterestedPartyUpdateTradeConfirmationDeliveryPreference = {}));
/** @internal */
exports.InterestedPartyUpdateStatementDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(InterestedPartyUpdateStatementDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.InterestedPartyUpdateStatementDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(InterestedPartyUpdateStatementDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InterestedPartyUpdateStatementDeliveryPreference$;
(function (InterestedPartyUpdateStatementDeliveryPreference$) {
    /** @deprecated use `InterestedPartyUpdateStatementDeliveryPreference$inboundSchema` instead. */
    InterestedPartyUpdateStatementDeliveryPreference$.inboundSchema = exports.InterestedPartyUpdateStatementDeliveryPreference$inboundSchema;
    /** @deprecated use `InterestedPartyUpdateStatementDeliveryPreference$outboundSchema` instead. */
    InterestedPartyUpdateStatementDeliveryPreference$.outboundSchema = exports.InterestedPartyUpdateStatementDeliveryPreference$outboundSchema;
})(InterestedPartyUpdateStatementDeliveryPreference$ || (exports.InterestedPartyUpdateStatementDeliveryPreference$ = InterestedPartyUpdateStatementDeliveryPreference$ = {}));
/** @internal */
exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(InterestedPartyUpdateTradeConfirmationDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(InterestedPartyUpdateTradeConfirmationDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InterestedPartyUpdateTradeConfirmationDeliveryPreference$;
(function (InterestedPartyUpdateTradeConfirmationDeliveryPreference$) {
    /** @deprecated use `InterestedPartyUpdateTradeConfirmationDeliveryPreference$inboundSchema` instead. */
    InterestedPartyUpdateTradeConfirmationDeliveryPreference$.inboundSchema = exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference$inboundSchema;
    /** @deprecated use `InterestedPartyUpdateTradeConfirmationDeliveryPreference$outboundSchema` instead. */
    InterestedPartyUpdateTradeConfirmationDeliveryPreference$.outboundSchema = exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference$outboundSchema;
})(InterestedPartyUpdateTradeConfirmationDeliveryPreference$ || (exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference$ = InterestedPartyUpdateTradeConfirmationDeliveryPreference$ = {}));
/** @internal */
exports.InterestedPartyUpdate$inboundSchema = z.object({
    mailing_address: postaladdressupdate_js_1.PostalAddressUpdate$inboundSchema.optional(),
    recipient: z.string().optional(),
    statement_delivery_preference: exports.InterestedPartyUpdateStatementDeliveryPreference$inboundSchema.optional(),
    trade_confirmation_delivery_preference: exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "mailing_address": "mailingAddress",
        "statement_delivery_preference": "statementDeliveryPreference",
        "trade_confirmation_delivery_preference": "tradeConfirmationDeliveryPreference",
    });
});
/** @internal */
exports.InterestedPartyUpdate$outboundSchema = z.object({
    mailingAddress: postaladdressupdate_js_1.PostalAddressUpdate$outboundSchema.optional(),
    recipient: z.string().optional(),
    statementDeliveryPreference: exports.InterestedPartyUpdateStatementDeliveryPreference$outboundSchema.optional(),
    tradeConfirmationDeliveryPreference: exports.InterestedPartyUpdateTradeConfirmationDeliveryPreference$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        mailingAddress: "mailing_address",
        statementDeliveryPreference: "statement_delivery_preference",
        tradeConfirmationDeliveryPreference: "trade_confirmation_delivery_preference",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InterestedPartyUpdate$;
(function (InterestedPartyUpdate$) {
    /** @deprecated use `InterestedPartyUpdate$inboundSchema` instead. */
    InterestedPartyUpdate$.inboundSchema = exports.InterestedPartyUpdate$inboundSchema;
    /** @deprecated use `InterestedPartyUpdate$outboundSchema` instead. */
    InterestedPartyUpdate$.outboundSchema = exports.InterestedPartyUpdate$outboundSchema;
})(InterestedPartyUpdate$ || (exports.InterestedPartyUpdate$ = InterestedPartyUpdate$ = {}));
