// @apexfintechsolutions/ascend-sdk/models/components/investmentprofileupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvestmentProfileUpdate$ = exports.InvestmentProfileUpdate$outboundSchema = exports.InvestmentProfileUpdate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const accountgoalsupdate_js_1 = require("./accountgoalsupdate.js");
const customerprofileupdate_js_1 = require("./customerprofileupdate.js");
/** @internal */
exports.InvestmentProfileUpdate$inboundSchema = z.object({
    account_goals: accountgoalsupdate_js_1.AccountGoalsUpdate$inboundSchema.optional(),
    customer_profile: customerprofileupdate_js_1.CustomerProfileUpdate$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_goals": "accountGoals",
        "customer_profile": "customerProfile",
    });
});
/** @internal */
exports.InvestmentProfileUpdate$outboundSchema = z.object({
    accountGoals: accountgoalsupdate_js_1.AccountGoalsUpdate$outboundSchema.optional(),
    customerProfile: customerprofileupdate_js_1.CustomerProfileUpdate$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountGoals: "account_goals",
        customerProfile: "customer_profile",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InvestmentProfileUpdate$;
(function (InvestmentProfileUpdate$) {
    /** @deprecated use `InvestmentProfileUpdate$inboundSchema` instead. */
    InvestmentProfileUpdate$.inboundSchema = exports.InvestmentProfileUpdate$inboundSchema;
    /** @deprecated use `InvestmentProfileUpdate$outboundSchema` instead. */
    InvestmentProfileUpdate$.outboundSchema = exports.InvestmentProfileUpdate$outboundSchema;
})(InvestmentProfileUpdate$ || (exports.InvestmentProfileUpdate$ = InvestmentProfileUpdate$ = {}));
