// @apexfintechsolutions/ascend-sdk/models/components/investmentvehicleexperiencecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvestmentVehicleExperienceCreate$ = exports.InvestmentVehicleExperienceCreate$outboundSchema = exports.InvestmentVehicleExperienceCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.InvestmentVehicleExperienceCreate$inboundSchema = z.object({
    commodities_average_annual_trade_count: z.number().int(),
    corporate_bonds_average_annual_trade_count: z.number().int(),
    funds_average_annual_trade_count: z.number().int(),
    government_bonds_average_annual_trade_count: z.number().int(),
    margin_average_annual_trade_count: z.number().int(),
    municipal_bonds_average_annual_trade_count: z.number().int(),
    options_average_annual_trade_count: z.number().int(),
    stocks_average_annual_trade_count: z.number().int(),
    tax_shelters_average_annual_trade_count: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "commodities_average_annual_trade_count": "commoditiesAverageAnnualTradeCount",
        "corporate_bonds_average_annual_trade_count": "corporateBondsAverageAnnualTradeCount",
        "funds_average_annual_trade_count": "fundsAverageAnnualTradeCount",
        "government_bonds_average_annual_trade_count": "governmentBondsAverageAnnualTradeCount",
        "margin_average_annual_trade_count": "marginAverageAnnualTradeCount",
        "municipal_bonds_average_annual_trade_count": "municipalBondsAverageAnnualTradeCount",
        "options_average_annual_trade_count": "optionsAverageAnnualTradeCount",
        "stocks_average_annual_trade_count": "stocksAverageAnnualTradeCount",
        "tax_shelters_average_annual_trade_count": "taxSheltersAverageAnnualTradeCount",
    });
});
/** @internal */
exports.InvestmentVehicleExperienceCreate$outboundSchema = z.object({
    commoditiesAverageAnnualTradeCount: z.number().int(),
    corporateBondsAverageAnnualTradeCount: z.number().int(),
    fundsAverageAnnualTradeCount: z.number().int(),
    governmentBondsAverageAnnualTradeCount: z.number().int(),
    marginAverageAnnualTradeCount: z.number().int(),
    municipalBondsAverageAnnualTradeCount: z.number().int(),
    optionsAverageAnnualTradeCount: z.number().int(),
    stocksAverageAnnualTradeCount: z.number().int(),
    taxSheltersAverageAnnualTradeCount: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        commoditiesAverageAnnualTradeCount: "commodities_average_annual_trade_count",
        corporateBondsAverageAnnualTradeCount: "corporate_bonds_average_annual_trade_count",
        fundsAverageAnnualTradeCount: "funds_average_annual_trade_count",
        governmentBondsAverageAnnualTradeCount: "government_bonds_average_annual_trade_count",
        marginAverageAnnualTradeCount: "margin_average_annual_trade_count",
        municipalBondsAverageAnnualTradeCount: "municipal_bonds_average_annual_trade_count",
        optionsAverageAnnualTradeCount: "options_average_annual_trade_count",
        stocksAverageAnnualTradeCount: "stocks_average_annual_trade_count",
        taxSheltersAverageAnnualTradeCount: "tax_shelters_average_annual_trade_count",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InvestmentVehicleExperienceCreate$;
(function (InvestmentVehicleExperienceCreate$) {
    /** @deprecated use `InvestmentVehicleExperienceCreate$inboundSchema` instead. */
    InvestmentVehicleExperienceCreate$.inboundSchema = exports.InvestmentVehicleExperienceCreate$inboundSchema;
    /** @deprecated use `InvestmentVehicleExperienceCreate$outboundSchema` instead. */
    InvestmentVehicleExperienceCreate$.outboundSchema = exports.InvestmentVehicleExperienceCreate$outboundSchema;
})(InvestmentVehicleExperienceCreate$ || (exports.InvestmentVehicleExperienceCreate$ = InvestmentVehicleExperienceCreate$ = {}));
