// @apexfintechsolutions/ascend-sdk/models/components/irabeneficiaryenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IRABeneficiaryEnrollmentMetadataCreate$ = exports.IRABeneficiaryEnrollmentMetadataCreate$outboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreate$inboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$ = exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep = exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const datecreate_js_1 = require("./datecreate.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan) {
    IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan = IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep;
(function (IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep) {
    IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep || (exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep = IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$ = IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$;
(function (IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$ || (exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$ = IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.IRABeneficiaryEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
    inherited_from_owner_birth_date: datecreate_js_1.DateCreate$inboundSchema.optional(),
    inherited_from_owner_death_date: datecreate_js_1.DateCreate$inboundSchema.optional(),
    inherited_from_owner_name: z.string(),
    inheritor_is_decedents_spouse: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
        "inherited_from_owner_birth_date": "inheritedFromOwnerBirthDate",
        "inherited_from_owner_death_date": "inheritedFromOwnerDeathDate",
        "inherited_from_owner_name": "inheritedFromOwnerName",
        "inheritor_is_decedents_spouse": "inheritorIsDecedentsSpouse",
    });
});
/** @internal */
exports.IRABeneficiaryEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.IRABeneficiaryEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.IRABeneficiaryEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
    inheritedFromOwnerBirthDate: datecreate_js_1.DateCreate$outboundSchema.optional(),
    inheritedFromOwnerDeathDate: datecreate_js_1.DateCreate$outboundSchema.optional(),
    inheritedFromOwnerName: z.string(),
    inheritorIsDecedentsSpouse: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
        inheritedFromOwnerBirthDate: "inherited_from_owner_birth_date",
        inheritedFromOwnerDeathDate: "inherited_from_owner_death_date",
        inheritedFromOwnerName: "inherited_from_owner_name",
        inheritorIsDecedentsSpouse: "inheritor_is_decedents_spouse",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRABeneficiaryEnrollmentMetadataCreate$;
(function (IRABeneficiaryEnrollmentMetadataCreate$) {
    /** @deprecated use `IRABeneficiaryEnrollmentMetadataCreate$inboundSchema` instead. */
    IRABeneficiaryEnrollmentMetadataCreate$.inboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `IRABeneficiaryEnrollmentMetadataCreate$outboundSchema` instead. */
    IRABeneficiaryEnrollmentMetadataCreate$.outboundSchema = exports.IRABeneficiaryEnrollmentMetadataCreate$outboundSchema;
})(IRABeneficiaryEnrollmentMetadataCreate$ || (exports.IRABeneficiaryEnrollmentMetadataCreate$ = IRABeneficiaryEnrollmentMetadataCreate$ = {}));
