// @apexfintechsolutions/ascend-sdk/models/components/irarothenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IRARothEnrollmentMetadataCreate$ = exports.IRARothEnrollmentMetadataCreate$outboundSchema = exports.IRARothEnrollmentMetadataCreate$inboundSchema = exports.IRARothEnrollmentMetadataCreateFdicCashSweep$ = exports.IRARothEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.IRARothEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.IRARothEnrollmentMetadataCreateFdicCashSweep = exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var IRARothEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (IRARothEnrollmentMetadataCreateDividendReinvestmentPlan) {
    IRARothEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    IRARothEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    IRARothEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(IRARothEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan = IRARothEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var IRARothEnrollmentMetadataCreateFdicCashSweep;
(function (IRARothEnrollmentMetadataCreateFdicCashSweep) {
    IRARothEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    IRARothEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    IRARothEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(IRARothEnrollmentMetadataCreateFdicCashSweep || (exports.IRARothEnrollmentMetadataCreateFdicCashSweep = IRARothEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(IRARothEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(IRARothEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$ = IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.IRARothEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(IRARothEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRARothEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(IRARothEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRARothEnrollmentMetadataCreateFdicCashSweep$;
(function (IRARothEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `IRARothEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    IRARothEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.IRARothEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `IRARothEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    IRARothEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.IRARothEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(IRARothEnrollmentMetadataCreateFdicCashSweep$ || (exports.IRARothEnrollmentMetadataCreateFdicCashSweep$ = IRARothEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.IRARothEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.IRARothEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
    });
});
/** @internal */
exports.IRARothEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.IRARothEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.IRARothEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRARothEnrollmentMetadataCreate$;
(function (IRARothEnrollmentMetadataCreate$) {
    /** @deprecated use `IRARothEnrollmentMetadataCreate$inboundSchema` instead. */
    IRARothEnrollmentMetadataCreate$.inboundSchema = exports.IRARothEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `IRARothEnrollmentMetadataCreate$outboundSchema` instead. */
    IRARothEnrollmentMetadataCreate$.outboundSchema = exports.IRARothEnrollmentMetadataCreate$outboundSchema;
})(IRARothEnrollmentMetadataCreate$ || (exports.IRARothEnrollmentMetadataCreate$ = IRARothEnrollmentMetadataCreate$ = {}));
