// @apexfintechsolutions/ascend-sdk/models/components/irarolloverenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IRARolloverEnrollmentMetadataCreate$ = exports.IRARolloverEnrollmentMetadataCreate$outboundSchema = exports.IRARolloverEnrollmentMetadataCreate$inboundSchema = exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep$ = exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep = exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan) {
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan = IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var IRARolloverEnrollmentMetadataCreateFdicCashSweep;
(function (IRARolloverEnrollmentMetadataCreateFdicCashSweep) {
    IRARolloverEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    IRARolloverEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    IRARolloverEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(IRARolloverEnrollmentMetadataCreateFdicCashSweep || (exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep = IRARolloverEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$ = IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(IRARolloverEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(IRARolloverEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRARolloverEnrollmentMetadataCreateFdicCashSweep$;
(function (IRARolloverEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `IRARolloverEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    IRARolloverEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `IRARolloverEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    IRARolloverEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(IRARolloverEnrollmentMetadataCreateFdicCashSweep$ || (exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep$ = IRARolloverEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.IRARolloverEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
    });
});
/** @internal */
exports.IRARolloverEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.IRARolloverEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.IRARolloverEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRARolloverEnrollmentMetadataCreate$;
(function (IRARolloverEnrollmentMetadataCreate$) {
    /** @deprecated use `IRARolloverEnrollmentMetadataCreate$inboundSchema` instead. */
    IRARolloverEnrollmentMetadataCreate$.inboundSchema = exports.IRARolloverEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `IRARolloverEnrollmentMetadataCreate$outboundSchema` instead. */
    IRARolloverEnrollmentMetadataCreate$.outboundSchema = exports.IRARolloverEnrollmentMetadataCreate$outboundSchema;
})(IRARolloverEnrollmentMetadataCreate$ || (exports.IRARolloverEnrollmentMetadataCreate$ = IRARolloverEnrollmentMetadataCreate$ = {}));
