// @apexfintechsolutions/ascend-sdk/models/components/irasepenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IRASEPEnrollmentMetadataCreate$ = exports.IRASEPEnrollmentMetadataCreate$outboundSchema = exports.IRASEPEnrollmentMetadataCreate$inboundSchema = exports.IRASEPEnrollmentMetadataCreateFdicCashSweep$ = exports.IRASEPEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.IRASEPEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.IRASEPEnrollmentMetadataCreateFdicCashSweep = exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan) {
    IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan = IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var IRASEPEnrollmentMetadataCreateFdicCashSweep;
(function (IRASEPEnrollmentMetadataCreateFdicCashSweep) {
    IRASEPEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    IRASEPEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    IRASEPEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(IRASEPEnrollmentMetadataCreateFdicCashSweep || (exports.IRASEPEnrollmentMetadataCreateFdicCashSweep = IRASEPEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$ = IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.IRASEPEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(IRASEPEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRASEPEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(IRASEPEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRASEPEnrollmentMetadataCreateFdicCashSweep$;
(function (IRASEPEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `IRASEPEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    IRASEPEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.IRASEPEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `IRASEPEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    IRASEPEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.IRASEPEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(IRASEPEnrollmentMetadataCreateFdicCashSweep$ || (exports.IRASEPEnrollmentMetadataCreateFdicCashSweep$ = IRASEPEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.IRASEPEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.IRASEPEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
    });
});
/** @internal */
exports.IRASEPEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.IRASEPEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.IRASEPEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRASEPEnrollmentMetadataCreate$;
(function (IRASEPEnrollmentMetadataCreate$) {
    /** @deprecated use `IRASEPEnrollmentMetadataCreate$inboundSchema` instead. */
    IRASEPEnrollmentMetadataCreate$.inboundSchema = exports.IRASEPEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `IRASEPEnrollmentMetadataCreate$outboundSchema` instead. */
    IRASEPEnrollmentMetadataCreate$.outboundSchema = exports.IRASEPEnrollmentMetadataCreate$outboundSchema;
})(IRASEPEnrollmentMetadataCreate$ || (exports.IRASEPEnrollmentMetadataCreate$ = IRASEPEnrollmentMetadataCreate$ = {}));
