// @apexfintechsolutions/ascend-sdk/models/components/irasimpleenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IRASimpleEnrollmentMetadataCreate$ = exports.IRASimpleEnrollmentMetadataCreate$outboundSchema = exports.IRASimpleEnrollmentMetadataCreate$inboundSchema = exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep$ = exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep = exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan) {
    IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan = IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var IRASimpleEnrollmentMetadataCreateFdicCashSweep;
(function (IRASimpleEnrollmentMetadataCreateFdicCashSweep) {
    IRASimpleEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    IRASimpleEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    IRASimpleEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(IRASimpleEnrollmentMetadataCreateFdicCashSweep || (exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep = IRASimpleEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$ = IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(IRASimpleEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(IRASimpleEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRASimpleEnrollmentMetadataCreateFdicCashSweep$;
(function (IRASimpleEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `IRASimpleEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    IRASimpleEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `IRASimpleEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    IRASimpleEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(IRASimpleEnrollmentMetadataCreateFdicCashSweep$ || (exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep$ = IRASimpleEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.IRASimpleEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
    });
});
/** @internal */
exports.IRASimpleEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.IRASimpleEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.IRASimpleEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRASimpleEnrollmentMetadataCreate$;
(function (IRASimpleEnrollmentMetadataCreate$) {
    /** @deprecated use `IRASimpleEnrollmentMetadataCreate$inboundSchema` instead. */
    IRASimpleEnrollmentMetadataCreate$.inboundSchema = exports.IRASimpleEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `IRASimpleEnrollmentMetadataCreate$outboundSchema` instead. */
    IRASimpleEnrollmentMetadataCreate$.outboundSchema = exports.IRASimpleEnrollmentMetadataCreate$outboundSchema;
})(IRASimpleEnrollmentMetadataCreate$ || (exports.IRASimpleEnrollmentMetadataCreate$ = IRASimpleEnrollmentMetadataCreate$ = {}));
