// @apexfintechsolutions/ascend-sdk/models/components/iratraditionalenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IRATraditionalEnrollmentMetadataCreate$ = exports.IRATraditionalEnrollmentMetadataCreate$outboundSchema = exports.IRATraditionalEnrollmentMetadataCreate$inboundSchema = exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep$ = exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep = exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan) {
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan = IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var IRATraditionalEnrollmentMetadataCreateFdicCashSweep;
(function (IRATraditionalEnrollmentMetadataCreateFdicCashSweep) {
    IRATraditionalEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    IRATraditionalEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    IRATraditionalEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(IRATraditionalEnrollmentMetadataCreateFdicCashSweep || (exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep = IRATraditionalEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$ = IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(IRATraditionalEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(IRATraditionalEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRATraditionalEnrollmentMetadataCreateFdicCashSweep$;
(function (IRATraditionalEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `IRATraditionalEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    IRATraditionalEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `IRATraditionalEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    IRATraditionalEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(IRATraditionalEnrollmentMetadataCreateFdicCashSweep$ || (exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep$ = IRATraditionalEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.IRATraditionalEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
    });
});
/** @internal */
exports.IRATraditionalEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.IRATraditionalEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.IRATraditionalEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IRATraditionalEnrollmentMetadataCreate$;
(function (IRATraditionalEnrollmentMetadataCreate$) {
    /** @deprecated use `IRATraditionalEnrollmentMetadataCreate$inboundSchema` instead. */
    IRATraditionalEnrollmentMetadataCreate$.inboundSchema = exports.IRATraditionalEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `IRATraditionalEnrollmentMetadataCreate$outboundSchema` instead. */
    IRATraditionalEnrollmentMetadataCreate$.outboundSchema = exports.IRATraditionalEnrollmentMetadataCreate$outboundSchema;
})(IRATraditionalEnrollmentMetadataCreate$ || (exports.IRATraditionalEnrollmentMetadataCreate$ = IRATraditionalEnrollmentMetadataCreate$ = {}));
