// @apexfintechsolutions/ascend-sdk/models/components/jointwithrightsofsurvivorshipenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$ = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$outboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$inboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$ = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan) {
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan = JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep;
(function (JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep) {
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep || (exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep = JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$ = JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$;
(function (JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$ || (exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$ = JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
    });
});
/** @internal */
exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$;
(function (JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$) {
    /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$inboundSchema` instead. */
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$.inboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$outboundSchema` instead. */
    JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$.outboundSchema = exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$outboundSchema;
})(JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$ || (exports.JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$ = JointWithRightsOfSurvivorshipEnrollmentMetadataCreate$ = {}));
