// @apexfintechsolutions/ascend-sdk/models/components/jointtenantsbyentiretyenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.JointTenantsByEntiretyEnrollmentMetadataCreate$ = exports.JointTenantsByEntiretyEnrollmentMetadataCreate$outboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreate$inboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$ = exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$outboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$inboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$ = exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple = exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep = exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan) {
    JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan = JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep;
(function (JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep) {
    JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep || (exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep = JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep = {}));
/**
 * The legal residency state of a married couple
 */
var JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple;
(function (JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple) {
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["LegalResidencyStateOfMarriedCoupleUnspecified"] = "LEGAL_RESIDENCY_STATE_OF_MARRIED_COUPLE_UNSPECIFIED";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Ak"] = "AK";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Az"] = "AZ";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Ar"] = "AR";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Ca"] = "CA";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["De"] = "DE";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Fl"] = "FL";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Hi"] = "HI";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Id"] = "ID";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Il"] = "IL";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["In"] = "IN";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Ky"] = "KY";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["La"] = "LA";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Md"] = "MD";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Ma"] = "MA";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Mi"] = "MI";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Ms"] = "MS";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Mo"] = "MO";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Nv"] = "NV";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Nj"] = "NJ";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Nm"] = "NM";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Ny"] = "NY";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Nc"] = "NC";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Oh"] = "OH";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Ok"] = "OK";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Or"] = "OR";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Pa"] = "PA";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Ri"] = "RI";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Tn"] = "TN";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Tx"] = "TX";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Vt"] = "VT";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Va"] = "VA";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Wa"] = "WA";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Wi"] = "WI";
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple["Wy"] = "WY";
})(JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple || (exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple = JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple = {}));
/** @internal */
exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$ = JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$;
(function (JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$ || (exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$ = JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$inboundSchema = z
    .union([
    z.nativeEnum(JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$outboundSchema = z.union([
    z.nativeEnum(JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$;
(function (JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$) {
    /** @deprecated use `JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$inboundSchema` instead. */
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$.inboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$inboundSchema;
    /** @deprecated use `JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$outboundSchema` instead. */
    JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$.outboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$outboundSchema;
})(JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$ || (exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$ = JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$ = {}));
/** @internal */
exports.JointTenantsByEntiretyEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
    legal_residency_state_of_married_couple: exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
        "legal_residency_state_of_married_couple": "legalResidencyStateOfMarriedCouple",
    });
});
/** @internal */
exports.JointTenantsByEntiretyEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.JointTenantsByEntiretyEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.JointTenantsByEntiretyEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
    legalResidencyStateOfMarriedCouple: exports.JointTenantsByEntiretyEnrollmentMetadataCreateLegalResidencyStateOfMarriedCouple$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
        legalResidencyStateOfMarriedCouple: "legal_residency_state_of_married_couple",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointTenantsByEntiretyEnrollmentMetadataCreate$;
(function (JointTenantsByEntiretyEnrollmentMetadataCreate$) {
    /** @deprecated use `JointTenantsByEntiretyEnrollmentMetadataCreate$inboundSchema` instead. */
    JointTenantsByEntiretyEnrollmentMetadataCreate$.inboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `JointTenantsByEntiretyEnrollmentMetadataCreate$outboundSchema` instead. */
    JointTenantsByEntiretyEnrollmentMetadataCreate$.outboundSchema = exports.JointTenantsByEntiretyEnrollmentMetadataCreate$outboundSchema;
})(JointTenantsByEntiretyEnrollmentMetadataCreate$ || (exports.JointTenantsByEntiretyEnrollmentMetadataCreate$ = JointTenantsByEntiretyEnrollmentMetadataCreate$ = {}));
