// @apexfintechsolutions/ascend-sdk/models/components/jointtenantsincommonenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.JointTenantsInCommonEnrollmentMetadataCreate$ = exports.JointTenantsInCommonEnrollmentMetadataCreate$outboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreate$inboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$ = exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep = exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan) {
    JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan = JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep;
(function (JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep) {
    JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep || (exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep = JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$ = JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$;
(function (JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$ || (exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$ = JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.JointTenantsInCommonEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
    });
});
/** @internal */
exports.JointTenantsInCommonEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.JointTenantsInCommonEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.JointTenantsInCommonEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointTenantsInCommonEnrollmentMetadataCreate$;
(function (JointTenantsInCommonEnrollmentMetadataCreate$) {
    /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreate$inboundSchema` instead. */
    JointTenantsInCommonEnrollmentMetadataCreate$.inboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `JointTenantsInCommonEnrollmentMetadataCreate$outboundSchema` instead. */
    JointTenantsInCommonEnrollmentMetadataCreate$.outboundSchema = exports.JointTenantsInCommonEnrollmentMetadataCreate$outboundSchema;
})(JointTenantsInCommonEnrollmentMetadataCreate$ || (exports.JointTenantsInCommonEnrollmentMetadataCreate$ = JointTenantsInCommonEnrollmentMetadataCreate$ = {}));
