// @apexfintechsolutions/ascend-sdk/models/components/jointcommunitypropertyenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.JointCommunityPropertyEnrollmentMetadataCreate$ = exports.JointCommunityPropertyEnrollmentMetadataCreate$outboundSchema = exports.JointCommunityPropertyEnrollmentMetadataCreate$inboundSchema = exports.LegalResidencyStateOfMarriedCouple$ = exports.LegalResidencyStateOfMarriedCouple$outboundSchema = exports.LegalResidencyStateOfMarriedCouple$inboundSchema = exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$ = exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.LegalResidencyStateOfMarriedCouple = exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep = exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan) {
    JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan = JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep;
(function (JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep) {
    JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep || (exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep = JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep = {}));
/**
 * The legal residency state of a married couple
 */
var LegalResidencyStateOfMarriedCouple;
(function (LegalResidencyStateOfMarriedCouple) {
    LegalResidencyStateOfMarriedCouple["LegalResidencyStateOfMarriedCoupleUnspecified"] = "LEGAL_RESIDENCY_STATE_OF_MARRIED_COUPLE_UNSPECIFIED";
    LegalResidencyStateOfMarriedCouple["Ak"] = "AK";
    LegalResidencyStateOfMarriedCouple["Az"] = "AZ";
    LegalResidencyStateOfMarriedCouple["Ar"] = "AR";
    LegalResidencyStateOfMarriedCouple["Ca"] = "CA";
    LegalResidencyStateOfMarriedCouple["De"] = "DE";
    LegalResidencyStateOfMarriedCouple["Fl"] = "FL";
    LegalResidencyStateOfMarriedCouple["Hi"] = "HI";
    LegalResidencyStateOfMarriedCouple["Id"] = "ID";
    LegalResidencyStateOfMarriedCouple["Il"] = "IL";
    LegalResidencyStateOfMarriedCouple["In"] = "IN";
    LegalResidencyStateOfMarriedCouple["Ky"] = "KY";
    LegalResidencyStateOfMarriedCouple["La"] = "LA";
    LegalResidencyStateOfMarriedCouple["Md"] = "MD";
    LegalResidencyStateOfMarriedCouple["Ma"] = "MA";
    LegalResidencyStateOfMarriedCouple["Mi"] = "MI";
    LegalResidencyStateOfMarriedCouple["Ms"] = "MS";
    LegalResidencyStateOfMarriedCouple["Mo"] = "MO";
    LegalResidencyStateOfMarriedCouple["Nv"] = "NV";
    LegalResidencyStateOfMarriedCouple["Nj"] = "NJ";
    LegalResidencyStateOfMarriedCouple["Nm"] = "NM";
    LegalResidencyStateOfMarriedCouple["Ny"] = "NY";
    LegalResidencyStateOfMarriedCouple["Nc"] = "NC";
    LegalResidencyStateOfMarriedCouple["Oh"] = "OH";
    LegalResidencyStateOfMarriedCouple["Ok"] = "OK";
    LegalResidencyStateOfMarriedCouple["Or"] = "OR";
    LegalResidencyStateOfMarriedCouple["Pa"] = "PA";
    LegalResidencyStateOfMarriedCouple["Ri"] = "RI";
    LegalResidencyStateOfMarriedCouple["Tn"] = "TN";
    LegalResidencyStateOfMarriedCouple["Tx"] = "TX";
    LegalResidencyStateOfMarriedCouple["Vt"] = "VT";
    LegalResidencyStateOfMarriedCouple["Va"] = "VA";
    LegalResidencyStateOfMarriedCouple["Wa"] = "WA";
    LegalResidencyStateOfMarriedCouple["Wi"] = "WI";
    LegalResidencyStateOfMarriedCouple["Wy"] = "WY";
})(LegalResidencyStateOfMarriedCouple || (exports.LegalResidencyStateOfMarriedCouple = LegalResidencyStateOfMarriedCouple = {}));
/** @internal */
exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$ = JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$;
(function (JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$ || (exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$ = JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.LegalResidencyStateOfMarriedCouple$inboundSchema = z
    .union([
    z.nativeEnum(LegalResidencyStateOfMarriedCouple),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalResidencyStateOfMarriedCouple$outboundSchema = z.union([
    z.nativeEnum(LegalResidencyStateOfMarriedCouple),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalResidencyStateOfMarriedCouple$;
(function (LegalResidencyStateOfMarriedCouple$) {
    /** @deprecated use `LegalResidencyStateOfMarriedCouple$inboundSchema` instead. */
    LegalResidencyStateOfMarriedCouple$.inboundSchema = exports.LegalResidencyStateOfMarriedCouple$inboundSchema;
    /** @deprecated use `LegalResidencyStateOfMarriedCouple$outboundSchema` instead. */
    LegalResidencyStateOfMarriedCouple$.outboundSchema = exports.LegalResidencyStateOfMarriedCouple$outboundSchema;
})(LegalResidencyStateOfMarriedCouple$ || (exports.LegalResidencyStateOfMarriedCouple$ = LegalResidencyStateOfMarriedCouple$ = {}));
/** @internal */
exports.JointCommunityPropertyEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
    legal_residency_state_of_married_couple: exports.LegalResidencyStateOfMarriedCouple$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
        "legal_residency_state_of_married_couple": "legalResidencyStateOfMarriedCouple",
    });
});
/** @internal */
exports.JointCommunityPropertyEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.JointCommunityPropertyEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.JointCommunityPropertyEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
    legalResidencyStateOfMarriedCouple: exports.LegalResidencyStateOfMarriedCouple$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
        legalResidencyStateOfMarriedCouple: "legal_residency_state_of_married_couple",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JointCommunityPropertyEnrollmentMetadataCreate$;
(function (JointCommunityPropertyEnrollmentMetadataCreate$) {
    /** @deprecated use `JointCommunityPropertyEnrollmentMetadataCreate$inboundSchema` instead. */
    JointCommunityPropertyEnrollmentMetadataCreate$.inboundSchema = exports.JointCommunityPropertyEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `JointCommunityPropertyEnrollmentMetadataCreate$outboundSchema` instead. */
    JointCommunityPropertyEnrollmentMetadataCreate$.outboundSchema = exports.JointCommunityPropertyEnrollmentMetadataCreate$outboundSchema;
})(JointCommunityPropertyEnrollmentMetadataCreate$ || (exports.JointCommunityPropertyEnrollmentMetadataCreate$ = JointCommunityPropertyEnrollmentMetadataCreate$ = {}));
