// @apexfintechsolutions/ascend-sdk/models/components/legalnaturalpersoncreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalNaturalPersonCreate$ = exports.LegalNaturalPersonCreate$outboundSchema = exports.LegalNaturalPersonCreate$inboundSchema = exports.TaxIdType$ = exports.TaxIdType$outboundSchema = exports.TaxIdType$inboundSchema = exports.NameSuffix$ = exports.NameSuffix$outboundSchema = exports.NameSuffix$inboundSchema = exports.MaritalStatus$ = exports.MaritalStatus$outboundSchema = exports.MaritalStatus$inboundSchema = exports.TaxIdType = exports.NameSuffix = exports.MaritalStatus = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const datecreate_js_1 = require("./datecreate.js");
const employmentcreate_js_1 = require("./employmentcreate.js");
const foreignidentificationcreate_js_1 = require("./foreignidentificationcreate.js");
const identityverificationresultcreate_js_1 = require("./identityverificationresultcreate.js");
const largetradercreate_js_1 = require("./largetradercreate.js");
const naturalpersonfddcreate_js_1 = require("./naturalpersonfddcreate.js");
const noncitizenresidencycreate_js_1 = require("./noncitizenresidencycreate.js");
const postaladdresscreate_js_1 = require("./postaladdresscreate.js");
const taxprofilecreate_js_1 = require("./taxprofilecreate.js");
/**
 * The legal marital status of an account-holder; Used in combination with state of domicile to determine qualification for account types and beneficiary exclusion rules.
 */
var MaritalStatus;
(function (MaritalStatus) {
    MaritalStatus["MaritalStatusUnspecified"] = "MARITAL_STATUS_UNSPECIFIED";
    MaritalStatus["Single"] = "SINGLE";
    MaritalStatus["Married"] = "MARRIED";
    MaritalStatus["Divorced"] = "DIVORCED";
    MaritalStatus["Widowed"] = "WIDOWED";
})(MaritalStatus || (exports.MaritalStatus = MaritalStatus = {}));
/**
 * The suffix of a natural person; A suffix in a name is any part of the name that comes after the last name
 */
var NameSuffix;
(function (NameSuffix) {
    NameSuffix["NameSuffixUnspecified"] = "NAME_SUFFIX_UNSPECIFIED";
    NameSuffix["Sr"] = "SR";
    NameSuffix["Jr"] = "JR";
    NameSuffix["Iii"] = "III";
    NameSuffix["Iv"] = "IV";
    NameSuffix["V"] = "V";
})(NameSuffix || (exports.NameSuffix = NameSuffix = {}));
/**
 * The nature of the U.S. Tax ID indicated in the related tax_id field; Examples include ITIN, SSN, EIN.
 */
var TaxIdType;
(function (TaxIdType) {
    TaxIdType["TaxIdTypeUnspecified"] = "TAX_ID_TYPE_UNSPECIFIED";
    TaxIdType["TaxIdTypeSsn"] = "TAX_ID_TYPE_SSN";
    TaxIdType["TaxIdTypeItin"] = "TAX_ID_TYPE_ITIN";
    TaxIdType["TaxIdTypeEin"] = "TAX_ID_TYPE_EIN";
})(TaxIdType || (exports.TaxIdType = TaxIdType = {}));
/** @internal */
exports.MaritalStatus$inboundSchema = z
    .union([
    z.nativeEnum(MaritalStatus),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.MaritalStatus$outboundSchema = z.union([
    z.nativeEnum(MaritalStatus),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MaritalStatus$;
(function (MaritalStatus$) {
    /** @deprecated use `MaritalStatus$inboundSchema` instead. */
    MaritalStatus$.inboundSchema = exports.MaritalStatus$inboundSchema;
    /** @deprecated use `MaritalStatus$outboundSchema` instead. */
    MaritalStatus$.outboundSchema = exports.MaritalStatus$outboundSchema;
})(MaritalStatus$ || (exports.MaritalStatus$ = MaritalStatus$ = {}));
/** @internal */
exports.NameSuffix$inboundSchema = z
    .union([
    z.nativeEnum(NameSuffix),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.NameSuffix$outboundSchema = z.union([
    z.nativeEnum(NameSuffix),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NameSuffix$;
(function (NameSuffix$) {
    /** @deprecated use `NameSuffix$inboundSchema` instead. */
    NameSuffix$.inboundSchema = exports.NameSuffix$inboundSchema;
    /** @deprecated use `NameSuffix$outboundSchema` instead. */
    NameSuffix$.outboundSchema = exports.NameSuffix$outboundSchema;
})(NameSuffix$ || (exports.NameSuffix$ = NameSuffix$ = {}));
/** @internal */
exports.TaxIdType$inboundSchema = z
    .union([
    z.nativeEnum(TaxIdType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TaxIdType$outboundSchema = z.union([
    z.nativeEnum(TaxIdType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TaxIdType$;
(function (TaxIdType$) {
    /** @deprecated use `TaxIdType$inboundSchema` instead. */
    TaxIdType$.inboundSchema = exports.TaxIdType$inboundSchema;
    /** @deprecated use `TaxIdType$outboundSchema` instead. */
    TaxIdType$.outboundSchema = exports.TaxIdType$outboundSchema;
})(TaxIdType$ || (exports.TaxIdType$ = TaxIdType$ = {}));
/** @internal */
exports.LegalNaturalPersonCreate$inboundSchema = z.object({
    accredited_investor: z.boolean().optional(),
    adviser: z.boolean().optional(),
    birth_date: datecreate_js_1.DateCreate$inboundSchema,
    citizenship_countries: z.array(z.string()),
    control_person_company_symbols: z.string().optional(),
    correspondent_employee: z.boolean().optional(),
    correspondent_id: z.string(),
    death_date: datecreate_js_1.DateCreate$inboundSchema.optional(),
    doing_business_as: z.array(z.string()).optional(),
    employment: employmentcreate_js_1.EmploymentCreate$inboundSchema,
    family_name: z.string(),
    finra_associated_entity: z.string().optional(),
    foreign_identification: foreignidentificationcreate_js_1.ForeignIdentificationCreate$inboundSchema.optional(),
    given_name: z.string(),
    identity_verification_result: identityverificationresultcreate_js_1.IdentityVerificationResultCreate$inboundSchema
        .optional(),
    institutional_customer: z.boolean().optional(),
    large_trader: largetradercreate_js_1.LargeTraderCreate$inboundSchema.optional(),
    marital_status: exports.MaritalStatus$inboundSchema.optional(),
    middle_names: z.string().optional(),
    name_suffix: exports.NameSuffix$inboundSchema.optional(),
    natural_person_fdd: naturalpersonfddcreate_js_1.NaturalPersonFddCreate$inboundSchema.optional(),
    non_citizen_residency: noncitizenresidencycreate_js_1.NonCitizenResidencyCreate$inboundSchema.optional(),
    personal_address: postaladdresscreate_js_1.PostalAddressCreate$inboundSchema,
    politically_exposed_immediate_family_names: z.array(z.string()).optional(),
    politically_exposed_organization: z.string().optional(),
    subject_to_backup_withholding: z.boolean().optional(),
    tax_id: z.string().optional(),
    tax_id_type: exports.TaxIdType$inboundSchema.optional(),
    tax_profile: taxprofilecreate_js_1.TaxProfileCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "accredited_investor": "accreditedInvestor",
        "birth_date": "birthDate",
        "citizenship_countries": "citizenshipCountries",
        "control_person_company_symbols": "controlPersonCompanySymbols",
        "correspondent_employee": "correspondentEmployee",
        "correspondent_id": "correspondentId",
        "death_date": "deathDate",
        "doing_business_as": "doingBusinessAs",
        "family_name": "familyName",
        "finra_associated_entity": "finraAssociatedEntity",
        "foreign_identification": "foreignIdentification",
        "given_name": "givenName",
        "identity_verification_result": "identityVerificationResult",
        "institutional_customer": "institutionalCustomer",
        "large_trader": "largeTrader",
        "marital_status": "maritalStatus",
        "middle_names": "middleNames",
        "name_suffix": "nameSuffix",
        "natural_person_fdd": "naturalPersonFdd",
        "non_citizen_residency": "nonCitizenResidency",
        "personal_address": "personalAddress",
        "politically_exposed_immediate_family_names": "politicallyExposedImmediateFamilyNames",
        "politically_exposed_organization": "politicallyExposedOrganization",
        "subject_to_backup_withholding": "subjectToBackupWithholding",
        "tax_id": "taxId",
        "tax_id_type": "taxIdType",
        "tax_profile": "taxProfile",
    });
});
/** @internal */
exports.LegalNaturalPersonCreate$outboundSchema = z.object({
    accreditedInvestor: z.boolean().optional(),
    adviser: z.boolean().optional(),
    birthDate: datecreate_js_1.DateCreate$outboundSchema,
    citizenshipCountries: z.array(z.string()),
    controlPersonCompanySymbols: z.string().optional(),
    correspondentEmployee: z.boolean().optional(),
    correspondentId: z.string(),
    deathDate: datecreate_js_1.DateCreate$outboundSchema.optional(),
    doingBusinessAs: z.array(z.string()).optional(),
    employment: employmentcreate_js_1.EmploymentCreate$outboundSchema,
    familyName: z.string(),
    finraAssociatedEntity: z.string().optional(),
    foreignIdentification: foreignidentificationcreate_js_1.ForeignIdentificationCreate$outboundSchema.optional(),
    givenName: z.string(),
    identityVerificationResult: identityverificationresultcreate_js_1.IdentityVerificationResultCreate$outboundSchema
        .optional(),
    institutionalCustomer: z.boolean().optional(),
    largeTrader: largetradercreate_js_1.LargeTraderCreate$outboundSchema.optional(),
    maritalStatus: exports.MaritalStatus$outboundSchema.optional(),
    middleNames: z.string().optional(),
    nameSuffix: exports.NameSuffix$outboundSchema.optional(),
    naturalPersonFdd: naturalpersonfddcreate_js_1.NaturalPersonFddCreate$outboundSchema.optional(),
    nonCitizenResidency: noncitizenresidencycreate_js_1.NonCitizenResidencyCreate$outboundSchema.optional(),
    personalAddress: postaladdresscreate_js_1.PostalAddressCreate$outboundSchema,
    politicallyExposedImmediateFamilyNames: z.array(z.string()).optional(),
    politicallyExposedOrganization: z.string().optional(),
    subjectToBackupWithholding: z.boolean().optional(),
    taxId: z.string().optional(),
    taxIdType: exports.TaxIdType$outboundSchema.optional(),
    taxProfile: taxprofilecreate_js_1.TaxProfileCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accreditedInvestor: "accredited_investor",
        birthDate: "birth_date",
        citizenshipCountries: "citizenship_countries",
        controlPersonCompanySymbols: "control_person_company_symbols",
        correspondentEmployee: "correspondent_employee",
        correspondentId: "correspondent_id",
        deathDate: "death_date",
        doingBusinessAs: "doing_business_as",
        familyName: "family_name",
        finraAssociatedEntity: "finra_associated_entity",
        foreignIdentification: "foreign_identification",
        givenName: "given_name",
        identityVerificationResult: "identity_verification_result",
        institutionalCustomer: "institutional_customer",
        largeTrader: "large_trader",
        maritalStatus: "marital_status",
        middleNames: "middle_names",
        nameSuffix: "name_suffix",
        naturalPersonFdd: "natural_person_fdd",
        nonCitizenResidency: "non_citizen_residency",
        personalAddress: "personal_address",
        politicallyExposedImmediateFamilyNames: "politically_exposed_immediate_family_names",
        politicallyExposedOrganization: "politically_exposed_organization",
        subjectToBackupWithholding: "subject_to_backup_withholding",
        taxId: "tax_id",
        taxIdType: "tax_id_type",
        taxProfile: "tax_profile",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonCreate$;
(function (LegalNaturalPersonCreate$) {
    /** @deprecated use `LegalNaturalPersonCreate$inboundSchema` instead. */
    LegalNaturalPersonCreate$.inboundSchema = exports.LegalNaturalPersonCreate$inboundSchema;
    /** @deprecated use `LegalNaturalPersonCreate$outboundSchema` instead. */
    LegalNaturalPersonCreate$.outboundSchema = exports.LegalNaturalPersonCreate$outboundSchema;
})(LegalNaturalPersonCreate$ || (exports.LegalNaturalPersonCreate$ = LegalNaturalPersonCreate$ = {}));
