// @apexfintechsolutions/ascend-sdk/models/components/legalagreement.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalAgreement$ = exports.LegalAgreement$outboundSchema = exports.LegalAgreement$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.LegalAgreement$inboundSchema = z.object({
    agreement_name: z.string().optional(),
    agreement_uri: z.string().optional(),
    agreement_version: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "agreement_name": "agreementName",
        "agreement_uri": "agreementUri",
        "agreement_version": "agreementVersion",
    });
});
/** @internal */
exports.LegalAgreement$outboundSchema = z.object({
    agreementName: z.string().optional(),
    agreementUri: z.string().optional(),
    agreementVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        agreementName: "agreement_name",
        agreementUri: "agreement_uri",
        agreementVersion: "agreement_version",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalAgreement$;
(function (LegalAgreement$) {
    /** @deprecated use `LegalAgreement$inboundSchema` instead. */
    LegalAgreement$.inboundSchema = exports.LegalAgreement$inboundSchema;
    /** @deprecated use `LegalAgreement$outboundSchema` instead. */
    LegalAgreement$.outboundSchema = exports.LegalAgreement$outboundSchema;
})(LegalAgreement$ || (exports.LegalAgreement$ = LegalAgreement$ = {}));
