// @apexfintechsolutions/ascend-sdk/models/components/legalnaturalperson.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalNaturalPersonLargeTrader$outboundSchema = exports.LegalNaturalPersonLargeTrader$inboundSchema = exports.LegalNaturalPersonEffectiveDate$ = exports.LegalNaturalPersonEffectiveDate$outboundSchema = exports.LegalNaturalPersonEffectiveDate$inboundSchema = exports.LegalNaturalPersonIdentityVerificationResult$ = exports.LegalNaturalPersonIdentityVerificationResult$outboundSchema = exports.LegalNaturalPersonIdentityVerificationResult$inboundSchema = exports.ExecutionDate$ = exports.ExecutionDate$outboundSchema = exports.ExecutionDate$inboundSchema = exports.ForeignIdentification$ = exports.ForeignIdentification$outboundSchema = exports.ForeignIdentification$inboundSchema = exports.LegalNaturalPersonType$ = exports.LegalNaturalPersonType$outboundSchema = exports.LegalNaturalPersonType$inboundSchema = exports.IssueDate$ = exports.IssueDate$outboundSchema = exports.IssueDate$inboundSchema = exports.ExpirationDate$ = exports.ExpirationDate$outboundSchema = exports.ExpirationDate$inboundSchema = exports.Employment$ = exports.Employment$outboundSchema = exports.Employment$inboundSchema = exports.LegalNaturalPersonEmploymentStatus$ = exports.LegalNaturalPersonEmploymentStatus$outboundSchema = exports.LegalNaturalPersonEmploymentStatus$inboundSchema = exports.EmployerAddress$ = exports.EmployerAddress$outboundSchema = exports.EmployerAddress$inboundSchema = exports.DeathDate$ = exports.DeathDate$outboundSchema = exports.DeathDate$inboundSchema = exports.BirthDate$ = exports.BirthDate$outboundSchema = exports.BirthDate$inboundSchema = exports.LegalNaturalPersonWithholdingState = exports.LegalNaturalPersonUsTinStatus = exports.TaxpayerCertificationState = exports.ReportingEligibility = exports.LegalNaturalPersonIrsFormType = exports.LegalNaturalPersonFederalTaxClassification = exports.LegalNaturalPersonTaxIdType = exports.LegalNaturalPersonResidencyStatus = exports.LegalNaturalPersonNameSuffix = exports.LegalNaturalPersonMaritalStatus = exports.LegalNaturalPersonType = exports.LegalNaturalPersonEmploymentStatus = void 0;
exports.TaxpayerCertificationState$inboundSchema = exports.TaxCertificationDate$ = exports.TaxCertificationDate$outboundSchema = exports.TaxCertificationDate$inboundSchema = exports.ReportingEligibility$ = exports.ReportingEligibility$outboundSchema = exports.ReportingEligibility$inboundSchema = exports.LegalNaturalPersonIrsFormType$ = exports.LegalNaturalPersonIrsFormType$outboundSchema = exports.LegalNaturalPersonIrsFormType$inboundSchema = exports.FirstBNoticeDate$ = exports.FirstBNoticeDate$outboundSchema = exports.FirstBNoticeDate$inboundSchema = exports.LegalNaturalPersonFederalTaxClassification$ = exports.LegalNaturalPersonFederalTaxClassification$outboundSchema = exports.LegalNaturalPersonFederalTaxClassification$inboundSchema = exports.CNoticeDate$ = exports.CNoticeDate$outboundSchema = exports.CNoticeDate$inboundSchema = exports.LegalNaturalPersonTaxIdType$ = exports.LegalNaturalPersonTaxIdType$outboundSchema = exports.LegalNaturalPersonTaxIdType$inboundSchema = exports.PersonalAddress$ = exports.PersonalAddress$outboundSchema = exports.PersonalAddress$inboundSchema = exports.NonCitizenResidency$ = exports.NonCitizenResidency$outboundSchema = exports.NonCitizenResidency$inboundSchema = exports.LegalNaturalPersonResidencyStatus$ = exports.LegalNaturalPersonResidencyStatus$outboundSchema = exports.LegalNaturalPersonResidencyStatus$inboundSchema = exports.NaturalPersonFdd$ = exports.NaturalPersonFdd$outboundSchema = exports.NaturalPersonFdd$inboundSchema = exports.OtherSourcesOfWealth$ = exports.OtherSourcesOfWealth$outboundSchema = exports.OtherSourcesOfWealth$inboundSchema = exports.NegativeNews$ = exports.NegativeNews$outboundSchema = exports.NegativeNews$inboundSchema = exports.CustomerReferralSource$ = exports.CustomerReferralSource$outboundSchema = exports.CustomerReferralSource$inboundSchema = exports.LegalNaturalPersonNameSuffix$ = exports.LegalNaturalPersonNameSuffix$outboundSchema = exports.LegalNaturalPersonNameSuffix$inboundSchema = exports.LegalNaturalPersonMaritalStatus$ = exports.LegalNaturalPersonMaritalStatus$outboundSchema = exports.LegalNaturalPersonMaritalStatus$inboundSchema = exports.LegalNaturalPersonLargeTrader$ = void 0;
exports.LegalNaturalPerson$ = exports.LegalNaturalPerson$outboundSchema = exports.LegalNaturalPerson$inboundSchema = exports.TaxProfile$ = exports.TaxProfile$outboundSchema = exports.TaxProfile$inboundSchema = exports.LegalNaturalPersonWithholdingState$ = exports.LegalNaturalPersonWithholdingState$outboundSchema = exports.LegalNaturalPersonWithholdingState$inboundSchema = exports.LegalNaturalPersonUsTinStatus$ = exports.LegalNaturalPersonUsTinStatus$outboundSchema = exports.LegalNaturalPersonUsTinStatus$inboundSchema = exports.TaxpayerCertificationState$ = exports.TaxpayerCertificationState$outboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Classifies in what capacity (or if) the underlying natural person holds a job
 */
var LegalNaturalPersonEmploymentStatus;
(function (LegalNaturalPersonEmploymentStatus) {
    LegalNaturalPersonEmploymentStatus["StatusUnspecified"] = "STATUS_UNSPECIFIED";
    LegalNaturalPersonEmploymentStatus["Employed"] = "EMPLOYED";
    LegalNaturalPersonEmploymentStatus["SelfEmployed"] = "SELF_EMPLOYED";
    LegalNaturalPersonEmploymentStatus["Unemployed"] = "UNEMPLOYED";
    LegalNaturalPersonEmploymentStatus["Retired"] = "RETIRED";
    LegalNaturalPersonEmploymentStatus["Student"] = "STUDENT";
})(LegalNaturalPersonEmploymentStatus || (exports.LegalNaturalPersonEmploymentStatus = LegalNaturalPersonEmploymentStatus = {}));
/**
 * Identification type
 */
var LegalNaturalPersonType;
(function (LegalNaturalPersonType) {
    LegalNaturalPersonType["IdentificationTypeUnspecified"] = "IDENTIFICATION_TYPE_UNSPECIFIED";
    LegalNaturalPersonType["Passport"] = "PASSPORT";
    LegalNaturalPersonType["NationalId"] = "NATIONAL_ID";
    LegalNaturalPersonType["DriversLicense"] = "DRIVERS_LICENSE";
})(LegalNaturalPersonType || (exports.LegalNaturalPersonType = LegalNaturalPersonType = {}));
/**
 * The legal marital status of an account-holder; Used in combination with state of domicile to determine qualification for account types and beneficiary exclusion rules.
 */
var LegalNaturalPersonMaritalStatus;
(function (LegalNaturalPersonMaritalStatus) {
    LegalNaturalPersonMaritalStatus["MaritalStatusUnspecified"] = "MARITAL_STATUS_UNSPECIFIED";
    LegalNaturalPersonMaritalStatus["Single"] = "SINGLE";
    LegalNaturalPersonMaritalStatus["Married"] = "MARRIED";
    LegalNaturalPersonMaritalStatus["Divorced"] = "DIVORCED";
    LegalNaturalPersonMaritalStatus["Widowed"] = "WIDOWED";
})(LegalNaturalPersonMaritalStatus || (exports.LegalNaturalPersonMaritalStatus = LegalNaturalPersonMaritalStatus = {}));
/**
 * The suffix of a natural person; A suffix in a name is any part of the name that comes after the last name
 */
var LegalNaturalPersonNameSuffix;
(function (LegalNaturalPersonNameSuffix) {
    LegalNaturalPersonNameSuffix["NameSuffixUnspecified"] = "NAME_SUFFIX_UNSPECIFIED";
    LegalNaturalPersonNameSuffix["Sr"] = "SR";
    LegalNaturalPersonNameSuffix["Jr"] = "JR";
    LegalNaturalPersonNameSuffix["Iii"] = "III";
    LegalNaturalPersonNameSuffix["Iv"] = "IV";
    LegalNaturalPersonNameSuffix["V"] = "V";
})(LegalNaturalPersonNameSuffix || (exports.LegalNaturalPersonNameSuffix = LegalNaturalPersonNameSuffix = {}));
var LegalNaturalPersonResidencyStatus;
(function (LegalNaturalPersonResidencyStatus) {
    LegalNaturalPersonResidencyStatus["ResidencyStatusUnspecified"] = "RESIDENCY_STATUS_UNSPECIFIED";
    LegalNaturalPersonResidencyStatus["UsPermanentResident"] = "US_PERMANENT_RESIDENT";
    LegalNaturalPersonResidencyStatus["UsTemporaryResident"] = "US_TEMPORARY_RESIDENT";
    LegalNaturalPersonResidencyStatus["UsNonResident"] = "US_NON_RESIDENT";
})(LegalNaturalPersonResidencyStatus || (exports.LegalNaturalPersonResidencyStatus = LegalNaturalPersonResidencyStatus = {}));
/**
 * The nature of the U.S. Tax ID indicated in the related tax_id field; Examples include ITIN, SSN, EIN.
 */
var LegalNaturalPersonTaxIdType;
(function (LegalNaturalPersonTaxIdType) {
    LegalNaturalPersonTaxIdType["TaxIdTypeUnspecified"] = "TAX_ID_TYPE_UNSPECIFIED";
    LegalNaturalPersonTaxIdType["TaxIdTypeSsn"] = "TAX_ID_TYPE_SSN";
    LegalNaturalPersonTaxIdType["TaxIdTypeItin"] = "TAX_ID_TYPE_ITIN";
    LegalNaturalPersonTaxIdType["TaxIdTypeEin"] = "TAX_ID_TYPE_EIN";
})(LegalNaturalPersonTaxIdType || (exports.LegalNaturalPersonTaxIdType = LegalNaturalPersonTaxIdType = {}));
/**
 * Federal tax classification.
 */
var LegalNaturalPersonFederalTaxClassification;
(function (LegalNaturalPersonFederalTaxClassification) {
    LegalNaturalPersonFederalTaxClassification["FederalTaxClassificationUnspecified"] = "FEDERAL_TAX_CLASSIFICATION_UNSPECIFIED";
    LegalNaturalPersonFederalTaxClassification["IndivSolepropOrSinglememberllc"] = "INDIV_SOLEPROP_OR_SINGLEMEMBERLLC";
    LegalNaturalPersonFederalTaxClassification["Partnership"] = "PARTNERSHIP";
    LegalNaturalPersonFederalTaxClassification["CCorporation"] = "C_CORPORATION";
    LegalNaturalPersonFederalTaxClassification["SCorporation"] = "S_CORPORATION";
    LegalNaturalPersonFederalTaxClassification["TrustEstate"] = "TRUST_ESTATE";
    LegalNaturalPersonFederalTaxClassification["LlcTaxedAsCCorp"] = "LLC_TAXED_AS_C_CORP";
    LegalNaturalPersonFederalTaxClassification["LlcTaxedAsSCorp"] = "LLC_TAXED_AS_S_CORP";
    LegalNaturalPersonFederalTaxClassification["LlcTaxedAsPartnership"] = "LLC_TAXED_AS_PARTNERSHIP";
    LegalNaturalPersonFederalTaxClassification["Other"] = "OTHER";
})(LegalNaturalPersonFederalTaxClassification || (exports.LegalNaturalPersonFederalTaxClassification = LegalNaturalPersonFederalTaxClassification = {}));
/**
 * IRS form type.
 */
var LegalNaturalPersonIrsFormType;
(function (LegalNaturalPersonIrsFormType) {
    LegalNaturalPersonIrsFormType["IrsFormTypeUnspecified"] = "IRS_FORM_TYPE_UNSPECIFIED";
    LegalNaturalPersonIrsFormType["W9"] = "W_9";
    LegalNaturalPersonIrsFormType["W8Ben"] = "W_8BEN";
})(LegalNaturalPersonIrsFormType || (exports.LegalNaturalPersonIrsFormType = LegalNaturalPersonIrsFormType = {}));
/**
 * Tax reporting eligibility.
 */
var ReportingEligibility;
(function (ReportingEligibility) {
    ReportingEligibility["TaxReportingEligibilityUnspecified"] = "TAX_REPORTING_ELIGIBILITY_UNSPECIFIED";
    ReportingEligibility["Eligible"] = "ELIGIBLE";
    ReportingEligibility["Ineligible"] = "INELIGIBLE";
})(ReportingEligibility || (exports.ReportingEligibility = ReportingEligibility = {}));
/**
 * Taxpayer certification status.
 */
var TaxpayerCertificationState;
(function (TaxpayerCertificationState) {
    TaxpayerCertificationState["TaxpayerCertificationStateUnspecified"] = "TAXPAYER_CERTIFICATION_STATE_UNSPECIFIED";
    TaxpayerCertificationState["Certified"] = "CERTIFIED";
    TaxpayerCertificationState["Uncertified"] = "UNCERTIFIED";
})(TaxpayerCertificationState || (exports.TaxpayerCertificationState = TaxpayerCertificationState = {}));
/**
 * United States Individual Taxpayer Identification Number (ITIN) status.
 */
var LegalNaturalPersonUsTinStatus;
(function (LegalNaturalPersonUsTinStatus) {
    LegalNaturalPersonUsTinStatus["UsTinStatusUnspecified"] = "US_TIN_STATUS_UNSPECIFIED";
    LegalNaturalPersonUsTinStatus["Passing"] = "PASSING";
    LegalNaturalPersonUsTinStatus["Failing"] = "FAILING";
})(LegalNaturalPersonUsTinStatus || (exports.LegalNaturalPersonUsTinStatus = LegalNaturalPersonUsTinStatus = {}));
/**
 * B/C Notice status.
 */
var LegalNaturalPersonWithholdingState;
(function (LegalNaturalPersonWithholdingState) {
    LegalNaturalPersonWithholdingState["WithholdingStateUnspecified"] = "WITHHOLDING_STATE_UNSPECIFIED";
    LegalNaturalPersonWithholdingState["FirstBNoticeReceived"] = "FIRST_B_NOTICE_RECEIVED";
    LegalNaturalPersonWithholdingState["SecondBNoticeReceived"] = "SECOND_B_NOTICE_RECEIVED";
    LegalNaturalPersonWithholdingState["CNoticeReceived"] = "C_NOTICE_RECEIVED";
    LegalNaturalPersonWithholdingState["CNoticeIndicatedByCustomer"] = "C_NOTICE_INDICATED_BY_CUSTOMER";
})(LegalNaturalPersonWithholdingState || (exports.LegalNaturalPersonWithholdingState = LegalNaturalPersonWithholdingState = {}));
/** @internal */
exports.BirthDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.BirthDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BirthDate$;
(function (BirthDate$) {
    /** @deprecated use `BirthDate$inboundSchema` instead. */
    BirthDate$.inboundSchema = exports.BirthDate$inboundSchema;
    /** @deprecated use `BirthDate$outboundSchema` instead. */
    BirthDate$.outboundSchema = exports.BirthDate$outboundSchema;
})(BirthDate$ || (exports.BirthDate$ = BirthDate$ = {}));
/** @internal */
exports.DeathDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.DeathDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeathDate$;
(function (DeathDate$) {
    /** @deprecated use `DeathDate$inboundSchema` instead. */
    DeathDate$.inboundSchema = exports.DeathDate$inboundSchema;
    /** @deprecated use `DeathDate$outboundSchema` instead. */
    DeathDate$.outboundSchema = exports.DeathDate$outboundSchema;
})(DeathDate$ || (exports.DeathDate$ = DeathDate$ = {}));
/** @internal */
exports.EmployerAddress$inboundSchema = z.object({
    address_lines: z.array(z.string()).optional(),
    administrative_area: z.string().optional(),
    language_code: z.string().optional(),
    locality: z.string().optional(),
    organization: z.string().optional(),
    postal_code: z.string().optional(),
    recipients: z.array(z.string()).optional(),
    region_code: z.string().optional(),
    revision: z.number().int().optional(),
    sorting_code: z.string().optional(),
    sublocality: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "address_lines": "addressLines",
        "administrative_area": "administrativeArea",
        "language_code": "languageCode",
        "postal_code": "postalCode",
        "region_code": "regionCode",
        "sorting_code": "sortingCode",
    });
});
/** @internal */
exports.EmployerAddress$outboundSchema = z.object({
    addressLines: z.array(z.string()).optional(),
    administrativeArea: z.string().optional(),
    languageCode: z.string().optional(),
    locality: z.string().optional(),
    organization: z.string().optional(),
    postalCode: z.string().optional(),
    recipients: z.array(z.string()).optional(),
    regionCode: z.string().optional(),
    revision: z.number().int().optional(),
    sortingCode: z.string().optional(),
    sublocality: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addressLines: "address_lines",
        administrativeArea: "administrative_area",
        languageCode: "language_code",
        postalCode: "postal_code",
        regionCode: "region_code",
        sortingCode: "sorting_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EmployerAddress$;
(function (EmployerAddress$) {
    /** @deprecated use `EmployerAddress$inboundSchema` instead. */
    EmployerAddress$.inboundSchema = exports.EmployerAddress$inboundSchema;
    /** @deprecated use `EmployerAddress$outboundSchema` instead. */
    EmployerAddress$.outboundSchema = exports.EmployerAddress$outboundSchema;
})(EmployerAddress$ || (exports.EmployerAddress$ = EmployerAddress$ = {}));
/** @internal */
exports.LegalNaturalPersonEmploymentStatus$inboundSchema = z
    .union([
    z.nativeEnum(LegalNaturalPersonEmploymentStatus),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalNaturalPersonEmploymentStatus$outboundSchema = z.union([
    z.nativeEnum(LegalNaturalPersonEmploymentStatus),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonEmploymentStatus$;
(function (LegalNaturalPersonEmploymentStatus$) {
    /** @deprecated use `LegalNaturalPersonEmploymentStatus$inboundSchema` instead. */
    LegalNaturalPersonEmploymentStatus$.inboundSchema = exports.LegalNaturalPersonEmploymentStatus$inboundSchema;
    /** @deprecated use `LegalNaturalPersonEmploymentStatus$outboundSchema` instead. */
    LegalNaturalPersonEmploymentStatus$.outboundSchema = exports.LegalNaturalPersonEmploymentStatus$outboundSchema;
})(LegalNaturalPersonEmploymentStatus$ || (exports.LegalNaturalPersonEmploymentStatus$ = LegalNaturalPersonEmploymentStatus$ = {}));
/** @internal */
exports.Employment$inboundSchema = z.object({
    employer: z.string().optional(),
    employer_address: z.nullable(z.lazy(() => exports.EmployerAddress$inboundSchema))
        .optional(),
    employment_id: z.string().optional(),
    employment_status: exports.LegalNaturalPersonEmploymentStatus$inboundSchema
        .optional(),
    occupation: z.string().optional(),
    start_year: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "employer_address": "employerAddress",
        "employment_id": "employmentId",
        "employment_status": "employmentStatus",
        "start_year": "startYear",
    });
});
/** @internal */
exports.Employment$outboundSchema = z.object({
    employer: z.string().optional(),
    employerAddress: z.nullable(z.lazy(() => exports.EmployerAddress$outboundSchema))
        .optional(),
    employmentId: z.string().optional(),
    employmentStatus: exports.LegalNaturalPersonEmploymentStatus$outboundSchema
        .optional(),
    occupation: z.string().optional(),
    startYear: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        employerAddress: "employer_address",
        employmentId: "employment_id",
        employmentStatus: "employment_status",
        startYear: "start_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Employment$;
(function (Employment$) {
    /** @deprecated use `Employment$inboundSchema` instead. */
    Employment$.inboundSchema = exports.Employment$inboundSchema;
    /** @deprecated use `Employment$outboundSchema` instead. */
    Employment$.outboundSchema = exports.Employment$outboundSchema;
})(Employment$ || (exports.Employment$ = Employment$ = {}));
/** @internal */
exports.ExpirationDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.ExpirationDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ExpirationDate$;
(function (ExpirationDate$) {
    /** @deprecated use `ExpirationDate$inboundSchema` instead. */
    ExpirationDate$.inboundSchema = exports.ExpirationDate$inboundSchema;
    /** @deprecated use `ExpirationDate$outboundSchema` instead. */
    ExpirationDate$.outboundSchema = exports.ExpirationDate$outboundSchema;
})(ExpirationDate$ || (exports.ExpirationDate$ = ExpirationDate$ = {}));
/** @internal */
exports.IssueDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.IssueDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IssueDate$;
(function (IssueDate$) {
    /** @deprecated use `IssueDate$inboundSchema` instead. */
    IssueDate$.inboundSchema = exports.IssueDate$inboundSchema;
    /** @deprecated use `IssueDate$outboundSchema` instead. */
    IssueDate$.outboundSchema = exports.IssueDate$outboundSchema;
})(IssueDate$ || (exports.IssueDate$ = IssueDate$ = {}));
/** @internal */
exports.LegalNaturalPersonType$inboundSchema = z
    .union([
    z.nativeEnum(LegalNaturalPersonType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalNaturalPersonType$outboundSchema = z.union([
    z.nativeEnum(LegalNaturalPersonType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonType$;
(function (LegalNaturalPersonType$) {
    /** @deprecated use `LegalNaturalPersonType$inboundSchema` instead. */
    LegalNaturalPersonType$.inboundSchema = exports.LegalNaturalPersonType$inboundSchema;
    /** @deprecated use `LegalNaturalPersonType$outboundSchema` instead. */
    LegalNaturalPersonType$.outboundSchema = exports.LegalNaturalPersonType$outboundSchema;
})(LegalNaturalPersonType$ || (exports.LegalNaturalPersonType$ = LegalNaturalPersonType$ = {}));
/** @internal */
exports.ForeignIdentification$inboundSchema = z.object({
    expiration_date: z.nullable(z.lazy(() => exports.ExpirationDate$inboundSchema))
        .optional(),
    ftin: z.boolean().optional(),
    identification_number: z.string().optional(),
    issue_date: z.nullable(z.lazy(() => exports.IssueDate$inboundSchema)).optional(),
    issuing_region_code: z.string().optional(),
    type: exports.LegalNaturalPersonType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "expiration_date": "expirationDate",
        "identification_number": "identificationNumber",
        "issue_date": "issueDate",
        "issuing_region_code": "issuingRegionCode",
    });
});
/** @internal */
exports.ForeignIdentification$outboundSchema = z.object({
    expirationDate: z.nullable(z.lazy(() => exports.ExpirationDate$outboundSchema))
        .optional(),
    ftin: z.boolean().optional(),
    identificationNumber: z.string().optional(),
    issueDate: z.nullable(z.lazy(() => exports.IssueDate$outboundSchema)).optional(),
    issuingRegionCode: z.string().optional(),
    type: exports.LegalNaturalPersonType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        expirationDate: "expiration_date",
        identificationNumber: "identification_number",
        issueDate: "issue_date",
        issuingRegionCode: "issuing_region_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ForeignIdentification$;
(function (ForeignIdentification$) {
    /** @deprecated use `ForeignIdentification$inboundSchema` instead. */
    ForeignIdentification$.inboundSchema = exports.ForeignIdentification$inboundSchema;
    /** @deprecated use `ForeignIdentification$outboundSchema` instead. */
    ForeignIdentification$.outboundSchema = exports.ForeignIdentification$outboundSchema;
})(ForeignIdentification$ || (exports.ForeignIdentification$ = ForeignIdentification$ = {}));
/** @internal */
exports.ExecutionDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.ExecutionDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ExecutionDate$;
(function (ExecutionDate$) {
    /** @deprecated use `ExecutionDate$inboundSchema` instead. */
    ExecutionDate$.inboundSchema = exports.ExecutionDate$inboundSchema;
    /** @deprecated use `ExecutionDate$outboundSchema` instead. */
    ExecutionDate$.outboundSchema = exports.ExecutionDate$outboundSchema;
})(ExecutionDate$ || (exports.ExecutionDate$ = ExecutionDate$ = {}));
/** @internal */
exports.LegalNaturalPersonIdentityVerificationResult$inboundSchema = z.object({
    address_verified: z.boolean().optional(),
    birth_date_verified: z.boolean().optional(),
    execution_date: z.nullable(z.lazy(() => exports.ExecutionDate$inboundSchema))
        .optional(),
    external_case_id: z.string().optional(),
    identity_verification_document_ids: z.array(z.string()).optional(),
    identity_verification_result_id: z.string().optional(),
    name_verified: z.boolean().optional(),
    raw_vendor_data_document_id: z.string().optional(),
    tax_id_verified: z.boolean().optional(),
    vendor: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "address_verified": "addressVerified",
        "birth_date_verified": "birthDateVerified",
        "execution_date": "executionDate",
        "external_case_id": "externalCaseId",
        "identity_verification_document_ids": "identityVerificationDocumentIds",
        "identity_verification_result_id": "identityVerificationResultId",
        "name_verified": "nameVerified",
        "raw_vendor_data_document_id": "rawVendorDataDocumentId",
        "tax_id_verified": "taxIdVerified",
    });
});
/** @internal */
exports.LegalNaturalPersonIdentityVerificationResult$outboundSchema = z.object({
    addressVerified: z.boolean().optional(),
    birthDateVerified: z.boolean().optional(),
    executionDate: z.nullable(z.lazy(() => exports.ExecutionDate$outboundSchema))
        .optional(),
    externalCaseId: z.string().optional(),
    identityVerificationDocumentIds: z.array(z.string()).optional(),
    identityVerificationResultId: z.string().optional(),
    nameVerified: z.boolean().optional(),
    rawVendorDataDocumentId: z.string().optional(),
    taxIdVerified: z.boolean().optional(),
    vendor: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addressVerified: "address_verified",
        birthDateVerified: "birth_date_verified",
        executionDate: "execution_date",
        externalCaseId: "external_case_id",
        identityVerificationDocumentIds: "identity_verification_document_ids",
        identityVerificationResultId: "identity_verification_result_id",
        nameVerified: "name_verified",
        rawVendorDataDocumentId: "raw_vendor_data_document_id",
        taxIdVerified: "tax_id_verified",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonIdentityVerificationResult$;
(function (LegalNaturalPersonIdentityVerificationResult$) {
    /** @deprecated use `LegalNaturalPersonIdentityVerificationResult$inboundSchema` instead. */
    LegalNaturalPersonIdentityVerificationResult$.inboundSchema = exports.LegalNaturalPersonIdentityVerificationResult$inboundSchema;
    /** @deprecated use `LegalNaturalPersonIdentityVerificationResult$outboundSchema` instead. */
    LegalNaturalPersonIdentityVerificationResult$.outboundSchema = exports.LegalNaturalPersonIdentityVerificationResult$outboundSchema;
})(LegalNaturalPersonIdentityVerificationResult$ || (exports.LegalNaturalPersonIdentityVerificationResult$ = LegalNaturalPersonIdentityVerificationResult$ = {}));
/** @internal */
exports.LegalNaturalPersonEffectiveDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.LegalNaturalPersonEffectiveDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonEffectiveDate$;
(function (LegalNaturalPersonEffectiveDate$) {
    /** @deprecated use `LegalNaturalPersonEffectiveDate$inboundSchema` instead. */
    LegalNaturalPersonEffectiveDate$.inboundSchema = exports.LegalNaturalPersonEffectiveDate$inboundSchema;
    /** @deprecated use `LegalNaturalPersonEffectiveDate$outboundSchema` instead. */
    LegalNaturalPersonEffectiveDate$.outboundSchema = exports.LegalNaturalPersonEffectiveDate$outboundSchema;
})(LegalNaturalPersonEffectiveDate$ || (exports.LegalNaturalPersonEffectiveDate$ = LegalNaturalPersonEffectiveDate$ = {}));
/** @internal */
exports.LegalNaturalPersonLargeTrader$inboundSchema = z.object({
    effective_date: z.nullable(z.lazy(() => exports.LegalNaturalPersonEffectiveDate$inboundSchema)).optional(),
    large_trader_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "effective_date": "effectiveDate",
        "large_trader_id": "largeTraderId",
    });
});
/** @internal */
exports.LegalNaturalPersonLargeTrader$outboundSchema = z.object({
    effectiveDate: z.nullable(z.lazy(() => exports.LegalNaturalPersonEffectiveDate$outboundSchema)).optional(),
    largeTraderId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        effectiveDate: "effective_date",
        largeTraderId: "large_trader_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonLargeTrader$;
(function (LegalNaturalPersonLargeTrader$) {
    /** @deprecated use `LegalNaturalPersonLargeTrader$inboundSchema` instead. */
    LegalNaturalPersonLargeTrader$.inboundSchema = exports.LegalNaturalPersonLargeTrader$inboundSchema;
    /** @deprecated use `LegalNaturalPersonLargeTrader$outboundSchema` instead. */
    LegalNaturalPersonLargeTrader$.outboundSchema = exports.LegalNaturalPersonLargeTrader$outboundSchema;
})(LegalNaturalPersonLargeTrader$ || (exports.LegalNaturalPersonLargeTrader$ = LegalNaturalPersonLargeTrader$ = {}));
/** @internal */
exports.LegalNaturalPersonMaritalStatus$inboundSchema = z
    .union([
    z.nativeEnum(LegalNaturalPersonMaritalStatus),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalNaturalPersonMaritalStatus$outboundSchema = z.union([
    z.nativeEnum(LegalNaturalPersonMaritalStatus),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonMaritalStatus$;
(function (LegalNaturalPersonMaritalStatus$) {
    /** @deprecated use `LegalNaturalPersonMaritalStatus$inboundSchema` instead. */
    LegalNaturalPersonMaritalStatus$.inboundSchema = exports.LegalNaturalPersonMaritalStatus$inboundSchema;
    /** @deprecated use `LegalNaturalPersonMaritalStatus$outboundSchema` instead. */
    LegalNaturalPersonMaritalStatus$.outboundSchema = exports.LegalNaturalPersonMaritalStatus$outboundSchema;
})(LegalNaturalPersonMaritalStatus$ || (exports.LegalNaturalPersonMaritalStatus$ = LegalNaturalPersonMaritalStatus$ = {}));
/** @internal */
exports.LegalNaturalPersonNameSuffix$inboundSchema = z
    .union([
    z.nativeEnum(LegalNaturalPersonNameSuffix),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalNaturalPersonNameSuffix$outboundSchema = z.union([
    z.nativeEnum(LegalNaturalPersonNameSuffix),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonNameSuffix$;
(function (LegalNaturalPersonNameSuffix$) {
    /** @deprecated use `LegalNaturalPersonNameSuffix$inboundSchema` instead. */
    LegalNaturalPersonNameSuffix$.inboundSchema = exports.LegalNaturalPersonNameSuffix$inboundSchema;
    /** @deprecated use `LegalNaturalPersonNameSuffix$outboundSchema` instead. */
    LegalNaturalPersonNameSuffix$.outboundSchema = exports.LegalNaturalPersonNameSuffix$outboundSchema;
})(LegalNaturalPersonNameSuffix$ || (exports.LegalNaturalPersonNameSuffix$ = LegalNaturalPersonNameSuffix$ = {}));
/** @internal */
exports.CustomerReferralSource$inboundSchema = z.object({
    name: z.string().optional(),
    relationship_to_applicant: z.string().optional(),
    relationship_years_with_applicant: z.number().int().optional(),
    relationship_years_with_broker: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "relationship_to_applicant": "relationshipToApplicant",
        "relationship_years_with_applicant": "relationshipYearsWithApplicant",
        "relationship_years_with_broker": "relationshipYearsWithBroker",
    });
});
/** @internal */
exports.CustomerReferralSource$outboundSchema = z.object({
    name: z.string().optional(),
    relationshipToApplicant: z.string().optional(),
    relationshipYearsWithApplicant: z.number().int().optional(),
    relationshipYearsWithBroker: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        relationshipToApplicant: "relationship_to_applicant",
        relationshipYearsWithApplicant: "relationship_years_with_applicant",
        relationshipYearsWithBroker: "relationship_years_with_broker",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerReferralSource$;
(function (CustomerReferralSource$) {
    /** @deprecated use `CustomerReferralSource$inboundSchema` instead. */
    CustomerReferralSource$.inboundSchema = exports.CustomerReferralSource$inboundSchema;
    /** @deprecated use `CustomerReferralSource$outboundSchema` instead. */
    CustomerReferralSource$.outboundSchema = exports.CustomerReferralSource$outboundSchema;
})(CustomerReferralSource$ || (exports.CustomerReferralSource$ = CustomerReferralSource$ = {}));
/** @internal */
exports.NegativeNews$inboundSchema = z.object({
    negative_news_against_related_parties: z.boolean().optional(),
    negative_news_against_related_parties_description: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "negative_news_against_related_parties": "negativeNewsAgainstRelatedParties",
        "negative_news_against_related_parties_description": "negativeNewsAgainstRelatedPartiesDescription",
    });
});
/** @internal */
exports.NegativeNews$outboundSchema = z.object({
    negativeNewsAgainstRelatedParties: z.boolean().optional(),
    negativeNewsAgainstRelatedPartiesDescription: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        negativeNewsAgainstRelatedParties: "negative_news_against_related_parties",
        negativeNewsAgainstRelatedPartiesDescription: "negative_news_against_related_parties_description",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NegativeNews$;
(function (NegativeNews$) {
    /** @deprecated use `NegativeNews$inboundSchema` instead. */
    NegativeNews$.inboundSchema = exports.NegativeNews$inboundSchema;
    /** @deprecated use `NegativeNews$outboundSchema` instead. */
    NegativeNews$.outboundSchema = exports.NegativeNews$outboundSchema;
})(NegativeNews$ || (exports.NegativeNews$ = NegativeNews$ = {}));
/** @internal */
exports.OtherSourcesOfWealth$inboundSchema = z.object({
    applicant_has_other_sources_of_wealth: z.boolean().optional(),
    other_sources_of_wealth: z.string().optional(),
    other_sources_of_wealth_verification: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "applicant_has_other_sources_of_wealth": "applicantHasOtherSourcesOfWealth",
        "other_sources_of_wealth": "otherSourcesOfWealth",
        "other_sources_of_wealth_verification": "otherSourcesOfWealthVerification",
    });
});
/** @internal */
exports.OtherSourcesOfWealth$outboundSchema = z.object({
    applicantHasOtherSourcesOfWealth: z.boolean().optional(),
    otherSourcesOfWealth: z.string().optional(),
    otherSourcesOfWealthVerification: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        applicantHasOtherSourcesOfWealth: "applicant_has_other_sources_of_wealth",
        otherSourcesOfWealth: "other_sources_of_wealth",
        otherSourcesOfWealthVerification: "other_sources_of_wealth_verification",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OtherSourcesOfWealth$;
(function (OtherSourcesOfWealth$) {
    /** @deprecated use `OtherSourcesOfWealth$inboundSchema` instead. */
    OtherSourcesOfWealth$.inboundSchema = exports.OtherSourcesOfWealth$inboundSchema;
    /** @deprecated use `OtherSourcesOfWealth$outboundSchema` instead. */
    OtherSourcesOfWealth$.outboundSchema = exports.OtherSourcesOfWealth$outboundSchema;
})(OtherSourcesOfWealth$ || (exports.OtherSourcesOfWealth$ = OtherSourcesOfWealth$ = {}));
/** @internal */
exports.NaturalPersonFdd$inboundSchema = z.object({
    customer_non_referral_source: z.string().optional(),
    customer_referral_source: z.nullable(z.lazy(() => exports.CustomerReferralSource$inboundSchema)).optional(),
    employment_and_employer_description: z.string().optional(),
    negative_news: z.nullable(z.lazy(() => exports.NegativeNews$inboundSchema))
        .optional(),
    other_sources_of_wealth: z.nullable(z.lazy(() => exports.OtherSourcesOfWealth$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_non_referral_source": "customerNonReferralSource",
        "customer_referral_source": "customerReferralSource",
        "employment_and_employer_description": "employmentAndEmployerDescription",
        "negative_news": "negativeNews",
        "other_sources_of_wealth": "otherSourcesOfWealth",
    });
});
/** @internal */
exports.NaturalPersonFdd$outboundSchema = z.object({
    customerNonReferralSource: z.string().optional(),
    customerReferralSource: z.nullable(z.lazy(() => exports.CustomerReferralSource$outboundSchema)).optional(),
    employmentAndEmployerDescription: z.string().optional(),
    negativeNews: z.nullable(z.lazy(() => exports.NegativeNews$outboundSchema))
        .optional(),
    otherSourcesOfWealth: z.nullable(z.lazy(() => exports.OtherSourcesOfWealth$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerNonReferralSource: "customer_non_referral_source",
        customerReferralSource: "customer_referral_source",
        employmentAndEmployerDescription: "employment_and_employer_description",
        negativeNews: "negative_news",
        otherSourcesOfWealth: "other_sources_of_wealth",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NaturalPersonFdd$;
(function (NaturalPersonFdd$) {
    /** @deprecated use `NaturalPersonFdd$inboundSchema` instead. */
    NaturalPersonFdd$.inboundSchema = exports.NaturalPersonFdd$inboundSchema;
    /** @deprecated use `NaturalPersonFdd$outboundSchema` instead. */
    NaturalPersonFdd$.outboundSchema = exports.NaturalPersonFdd$outboundSchema;
})(NaturalPersonFdd$ || (exports.NaturalPersonFdd$ = NaturalPersonFdd$ = {}));
/** @internal */
exports.LegalNaturalPersonResidencyStatus$inboundSchema = z
    .union([
    z.nativeEnum(LegalNaturalPersonResidencyStatus),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalNaturalPersonResidencyStatus$outboundSchema = z.union([
    z.nativeEnum(LegalNaturalPersonResidencyStatus),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonResidencyStatus$;
(function (LegalNaturalPersonResidencyStatus$) {
    /** @deprecated use `LegalNaturalPersonResidencyStatus$inboundSchema` instead. */
    LegalNaturalPersonResidencyStatus$.inboundSchema = exports.LegalNaturalPersonResidencyStatus$inboundSchema;
    /** @deprecated use `LegalNaturalPersonResidencyStatus$outboundSchema` instead. */
    LegalNaturalPersonResidencyStatus$.outboundSchema = exports.LegalNaturalPersonResidencyStatus$outboundSchema;
})(LegalNaturalPersonResidencyStatus$ || (exports.LegalNaturalPersonResidencyStatus$ = LegalNaturalPersonResidencyStatus$ = {}));
/** @internal */
exports.NonCitizenResidency$inboundSchema = z.object({
    residency_status: exports.LegalNaturalPersonResidencyStatus$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "residency_status": "residencyStatus",
    });
});
/** @internal */
exports.NonCitizenResidency$outboundSchema = z.object({
    residencyStatus: exports.LegalNaturalPersonResidencyStatus$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        residencyStatus: "residency_status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NonCitizenResidency$;
(function (NonCitizenResidency$) {
    /** @deprecated use `NonCitizenResidency$inboundSchema` instead. */
    NonCitizenResidency$.inboundSchema = exports.NonCitizenResidency$inboundSchema;
    /** @deprecated use `NonCitizenResidency$outboundSchema` instead. */
    NonCitizenResidency$.outboundSchema = exports.NonCitizenResidency$outboundSchema;
})(NonCitizenResidency$ || (exports.NonCitizenResidency$ = NonCitizenResidency$ = {}));
/** @internal */
exports.PersonalAddress$inboundSchema = z.object({
    address_lines: z.array(z.string()).optional(),
    administrative_area: z.string().optional(),
    language_code: z.string().optional(),
    locality: z.string().optional(),
    organization: z.string().optional(),
    postal_code: z.string().optional(),
    recipients: z.array(z.string()).optional(),
    region_code: z.string().optional(),
    revision: z.number().int().optional(),
    sorting_code: z.string().optional(),
    sublocality: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "address_lines": "addressLines",
        "administrative_area": "administrativeArea",
        "language_code": "languageCode",
        "postal_code": "postalCode",
        "region_code": "regionCode",
        "sorting_code": "sortingCode",
    });
});
/** @internal */
exports.PersonalAddress$outboundSchema = z.object({
    addressLines: z.array(z.string()).optional(),
    administrativeArea: z.string().optional(),
    languageCode: z.string().optional(),
    locality: z.string().optional(),
    organization: z.string().optional(),
    postalCode: z.string().optional(),
    recipients: z.array(z.string()).optional(),
    regionCode: z.string().optional(),
    revision: z.number().int().optional(),
    sortingCode: z.string().optional(),
    sublocality: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addressLines: "address_lines",
        administrativeArea: "administrative_area",
        languageCode: "language_code",
        postalCode: "postal_code",
        regionCode: "region_code",
        sortingCode: "sorting_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PersonalAddress$;
(function (PersonalAddress$) {
    /** @deprecated use `PersonalAddress$inboundSchema` instead. */
    PersonalAddress$.inboundSchema = exports.PersonalAddress$inboundSchema;
    /** @deprecated use `PersonalAddress$outboundSchema` instead. */
    PersonalAddress$.outboundSchema = exports.PersonalAddress$outboundSchema;
})(PersonalAddress$ || (exports.PersonalAddress$ = PersonalAddress$ = {}));
/** @internal */
exports.LegalNaturalPersonTaxIdType$inboundSchema = z
    .union([
    z.nativeEnum(LegalNaturalPersonTaxIdType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalNaturalPersonTaxIdType$outboundSchema = z.union([
    z.nativeEnum(LegalNaturalPersonTaxIdType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonTaxIdType$;
(function (LegalNaturalPersonTaxIdType$) {
    /** @deprecated use `LegalNaturalPersonTaxIdType$inboundSchema` instead. */
    LegalNaturalPersonTaxIdType$.inboundSchema = exports.LegalNaturalPersonTaxIdType$inboundSchema;
    /** @deprecated use `LegalNaturalPersonTaxIdType$outboundSchema` instead. */
    LegalNaturalPersonTaxIdType$.outboundSchema = exports.LegalNaturalPersonTaxIdType$outboundSchema;
})(LegalNaturalPersonTaxIdType$ || (exports.LegalNaturalPersonTaxIdType$ = LegalNaturalPersonTaxIdType$ = {}));
/** @internal */
exports.CNoticeDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.CNoticeDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CNoticeDate$;
(function (CNoticeDate$) {
    /** @deprecated use `CNoticeDate$inboundSchema` instead. */
    CNoticeDate$.inboundSchema = exports.CNoticeDate$inboundSchema;
    /** @deprecated use `CNoticeDate$outboundSchema` instead. */
    CNoticeDate$.outboundSchema = exports.CNoticeDate$outboundSchema;
})(CNoticeDate$ || (exports.CNoticeDate$ = CNoticeDate$ = {}));
/** @internal */
exports.LegalNaturalPersonFederalTaxClassification$inboundSchema = z
    .union([
    z.nativeEnum(LegalNaturalPersonFederalTaxClassification),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalNaturalPersonFederalTaxClassification$outboundSchema = z.union([
    z.nativeEnum(LegalNaturalPersonFederalTaxClassification),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonFederalTaxClassification$;
(function (LegalNaturalPersonFederalTaxClassification$) {
    /** @deprecated use `LegalNaturalPersonFederalTaxClassification$inboundSchema` instead. */
    LegalNaturalPersonFederalTaxClassification$.inboundSchema = exports.LegalNaturalPersonFederalTaxClassification$inboundSchema;
    /** @deprecated use `LegalNaturalPersonFederalTaxClassification$outboundSchema` instead. */
    LegalNaturalPersonFederalTaxClassification$.outboundSchema = exports.LegalNaturalPersonFederalTaxClassification$outboundSchema;
})(LegalNaturalPersonFederalTaxClassification$ || (exports.LegalNaturalPersonFederalTaxClassification$ = LegalNaturalPersonFederalTaxClassification$ = {}));
/** @internal */
exports.FirstBNoticeDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.FirstBNoticeDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FirstBNoticeDate$;
(function (FirstBNoticeDate$) {
    /** @deprecated use `FirstBNoticeDate$inboundSchema` instead. */
    FirstBNoticeDate$.inboundSchema = exports.FirstBNoticeDate$inboundSchema;
    /** @deprecated use `FirstBNoticeDate$outboundSchema` instead. */
    FirstBNoticeDate$.outboundSchema = exports.FirstBNoticeDate$outboundSchema;
})(FirstBNoticeDate$ || (exports.FirstBNoticeDate$ = FirstBNoticeDate$ = {}));
/** @internal */
exports.LegalNaturalPersonIrsFormType$inboundSchema = z
    .union([
    z.nativeEnum(LegalNaturalPersonIrsFormType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalNaturalPersonIrsFormType$outboundSchema = z.union([
    z.nativeEnum(LegalNaturalPersonIrsFormType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonIrsFormType$;
(function (LegalNaturalPersonIrsFormType$) {
    /** @deprecated use `LegalNaturalPersonIrsFormType$inboundSchema` instead. */
    LegalNaturalPersonIrsFormType$.inboundSchema = exports.LegalNaturalPersonIrsFormType$inboundSchema;
    /** @deprecated use `LegalNaturalPersonIrsFormType$outboundSchema` instead. */
    LegalNaturalPersonIrsFormType$.outboundSchema = exports.LegalNaturalPersonIrsFormType$outboundSchema;
})(LegalNaturalPersonIrsFormType$ || (exports.LegalNaturalPersonIrsFormType$ = LegalNaturalPersonIrsFormType$ = {}));
/** @internal */
exports.ReportingEligibility$inboundSchema = z
    .union([
    z.nativeEnum(ReportingEligibility),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ReportingEligibility$outboundSchema = z.union([
    z.nativeEnum(ReportingEligibility),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ReportingEligibility$;
(function (ReportingEligibility$) {
    /** @deprecated use `ReportingEligibility$inboundSchema` instead. */
    ReportingEligibility$.inboundSchema = exports.ReportingEligibility$inboundSchema;
    /** @deprecated use `ReportingEligibility$outboundSchema` instead. */
    ReportingEligibility$.outboundSchema = exports.ReportingEligibility$outboundSchema;
})(ReportingEligibility$ || (exports.ReportingEligibility$ = ReportingEligibility$ = {}));
/** @internal */
exports.TaxCertificationDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.TaxCertificationDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TaxCertificationDate$;
(function (TaxCertificationDate$) {
    /** @deprecated use `TaxCertificationDate$inboundSchema` instead. */
    TaxCertificationDate$.inboundSchema = exports.TaxCertificationDate$inboundSchema;
    /** @deprecated use `TaxCertificationDate$outboundSchema` instead. */
    TaxCertificationDate$.outboundSchema = exports.TaxCertificationDate$outboundSchema;
})(TaxCertificationDate$ || (exports.TaxCertificationDate$ = TaxCertificationDate$ = {}));
/** @internal */
exports.TaxpayerCertificationState$inboundSchema = z
    .union([
    z.nativeEnum(TaxpayerCertificationState),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TaxpayerCertificationState$outboundSchema = z.union([
    z.nativeEnum(TaxpayerCertificationState),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TaxpayerCertificationState$;
(function (TaxpayerCertificationState$) {
    /** @deprecated use `TaxpayerCertificationState$inboundSchema` instead. */
    TaxpayerCertificationState$.inboundSchema = exports.TaxpayerCertificationState$inboundSchema;
    /** @deprecated use `TaxpayerCertificationState$outboundSchema` instead. */
    TaxpayerCertificationState$.outboundSchema = exports.TaxpayerCertificationState$outboundSchema;
})(TaxpayerCertificationState$ || (exports.TaxpayerCertificationState$ = TaxpayerCertificationState$ = {}));
/** @internal */
exports.LegalNaturalPersonUsTinStatus$inboundSchema = z
    .union([
    z.nativeEnum(LegalNaturalPersonUsTinStatus),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalNaturalPersonUsTinStatus$outboundSchema = z.union([
    z.nativeEnum(LegalNaturalPersonUsTinStatus),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonUsTinStatus$;
(function (LegalNaturalPersonUsTinStatus$) {
    /** @deprecated use `LegalNaturalPersonUsTinStatus$inboundSchema` instead. */
    LegalNaturalPersonUsTinStatus$.inboundSchema = exports.LegalNaturalPersonUsTinStatus$inboundSchema;
    /** @deprecated use `LegalNaturalPersonUsTinStatus$outboundSchema` instead. */
    LegalNaturalPersonUsTinStatus$.outboundSchema = exports.LegalNaturalPersonUsTinStatus$outboundSchema;
})(LegalNaturalPersonUsTinStatus$ || (exports.LegalNaturalPersonUsTinStatus$ = LegalNaturalPersonUsTinStatus$ = {}));
/** @internal */
exports.LegalNaturalPersonWithholdingState$inboundSchema = z
    .union([
    z.nativeEnum(LegalNaturalPersonWithholdingState),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalNaturalPersonWithholdingState$outboundSchema = z.union([
    z.nativeEnum(LegalNaturalPersonWithholdingState),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPersonWithholdingState$;
(function (LegalNaturalPersonWithholdingState$) {
    /** @deprecated use `LegalNaturalPersonWithholdingState$inboundSchema` instead. */
    LegalNaturalPersonWithholdingState$.inboundSchema = exports.LegalNaturalPersonWithholdingState$inboundSchema;
    /** @deprecated use `LegalNaturalPersonWithholdingState$outboundSchema` instead. */
    LegalNaturalPersonWithholdingState$.outboundSchema = exports.LegalNaturalPersonWithholdingState$outboundSchema;
})(LegalNaturalPersonWithholdingState$ || (exports.LegalNaturalPersonWithholdingState$ = LegalNaturalPersonWithholdingState$ = {}));
/** @internal */
exports.TaxProfile$inboundSchema = z.object({
    c_notice_date: z.nullable(z.lazy(() => exports.CNoticeDate$inboundSchema)).optional(),
    federal_tax_classification: exports.LegalNaturalPersonFederalTaxClassification$inboundSchema.optional(),
    first_b_notice_date: z.nullable(z.lazy(() => exports.FirstBNoticeDate$inboundSchema))
        .optional(),
    irs_form_type: exports.LegalNaturalPersonIrsFormType$inboundSchema.optional(),
    legal_tax_region_code: z.string().optional(),
    reporting_eligibility: exports.ReportingEligibility$inboundSchema.optional(),
    tax_certification_date: z.nullable(z.lazy(() => exports.TaxCertificationDate$inboundSchema)).optional(),
    taxpayer_certification_state: exports.TaxpayerCertificationState$inboundSchema
        .optional(),
    us_tin_status: exports.LegalNaturalPersonUsTinStatus$inboundSchema.optional(),
    withholding_state: exports.LegalNaturalPersonWithholdingState$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "c_notice_date": "cNoticeDate",
        "federal_tax_classification": "federalTaxClassification",
        "first_b_notice_date": "firstBNoticeDate",
        "irs_form_type": "irsFormType",
        "legal_tax_region_code": "legalTaxRegionCode",
        "reporting_eligibility": "reportingEligibility",
        "tax_certification_date": "taxCertificationDate",
        "taxpayer_certification_state": "taxpayerCertificationState",
        "us_tin_status": "usTinStatus",
        "withholding_state": "withholdingState",
    });
});
/** @internal */
exports.TaxProfile$outboundSchema = z.object({
    cNoticeDate: z.nullable(z.lazy(() => exports.CNoticeDate$outboundSchema)).optional(),
    federalTaxClassification: exports.LegalNaturalPersonFederalTaxClassification$outboundSchema.optional(),
    firstBNoticeDate: z.nullable(z.lazy(() => exports.FirstBNoticeDate$outboundSchema))
        .optional(),
    irsFormType: exports.LegalNaturalPersonIrsFormType$outboundSchema.optional(),
    legalTaxRegionCode: z.string().optional(),
    reportingEligibility: exports.ReportingEligibility$outboundSchema.optional(),
    taxCertificationDate: z.nullable(z.lazy(() => exports.TaxCertificationDate$outboundSchema)).optional(),
    taxpayerCertificationState: exports.TaxpayerCertificationState$outboundSchema
        .optional(),
    usTinStatus: exports.LegalNaturalPersonUsTinStatus$outboundSchema.optional(),
    withholdingState: exports.LegalNaturalPersonWithholdingState$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        cNoticeDate: "c_notice_date",
        federalTaxClassification: "federal_tax_classification",
        firstBNoticeDate: "first_b_notice_date",
        irsFormType: "irs_form_type",
        legalTaxRegionCode: "legal_tax_region_code",
        reportingEligibility: "reporting_eligibility",
        taxCertificationDate: "tax_certification_date",
        taxpayerCertificationState: "taxpayer_certification_state",
        usTinStatus: "us_tin_status",
        withholdingState: "withholding_state",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TaxProfile$;
(function (TaxProfile$) {
    /** @deprecated use `TaxProfile$inboundSchema` instead. */
    TaxProfile$.inboundSchema = exports.TaxProfile$inboundSchema;
    /** @deprecated use `TaxProfile$outboundSchema` instead. */
    TaxProfile$.outboundSchema = exports.TaxProfile$outboundSchema;
})(TaxProfile$ || (exports.TaxProfile$ = TaxProfile$ = {}));
/** @internal */
exports.LegalNaturalPerson$inboundSchema = z.object({
    accredited_investor: z.boolean().optional(),
    adviser: z.boolean().optional(),
    birth_date: z.nullable(z.lazy(() => exports.BirthDate$inboundSchema)).optional(),
    citizenship_countries: z.array(z.string()).optional(),
    control_person_company_symbols: z.string().optional(),
    correspondent_employee: z.boolean().optional(),
    correspondent_id: z.string().optional(),
    custodian_employee: z.boolean().optional(),
    death_date: z.nullable(z.lazy(() => exports.DeathDate$inboundSchema)).optional(),
    doing_business_as: z.array(z.string()).optional(),
    employment: z.nullable(z.lazy(() => exports.Employment$inboundSchema)).optional(),
    family_name: z.string().optional(),
    finra_associated_entity: z.string().optional(),
    foreign_identification: z.nullable(z.lazy(() => exports.ForeignIdentification$inboundSchema)).optional(),
    given_name: z.string().optional(),
    global_person_id: z.string().optional(),
    identity_verification_result: z.nullable(z.lazy(() => exports.LegalNaturalPersonIdentityVerificationResult$inboundSchema)).optional(),
    institutional_customer: z.boolean().optional(),
    investigation_id: z.string().optional(),
    large_trader: z.nullable(z.lazy(() => exports.LegalNaturalPersonLargeTrader$inboundSchema)).optional(),
    legal_natural_person_id: z.string().optional(),
    marital_status: exports.LegalNaturalPersonMaritalStatus$inboundSchema.optional(),
    middle_names: z.string().optional(),
    name: z.string().optional(),
    name_suffix: exports.LegalNaturalPersonNameSuffix$inboundSchema.optional(),
    natural_person_fdd: z.nullable(z.lazy(() => exports.NaturalPersonFdd$inboundSchema))
        .optional(),
    non_citizen_residency: z.nullable(z.lazy(() => exports.NonCitizenResidency$inboundSchema)).optional(),
    personal_address: z.nullable(z.lazy(() => exports.PersonalAddress$inboundSchema))
        .optional(),
    politically_exposed_immediate_family_names: z.array(z.string()).optional(),
    politically_exposed_organization: z.string().optional(),
    tax_id: z.string().optional(),
    tax_id_last_four: z.string().optional(),
    tax_id_type: exports.LegalNaturalPersonTaxIdType$inboundSchema.optional(),
    tax_profile: z.nullable(z.lazy(() => exports.TaxProfile$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "accredited_investor": "accreditedInvestor",
        "birth_date": "birthDate",
        "citizenship_countries": "citizenshipCountries",
        "control_person_company_symbols": "controlPersonCompanySymbols",
        "correspondent_employee": "correspondentEmployee",
        "correspondent_id": "correspondentId",
        "custodian_employee": "custodianEmployee",
        "death_date": "deathDate",
        "doing_business_as": "doingBusinessAs",
        "family_name": "familyName",
        "finra_associated_entity": "finraAssociatedEntity",
        "foreign_identification": "foreignIdentification",
        "given_name": "givenName",
        "global_person_id": "globalPersonId",
        "identity_verification_result": "identityVerificationResult",
        "institutional_customer": "institutionalCustomer",
        "investigation_id": "investigationId",
        "large_trader": "largeTrader",
        "legal_natural_person_id": "legalNaturalPersonId",
        "marital_status": "maritalStatus",
        "middle_names": "middleNames",
        "name_suffix": "nameSuffix",
        "natural_person_fdd": "naturalPersonFdd",
        "non_citizen_residency": "nonCitizenResidency",
        "personal_address": "personalAddress",
        "politically_exposed_immediate_family_names": "politicallyExposedImmediateFamilyNames",
        "politically_exposed_organization": "politicallyExposedOrganization",
        "tax_id": "taxId",
        "tax_id_last_four": "taxIdLastFour",
        "tax_id_type": "taxIdType",
        "tax_profile": "taxProfile",
    });
});
/** @internal */
exports.LegalNaturalPerson$outboundSchema = z.object({
    accreditedInvestor: z.boolean().optional(),
    adviser: z.boolean().optional(),
    birthDate: z.nullable(z.lazy(() => exports.BirthDate$outboundSchema)).optional(),
    citizenshipCountries: z.array(z.string()).optional(),
    controlPersonCompanySymbols: z.string().optional(),
    correspondentEmployee: z.boolean().optional(),
    correspondentId: z.string().optional(),
    custodianEmployee: z.boolean().optional(),
    deathDate: z.nullable(z.lazy(() => exports.DeathDate$outboundSchema)).optional(),
    doingBusinessAs: z.array(z.string()).optional(),
    employment: z.nullable(z.lazy(() => exports.Employment$outboundSchema)).optional(),
    familyName: z.string().optional(),
    finraAssociatedEntity: z.string().optional(),
    foreignIdentification: z.nullable(z.lazy(() => exports.ForeignIdentification$outboundSchema)).optional(),
    givenName: z.string().optional(),
    globalPersonId: z.string().optional(),
    identityVerificationResult: z.nullable(z.lazy(() => exports.LegalNaturalPersonIdentityVerificationResult$outboundSchema)).optional(),
    institutionalCustomer: z.boolean().optional(),
    investigationId: z.string().optional(),
    largeTrader: z.nullable(z.lazy(() => exports.LegalNaturalPersonLargeTrader$outboundSchema)).optional(),
    legalNaturalPersonId: z.string().optional(),
    maritalStatus: exports.LegalNaturalPersonMaritalStatus$outboundSchema.optional(),
    middleNames: z.string().optional(),
    name: z.string().optional(),
    nameSuffix: exports.LegalNaturalPersonNameSuffix$outboundSchema.optional(),
    naturalPersonFdd: z.nullable(z.lazy(() => exports.NaturalPersonFdd$outboundSchema))
        .optional(),
    nonCitizenResidency: z.nullable(z.lazy(() => exports.NonCitizenResidency$outboundSchema)).optional(),
    personalAddress: z.nullable(z.lazy(() => exports.PersonalAddress$outboundSchema))
        .optional(),
    politicallyExposedImmediateFamilyNames: z.array(z.string()).optional(),
    politicallyExposedOrganization: z.string().optional(),
    taxId: z.string().optional(),
    taxIdLastFour: z.string().optional(),
    taxIdType: exports.LegalNaturalPersonTaxIdType$outboundSchema.optional(),
    taxProfile: z.nullable(z.lazy(() => exports.TaxProfile$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accreditedInvestor: "accredited_investor",
        birthDate: "birth_date",
        citizenshipCountries: "citizenship_countries",
        controlPersonCompanySymbols: "control_person_company_symbols",
        correspondentEmployee: "correspondent_employee",
        correspondentId: "correspondent_id",
        custodianEmployee: "custodian_employee",
        deathDate: "death_date",
        doingBusinessAs: "doing_business_as",
        familyName: "family_name",
        finraAssociatedEntity: "finra_associated_entity",
        foreignIdentification: "foreign_identification",
        givenName: "given_name",
        globalPersonId: "global_person_id",
        identityVerificationResult: "identity_verification_result",
        institutionalCustomer: "institutional_customer",
        investigationId: "investigation_id",
        largeTrader: "large_trader",
        legalNaturalPersonId: "legal_natural_person_id",
        maritalStatus: "marital_status",
        middleNames: "middle_names",
        nameSuffix: "name_suffix",
        naturalPersonFdd: "natural_person_fdd",
        nonCitizenResidency: "non_citizen_residency",
        personalAddress: "personal_address",
        politicallyExposedImmediateFamilyNames: "politically_exposed_immediate_family_names",
        politicallyExposedOrganization: "politically_exposed_organization",
        taxId: "tax_id",
        taxIdLastFour: "tax_id_last_four",
        taxIdType: "tax_id_type",
        taxProfile: "tax_profile",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalNaturalPerson$;
(function (LegalNaturalPerson$) {
    /** @deprecated use `LegalNaturalPerson$inboundSchema` instead. */
    LegalNaturalPerson$.inboundSchema = exports.LegalNaturalPerson$inboundSchema;
    /** @deprecated use `LegalNaturalPerson$outboundSchema` instead. */
    LegalNaturalPerson$.outboundSchema = exports.LegalNaturalPerson$outboundSchema;
})(LegalNaturalPerson$ || (exports.LegalNaturalPerson$ = LegalNaturalPerson$ = {}));
