// @apexfintechsolutions/ascend-sdk/models/components/legalentitycreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalEntityCreate$ = exports.LegalEntityCreate$outboundSchema = exports.LegalEntityCreate$inboundSchema = exports.LegalEntityCreateTaxIdType$ = exports.LegalEntityCreateTaxIdType$outboundSchema = exports.LegalEntityCreateTaxIdType$inboundSchema = exports.ExemptCustomerReason$ = exports.ExemptCustomerReason$outboundSchema = exports.ExemptCustomerReason$inboundSchema = exports.EntityType$ = exports.EntityType$outboundSchema = exports.EntityType$inboundSchema = exports.CorporateStructure$ = exports.CorporateStructure$outboundSchema = exports.CorporateStructure$inboundSchema = exports.BusinessIndustrialClassification$ = exports.BusinessIndustrialClassification$outboundSchema = exports.BusinessIndustrialClassification$inboundSchema = exports.LegalEntityCreateTaxIdType = exports.ExemptCustomerReason = exports.EntityType = exports.CorporateStructure = exports.BusinessIndustrialClassification = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const datecreate_js_1 = require("./datecreate.js");
const entityduediligencecreate_js_1 = require("./entityduediligencecreate.js");
const largetradercreate_js_1 = require("./largetradercreate.js");
const postaladdresscreate_js_1 = require("./postaladdresscreate.js");
const taxprofilecreate_js_1 = require("./taxprofilecreate.js");
var BusinessIndustrialClassification;
(function (BusinessIndustrialClassification) {
    BusinessIndustrialClassification["BusinessIndustrialClassificationUnspecified"] = "BUSINESS_INDUSTRIAL_CLASSIFICATION_UNSPECIFIED";
    BusinessIndustrialClassification["AgricultureForestryAndFishing"] = "AGRICULTURE_FORESTRY_AND_FISHING";
    BusinessIndustrialClassification["Mining"] = "MINING";
    BusinessIndustrialClassification["Construction"] = "CONSTRUCTION";
    BusinessIndustrialClassification["Manufacturing"] = "MANUFACTURING";
    BusinessIndustrialClassification["TransportationCommunicationsElectricGasAndSanitaryServices"] = "TRANSPORTATION_COMMUNICATIONS_ELECTRIC_GAS_AND_SANITARY_SERVICES";
    BusinessIndustrialClassification["WholesaleTrade"] = "WHOLESALE_TRADE";
    BusinessIndustrialClassification["RetailTrade"] = "RETAIL_TRADE";
    BusinessIndustrialClassification["FinanceInsuranceAndRealEstate"] = "FINANCE_INSURANCE_AND_REAL_ESTATE";
    BusinessIndustrialClassification["Services"] = "SERVICES";
    BusinessIndustrialClassification["PublicAdministration"] = "PUBLIC_ADMINISTRATION";
})(BusinessIndustrialClassification || (exports.BusinessIndustrialClassification = BusinessIndustrialClassification = {}));
/**
 * Corporate structure of the entity.
 */
var CorporateStructure;
(function (CorporateStructure) {
    CorporateStructure["EntityCorporateStructureUnspecified"] = "ENTITY_CORPORATE_STRUCTURE_UNSPECIFIED";
    CorporateStructure["CorporationCCorp"] = "CORPORATION_C_CORP";
    CorporateStructure["CorporationSCorp"] = "CORPORATION_S_CORP";
    CorporateStructure["CorporationBCorp"] = "CORPORATION_B_CORP";
    CorporateStructure["CorporationNonprofit"] = "CORPORATION_NONPROFIT";
})(CorporateStructure || (exports.CorporateStructure = CorporateStructure = {}));
/**
 * The entity type.
 */
var EntityType;
(function (EntityType) {
    EntityType["EntityTypeUnspecified"] = "ENTITY_TYPE_UNSPECIFIED";
    EntityType["Corporation"] = "CORPORATION";
    EntityType["LimitedLiabilityCompany"] = "LIMITED_LIABILITY_COMPANY";
    EntityType["Partnership"] = "PARTNERSHIP";
    EntityType["Trust"] = "TRUST";
    EntityType["Estate"] = "ESTATE";
})(EntityType || (exports.EntityType = EntityType = {}));
/**
 * The reason the customer is exempt from verifying beneficial owners, if applicable.
 */
var ExemptCustomerReason;
(function (ExemptCustomerReason) {
    ExemptCustomerReason["ExemptReasonUnspecified"] = "EXEMPT_REASON_UNSPECIFIED";
    ExemptCustomerReason["RegulatedFinancialInstitution"] = "REGULATED_FINANCIAL_INSTITUTION";
    ExemptCustomerReason["DepartmentOrAgencyOfFederalStateOrSubdivision"] = "DEPARTMENT_OR_AGENCY_OF_FEDERAL_STATE_OR_SUBDIVISION";
    ExemptCustomerReason["NonBankListedEntity"] = "NON_BANK_LISTED_ENTITY";
    ExemptCustomerReason["Section12SecuritiesExchangeAct1934Or15D"] = "SECTION_12_SECURITIES_EXCHANGE_ACT_1934_OR_15D";
    ExemptCustomerReason["Section3InvestmentCompanyAct1940"] = "SECTION_3_INVESTMENT_COMPANY_ACT_1940";
    ExemptCustomerReason["Section202AInvestmentAdvisorsAct1940"] = "SECTION_202A_INVESTMENT_ADVISORS_ACT_1940";
    ExemptCustomerReason["Section3SecuritiesExchangeAct1934Section6Or17A"] = "SECTION_3_SECURITIES_EXCHANGE_ACT_1934_SECTION_6_OR_17A";
    ExemptCustomerReason["AnyOtherSecuritiesExchangeAct1934"] = "ANY_OTHER_SECURITIES_EXCHANGE_ACT_1934";
    ExemptCustomerReason["CommodityFuturesTradingCommissionRegistered"] = "COMMODITY_FUTURES_TRADING_COMMISSION_REGISTERED";
    ExemptCustomerReason["PublicAccountingFirmSection102SarbanesOxley"] = "PUBLIC_ACCOUNTING_FIRM_SECTION_102_SARBANES_OXLEY";
    ExemptCustomerReason["StateRegulatedInsuranceCompany"] = "STATE_REGULATED_INSURANCE_COMPANY";
})(ExemptCustomerReason || (exports.ExemptCustomerReason = ExemptCustomerReason = {}));
/**
 * The nature of the U.S. Tax ID indicated in the related tax_id field; Examples include ITIN, SSN, EIN.
 */
var LegalEntityCreateTaxIdType;
(function (LegalEntityCreateTaxIdType) {
    LegalEntityCreateTaxIdType["TaxIdTypeUnspecified"] = "TAX_ID_TYPE_UNSPECIFIED";
    LegalEntityCreateTaxIdType["TaxIdTypeSsn"] = "TAX_ID_TYPE_SSN";
    LegalEntityCreateTaxIdType["TaxIdTypeItin"] = "TAX_ID_TYPE_ITIN";
    LegalEntityCreateTaxIdType["TaxIdTypeEin"] = "TAX_ID_TYPE_EIN";
})(LegalEntityCreateTaxIdType || (exports.LegalEntityCreateTaxIdType = LegalEntityCreateTaxIdType = {}));
/** @internal */
exports.BusinessIndustrialClassification$inboundSchema = z
    .union([
    z.nativeEnum(BusinessIndustrialClassification),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.BusinessIndustrialClassification$outboundSchema = z.union([
    z.nativeEnum(BusinessIndustrialClassification),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BusinessIndustrialClassification$;
(function (BusinessIndustrialClassification$) {
    /** @deprecated use `BusinessIndustrialClassification$inboundSchema` instead. */
    BusinessIndustrialClassification$.inboundSchema = exports.BusinessIndustrialClassification$inboundSchema;
    /** @deprecated use `BusinessIndustrialClassification$outboundSchema` instead. */
    BusinessIndustrialClassification$.outboundSchema = exports.BusinessIndustrialClassification$outboundSchema;
})(BusinessIndustrialClassification$ || (exports.BusinessIndustrialClassification$ = BusinessIndustrialClassification$ = {}));
/** @internal */
exports.CorporateStructure$inboundSchema = z
    .union([
    z.nativeEnum(CorporateStructure),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.CorporateStructure$outboundSchema = z.union([
    z.nativeEnum(CorporateStructure),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CorporateStructure$;
(function (CorporateStructure$) {
    /** @deprecated use `CorporateStructure$inboundSchema` instead. */
    CorporateStructure$.inboundSchema = exports.CorporateStructure$inboundSchema;
    /** @deprecated use `CorporateStructure$outboundSchema` instead. */
    CorporateStructure$.outboundSchema = exports.CorporateStructure$outboundSchema;
})(CorporateStructure$ || (exports.CorporateStructure$ = CorporateStructure$ = {}));
/** @internal */
exports.EntityType$inboundSchema = z
    .union([
    z.nativeEnum(EntityType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.EntityType$outboundSchema = z.union([
    z.nativeEnum(EntityType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EntityType$;
(function (EntityType$) {
    /** @deprecated use `EntityType$inboundSchema` instead. */
    EntityType$.inboundSchema = exports.EntityType$inboundSchema;
    /** @deprecated use `EntityType$outboundSchema` instead. */
    EntityType$.outboundSchema = exports.EntityType$outboundSchema;
})(EntityType$ || (exports.EntityType$ = EntityType$ = {}));
/** @internal */
exports.ExemptCustomerReason$inboundSchema = z
    .union([
    z.nativeEnum(ExemptCustomerReason),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ExemptCustomerReason$outboundSchema = z.union([
    z.nativeEnum(ExemptCustomerReason),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ExemptCustomerReason$;
(function (ExemptCustomerReason$) {
    /** @deprecated use `ExemptCustomerReason$inboundSchema` instead. */
    ExemptCustomerReason$.inboundSchema = exports.ExemptCustomerReason$inboundSchema;
    /** @deprecated use `ExemptCustomerReason$outboundSchema` instead. */
    ExemptCustomerReason$.outboundSchema = exports.ExemptCustomerReason$outboundSchema;
})(ExemptCustomerReason$ || (exports.ExemptCustomerReason$ = ExemptCustomerReason$ = {}));
/** @internal */
exports.LegalEntityCreateTaxIdType$inboundSchema = z
    .union([
    z.nativeEnum(LegalEntityCreateTaxIdType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LegalEntityCreateTaxIdType$outboundSchema = z.union([
    z.nativeEnum(LegalEntityCreateTaxIdType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalEntityCreateTaxIdType$;
(function (LegalEntityCreateTaxIdType$) {
    /** @deprecated use `LegalEntityCreateTaxIdType$inboundSchema` instead. */
    LegalEntityCreateTaxIdType$.inboundSchema = exports.LegalEntityCreateTaxIdType$inboundSchema;
    /** @deprecated use `LegalEntityCreateTaxIdType$outboundSchema` instead. */
    LegalEntityCreateTaxIdType$.outboundSchema = exports.LegalEntityCreateTaxIdType$outboundSchema;
})(LegalEntityCreateTaxIdType$ || (exports.LegalEntityCreateTaxIdType$ = LegalEntityCreateTaxIdType$ = {}));
/** @internal */
exports.LegalEntityCreate$inboundSchema = z.object({
    accredited_investor: z.boolean().optional(),
    adviser: z.boolean().optional(),
    broker_dealer: z.boolean().optional(),
    business_industrial_classification: exports.BusinessIndustrialClassification$inboundSchema.optional(),
    corporate_structure: exports.CorporateStructure$inboundSchema.optional(),
    correspondent_id: z.string(),
    doing_business_as: z.array(z.string()).optional(),
    entity_due_diligence: entityduediligencecreate_js_1.EntityDueDiligenceCreate$inboundSchema.optional(),
    entity_name: z.string(),
    entity_type: exports.EntityType$inboundSchema,
    exempt_customer_reason: exports.ExemptCustomerReason$inboundSchema.optional(),
    exempt_verifying_beneficial_owners: z.boolean().optional(),
    for_the_benefit_of: z.string().optional(),
    foreign_financial_institution: z.boolean().optional(),
    formation_date: datecreate_js_1.DateCreate$inboundSchema.optional(),
    institutional_customer: z.boolean().optional(),
    large_trader: largetradercreate_js_1.LargeTraderCreate$inboundSchema.optional(),
    legal_address: postaladdresscreate_js_1.PostalAddressCreate$inboundSchema,
    lei_code: z.string().optional(),
    operating_regions: z.array(z.string()),
    registration_region: z.string(),
    regulated_investment_company: z.boolean().optional(),
    related_document_ids: z.array(z.string()).optional(),
    revocable_trust: z.boolean().optional(),
    subject_to_backup_withholding: z.boolean().optional(),
    tax_id: z.string(),
    tax_id_type: exports.LegalEntityCreateTaxIdType$inboundSchema,
    tax_profile: taxprofilecreate_js_1.TaxProfileCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "accredited_investor": "accreditedInvestor",
        "broker_dealer": "brokerDealer",
        "business_industrial_classification": "businessIndustrialClassification",
        "corporate_structure": "corporateStructure",
        "correspondent_id": "correspondentId",
        "doing_business_as": "doingBusinessAs",
        "entity_due_diligence": "entityDueDiligence",
        "entity_name": "entityName",
        "entity_type": "entityType",
        "exempt_customer_reason": "exemptCustomerReason",
        "exempt_verifying_beneficial_owners": "exemptVerifyingBeneficialOwners",
        "for_the_benefit_of": "forTheBenefitOf",
        "foreign_financial_institution": "foreignFinancialInstitution",
        "formation_date": "formationDate",
        "institutional_customer": "institutionalCustomer",
        "large_trader": "largeTrader",
        "legal_address": "legalAddress",
        "lei_code": "leiCode",
        "operating_regions": "operatingRegions",
        "registration_region": "registrationRegion",
        "regulated_investment_company": "regulatedInvestmentCompany",
        "related_document_ids": "relatedDocumentIds",
        "revocable_trust": "revocableTrust",
        "subject_to_backup_withholding": "subjectToBackupWithholding",
        "tax_id": "taxId",
        "tax_id_type": "taxIdType",
        "tax_profile": "taxProfile",
    });
});
/** @internal */
exports.LegalEntityCreate$outboundSchema = z.object({
    accreditedInvestor: z.boolean().optional(),
    adviser: z.boolean().optional(),
    brokerDealer: z.boolean().optional(),
    businessIndustrialClassification: exports.BusinessIndustrialClassification$outboundSchema.optional(),
    corporateStructure: exports.CorporateStructure$outboundSchema.optional(),
    correspondentId: z.string(),
    doingBusinessAs: z.array(z.string()).optional(),
    entityDueDiligence: entityduediligencecreate_js_1.EntityDueDiligenceCreate$outboundSchema.optional(),
    entityName: z.string(),
    entityType: exports.EntityType$outboundSchema,
    exemptCustomerReason: exports.ExemptCustomerReason$outboundSchema.optional(),
    exemptVerifyingBeneficialOwners: z.boolean().optional(),
    forTheBenefitOf: z.string().optional(),
    foreignFinancialInstitution: z.boolean().optional(),
    formationDate: datecreate_js_1.DateCreate$outboundSchema.optional(),
    institutionalCustomer: z.boolean().optional(),
    largeTrader: largetradercreate_js_1.LargeTraderCreate$outboundSchema.optional(),
    legalAddress: postaladdresscreate_js_1.PostalAddressCreate$outboundSchema,
    leiCode: z.string().optional(),
    operatingRegions: z.array(z.string()),
    registrationRegion: z.string(),
    regulatedInvestmentCompany: z.boolean().optional(),
    relatedDocumentIds: z.array(z.string()).optional(),
    revocableTrust: z.boolean().optional(),
    subjectToBackupWithholding: z.boolean().optional(),
    taxId: z.string(),
    taxIdType: exports.LegalEntityCreateTaxIdType$outboundSchema,
    taxProfile: taxprofilecreate_js_1.TaxProfileCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accreditedInvestor: "accredited_investor",
        brokerDealer: "broker_dealer",
        businessIndustrialClassification: "business_industrial_classification",
        corporateStructure: "corporate_structure",
        correspondentId: "correspondent_id",
        doingBusinessAs: "doing_business_as",
        entityDueDiligence: "entity_due_diligence",
        entityName: "entity_name",
        entityType: "entity_type",
        exemptCustomerReason: "exempt_customer_reason",
        exemptVerifyingBeneficialOwners: "exempt_verifying_beneficial_owners",
        forTheBenefitOf: "for_the_benefit_of",
        foreignFinancialInstitution: "foreign_financial_institution",
        formationDate: "formation_date",
        institutionalCustomer: "institutional_customer",
        largeTrader: "large_trader",
        legalAddress: "legal_address",
        leiCode: "lei_code",
        operatingRegions: "operating_regions",
        registrationRegion: "registration_region",
        regulatedInvestmentCompany: "regulated_investment_company",
        relatedDocumentIds: "related_document_ids",
        revocableTrust: "revocable_trust",
        subjectToBackupWithholding: "subject_to_backup_withholding",
        taxId: "tax_id",
        taxIdType: "tax_id_type",
        taxProfile: "tax_profile",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LegalEntityCreate$;
(function (LegalEntityCreate$) {
    /** @deprecated use `LegalEntityCreate$inboundSchema` instead. */
    LegalEntityCreate$.inboundSchema = exports.LegalEntityCreate$inboundSchema;
    /** @deprecated use `LegalEntityCreate$outboundSchema` instead. */
    LegalEntityCreate$.outboundSchema = exports.LegalEntityCreate$outboundSchema;
})(LegalEntityCreate$ || (exports.LegalEntityCreate$ = LegalEntityCreate$ = {}));
