// @apexfintechsolutions/ascend-sdk/models/components/limitpricecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LimitPriceCreate$ = exports.LimitPriceCreate$outboundSchema = exports.LimitPriceCreate$inboundSchema = exports.LimitPriceCreateType$ = exports.LimitPriceCreateType$outboundSchema = exports.LimitPriceCreateType$inboundSchema = exports.LimitPriceCreateType = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
/**
 * The type of this price, which must be PRICE_PER_UNIT for equity orders, or PERCENTAGE_OF_PAR for fixed income orders.
 */
var LimitPriceCreateType;
(function (LimitPriceCreateType) {
    LimitPriceCreateType["LimitPriceTypeUnspecified"] = "LIMIT_PRICE_TYPE_UNSPECIFIED";
    LimitPriceCreateType["PricePerUnit"] = "PRICE_PER_UNIT";
    LimitPriceCreateType["PercentageOfPar"] = "PERCENTAGE_OF_PAR";
})(LimitPriceCreateType || (exports.LimitPriceCreateType = LimitPriceCreateType = {}));
/** @internal */
exports.LimitPriceCreateType$inboundSchema = z
    .union([
    z.nativeEnum(LimitPriceCreateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LimitPriceCreateType$outboundSchema = z.union([
    z.nativeEnum(LimitPriceCreateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LimitPriceCreateType$;
(function (LimitPriceCreateType$) {
    /** @deprecated use `LimitPriceCreateType$inboundSchema` instead. */
    LimitPriceCreateType$.inboundSchema = exports.LimitPriceCreateType$inboundSchema;
    /** @deprecated use `LimitPriceCreateType$outboundSchema` instead. */
    LimitPriceCreateType$.outboundSchema = exports.LimitPriceCreateType$outboundSchema;
})(LimitPriceCreateType$ || (exports.LimitPriceCreateType$ = LimitPriceCreateType$ = {}));
/** @internal */
exports.LimitPriceCreate$inboundSchema = z.object({
    price: decimalcreate_js_1.DecimalCreate$inboundSchema,
    type: exports.LimitPriceCreateType$inboundSchema,
});
/** @internal */
exports.LimitPriceCreate$outboundSchema = z.object({
    price: decimalcreate_js_1.DecimalCreate$outboundSchema,
    type: exports.LimitPriceCreateType$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LimitPriceCreate$;
(function (LimitPriceCreate$) {
    /** @deprecated use `LimitPriceCreate$inboundSchema` instead. */
    LimitPriceCreate$.inboundSchema = exports.LimitPriceCreate$inboundSchema;
    /** @deprecated use `LimitPriceCreate$outboundSchema` instead. */
    LimitPriceCreate$.outboundSchema = exports.LimitPriceCreate$outboundSchema;
})(LimitPriceCreate$ || (exports.LimitPriceCreate$ = LimitPriceCreate$ = {}));
