// @apexfintechsolutions/ascend-sdk/models/components/llcenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLCEnrollmentMetadataCreate$ = exports.LLCEnrollmentMetadataCreate$outboundSchema = exports.LLCEnrollmentMetadataCreate$inboundSchema = exports.LLCEnrollmentMetadataCreateFdicCashSweep$ = exports.LLCEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.LLCEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.LLCEnrollmentMetadataCreateFdicCashSweep = exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const eddaccountenrollmentmetadatacreate_js_1 = require("./eddaccountenrollmentmetadatacreate.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var LLCEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (LLCEnrollmentMetadataCreateDividendReinvestmentPlan) {
    LLCEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    LLCEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    LLCEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(LLCEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan = LLCEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var LLCEnrollmentMetadataCreateFdicCashSweep;
(function (LLCEnrollmentMetadataCreateFdicCashSweep) {
    LLCEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    LLCEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    LLCEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(LLCEnrollmentMetadataCreateFdicCashSweep || (exports.LLCEnrollmentMetadataCreateFdicCashSweep = LLCEnrollmentMetadataCreateFdicCashSweep = {}));
/** @internal */
exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(LLCEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(LLCEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LLCEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (LLCEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `LLCEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    LLCEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `LLCEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    LLCEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(LLCEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan$ = LLCEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.LLCEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(LLCEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.LLCEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(LLCEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LLCEnrollmentMetadataCreateFdicCashSweep$;
(function (LLCEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `LLCEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    LLCEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.LLCEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `LLCEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    LLCEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.LLCEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(LLCEnrollmentMetadataCreateFdicCashSweep$ || (exports.LLCEnrollmentMetadataCreateFdicCashSweep$ = LLCEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.LLCEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    edd_account_enrollment_metadata: eddaccountenrollmentmetadatacreate_js_1.EddAccountEnrollmentMetadataCreate$inboundSchema.optional(),
    fdic_cash_sweep: exports.LLCEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "edd_account_enrollment_metadata": "eddAccountEnrollmentMetadata",
        "fdic_cash_sweep": "fdicCashSweep",
    });
});
/** @internal */
exports.LLCEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.LLCEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    eddAccountEnrollmentMetadata: eddaccountenrollmentmetadatacreate_js_1.EddAccountEnrollmentMetadataCreate$outboundSchema.optional(),
    fdicCashSweep: exports.LLCEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        eddAccountEnrollmentMetadata: "edd_account_enrollment_metadata",
        fdicCashSweep: "fdic_cash_sweep",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LLCEnrollmentMetadataCreate$;
(function (LLCEnrollmentMetadataCreate$) {
    /** @deprecated use `LLCEnrollmentMetadataCreate$inboundSchema` instead. */
    LLCEnrollmentMetadataCreate$.inboundSchema = exports.LLCEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `LLCEnrollmentMetadataCreate$outboundSchema` instead. */
    LLCEnrollmentMetadataCreate$.outboundSchema = exports.LLCEnrollmentMetadataCreate$outboundSchema;
})(LLCEnrollmentMetadataCreate$ || (exports.LLCEnrollmentMetadataCreate$ = LLCEnrollmentMetadataCreate$ = {}));
