// @apexfintechsolutions/ascend-sdk/models/components/lotcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LotCreate$ = exports.LotCreate$outboundSchema = exports.LotCreate$inboundSchema = exports.Money$ = exports.Money$outboundSchema = exports.Money$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const datecreate_js_1 = require("./datecreate.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
/** @internal */
exports.Money$inboundSchema = z
    .object({
    currency_code: z.string().optional(),
    price: decimalcreate_js_1.DecimalCreate$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "currency_code": "currencyCode",
    });
});
/** @internal */
exports.Money$outboundSchema = z.object({
    currencyCode: z.string().optional(),
    price: decimalcreate_js_1.DecimalCreate$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        currencyCode: "currency_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Money$;
(function (Money$) {
    /** @deprecated use `Money$inboundSchema` instead. */
    Money$.inboundSchema = exports.Money$inboundSchema;
    /** @deprecated use `Money$outboundSchema` instead. */
    Money$.outboundSchema = exports.Money$outboundSchema;
})(Money$ || (exports.Money$ = Money$ = {}));
/** @internal */
exports.LotCreate$inboundSchema = z.object({
    id: z.string().optional(),
    money: z.lazy(() => exports.Money$inboundSchema).optional(),
    price: decimalcreate_js_1.DecimalCreate$inboundSchema.optional(),
    quantity: decimalcreate_js_1.DecimalCreate$inboundSchema.optional(),
    trade_date: datecreate_js_1.DateCreate$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "trade_date": "tradeDate",
    });
});
/** @internal */
exports.LotCreate$outboundSchema = z.object({
    id: z.string().optional(),
    money: z.lazy(() => exports.Money$outboundSchema).optional(),
    price: decimalcreate_js_1.DecimalCreate$outboundSchema.optional(),
    quantity: decimalcreate_js_1.DecimalCreate$outboundSchema.optional(),
    tradeDate: datecreate_js_1.DateCreate$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        tradeDate: "trade_date",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LotCreate$;
(function (LotCreate$) {
    /** @deprecated use `LotCreate$inboundSchema` instead. */
    LotCreate$.inboundSchema = exports.LotCreate$inboundSchema;
    /** @deprecated use `LotCreate$outboundSchema` instead. */
    LotCreate$.outboundSchema = exports.LotCreate$outboundSchema;
})(LotCreate$ || (exports.LotCreate$ = LotCreate$ = {}));
