// @apexfintechsolutions/ascend-sdk/models/components/markprice.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MarkPrice$ = exports.MarkPrice$outboundSchema = exports.MarkPrice$inboundSchema = exports.MarkPriceType$ = exports.MarkPriceType$outboundSchema = exports.MarkPriceType$inboundSchema = exports.MarkPricePrice$ = exports.MarkPricePrice$outboundSchema = exports.MarkPricePrice$inboundSchema = exports.MarkPriceType = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
/**
 * The calculation type of this price
 */
var MarkPriceType;
(function (MarkPriceType) {
    MarkPriceType["PercentageOfPar"] = "PERCENTAGE_OF_PAR";
    MarkPriceType["DirtyPercentageOfPar"] = "DIRTY_PERCENTAGE_OF_PAR";
    MarkPriceType["YieldToWorst"] = "YIELD_TO_WORST";
    MarkPriceType["YieldToMaturity"] = "YIELD_TO_MATURITY";
})(MarkPriceType || (exports.MarkPriceType = MarkPriceType = {}));
/** @internal */
exports.MarkPricePrice$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.MarkPricePrice$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MarkPricePrice$;
(function (MarkPricePrice$) {
    /** @deprecated use `MarkPricePrice$inboundSchema` instead. */
    MarkPricePrice$.inboundSchema = exports.MarkPricePrice$inboundSchema;
    /** @deprecated use `MarkPricePrice$outboundSchema` instead. */
    MarkPricePrice$.outboundSchema = exports.MarkPricePrice$outboundSchema;
})(MarkPricePrice$ || (exports.MarkPricePrice$ = MarkPricePrice$ = {}));
/** @internal */
exports.MarkPriceType$inboundSchema = z
    .union([
    z.nativeEnum(MarkPriceType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.MarkPriceType$outboundSchema = z.union([
    z.nativeEnum(MarkPriceType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MarkPriceType$;
(function (MarkPriceType$) {
    /** @deprecated use `MarkPriceType$inboundSchema` instead. */
    MarkPriceType$.inboundSchema = exports.MarkPriceType$inboundSchema;
    /** @deprecated use `MarkPriceType$outboundSchema` instead. */
    MarkPriceType$.outboundSchema = exports.MarkPriceType$outboundSchema;
})(MarkPriceType$ || (exports.MarkPriceType$ = MarkPriceType$ = {}));
/** @internal */
exports.MarkPrice$inboundSchema = z.object({
    price: z.nullable(z.lazy(() => exports.MarkPricePrice$inboundSchema)).optional(),
    type: exports.MarkPriceType$inboundSchema.optional(),
});
/** @internal */
exports.MarkPrice$outboundSchema = z.object({
    price: z.nullable(z.lazy(() => exports.MarkPricePrice$outboundSchema)).optional(),
    type: exports.MarkPriceType$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MarkPrice$;
(function (MarkPrice$) {
    /** @deprecated use `MarkPrice$inboundSchema` instead. */
    MarkPrice$.inboundSchema = exports.MarkPrice$inboundSchema;
    /** @deprecated use `MarkPrice$outboundSchema` instead. */
    MarkPrice$.outboundSchema = exports.MarkPrice$outboundSchema;
})(MarkPrice$ || (exports.MarkPrice$ = MarkPrice$ = {}));
