// @apexfintechsolutions/ascend-sdk/models/components/nachareturncreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.NachaReturnCreate$ = exports.NachaReturnCreate$outboundSchema = exports.NachaReturnCreate$inboundSchema = exports.NachaReturnCreateCode$ = exports.NachaReturnCreateCode$outboundSchema = exports.NachaReturnCreateCode$inboundSchema = exports.NachaReturnCreateCode = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
/**
 * The cause of the return.
 */
var NachaReturnCreateCode;
(function (NachaReturnCreateCode) {
    NachaReturnCreateCode["CodeUnspecified"] = "CODE_UNSPECIFIED";
    NachaReturnCreateCode["R01"] = "R01";
    NachaReturnCreateCode["R02"] = "R02";
    NachaReturnCreateCode["R03"] = "R03";
    NachaReturnCreateCode["R04"] = "R04";
    NachaReturnCreateCode["R05"] = "R05";
    NachaReturnCreateCode["R06"] = "R06";
    NachaReturnCreateCode["R07"] = "R07";
    NachaReturnCreateCode["R08"] = "R08";
    NachaReturnCreateCode["R09"] = "R09";
    NachaReturnCreateCode["R10"] = "R10";
    NachaReturnCreateCode["R11"] = "R11";
    NachaReturnCreateCode["R12"] = "R12";
    NachaReturnCreateCode["R13"] = "R13";
    NachaReturnCreateCode["R14"] = "R14";
    NachaReturnCreateCode["R15"] = "R15";
    NachaReturnCreateCode["R16"] = "R16";
    NachaReturnCreateCode["R17"] = "R17";
    NachaReturnCreateCode["R18"] = "R18";
    NachaReturnCreateCode["R19"] = "R19";
    NachaReturnCreateCode["R20"] = "R20";
    NachaReturnCreateCode["R21"] = "R21";
    NachaReturnCreateCode["R22"] = "R22";
    NachaReturnCreateCode["R23"] = "R23";
    NachaReturnCreateCode["R24"] = "R24";
    NachaReturnCreateCode["R25"] = "R25";
    NachaReturnCreateCode["R26"] = "R26";
    NachaReturnCreateCode["R27"] = "R27";
    NachaReturnCreateCode["R28"] = "R28";
    NachaReturnCreateCode["R29"] = "R29";
    NachaReturnCreateCode["R30"] = "R30";
    NachaReturnCreateCode["R31"] = "R31";
    NachaReturnCreateCode["R32"] = "R32";
    NachaReturnCreateCode["R33"] = "R33";
    NachaReturnCreateCode["R34"] = "R34";
    NachaReturnCreateCode["R35"] = "R35";
    NachaReturnCreateCode["R36"] = "R36";
    NachaReturnCreateCode["R37"] = "R37";
    NachaReturnCreateCode["R38"] = "R38";
    NachaReturnCreateCode["R39"] = "R39";
    NachaReturnCreateCode["R40"] = "R40";
    NachaReturnCreateCode["R41"] = "R41";
    NachaReturnCreateCode["R42"] = "R42";
    NachaReturnCreateCode["R43"] = "R43";
    NachaReturnCreateCode["R44"] = "R44";
    NachaReturnCreateCode["R45"] = "R45";
    NachaReturnCreateCode["R46"] = "R46";
    NachaReturnCreateCode["R47"] = "R47";
    NachaReturnCreateCode["R50"] = "R50";
    NachaReturnCreateCode["R51"] = "R51";
    NachaReturnCreateCode["R52"] = "R52";
    NachaReturnCreateCode["R53"] = "R53";
    NachaReturnCreateCode["R61"] = "R61";
    NachaReturnCreateCode["R67"] = "R67";
    NachaReturnCreateCode["R68"] = "R68";
    NachaReturnCreateCode["R69"] = "R69";
    NachaReturnCreateCode["R70"] = "R70";
    NachaReturnCreateCode["R71"] = "R71";
    NachaReturnCreateCode["R72"] = "R72";
    NachaReturnCreateCode["R73"] = "R73";
    NachaReturnCreateCode["R74"] = "R74";
    NachaReturnCreateCode["R75"] = "R75";
    NachaReturnCreateCode["R76"] = "R76";
    NachaReturnCreateCode["R80"] = "R80";
    NachaReturnCreateCode["R81"] = "R81";
    NachaReturnCreateCode["R82"] = "R82";
    NachaReturnCreateCode["R83"] = "R83";
    NachaReturnCreateCode["R84"] = "R84";
    NachaReturnCreateCode["R85"] = "R85";
})(NachaReturnCreateCode || (exports.NachaReturnCreateCode = NachaReturnCreateCode = {}));
/** @internal */
exports.NachaReturnCreateCode$inboundSchema = z
    .union([
    z.nativeEnum(NachaReturnCreateCode),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.NachaReturnCreateCode$outboundSchema = z.union([
    z.nativeEnum(NachaReturnCreateCode),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NachaReturnCreateCode$;
(function (NachaReturnCreateCode$) {
    /** @deprecated use `NachaReturnCreateCode$inboundSchema` instead. */
    NachaReturnCreateCode$.inboundSchema = exports.NachaReturnCreateCode$inboundSchema;
    /** @deprecated use `NachaReturnCreateCode$outboundSchema` instead. */
    NachaReturnCreateCode$.outboundSchema = exports.NachaReturnCreateCode$outboundSchema;
})(NachaReturnCreateCode$ || (exports.NachaReturnCreateCode$ = NachaReturnCreateCode$ = {}));
/** @internal */
exports.NachaReturnCreate$inboundSchema = z.object({
    code: exports.NachaReturnCreateCode$inboundSchema,
});
/** @internal */
exports.NachaReturnCreate$outboundSchema = z.object({
    code: exports.NachaReturnCreateCode$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NachaReturnCreate$;
(function (NachaReturnCreate$) {
    /** @deprecated use `NachaReturnCreate$inboundSchema` instead. */
    NachaReturnCreate$.inboundSchema = exports.NachaReturnCreate$inboundSchema;
    /** @deprecated use `NachaReturnCreate$outboundSchema` instead. */
    NachaReturnCreate$.outboundSchema = exports.NachaReturnCreate$outboundSchema;
})(NachaReturnCreate$ || (exports.NachaReturnCreate$ = NachaReturnCreate$ = {}));
