// @apexfintechsolutions/ascend-sdk/models/components/namedetail.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.NameDetail$ = exports.NameDetail$outboundSchema = exports.NameDetail$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.NameDetail$inboundSchema = z.object({
    given_name: z.string().optional(),
    middle_names: z.string().optional(),
    name_suffix: z.string().optional(),
    name_type: z.string().optional(),
    surname: z.string().optional(),
    title_honorific: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "given_name": "givenName",
        "middle_names": "middleNames",
        "name_suffix": "nameSuffix",
        "name_type": "nameType",
        "title_honorific": "titleHonorific",
    });
});
/** @internal */
exports.NameDetail$outboundSchema = z.object({
    givenName: z.string().optional(),
    middleNames: z.string().optional(),
    nameSuffix: z.string().optional(),
    nameType: z.string().optional(),
    surname: z.string().optional(),
    titleHonorific: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        givenName: "given_name",
        middleNames: "middle_names",
        nameSuffix: "name_suffix",
        nameType: "name_type",
        titleHonorific: "title_honorific",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NameDetail$;
(function (NameDetail$) {
    /** @deprecated use `NameDetail$inboundSchema` instead. */
    NameDetail$.inboundSchema = exports.NameDetail$inboundSchema;
    /** @deprecated use `NameDetail$outboundSchema` instead. */
    NameDetail$.outboundSchema = exports.NameDetail$outboundSchema;
})(NameDetail$ || (exports.NameDetail$ = NameDetail$ = {}));
