// @apexfintechsolutions/ascend-sdk/models/components/naturalpersonfddupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.NaturalPersonFddUpdate$ = exports.NaturalPersonFddUpdate$outboundSchema = exports.NaturalPersonFddUpdate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const customerreferralsourceupdate_js_1 = require("./customerreferralsourceupdate.js");
const negativenewsupdate_js_1 = require("./negativenewsupdate.js");
const othersourcesofwealthupdate_js_1 = require("./othersourcesofwealthupdate.js");
/** @internal */
exports.NaturalPersonFddUpdate$inboundSchema = z.object({
    customer_non_referral_source: z.string().optional(),
    customer_referral_source: customerreferralsourceupdate_js_1.CustomerReferralSourceUpdate$inboundSchema
        .optional(),
    employment_and_employer_description: z.string().optional(),
    negative_news: negativenewsupdate_js_1.NegativeNewsUpdate$inboundSchema.optional(),
    other_sources_of_wealth: othersourcesofwealthupdate_js_1.OtherSourcesOfWealthUpdate$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_non_referral_source": "customerNonReferralSource",
        "customer_referral_source": "customerReferralSource",
        "employment_and_employer_description": "employmentAndEmployerDescription",
        "negative_news": "negativeNews",
        "other_sources_of_wealth": "otherSourcesOfWealth",
    });
});
/** @internal */
exports.NaturalPersonFddUpdate$outboundSchema = z.object({
    customerNonReferralSource: z.string().optional(),
    customerReferralSource: customerreferralsourceupdate_js_1.CustomerReferralSourceUpdate$outboundSchema
        .optional(),
    employmentAndEmployerDescription: z.string().optional(),
    negativeNews: negativenewsupdate_js_1.NegativeNewsUpdate$outboundSchema.optional(),
    otherSourcesOfWealth: othersourcesofwealthupdate_js_1.OtherSourcesOfWealthUpdate$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerNonReferralSource: "customer_non_referral_source",
        customerReferralSource: "customer_referral_source",
        employmentAndEmployerDescription: "employment_and_employer_description",
        negativeNews: "negative_news",
        otherSourcesOfWealth: "other_sources_of_wealth",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NaturalPersonFddUpdate$;
(function (NaturalPersonFddUpdate$) {
    /** @deprecated use `NaturalPersonFddUpdate$inboundSchema` instead. */
    NaturalPersonFddUpdate$.inboundSchema = exports.NaturalPersonFddUpdate$inboundSchema;
    /** @deprecated use `NaturalPersonFddUpdate$outboundSchema` instead. */
    NaturalPersonFddUpdate$.outboundSchema = exports.NaturalPersonFddUpdate$outboundSchema;
})(NaturalPersonFddUpdate$ || (exports.NaturalPersonFddUpdate$ = NaturalPersonFddUpdate$ = {}));
