// @apexfintechsolutions/ascend-sdk/models/components/noncitizenresidencyupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.NonCitizenResidencyUpdate$ = exports.NonCitizenResidencyUpdate$outboundSchema = exports.NonCitizenResidencyUpdate$inboundSchema = exports.NonCitizenResidencyUpdateResidencyStatus$ = exports.NonCitizenResidencyUpdateResidencyStatus$outboundSchema = exports.NonCitizenResidencyUpdateResidencyStatus$inboundSchema = exports.NonCitizenResidencyUpdateResidencyStatus = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
var NonCitizenResidencyUpdateResidencyStatus;
(function (NonCitizenResidencyUpdateResidencyStatus) {
    NonCitizenResidencyUpdateResidencyStatus["ResidencyStatusUnspecified"] = "RESIDENCY_STATUS_UNSPECIFIED";
    NonCitizenResidencyUpdateResidencyStatus["UsPermanentResident"] = "US_PERMANENT_RESIDENT";
    NonCitizenResidencyUpdateResidencyStatus["UsTemporaryResident"] = "US_TEMPORARY_RESIDENT";
    NonCitizenResidencyUpdateResidencyStatus["UsNonResident"] = "US_NON_RESIDENT";
})(NonCitizenResidencyUpdateResidencyStatus || (exports.NonCitizenResidencyUpdateResidencyStatus = NonCitizenResidencyUpdateResidencyStatus = {}));
/** @internal */
exports.NonCitizenResidencyUpdateResidencyStatus$inboundSchema = z
    .union([
    z.nativeEnum(NonCitizenResidencyUpdateResidencyStatus),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.NonCitizenResidencyUpdateResidencyStatus$outboundSchema = z.union([
    z.nativeEnum(NonCitizenResidencyUpdateResidencyStatus),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NonCitizenResidencyUpdateResidencyStatus$;
(function (NonCitizenResidencyUpdateResidencyStatus$) {
    /** @deprecated use `NonCitizenResidencyUpdateResidencyStatus$inboundSchema` instead. */
    NonCitizenResidencyUpdateResidencyStatus$.inboundSchema = exports.NonCitizenResidencyUpdateResidencyStatus$inboundSchema;
    /** @deprecated use `NonCitizenResidencyUpdateResidencyStatus$outboundSchema` instead. */
    NonCitizenResidencyUpdateResidencyStatus$.outboundSchema = exports.NonCitizenResidencyUpdateResidencyStatus$outboundSchema;
})(NonCitizenResidencyUpdateResidencyStatus$ || (exports.NonCitizenResidencyUpdateResidencyStatus$ = NonCitizenResidencyUpdateResidencyStatus$ = {}));
/** @internal */
exports.NonCitizenResidencyUpdate$inboundSchema = z.object({
    residency_status: exports.NonCitizenResidencyUpdateResidencyStatus$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "residency_status": "residencyStatus",
    });
});
/** @internal */
exports.NonCitizenResidencyUpdate$outboundSchema = z.object({
    residencyStatus: exports.NonCitizenResidencyUpdateResidencyStatus$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        residencyStatus: "residency_status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NonCitizenResidencyUpdate$;
(function (NonCitizenResidencyUpdate$) {
    /** @deprecated use `NonCitizenResidencyUpdate$inboundSchema` instead. */
    NonCitizenResidencyUpdate$.inboundSchema = exports.NonCitizenResidencyUpdate$inboundSchema;
    /** @deprecated use `NonCitizenResidencyUpdate$outboundSchema` instead. */
    NonCitizenResidencyUpdate$.outboundSchema = exports.NonCitizenResidencyUpdate$outboundSchema;
})(NonCitizenResidencyUpdate$ || (exports.NonCitizenResidencyUpdate$ = NonCitizenResidencyUpdate$ = {}));
