// @apexfintechsolutions/ascend-sdk/models/components/noncitizenresidencycreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.NonCitizenResidencyCreate$ = exports.NonCitizenResidencyCreate$outboundSchema = exports.NonCitizenResidencyCreate$inboundSchema = exports.ResidencyStatus$ = exports.ResidencyStatus$outboundSchema = exports.ResidencyStatus$inboundSchema = exports.ResidencyStatus = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
var ResidencyStatus;
(function (ResidencyStatus) {
    ResidencyStatus["ResidencyStatusUnspecified"] = "RESIDENCY_STATUS_UNSPECIFIED";
    ResidencyStatus["UsPermanentResident"] = "US_PERMANENT_RESIDENT";
    ResidencyStatus["UsTemporaryResident"] = "US_TEMPORARY_RESIDENT";
    ResidencyStatus["UsNonResident"] = "US_NON_RESIDENT";
})(ResidencyStatus || (exports.ResidencyStatus = ResidencyStatus = {}));
/** @internal */
exports.ResidencyStatus$inboundSchema = z
    .union([
    z.nativeEnum(ResidencyStatus),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ResidencyStatus$outboundSchema = z.union([
    z.nativeEnum(ResidencyStatus),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ResidencyStatus$;
(function (ResidencyStatus$) {
    /** @deprecated use `ResidencyStatus$inboundSchema` instead. */
    ResidencyStatus$.inboundSchema = exports.ResidencyStatus$inboundSchema;
    /** @deprecated use `ResidencyStatus$outboundSchema` instead. */
    ResidencyStatus$.outboundSchema = exports.ResidencyStatus$outboundSchema;
})(ResidencyStatus$ || (exports.ResidencyStatus$ = ResidencyStatus$ = {}));
/** @internal */
exports.NonCitizenResidencyCreate$inboundSchema = z.object({
    residency_status: exports.ResidencyStatus$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "residency_status": "residencyStatus",
    });
});
/** @internal */
exports.NonCitizenResidencyCreate$outboundSchema = z.object({
    residencyStatus: exports.ResidencyStatus$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        residencyStatus: "residency_status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var NonCitizenResidencyCreate$;
(function (NonCitizenResidencyCreate$) {
    /** @deprecated use `NonCitizenResidencyCreate$inboundSchema` instead. */
    NonCitizenResidencyCreate$.inboundSchema = exports.NonCitizenResidencyCreate$inboundSchema;
    /** @deprecated use `NonCitizenResidencyCreate$outboundSchema` instead. */
    NonCitizenResidencyCreate$.outboundSchema = exports.NonCitizenResidencyCreate$outboundSchema;
})(NonCitizenResidencyCreate$ || (exports.NonCitizenResidencyCreate$ = NonCitizenResidencyCreate$ = {}));
