// @apexfintechsolutions/ascend-sdk/models/components/ordercostpreviewresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderCostPreviewResponse$ = exports.OrderCostPreviewResponse$outboundSchema = exports.OrderCostPreviewResponse$inboundSchema = exports.OrderCostPreviewResponseQuantity$ = exports.OrderCostPreviewResponseQuantity$outboundSchema = exports.OrderCostPreviewResponseQuantity$inboundSchema = exports.OrderCostPreviewResponseLimitPrice$ = exports.OrderCostPreviewResponseLimitPrice$outboundSchema = exports.OrderCostPreviewResponseLimitPrice$inboundSchema = exports.OrderCostPreviewResponseType$ = exports.OrderCostPreviewResponseType$outboundSchema = exports.OrderCostPreviewResponseType$inboundSchema = exports.OrderCostPreviewResponsePrice$ = exports.OrderCostPreviewResponsePrice$outboundSchema = exports.OrderCostPreviewResponsePrice$inboundSchema = exports.OrderCostPreviewResponseIdentifierType$ = exports.OrderCostPreviewResponseIdentifierType$outboundSchema = exports.OrderCostPreviewResponseIdentifierType$inboundSchema = exports.EstimatedNetAmount$ = exports.EstimatedNetAmount$outboundSchema = exports.EstimatedNetAmount$inboundSchema = exports.EstimatedGrossCreditAmount$ = exports.EstimatedGrossCreditAmount$outboundSchema = exports.EstimatedGrossCreditAmount$inboundSchema = exports.EstimatedGrossAmount$ = exports.EstimatedGrossAmount$outboundSchema = exports.EstimatedGrossAmount$inboundSchema = exports.EstimatedAccruedInterest$ = exports.EstimatedAccruedInterest$outboundSchema = exports.EstimatedAccruedInterest$inboundSchema = exports.OrderCostPreviewResponseBrokerCapacity$ = exports.OrderCostPreviewResponseBrokerCapacity$outboundSchema = exports.OrderCostPreviewResponseBrokerCapacity$inboundSchema = exports.OrderCostPreviewResponseAssetType$ = exports.OrderCostPreviewResponseAssetType$outboundSchema = exports.OrderCostPreviewResponseAssetType$inboundSchema = exports.OrderCostPreviewResponseType = exports.OrderCostPreviewResponseIdentifierType = exports.OrderCostPreviewResponseBrokerCapacity = exports.OrderCostPreviewResponseAssetType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * The type of asset referenced by the security identifier
 */
var OrderCostPreviewResponseAssetType;
(function (OrderCostPreviewResponseAssetType) {
    OrderCostPreviewResponseAssetType["AssetTypeUnspecified"] = "ASSET_TYPE_UNSPECIFIED";
    OrderCostPreviewResponseAssetType["Equity"] = "EQUITY";
    OrderCostPreviewResponseAssetType["FixedIncome"] = "FIXED_INCOME";
    OrderCostPreviewResponseAssetType["MutualFund"] = "MUTUAL_FUND";
    OrderCostPreviewResponseAssetType["Option"] = "OPTION";
})(OrderCostPreviewResponseAssetType || (exports.OrderCostPreviewResponseAssetType = OrderCostPreviewResponseAssetType = {}));
/**
 * Capacity used in determining the order cost. Defaults to "AGENCY" if no value specified.
 */
var OrderCostPreviewResponseBrokerCapacity;
(function (OrderCostPreviewResponseBrokerCapacity) {
    OrderCostPreviewResponseBrokerCapacity["BrokerCapacityUnspecified"] = "BROKER_CAPACITY_UNSPECIFIED";
    OrderCostPreviewResponseBrokerCapacity["Agency"] = "AGENCY";
    OrderCostPreviewResponseBrokerCapacity["Principal"] = "PRINCIPAL";
})(OrderCostPreviewResponseBrokerCapacity || (exports.OrderCostPreviewResponseBrokerCapacity = OrderCostPreviewResponseBrokerCapacity = {}));
/**
 * The identifier type of the asset used for price calculations. This will be the same value as what was sent on the request.
 */
var OrderCostPreviewResponseIdentifierType;
(function (OrderCostPreviewResponseIdentifierType) {
    OrderCostPreviewResponseIdentifierType["AssetId"] = "ASSET_ID";
    OrderCostPreviewResponseIdentifierType["Cusip"] = "CUSIP";
    OrderCostPreviewResponseIdentifierType["Isin"] = "ISIN";
})(OrderCostPreviewResponseIdentifierType || (exports.OrderCostPreviewResponseIdentifierType = OrderCostPreviewResponseIdentifierType = {}));
/**
 * The type of this price, which must be PRICE_PER_UNIT for equity orders, or PERCENTAGE_OF_PAR for fixed income orders.
 */
var OrderCostPreviewResponseType;
(function (OrderCostPreviewResponseType) {
    OrderCostPreviewResponseType["LimitPriceTypeUnspecified"] = "LIMIT_PRICE_TYPE_UNSPECIFIED";
    OrderCostPreviewResponseType["PricePerUnit"] = "PRICE_PER_UNIT";
    OrderCostPreviewResponseType["PercentageOfPar"] = "PERCENTAGE_OF_PAR";
})(OrderCostPreviewResponseType || (exports.OrderCostPreviewResponseType = OrderCostPreviewResponseType = {}));
/** @internal */
exports.OrderCostPreviewResponseAssetType$inboundSchema = z
    .union([
    z.nativeEnum(OrderCostPreviewResponseAssetType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.OrderCostPreviewResponseAssetType$outboundSchema = z.union([
    z.nativeEnum(OrderCostPreviewResponseAssetType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewResponseAssetType$;
(function (OrderCostPreviewResponseAssetType$) {
    /** @deprecated use `OrderCostPreviewResponseAssetType$inboundSchema` instead. */
    OrderCostPreviewResponseAssetType$.inboundSchema = exports.OrderCostPreviewResponseAssetType$inboundSchema;
    /** @deprecated use `OrderCostPreviewResponseAssetType$outboundSchema` instead. */
    OrderCostPreviewResponseAssetType$.outboundSchema = exports.OrderCostPreviewResponseAssetType$outboundSchema;
})(OrderCostPreviewResponseAssetType$ || (exports.OrderCostPreviewResponseAssetType$ = OrderCostPreviewResponseAssetType$ = {}));
/** @internal */
exports.OrderCostPreviewResponseBrokerCapacity$inboundSchema = z
    .union([
    z.nativeEnum(OrderCostPreviewResponseBrokerCapacity),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.OrderCostPreviewResponseBrokerCapacity$outboundSchema = z.union([
    z.nativeEnum(OrderCostPreviewResponseBrokerCapacity),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewResponseBrokerCapacity$;
(function (OrderCostPreviewResponseBrokerCapacity$) {
    /** @deprecated use `OrderCostPreviewResponseBrokerCapacity$inboundSchema` instead. */
    OrderCostPreviewResponseBrokerCapacity$.inboundSchema = exports.OrderCostPreviewResponseBrokerCapacity$inboundSchema;
    /** @deprecated use `OrderCostPreviewResponseBrokerCapacity$outboundSchema` instead. */
    OrderCostPreviewResponseBrokerCapacity$.outboundSchema = exports.OrderCostPreviewResponseBrokerCapacity$outboundSchema;
})(OrderCostPreviewResponseBrokerCapacity$ || (exports.OrderCostPreviewResponseBrokerCapacity$ = OrderCostPreviewResponseBrokerCapacity$ = {}));
/** @internal */
exports.EstimatedAccruedInterest$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.EstimatedAccruedInterest$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EstimatedAccruedInterest$;
(function (EstimatedAccruedInterest$) {
    /** @deprecated use `EstimatedAccruedInterest$inboundSchema` instead. */
    EstimatedAccruedInterest$.inboundSchema = exports.EstimatedAccruedInterest$inboundSchema;
    /** @deprecated use `EstimatedAccruedInterest$outboundSchema` instead. */
    EstimatedAccruedInterest$.outboundSchema = exports.EstimatedAccruedInterest$outboundSchema;
})(EstimatedAccruedInterest$ || (exports.EstimatedAccruedInterest$ = EstimatedAccruedInterest$ = {}));
/** @internal */
exports.EstimatedGrossAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.EstimatedGrossAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EstimatedGrossAmount$;
(function (EstimatedGrossAmount$) {
    /** @deprecated use `EstimatedGrossAmount$inboundSchema` instead. */
    EstimatedGrossAmount$.inboundSchema = exports.EstimatedGrossAmount$inboundSchema;
    /** @deprecated use `EstimatedGrossAmount$outboundSchema` instead. */
    EstimatedGrossAmount$.outboundSchema = exports.EstimatedGrossAmount$outboundSchema;
})(EstimatedGrossAmount$ || (exports.EstimatedGrossAmount$ = EstimatedGrossAmount$ = {}));
/** @internal */
exports.EstimatedGrossCreditAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.EstimatedGrossCreditAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EstimatedGrossCreditAmount$;
(function (EstimatedGrossCreditAmount$) {
    /** @deprecated use `EstimatedGrossCreditAmount$inboundSchema` instead. */
    EstimatedGrossCreditAmount$.inboundSchema = exports.EstimatedGrossCreditAmount$inboundSchema;
    /** @deprecated use `EstimatedGrossCreditAmount$outboundSchema` instead. */
    EstimatedGrossCreditAmount$.outboundSchema = exports.EstimatedGrossCreditAmount$outboundSchema;
})(EstimatedGrossCreditAmount$ || (exports.EstimatedGrossCreditAmount$ = EstimatedGrossCreditAmount$ = {}));
/** @internal */
exports.EstimatedNetAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.EstimatedNetAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var EstimatedNetAmount$;
(function (EstimatedNetAmount$) {
    /** @deprecated use `EstimatedNetAmount$inboundSchema` instead. */
    EstimatedNetAmount$.inboundSchema = exports.EstimatedNetAmount$inboundSchema;
    /** @deprecated use `EstimatedNetAmount$outboundSchema` instead. */
    EstimatedNetAmount$.outboundSchema = exports.EstimatedNetAmount$outboundSchema;
})(EstimatedNetAmount$ || (exports.EstimatedNetAmount$ = EstimatedNetAmount$ = {}));
/** @internal */
exports.OrderCostPreviewResponseIdentifierType$inboundSchema = z
    .union([
    z.nativeEnum(OrderCostPreviewResponseIdentifierType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.OrderCostPreviewResponseIdentifierType$outboundSchema = z.union([
    z.nativeEnum(OrderCostPreviewResponseIdentifierType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewResponseIdentifierType$;
(function (OrderCostPreviewResponseIdentifierType$) {
    /** @deprecated use `OrderCostPreviewResponseIdentifierType$inboundSchema` instead. */
    OrderCostPreviewResponseIdentifierType$.inboundSchema = exports.OrderCostPreviewResponseIdentifierType$inboundSchema;
    /** @deprecated use `OrderCostPreviewResponseIdentifierType$outboundSchema` instead. */
    OrderCostPreviewResponseIdentifierType$.outboundSchema = exports.OrderCostPreviewResponseIdentifierType$outboundSchema;
})(OrderCostPreviewResponseIdentifierType$ || (exports.OrderCostPreviewResponseIdentifierType$ = OrderCostPreviewResponseIdentifierType$ = {}));
/** @internal */
exports.OrderCostPreviewResponsePrice$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.OrderCostPreviewResponsePrice$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewResponsePrice$;
(function (OrderCostPreviewResponsePrice$) {
    /** @deprecated use `OrderCostPreviewResponsePrice$inboundSchema` instead. */
    OrderCostPreviewResponsePrice$.inboundSchema = exports.OrderCostPreviewResponsePrice$inboundSchema;
    /** @deprecated use `OrderCostPreviewResponsePrice$outboundSchema` instead. */
    OrderCostPreviewResponsePrice$.outboundSchema = exports.OrderCostPreviewResponsePrice$outboundSchema;
})(OrderCostPreviewResponsePrice$ || (exports.OrderCostPreviewResponsePrice$ = OrderCostPreviewResponsePrice$ = {}));
/** @internal */
exports.OrderCostPreviewResponseType$inboundSchema = z
    .union([
    z.nativeEnum(OrderCostPreviewResponseType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.OrderCostPreviewResponseType$outboundSchema = z.union([
    z.nativeEnum(OrderCostPreviewResponseType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewResponseType$;
(function (OrderCostPreviewResponseType$) {
    /** @deprecated use `OrderCostPreviewResponseType$inboundSchema` instead. */
    OrderCostPreviewResponseType$.inboundSchema = exports.OrderCostPreviewResponseType$inboundSchema;
    /** @deprecated use `OrderCostPreviewResponseType$outboundSchema` instead. */
    OrderCostPreviewResponseType$.outboundSchema = exports.OrderCostPreviewResponseType$outboundSchema;
})(OrderCostPreviewResponseType$ || (exports.OrderCostPreviewResponseType$ = OrderCostPreviewResponseType$ = {}));
/** @internal */
exports.OrderCostPreviewResponseLimitPrice$inboundSchema = z.object({
    price: z.nullable(z.lazy(() => exports.OrderCostPreviewResponsePrice$inboundSchema))
        .optional(),
    type: exports.OrderCostPreviewResponseType$inboundSchema.optional(),
});
/** @internal */
exports.OrderCostPreviewResponseLimitPrice$outboundSchema = z.object({
    price: z.nullable(z.lazy(() => exports.OrderCostPreviewResponsePrice$outboundSchema))
        .optional(),
    type: exports.OrderCostPreviewResponseType$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewResponseLimitPrice$;
(function (OrderCostPreviewResponseLimitPrice$) {
    /** @deprecated use `OrderCostPreviewResponseLimitPrice$inboundSchema` instead. */
    OrderCostPreviewResponseLimitPrice$.inboundSchema = exports.OrderCostPreviewResponseLimitPrice$inboundSchema;
    /** @deprecated use `OrderCostPreviewResponseLimitPrice$outboundSchema` instead. */
    OrderCostPreviewResponseLimitPrice$.outboundSchema = exports.OrderCostPreviewResponseLimitPrice$outboundSchema;
})(OrderCostPreviewResponseLimitPrice$ || (exports.OrderCostPreviewResponseLimitPrice$ = OrderCostPreviewResponseLimitPrice$ = {}));
/** @internal */
exports.OrderCostPreviewResponseQuantity$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.OrderCostPreviewResponseQuantity$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewResponseQuantity$;
(function (OrderCostPreviewResponseQuantity$) {
    /** @deprecated use `OrderCostPreviewResponseQuantity$inboundSchema` instead. */
    OrderCostPreviewResponseQuantity$.inboundSchema = exports.OrderCostPreviewResponseQuantity$inboundSchema;
    /** @deprecated use `OrderCostPreviewResponseQuantity$outboundSchema` instead. */
    OrderCostPreviewResponseQuantity$.outboundSchema = exports.OrderCostPreviewResponseQuantity$outboundSchema;
})(OrderCostPreviewResponseQuantity$ || (exports.OrderCostPreviewResponseQuantity$ = OrderCostPreviewResponseQuantity$ = {}));
/** @internal */
exports.OrderCostPreviewResponse$inboundSchema = z.object({
    account_id: z.string().optional(),
    asset_id: z.string().optional(),
    asset_type: exports.OrderCostPreviewResponseAssetType$inboundSchema.optional(),
    broker_capacity: exports.OrderCostPreviewResponseBrokerCapacity$inboundSchema
        .optional(),
    estimated_accrued_interest: z.nullable(z.lazy(() => exports.EstimatedAccruedInterest$inboundSchema)).optional(),
    estimated_gross_amount: z.nullable(z.lazy(() => exports.EstimatedGrossAmount$inboundSchema)).optional(),
    estimated_gross_credit_amount: z.nullable(z.lazy(() => exports.EstimatedGrossCreditAmount$inboundSchema)).optional(),
    estimated_net_amount: z.nullable(z.lazy(() => exports.EstimatedNetAmount$inboundSchema)).optional(),
    identifier: z.string().optional(),
    identifier_type: exports.OrderCostPreviewResponseIdentifierType$inboundSchema
        .optional(),
    limit_price: z.nullable(z.lazy(() => exports.OrderCostPreviewResponseLimitPrice$inboundSchema)).optional(),
    quantity: z.nullable(z.lazy(() => exports.OrderCostPreviewResponseQuantity$inboundSchema)).optional(),
    response_generation_time: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "asset_id": "assetId",
        "asset_type": "assetType",
        "broker_capacity": "brokerCapacity",
        "estimated_accrued_interest": "estimatedAccruedInterest",
        "estimated_gross_amount": "estimatedGrossAmount",
        "estimated_gross_credit_amount": "estimatedGrossCreditAmount",
        "estimated_net_amount": "estimatedNetAmount",
        "identifier_type": "identifierType",
        "limit_price": "limitPrice",
        "response_generation_time": "responseGenerationTime",
    });
});
/** @internal */
exports.OrderCostPreviewResponse$outboundSchema = z.object({
    accountId: z.string().optional(),
    assetId: z.string().optional(),
    assetType: exports.OrderCostPreviewResponseAssetType$outboundSchema.optional(),
    brokerCapacity: exports.OrderCostPreviewResponseBrokerCapacity$outboundSchema
        .optional(),
    estimatedAccruedInterest: z.nullable(z.lazy(() => exports.EstimatedAccruedInterest$outboundSchema)).optional(),
    estimatedGrossAmount: z.nullable(z.lazy(() => exports.EstimatedGrossAmount$outboundSchema)).optional(),
    estimatedGrossCreditAmount: z.nullable(z.lazy(() => exports.EstimatedGrossCreditAmount$outboundSchema)).optional(),
    estimatedNetAmount: z.nullable(z.lazy(() => exports.EstimatedNetAmount$outboundSchema)).optional(),
    identifier: z.string().optional(),
    identifierType: exports.OrderCostPreviewResponseIdentifierType$outboundSchema
        .optional(),
    limitPrice: z.nullable(z.lazy(() => exports.OrderCostPreviewResponseLimitPrice$outboundSchema)).optional(),
    quantity: z.nullable(z.lazy(() => exports.OrderCostPreviewResponseQuantity$outboundSchema)).optional(),
    responseGenerationTime: z.nullable(z.date().transform(v => v.toISOString()))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        assetId: "asset_id",
        assetType: "asset_type",
        brokerCapacity: "broker_capacity",
        estimatedAccruedInterest: "estimated_accrued_interest",
        estimatedGrossAmount: "estimated_gross_amount",
        estimatedGrossCreditAmount: "estimated_gross_credit_amount",
        estimatedNetAmount: "estimated_net_amount",
        identifierType: "identifier_type",
        limitPrice: "limit_price",
        responseGenerationTime: "response_generation_time",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewResponse$;
(function (OrderCostPreviewResponse$) {
    /** @deprecated use `OrderCostPreviewResponse$inboundSchema` instead. */
    OrderCostPreviewResponse$.inboundSchema = exports.OrderCostPreviewResponse$inboundSchema;
    /** @deprecated use `OrderCostPreviewResponse$outboundSchema` instead. */
    OrderCostPreviewResponse$.outboundSchema = exports.OrderCostPreviewResponse$outboundSchema;
})(OrderCostPreviewResponse$ || (exports.OrderCostPreviewResponse$ = OrderCostPreviewResponse$ = {}));
