// @apexfintechsolutions/ascend-sdk/models/components/ordersoptionstradingenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrdersOptionsTradingEnrollmentMetadataCreate$ = exports.OrdersOptionsTradingEnrollmentMetadataCreate$outboundSchema = exports.OrdersOptionsTradingEnrollmentMetadataCreate$inboundSchema = exports.OptionsLevel$ = exports.OptionsLevel$outboundSchema = exports.OptionsLevel$inboundSchema = exports.OptionsLevel = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const investmentvehicleexperiencecreate_js_1 = require("./investmentvehicleexperiencecreate.js");
/**
 * Requested options level
 */
var OptionsLevel;
(function (OptionsLevel) {
    OptionsLevel["OptionsLevelUnspecified"] = "OPTIONS_LEVEL_UNSPECIFIED";
    OptionsLevel["Level1BasicDirectionalTrades"] = "LEVEL_1_BASIC_DIRECTIONAL_TRADES";
})(OptionsLevel || (exports.OptionsLevel = OptionsLevel = {}));
/** @internal */
exports.OptionsLevel$inboundSchema = z
    .union([
    z.nativeEnum(OptionsLevel),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.OptionsLevel$outboundSchema = z.union([
    z.nativeEnum(OptionsLevel),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OptionsLevel$;
(function (OptionsLevel$) {
    /** @deprecated use `OptionsLevel$inboundSchema` instead. */
    OptionsLevel$.inboundSchema = exports.OptionsLevel$inboundSchema;
    /** @deprecated use `OptionsLevel$outboundSchema` instead. */
    OptionsLevel$.outboundSchema = exports.OptionsLevel$outboundSchema;
})(OptionsLevel$ || (exports.OptionsLevel$ = OptionsLevel$ = {}));
/** @internal */
exports.OrdersOptionsTradingEnrollmentMetadataCreate$inboundSchema = z.object({
    investment_vehicle_experience: investmentvehicleexperiencecreate_js_1.InvestmentVehicleExperienceCreate$inboundSchema,
    options_level: exports.OptionsLevel$inboundSchema,
    total_years_options_trading_experience: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "investment_vehicle_experience": "investmentVehicleExperience",
        "options_level": "optionsLevel",
        "total_years_options_trading_experience": "totalYearsOptionsTradingExperience",
    });
});
/** @internal */
exports.OrdersOptionsTradingEnrollmentMetadataCreate$outboundSchema = z.object({
    investmentVehicleExperience: investmentvehicleexperiencecreate_js_1.InvestmentVehicleExperienceCreate$outboundSchema,
    optionsLevel: exports.OptionsLevel$outboundSchema,
    totalYearsOptionsTradingExperience: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        investmentVehicleExperience: "investment_vehicle_experience",
        optionsLevel: "options_level",
        totalYearsOptionsTradingExperience: "total_years_options_trading_experience",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrdersOptionsTradingEnrollmentMetadataCreate$;
(function (OrdersOptionsTradingEnrollmentMetadataCreate$) {
    /** @deprecated use `OrdersOptionsTradingEnrollmentMetadataCreate$inboundSchema` instead. */
    OrdersOptionsTradingEnrollmentMetadataCreate$.inboundSchema = exports.OrdersOptionsTradingEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `OrdersOptionsTradingEnrollmentMetadataCreate$outboundSchema` instead. */
    OrdersOptionsTradingEnrollmentMetadataCreate$.outboundSchema = exports.OrdersOptionsTradingEnrollmentMetadataCreate$outboundSchema;
})(OrdersOptionsTradingEnrollmentMetadataCreate$ || (exports.OrdersOptionsTradingEnrollmentMetadataCreate$ = OrdersOptionsTradingEnrollmentMetadataCreate$ = {}));
