// @apexfintechsolutions/ascend-sdk/models/components/ordercostpreviewrequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderCostPreviewRequestCreate$ = exports.OrderCostPreviewRequestCreate$outboundSchema = exports.OrderCostPreviewRequestCreate$inboundSchema = exports.OrderCostPreviewRequestCreateIdentifierType$ = exports.OrderCostPreviewRequestCreateIdentifierType$outboundSchema = exports.OrderCostPreviewRequestCreateIdentifierType$inboundSchema = exports.OrderCostPreviewRequestCreateBrokerCapacity$ = exports.OrderCostPreviewRequestCreateBrokerCapacity$outboundSchema = exports.OrderCostPreviewRequestCreateBrokerCapacity$inboundSchema = exports.OrderCostPreviewRequestCreateAssetType$ = exports.OrderCostPreviewRequestCreateAssetType$outboundSchema = exports.OrderCostPreviewRequestCreateAssetType$inboundSchema = exports.OrderCostPreviewRequestCreateIdentifierType = exports.OrderCostPreviewRequestCreateBrokerCapacity = exports.OrderCostPreviewRequestCreateAssetType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
const limitpricecreate_js_1 = require("./limitpricecreate.js");
/**
 * The type of asset being evaluated in cost calculations.
 */
var OrderCostPreviewRequestCreateAssetType;
(function (OrderCostPreviewRequestCreateAssetType) {
    OrderCostPreviewRequestCreateAssetType["FixedIncome"] = "FIXED_INCOME";
})(OrderCostPreviewRequestCreateAssetType || (exports.OrderCostPreviewRequestCreateAssetType = OrderCostPreviewRequestCreateAssetType = {}));
/**
 * Capacity used in determining the order cost. Defaults to "AGENCY" if no value specified.
 */
var OrderCostPreviewRequestCreateBrokerCapacity;
(function (OrderCostPreviewRequestCreateBrokerCapacity) {
    OrderCostPreviewRequestCreateBrokerCapacity["BrokerCapacityUnspecified"] = "BROKER_CAPACITY_UNSPECIFIED";
    OrderCostPreviewRequestCreateBrokerCapacity["Agency"] = "AGENCY";
    OrderCostPreviewRequestCreateBrokerCapacity["Principal"] = "PRINCIPAL";
})(OrderCostPreviewRequestCreateBrokerCapacity || (exports.OrderCostPreviewRequestCreateBrokerCapacity = OrderCostPreviewRequestCreateBrokerCapacity = {}));
/**
 * The identifier type of the asset being sought
 */
var OrderCostPreviewRequestCreateIdentifierType;
(function (OrderCostPreviewRequestCreateIdentifierType) {
    OrderCostPreviewRequestCreateIdentifierType["AssetId"] = "ASSET_ID";
    OrderCostPreviewRequestCreateIdentifierType["Cusip"] = "CUSIP";
    OrderCostPreviewRequestCreateIdentifierType["Isin"] = "ISIN";
})(OrderCostPreviewRequestCreateIdentifierType || (exports.OrderCostPreviewRequestCreateIdentifierType = OrderCostPreviewRequestCreateIdentifierType = {}));
/** @internal */
exports.OrderCostPreviewRequestCreateAssetType$inboundSchema = z
    .union([
    z.nativeEnum(OrderCostPreviewRequestCreateAssetType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.OrderCostPreviewRequestCreateAssetType$outboundSchema = z.union([
    z.nativeEnum(OrderCostPreviewRequestCreateAssetType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewRequestCreateAssetType$;
(function (OrderCostPreviewRequestCreateAssetType$) {
    /** @deprecated use `OrderCostPreviewRequestCreateAssetType$inboundSchema` instead. */
    OrderCostPreviewRequestCreateAssetType$.inboundSchema = exports.OrderCostPreviewRequestCreateAssetType$inboundSchema;
    /** @deprecated use `OrderCostPreviewRequestCreateAssetType$outboundSchema` instead. */
    OrderCostPreviewRequestCreateAssetType$.outboundSchema = exports.OrderCostPreviewRequestCreateAssetType$outboundSchema;
})(OrderCostPreviewRequestCreateAssetType$ || (exports.OrderCostPreviewRequestCreateAssetType$ = OrderCostPreviewRequestCreateAssetType$ = {}));
/** @internal */
exports.OrderCostPreviewRequestCreateBrokerCapacity$inboundSchema = z
    .union([
    z.nativeEnum(OrderCostPreviewRequestCreateBrokerCapacity),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.OrderCostPreviewRequestCreateBrokerCapacity$outboundSchema = z.union([
    z.nativeEnum(OrderCostPreviewRequestCreateBrokerCapacity),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewRequestCreateBrokerCapacity$;
(function (OrderCostPreviewRequestCreateBrokerCapacity$) {
    /** @deprecated use `OrderCostPreviewRequestCreateBrokerCapacity$inboundSchema` instead. */
    OrderCostPreviewRequestCreateBrokerCapacity$.inboundSchema = exports.OrderCostPreviewRequestCreateBrokerCapacity$inboundSchema;
    /** @deprecated use `OrderCostPreviewRequestCreateBrokerCapacity$outboundSchema` instead. */
    OrderCostPreviewRequestCreateBrokerCapacity$.outboundSchema = exports.OrderCostPreviewRequestCreateBrokerCapacity$outboundSchema;
})(OrderCostPreviewRequestCreateBrokerCapacity$ || (exports.OrderCostPreviewRequestCreateBrokerCapacity$ = OrderCostPreviewRequestCreateBrokerCapacity$ = {}));
/** @internal */
exports.OrderCostPreviewRequestCreateIdentifierType$inboundSchema = z
    .union([
    z.nativeEnum(OrderCostPreviewRequestCreateIdentifierType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.OrderCostPreviewRequestCreateIdentifierType$outboundSchema = z.union([
    z.nativeEnum(OrderCostPreviewRequestCreateIdentifierType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewRequestCreateIdentifierType$;
(function (OrderCostPreviewRequestCreateIdentifierType$) {
    /** @deprecated use `OrderCostPreviewRequestCreateIdentifierType$inboundSchema` instead. */
    OrderCostPreviewRequestCreateIdentifierType$.inboundSchema = exports.OrderCostPreviewRequestCreateIdentifierType$inboundSchema;
    /** @deprecated use `OrderCostPreviewRequestCreateIdentifierType$outboundSchema` instead. */
    OrderCostPreviewRequestCreateIdentifierType$.outboundSchema = exports.OrderCostPreviewRequestCreateIdentifierType$outboundSchema;
})(OrderCostPreviewRequestCreateIdentifierType$ || (exports.OrderCostPreviewRequestCreateIdentifierType$ = OrderCostPreviewRequestCreateIdentifierType$ = {}));
/** @internal */
exports.OrderCostPreviewRequestCreate$inboundSchema = z.object({
    asset_type: exports.OrderCostPreviewRequestCreateAssetType$inboundSchema,
    broker_capacity: exports.OrderCostPreviewRequestCreateBrokerCapacity$inboundSchema
        .optional(),
    identifier: z.string(),
    identifier_type: exports.OrderCostPreviewRequestCreateIdentifierType$inboundSchema,
    limit_price: limitpricecreate_js_1.LimitPriceCreate$inboundSchema,
    parent: z.string(),
    quantity: decimalcreate_js_1.DecimalCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "asset_type": "assetType",
        "broker_capacity": "brokerCapacity",
        "identifier_type": "identifierType",
        "limit_price": "limitPrice",
    });
});
/** @internal */
exports.OrderCostPreviewRequestCreate$outboundSchema = z.object({
    assetType: exports.OrderCostPreviewRequestCreateAssetType$outboundSchema,
    brokerCapacity: exports.OrderCostPreviewRequestCreateBrokerCapacity$outboundSchema
        .optional(),
    identifier: z.string(),
    identifierType: exports.OrderCostPreviewRequestCreateIdentifierType$outboundSchema,
    limitPrice: limitpricecreate_js_1.LimitPriceCreate$outboundSchema,
    parent: z.string(),
    quantity: decimalcreate_js_1.DecimalCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        assetType: "asset_type",
        brokerCapacity: "broker_capacity",
        identifierType: "identifier_type",
        limitPrice: "limit_price",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCostPreviewRequestCreate$;
(function (OrderCostPreviewRequestCreate$) {
    /** @deprecated use `OrderCostPreviewRequestCreate$inboundSchema` instead. */
    OrderCostPreviewRequestCreate$.inboundSchema = exports.OrderCostPreviewRequestCreate$inboundSchema;
    /** @deprecated use `OrderCostPreviewRequestCreate$outboundSchema` instead. */
    OrderCostPreviewRequestCreate$.outboundSchema = exports.OrderCostPreviewRequestCreate$outboundSchema;
})(OrderCostPreviewRequestCreate$ || (exports.OrderCostPreviewRequestCreate$ = OrderCostPreviewRequestCreate$ = {}));
