// @apexfintechsolutions/ascend-sdk/models/components/otheraccountscreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OtherAccountsCreate$ = exports.OtherAccountsCreate$outboundSchema = exports.OtherAccountsCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.OtherAccountsCreate$inboundSchema = z.object({
    account_names: z.array(z.string()).optional(),
    account_numbers: z.array(z.string()).optional(),
    owner_has_other_accounts_at_apex: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_names": "accountNames",
        "account_numbers": "accountNumbers",
        "owner_has_other_accounts_at_apex": "ownerHasOtherAccountsAtApex",
    });
});
/** @internal */
exports.OtherAccountsCreate$outboundSchema = z.object({
    accountNames: z.array(z.string()).optional(),
    accountNumbers: z.array(z.string()).optional(),
    ownerHasOtherAccountsAtApex: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountNames: "account_names",
        accountNumbers: "account_numbers",
        ownerHasOtherAccountsAtApex: "owner_has_other_accounts_at_apex",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OtherAccountsCreate$;
(function (OtherAccountsCreate$) {
    /** @deprecated use `OtherAccountsCreate$inboundSchema` instead. */
    OtherAccountsCreate$.inboundSchema = exports.OtherAccountsCreate$inboundSchema;
    /** @deprecated use `OtherAccountsCreate$outboundSchema` instead. */
    OtherAccountsCreate$.outboundSchema = exports.OtherAccountsCreate$outboundSchema;
})(OtherAccountsCreate$ || (exports.OtherAccountsCreate$ = OtherAccountsCreate$ = {}));
