// @apexfintechsolutions/ascend-sdk/models/components/partyrequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartyRequestCreate$ = exports.PartyRequestCreate$outboundSchema = exports.PartyRequestCreate$inboundSchema = exports.PartyRequestCreateTradeConfirmationDeliveryPreference$ = exports.PartyRequestCreateTradeConfirmationDeliveryPreference$outboundSchema = exports.PartyRequestCreateTradeConfirmationDeliveryPreference$inboundSchema = exports.TaxDocumentDeliveryPreference$ = exports.TaxDocumentDeliveryPreference$outboundSchema = exports.TaxDocumentDeliveryPreference$inboundSchema = exports.PartyRequestCreateStatementDeliveryPreference$ = exports.PartyRequestCreateStatementDeliveryPreference$outboundSchema = exports.PartyRequestCreateStatementDeliveryPreference$inboundSchema = exports.RelationType$ = exports.RelationType$outboundSchema = exports.RelationType$inboundSchema = exports.ProxyDeliveryPreference$ = exports.ProxyDeliveryPreference$outboundSchema = exports.ProxyDeliveryPreference$inboundSchema = exports.ProspectusDeliveryPreference$ = exports.ProspectusDeliveryPreference$outboundSchema = exports.ProspectusDeliveryPreference$inboundSchema = exports.PartyRequestCreateTradeConfirmationDeliveryPreference = exports.TaxDocumentDeliveryPreference = exports.PartyRequestCreateStatementDeliveryPreference = exports.RelationType = exports.ProxyDeliveryPreference = exports.ProspectusDeliveryPreference = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const phonenumbercreate_js_1 = require("./phonenumbercreate.js");
const postaladdresscreate_js_1 = require("./postaladdresscreate.js");
/**
 * Delivery method instruction for prospectuses for a given Party record; Can be `DIGITAL`, `PHYSICAL`, `SUPPRESS`; Defaults to `DIGITAL` on account creation but may be updated
 */
var ProspectusDeliveryPreference;
(function (ProspectusDeliveryPreference) {
    ProspectusDeliveryPreference["DeliveryPreferenceUnspecified"] = "DELIVERY_PREFERENCE_UNSPECIFIED";
    ProspectusDeliveryPreference["Digital"] = "DIGITAL";
    ProspectusDeliveryPreference["Physical"] = "PHYSICAL";
    ProspectusDeliveryPreference["Suppress"] = "SUPPRESS";
})(ProspectusDeliveryPreference || (exports.ProspectusDeliveryPreference = ProspectusDeliveryPreference = {}));
/**
 * Delivery method instruction for proxy voting for a given Party record; Can be `DIGITAL`, `PHYSICAL`, `SUPPRESS`; Defaults to `DIGITAL` on account creation but may be updated
 */
var ProxyDeliveryPreference;
(function (ProxyDeliveryPreference) {
    ProxyDeliveryPreference["DeliveryPreferenceUnspecified"] = "DELIVERY_PREFERENCE_UNSPECIFIED";
    ProxyDeliveryPreference["Digital"] = "DIGITAL";
    ProxyDeliveryPreference["Physical"] = "PHYSICAL";
    ProxyDeliveryPreference["Suppress"] = "SUPPRESS";
})(ProxyDeliveryPreference || (exports.ProxyDeliveryPreference = ProxyDeliveryPreference = {}));
/**
 * Conveys how a person is related to account; Located on each account Party record; Examples are `PRIMARY_OWNER`, `JOINT_OWNER`, `EXECUTOR`, etc.
 */
var RelationType;
(function (RelationType) {
    RelationType["PartyRelationTypeUnspecified"] = "PARTY_RELATION_TYPE_UNSPECIFIED";
    RelationType["PrimaryOwner"] = "PRIMARY_OWNER";
    RelationType["JointOwner"] = "JOINT_OWNER";
    RelationType["Custodian"] = "CUSTODIAN";
    RelationType["Executor"] = "EXECUTOR";
    RelationType["AuthorizedSigner"] = "AUTHORIZED_SIGNER";
    RelationType["BeneficialOwner"] = "BENEFICIAL_OWNER";
    RelationType["ControlPerson"] = "CONTROL_PERSON";
    RelationType["AuthorizedRepresentative"] = "AUTHORIZED_REPRESENTATIVE";
    RelationType["Trustee"] = "TRUSTEE";
    RelationType["AuthTrusteeRep"] = "AUTH_TRUSTEE_REP";
})(RelationType || (exports.RelationType = RelationType = {}));
/**
 * Delivery method instruction for account statements for a given Party record; Can be `DIGITAL`, `PHYSICAL`, `SUPPRESS`; Defaults to `DIGITAL` on account creation but may be updated
 */
var PartyRequestCreateStatementDeliveryPreference;
(function (PartyRequestCreateStatementDeliveryPreference) {
    PartyRequestCreateStatementDeliveryPreference["DeliveryPreferenceUnspecified"] = "DELIVERY_PREFERENCE_UNSPECIFIED";
    PartyRequestCreateStatementDeliveryPreference["Digital"] = "DIGITAL";
    PartyRequestCreateStatementDeliveryPreference["Physical"] = "PHYSICAL";
    PartyRequestCreateStatementDeliveryPreference["Suppress"] = "SUPPRESS";
})(PartyRequestCreateStatementDeliveryPreference || (exports.PartyRequestCreateStatementDeliveryPreference = PartyRequestCreateStatementDeliveryPreference = {}));
/**
 * Delivery method instruction for tax documents for a given Party record; Can be `DIGITAL`, `PHYSICAL`, `SUPPRESS`; Defaults to `DIGITAL` on account creation but may be updated; Per regulation, selected tax forms will be mailed by regulation regardless of this setting
 */
var TaxDocumentDeliveryPreference;
(function (TaxDocumentDeliveryPreference) {
    TaxDocumentDeliveryPreference["DeliveryPreferenceUnspecified"] = "DELIVERY_PREFERENCE_UNSPECIFIED";
    TaxDocumentDeliveryPreference["Digital"] = "DIGITAL";
    TaxDocumentDeliveryPreference["Physical"] = "PHYSICAL";
    TaxDocumentDeliveryPreference["Suppress"] = "SUPPRESS";
})(TaxDocumentDeliveryPreference || (exports.TaxDocumentDeliveryPreference = TaxDocumentDeliveryPreference = {}));
/**
 * Delivery method instruction for trade confirmations for a given Party record; Can be `DIGITAL`, `PHYSICAL`, `SUPPRESS`; Defaults to `DIGITAL` on account creation but may be updated
 */
var PartyRequestCreateTradeConfirmationDeliveryPreference;
(function (PartyRequestCreateTradeConfirmationDeliveryPreference) {
    PartyRequestCreateTradeConfirmationDeliveryPreference["DeliveryPreferenceUnspecified"] = "DELIVERY_PREFERENCE_UNSPECIFIED";
    PartyRequestCreateTradeConfirmationDeliveryPreference["Digital"] = "DIGITAL";
    PartyRequestCreateTradeConfirmationDeliveryPreference["Physical"] = "PHYSICAL";
    PartyRequestCreateTradeConfirmationDeliveryPreference["Suppress"] = "SUPPRESS";
})(PartyRequestCreateTradeConfirmationDeliveryPreference || (exports.PartyRequestCreateTradeConfirmationDeliveryPreference = PartyRequestCreateTradeConfirmationDeliveryPreference = {}));
/** @internal */
exports.ProspectusDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(ProspectusDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ProspectusDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(ProspectusDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProspectusDeliveryPreference$;
(function (ProspectusDeliveryPreference$) {
    /** @deprecated use `ProspectusDeliveryPreference$inboundSchema` instead. */
    ProspectusDeliveryPreference$.inboundSchema = exports.ProspectusDeliveryPreference$inboundSchema;
    /** @deprecated use `ProspectusDeliveryPreference$outboundSchema` instead. */
    ProspectusDeliveryPreference$.outboundSchema = exports.ProspectusDeliveryPreference$outboundSchema;
})(ProspectusDeliveryPreference$ || (exports.ProspectusDeliveryPreference$ = ProspectusDeliveryPreference$ = {}));
/** @internal */
exports.ProxyDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(ProxyDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ProxyDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(ProxyDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProxyDeliveryPreference$;
(function (ProxyDeliveryPreference$) {
    /** @deprecated use `ProxyDeliveryPreference$inboundSchema` instead. */
    ProxyDeliveryPreference$.inboundSchema = exports.ProxyDeliveryPreference$inboundSchema;
    /** @deprecated use `ProxyDeliveryPreference$outboundSchema` instead. */
    ProxyDeliveryPreference$.outboundSchema = exports.ProxyDeliveryPreference$outboundSchema;
})(ProxyDeliveryPreference$ || (exports.ProxyDeliveryPreference$ = ProxyDeliveryPreference$ = {}));
/** @internal */
exports.RelationType$inboundSchema = z
    .union([
    z.nativeEnum(RelationType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.RelationType$outboundSchema = z.union([
    z.nativeEnum(RelationType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RelationType$;
(function (RelationType$) {
    /** @deprecated use `RelationType$inboundSchema` instead. */
    RelationType$.inboundSchema = exports.RelationType$inboundSchema;
    /** @deprecated use `RelationType$outboundSchema` instead. */
    RelationType$.outboundSchema = exports.RelationType$outboundSchema;
})(RelationType$ || (exports.RelationType$ = RelationType$ = {}));
/** @internal */
exports.PartyRequestCreateStatementDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(PartyRequestCreateStatementDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.PartyRequestCreateStatementDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(PartyRequestCreateStatementDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PartyRequestCreateStatementDeliveryPreference$;
(function (PartyRequestCreateStatementDeliveryPreference$) {
    /** @deprecated use `PartyRequestCreateStatementDeliveryPreference$inboundSchema` instead. */
    PartyRequestCreateStatementDeliveryPreference$.inboundSchema = exports.PartyRequestCreateStatementDeliveryPreference$inboundSchema;
    /** @deprecated use `PartyRequestCreateStatementDeliveryPreference$outboundSchema` instead. */
    PartyRequestCreateStatementDeliveryPreference$.outboundSchema = exports.PartyRequestCreateStatementDeliveryPreference$outboundSchema;
})(PartyRequestCreateStatementDeliveryPreference$ || (exports.PartyRequestCreateStatementDeliveryPreference$ = PartyRequestCreateStatementDeliveryPreference$ = {}));
/** @internal */
exports.TaxDocumentDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(TaxDocumentDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TaxDocumentDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(TaxDocumentDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TaxDocumentDeliveryPreference$;
(function (TaxDocumentDeliveryPreference$) {
    /** @deprecated use `TaxDocumentDeliveryPreference$inboundSchema` instead. */
    TaxDocumentDeliveryPreference$.inboundSchema = exports.TaxDocumentDeliveryPreference$inboundSchema;
    /** @deprecated use `TaxDocumentDeliveryPreference$outboundSchema` instead. */
    TaxDocumentDeliveryPreference$.outboundSchema = exports.TaxDocumentDeliveryPreference$outboundSchema;
})(TaxDocumentDeliveryPreference$ || (exports.TaxDocumentDeliveryPreference$ = TaxDocumentDeliveryPreference$ = {}));
/** @internal */
exports.PartyRequestCreateTradeConfirmationDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(PartyRequestCreateTradeConfirmationDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.PartyRequestCreateTradeConfirmationDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(PartyRequestCreateTradeConfirmationDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PartyRequestCreateTradeConfirmationDeliveryPreference$;
(function (PartyRequestCreateTradeConfirmationDeliveryPreference$) {
    /** @deprecated use `PartyRequestCreateTradeConfirmationDeliveryPreference$inboundSchema` instead. */
    PartyRequestCreateTradeConfirmationDeliveryPreference$.inboundSchema = exports.PartyRequestCreateTradeConfirmationDeliveryPreference$inboundSchema;
    /** @deprecated use `PartyRequestCreateTradeConfirmationDeliveryPreference$outboundSchema` instead. */
    PartyRequestCreateTradeConfirmationDeliveryPreference$.outboundSchema = exports.PartyRequestCreateTradeConfirmationDeliveryPreference$outboundSchema;
})(PartyRequestCreateTradeConfirmationDeliveryPreference$ || (exports.PartyRequestCreateTradeConfirmationDeliveryPreference$ = PartyRequestCreateTradeConfirmationDeliveryPreference$ = {}));
/** @internal */
exports.PartyRequestCreate$inboundSchema = z.object({
    email_address: z.string(),
    legal_entity_id: z.string().optional(),
    legal_natural_person_id: z.string().optional(),
    mailing_address: postaladdresscreate_js_1.PostalAddressCreate$inboundSchema,
    phone_number: phonenumbercreate_js_1.PhoneNumberCreate$inboundSchema,
    prospectus_delivery_preference: exports.ProspectusDeliveryPreference$inboundSchema
        .optional(),
    proxy_delivery_preference: exports.ProxyDeliveryPreference$inboundSchema.optional(),
    relation_type: exports.RelationType$inboundSchema,
    statement_delivery_preference: exports.PartyRequestCreateStatementDeliveryPreference$inboundSchema.optional(),
    tax_document_delivery_preference: exports.TaxDocumentDeliveryPreference$inboundSchema
        .optional(),
    trade_confirmation_delivery_preference: exports.PartyRequestCreateTradeConfirmationDeliveryPreference$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "email_address": "emailAddress",
        "legal_entity_id": "legalEntityId",
        "legal_natural_person_id": "legalNaturalPersonId",
        "mailing_address": "mailingAddress",
        "phone_number": "phoneNumber",
        "prospectus_delivery_preference": "prospectusDeliveryPreference",
        "proxy_delivery_preference": "proxyDeliveryPreference",
        "relation_type": "relationType",
        "statement_delivery_preference": "statementDeliveryPreference",
        "tax_document_delivery_preference": "taxDocumentDeliveryPreference",
        "trade_confirmation_delivery_preference": "tradeConfirmationDeliveryPreference",
    });
});
/** @internal */
exports.PartyRequestCreate$outboundSchema = z.object({
    emailAddress: z.string(),
    legalEntityId: z.string().optional(),
    legalNaturalPersonId: z.string().optional(),
    mailingAddress: postaladdresscreate_js_1.PostalAddressCreate$outboundSchema,
    phoneNumber: phonenumbercreate_js_1.PhoneNumberCreate$outboundSchema,
    prospectusDeliveryPreference: exports.ProspectusDeliveryPreference$outboundSchema
        .optional(),
    proxyDeliveryPreference: exports.ProxyDeliveryPreference$outboundSchema.optional(),
    relationType: exports.RelationType$outboundSchema,
    statementDeliveryPreference: exports.PartyRequestCreateStatementDeliveryPreference$outboundSchema.optional(),
    taxDocumentDeliveryPreference: exports.TaxDocumentDeliveryPreference$outboundSchema
        .optional(),
    tradeConfirmationDeliveryPreference: exports.PartyRequestCreateTradeConfirmationDeliveryPreference$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        emailAddress: "email_address",
        legalEntityId: "legal_entity_id",
        legalNaturalPersonId: "legal_natural_person_id",
        mailingAddress: "mailing_address",
        phoneNumber: "phone_number",
        prospectusDeliveryPreference: "prospectus_delivery_preference",
        proxyDeliveryPreference: "proxy_delivery_preference",
        relationType: "relation_type",
        statementDeliveryPreference: "statement_delivery_preference",
        taxDocumentDeliveryPreference: "tax_document_delivery_preference",
        tradeConfirmationDeliveryPreference: "trade_confirmation_delivery_preference",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PartyRequestCreate$;
(function (PartyRequestCreate$) {
    /** @deprecated use `PartyRequestCreate$inboundSchema` instead. */
    PartyRequestCreate$.inboundSchema = exports.PartyRequestCreate$inboundSchema;
    /** @deprecated use `PartyRequestCreate$outboundSchema` instead. */
    PartyRequestCreate$.outboundSchema = exports.PartyRequestCreate$outboundSchema;
})(PartyRequestCreate$ || (exports.PartyRequestCreate$ = PartyRequestCreate$ = {}));
