// @apexfintechsolutions/ascend-sdk/models/components/partyrequestupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartyRequestUpdate$ = exports.PartyRequestUpdate$outboundSchema = exports.PartyRequestUpdate$inboundSchema = exports.PartyRequestUpdateTradeConfirmationDeliveryPreference$ = exports.PartyRequestUpdateTradeConfirmationDeliveryPreference$outboundSchema = exports.PartyRequestUpdateTradeConfirmationDeliveryPreference$inboundSchema = exports.PartyRequestUpdateTaxDocumentDeliveryPreference$ = exports.PartyRequestUpdateTaxDocumentDeliveryPreference$outboundSchema = exports.PartyRequestUpdateTaxDocumentDeliveryPreference$inboundSchema = exports.PartyRequestUpdateStatementDeliveryPreference$ = exports.PartyRequestUpdateStatementDeliveryPreference$outboundSchema = exports.PartyRequestUpdateStatementDeliveryPreference$inboundSchema = exports.PartyRequestUpdateRelationType$ = exports.PartyRequestUpdateRelationType$outboundSchema = exports.PartyRequestUpdateRelationType$inboundSchema = exports.PartyRequestUpdateProxyDeliveryPreference$ = exports.PartyRequestUpdateProxyDeliveryPreference$outboundSchema = exports.PartyRequestUpdateProxyDeliveryPreference$inboundSchema = exports.PartyRequestUpdateProspectusDeliveryPreference$ = exports.PartyRequestUpdateProspectusDeliveryPreference$outboundSchema = exports.PartyRequestUpdateProspectusDeliveryPreference$inboundSchema = exports.PartyRequestUpdateTradeConfirmationDeliveryPreference = exports.PartyRequestUpdateTaxDocumentDeliveryPreference = exports.PartyRequestUpdateStatementDeliveryPreference = exports.PartyRequestUpdateRelationType = exports.PartyRequestUpdateProxyDeliveryPreference = exports.PartyRequestUpdateProspectusDeliveryPreference = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const phonenumberupdate_js_1 = require("./phonenumberupdate.js");
const postaladdressupdate_js_1 = require("./postaladdressupdate.js");
/**
 * Delivery method instruction for prospectuses for a given Party record; Can be `DIGITAL`, `PHYSICAL`, `SUPPRESS`; Defaults to `DIGITAL` on account creation but may be updated
 */
var PartyRequestUpdateProspectusDeliveryPreference;
(function (PartyRequestUpdateProspectusDeliveryPreference) {
    PartyRequestUpdateProspectusDeliveryPreference["DeliveryPreferenceUnspecified"] = "DELIVERY_PREFERENCE_UNSPECIFIED";
    PartyRequestUpdateProspectusDeliveryPreference["Digital"] = "DIGITAL";
    PartyRequestUpdateProspectusDeliveryPreference["Physical"] = "PHYSICAL";
    PartyRequestUpdateProspectusDeliveryPreference["Suppress"] = "SUPPRESS";
})(PartyRequestUpdateProspectusDeliveryPreference || (exports.PartyRequestUpdateProspectusDeliveryPreference = PartyRequestUpdateProspectusDeliveryPreference = {}));
/**
 * Delivery method instruction for proxy voting for a given Party record; Can be `DIGITAL`, `PHYSICAL`, `SUPPRESS`; Defaults to `DIGITAL` on account creation but may be updated
 */
var PartyRequestUpdateProxyDeliveryPreference;
(function (PartyRequestUpdateProxyDeliveryPreference) {
    PartyRequestUpdateProxyDeliveryPreference["DeliveryPreferenceUnspecified"] = "DELIVERY_PREFERENCE_UNSPECIFIED";
    PartyRequestUpdateProxyDeliveryPreference["Digital"] = "DIGITAL";
    PartyRequestUpdateProxyDeliveryPreference["Physical"] = "PHYSICAL";
    PartyRequestUpdateProxyDeliveryPreference["Suppress"] = "SUPPRESS";
})(PartyRequestUpdateProxyDeliveryPreference || (exports.PartyRequestUpdateProxyDeliveryPreference = PartyRequestUpdateProxyDeliveryPreference = {}));
/**
 * Conveys how a person is related to account; Located on each account Party record; Examples are `PRIMARY_OWNER`, `JOINT_OWNER`, `EXECUTOR`, etc.
 */
var PartyRequestUpdateRelationType;
(function (PartyRequestUpdateRelationType) {
    PartyRequestUpdateRelationType["PartyRelationTypeUnspecified"] = "PARTY_RELATION_TYPE_UNSPECIFIED";
    PartyRequestUpdateRelationType["PrimaryOwner"] = "PRIMARY_OWNER";
    PartyRequestUpdateRelationType["JointOwner"] = "JOINT_OWNER";
    PartyRequestUpdateRelationType["Custodian"] = "CUSTODIAN";
    PartyRequestUpdateRelationType["Executor"] = "EXECUTOR";
    PartyRequestUpdateRelationType["AuthorizedSigner"] = "AUTHORIZED_SIGNER";
    PartyRequestUpdateRelationType["BeneficialOwner"] = "BENEFICIAL_OWNER";
    PartyRequestUpdateRelationType["ControlPerson"] = "CONTROL_PERSON";
    PartyRequestUpdateRelationType["AuthorizedRepresentative"] = "AUTHORIZED_REPRESENTATIVE";
    PartyRequestUpdateRelationType["Trustee"] = "TRUSTEE";
    PartyRequestUpdateRelationType["AuthTrusteeRep"] = "AUTH_TRUSTEE_REP";
})(PartyRequestUpdateRelationType || (exports.PartyRequestUpdateRelationType = PartyRequestUpdateRelationType = {}));
/**
 * Delivery method instruction for account statements for a given Party record; Can be `DIGITAL`, `PHYSICAL`, `SUPPRESS`; Defaults to `DIGITAL` on account creation but may be updated
 */
var PartyRequestUpdateStatementDeliveryPreference;
(function (PartyRequestUpdateStatementDeliveryPreference) {
    PartyRequestUpdateStatementDeliveryPreference["DeliveryPreferenceUnspecified"] = "DELIVERY_PREFERENCE_UNSPECIFIED";
    PartyRequestUpdateStatementDeliveryPreference["Digital"] = "DIGITAL";
    PartyRequestUpdateStatementDeliveryPreference["Physical"] = "PHYSICAL";
    PartyRequestUpdateStatementDeliveryPreference["Suppress"] = "SUPPRESS";
})(PartyRequestUpdateStatementDeliveryPreference || (exports.PartyRequestUpdateStatementDeliveryPreference = PartyRequestUpdateStatementDeliveryPreference = {}));
/**
 * Delivery method instruction for tax documents for a given Party record; Can be `DIGITAL`, `PHYSICAL`, `SUPPRESS`; Defaults to `DIGITAL` on account creation but may be updated; Per regulation, selected tax forms will be mailed by regulation regardless of this setting
 */
var PartyRequestUpdateTaxDocumentDeliveryPreference;
(function (PartyRequestUpdateTaxDocumentDeliveryPreference) {
    PartyRequestUpdateTaxDocumentDeliveryPreference["DeliveryPreferenceUnspecified"] = "DELIVERY_PREFERENCE_UNSPECIFIED";
    PartyRequestUpdateTaxDocumentDeliveryPreference["Digital"] = "DIGITAL";
    PartyRequestUpdateTaxDocumentDeliveryPreference["Physical"] = "PHYSICAL";
    PartyRequestUpdateTaxDocumentDeliveryPreference["Suppress"] = "SUPPRESS";
})(PartyRequestUpdateTaxDocumentDeliveryPreference || (exports.PartyRequestUpdateTaxDocumentDeliveryPreference = PartyRequestUpdateTaxDocumentDeliveryPreference = {}));
/**
 * Delivery method instruction for trade confirmations for a given Party record; Can be `DIGITAL`, `PHYSICAL`, `SUPPRESS`; Defaults to `DIGITAL` on account creation but may be updated
 */
var PartyRequestUpdateTradeConfirmationDeliveryPreference;
(function (PartyRequestUpdateTradeConfirmationDeliveryPreference) {
    PartyRequestUpdateTradeConfirmationDeliveryPreference["DeliveryPreferenceUnspecified"] = "DELIVERY_PREFERENCE_UNSPECIFIED";
    PartyRequestUpdateTradeConfirmationDeliveryPreference["Digital"] = "DIGITAL";
    PartyRequestUpdateTradeConfirmationDeliveryPreference["Physical"] = "PHYSICAL";
    PartyRequestUpdateTradeConfirmationDeliveryPreference["Suppress"] = "SUPPRESS";
})(PartyRequestUpdateTradeConfirmationDeliveryPreference || (exports.PartyRequestUpdateTradeConfirmationDeliveryPreference = PartyRequestUpdateTradeConfirmationDeliveryPreference = {}));
/** @internal */
exports.PartyRequestUpdateProspectusDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(PartyRequestUpdateProspectusDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.PartyRequestUpdateProspectusDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(PartyRequestUpdateProspectusDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PartyRequestUpdateProspectusDeliveryPreference$;
(function (PartyRequestUpdateProspectusDeliveryPreference$) {
    /** @deprecated use `PartyRequestUpdateProspectusDeliveryPreference$inboundSchema` instead. */
    PartyRequestUpdateProspectusDeliveryPreference$.inboundSchema = exports.PartyRequestUpdateProspectusDeliveryPreference$inboundSchema;
    /** @deprecated use `PartyRequestUpdateProspectusDeliveryPreference$outboundSchema` instead. */
    PartyRequestUpdateProspectusDeliveryPreference$.outboundSchema = exports.PartyRequestUpdateProspectusDeliveryPreference$outboundSchema;
})(PartyRequestUpdateProspectusDeliveryPreference$ || (exports.PartyRequestUpdateProspectusDeliveryPreference$ = PartyRequestUpdateProspectusDeliveryPreference$ = {}));
/** @internal */
exports.PartyRequestUpdateProxyDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(PartyRequestUpdateProxyDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.PartyRequestUpdateProxyDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(PartyRequestUpdateProxyDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PartyRequestUpdateProxyDeliveryPreference$;
(function (PartyRequestUpdateProxyDeliveryPreference$) {
    /** @deprecated use `PartyRequestUpdateProxyDeliveryPreference$inboundSchema` instead. */
    PartyRequestUpdateProxyDeliveryPreference$.inboundSchema = exports.PartyRequestUpdateProxyDeliveryPreference$inboundSchema;
    /** @deprecated use `PartyRequestUpdateProxyDeliveryPreference$outboundSchema` instead. */
    PartyRequestUpdateProxyDeliveryPreference$.outboundSchema = exports.PartyRequestUpdateProxyDeliveryPreference$outboundSchema;
})(PartyRequestUpdateProxyDeliveryPreference$ || (exports.PartyRequestUpdateProxyDeliveryPreference$ = PartyRequestUpdateProxyDeliveryPreference$ = {}));
/** @internal */
exports.PartyRequestUpdateRelationType$inboundSchema = z
    .union([
    z.nativeEnum(PartyRequestUpdateRelationType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.PartyRequestUpdateRelationType$outboundSchema = z.union([
    z.nativeEnum(PartyRequestUpdateRelationType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PartyRequestUpdateRelationType$;
(function (PartyRequestUpdateRelationType$) {
    /** @deprecated use `PartyRequestUpdateRelationType$inboundSchema` instead. */
    PartyRequestUpdateRelationType$.inboundSchema = exports.PartyRequestUpdateRelationType$inboundSchema;
    /** @deprecated use `PartyRequestUpdateRelationType$outboundSchema` instead. */
    PartyRequestUpdateRelationType$.outboundSchema = exports.PartyRequestUpdateRelationType$outboundSchema;
})(PartyRequestUpdateRelationType$ || (exports.PartyRequestUpdateRelationType$ = PartyRequestUpdateRelationType$ = {}));
/** @internal */
exports.PartyRequestUpdateStatementDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(PartyRequestUpdateStatementDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.PartyRequestUpdateStatementDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(PartyRequestUpdateStatementDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PartyRequestUpdateStatementDeliveryPreference$;
(function (PartyRequestUpdateStatementDeliveryPreference$) {
    /** @deprecated use `PartyRequestUpdateStatementDeliveryPreference$inboundSchema` instead. */
    PartyRequestUpdateStatementDeliveryPreference$.inboundSchema = exports.PartyRequestUpdateStatementDeliveryPreference$inboundSchema;
    /** @deprecated use `PartyRequestUpdateStatementDeliveryPreference$outboundSchema` instead. */
    PartyRequestUpdateStatementDeliveryPreference$.outboundSchema = exports.PartyRequestUpdateStatementDeliveryPreference$outboundSchema;
})(PartyRequestUpdateStatementDeliveryPreference$ || (exports.PartyRequestUpdateStatementDeliveryPreference$ = PartyRequestUpdateStatementDeliveryPreference$ = {}));
/** @internal */
exports.PartyRequestUpdateTaxDocumentDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(PartyRequestUpdateTaxDocumentDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.PartyRequestUpdateTaxDocumentDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(PartyRequestUpdateTaxDocumentDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PartyRequestUpdateTaxDocumentDeliveryPreference$;
(function (PartyRequestUpdateTaxDocumentDeliveryPreference$) {
    /** @deprecated use `PartyRequestUpdateTaxDocumentDeliveryPreference$inboundSchema` instead. */
    PartyRequestUpdateTaxDocumentDeliveryPreference$.inboundSchema = exports.PartyRequestUpdateTaxDocumentDeliveryPreference$inboundSchema;
    /** @deprecated use `PartyRequestUpdateTaxDocumentDeliveryPreference$outboundSchema` instead. */
    PartyRequestUpdateTaxDocumentDeliveryPreference$.outboundSchema = exports.PartyRequestUpdateTaxDocumentDeliveryPreference$outboundSchema;
})(PartyRequestUpdateTaxDocumentDeliveryPreference$ || (exports.PartyRequestUpdateTaxDocumentDeliveryPreference$ = PartyRequestUpdateTaxDocumentDeliveryPreference$ = {}));
/** @internal */
exports.PartyRequestUpdateTradeConfirmationDeliveryPreference$inboundSchema = z
    .union([
    z.nativeEnum(PartyRequestUpdateTradeConfirmationDeliveryPreference),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.PartyRequestUpdateTradeConfirmationDeliveryPreference$outboundSchema = z.union([
    z.nativeEnum(PartyRequestUpdateTradeConfirmationDeliveryPreference),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PartyRequestUpdateTradeConfirmationDeliveryPreference$;
(function (PartyRequestUpdateTradeConfirmationDeliveryPreference$) {
    /** @deprecated use `PartyRequestUpdateTradeConfirmationDeliveryPreference$inboundSchema` instead. */
    PartyRequestUpdateTradeConfirmationDeliveryPreference$.inboundSchema = exports.PartyRequestUpdateTradeConfirmationDeliveryPreference$inboundSchema;
    /** @deprecated use `PartyRequestUpdateTradeConfirmationDeliveryPreference$outboundSchema` instead. */
    PartyRequestUpdateTradeConfirmationDeliveryPreference$.outboundSchema = exports.PartyRequestUpdateTradeConfirmationDeliveryPreference$outboundSchema;
})(PartyRequestUpdateTradeConfirmationDeliveryPreference$ || (exports.PartyRequestUpdateTradeConfirmationDeliveryPreference$ = PartyRequestUpdateTradeConfirmationDeliveryPreference$ = {}));
/** @internal */
exports.PartyRequestUpdate$inboundSchema = z.object({
    email_address: z.string().optional(),
    legal_entity_id: z.string().optional(),
    legal_natural_person_id: z.string().optional(),
    mailing_address: postaladdressupdate_js_1.PostalAddressUpdate$inboundSchema.optional(),
    phone_number: phonenumberupdate_js_1.PhoneNumberUpdate$inboundSchema.optional(),
    prospectus_delivery_preference: exports.PartyRequestUpdateProspectusDeliveryPreference$inboundSchema.optional(),
    proxy_delivery_preference: exports.PartyRequestUpdateProxyDeliveryPreference$inboundSchema.optional(),
    relation_type: exports.PartyRequestUpdateRelationType$inboundSchema.optional(),
    statement_delivery_preference: exports.PartyRequestUpdateStatementDeliveryPreference$inboundSchema.optional(),
    tax_document_delivery_preference: exports.PartyRequestUpdateTaxDocumentDeliveryPreference$inboundSchema.optional(),
    trade_confirmation_delivery_preference: exports.PartyRequestUpdateTradeConfirmationDeliveryPreference$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "email_address": "emailAddress",
        "legal_entity_id": "legalEntityId",
        "legal_natural_person_id": "legalNaturalPersonId",
        "mailing_address": "mailingAddress",
        "phone_number": "phoneNumber",
        "prospectus_delivery_preference": "prospectusDeliveryPreference",
        "proxy_delivery_preference": "proxyDeliveryPreference",
        "relation_type": "relationType",
        "statement_delivery_preference": "statementDeliveryPreference",
        "tax_document_delivery_preference": "taxDocumentDeliveryPreference",
        "trade_confirmation_delivery_preference": "tradeConfirmationDeliveryPreference",
    });
});
/** @internal */
exports.PartyRequestUpdate$outboundSchema = z.object({
    emailAddress: z.string().optional(),
    legalEntityId: z.string().optional(),
    legalNaturalPersonId: z.string().optional(),
    mailingAddress: postaladdressupdate_js_1.PostalAddressUpdate$outboundSchema.optional(),
    phoneNumber: phonenumberupdate_js_1.PhoneNumberUpdate$outboundSchema.optional(),
    prospectusDeliveryPreference: exports.PartyRequestUpdateProspectusDeliveryPreference$outboundSchema.optional(),
    proxyDeliveryPreference: exports.PartyRequestUpdateProxyDeliveryPreference$outboundSchema.optional(),
    relationType: exports.PartyRequestUpdateRelationType$outboundSchema.optional(),
    statementDeliveryPreference: exports.PartyRequestUpdateStatementDeliveryPreference$outboundSchema.optional(),
    taxDocumentDeliveryPreference: exports.PartyRequestUpdateTaxDocumentDeliveryPreference$outboundSchema.optional(),
    tradeConfirmationDeliveryPreference: exports.PartyRequestUpdateTradeConfirmationDeliveryPreference$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        emailAddress: "email_address",
        legalEntityId: "legal_entity_id",
        legalNaturalPersonId: "legal_natural_person_id",
        mailingAddress: "mailing_address",
        phoneNumber: "phone_number",
        prospectusDeliveryPreference: "prospectus_delivery_preference",
        proxyDeliveryPreference: "proxy_delivery_preference",
        relationType: "relation_type",
        statementDeliveryPreference: "statement_delivery_preference",
        taxDocumentDeliveryPreference: "tax_document_delivery_preference",
        tradeConfirmationDeliveryPreference: "trade_confirmation_delivery_preference",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PartyRequestUpdate$;
(function (PartyRequestUpdate$) {
    /** @deprecated use `PartyRequestUpdate$inboundSchema` instead. */
    PartyRequestUpdate$.inboundSchema = exports.PartyRequestUpdate$inboundSchema;
    /** @deprecated use `PartyRequestUpdate$outboundSchema` instead. */
    PartyRequestUpdate$.outboundSchema = exports.PartyRequestUpdate$outboundSchema;
})(PartyRequestUpdate$ || (exports.PartyRequestUpdate$ = PartyRequestUpdate$ = {}));
