// @apexfintechsolutions/ascend-sdk/models/components/phonenumberupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneNumberUpdate$ = exports.PhoneNumberUpdate$outboundSchema = exports.PhoneNumberUpdate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const phonenumbershortcodeupdate_js_1 = require("./phonenumbershortcodeupdate.js");
/** @internal */
exports.PhoneNumberUpdate$inboundSchema = z.object({
    e164_number: z.string().optional(),
    extension: z.string().optional(),
    short_code: phonenumbershortcodeupdate_js_1.PhoneNumberShortCodeUpdate$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "e164_number": "e164Number",
        "short_code": "shortCode",
    });
});
/** @internal */
exports.PhoneNumberUpdate$outboundSchema = z.object({
    e164Number: z.string().optional(),
    extension: z.string().optional(),
    shortCode: phonenumbershortcodeupdate_js_1.PhoneNumberShortCodeUpdate$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        e164Number: "e164_number",
        shortCode: "short_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PhoneNumberUpdate$;
(function (PhoneNumberUpdate$) {
    /** @deprecated use `PhoneNumberUpdate$inboundSchema` instead. */
    PhoneNumberUpdate$.inboundSchema = exports.PhoneNumberUpdate$inboundSchema;
    /** @deprecated use `PhoneNumberUpdate$outboundSchema` instead. */
    PhoneNumberUpdate$.outboundSchema = exports.PhoneNumberUpdate$outboundSchema;
})(PhoneNumberUpdate$ || (exports.PhoneNumberUpdate$ = PhoneNumberUpdate$ = {}));
