// @apexfintechsolutions/ascend-sdk/models/components/priceadjustmentcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PriceAdjustmentCreate$ = exports.PriceAdjustmentCreate$outboundSchema = exports.PriceAdjustmentCreate$inboundSchema = exports.PriceAdjustmentType$ = exports.PriceAdjustmentType$outboundSchema = exports.PriceAdjustmentType$inboundSchema = exports.PriceAdjustmentType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
/**
 * The type of price adjustment being applied by the broker to the net price of the security.
 */
var PriceAdjustmentType;
(function (PriceAdjustmentType) {
    PriceAdjustmentType["PriceAdjustmentTypeUnspecified"] = "PRICE_ADJUSTMENT_TYPE_UNSPECIFIED";
    PriceAdjustmentType["Markup"] = "MARKUP";
    PriceAdjustmentType["Markdown"] = "MARKDOWN";
    PriceAdjustmentType["GrossCredit"] = "GROSS_CREDIT";
})(PriceAdjustmentType || (exports.PriceAdjustmentType = PriceAdjustmentType = {}));
/** @internal */
exports.PriceAdjustmentType$inboundSchema = z
    .union([
    z.nativeEnum(PriceAdjustmentType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.PriceAdjustmentType$outboundSchema = z.union([
    z.nativeEnum(PriceAdjustmentType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PriceAdjustmentType$;
(function (PriceAdjustmentType$) {
    /** @deprecated use `PriceAdjustmentType$inboundSchema` instead. */
    PriceAdjustmentType$.inboundSchema = exports.PriceAdjustmentType$inboundSchema;
    /** @deprecated use `PriceAdjustmentType$outboundSchema` instead. */
    PriceAdjustmentType$.outboundSchema = exports.PriceAdjustmentType$outboundSchema;
})(PriceAdjustmentType$ || (exports.PriceAdjustmentType$ = PriceAdjustmentType$ = {}));
/** @internal */
exports.PriceAdjustmentCreate$inboundSchema = z.object({
    price_adjustment_amount: decimalcreate_js_1.DecimalCreate$inboundSchema,
    price_adjustment_type: exports.PriceAdjustmentType$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "price_adjustment_amount": "priceAdjustmentAmount",
        "price_adjustment_type": "priceAdjustmentType",
    });
});
/** @internal */
exports.PriceAdjustmentCreate$outboundSchema = z.object({
    priceAdjustmentAmount: decimalcreate_js_1.DecimalCreate$outboundSchema,
    priceAdjustmentType: exports.PriceAdjustmentType$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        priceAdjustmentAmount: "price_adjustment_amount",
        priceAdjustmentType: "price_adjustment_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PriceAdjustmentCreate$;
(function (PriceAdjustmentCreate$) {
    /** @deprecated use `PriceAdjustmentCreate$inboundSchema` instead. */
    PriceAdjustmentCreate$.inboundSchema = exports.PriceAdjustmentCreate$inboundSchema;
    /** @deprecated use `PriceAdjustmentCreate$outboundSchema` instead. */
    PriceAdjustmentCreate$.outboundSchema = exports.PriceAdjustmentCreate$outboundSchema;
})(PriceAdjustmentCreate$ || (exports.PriceAdjustmentCreate$ = PriceAdjustmentCreate$ = {}));
