// @apexfintechsolutions/ascend-sdk/models/components/pushsubscriptiondelivery.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PushSubscriptionDelivery$ = exports.PushSubscriptionDelivery$outboundSchema = exports.PushSubscriptionDelivery$inboundSchema = exports.Result$ = exports.Result$outboundSchema = exports.Result$inboundSchema = exports.Result = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * The outcome of the delivery
 */
var Result;
(function (Result) {
    Result["PushSubscriptionDeliveryResultUnspecified"] = "PUSH_SUBSCRIPTION_DELIVERY_RESULT_UNSPECIFIED";
    Result["Succeeded"] = "SUCCEEDED";
    Result["Failed"] = "FAILED";
})(Result || (exports.Result = Result = {}));
/** @internal */
exports.Result$inboundSchema = z
    .union([
    z.nativeEnum(Result),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.Result$outboundSchema = z.union([
    z.nativeEnum(Result),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Result$;
(function (Result$) {
    /** @deprecated use `Result$inboundSchema` instead. */
    Result$.inboundSchema = exports.Result$inboundSchema;
    /** @deprecated use `Result$outboundSchema` instead. */
    Result$.outboundSchema = exports.Result$outboundSchema;
})(Result$ || (exports.Result$ = Result$ = {}));
/** @internal */
exports.PushSubscriptionDelivery$inboundSchema = z.object({
    delivery_id: z.string().optional(),
    duration: z.string().optional(),
    event: z.string().optional(),
    event_publish_time: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    last_response: z.string().optional(),
    last_send_time: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    name: z.string().optional(),
    result: exports.Result$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "delivery_id": "deliveryId",
        "event_publish_time": "eventPublishTime",
        "last_response": "lastResponse",
        "last_send_time": "lastSendTime",
    });
});
/** @internal */
exports.PushSubscriptionDelivery$outboundSchema = z.object({
    deliveryId: z.string().optional(),
    duration: z.string().optional(),
    event: z.string().optional(),
    eventPublishTime: z.nullable(z.date().transform(v => v.toISOString()))
        .optional(),
    lastResponse: z.string().optional(),
    lastSendTime: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    name: z.string().optional(),
    result: exports.Result$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        deliveryId: "delivery_id",
        eventPublishTime: "event_publish_time",
        lastResponse: "last_response",
        lastSendTime: "last_send_time",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PushSubscriptionDelivery$;
(function (PushSubscriptionDelivery$) {
    /** @deprecated use `PushSubscriptionDelivery$inboundSchema` instead. */
    PushSubscriptionDelivery$.inboundSchema = exports.PushSubscriptionDelivery$inboundSchema;
    /** @deprecated use `PushSubscriptionDelivery$outboundSchema` instead. */
    PushSubscriptionDelivery$.outboundSchema = exports.PushSubscriptionDelivery$outboundSchema;
})(PushSubscriptionDelivery$ || (exports.PushSubscriptionDelivery$ = PushSubscriptionDelivery$ = {}));
