// @apexfintechsolutions/ascend-sdk/models/components/pushsubscriptionupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PushSubscriptionUpdate$ = exports.PushSubscriptionUpdate$outboundSchema = exports.PushSubscriptionUpdate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const httppushcallbackupdate_js_1 = require("./httppushcallbackupdate.js");
/** @internal */
exports.PushSubscriptionUpdate$inboundSchema = z.object({
    display_name: z.string().optional(),
    event_types: z.array(z.string()).optional(),
    http_callback: httppushcallbackupdate_js_1.HttpPushCallbackUpdate$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "display_name": "displayName",
        "event_types": "eventTypes",
        "http_callback": "httpCallback",
    });
});
/** @internal */
exports.PushSubscriptionUpdate$outboundSchema = z.object({
    displayName: z.string().optional(),
    eventTypes: z.array(z.string()).optional(),
    httpCallback: httppushcallbackupdate_js_1.HttpPushCallbackUpdate$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        displayName: "display_name",
        eventTypes: "event_types",
        httpCallback: "http_callback",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PushSubscriptionUpdate$;
(function (PushSubscriptionUpdate$) {
    /** @deprecated use `PushSubscriptionUpdate$inboundSchema` instead. */
    PushSubscriptionUpdate$.inboundSchema = exports.PushSubscriptionUpdate$inboundSchema;
    /** @deprecated use `PushSubscriptionUpdate$outboundSchema` instead. */
    PushSubscriptionUpdate$.outboundSchema = exports.PushSubscriptionUpdate$outboundSchema;
})(PushSubscriptionUpdate$ || (exports.PushSubscriptionUpdate$ = PushSubscriptionUpdate$ = {}));
