// @apexfintechsolutions/ascend-sdk/models/components/rebooktradeallocationrequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RebookTradeAllocationRequestCreate$ = exports.RebookTradeAllocationRequestCreate$outboundSchema = exports.RebookTradeAllocationRequestCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const tradeallocationcreate_js_1 = require("./tradeallocationcreate.js");
/** @internal */
exports.RebookTradeAllocationRequestCreate$inboundSchema = z.object({
    name: z.string(),
    request_id: z.string(),
    trade_allocation: tradeallocationcreate_js_1.TradeAllocationCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "request_id": "requestId",
        "trade_allocation": "tradeAllocation",
    });
});
/** @internal */
exports.RebookTradeAllocationRequestCreate$outboundSchema = z.object({
    name: z.string(),
    requestId: z.string(),
    tradeAllocation: tradeallocationcreate_js_1.TradeAllocationCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        requestId: "request_id",
        tradeAllocation: "trade_allocation",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RebookTradeAllocationRequestCreate$;
(function (RebookTradeAllocationRequestCreate$) {
    /** @deprecated use `RebookTradeAllocationRequestCreate$inboundSchema` instead. */
    RebookTradeAllocationRequestCreate$.inboundSchema = exports.RebookTradeAllocationRequestCreate$inboundSchema;
    /** @deprecated use `RebookTradeAllocationRequestCreate$outboundSchema` instead. */
    RebookTradeAllocationRequestCreate$.outboundSchema = exports.RebookTradeAllocationRequestCreate$outboundSchema;
})(RebookTradeAllocationRequestCreate$ || (exports.RebookTradeAllocationRequestCreate$ = RebookTradeAllocationRequestCreate$ = {}));
