// @apexfintechsolutions/ascend-sdk/models/components/recipientbankbankdetailscreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecipientBankBankDetailsCreate$ = exports.RecipientBankBankDetailsCreate$outboundSchema = exports.RecipientBankBankDetailsCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const addresscreate_js_1 = require("./addresscreate.js");
/** @internal */
exports.RecipientBankBankDetailsCreate$inboundSchema = z.object({
    additional_info: z.string().optional(),
    address: addresscreate_js_1.AddressCreate$inboundSchema,
    bank_name: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "additional_info": "additionalInfo",
        "bank_name": "bankName",
    });
});
/** @internal */
exports.RecipientBankBankDetailsCreate$outboundSchema = z.object({
    additionalInfo: z.string().optional(),
    address: addresscreate_js_1.AddressCreate$outboundSchema,
    bankName: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        additionalInfo: "additional_info",
        bankName: "bank_name",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RecipientBankBankDetailsCreate$;
(function (RecipientBankBankDetailsCreate$) {
    /** @deprecated use `RecipientBankBankDetailsCreate$inboundSchema` instead. */
    RecipientBankBankDetailsCreate$.inboundSchema = exports.RecipientBankBankDetailsCreate$inboundSchema;
    /** @deprecated use `RecipientBankBankDetailsCreate$outboundSchema` instead. */
    RecipientBankBankDetailsCreate$.outboundSchema = exports.RecipientBankBankDetailsCreate$outboundSchema;
})(RecipientBankBankDetailsCreate$ || (exports.RecipientBankBankDetailsCreate$ = RecipientBankBankDetailsCreate$ = {}));
