// @apexfintechsolutions/ascend-sdk/models/components/relatedprofile.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RelatedProfile$ = exports.RelatedProfile$outboundSchema = exports.RelatedProfile$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const descriptiondetail_js_1 = require("./descriptiondetail.js");
const sanctionslistdetail_js_1 = require("./sanctionslistdetail.js");
/** @internal */
exports.RelatedProfile$inboundSchema = z.object({
    description_details: z.array(descriptiondetail_js_1.DescriptionDetail$inboundSchema).optional(),
    dow_jones_person_id: z.number().int().optional(),
    given_name: z.string().optional(),
    middle_names: z.string().optional(),
    name_suffix: z.string().optional(),
    relationship_type: z.string().optional(),
    sanctions_list_details: z.array(sanctionslistdetail_js_1.SanctionsListDetail$inboundSchema).optional(),
    surname: z.string().optional(),
    title_honorific: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "description_details": "descriptionDetails",
        "dow_jones_person_id": "dowJonesPersonId",
        "given_name": "givenName",
        "middle_names": "middleNames",
        "name_suffix": "nameSuffix",
        "relationship_type": "relationshipType",
        "sanctions_list_details": "sanctionsListDetails",
        "title_honorific": "titleHonorific",
    });
});
/** @internal */
exports.RelatedProfile$outboundSchema = z.object({
    descriptionDetails: z.array(descriptiondetail_js_1.DescriptionDetail$outboundSchema).optional(),
    dowJonesPersonId: z.number().int().optional(),
    givenName: z.string().optional(),
    middleNames: z.string().optional(),
    nameSuffix: z.string().optional(),
    relationshipType: z.string().optional(),
    sanctionsListDetails: z.array(sanctionslistdetail_js_1.SanctionsListDetail$outboundSchema).optional(),
    surname: z.string().optional(),
    titleHonorific: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        descriptionDetails: "description_details",
        dowJonesPersonId: "dow_jones_person_id",
        givenName: "given_name",
        middleNames: "middle_names",
        nameSuffix: "name_suffix",
        relationshipType: "relationship_type",
        sanctionsListDetails: "sanctions_list_details",
        titleHonorific: "title_honorific",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RelatedProfile$;
(function (RelatedProfile$) {
    /** @deprecated use `RelatedProfile$inboundSchema` instead. */
    RelatedProfile$.inboundSchema = exports.RelatedProfile$inboundSchema;
    /** @deprecated use `RelatedProfile$outboundSchema` instead. */
    RelatedProfile$.outboundSchema = exports.RelatedProfile$outboundSchema;
})(RelatedProfile$ || (exports.RelatedProfile$ = RelatedProfile$ = {}));
