// @apexfintechsolutions/ascend-sdk/models/components/replacepartyrequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplacePartyRequestCreate$ = exports.ReplacePartyRequestCreate$outboundSchema = exports.ReplacePartyRequestCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const partyrequestcreate_js_1 = require("./partyrequestcreate.js");
/** @internal */
exports.ReplacePartyRequestCreate$inboundSchema = z.object({
    authorized_by_party_ids: z.array(z.string()).optional(),
    name: z.string(),
    party: partyrequestcreate_js_1.PartyRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "authorized_by_party_ids": "authorizedByPartyIds",
    });
});
/** @internal */
exports.ReplacePartyRequestCreate$outboundSchema = z.object({
    authorizedByPartyIds: z.array(z.string()).optional(),
    name: z.string(),
    party: partyrequestcreate_js_1.PartyRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        authorizedByPartyIds: "authorized_by_party_ids",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ReplacePartyRequestCreate$;
(function (ReplacePartyRequestCreate$) {
    /** @deprecated use `ReplacePartyRequestCreate$inboundSchema` instead. */
    ReplacePartyRequestCreate$.inboundSchema = exports.ReplacePartyRequestCreate$inboundSchema;
    /** @deprecated use `ReplacePartyRequestCreate$outboundSchema` instead. */
    ReplacePartyRequestCreate$.outboundSchema = exports.ReplacePartyRequestCreate$outboundSchema;
})(ReplacePartyRequestCreate$ || (exports.ReplacePartyRequestCreate$ = ReplacePartyRequestCreate$ = {}));
