// @apexfintechsolutions/ascend-sdk/models/components/restriction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Restriction$ = exports.Restriction$outboundSchema = exports.Restriction$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.Restriction$inboundSchema = z.object({
    active: z.boolean().optional(),
    create_reason: z.string().optional(),
    create_time: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    create_user_id: z.string().optional(),
    ended_reason: z.string().optional(),
    ended_time: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    ended_user_id: z.string().optional(),
    restriction_code: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "create_reason": "createReason",
        "create_time": "createTime",
        "create_user_id": "createUserId",
        "ended_reason": "endedReason",
        "ended_time": "endedTime",
        "ended_user_id": "endedUserId",
        "restriction_code": "restrictionCode",
    });
});
/** @internal */
exports.Restriction$outboundSchema = z.object({
    active: z.boolean().optional(),
    createReason: z.string().optional(),
    createTime: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    createUserId: z.string().optional(),
    endedReason: z.string().optional(),
    endedTime: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    endedUserId: z.string().optional(),
    restrictionCode: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createReason: "create_reason",
        createTime: "create_time",
        createUserId: "create_user_id",
        endedReason: "ended_reason",
        endedTime: "ended_time",
        endedUserId: "ended_user_id",
        restrictionCode: "restriction_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Restriction$;
(function (Restriction$) {
    /** @deprecated use `Restriction$inboundSchema` instead. */
    Restriction$.inboundSchema = exports.Restriction$inboundSchema;
    /** @deprecated use `Restriction$outboundSchema` instead. */
    Restriction$.outboundSchema = exports.Restriction$outboundSchema;
})(Restriction$ || (exports.Restriction$ = Restriction$ = {}));
