// @apexfintechsolutions/ascend-sdk/models/components/retirementdistributioncreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetirementDistributionCreate$ = exports.RetirementDistributionCreate$outboundSchema = exports.RetirementDistributionCreate$inboundSchema = exports.RetirementDistributionCreateType$ = exports.RetirementDistributionCreateType$outboundSchema = exports.RetirementDistributionCreateType$inboundSchema = exports.RetirementDistributionCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const retirementdistributiontaxwithholdingcreate_js_1 = require("./retirementdistributiontaxwithholdingcreate.js");
/**
 * The type of retirement distribution.
 */
var RetirementDistributionCreateType;
(function (RetirementDistributionCreateType) {
    RetirementDistributionCreateType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    RetirementDistributionCreateType["Normal"] = "NORMAL";
    RetirementDistributionCreateType["Disability"] = "DISABILITY";
    RetirementDistributionCreateType["Sosepp"] = "SOSEPP";
    RetirementDistributionCreateType["Premature"] = "PREMATURE";
    RetirementDistributionCreateType["Death"] = "DEATH";
    RetirementDistributionCreateType["ExcessContributionRemovalBeforeTaxDeadline"] = "EXCESS_CONTRIBUTION_REMOVAL_BEFORE_TAX_DEADLINE";
    RetirementDistributionCreateType["ExcessContributionRemovalAfterTaxDeadline"] = "EXCESS_CONTRIBUTION_REMOVAL_AFTER_TAX_DEADLINE";
    RetirementDistributionCreateType["RolloverToQualifiedPlan"] = "ROLLOVER_TO_QUALIFIED_PLAN";
    RetirementDistributionCreateType["RolloverToIra"] = "ROLLOVER_TO_IRA";
    RetirementDistributionCreateType["DistributionTransfer"] = "DISTRIBUTION_TRANSFER";
    RetirementDistributionCreateType["RecharacterizationPriorYear"] = "RECHARACTERIZATION_PRIOR_YEAR";
    RetirementDistributionCreateType["RecharacterizationCurrentYear"] = "RECHARACTERIZATION_CURRENT_YEAR";
    RetirementDistributionCreateType["DistributionConversion"] = "DISTRIBUTION_CONVERSION";
    RetirementDistributionCreateType["ManagementFee"] = "MANAGEMENT_FEE";
    RetirementDistributionCreateType["PlanLoan401K"] = "PLAN_LOAN_401K";
    RetirementDistributionCreateType["PrematureSimpleIraLessThan2Years"] = "PREMATURE_SIMPLE_IRA_LESS_THAN_2_YEARS";
    RetirementDistributionCreateType["NormalRothIraGreaterThan5Years"] = "NORMAL_ROTH_IRA_GREATER_THAN_5_YEARS";
    RetirementDistributionCreateType["NetIncomeAttributable"] = "NET_INCOME_ATTRIBUTABLE";
    RetirementDistributionCreateType["Revocation"] = "REVOCATION";
    RetirementDistributionCreateType["NonReportable"] = "NON_REPORTABLE";
})(RetirementDistributionCreateType || (exports.RetirementDistributionCreateType = RetirementDistributionCreateType = {}));
/** @internal */
exports.RetirementDistributionCreateType$inboundSchema = z
    .union([
    z.nativeEnum(RetirementDistributionCreateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.RetirementDistributionCreateType$outboundSchema = z.union([
    z.nativeEnum(RetirementDistributionCreateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetirementDistributionCreateType$;
(function (RetirementDistributionCreateType$) {
    /** @deprecated use `RetirementDistributionCreateType$inboundSchema` instead. */
    RetirementDistributionCreateType$.inboundSchema = exports.RetirementDistributionCreateType$inboundSchema;
    /** @deprecated use `RetirementDistributionCreateType$outboundSchema` instead. */
    RetirementDistributionCreateType$.outboundSchema = exports.RetirementDistributionCreateType$outboundSchema;
})(RetirementDistributionCreateType$ || (exports.RetirementDistributionCreateType$ = RetirementDistributionCreateType$ = {}));
/** @internal */
exports.RetirementDistributionCreate$inboundSchema = z.object({
    federal_tax_withholding: retirementdistributiontaxwithholdingcreate_js_1.RetirementDistributionTaxWithholdingCreate$inboundSchema.optional(),
    receiving_institution: z.string().optional(),
    state_tax_withholding: retirementdistributiontaxwithholdingcreate_js_1.RetirementDistributionTaxWithholdingCreate$inboundSchema.optional(),
    state_withholding_waiver: z.boolean().optional(),
    type: exports.RetirementDistributionCreateType$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "federal_tax_withholding": "federalTaxWithholding",
        "receiving_institution": "receivingInstitution",
        "state_tax_withholding": "stateTaxWithholding",
        "state_withholding_waiver": "stateWithholdingWaiver",
    });
});
/** @internal */
exports.RetirementDistributionCreate$outboundSchema = z.object({
    federalTaxWithholding: retirementdistributiontaxwithholdingcreate_js_1.RetirementDistributionTaxWithholdingCreate$outboundSchema.optional(),
    receivingInstitution: z.string().optional(),
    stateTaxWithholding: retirementdistributiontaxwithholdingcreate_js_1.RetirementDistributionTaxWithholdingCreate$outboundSchema
        .optional(),
    stateWithholdingWaiver: z.boolean().optional(),
    type: exports.RetirementDistributionCreateType$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        federalTaxWithholding: "federal_tax_withholding",
        receivingInstitution: "receiving_institution",
        stateTaxWithholding: "state_tax_withholding",
        stateWithholdingWaiver: "state_withholding_waiver",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetirementDistributionCreate$;
(function (RetirementDistributionCreate$) {
    /** @deprecated use `RetirementDistributionCreate$inboundSchema` instead. */
    RetirementDistributionCreate$.inboundSchema = exports.RetirementDistributionCreate$inboundSchema;
    /** @deprecated use `RetirementDistributionCreate$outboundSchema` instead. */
    RetirementDistributionCreate$.outboundSchema = exports.RetirementDistributionCreate$outboundSchema;
})(RetirementDistributionCreate$ || (exports.RetirementDistributionCreate$ = RetirementDistributionCreate$ = {}));
