// @apexfintechsolutions/ascend-sdk/models/components/retrievefixedincomemarksrequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetrieveFixedIncomeMarksRequestCreate$ = exports.RetrieveFixedIncomeMarksRequestCreate$outboundSchema = exports.RetrieveFixedIncomeMarksRequestCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const retrievefixedincomemarksrequestsecurityidentifierscreate_js_1 = require("./retrievefixedincomemarksrequestsecurityidentifierscreate.js");
/** @internal */
exports.RetrieveFixedIncomeMarksRequestCreate$inboundSchema = z.object({
    parent: z.string(),
    security_identifiers: z.array(retrievefixedincomemarksrequestsecurityidentifierscreate_js_1.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "security_identifiers": "securityIdentifiers",
    });
});
/** @internal */
exports.RetrieveFixedIncomeMarksRequestCreate$outboundSchema = z.object({
    parent: z.string(),
    securityIdentifiers: z.array(retrievefixedincomemarksrequestsecurityidentifierscreate_js_1.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        securityIdentifiers: "security_identifiers",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetrieveFixedIncomeMarksRequestCreate$;
(function (RetrieveFixedIncomeMarksRequestCreate$) {
    /** @deprecated use `RetrieveFixedIncomeMarksRequestCreate$inboundSchema` instead. */
    RetrieveFixedIncomeMarksRequestCreate$.inboundSchema = exports.RetrieveFixedIncomeMarksRequestCreate$inboundSchema;
    /** @deprecated use `RetrieveFixedIncomeMarksRequestCreate$outboundSchema` instead. */
    RetrieveFixedIncomeMarksRequestCreate$.outboundSchema = exports.RetrieveFixedIncomeMarksRequestCreate$outboundSchema;
})(RetrieveFixedIncomeMarksRequestCreate$ || (exports.RetrieveFixedIncomeMarksRequestCreate$ = RetrieveFixedIncomeMarksRequestCreate$ = {}));
