// @apexfintechsolutions/ascend-sdk/models/components/retrievequoterequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetrieveQuoteRequestCreate$ = exports.RetrieveQuoteRequestCreate$outboundSchema = exports.RetrieveQuoteRequestCreate$inboundSchema = exports.RetrieveQuoteRequestCreateIdentifierType$ = exports.RetrieveQuoteRequestCreateIdentifierType$outboundSchema = exports.RetrieveQuoteRequestCreateIdentifierType$inboundSchema = exports.RetrieveQuoteRequestCreateBrokerCapacity$ = exports.RetrieveQuoteRequestCreateBrokerCapacity$outboundSchema = exports.RetrieveQuoteRequestCreateBrokerCapacity$inboundSchema = exports.RetrieveQuoteRequestCreateAssetType$ = exports.RetrieveQuoteRequestCreateAssetType$outboundSchema = exports.RetrieveQuoteRequestCreateAssetType$inboundSchema = exports.RetrieveQuoteRequestCreateIdentifierType = exports.RetrieveQuoteRequestCreateBrokerCapacity = exports.RetrieveQuoteRequestCreateAssetType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
/**
 * The type of asset being evaluated in cost calculations
 */
var RetrieveQuoteRequestCreateAssetType;
(function (RetrieveQuoteRequestCreateAssetType) {
    RetrieveQuoteRequestCreateAssetType["FixedIncome"] = "FIXED_INCOME";
})(RetrieveQuoteRequestCreateAssetType || (exports.RetrieveQuoteRequestCreateAssetType = RetrieveQuoteRequestCreateAssetType = {}));
/**
 * Capacity used in determining bid and ask prices. Defaults to "AGENCY" if no value specified.
 */
var RetrieveQuoteRequestCreateBrokerCapacity;
(function (RetrieveQuoteRequestCreateBrokerCapacity) {
    RetrieveQuoteRequestCreateBrokerCapacity["BrokerCapacityUnspecified"] = "BROKER_CAPACITY_UNSPECIFIED";
    RetrieveQuoteRequestCreateBrokerCapacity["Agency"] = "AGENCY";
    RetrieveQuoteRequestCreateBrokerCapacity["Principal"] = "PRINCIPAL";
})(RetrieveQuoteRequestCreateBrokerCapacity || (exports.RetrieveQuoteRequestCreateBrokerCapacity = RetrieveQuoteRequestCreateBrokerCapacity = {}));
/**
 * The identifier type of the asset being sought
 */
var RetrieveQuoteRequestCreateIdentifierType;
(function (RetrieveQuoteRequestCreateIdentifierType) {
    RetrieveQuoteRequestCreateIdentifierType["AssetId"] = "ASSET_ID";
    RetrieveQuoteRequestCreateIdentifierType["Cusip"] = "CUSIP";
    RetrieveQuoteRequestCreateIdentifierType["Isin"] = "ISIN";
})(RetrieveQuoteRequestCreateIdentifierType || (exports.RetrieveQuoteRequestCreateIdentifierType = RetrieveQuoteRequestCreateIdentifierType = {}));
/** @internal */
exports.RetrieveQuoteRequestCreateAssetType$inboundSchema = z
    .union([
    z.nativeEnum(RetrieveQuoteRequestCreateAssetType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.RetrieveQuoteRequestCreateAssetType$outboundSchema = z.union([
    z.nativeEnum(RetrieveQuoteRequestCreateAssetType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetrieveQuoteRequestCreateAssetType$;
(function (RetrieveQuoteRequestCreateAssetType$) {
    /** @deprecated use `RetrieveQuoteRequestCreateAssetType$inboundSchema` instead. */
    RetrieveQuoteRequestCreateAssetType$.inboundSchema = exports.RetrieveQuoteRequestCreateAssetType$inboundSchema;
    /** @deprecated use `RetrieveQuoteRequestCreateAssetType$outboundSchema` instead. */
    RetrieveQuoteRequestCreateAssetType$.outboundSchema = exports.RetrieveQuoteRequestCreateAssetType$outboundSchema;
})(RetrieveQuoteRequestCreateAssetType$ || (exports.RetrieveQuoteRequestCreateAssetType$ = RetrieveQuoteRequestCreateAssetType$ = {}));
/** @internal */
exports.RetrieveQuoteRequestCreateBrokerCapacity$inboundSchema = z
    .union([
    z.nativeEnum(RetrieveQuoteRequestCreateBrokerCapacity),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.RetrieveQuoteRequestCreateBrokerCapacity$outboundSchema = z.union([
    z.nativeEnum(RetrieveQuoteRequestCreateBrokerCapacity),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetrieveQuoteRequestCreateBrokerCapacity$;
(function (RetrieveQuoteRequestCreateBrokerCapacity$) {
    /** @deprecated use `RetrieveQuoteRequestCreateBrokerCapacity$inboundSchema` instead. */
    RetrieveQuoteRequestCreateBrokerCapacity$.inboundSchema = exports.RetrieveQuoteRequestCreateBrokerCapacity$inboundSchema;
    /** @deprecated use `RetrieveQuoteRequestCreateBrokerCapacity$outboundSchema` instead. */
    RetrieveQuoteRequestCreateBrokerCapacity$.outboundSchema = exports.RetrieveQuoteRequestCreateBrokerCapacity$outboundSchema;
})(RetrieveQuoteRequestCreateBrokerCapacity$ || (exports.RetrieveQuoteRequestCreateBrokerCapacity$ = RetrieveQuoteRequestCreateBrokerCapacity$ = {}));
/** @internal */
exports.RetrieveQuoteRequestCreateIdentifierType$inboundSchema = z
    .union([
    z.nativeEnum(RetrieveQuoteRequestCreateIdentifierType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.RetrieveQuoteRequestCreateIdentifierType$outboundSchema = z.union([
    z.nativeEnum(RetrieveQuoteRequestCreateIdentifierType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetrieveQuoteRequestCreateIdentifierType$;
(function (RetrieveQuoteRequestCreateIdentifierType$) {
    /** @deprecated use `RetrieveQuoteRequestCreateIdentifierType$inboundSchema` instead. */
    RetrieveQuoteRequestCreateIdentifierType$.inboundSchema = exports.RetrieveQuoteRequestCreateIdentifierType$inboundSchema;
    /** @deprecated use `RetrieveQuoteRequestCreateIdentifierType$outboundSchema` instead. */
    RetrieveQuoteRequestCreateIdentifierType$.outboundSchema = exports.RetrieveQuoteRequestCreateIdentifierType$outboundSchema;
})(RetrieveQuoteRequestCreateIdentifierType$ || (exports.RetrieveQuoteRequestCreateIdentifierType$ = RetrieveQuoteRequestCreateIdentifierType$ = {}));
/** @internal */
exports.RetrieveQuoteRequestCreate$inboundSchema = z.object({
    asset_type: exports.RetrieveQuoteRequestCreateAssetType$inboundSchema,
    broker_capacity: exports.RetrieveQuoteRequestCreateBrokerCapacity$inboundSchema
        .optional(),
    identifier: z.string(),
    identifier_type: exports.RetrieveQuoteRequestCreateIdentifierType$inboundSchema,
    minimum_quantity: decimalcreate_js_1.DecimalCreate$inboundSchema.optional(),
    parent: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "asset_type": "assetType",
        "broker_capacity": "brokerCapacity",
        "identifier_type": "identifierType",
        "minimum_quantity": "minimumQuantity",
    });
});
/** @internal */
exports.RetrieveQuoteRequestCreate$outboundSchema = z.object({
    assetType: exports.RetrieveQuoteRequestCreateAssetType$outboundSchema,
    brokerCapacity: exports.RetrieveQuoteRequestCreateBrokerCapacity$outboundSchema
        .optional(),
    identifier: z.string(),
    identifierType: exports.RetrieveQuoteRequestCreateIdentifierType$outboundSchema,
    minimumQuantity: decimalcreate_js_1.DecimalCreate$outboundSchema.optional(),
    parent: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        assetType: "asset_type",
        brokerCapacity: "broker_capacity",
        identifierType: "identifier_type",
        minimumQuantity: "minimum_quantity",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetrieveQuoteRequestCreate$;
(function (RetrieveQuoteRequestCreate$) {
    /** @deprecated use `RetrieveQuoteRequestCreate$inboundSchema` instead. */
    RetrieveQuoteRequestCreate$.inboundSchema = exports.RetrieveQuoteRequestCreate$inboundSchema;
    /** @deprecated use `RetrieveQuoteRequestCreate$outboundSchema` instead. */
    RetrieveQuoteRequestCreate$.outboundSchema = exports.RetrieveQuoteRequestCreate$outboundSchema;
})(RetrieveQuoteRequestCreate$ || (exports.RetrieveQuoteRequestCreate$ = RetrieveQuoteRequestCreate$ = {}));
