// @apexfintechsolutions/ascend-sdk/models/components/retrievedistributionconstraintsrequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetrieveDistributionConstraintsRequestCreate$ = exports.RetrieveDistributionConstraintsRequestCreate$outboundSchema = exports.RetrieveDistributionConstraintsRequestCreate$inboundSchema = exports.RetrieveDistributionConstraintsRequestCreateMechanism$ = exports.RetrieveDistributionConstraintsRequestCreateMechanism$outboundSchema = exports.RetrieveDistributionConstraintsRequestCreateMechanism$inboundSchema = exports.RetrieveDistributionConstraintsRequestCreateMechanism = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
/**
 * Cash transfer mechanism to search constraints for
 */
var RetrieveDistributionConstraintsRequestCreateMechanism;
(function (RetrieveDistributionConstraintsRequestCreateMechanism) {
    RetrieveDistributionConstraintsRequestCreateMechanism["Ach"] = "ACH";
    RetrieveDistributionConstraintsRequestCreateMechanism["Ict"] = "ICT";
})(RetrieveDistributionConstraintsRequestCreateMechanism || (exports.RetrieveDistributionConstraintsRequestCreateMechanism = RetrieveDistributionConstraintsRequestCreateMechanism = {}));
/** @internal */
exports.RetrieveDistributionConstraintsRequestCreateMechanism$inboundSchema = z
    .union([
    z.nativeEnum(RetrieveDistributionConstraintsRequestCreateMechanism),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.RetrieveDistributionConstraintsRequestCreateMechanism$outboundSchema = z.union([
    z.nativeEnum(RetrieveDistributionConstraintsRequestCreateMechanism),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetrieveDistributionConstraintsRequestCreateMechanism$;
(function (RetrieveDistributionConstraintsRequestCreateMechanism$) {
    /** @deprecated use `RetrieveDistributionConstraintsRequestCreateMechanism$inboundSchema` instead. */
    RetrieveDistributionConstraintsRequestCreateMechanism$.inboundSchema = exports.RetrieveDistributionConstraintsRequestCreateMechanism$inboundSchema;
    /** @deprecated use `RetrieveDistributionConstraintsRequestCreateMechanism$outboundSchema` instead. */
    RetrieveDistributionConstraintsRequestCreateMechanism$.outboundSchema = exports.RetrieveDistributionConstraintsRequestCreateMechanism$outboundSchema;
})(RetrieveDistributionConstraintsRequestCreateMechanism$ || (exports.RetrieveDistributionConstraintsRequestCreateMechanism$ = RetrieveDistributionConstraintsRequestCreateMechanism$ = {}));
/** @internal */
exports.RetrieveDistributionConstraintsRequestCreate$inboundSchema = z.object({
    mechanism: exports.RetrieveDistributionConstraintsRequestCreateMechanism$inboundSchema,
    name: z.string(),
});
/** @internal */
exports.RetrieveDistributionConstraintsRequestCreate$outboundSchema = z.object({
    mechanism: exports.RetrieveDistributionConstraintsRequestCreateMechanism$outboundSchema,
    name: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetrieveDistributionConstraintsRequestCreate$;
(function (RetrieveDistributionConstraintsRequestCreate$) {
    /** @deprecated use `RetrieveDistributionConstraintsRequestCreate$inboundSchema` instead. */
    RetrieveDistributionConstraintsRequestCreate$.inboundSchema = exports.RetrieveDistributionConstraintsRequestCreate$inboundSchema;
    /** @deprecated use `RetrieveDistributionConstraintsRequestCreate$outboundSchema` instead. */
    RetrieveDistributionConstraintsRequestCreate$.outboundSchema = exports.RetrieveDistributionConstraintsRequestCreate$outboundSchema;
})(RetrieveDistributionConstraintsRequestCreate$ || (exports.RetrieveDistributionConstraintsRequestCreate$ = RetrieveDistributionConstraintsRequestCreate$ = {}));
