// @apexfintechsolutions/ascend-sdk/models/components/retrievecontributionconstraintsrequestcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetrieveContributionConstraintsRequestCreate$ = exports.RetrieveContributionConstraintsRequestCreate$outboundSchema = exports.RetrieveContributionConstraintsRequestCreate$inboundSchema = exports.Mechanism$ = exports.Mechanism$outboundSchema = exports.Mechanism$inboundSchema = exports.Mechanism = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
/**
 * Cash transfer mechanism to search constraints for
 */
var Mechanism;
(function (Mechanism) {
    Mechanism["Ach"] = "ACH";
    Mechanism["Ict"] = "ICT";
})(Mechanism || (exports.Mechanism = Mechanism = {}));
/** @internal */
exports.Mechanism$inboundSchema = z
    .union([
    z.nativeEnum(Mechanism),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.Mechanism$outboundSchema = z.union([
    z.nativeEnum(Mechanism),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Mechanism$;
(function (Mechanism$) {
    /** @deprecated use `Mechanism$inboundSchema` instead. */
    Mechanism$.inboundSchema = exports.Mechanism$inboundSchema;
    /** @deprecated use `Mechanism$outboundSchema` instead. */
    Mechanism$.outboundSchema = exports.Mechanism$outboundSchema;
})(Mechanism$ || (exports.Mechanism$ = Mechanism$ = {}));
/** @internal */
exports.RetrieveContributionConstraintsRequestCreate$inboundSchema = z.object({
    mechanism: exports.Mechanism$inboundSchema,
    name: z.string(),
});
/** @internal */
exports.RetrieveContributionConstraintsRequestCreate$outboundSchema = z.object({
    mechanism: exports.Mechanism$outboundSchema,
    name: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetrieveContributionConstraintsRequestCreate$;
(function (RetrieveContributionConstraintsRequestCreate$) {
    /** @deprecated use `RetrieveContributionConstraintsRequestCreate$inboundSchema` instead. */
    RetrieveContributionConstraintsRequestCreate$.inboundSchema = exports.RetrieveContributionConstraintsRequestCreate$inboundSchema;
    /** @deprecated use `RetrieveContributionConstraintsRequestCreate$outboundSchema` instead. */
    RetrieveContributionConstraintsRequestCreate$.outboundSchema = exports.RetrieveContributionConstraintsRequestCreate$outboundSchema;
})(RetrieveContributionConstraintsRequestCreate$ || (exports.RetrieveContributionConstraintsRequestCreate$ = RetrieveContributionConstraintsRequestCreate$ = {}));
