// @apexfintechsolutions/ascend-sdk/models/components/retrievefixedincomemarksrequestsecurityidentifierscreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$ = exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$outboundSchema = exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$inboundSchema = exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$ = exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$outboundSchema = exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$inboundSchema = exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * The identifier type of the asset being sought
 */
var RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType;
(function (RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType) {
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType["AssetId"] = "ASSET_ID";
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType["Cusip"] = "CUSIP";
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType["Isin"] = "ISIN";
})(RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType || (exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType = RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType = {}));
/** @internal */
exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$inboundSchema = z
    .union([
    z.nativeEnum(RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$outboundSchema = z.union([
    z.nativeEnum(RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$;
(function (RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$) {
    /** @deprecated use `RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$inboundSchema` instead. */
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$.inboundSchema = exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$inboundSchema;
    /** @deprecated use `RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$outboundSchema` instead. */
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$.outboundSchema = exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$outboundSchema;
})(RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$ || (exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$ = RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$ = {}));
/** @internal */
exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$inboundSchema = z.object({
    identifier: z.string(),
    identifier_type: exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "identifier_type": "identifierType",
    });
});
/** @internal */
exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$outboundSchema = z.object({
    identifier: z.string(),
    identifierType: exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreateIdentifierType$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        identifierType: "identifier_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$;
(function (RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$) {
    /** @deprecated use `RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$inboundSchema` instead. */
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$.inboundSchema = exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$inboundSchema;
    /** @deprecated use `RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$outboundSchema` instead. */
    RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$.outboundSchema = exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$outboundSchema;
})(RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$ || (exports.RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$ = RetrieveFixedIncomeMarksRequestSecurityIdentifiersCreate$ = {}));
