// @apexfintechsolutions/ascend-sdk/models/components/roledetail.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleDetail$ = exports.RoleDetail$outboundSchema = exports.RoleDetail$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.RoleDetail$inboundSchema = z.object({
    end_day: z.string().optional(),
    end_month: z.string().optional(),
    end_year: z.string().optional(),
    occupation_category: z.string().optional(),
    occupation_description: z.string().optional(),
    role_type: z.string().optional(),
    start_day: z.string().optional(),
    start_month: z.string().optional(),
    start_year: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "end_day": "endDay",
        "end_month": "endMonth",
        "end_year": "endYear",
        "occupation_category": "occupationCategory",
        "occupation_description": "occupationDescription",
        "role_type": "roleType",
        "start_day": "startDay",
        "start_month": "startMonth",
        "start_year": "startYear",
    });
});
/** @internal */
exports.RoleDetail$outboundSchema = z.object({
    endDay: z.string().optional(),
    endMonth: z.string().optional(),
    endYear: z.string().optional(),
    occupationCategory: z.string().optional(),
    occupationDescription: z.string().optional(),
    roleType: z.string().optional(),
    startDay: z.string().optional(),
    startMonth: z.string().optional(),
    startYear: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        endDay: "end_day",
        endMonth: "end_month",
        endYear: "end_year",
        occupationCategory: "occupation_category",
        occupationDescription: "occupation_description",
        roleType: "role_type",
        startDay: "start_day",
        startMonth: "start_month",
        startYear: "start_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RoleDetail$;
(function (RoleDetail$) {
    /** @deprecated use `RoleDetail$inboundSchema` instead. */
    RoleDetail$.inboundSchema = exports.RoleDetail$inboundSchema;
    /** @deprecated use `RoleDetail$outboundSchema` instead. */
    RoleDetail$.outboundSchema = exports.RoleDetail$outboundSchema;
})(RoleDetail$ || (exports.RoleDetail$ = RoleDetail$ = {}));
