// @apexfintechsolutions/ascend-sdk/models/components/scheduledretirementcontributioncreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduledRetirementContributionCreate$ = exports.ScheduledRetirementContributionCreate$outboundSchema = exports.ScheduledRetirementContributionCreate$inboundSchema = exports.ScheduledRetirementContributionCreateType$ = exports.ScheduledRetirementContributionCreateType$outboundSchema = exports.ScheduledRetirementContributionCreateType$inboundSchema = exports.TemporalTaxYear$ = exports.TemporalTaxYear$outboundSchema = exports.TemporalTaxYear$inboundSchema = exports.ScheduledRetirementContributionCreateType = exports.TemporalTaxYear = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * A temporal tax year value. This will always evaluate to a year based on the date the transfer was initiated.
 */
var TemporalTaxYear;
(function (TemporalTaxYear) {
    TemporalTaxYear["TemporalTaxYearUnspecified"] = "TEMPORAL_TAX_YEAR_UNSPECIFIED";
    TemporalTaxYear["CurrentCalendarYear"] = "CURRENT_CALENDAR_YEAR";
    TemporalTaxYear["MinimumTaxYear"] = "MINIMUM_TAX_YEAR";
})(TemporalTaxYear || (exports.TemporalTaxYear = TemporalTaxYear = {}));
/**
 * The type of retirement contribution.
 */
var ScheduledRetirementContributionCreateType;
(function (ScheduledRetirementContributionCreateType) {
    ScheduledRetirementContributionCreateType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    ScheduledRetirementContributionCreateType["Regular"] = "REGULAR";
    ScheduledRetirementContributionCreateType["Employee"] = "EMPLOYEE";
    ScheduledRetirementContributionCreateType["Employer"] = "EMPLOYER";
    ScheduledRetirementContributionCreateType["Recharacterization"] = "RECHARACTERIZATION";
    ScheduledRetirementContributionCreateType["Rollover60Day"] = "ROLLOVER_60_DAY";
    ScheduledRetirementContributionCreateType["RolloverDirect"] = "ROLLOVER_DIRECT";
    ScheduledRetirementContributionCreateType["Transfer"] = "TRANSFER";
    ScheduledRetirementContributionCreateType["TrusteeFee"] = "TRUSTEE_FEE";
    ScheduledRetirementContributionCreateType["Conversion"] = "CONVERSION";
    ScheduledRetirementContributionCreateType["Repayment"] = "REPAYMENT";
    ScheduledRetirementContributionCreateType["ContributionNonReportable"] = "CONTRIBUTION_NON_REPORTABLE";
})(ScheduledRetirementContributionCreateType || (exports.ScheduledRetirementContributionCreateType = ScheduledRetirementContributionCreateType = {}));
/** @internal */
exports.TemporalTaxYear$inboundSchema = z
    .union([
    z.nativeEnum(TemporalTaxYear),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TemporalTaxYear$outboundSchema = z.union([
    z.nativeEnum(TemporalTaxYear),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TemporalTaxYear$;
(function (TemporalTaxYear$) {
    /** @deprecated use `TemporalTaxYear$inboundSchema` instead. */
    TemporalTaxYear$.inboundSchema = exports.TemporalTaxYear$inboundSchema;
    /** @deprecated use `TemporalTaxYear$outboundSchema` instead. */
    TemporalTaxYear$.outboundSchema = exports.TemporalTaxYear$outboundSchema;
})(TemporalTaxYear$ || (exports.TemporalTaxYear$ = TemporalTaxYear$ = {}));
/** @internal */
exports.ScheduledRetirementContributionCreateType$inboundSchema = z
    .union([
    z.nativeEnum(ScheduledRetirementContributionCreateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.ScheduledRetirementContributionCreateType$outboundSchema = z.union([
    z.nativeEnum(ScheduledRetirementContributionCreateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ScheduledRetirementContributionCreateType$;
(function (ScheduledRetirementContributionCreateType$) {
    /** @deprecated use `ScheduledRetirementContributionCreateType$inboundSchema` instead. */
    ScheduledRetirementContributionCreateType$.inboundSchema = exports.ScheduledRetirementContributionCreateType$inboundSchema;
    /** @deprecated use `ScheduledRetirementContributionCreateType$outboundSchema` instead. */
    ScheduledRetirementContributionCreateType$.outboundSchema = exports.ScheduledRetirementContributionCreateType$outboundSchema;
})(ScheduledRetirementContributionCreateType$ || (exports.ScheduledRetirementContributionCreateType$ = ScheduledRetirementContributionCreateType$ = {}));
/** @internal */
exports.ScheduledRetirementContributionCreate$inboundSchema = z.object({
    tax_year: z.number().int().optional(),
    temporal_tax_year: exports.TemporalTaxYear$inboundSchema.optional(),
    type: exports.ScheduledRetirementContributionCreateType$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "tax_year": "taxYear",
        "temporal_tax_year": "temporalTaxYear",
    });
});
/** @internal */
exports.ScheduledRetirementContributionCreate$outboundSchema = z.object({
    taxYear: z.number().int().optional(),
    temporalTaxYear: exports.TemporalTaxYear$outboundSchema.optional(),
    type: exports.ScheduledRetirementContributionCreateType$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        taxYear: "tax_year",
        temporalTaxYear: "temporal_tax_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ScheduledRetirementContributionCreate$;
(function (ScheduledRetirementContributionCreate$) {
    /** @deprecated use `ScheduledRetirementContributionCreate$inboundSchema` instead. */
    ScheduledRetirementContributionCreate$.inboundSchema = exports.ScheduledRetirementContributionCreate$inboundSchema;
    /** @deprecated use `ScheduledRetirementContributionCreate$outboundSchema` instead. */
    ScheduledRetirementContributionCreate$.outboundSchema = exports.ScheduledRetirementContributionCreate$outboundSchema;
})(ScheduledRetirementContributionCreate$ || (exports.ScheduledRetirementContributionCreate$ = ScheduledRetirementContributionCreate$ = {}));
