// @apexfintechsolutions/ascend-sdk/models/components/schedulepropertiescreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchedulePropertiesCreate$ = exports.SchedulePropertiesCreate$outboundSchema = exports.SchedulePropertiesCreate$inboundSchema = exports.TimeUnit$ = exports.TimeUnit$outboundSchema = exports.TimeUnit$inboundSchema = exports.TimeUnit = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const datecreate_js_1 = require("./datecreate.js");
/**
 * The time unit used to calculate the interval between transfers. The time period between transfers in a scheduled series is the unit of time times the multiplier
 */
var TimeUnit;
(function (TimeUnit) {
    TimeUnit["Day"] = "DAY";
    TimeUnit["Week"] = "WEEK";
    TimeUnit["Month"] = "MONTH";
})(TimeUnit || (exports.TimeUnit = TimeUnit = {}));
/** @internal */
exports.TimeUnit$inboundSchema = z
    .union([
    z.nativeEnum(TimeUnit),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TimeUnit$outboundSchema = z.union([
    z.nativeEnum(TimeUnit),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TimeUnit$;
(function (TimeUnit$) {
    /** @deprecated use `TimeUnit$inboundSchema` instead. */
    TimeUnit$.inboundSchema = exports.TimeUnit$inboundSchema;
    /** @deprecated use `TimeUnit$outboundSchema` instead. */
    TimeUnit$.outboundSchema = exports.TimeUnit$outboundSchema;
})(TimeUnit$ || (exports.TimeUnit$ = TimeUnit$ = {}));
/** @internal */
exports.SchedulePropertiesCreate$inboundSchema = z.object({
    occurrences: z.number().int().optional(),
    start_date: datecreate_js_1.DateCreate$inboundSchema,
    time_unit: exports.TimeUnit$inboundSchema,
    unit_multiplier: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "start_date": "startDate",
        "time_unit": "timeUnit",
        "unit_multiplier": "unitMultiplier",
    });
});
/** @internal */
exports.SchedulePropertiesCreate$outboundSchema = z.object({
    occurrences: z.number().int().optional(),
    startDate: datecreate_js_1.DateCreate$outboundSchema,
    timeUnit: exports.TimeUnit$outboundSchema,
    unitMultiplier: z.number().int(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        startDate: "start_date",
        timeUnit: "time_unit",
        unitMultiplier: "unit_multiplier",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SchedulePropertiesCreate$;
(function (SchedulePropertiesCreate$) {
    /** @deprecated use `SchedulePropertiesCreate$inboundSchema` instead. */
    SchedulePropertiesCreate$.inboundSchema = exports.SchedulePropertiesCreate$inboundSchema;
    /** @deprecated use `SchedulePropertiesCreate$outboundSchema` instead. */
    SchedulePropertiesCreate$.outboundSchema = exports.SchedulePropertiesCreate$outboundSchema;
})(SchedulePropertiesCreate$ || (exports.SchedulePropertiesCreate$ = SchedulePropertiesCreate$ = {}));
