// @apexfintechsolutions/ascend-sdk/models/components/stoppricecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.StopPriceCreate$ = exports.StopPriceCreate$outboundSchema = exports.StopPriceCreate$inboundSchema = exports.StopPriceCreateType$ = exports.StopPriceCreateType$outboundSchema = exports.StopPriceCreateType$inboundSchema = exports.StopPriceCreateType = void 0;
const z = __importStar(require("zod"));
const enums_js_1 = require("../../types/enums.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
/**
 * The type of this price, which must PRICE_PER_UNIT for equity orders. (Fixed income and mutual fund assets do not support stop orders.)
 */
var StopPriceCreateType;
(function (StopPriceCreateType) {
    StopPriceCreateType["StopPriceTypeUnspecified"] = "STOP_PRICE_TYPE_UNSPECIFIED";
    StopPriceCreateType["PricePerUnit"] = "PRICE_PER_UNIT";
})(StopPriceCreateType || (exports.StopPriceCreateType = StopPriceCreateType = {}));
/** @internal */
exports.StopPriceCreateType$inboundSchema = z
    .union([
    z.nativeEnum(StopPriceCreateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.StopPriceCreateType$outboundSchema = z.union([
    z.nativeEnum(StopPriceCreateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StopPriceCreateType$;
(function (StopPriceCreateType$) {
    /** @deprecated use `StopPriceCreateType$inboundSchema` instead. */
    StopPriceCreateType$.inboundSchema = exports.StopPriceCreateType$inboundSchema;
    /** @deprecated use `StopPriceCreateType$outboundSchema` instead. */
    StopPriceCreateType$.outboundSchema = exports.StopPriceCreateType$outboundSchema;
})(StopPriceCreateType$ || (exports.StopPriceCreateType$ = StopPriceCreateType$ = {}));
/** @internal */
exports.StopPriceCreate$inboundSchema = z.object({
    price: decimalcreate_js_1.DecimalCreate$inboundSchema,
    type: exports.StopPriceCreateType$inboundSchema,
});
/** @internal */
exports.StopPriceCreate$outboundSchema = z.object({
    price: decimalcreate_js_1.DecimalCreate$outboundSchema,
    type: exports.StopPriceCreateType$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var StopPriceCreate$;
(function (StopPriceCreate$) {
    /** @deprecated use `StopPriceCreate$inboundSchema` instead. */
    StopPriceCreate$.inboundSchema = exports.StopPriceCreate$inboundSchema;
    /** @deprecated use `StopPriceCreate$outboundSchema` instead. */
    StopPriceCreate$.outboundSchema = exports.StopPriceCreate$outboundSchema;
})(StopPriceCreate$ || (exports.StopPriceCreate$ = StopPriceCreate$ = {}));
