// @apexfintechsolutions/ascend-sdk/models/components/taxprofileupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxProfileUpdate$ = exports.TaxProfileUpdate$outboundSchema = exports.TaxProfileUpdate$inboundSchema = exports.TaxProfileUpdateUsTinStatus$ = exports.TaxProfileUpdateUsTinStatus$outboundSchema = exports.TaxProfileUpdateUsTinStatus$inboundSchema = exports.TaxProfileUpdateIrsFormType$ = exports.TaxProfileUpdateIrsFormType$outboundSchema = exports.TaxProfileUpdateIrsFormType$inboundSchema = exports.TaxProfileUpdateFederalTaxClassification$ = exports.TaxProfileUpdateFederalTaxClassification$outboundSchema = exports.TaxProfileUpdateFederalTaxClassification$inboundSchema = exports.TaxProfileUpdateUsTinStatus = exports.TaxProfileUpdateIrsFormType = exports.TaxProfileUpdateFederalTaxClassification = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Federal tax classification.
 */
var TaxProfileUpdateFederalTaxClassification;
(function (TaxProfileUpdateFederalTaxClassification) {
    TaxProfileUpdateFederalTaxClassification["FederalTaxClassificationUnspecified"] = "FEDERAL_TAX_CLASSIFICATION_UNSPECIFIED";
    TaxProfileUpdateFederalTaxClassification["IndivSolepropOrSinglememberllc"] = "INDIV_SOLEPROP_OR_SINGLEMEMBERLLC";
    TaxProfileUpdateFederalTaxClassification["Partnership"] = "PARTNERSHIP";
    TaxProfileUpdateFederalTaxClassification["CCorporation"] = "C_CORPORATION";
    TaxProfileUpdateFederalTaxClassification["SCorporation"] = "S_CORPORATION";
    TaxProfileUpdateFederalTaxClassification["TrustEstate"] = "TRUST_ESTATE";
    TaxProfileUpdateFederalTaxClassification["LlcTaxedAsCCorp"] = "LLC_TAXED_AS_C_CORP";
    TaxProfileUpdateFederalTaxClassification["LlcTaxedAsSCorp"] = "LLC_TAXED_AS_S_CORP";
    TaxProfileUpdateFederalTaxClassification["LlcTaxedAsPartnership"] = "LLC_TAXED_AS_PARTNERSHIP";
    TaxProfileUpdateFederalTaxClassification["Other"] = "OTHER";
})(TaxProfileUpdateFederalTaxClassification || (exports.TaxProfileUpdateFederalTaxClassification = TaxProfileUpdateFederalTaxClassification = {}));
/**
 * IRS form type.
 */
var TaxProfileUpdateIrsFormType;
(function (TaxProfileUpdateIrsFormType) {
    TaxProfileUpdateIrsFormType["IrsFormTypeUnspecified"] = "IRS_FORM_TYPE_UNSPECIFIED";
    TaxProfileUpdateIrsFormType["W9"] = "W_9";
    TaxProfileUpdateIrsFormType["W8Ben"] = "W_8BEN";
})(TaxProfileUpdateIrsFormType || (exports.TaxProfileUpdateIrsFormType = TaxProfileUpdateIrsFormType = {}));
/**
 * United States Individual Taxpayer Identification Number (ITIN) status.
 */
var TaxProfileUpdateUsTinStatus;
(function (TaxProfileUpdateUsTinStatus) {
    TaxProfileUpdateUsTinStatus["UsTinStatusUnspecified"] = "US_TIN_STATUS_UNSPECIFIED";
    TaxProfileUpdateUsTinStatus["Passing"] = "PASSING";
    TaxProfileUpdateUsTinStatus["Failing"] = "FAILING";
})(TaxProfileUpdateUsTinStatus || (exports.TaxProfileUpdateUsTinStatus = TaxProfileUpdateUsTinStatus = {}));
/** @internal */
exports.TaxProfileUpdateFederalTaxClassification$inboundSchema = z
    .union([
    z.nativeEnum(TaxProfileUpdateFederalTaxClassification),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TaxProfileUpdateFederalTaxClassification$outboundSchema = z.union([
    z.nativeEnum(TaxProfileUpdateFederalTaxClassification),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TaxProfileUpdateFederalTaxClassification$;
(function (TaxProfileUpdateFederalTaxClassification$) {
    /** @deprecated use `TaxProfileUpdateFederalTaxClassification$inboundSchema` instead. */
    TaxProfileUpdateFederalTaxClassification$.inboundSchema = exports.TaxProfileUpdateFederalTaxClassification$inboundSchema;
    /** @deprecated use `TaxProfileUpdateFederalTaxClassification$outboundSchema` instead. */
    TaxProfileUpdateFederalTaxClassification$.outboundSchema = exports.TaxProfileUpdateFederalTaxClassification$outboundSchema;
})(TaxProfileUpdateFederalTaxClassification$ || (exports.TaxProfileUpdateFederalTaxClassification$ = TaxProfileUpdateFederalTaxClassification$ = {}));
/** @internal */
exports.TaxProfileUpdateIrsFormType$inboundSchema = z
    .union([
    z.nativeEnum(TaxProfileUpdateIrsFormType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TaxProfileUpdateIrsFormType$outboundSchema = z.union([
    z.nativeEnum(TaxProfileUpdateIrsFormType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TaxProfileUpdateIrsFormType$;
(function (TaxProfileUpdateIrsFormType$) {
    /** @deprecated use `TaxProfileUpdateIrsFormType$inboundSchema` instead. */
    TaxProfileUpdateIrsFormType$.inboundSchema = exports.TaxProfileUpdateIrsFormType$inboundSchema;
    /** @deprecated use `TaxProfileUpdateIrsFormType$outboundSchema` instead. */
    TaxProfileUpdateIrsFormType$.outboundSchema = exports.TaxProfileUpdateIrsFormType$outboundSchema;
})(TaxProfileUpdateIrsFormType$ || (exports.TaxProfileUpdateIrsFormType$ = TaxProfileUpdateIrsFormType$ = {}));
/** @internal */
exports.TaxProfileUpdateUsTinStatus$inboundSchema = z
    .union([
    z.nativeEnum(TaxProfileUpdateUsTinStatus),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TaxProfileUpdateUsTinStatus$outboundSchema = z.union([
    z.nativeEnum(TaxProfileUpdateUsTinStatus),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TaxProfileUpdateUsTinStatus$;
(function (TaxProfileUpdateUsTinStatus$) {
    /** @deprecated use `TaxProfileUpdateUsTinStatus$inboundSchema` instead. */
    TaxProfileUpdateUsTinStatus$.inboundSchema = exports.TaxProfileUpdateUsTinStatus$inboundSchema;
    /** @deprecated use `TaxProfileUpdateUsTinStatus$outboundSchema` instead. */
    TaxProfileUpdateUsTinStatus$.outboundSchema = exports.TaxProfileUpdateUsTinStatus$outboundSchema;
})(TaxProfileUpdateUsTinStatus$ || (exports.TaxProfileUpdateUsTinStatus$ = TaxProfileUpdateUsTinStatus$ = {}));
/** @internal */
exports.TaxProfileUpdate$inboundSchema = z.object({
    federal_tax_classification: exports.TaxProfileUpdateFederalTaxClassification$inboundSchema.optional(),
    irs_form_type: exports.TaxProfileUpdateIrsFormType$inboundSchema.optional(),
    legal_tax_region_code: z.string().optional(),
    us_tin_status: exports.TaxProfileUpdateUsTinStatus$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "federal_tax_classification": "federalTaxClassification",
        "irs_form_type": "irsFormType",
        "legal_tax_region_code": "legalTaxRegionCode",
        "us_tin_status": "usTinStatus",
    });
});
/** @internal */
exports.TaxProfileUpdate$outboundSchema = z.object({
    federalTaxClassification: exports.TaxProfileUpdateFederalTaxClassification$outboundSchema.optional(),
    irsFormType: exports.TaxProfileUpdateIrsFormType$outboundSchema.optional(),
    legalTaxRegionCode: z.string().optional(),
    usTinStatus: exports.TaxProfileUpdateUsTinStatus$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        federalTaxClassification: "federal_tax_classification",
        irsFormType: "irs_form_type",
        legalTaxRegionCode: "legal_tax_region_code",
        usTinStatus: "us_tin_status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TaxProfileUpdate$;
(function (TaxProfileUpdate$) {
    /** @deprecated use `TaxProfileUpdate$inboundSchema` instead. */
    TaxProfileUpdate$.inboundSchema = exports.TaxProfileUpdate$inboundSchema;
    /** @deprecated use `TaxProfileUpdate$outboundSchema` instead. */
    TaxProfileUpdate$.outboundSchema = exports.TaxProfileUpdate$outboundSchema;
})(TaxProfileUpdate$ || (exports.TaxProfileUpdate$ = TaxProfileUpdate$ = {}));
