// @apexfintechsolutions/ascend-sdk/models/components/trade.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Trade$ = exports.Trade$outboundSchema = exports.Trade$inboundSchema = exports.PriceAdjustmentRecord$ = exports.PriceAdjustmentRecord$outboundSchema = exports.PriceAdjustmentRecord$inboundSchema = exports.TradePriceAdjustmentType$ = exports.TradePriceAdjustmentType$outboundSchema = exports.TradePriceAdjustmentType$inboundSchema = exports.PriceAdjustmentPercent$ = exports.PriceAdjustmentPercent$outboundSchema = exports.PriceAdjustmentPercent$inboundSchema = exports.PriceAdjustmentAmount$ = exports.PriceAdjustmentAmount$outboundSchema = exports.PriceAdjustmentAmount$inboundSchema = exports.TradePrevailingMarketPrice$ = exports.TradePrevailingMarketPrice$outboundSchema = exports.TradePrevailingMarketPrice$inboundSchema = exports.TradeBrokerCapacity$ = exports.TradeBrokerCapacity$outboundSchema = exports.TradeBrokerCapacity$inboundSchema = exports.TradePriceAdjustmentType = exports.TradeBrokerCapacity = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const lot_js_1 = require("./lot.js");
const yieldrecord_js_1 = require("./yieldrecord.js");
/**
 * Used to calculate broadridge blotter code
 */
var TradeBrokerCapacity;
(function (TradeBrokerCapacity) {
    TradeBrokerCapacity["CapacityUnspecified"] = "CAPACITY_UNSPECIFIED";
    TradeBrokerCapacity["Agency"] = "AGENCY";
    TradeBrokerCapacity["Principal"] = "PRINCIPAL";
    TradeBrokerCapacity["Mixed"] = "MIXED";
})(TradeBrokerCapacity || (exports.TradeBrokerCapacity = TradeBrokerCapacity = {}));
/**
 * The type of price adjustment being applied by the broker to the net price of the security
 */
var TradePriceAdjustmentType;
(function (TradePriceAdjustmentType) {
    TradePriceAdjustmentType["PriceAdjustmentTypeUnspecified"] = "PRICE_ADJUSTMENT_TYPE_UNSPECIFIED";
    TradePriceAdjustmentType["Markup"] = "MARKUP";
    TradePriceAdjustmentType["Markdown"] = "MARKDOWN";
    TradePriceAdjustmentType["SalesLoad"] = "SALES_LOAD";
})(TradePriceAdjustmentType || (exports.TradePriceAdjustmentType = TradePriceAdjustmentType = {}));
/** @internal */
exports.TradeBrokerCapacity$inboundSchema = z
    .union([
    z.nativeEnum(TradeBrokerCapacity),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TradeBrokerCapacity$outboundSchema = z.union([
    z.nativeEnum(TradeBrokerCapacity),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TradeBrokerCapacity$;
(function (TradeBrokerCapacity$) {
    /** @deprecated use `TradeBrokerCapacity$inboundSchema` instead. */
    TradeBrokerCapacity$.inboundSchema = exports.TradeBrokerCapacity$inboundSchema;
    /** @deprecated use `TradeBrokerCapacity$outboundSchema` instead. */
    TradeBrokerCapacity$.outboundSchema = exports.TradeBrokerCapacity$outboundSchema;
})(TradeBrokerCapacity$ || (exports.TradeBrokerCapacity$ = TradeBrokerCapacity$ = {}));
/** @internal */
exports.TradePrevailingMarketPrice$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.TradePrevailingMarketPrice$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TradePrevailingMarketPrice$;
(function (TradePrevailingMarketPrice$) {
    /** @deprecated use `TradePrevailingMarketPrice$inboundSchema` instead. */
    TradePrevailingMarketPrice$.inboundSchema = exports.TradePrevailingMarketPrice$inboundSchema;
    /** @deprecated use `TradePrevailingMarketPrice$outboundSchema` instead. */
    TradePrevailingMarketPrice$.outboundSchema = exports.TradePrevailingMarketPrice$outboundSchema;
})(TradePrevailingMarketPrice$ || (exports.TradePrevailingMarketPrice$ = TradePrevailingMarketPrice$ = {}));
/** @internal */
exports.PriceAdjustmentAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.PriceAdjustmentAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PriceAdjustmentAmount$;
(function (PriceAdjustmentAmount$) {
    /** @deprecated use `PriceAdjustmentAmount$inboundSchema` instead. */
    PriceAdjustmentAmount$.inboundSchema = exports.PriceAdjustmentAmount$inboundSchema;
    /** @deprecated use `PriceAdjustmentAmount$outboundSchema` instead. */
    PriceAdjustmentAmount$.outboundSchema = exports.PriceAdjustmentAmount$outboundSchema;
})(PriceAdjustmentAmount$ || (exports.PriceAdjustmentAmount$ = PriceAdjustmentAmount$ = {}));
/** @internal */
exports.PriceAdjustmentPercent$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.PriceAdjustmentPercent$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PriceAdjustmentPercent$;
(function (PriceAdjustmentPercent$) {
    /** @deprecated use `PriceAdjustmentPercent$inboundSchema` instead. */
    PriceAdjustmentPercent$.inboundSchema = exports.PriceAdjustmentPercent$inboundSchema;
    /** @deprecated use `PriceAdjustmentPercent$outboundSchema` instead. */
    PriceAdjustmentPercent$.outboundSchema = exports.PriceAdjustmentPercent$outboundSchema;
})(PriceAdjustmentPercent$ || (exports.PriceAdjustmentPercent$ = PriceAdjustmentPercent$ = {}));
/** @internal */
exports.TradePriceAdjustmentType$inboundSchema = z
    .union([
    z.nativeEnum(TradePriceAdjustmentType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TradePriceAdjustmentType$outboundSchema = z.union([
    z.nativeEnum(TradePriceAdjustmentType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TradePriceAdjustmentType$;
(function (TradePriceAdjustmentType$) {
    /** @deprecated use `TradePriceAdjustmentType$inboundSchema` instead. */
    TradePriceAdjustmentType$.inboundSchema = exports.TradePriceAdjustmentType$inboundSchema;
    /** @deprecated use `TradePriceAdjustmentType$outboundSchema` instead. */
    TradePriceAdjustmentType$.outboundSchema = exports.TradePriceAdjustmentType$outboundSchema;
})(TradePriceAdjustmentType$ || (exports.TradePriceAdjustmentType$ = TradePriceAdjustmentType$ = {}));
/** @internal */
exports.PriceAdjustmentRecord$inboundSchema = z.object({
    price_adjustment_amount: z.nullable(z.lazy(() => exports.PriceAdjustmentAmount$inboundSchema)).optional(),
    price_adjustment_percent: z.nullable(z.lazy(() => exports.PriceAdjustmentPercent$inboundSchema)).optional(),
    price_adjustment_type: exports.TradePriceAdjustmentType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "price_adjustment_amount": "priceAdjustmentAmount",
        "price_adjustment_percent": "priceAdjustmentPercent",
        "price_adjustment_type": "priceAdjustmentType",
    });
});
/** @internal */
exports.PriceAdjustmentRecord$outboundSchema = z.object({
    priceAdjustmentAmount: z.nullable(z.lazy(() => exports.PriceAdjustmentAmount$outboundSchema)).optional(),
    priceAdjustmentPercent: z.nullable(z.lazy(() => exports.PriceAdjustmentPercent$outboundSchema)).optional(),
    priceAdjustmentType: exports.TradePriceAdjustmentType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        priceAdjustmentAmount: "price_adjustment_amount",
        priceAdjustmentPercent: "price_adjustment_percent",
        priceAdjustmentType: "price_adjustment_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PriceAdjustmentRecord$;
(function (PriceAdjustmentRecord$) {
    /** @deprecated use `PriceAdjustmentRecord$inboundSchema` instead. */
    PriceAdjustmentRecord$.inboundSchema = exports.PriceAdjustmentRecord$inboundSchema;
    /** @deprecated use `PriceAdjustmentRecord$outboundSchema` instead. */
    PriceAdjustmentRecord$.outboundSchema = exports.PriceAdjustmentRecord$outboundSchema;
})(PriceAdjustmentRecord$ || (exports.PriceAdjustmentRecord$ = PriceAdjustmentRecord$ = {}));
/** @internal */
exports.Trade$inboundSchema = z
    .object({
    additional_instructions: z.array(z.string()).optional(),
    alternate_order_id: z.string().optional(),
    booking_api_trade_allocation_id: z.string().optional(),
    booking_api_trade_id: z.string().optional(),
    broker: z.string().optional(),
    broker_capacity: exports.TradeBrokerCapacity$inboundSchema.optional(),
    client_memos: z.array(z.string()).optional(),
    client_order_id: z.string().optional(),
    exchange: z.string().optional(),
    execution_id: z.string().optional(),
    execution_only: z.boolean().optional(),
    external_id: z.string().optional(),
    fund_confirmation_number: z.string().optional(),
    gateway_client_order_id: z.string().optional(),
    internal_error: z.boolean().optional(),
    is_writeoff: z.boolean().optional(),
    lots: z.array(lot_js_1.Lot$inboundSchema).optional(),
    order_id: z.string().optional(),
    prevailing_market_price: z.nullable(z.lazy(() => exports.TradePrevailingMarketPrice$inboundSchema)).optional(),
    price_adjustment_record: z.nullable(z.lazy(() => exports.PriceAdjustmentRecord$inboundSchema)).optional(),
    route: z.string().optional(),
    special_instructions: z.array(z.string()).optional(),
    symbol_description: z.string().optional(),
    when_issued: z.boolean().optional(),
    yield_records: z.array(yieldrecord_js_1.YieldRecord$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "additional_instructions": "additionalInstructions",
        "alternate_order_id": "alternateOrderId",
        "booking_api_trade_allocation_id": "bookingApiTradeAllocationId",
        "booking_api_trade_id": "bookingApiTradeId",
        "broker_capacity": "brokerCapacity",
        "client_memos": "clientMemos",
        "client_order_id": "clientOrderId",
        "execution_id": "executionId",
        "execution_only": "executionOnly",
        "external_id": "externalId",
        "fund_confirmation_number": "fundConfirmationNumber",
        "gateway_client_order_id": "gatewayClientOrderId",
        "internal_error": "internalError",
        "is_writeoff": "isWriteoff",
        "order_id": "orderId",
        "prevailing_market_price": "prevailingMarketPrice",
        "price_adjustment_record": "priceAdjustmentRecord",
        "special_instructions": "specialInstructions",
        "symbol_description": "symbolDescription",
        "when_issued": "whenIssued",
        "yield_records": "yieldRecords",
    });
});
/** @internal */
exports.Trade$outboundSchema = z.object({
    additionalInstructions: z.array(z.string()).optional(),
    alternateOrderId: z.string().optional(),
    bookingApiTradeAllocationId: z.string().optional(),
    bookingApiTradeId: z.string().optional(),
    broker: z.string().optional(),
    brokerCapacity: exports.TradeBrokerCapacity$outboundSchema.optional(),
    clientMemos: z.array(z.string()).optional(),
    clientOrderId: z.string().optional(),
    exchange: z.string().optional(),
    executionId: z.string().optional(),
    executionOnly: z.boolean().optional(),
    externalId: z.string().optional(),
    fundConfirmationNumber: z.string().optional(),
    gatewayClientOrderId: z.string().optional(),
    internalError: z.boolean().optional(),
    isWriteoff: z.boolean().optional(),
    lots: z.array(lot_js_1.Lot$outboundSchema).optional(),
    orderId: z.string().optional(),
    prevailingMarketPrice: z.nullable(z.lazy(() => exports.TradePrevailingMarketPrice$outboundSchema)).optional(),
    priceAdjustmentRecord: z.nullable(z.lazy(() => exports.PriceAdjustmentRecord$outboundSchema)).optional(),
    route: z.string().optional(),
    specialInstructions: z.array(z.string()).optional(),
    symbolDescription: z.string().optional(),
    whenIssued: z.boolean().optional(),
    yieldRecords: z.array(yieldrecord_js_1.YieldRecord$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        additionalInstructions: "additional_instructions",
        alternateOrderId: "alternate_order_id",
        bookingApiTradeAllocationId: "booking_api_trade_allocation_id",
        bookingApiTradeId: "booking_api_trade_id",
        brokerCapacity: "broker_capacity",
        clientMemos: "client_memos",
        clientOrderId: "client_order_id",
        executionId: "execution_id",
        executionOnly: "execution_only",
        externalId: "external_id",
        fundConfirmationNumber: "fund_confirmation_number",
        gatewayClientOrderId: "gateway_client_order_id",
        internalError: "internal_error",
        isWriteoff: "is_writeoff",
        orderId: "order_id",
        prevailingMarketPrice: "prevailing_market_price",
        priceAdjustmentRecord: "price_adjustment_record",
        specialInstructions: "special_instructions",
        symbolDescription: "symbol_description",
        whenIssued: "when_issued",
        yieldRecords: "yield_records",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Trade$;
(function (Trade$) {
    /** @deprecated use `Trade$inboundSchema` instead. */
    Trade$.inboundSchema = exports.Trade$inboundSchema;
    /** @deprecated use `Trade$outboundSchema` instead. */
    Trade$.outboundSchema = exports.Trade$outboundSchema;
})(Trade$ || (exports.Trade$ = Trade$ = {}));
