// @apexfintechsolutions/ascend-sdk/models/components/transfersfeecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransfersFeeCreate$ = exports.TransfersFeeCreate$outboundSchema = exports.TransfersFeeCreate$inboundSchema = exports.TransfersFeeCreateType$ = exports.TransfersFeeCreateType$outboundSchema = exports.TransfersFeeCreateType$inboundSchema = exports.TransfersFeeCreateType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
/**
 * The type of the fee being charged
 */
var TransfersFeeCreateType;
(function (TransfersFeeCreateType) {
    TransfersFeeCreateType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    TransfersFeeCreateType["Management"] = "MANAGEMENT";
    TransfersFeeCreateType["Advisory"] = "ADVISORY";
    TransfersFeeCreateType["Exchange"] = "EXCHANGE";
    TransfersFeeCreateType["Platform"] = "PLATFORM";
    TransfersFeeCreateType["Inactivity"] = "INACTIVITY";
})(TransfersFeeCreateType || (exports.TransfersFeeCreateType = TransfersFeeCreateType = {}));
/** @internal */
exports.TransfersFeeCreateType$inboundSchema = z
    .union([
    z.nativeEnum(TransfersFeeCreateType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TransfersFeeCreateType$outboundSchema = z.union([
    z.nativeEnum(TransfersFeeCreateType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TransfersFeeCreateType$;
(function (TransfersFeeCreateType$) {
    /** @deprecated use `TransfersFeeCreateType$inboundSchema` instead. */
    TransfersFeeCreateType$.inboundSchema = exports.TransfersFeeCreateType$inboundSchema;
    /** @deprecated use `TransfersFeeCreateType$outboundSchema` instead. */
    TransfersFeeCreateType$.outboundSchema = exports.TransfersFeeCreateType$outboundSchema;
})(TransfersFeeCreateType$ || (exports.TransfersFeeCreateType$ = TransfersFeeCreateType$ = {}));
/** @internal */
exports.TransfersFeeCreate$inboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$inboundSchema,
    client_transfer_id: z.string(),
    description: z.string().optional(),
    fee_operating_account: z.string().optional(),
    type: exports.TransfersFeeCreateType$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_transfer_id": "clientTransferId",
        "fee_operating_account": "feeOperatingAccount",
    });
});
/** @internal */
exports.TransfersFeeCreate$outboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$outboundSchema,
    clientTransferId: z.string(),
    description: z.string().optional(),
    feeOperatingAccount: z.string().optional(),
    type: exports.TransfersFeeCreateType$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientTransferId: "client_transfer_id",
        feeOperatingAccount: "fee_operating_account",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TransfersFeeCreate$;
(function (TransfersFeeCreate$) {
    /** @deprecated use `TransfersFeeCreate$inboundSchema` instead. */
    TransfersFeeCreate$.inboundSchema = exports.TransfersFeeCreate$inboundSchema;
    /** @deprecated use `TransfersFeeCreate$outboundSchema` instead. */
    TransfersFeeCreate$.outboundSchema = exports.TransfersFeeCreate$outboundSchema;
})(TransfersFeeCreate$ || (exports.TransfersFeeCreate$ = TransfersFeeCreate$ = {}));
