// @apexfintechsolutions/ascend-sdk/models/components/transfercreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferCreate$ = exports.TransferCreate$outboundSchema = exports.TransferCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const assetcreate_js_1 = require("./assetcreate.js");
const transferaccountcreate_js_1 = require("./transferaccountcreate.js");
/** @internal */
exports.TransferCreate$inboundSchema = z.object({
    assets: z.array(assetcreate_js_1.AssetCreate$inboundSchema).optional(),
    comment: z.string().optional(),
    deliverer: transferaccountcreate_js_1.TransferAccountCreate$inboundSchema,
    original_control_number: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "original_control_number": "originalControlNumber",
    });
});
/** @internal */
exports.TransferCreate$outboundSchema = z.object({
    assets: z.array(assetcreate_js_1.AssetCreate$outboundSchema).optional(),
    comment: z.string().optional(),
    deliverer: transferaccountcreate_js_1.TransferAccountCreate$outboundSchema,
    originalControlNumber: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        originalControlNumber: "original_control_number",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TransferCreate$;
(function (TransferCreate$) {
    /** @deprecated use `TransferCreate$inboundSchema` instead. */
    TransferCreate$.inboundSchema = exports.TransferCreate$inboundSchema;
    /** @deprecated use `TransferCreate$outboundSchema` instead. */
    TransferCreate$.outboundSchema = exports.TransferCreate$outboundSchema;
})(TransferCreate$ || (exports.TransferCreate$ = TransferCreate$ = {}));
