// @apexfintechsolutions/ascend-sdk/models/components/trustenrollmentmetadatacreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrustEnrollmentMetadataCreate$ = exports.TrustEnrollmentMetadataCreate$outboundSchema = exports.TrustEnrollmentMetadataCreate$inboundSchema = exports.OpenedOnBehalfOf$ = exports.OpenedOnBehalfOf$outboundSchema = exports.OpenedOnBehalfOf$inboundSchema = exports.TrustEnrollmentMetadataCreateFdicCashSweep$ = exports.TrustEnrollmentMetadataCreateFdicCashSweep$outboundSchema = exports.TrustEnrollmentMetadataCreateFdicCashSweep$inboundSchema = exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan$ = exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = exports.OpenedOnBehalfOf = exports.TrustEnrollmentMetadataCreateFdicCashSweep = exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Option to auto-enroll in Dividend Reinvestment; defaults to DIVIDEND_REINVESTMENT_ENROLL
 */
var TrustEnrollmentMetadataCreateDividendReinvestmentPlan;
(function (TrustEnrollmentMetadataCreateDividendReinvestmentPlan) {
    TrustEnrollmentMetadataCreateDividendReinvestmentPlan["AutoEnrollDividendReinvestmentUnspecified"] = "AUTO_ENROLL_DIVIDEND_REINVESTMENT_UNSPECIFIED";
    TrustEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentEnroll"] = "DIVIDEND_REINVESTMENT_ENROLL";
    TrustEnrollmentMetadataCreateDividendReinvestmentPlan["DividendReinvestmentDecline"] = "DIVIDEND_REINVESTMENT_DECLINE";
})(TrustEnrollmentMetadataCreateDividendReinvestmentPlan || (exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan = TrustEnrollmentMetadataCreateDividendReinvestmentPlan = {}));
/**
 * Option to auto-enroll in FDIC cash sweep; defaults to FDIC_CASH_SWEEP_ENROLL
 */
var TrustEnrollmentMetadataCreateFdicCashSweep;
(function (TrustEnrollmentMetadataCreateFdicCashSweep) {
    TrustEnrollmentMetadataCreateFdicCashSweep["AutoEnrollFdicCashSweepUnspecified"] = "AUTO_ENROLL_FDIC_CASH_SWEEP_UNSPECIFIED";
    TrustEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepEnroll"] = "FDIC_CASH_SWEEP_ENROLL";
    TrustEnrollmentMetadataCreateFdicCashSweep["FdicCashSweepDecline"] = "FDIC_CASH_SWEEP_DECLINE";
})(TrustEnrollmentMetadataCreateFdicCashSweep || (exports.TrustEnrollmentMetadataCreateFdicCashSweep = TrustEnrollmentMetadataCreateFdicCashSweep = {}));
/**
 * Trust account is opened on behalf of
 */
var OpenedOnBehalfOf;
(function (OpenedOnBehalfOf) {
    OpenedOnBehalfOf["OpenedOnBehalfOfUnspecified"] = "OPENED_ON_BEHALF_OF_UNSPECIFIED";
    OpenedOnBehalfOf["PersonalTrust"] = "PERSONAL_TRUST";
    OpenedOnBehalfOf["BusinessTrust"] = "BUSINESS_TRUST";
    OpenedOnBehalfOf["ThirdPartyAdministrator"] = "THIRD_PARTY_ADMINISTRATOR";
})(OpenedOnBehalfOf || (exports.OpenedOnBehalfOf = OpenedOnBehalfOf = {}));
/** @internal */
exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema = z
    .union([
    z.nativeEnum(TrustEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema = z.union([
    z.nativeEnum(TrustEnrollmentMetadataCreateDividendReinvestmentPlan),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TrustEnrollmentMetadataCreateDividendReinvestmentPlan$;
(function (TrustEnrollmentMetadataCreateDividendReinvestmentPlan$) {
    /** @deprecated use `TrustEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema` instead. */
    TrustEnrollmentMetadataCreateDividendReinvestmentPlan$.inboundSchema = exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema;
    /** @deprecated use `TrustEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema` instead. */
    TrustEnrollmentMetadataCreateDividendReinvestmentPlan$.outboundSchema = exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema;
})(TrustEnrollmentMetadataCreateDividendReinvestmentPlan$ || (exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan$ = TrustEnrollmentMetadataCreateDividendReinvestmentPlan$ = {}));
/** @internal */
exports.TrustEnrollmentMetadataCreateFdicCashSweep$inboundSchema = z
    .union([
    z.nativeEnum(TrustEnrollmentMetadataCreateFdicCashSweep),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.TrustEnrollmentMetadataCreateFdicCashSweep$outboundSchema = z.union([
    z.nativeEnum(TrustEnrollmentMetadataCreateFdicCashSweep),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TrustEnrollmentMetadataCreateFdicCashSweep$;
(function (TrustEnrollmentMetadataCreateFdicCashSweep$) {
    /** @deprecated use `TrustEnrollmentMetadataCreateFdicCashSweep$inboundSchema` instead. */
    TrustEnrollmentMetadataCreateFdicCashSweep$.inboundSchema = exports.TrustEnrollmentMetadataCreateFdicCashSweep$inboundSchema;
    /** @deprecated use `TrustEnrollmentMetadataCreateFdicCashSweep$outboundSchema` instead. */
    TrustEnrollmentMetadataCreateFdicCashSweep$.outboundSchema = exports.TrustEnrollmentMetadataCreateFdicCashSweep$outboundSchema;
})(TrustEnrollmentMetadataCreateFdicCashSweep$ || (exports.TrustEnrollmentMetadataCreateFdicCashSweep$ = TrustEnrollmentMetadataCreateFdicCashSweep$ = {}));
/** @internal */
exports.OpenedOnBehalfOf$inboundSchema = z
    .union([
    z.nativeEnum(OpenedOnBehalfOf),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.OpenedOnBehalfOf$outboundSchema = z.union([
    z.nativeEnum(OpenedOnBehalfOf),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OpenedOnBehalfOf$;
(function (OpenedOnBehalfOf$) {
    /** @deprecated use `OpenedOnBehalfOf$inboundSchema` instead. */
    OpenedOnBehalfOf$.inboundSchema = exports.OpenedOnBehalfOf$inboundSchema;
    /** @deprecated use `OpenedOnBehalfOf$outboundSchema` instead. */
    OpenedOnBehalfOf$.outboundSchema = exports.OpenedOnBehalfOf$outboundSchema;
})(OpenedOnBehalfOf$ || (exports.OpenedOnBehalfOf$ = OpenedOnBehalfOf$ = {}));
/** @internal */
exports.TrustEnrollmentMetadataCreate$inboundSchema = z.object({
    dividend_reinvestment_plan: exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan$inboundSchema
        .optional(),
    fdic_cash_sweep: exports.TrustEnrollmentMetadataCreateFdicCashSweep$inboundSchema
        .optional(),
    opened_on_behalf_of: exports.OpenedOnBehalfOf$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "dividend_reinvestment_plan": "dividendReinvestmentPlan",
        "fdic_cash_sweep": "fdicCashSweep",
        "opened_on_behalf_of": "openedOnBehalfOf",
    });
});
/** @internal */
exports.TrustEnrollmentMetadataCreate$outboundSchema = z.object({
    dividendReinvestmentPlan: exports.TrustEnrollmentMetadataCreateDividendReinvestmentPlan$outboundSchema
        .optional(),
    fdicCashSweep: exports.TrustEnrollmentMetadataCreateFdicCashSweep$outboundSchema
        .optional(),
    openedOnBehalfOf: exports.OpenedOnBehalfOf$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        dividendReinvestmentPlan: "dividend_reinvestment_plan",
        fdicCashSweep: "fdic_cash_sweep",
        openedOnBehalfOf: "opened_on_behalf_of",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TrustEnrollmentMetadataCreate$;
(function (TrustEnrollmentMetadataCreate$) {
    /** @deprecated use `TrustEnrollmentMetadataCreate$inboundSchema` instead. */
    TrustEnrollmentMetadataCreate$.inboundSchema = exports.TrustEnrollmentMetadataCreate$inboundSchema;
    /** @deprecated use `TrustEnrollmentMetadataCreate$outboundSchema` instead. */
    TrustEnrollmentMetadataCreate$.outboundSchema = exports.TrustEnrollmentMetadataCreate$outboundSchema;
})(TrustEnrollmentMetadataCreate$ || (exports.TrustEnrollmentMetadataCreate$ = TrustEnrollmentMetadataCreate$ = {}));
