// @apexfintechsolutions/ascend-sdk/models/components/trustedcontactcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrustedContactCreate$ = exports.TrustedContactCreate$outboundSchema = exports.TrustedContactCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const phonenumbercreate_js_1 = require("./phonenumbercreate.js");
const postaladdresscreate_js_1 = require("./postaladdresscreate.js");
/** @internal */
exports.TrustedContactCreate$inboundSchema = z.object({
    email_address: z.string().optional(),
    family_name: z.string(),
    given_name: z.string(),
    mailing_address: postaladdresscreate_js_1.PostalAddressCreate$inboundSchema.optional(),
    middle_names: z.string().optional(),
    phone_number: phonenumbercreate_js_1.PhoneNumberCreate$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "email_address": "emailAddress",
        "family_name": "familyName",
        "given_name": "givenName",
        "mailing_address": "mailingAddress",
        "middle_names": "middleNames",
        "phone_number": "phoneNumber",
    });
});
/** @internal */
exports.TrustedContactCreate$outboundSchema = z.object({
    emailAddress: z.string().optional(),
    familyName: z.string(),
    givenName: z.string(),
    mailingAddress: postaladdresscreate_js_1.PostalAddressCreate$outboundSchema.optional(),
    middleNames: z.string().optional(),
    phoneNumber: phonenumbercreate_js_1.PhoneNumberCreate$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        emailAddress: "email_address",
        familyName: "family_name",
        givenName: "given_name",
        mailingAddress: "mailing_address",
        middleNames: "middle_names",
        phoneNumber: "phone_number",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TrustedContactCreate$;
(function (TrustedContactCreate$) {
    /** @deprecated use `TrustedContactCreate$inboundSchema` instead. */
    TrustedContactCreate$.inboundSchema = exports.TrustedContactCreate$inboundSchema;
    /** @deprecated use `TrustedContactCreate$outboundSchema` instead. */
    TrustedContactCreate$.outboundSchema = exports.TrustedContactCreate$outboundSchema;
})(TrustedContactCreate$ || (exports.TrustedContactCreate$ = TrustedContactCreate$ = {}));
