// @apexfintechsolutions/ascend-sdk/models/components/watchlistmatchupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WatchlistMatchUpdate$ = exports.WatchlistMatchUpdate$outboundSchema = exports.WatchlistMatchUpdate$inboundSchema = exports.WatchlistMatchUpdateMatchState$ = exports.WatchlistMatchUpdateMatchState$outboundSchema = exports.WatchlistMatchUpdateMatchState$inboundSchema = exports.WatchlistMatchUpdateMatchState = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * Match state - whether or not the match is confirmed
 */
var WatchlistMatchUpdateMatchState;
(function (WatchlistMatchUpdateMatchState) {
    WatchlistMatchUpdateMatchState["MatchUnspecified"] = "MATCH_UNSPECIFIED";
    WatchlistMatchUpdateMatchState["ConfirmedMatch"] = "CONFIRMED_MATCH";
    WatchlistMatchUpdateMatchState["PotentialMatch"] = "POTENTIAL_MATCH";
    WatchlistMatchUpdateMatchState["NoMatch"] = "NO_MATCH";
    WatchlistMatchUpdateMatchState["Inconclusive"] = "INCONCLUSIVE";
})(WatchlistMatchUpdateMatchState || (exports.WatchlistMatchUpdateMatchState = WatchlistMatchUpdateMatchState = {}));
/** @internal */
exports.WatchlistMatchUpdateMatchState$inboundSchema = z
    .union([
    z.nativeEnum(WatchlistMatchUpdateMatchState),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.WatchlistMatchUpdateMatchState$outboundSchema = z.union([
    z.nativeEnum(WatchlistMatchUpdateMatchState),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WatchlistMatchUpdateMatchState$;
(function (WatchlistMatchUpdateMatchState$) {
    /** @deprecated use `WatchlistMatchUpdateMatchState$inboundSchema` instead. */
    WatchlistMatchUpdateMatchState$.inboundSchema = exports.WatchlistMatchUpdateMatchState$inboundSchema;
    /** @deprecated use `WatchlistMatchUpdateMatchState$outboundSchema` instead. */
    WatchlistMatchUpdateMatchState$.outboundSchema = exports.WatchlistMatchUpdateMatchState$outboundSchema;
})(WatchlistMatchUpdateMatchState$ || (exports.WatchlistMatchUpdateMatchState$ = WatchlistMatchUpdateMatchState$ = {}));
/** @internal */
exports.WatchlistMatchUpdate$inboundSchema = z.object({
    exclude_from_screening: z.boolean().optional(),
    match_state: exports.WatchlistMatchUpdateMatchState$inboundSchema.optional(),
    watchlist_id: z.string().optional(),
    watchlist_item_id: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "exclude_from_screening": "excludeFromScreening",
        "match_state": "matchState",
        "watchlist_id": "watchlistId",
        "watchlist_item_id": "watchlistItemId",
    });
});
/** @internal */
exports.WatchlistMatchUpdate$outboundSchema = z.object({
    excludeFromScreening: z.boolean().optional(),
    matchState: exports.WatchlistMatchUpdateMatchState$outboundSchema.optional(),
    watchlistId: z.string().optional(),
    watchlistItemId: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        excludeFromScreening: "exclude_from_screening",
        matchState: "match_state",
        watchlistId: "watchlist_id",
        watchlistItemId: "watchlist_item_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WatchlistMatchUpdate$;
(function (WatchlistMatchUpdate$) {
    /** @deprecated use `WatchlistMatchUpdate$inboundSchema` instead. */
    WatchlistMatchUpdate$.inboundSchema = exports.WatchlistMatchUpdate$inboundSchema;
    /** @deprecated use `WatchlistMatchUpdate$outboundSchema` instead. */
    WatchlistMatchUpdate$.outboundSchema = exports.WatchlistMatchUpdate$outboundSchema;
})(WatchlistMatchUpdate$ || (exports.WatchlistMatchUpdate$ = WatchlistMatchUpdate$ = {}));
