// @apexfintechsolutions/ascend-sdk/models/components/watchlistmatch.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WatchlistMatch$ = exports.WatchlistMatch$outboundSchema = exports.WatchlistMatch$inboundSchema = exports.MatchState$ = exports.MatchState$outboundSchema = exports.MatchState$inboundSchema = exports.MatchAttributes$ = exports.MatchAttributes$outboundSchema = exports.MatchAttributes$inboundSchema = exports.MatchState = exports.MatchAttributes = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
var MatchAttributes;
(function (MatchAttributes) {
    MatchAttributes["MatchAttributeUnspecified"] = "MATCH_ATTRIBUTE_UNSPECIFIED";
    MatchAttributes["Name"] = "NAME";
    MatchAttributes["PhoneNumber"] = "PHONE_NUMBER";
    MatchAttributes["BirthDate"] = "BIRTH_DATE";
    MatchAttributes["Identification"] = "IDENTIFICATION";
    MatchAttributes["Email"] = "EMAIL";
    MatchAttributes["Address"] = "ADDRESS";
})(MatchAttributes || (exports.MatchAttributes = MatchAttributes = {}));
/**
 * Match state - whether or not the match is confirmed
 */
var MatchState;
(function (MatchState) {
    MatchState["MatchUnspecified"] = "MATCH_UNSPECIFIED";
    MatchState["ConfirmedMatch"] = "CONFIRMED_MATCH";
    MatchState["PotentialMatch"] = "POTENTIAL_MATCH";
    MatchState["NoMatch"] = "NO_MATCH";
    MatchState["Inconclusive"] = "INCONCLUSIVE";
})(MatchState || (exports.MatchState = MatchState = {}));
/** @internal */
exports.MatchAttributes$inboundSchema = z
    .union([
    z.nativeEnum(MatchAttributes),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.MatchAttributes$outboundSchema = z.union([
    z.nativeEnum(MatchAttributes),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MatchAttributes$;
(function (MatchAttributes$) {
    /** @deprecated use `MatchAttributes$inboundSchema` instead. */
    MatchAttributes$.inboundSchema = exports.MatchAttributes$inboundSchema;
    /** @deprecated use `MatchAttributes$outboundSchema` instead. */
    MatchAttributes$.outboundSchema = exports.MatchAttributes$outboundSchema;
})(MatchAttributes$ || (exports.MatchAttributes$ = MatchAttributes$ = {}));
/** @internal */
exports.MatchState$inboundSchema = z
    .union([
    z.nativeEnum(MatchState),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.MatchState$outboundSchema = z.union([
    z.nativeEnum(MatchState),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var MatchState$;
(function (MatchState$) {
    /** @deprecated use `MatchState$inboundSchema` instead. */
    MatchState$.inboundSchema = exports.MatchState$inboundSchema;
    /** @deprecated use `MatchState$outboundSchema` instead. */
    MatchState$.outboundSchema = exports.MatchState$outboundSchema;
})(MatchState$ || (exports.MatchState$ = MatchState$ = {}));
/** @internal */
exports.WatchlistMatch$inboundSchema = z.object({
    active: z.boolean().optional(),
    created_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    exclude_from_screening: z.boolean().optional(),
    match_attributes: z.array(exports.MatchAttributes$inboundSchema).optional(),
    match_state: exports.MatchState$inboundSchema.optional(),
    updated_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    watchlist_id: z.string().optional(),
    watchlist_item_id: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "exclude_from_screening": "excludeFromScreening",
        "match_attributes": "matchAttributes",
        "match_state": "matchState",
        "updated_at": "updatedAt",
        "watchlist_id": "watchlistId",
        "watchlist_item_id": "watchlistItemId",
    });
});
/** @internal */
exports.WatchlistMatch$outboundSchema = z.object({
    active: z.boolean().optional(),
    createdAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    excludeFromScreening: z.boolean().optional(),
    matchAttributes: z.array(exports.MatchAttributes$outboundSchema).optional(),
    matchState: exports.MatchState$outboundSchema.optional(),
    updatedAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    watchlistId: z.string().optional(),
    watchlistItemId: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        excludeFromScreening: "exclude_from_screening",
        matchAttributes: "match_attributes",
        matchState: "match_state",
        updatedAt: "updated_at",
        watchlistId: "watchlist_id",
        watchlistItemId: "watchlist_item_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WatchlistMatch$;
(function (WatchlistMatch$) {
    /** @deprecated use `WatchlistMatch$inboundSchema` instead. */
    WatchlistMatch$.inboundSchema = exports.WatchlistMatch$inboundSchema;
    /** @deprecated use `WatchlistMatch$outboundSchema` instead. */
    WatchlistMatch$.outboundSchema = exports.WatchlistMatch$outboundSchema;
})(WatchlistMatch$ || (exports.WatchlistMatch$ = WatchlistMatch$ = {}));
