// @apexfintechsolutions/ascend-sdk/models/components/wirewithdrawalschedule.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WireWithdrawalScheduleType$inboundSchema = exports.WireWithdrawalScheduleStateTaxWithholding$ = exports.WireWithdrawalScheduleStateTaxWithholding$outboundSchema = exports.WireWithdrawalScheduleStateTaxWithholding$inboundSchema = exports.WireWithdrawalScheduleRetirementDistributionPercentage$ = exports.WireWithdrawalScheduleRetirementDistributionPercentage$outboundSchema = exports.WireWithdrawalScheduleRetirementDistributionPercentage$inboundSchema = exports.WireWithdrawalScheduleRetirementDistributionAmount$ = exports.WireWithdrawalScheduleRetirementDistributionAmount$outboundSchema = exports.WireWithdrawalScheduleRetirementDistributionAmount$inboundSchema = exports.WireWithdrawalScheduleFederalTaxWithholding$ = exports.WireWithdrawalScheduleFederalTaxWithholding$outboundSchema = exports.WireWithdrawalScheduleFederalTaxWithholding$inboundSchema = exports.WireWithdrawalSchedulePercentage$ = exports.WireWithdrawalSchedulePercentage$outboundSchema = exports.WireWithdrawalSchedulePercentage$inboundSchema = exports.WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$ = exports.WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$outboundSchema = exports.WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$inboundSchema = exports.RecipientBank$ = exports.RecipientBank$outboundSchema = exports.RecipientBank$inboundSchema = exports.InternationalBankDetails$ = exports.InternationalBankDetails$outboundSchema = exports.InternationalBankDetails$inboundSchema = exports.WireWithdrawalScheduleRecipientBankAddress$ = exports.WireWithdrawalScheduleRecipientBankAddress$outboundSchema = exports.WireWithdrawalScheduleRecipientBankAddress$inboundSchema = exports.BankId$ = exports.BankId$outboundSchema = exports.BankId$inboundSchema = exports.WireWithdrawalScheduleRecipientBankType$ = exports.WireWithdrawalScheduleRecipientBankType$outboundSchema = exports.WireWithdrawalScheduleRecipientBankType$inboundSchema = exports.Intermediary$ = exports.Intermediary$outboundSchema = exports.Intermediary$inboundSchema = exports.WireWithdrawalScheduleAddress$ = exports.WireWithdrawalScheduleAddress$outboundSchema = exports.WireWithdrawalScheduleAddress$inboundSchema = exports.WireWithdrawalScheduleBeneficiary$ = exports.WireWithdrawalScheduleBeneficiary$outboundSchema = exports.WireWithdrawalScheduleBeneficiary$inboundSchema = exports.Address$ = exports.Address$outboundSchema = exports.Address$inboundSchema = exports.WireWithdrawalScheduleTimeUnit = exports.WireWithdrawalScheduleState = exports.WireWithdrawalScheduleType = exports.WireWithdrawalScheduleRecipientBankType = void 0;
exports.WireWithdrawalSchedule$ = exports.WireWithdrawalSchedule$outboundSchema = exports.WireWithdrawalSchedule$inboundSchema = exports.WireWithdrawalScheduleScheduleDetails$ = exports.WireWithdrawalScheduleScheduleDetails$outboundSchema = exports.WireWithdrawalScheduleScheduleDetails$inboundSchema = exports.WireWithdrawalScheduleScheduleProperties$ = exports.WireWithdrawalScheduleScheduleProperties$outboundSchema = exports.WireWithdrawalScheduleScheduleProperties$inboundSchema = exports.WireWithdrawalScheduleTimeUnit$ = exports.WireWithdrawalScheduleTimeUnit$outboundSchema = exports.WireWithdrawalScheduleTimeUnit$inboundSchema = exports.WireWithdrawalScheduleState$ = exports.WireWithdrawalScheduleState$outboundSchema = exports.WireWithdrawalScheduleState$inboundSchema = exports.WireWithdrawalScheduleStartDate$ = exports.WireWithdrawalScheduleStartDate$outboundSchema = exports.WireWithdrawalScheduleStartDate$inboundSchema = exports.WireWithdrawalScheduleAmount$ = exports.WireWithdrawalScheduleAmount$outboundSchema = exports.WireWithdrawalScheduleAmount$inboundSchema = exports.WireWithdrawalScheduleRetirementDistribution$ = exports.WireWithdrawalScheduleRetirementDistribution$outboundSchema = exports.WireWithdrawalScheduleRetirementDistribution$inboundSchema = exports.WireWithdrawalScheduleType$ = exports.WireWithdrawalScheduleType$outboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const enums_js_1 = require("../../types/enums.js");
/**
 * The type of bank identifier specified
 */
var WireWithdrawalScheduleRecipientBankType;
(function (WireWithdrawalScheduleRecipientBankType) {
    WireWithdrawalScheduleRecipientBankType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    WireWithdrawalScheduleRecipientBankType["Aba"] = "ABA";
    WireWithdrawalScheduleRecipientBankType["Bic"] = "BIC";
})(WireWithdrawalScheduleRecipientBankType || (exports.WireWithdrawalScheduleRecipientBankType = WireWithdrawalScheduleRecipientBankType = {}));
/**
 * The type of retirement distribution.
 */
var WireWithdrawalScheduleType;
(function (WireWithdrawalScheduleType) {
    WireWithdrawalScheduleType["TypeUnspecified"] = "TYPE_UNSPECIFIED";
    WireWithdrawalScheduleType["Normal"] = "NORMAL";
    WireWithdrawalScheduleType["Disability"] = "DISABILITY";
    WireWithdrawalScheduleType["Sosepp"] = "SOSEPP";
    WireWithdrawalScheduleType["Premature"] = "PREMATURE";
    WireWithdrawalScheduleType["Death"] = "DEATH";
    WireWithdrawalScheduleType["ExcessContributionRemovalBeforeTaxDeadline"] = "EXCESS_CONTRIBUTION_REMOVAL_BEFORE_TAX_DEADLINE";
    WireWithdrawalScheduleType["ExcessContributionRemovalAfterTaxDeadline"] = "EXCESS_CONTRIBUTION_REMOVAL_AFTER_TAX_DEADLINE";
    WireWithdrawalScheduleType["RolloverToQualifiedPlan"] = "ROLLOVER_TO_QUALIFIED_PLAN";
    WireWithdrawalScheduleType["RolloverToIra"] = "ROLLOVER_TO_IRA";
    WireWithdrawalScheduleType["DistributionTransfer"] = "DISTRIBUTION_TRANSFER";
    WireWithdrawalScheduleType["RecharacterizationPriorYear"] = "RECHARACTERIZATION_PRIOR_YEAR";
    WireWithdrawalScheduleType["RecharacterizationCurrentYear"] = "RECHARACTERIZATION_CURRENT_YEAR";
    WireWithdrawalScheduleType["DistributionConversion"] = "DISTRIBUTION_CONVERSION";
    WireWithdrawalScheduleType["ManagementFee"] = "MANAGEMENT_FEE";
    WireWithdrawalScheduleType["PlanLoan401K"] = "PLAN_LOAN_401K";
    WireWithdrawalScheduleType["PrematureSimpleIraLessThan2Years"] = "PREMATURE_SIMPLE_IRA_LESS_THAN_2_YEARS";
    WireWithdrawalScheduleType["NormalRothIraGreaterThan5Years"] = "NORMAL_ROTH_IRA_GREATER_THAN_5_YEARS";
    WireWithdrawalScheduleType["NetIncomeAttributable"] = "NET_INCOME_ATTRIBUTABLE";
    WireWithdrawalScheduleType["Revocation"] = "REVOCATION";
    WireWithdrawalScheduleType["NonReportable"] = "NON_REPORTABLE";
})(WireWithdrawalScheduleType || (exports.WireWithdrawalScheduleType = WireWithdrawalScheduleType = {}));
/**
 * The state of the represented schedule
 */
var WireWithdrawalScheduleState;
(function (WireWithdrawalScheduleState) {
    WireWithdrawalScheduleState["Active"] = "ACTIVE";
    WireWithdrawalScheduleState["Canceled"] = "CANCELED";
    WireWithdrawalScheduleState["Completed"] = "COMPLETED";
})(WireWithdrawalScheduleState || (exports.WireWithdrawalScheduleState = WireWithdrawalScheduleState = {}));
/**
 * The time unit used to calculate the interval between transfers. The time period between transfers in a scheduled series is the unit of time times the multiplier
 */
var WireWithdrawalScheduleTimeUnit;
(function (WireWithdrawalScheduleTimeUnit) {
    WireWithdrawalScheduleTimeUnit["Day"] = "DAY";
    WireWithdrawalScheduleTimeUnit["Week"] = "WEEK";
    WireWithdrawalScheduleTimeUnit["Month"] = "MONTH";
})(WireWithdrawalScheduleTimeUnit || (exports.WireWithdrawalScheduleTimeUnit = WireWithdrawalScheduleTimeUnit = {}));
/** @internal */
exports.Address$inboundSchema = z.object({
    city: z.string().optional(),
    country: z.string().optional(),
    postal_code: z.string().optional(),
    state: z.string().optional(),
    streetAddress: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "postal_code": "postalCode",
    });
});
/** @internal */
exports.Address$outboundSchema = z.object({
    city: z.string().optional(),
    country: z.string().optional(),
    postalCode: z.string().optional(),
    state: z.string().optional(),
    streetAddress: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        postalCode: "postal_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Address$;
(function (Address$) {
    /** @deprecated use `Address$inboundSchema` instead. */
    Address$.inboundSchema = exports.Address$inboundSchema;
    /** @deprecated use `Address$outboundSchema` instead. */
    Address$.outboundSchema = exports.Address$outboundSchema;
})(Address$ || (exports.Address$ = Address$ = {}));
/** @internal */
exports.WireWithdrawalScheduleBeneficiary$inboundSchema = z.object({
    account: z.string().optional(),
    account_title: z.string().optional(),
    address: z.nullable(z.lazy(() => exports.Address$inboundSchema)).optional(),
    third_party: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_title": "accountTitle",
        "third_party": "thirdParty",
    });
});
/** @internal */
exports.WireWithdrawalScheduleBeneficiary$outboundSchema = z.object({
    account: z.string().optional(),
    accountTitle: z.string().optional(),
    address: z.nullable(z.lazy(() => exports.Address$outboundSchema)).optional(),
    thirdParty: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountTitle: "account_title",
        thirdParty: "third_party",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleBeneficiary$;
(function (WireWithdrawalScheduleBeneficiary$) {
    /** @deprecated use `WireWithdrawalScheduleBeneficiary$inboundSchema` instead. */
    WireWithdrawalScheduleBeneficiary$.inboundSchema = exports.WireWithdrawalScheduleBeneficiary$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleBeneficiary$outboundSchema` instead. */
    WireWithdrawalScheduleBeneficiary$.outboundSchema = exports.WireWithdrawalScheduleBeneficiary$outboundSchema;
})(WireWithdrawalScheduleBeneficiary$ || (exports.WireWithdrawalScheduleBeneficiary$ = WireWithdrawalScheduleBeneficiary$ = {}));
/** @internal */
exports.WireWithdrawalScheduleAddress$inboundSchema = z.object({
    city: z.string().optional(),
    country: z.string().optional(),
    postal_code: z.string().optional(),
    state: z.string().optional(),
    streetAddress: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "postal_code": "postalCode",
    });
});
/** @internal */
exports.WireWithdrawalScheduleAddress$outboundSchema = z.object({
    city: z.string().optional(),
    country: z.string().optional(),
    postalCode: z.string().optional(),
    state: z.string().optional(),
    streetAddress: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        postalCode: "postal_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleAddress$;
(function (WireWithdrawalScheduleAddress$) {
    /** @deprecated use `WireWithdrawalScheduleAddress$inboundSchema` instead. */
    WireWithdrawalScheduleAddress$.inboundSchema = exports.WireWithdrawalScheduleAddress$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleAddress$outboundSchema` instead. */
    WireWithdrawalScheduleAddress$.outboundSchema = exports.WireWithdrawalScheduleAddress$outboundSchema;
})(WireWithdrawalScheduleAddress$ || (exports.WireWithdrawalScheduleAddress$ = WireWithdrawalScheduleAddress$ = {}));
/** @internal */
exports.Intermediary$inboundSchema = z.object({
    account: z.string().optional(),
    account_title: z.string().optional(),
    address: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleAddress$inboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_title": "accountTitle",
    });
});
/** @internal */
exports.Intermediary$outboundSchema = z.object({
    account: z.string().optional(),
    accountTitle: z.string().optional(),
    address: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleAddress$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountTitle: "account_title",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Intermediary$;
(function (Intermediary$) {
    /** @deprecated use `Intermediary$inboundSchema` instead. */
    Intermediary$.inboundSchema = exports.Intermediary$inboundSchema;
    /** @deprecated use `Intermediary$outboundSchema` instead. */
    Intermediary$.outboundSchema = exports.Intermediary$outboundSchema;
})(Intermediary$ || (exports.Intermediary$ = Intermediary$ = {}));
/** @internal */
exports.WireWithdrawalScheduleRecipientBankType$inboundSchema = z
    .union([
    z.nativeEnum(WireWithdrawalScheduleRecipientBankType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.WireWithdrawalScheduleRecipientBankType$outboundSchema = z.union([
    z.nativeEnum(WireWithdrawalScheduleRecipientBankType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleRecipientBankType$;
(function (WireWithdrawalScheduleRecipientBankType$) {
    /** @deprecated use `WireWithdrawalScheduleRecipientBankType$inboundSchema` instead. */
    WireWithdrawalScheduleRecipientBankType$.inboundSchema = exports.WireWithdrawalScheduleRecipientBankType$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleRecipientBankType$outboundSchema` instead. */
    WireWithdrawalScheduleRecipientBankType$.outboundSchema = exports.WireWithdrawalScheduleRecipientBankType$outboundSchema;
})(WireWithdrawalScheduleRecipientBankType$ || (exports.WireWithdrawalScheduleRecipientBankType$ = WireWithdrawalScheduleRecipientBankType$ = {}));
/** @internal */
exports.BankId$inboundSchema = z
    .object({
    id: z.string().optional(),
    type: exports.WireWithdrawalScheduleRecipientBankType$inboundSchema.optional(),
});
/** @internal */
exports.BankId$outboundSchema = z.object({
    id: z.string().optional(),
    type: exports.WireWithdrawalScheduleRecipientBankType$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BankId$;
(function (BankId$) {
    /** @deprecated use `BankId$inboundSchema` instead. */
    BankId$.inboundSchema = exports.BankId$inboundSchema;
    /** @deprecated use `BankId$outboundSchema` instead. */
    BankId$.outboundSchema = exports.BankId$outboundSchema;
})(BankId$ || (exports.BankId$ = BankId$ = {}));
/** @internal */
exports.WireWithdrawalScheduleRecipientBankAddress$inboundSchema = z.object({
    city: z.string().optional(),
    country: z.string().optional(),
    postal_code: z.string().optional(),
    state: z.string().optional(),
    streetAddress: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "postal_code": "postalCode",
    });
});
/** @internal */
exports.WireWithdrawalScheduleRecipientBankAddress$outboundSchema = z.object({
    city: z.string().optional(),
    country: z.string().optional(),
    postalCode: z.string().optional(),
    state: z.string().optional(),
    streetAddress: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        postalCode: "postal_code",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleRecipientBankAddress$;
(function (WireWithdrawalScheduleRecipientBankAddress$) {
    /** @deprecated use `WireWithdrawalScheduleRecipientBankAddress$inboundSchema` instead. */
    WireWithdrawalScheduleRecipientBankAddress$.inboundSchema = exports.WireWithdrawalScheduleRecipientBankAddress$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleRecipientBankAddress$outboundSchema` instead. */
    WireWithdrawalScheduleRecipientBankAddress$.outboundSchema = exports.WireWithdrawalScheduleRecipientBankAddress$outboundSchema;
})(WireWithdrawalScheduleRecipientBankAddress$ || (exports.WireWithdrawalScheduleRecipientBankAddress$ = WireWithdrawalScheduleRecipientBankAddress$ = {}));
/** @internal */
exports.InternationalBankDetails$inboundSchema = z.object({
    additional_info: z.string().optional(),
    address: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleRecipientBankAddress$inboundSchema)).optional(),
    bank_name: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "additional_info": "additionalInfo",
        "bank_name": "bankName",
    });
});
/** @internal */
exports.InternationalBankDetails$outboundSchema = z.object({
    additionalInfo: z.string().optional(),
    address: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleRecipientBankAddress$outboundSchema)).optional(),
    bankName: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        additionalInfo: "additional_info",
        bankName: "bank_name",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InternationalBankDetails$;
(function (InternationalBankDetails$) {
    /** @deprecated use `InternationalBankDetails$inboundSchema` instead. */
    InternationalBankDetails$.inboundSchema = exports.InternationalBankDetails$inboundSchema;
    /** @deprecated use `InternationalBankDetails$outboundSchema` instead. */
    InternationalBankDetails$.outboundSchema = exports.InternationalBankDetails$outboundSchema;
})(InternationalBankDetails$ || (exports.InternationalBankDetails$ = InternationalBankDetails$ = {}));
/** @internal */
exports.RecipientBank$inboundSchema = z.object({
    bank_id: z.nullable(z.lazy(() => exports.BankId$inboundSchema)).optional(),
    international_bank_details: z.nullable(z.lazy(() => exports.InternationalBankDetails$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "bank_id": "bankId",
        "international_bank_details": "internationalBankDetails",
    });
});
/** @internal */
exports.RecipientBank$outboundSchema = z.object({
    bankId: z.nullable(z.lazy(() => exports.BankId$outboundSchema)).optional(),
    internationalBankDetails: z.nullable(z.lazy(() => exports.InternationalBankDetails$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        bankId: "bank_id",
        internationalBankDetails: "international_bank_details",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RecipientBank$;
(function (RecipientBank$) {
    /** @deprecated use `RecipientBank$inboundSchema` instead. */
    RecipientBank$.inboundSchema = exports.RecipientBank$inboundSchema;
    /** @deprecated use `RecipientBank$outboundSchema` instead. */
    RecipientBank$.outboundSchema = exports.RecipientBank$outboundSchema;
})(RecipientBank$ || (exports.RecipientBank$ = RecipientBank$ = {}));
/** @internal */
exports.WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$;
(function (WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$) {
    /** @deprecated use `WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$inboundSchema` instead. */
    WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$.inboundSchema = exports.WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$outboundSchema` instead. */
    WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$.outboundSchema = exports.WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$outboundSchema;
})(WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$ || (exports.WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$ = WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$ = {}));
/** @internal */
exports.WireWithdrawalSchedulePercentage$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.WireWithdrawalSchedulePercentage$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalSchedulePercentage$;
(function (WireWithdrawalSchedulePercentage$) {
    /** @deprecated use `WireWithdrawalSchedulePercentage$inboundSchema` instead. */
    WireWithdrawalSchedulePercentage$.inboundSchema = exports.WireWithdrawalSchedulePercentage$inboundSchema;
    /** @deprecated use `WireWithdrawalSchedulePercentage$outboundSchema` instead. */
    WireWithdrawalSchedulePercentage$.outboundSchema = exports.WireWithdrawalSchedulePercentage$outboundSchema;
})(WireWithdrawalSchedulePercentage$ || (exports.WireWithdrawalSchedulePercentage$ = WireWithdrawalSchedulePercentage$ = {}));
/** @internal */
exports.WireWithdrawalScheduleFederalTaxWithholding$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$inboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.WireWithdrawalSchedulePercentage$inboundSchema)).optional(),
});
/** @internal */
exports.WireWithdrawalScheduleFederalTaxWithholding$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleRetirementDistributionFederalTaxWithholdingAmount$outboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.WireWithdrawalSchedulePercentage$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleFederalTaxWithholding$;
(function (WireWithdrawalScheduleFederalTaxWithholding$) {
    /** @deprecated use `WireWithdrawalScheduleFederalTaxWithholding$inboundSchema` instead. */
    WireWithdrawalScheduleFederalTaxWithholding$.inboundSchema = exports.WireWithdrawalScheduleFederalTaxWithholding$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleFederalTaxWithholding$outboundSchema` instead. */
    WireWithdrawalScheduleFederalTaxWithholding$.outboundSchema = exports.WireWithdrawalScheduleFederalTaxWithholding$outboundSchema;
})(WireWithdrawalScheduleFederalTaxWithholding$ || (exports.WireWithdrawalScheduleFederalTaxWithholding$ = WireWithdrawalScheduleFederalTaxWithholding$ = {}));
/** @internal */
exports.WireWithdrawalScheduleRetirementDistributionAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.WireWithdrawalScheduleRetirementDistributionAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleRetirementDistributionAmount$;
(function (WireWithdrawalScheduleRetirementDistributionAmount$) {
    /** @deprecated use `WireWithdrawalScheduleRetirementDistributionAmount$inboundSchema` instead. */
    WireWithdrawalScheduleRetirementDistributionAmount$.inboundSchema = exports.WireWithdrawalScheduleRetirementDistributionAmount$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleRetirementDistributionAmount$outboundSchema` instead. */
    WireWithdrawalScheduleRetirementDistributionAmount$.outboundSchema = exports.WireWithdrawalScheduleRetirementDistributionAmount$outboundSchema;
})(WireWithdrawalScheduleRetirementDistributionAmount$ || (exports.WireWithdrawalScheduleRetirementDistributionAmount$ = WireWithdrawalScheduleRetirementDistributionAmount$ = {}));
/** @internal */
exports.WireWithdrawalScheduleRetirementDistributionPercentage$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.WireWithdrawalScheduleRetirementDistributionPercentage$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleRetirementDistributionPercentage$;
(function (WireWithdrawalScheduleRetirementDistributionPercentage$) {
    /** @deprecated use `WireWithdrawalScheduleRetirementDistributionPercentage$inboundSchema` instead. */
    WireWithdrawalScheduleRetirementDistributionPercentage$.inboundSchema = exports.WireWithdrawalScheduleRetirementDistributionPercentage$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleRetirementDistributionPercentage$outboundSchema` instead. */
    WireWithdrawalScheduleRetirementDistributionPercentage$.outboundSchema = exports.WireWithdrawalScheduleRetirementDistributionPercentage$outboundSchema;
})(WireWithdrawalScheduleRetirementDistributionPercentage$ || (exports.WireWithdrawalScheduleRetirementDistributionPercentage$ = WireWithdrawalScheduleRetirementDistributionPercentage$ = {}));
/** @internal */
exports.WireWithdrawalScheduleStateTaxWithholding$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleRetirementDistributionAmount$inboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleRetirementDistributionPercentage$inboundSchema)).optional(),
});
/** @internal */
exports.WireWithdrawalScheduleStateTaxWithholding$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleRetirementDistributionAmount$outboundSchema)).optional(),
    percentage: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleRetirementDistributionPercentage$outboundSchema)).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleStateTaxWithholding$;
(function (WireWithdrawalScheduleStateTaxWithholding$) {
    /** @deprecated use `WireWithdrawalScheduleStateTaxWithholding$inboundSchema` instead. */
    WireWithdrawalScheduleStateTaxWithholding$.inboundSchema = exports.WireWithdrawalScheduleStateTaxWithholding$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleStateTaxWithholding$outboundSchema` instead. */
    WireWithdrawalScheduleStateTaxWithholding$.outboundSchema = exports.WireWithdrawalScheduleStateTaxWithholding$outboundSchema;
})(WireWithdrawalScheduleStateTaxWithholding$ || (exports.WireWithdrawalScheduleStateTaxWithholding$ = WireWithdrawalScheduleStateTaxWithholding$ = {}));
/** @internal */
exports.WireWithdrawalScheduleType$inboundSchema = z
    .union([
    z.nativeEnum(WireWithdrawalScheduleType),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.WireWithdrawalScheduleType$outboundSchema = z.union([
    z.nativeEnum(WireWithdrawalScheduleType),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleType$;
(function (WireWithdrawalScheduleType$) {
    /** @deprecated use `WireWithdrawalScheduleType$inboundSchema` instead. */
    WireWithdrawalScheduleType$.inboundSchema = exports.WireWithdrawalScheduleType$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleType$outboundSchema` instead. */
    WireWithdrawalScheduleType$.outboundSchema = exports.WireWithdrawalScheduleType$outboundSchema;
})(WireWithdrawalScheduleType$ || (exports.WireWithdrawalScheduleType$ = WireWithdrawalScheduleType$ = {}));
/** @internal */
exports.WireWithdrawalScheduleRetirementDistribution$inboundSchema = z.object({
    federal_tax_withholding: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleFederalTaxWithholding$inboundSchema)).optional(),
    receiving_institution: z.string().optional(),
    state_tax_withholding: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleStateTaxWithholding$inboundSchema)).optional(),
    state_withholding_waiver: z.boolean().optional(),
    tax_year: z.number().int().optional(),
    type: exports.WireWithdrawalScheduleType$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "federal_tax_withholding": "federalTaxWithholding",
        "receiving_institution": "receivingInstitution",
        "state_tax_withholding": "stateTaxWithholding",
        "state_withholding_waiver": "stateWithholdingWaiver",
        "tax_year": "taxYear",
    });
});
/** @internal */
exports.WireWithdrawalScheduleRetirementDistribution$outboundSchema = z.object({
    federalTaxWithholding: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleFederalTaxWithholding$outboundSchema)).optional(),
    receivingInstitution: z.string().optional(),
    stateTaxWithholding: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleStateTaxWithholding$outboundSchema)).optional(),
    stateWithholdingWaiver: z.boolean().optional(),
    taxYear: z.number().int().optional(),
    type: exports.WireWithdrawalScheduleType$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        federalTaxWithholding: "federal_tax_withholding",
        receivingInstitution: "receiving_institution",
        stateTaxWithholding: "state_tax_withholding",
        stateWithholdingWaiver: "state_withholding_waiver",
        taxYear: "tax_year",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleRetirementDistribution$;
(function (WireWithdrawalScheduleRetirementDistribution$) {
    /** @deprecated use `WireWithdrawalScheduleRetirementDistribution$inboundSchema` instead. */
    WireWithdrawalScheduleRetirementDistribution$.inboundSchema = exports.WireWithdrawalScheduleRetirementDistribution$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleRetirementDistribution$outboundSchema` instead. */
    WireWithdrawalScheduleRetirementDistribution$.outboundSchema = exports.WireWithdrawalScheduleRetirementDistribution$outboundSchema;
})(WireWithdrawalScheduleRetirementDistribution$ || (exports.WireWithdrawalScheduleRetirementDistribution$ = WireWithdrawalScheduleRetirementDistribution$ = {}));
/** @internal */
exports.WireWithdrawalScheduleAmount$inboundSchema = z.object({
    value: z.string().optional(),
});
/** @internal */
exports.WireWithdrawalScheduleAmount$outboundSchema = z.object({
    value: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleAmount$;
(function (WireWithdrawalScheduleAmount$) {
    /** @deprecated use `WireWithdrawalScheduleAmount$inboundSchema` instead. */
    WireWithdrawalScheduleAmount$.inboundSchema = exports.WireWithdrawalScheduleAmount$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleAmount$outboundSchema` instead. */
    WireWithdrawalScheduleAmount$.outboundSchema = exports.WireWithdrawalScheduleAmount$outboundSchema;
})(WireWithdrawalScheduleAmount$ || (exports.WireWithdrawalScheduleAmount$ = WireWithdrawalScheduleAmount$ = {}));
/** @internal */
exports.WireWithdrawalScheduleStartDate$inboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/** @internal */
exports.WireWithdrawalScheduleStartDate$outboundSchema = z.object({
    day: z.number().int().optional(),
    month: z.number().int().optional(),
    year: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleStartDate$;
(function (WireWithdrawalScheduleStartDate$) {
    /** @deprecated use `WireWithdrawalScheduleStartDate$inboundSchema` instead. */
    WireWithdrawalScheduleStartDate$.inboundSchema = exports.WireWithdrawalScheduleStartDate$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleStartDate$outboundSchema` instead. */
    WireWithdrawalScheduleStartDate$.outboundSchema = exports.WireWithdrawalScheduleStartDate$outboundSchema;
})(WireWithdrawalScheduleStartDate$ || (exports.WireWithdrawalScheduleStartDate$ = WireWithdrawalScheduleStartDate$ = {}));
/** @internal */
exports.WireWithdrawalScheduleState$inboundSchema = z
    .union([
    z.nativeEnum(WireWithdrawalScheduleState),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.WireWithdrawalScheduleState$outboundSchema = z.union([
    z.nativeEnum(WireWithdrawalScheduleState),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleState$;
(function (WireWithdrawalScheduleState$) {
    /** @deprecated use `WireWithdrawalScheduleState$inboundSchema` instead. */
    WireWithdrawalScheduleState$.inboundSchema = exports.WireWithdrawalScheduleState$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleState$outboundSchema` instead. */
    WireWithdrawalScheduleState$.outboundSchema = exports.WireWithdrawalScheduleState$outboundSchema;
})(WireWithdrawalScheduleState$ || (exports.WireWithdrawalScheduleState$ = WireWithdrawalScheduleState$ = {}));
/** @internal */
exports.WireWithdrawalScheduleTimeUnit$inboundSchema = z
    .union([
    z.nativeEnum(WireWithdrawalScheduleTimeUnit),
    z.string().transform(enums_js_1.catchUnrecognizedEnum),
]);
/** @internal */
exports.WireWithdrawalScheduleTimeUnit$outboundSchema = z.union([
    z.nativeEnum(WireWithdrawalScheduleTimeUnit),
    z.string().and(z.custom()),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleTimeUnit$;
(function (WireWithdrawalScheduleTimeUnit$) {
    /** @deprecated use `WireWithdrawalScheduleTimeUnit$inboundSchema` instead. */
    WireWithdrawalScheduleTimeUnit$.inboundSchema = exports.WireWithdrawalScheduleTimeUnit$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleTimeUnit$outboundSchema` instead. */
    WireWithdrawalScheduleTimeUnit$.outboundSchema = exports.WireWithdrawalScheduleTimeUnit$outboundSchema;
})(WireWithdrawalScheduleTimeUnit$ || (exports.WireWithdrawalScheduleTimeUnit$ = WireWithdrawalScheduleTimeUnit$ = {}));
/** @internal */
exports.WireWithdrawalScheduleScheduleProperties$inboundSchema = z.object({
    occurrences: z.number().int().optional(),
    start_date: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleStartDate$inboundSchema)).optional(),
    state: exports.WireWithdrawalScheduleState$inboundSchema.optional(),
    time_unit: exports.WireWithdrawalScheduleTimeUnit$inboundSchema.optional(),
    unit_multiplier: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "start_date": "startDate",
        "time_unit": "timeUnit",
        "unit_multiplier": "unitMultiplier",
    });
});
/** @internal */
exports.WireWithdrawalScheduleScheduleProperties$outboundSchema = z.object({
    occurrences: z.number().int().optional(),
    startDate: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleStartDate$outboundSchema)).optional(),
    state: exports.WireWithdrawalScheduleState$outboundSchema.optional(),
    timeUnit: exports.WireWithdrawalScheduleTimeUnit$outboundSchema.optional(),
    unitMultiplier: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        startDate: "start_date",
        timeUnit: "time_unit",
        unitMultiplier: "unit_multiplier",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleScheduleProperties$;
(function (WireWithdrawalScheduleScheduleProperties$) {
    /** @deprecated use `WireWithdrawalScheduleScheduleProperties$inboundSchema` instead. */
    WireWithdrawalScheduleScheduleProperties$.inboundSchema = exports.WireWithdrawalScheduleScheduleProperties$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleScheduleProperties$outboundSchema` instead. */
    WireWithdrawalScheduleScheduleProperties$.outboundSchema = exports.WireWithdrawalScheduleScheduleProperties$outboundSchema;
})(WireWithdrawalScheduleScheduleProperties$ || (exports.WireWithdrawalScheduleScheduleProperties$ = WireWithdrawalScheduleScheduleProperties$ = {}));
/** @internal */
exports.WireWithdrawalScheduleScheduleDetails$inboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleAmount$inboundSchema))
        .optional(),
    client_schedule_id: z.string().optional(),
    full_disbursement: z.boolean().optional(),
    schedule_properties: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleScheduleProperties$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_schedule_id": "clientScheduleId",
        "full_disbursement": "fullDisbursement",
        "schedule_properties": "scheduleProperties",
    });
});
/** @internal */
exports.WireWithdrawalScheduleScheduleDetails$outboundSchema = z.object({
    amount: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleAmount$outboundSchema))
        .optional(),
    clientScheduleId: z.string().optional(),
    fullDisbursement: z.boolean().optional(),
    scheduleProperties: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleScheduleProperties$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientScheduleId: "client_schedule_id",
        fullDisbursement: "full_disbursement",
        scheduleProperties: "schedule_properties",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalScheduleScheduleDetails$;
(function (WireWithdrawalScheduleScheduleDetails$) {
    /** @deprecated use `WireWithdrawalScheduleScheduleDetails$inboundSchema` instead. */
    WireWithdrawalScheduleScheduleDetails$.inboundSchema = exports.WireWithdrawalScheduleScheduleDetails$inboundSchema;
    /** @deprecated use `WireWithdrawalScheduleScheduleDetails$outboundSchema` instead. */
    WireWithdrawalScheduleScheduleDetails$.outboundSchema = exports.WireWithdrawalScheduleScheduleDetails$outboundSchema;
})(WireWithdrawalScheduleScheduleDetails$ || (exports.WireWithdrawalScheduleScheduleDetails$ = WireWithdrawalScheduleScheduleDetails$ = {}));
/** @internal */
exports.WireWithdrawalSchedule$inboundSchema = z.object({
    beneficiary: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleBeneficiary$inboundSchema)).optional(),
    intermediary: z.nullable(z.lazy(() => exports.Intermediary$inboundSchema)).optional(),
    name: z.string().optional(),
    recipient_bank: z.nullable(z.lazy(() => exports.RecipientBank$inboundSchema))
        .optional(),
    retirement_distribution: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleRetirementDistribution$inboundSchema)).optional(),
    schedule_details: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleScheduleDetails$inboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "recipient_bank": "recipientBank",
        "retirement_distribution": "retirementDistribution",
        "schedule_details": "scheduleDetails",
    });
});
/** @internal */
exports.WireWithdrawalSchedule$outboundSchema = z.object({
    beneficiary: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleBeneficiary$outboundSchema)).optional(),
    intermediary: z.nullable(z.lazy(() => exports.Intermediary$outboundSchema))
        .optional(),
    name: z.string().optional(),
    recipientBank: z.nullable(z.lazy(() => exports.RecipientBank$outboundSchema))
        .optional(),
    retirementDistribution: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleRetirementDistribution$outboundSchema)).optional(),
    scheduleDetails: z.nullable(z.lazy(() => exports.WireWithdrawalScheduleScheduleDetails$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        recipientBank: "recipient_bank",
        retirementDistribution: "retirement_distribution",
        scheduleDetails: "schedule_details",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalSchedule$;
(function (WireWithdrawalSchedule$) {
    /** @deprecated use `WireWithdrawalSchedule$inboundSchema` instead. */
    WireWithdrawalSchedule$.inboundSchema = exports.WireWithdrawalSchedule$inboundSchema;
    /** @deprecated use `WireWithdrawalSchedule$outboundSchema` instead. */
    WireWithdrawalSchedule$.outboundSchema = exports.WireWithdrawalSchedule$outboundSchema;
})(WireWithdrawalSchedule$ || (exports.WireWithdrawalSchedule$ = WireWithdrawalSchedule$ = {}));
