// @apexfintechsolutions/ascend-sdk/models/components/wirewithdrawalcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WireWithdrawalCreate$ = exports.WireWithdrawalCreate$outboundSchema = exports.WireWithdrawalCreate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const decimalcreate_js_1 = require("./decimalcreate.js");
const retirementdistributioncreate_js_1 = require("./retirementdistributioncreate.js");
const wirewithdrawalbeneficiarycreate_js_1 = require("./wirewithdrawalbeneficiarycreate.js");
const wirewithdrawalintermediarycreate_js_1 = require("./wirewithdrawalintermediarycreate.js");
const wirewithdrawalrecipientbankcreate_js_1 = require("./wirewithdrawalrecipientbankcreate.js");
/** @internal */
exports.WireWithdrawalCreate$inboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$inboundSchema.optional(),
    beneficiary: wirewithdrawalbeneficiarycreate_js_1.WireWithdrawalBeneficiaryCreate$inboundSchema,
    client_transfer_id: z.string(),
    full_disbursement: z.boolean().optional(),
    intermediary: wirewithdrawalintermediarycreate_js_1.WireWithdrawalIntermediaryCreate$inboundSchema.optional(),
    ira_distribution: retirementdistributioncreate_js_1.RetirementDistributionCreate$inboundSchema.optional(),
    recipient_bank: wirewithdrawalrecipientbankcreate_js_1.WireWithdrawalRecipientBankCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "client_transfer_id": "clientTransferId",
        "full_disbursement": "fullDisbursement",
        "ira_distribution": "iraDistribution",
        "recipient_bank": "recipientBank",
    });
});
/** @internal */
exports.WireWithdrawalCreate$outboundSchema = z.object({
    amount: decimalcreate_js_1.DecimalCreate$outboundSchema.optional(),
    beneficiary: wirewithdrawalbeneficiarycreate_js_1.WireWithdrawalBeneficiaryCreate$outboundSchema,
    clientTransferId: z.string(),
    fullDisbursement: z.boolean().optional(),
    intermediary: wirewithdrawalintermediarycreate_js_1.WireWithdrawalIntermediaryCreate$outboundSchema.optional(),
    iraDistribution: retirementdistributioncreate_js_1.RetirementDistributionCreate$outboundSchema.optional(),
    recipientBank: wirewithdrawalrecipientbankcreate_js_1.WireWithdrawalRecipientBankCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        clientTransferId: "client_transfer_id",
        fullDisbursement: "full_disbursement",
        iraDistribution: "ira_distribution",
        recipientBank: "recipient_bank",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalCreate$;
(function (WireWithdrawalCreate$) {
    /** @deprecated use `WireWithdrawalCreate$inboundSchema` instead. */
    WireWithdrawalCreate$.inboundSchema = exports.WireWithdrawalCreate$inboundSchema;
    /** @deprecated use `WireWithdrawalCreate$outboundSchema` instead. */
    WireWithdrawalCreate$.outboundSchema = exports.WireWithdrawalCreate$outboundSchema;
})(WireWithdrawalCreate$ || (exports.WireWithdrawalCreate$ = WireWithdrawalCreate$ = {}));
