// @apexfintechsolutions/ascend-sdk/src/models/operations/accountscreateaccount.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AccountsCreateAccountResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  account?: components.Account | undefined;
  /**
   * INVALID_ARGUMENT: The request is not valid, additional information may be present in the BadRequest details.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AccountsCreateAccountResponse$inboundSchema: z.ZodType<
  AccountsCreateAccountResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  Account: components.Account$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "Account": "account",
    "Status": "status",
  });
});

/** @internal */
export type AccountsCreateAccountResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  Account?: components.Account$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AccountsCreateAccountResponse$outboundSchema: z.ZodType<
  AccountsCreateAccountResponse$Outbound,
  z.ZodTypeDef,
  AccountsCreateAccountResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  account: components.Account$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    account: "Account",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AccountsCreateAccountResponse$ {
  /** @deprecated use `AccountsCreateAccountResponse$inboundSchema` instead. */
  export const inboundSchema = AccountsCreateAccountResponse$inboundSchema;
  /** @deprecated use `AccountsCreateAccountResponse$outboundSchema` instead. */
  export const outboundSchema = AccountsCreateAccountResponse$outboundSchema;
  /** @deprecated use `AccountsCreateAccountResponse$Outbound` instead. */
  export type Outbound = AccountsCreateAccountResponse$Outbound;
}
