// @apexfintechsolutions/ascend-sdk/src/models/operations/accountslistenrollments.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AccountsListEnrollmentsRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The maximum number of enrollments to return.
   */
  pageSize?: number | undefined;
  /**
   * A page token, received from a previous `ListEnrollments` call. Provide this to retrieve the subsequent page.
   *
   * @remarks
   *
   *  When paginating, all other parameters provided to `ListEnrollments` must match the call that provided the page token.
   */
  pageToken?: string | undefined;
};

export type AccountsListEnrollmentsResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  listEnrollmentsResponse?: components.ListEnrollmentsResponse | undefined;
  /**
   * INVALID_ARGUMENT: The request is not valid, additional information may be present in the BadRequest details.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AccountsListEnrollmentsRequest$inboundSchema: z.ZodType<
  AccountsListEnrollmentsRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  page_size: z.number().int().optional(),
  page_token: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "page_size": "pageSize",
    "page_token": "pageToken",
  });
});

/** @internal */
export type AccountsListEnrollmentsRequest$Outbound = {
  account_id: string;
  page_size?: number | undefined;
  page_token?: string | undefined;
};

/** @internal */
export const AccountsListEnrollmentsRequest$outboundSchema: z.ZodType<
  AccountsListEnrollmentsRequest$Outbound,
  z.ZodTypeDef,
  AccountsListEnrollmentsRequest
> = z.object({
  accountId: z.string(),
  pageSize: z.number().int().optional(),
  pageToken: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    pageSize: "page_size",
    pageToken: "page_token",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AccountsListEnrollmentsRequest$ {
  /** @deprecated use `AccountsListEnrollmentsRequest$inboundSchema` instead. */
  export const inboundSchema = AccountsListEnrollmentsRequest$inboundSchema;
  /** @deprecated use `AccountsListEnrollmentsRequest$outboundSchema` instead. */
  export const outboundSchema = AccountsListEnrollmentsRequest$outboundSchema;
  /** @deprecated use `AccountsListEnrollmentsRequest$Outbound` instead. */
  export type Outbound = AccountsListEnrollmentsRequest$Outbound;
}

/** @internal */
export const AccountsListEnrollmentsResponse$inboundSchema: z.ZodType<
  AccountsListEnrollmentsResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  ListEnrollmentsResponse: components.ListEnrollmentsResponse$inboundSchema
    .optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "ListEnrollmentsResponse": "listEnrollmentsResponse",
    "Status": "status",
  });
});

/** @internal */
export type AccountsListEnrollmentsResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  ListEnrollmentsResponse?:
    | components.ListEnrollmentsResponse$Outbound
    | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AccountsListEnrollmentsResponse$outboundSchema: z.ZodType<
  AccountsListEnrollmentsResponse$Outbound,
  z.ZodTypeDef,
  AccountsListEnrollmentsResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  listEnrollmentsResponse: components.ListEnrollmentsResponse$outboundSchema
    .optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    listEnrollmentsResponse: "ListEnrollmentsResponse",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AccountsListEnrollmentsResponse$ {
  /** @deprecated use `AccountsListEnrollmentsResponse$inboundSchema` instead. */
  export const inboundSchema = AccountsListEnrollmentsResponse$inboundSchema;
  /** @deprecated use `AccountsListEnrollmentsResponse$outboundSchema` instead. */
  export const outboundSchema = AccountsListEnrollmentsResponse$outboundSchema;
  /** @deprecated use `AccountsListEnrollmentsResponse$Outbound` instead. */
  export type Outbound = AccountsListEnrollmentsResponse$Outbound;
}
