// @apexfintechsolutions/ascend-sdk/src/models/operations/accountsenrollaccount.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AccountsEnrollAccountRequest = {
  /**
   * The account id.
   */
  accountId: string;
  enrollAccountRequestCreate: components.EnrollAccountRequestCreate;
};

export type AccountsEnrollAccountResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  enrollAccountResponse?: components.EnrollAccountResponse | undefined;
  /**
   * INVALID_ARGUMENT: The request is not valid, additional information may be present in the BadRequest details.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AccountsEnrollAccountRequest$inboundSchema: z.ZodType<
  AccountsEnrollAccountRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  EnrollAccountRequestCreate:
    components.EnrollAccountRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "EnrollAccountRequestCreate": "enrollAccountRequestCreate",
  });
});

/** @internal */
export type AccountsEnrollAccountRequest$Outbound = {
  account_id: string;
  EnrollAccountRequestCreate: components.EnrollAccountRequestCreate$Outbound;
};

/** @internal */
export const AccountsEnrollAccountRequest$outboundSchema: z.ZodType<
  AccountsEnrollAccountRequest$Outbound,
  z.ZodTypeDef,
  AccountsEnrollAccountRequest
> = z.object({
  accountId: z.string(),
  enrollAccountRequestCreate:
    components.EnrollAccountRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    enrollAccountRequestCreate: "EnrollAccountRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AccountsEnrollAccountRequest$ {
  /** @deprecated use `AccountsEnrollAccountRequest$inboundSchema` instead. */
  export const inboundSchema = AccountsEnrollAccountRequest$inboundSchema;
  /** @deprecated use `AccountsEnrollAccountRequest$outboundSchema` instead. */
  export const outboundSchema = AccountsEnrollAccountRequest$outboundSchema;
  /** @deprecated use `AccountsEnrollAccountRequest$Outbound` instead. */
  export type Outbound = AccountsEnrollAccountRequest$Outbound;
}

/** @internal */
export const AccountsEnrollAccountResponse$inboundSchema: z.ZodType<
  AccountsEnrollAccountResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  EnrollAccountResponse: components.EnrollAccountResponse$inboundSchema
    .optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "EnrollAccountResponse": "enrollAccountResponse",
    "Status": "status",
  });
});

/** @internal */
export type AccountsEnrollAccountResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  EnrollAccountResponse?: components.EnrollAccountResponse$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AccountsEnrollAccountResponse$outboundSchema: z.ZodType<
  AccountsEnrollAccountResponse$Outbound,
  z.ZodTypeDef,
  AccountsEnrollAccountResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  enrollAccountResponse: components.EnrollAccountResponse$outboundSchema
    .optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    enrollAccountResponse: "EnrollAccountResponse",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AccountsEnrollAccountResponse$ {
  /** @deprecated use `AccountsEnrollAccountResponse$inboundSchema` instead. */
  export const inboundSchema = AccountsEnrollAccountResponse$inboundSchema;
  /** @deprecated use `AccountsEnrollAccountResponse$outboundSchema` instead. */
  export const outboundSchema = AccountsEnrollAccountResponse$outboundSchema;
  /** @deprecated use `AccountsEnrollAccountResponse$Outbound` instead. */
  export type Outbound = AccountsEnrollAccountResponse$Outbound;
}
