// @apexfintechsolutions/ascend-sdk/src/models/operations/achdepositschedulesupdateachdepositschedule.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AchDepositSchedulesUpdateAchDepositScheduleRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The achDepositSchedule id.
   */
  achDepositScheduleId: string;
  /**
   * A field mask representing the update. Note: only the 'schedule_details.amount' field of a schedule is currently updatable
   */
  updateMask?: string | undefined;
  achDepositScheduleUpdate: components.AchDepositScheduleUpdate;
};

export type AchDepositSchedulesUpdateAchDepositScheduleResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  achDepositSchedule?: components.AchDepositSchedule | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AchDepositSchedulesUpdateAchDepositScheduleRequest$inboundSchema:
  z.ZodType<
    AchDepositSchedulesUpdateAchDepositScheduleRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    achDepositSchedule_id: z.string(),
    update_mask: z.string().optional(),
    AchDepositScheduleUpdate: components.AchDepositScheduleUpdate$inboundSchema,
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "achDepositSchedule_id": "achDepositScheduleId",
      "update_mask": "updateMask",
      "AchDepositScheduleUpdate": "achDepositScheduleUpdate",
    });
  });

/** @internal */
export type AchDepositSchedulesUpdateAchDepositScheduleRequest$Outbound = {
  account_id: string;
  achDepositSchedule_id: string;
  update_mask?: string | undefined;
  AchDepositScheduleUpdate: components.AchDepositScheduleUpdate$Outbound;
};

/** @internal */
export const AchDepositSchedulesUpdateAchDepositScheduleRequest$outboundSchema:
  z.ZodType<
    AchDepositSchedulesUpdateAchDepositScheduleRequest$Outbound,
    z.ZodTypeDef,
    AchDepositSchedulesUpdateAchDepositScheduleRequest
  > = z.object({
    accountId: z.string(),
    achDepositScheduleId: z.string(),
    updateMask: z.string().optional(),
    achDepositScheduleUpdate:
      components.AchDepositScheduleUpdate$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      achDepositScheduleId: "achDepositSchedule_id",
      updateMask: "update_mask",
      achDepositScheduleUpdate: "AchDepositScheduleUpdate",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositSchedulesUpdateAchDepositScheduleRequest$ {
  /** @deprecated use `AchDepositSchedulesUpdateAchDepositScheduleRequest$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositSchedulesUpdateAchDepositScheduleRequest$inboundSchema;
  /** @deprecated use `AchDepositSchedulesUpdateAchDepositScheduleRequest$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositSchedulesUpdateAchDepositScheduleRequest$outboundSchema;
  /** @deprecated use `AchDepositSchedulesUpdateAchDepositScheduleRequest$Outbound` instead. */
  export type Outbound =
    AchDepositSchedulesUpdateAchDepositScheduleRequest$Outbound;
}

/** @internal */
export const AchDepositSchedulesUpdateAchDepositScheduleResponse$inboundSchema:
  z.ZodType<
    AchDepositSchedulesUpdateAchDepositScheduleResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchDepositSchedule: components.AchDepositSchedule$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "AchDepositSchedule": "achDepositSchedule",
      "Status": "status",
    });
  });

/** @internal */
export type AchDepositSchedulesUpdateAchDepositScheduleResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  AchDepositSchedule?: components.AchDepositSchedule$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AchDepositSchedulesUpdateAchDepositScheduleResponse$outboundSchema:
  z.ZodType<
    AchDepositSchedulesUpdateAchDepositScheduleResponse$Outbound,
    z.ZodTypeDef,
    AchDepositSchedulesUpdateAchDepositScheduleResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achDepositSchedule: components.AchDepositSchedule$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      achDepositSchedule: "AchDepositSchedule",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositSchedulesUpdateAchDepositScheduleResponse$ {
  /** @deprecated use `AchDepositSchedulesUpdateAchDepositScheduleResponse$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositSchedulesUpdateAchDepositScheduleResponse$inboundSchema;
  /** @deprecated use `AchDepositSchedulesUpdateAchDepositScheduleResponse$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositSchedulesUpdateAchDepositScheduleResponse$outboundSchema;
  /** @deprecated use `AchDepositSchedulesUpdateAchDepositScheduleResponse$Outbound` instead. */
  export type Outbound =
    AchDepositSchedulesUpdateAchDepositScheduleResponse$Outbound;
}
