// @apexfintechsolutions/ascend-sdk/src/models/operations/achdepositscancelachdeposit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AchDepositsCancelAchDepositRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The achDeposit id.
   */
  achDepositId: string;
  cancelAchDepositRequestCreate: components.CancelAchDepositRequestCreate;
};

export type AchDepositsCancelAchDepositResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  achDeposit?: components.AchDeposit | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AchDepositsCancelAchDepositRequest$inboundSchema: z.ZodType<
  AchDepositsCancelAchDepositRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  achDeposit_id: z.string(),
  CancelAchDepositRequestCreate:
    components.CancelAchDepositRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "achDeposit_id": "achDepositId",
    "CancelAchDepositRequestCreate": "cancelAchDepositRequestCreate",
  });
});

/** @internal */
export type AchDepositsCancelAchDepositRequest$Outbound = {
  account_id: string;
  achDeposit_id: string;
  CancelAchDepositRequestCreate:
    components.CancelAchDepositRequestCreate$Outbound;
};

/** @internal */
export const AchDepositsCancelAchDepositRequest$outboundSchema: z.ZodType<
  AchDepositsCancelAchDepositRequest$Outbound,
  z.ZodTypeDef,
  AchDepositsCancelAchDepositRequest
> = z.object({
  accountId: z.string(),
  achDepositId: z.string(),
  cancelAchDepositRequestCreate:
    components.CancelAchDepositRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    achDepositId: "achDeposit_id",
    cancelAchDepositRequestCreate: "CancelAchDepositRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositsCancelAchDepositRequest$ {
  /** @deprecated use `AchDepositsCancelAchDepositRequest$inboundSchema` instead. */
  export const inboundSchema = AchDepositsCancelAchDepositRequest$inboundSchema;
  /** @deprecated use `AchDepositsCancelAchDepositRequest$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositsCancelAchDepositRequest$outboundSchema;
  /** @deprecated use `AchDepositsCancelAchDepositRequest$Outbound` instead. */
  export type Outbound = AchDepositsCancelAchDepositRequest$Outbound;
}

/** @internal */
export const AchDepositsCancelAchDepositResponse$inboundSchema: z.ZodType<
  AchDepositsCancelAchDepositResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  AchDeposit: components.AchDeposit$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "AchDeposit": "achDeposit",
    "Status": "status",
  });
});

/** @internal */
export type AchDepositsCancelAchDepositResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  AchDeposit?: components.AchDeposit$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AchDepositsCancelAchDepositResponse$outboundSchema: z.ZodType<
  AchDepositsCancelAchDepositResponse$Outbound,
  z.ZodTypeDef,
  AchDepositsCancelAchDepositResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  achDeposit: components.AchDeposit$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    achDeposit: "AchDeposit",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositsCancelAchDepositResponse$ {
  /** @deprecated use `AchDepositsCancelAchDepositResponse$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositsCancelAchDepositResponse$inboundSchema;
  /** @deprecated use `AchDepositsCancelAchDepositResponse$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositsCancelAchDepositResponse$outboundSchema;
  /** @deprecated use `AchDepositsCancelAchDepositResponse$Outbound` instead. */
  export type Outbound = AchDepositsCancelAchDepositResponse$Outbound;
}
