// @apexfintechsolutions/ascend-sdk/src/models/operations/achdepositsforcereturnachdeposit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AchDepositsForceReturnAchDepositRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The achDeposit id.
   */
  achDepositId: string;
  forceReturnAchDepositRequestCreate:
    components.ForceReturnAchDepositRequestCreate;
};

export type AchDepositsForceReturnAchDepositResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  achDeposit?: components.AchDeposit | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AchDepositsForceReturnAchDepositRequest$inboundSchema: z.ZodType<
  AchDepositsForceReturnAchDepositRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  achDeposit_id: z.string(),
  ForceReturnAchDepositRequestCreate:
    components.ForceReturnAchDepositRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "achDeposit_id": "achDepositId",
    "ForceReturnAchDepositRequestCreate": "forceReturnAchDepositRequestCreate",
  });
});

/** @internal */
export type AchDepositsForceReturnAchDepositRequest$Outbound = {
  account_id: string;
  achDeposit_id: string;
  ForceReturnAchDepositRequestCreate:
    components.ForceReturnAchDepositRequestCreate$Outbound;
};

/** @internal */
export const AchDepositsForceReturnAchDepositRequest$outboundSchema: z.ZodType<
  AchDepositsForceReturnAchDepositRequest$Outbound,
  z.ZodTypeDef,
  AchDepositsForceReturnAchDepositRequest
> = z.object({
  accountId: z.string(),
  achDepositId: z.string(),
  forceReturnAchDepositRequestCreate:
    components.ForceReturnAchDepositRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    achDepositId: "achDeposit_id",
    forceReturnAchDepositRequestCreate: "ForceReturnAchDepositRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositsForceReturnAchDepositRequest$ {
  /** @deprecated use `AchDepositsForceReturnAchDepositRequest$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositsForceReturnAchDepositRequest$inboundSchema;
  /** @deprecated use `AchDepositsForceReturnAchDepositRequest$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositsForceReturnAchDepositRequest$outboundSchema;
  /** @deprecated use `AchDepositsForceReturnAchDepositRequest$Outbound` instead. */
  export type Outbound = AchDepositsForceReturnAchDepositRequest$Outbound;
}

/** @internal */
export const AchDepositsForceReturnAchDepositResponse$inboundSchema: z.ZodType<
  AchDepositsForceReturnAchDepositResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  AchDeposit: components.AchDeposit$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "AchDeposit": "achDeposit",
    "Status": "status",
  });
});

/** @internal */
export type AchDepositsForceReturnAchDepositResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  AchDeposit?: components.AchDeposit$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AchDepositsForceReturnAchDepositResponse$outboundSchema: z.ZodType<
  AchDepositsForceReturnAchDepositResponse$Outbound,
  z.ZodTypeDef,
  AchDepositsForceReturnAchDepositResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  achDeposit: components.AchDeposit$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    achDeposit: "AchDeposit",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositsForceReturnAchDepositResponse$ {
  /** @deprecated use `AchDepositsForceReturnAchDepositResponse$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositsForceReturnAchDepositResponse$inboundSchema;
  /** @deprecated use `AchDepositsForceReturnAchDepositResponse$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositsForceReturnAchDepositResponse$outboundSchema;
  /** @deprecated use `AchDepositsForceReturnAchDepositResponse$Outbound` instead. */
  export type Outbound = AchDepositsForceReturnAchDepositResponse$Outbound;
}
