// @apexfintechsolutions/ascend-sdk/src/models/operations/achdepositsforcerejectachdeposit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AchDepositsForceRejectAchDepositRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The achDeposit id.
   */
  achDepositId: string;
  forceRejectAchDepositRequestCreate:
    components.ForceRejectAchDepositRequestCreate;
};

export type AchDepositsForceRejectAchDepositResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  achDeposit?: components.AchDeposit | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AchDepositsForceRejectAchDepositRequest$inboundSchema: z.ZodType<
  AchDepositsForceRejectAchDepositRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  achDeposit_id: z.string(),
  ForceRejectAchDepositRequestCreate:
    components.ForceRejectAchDepositRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "achDeposit_id": "achDepositId",
    "ForceRejectAchDepositRequestCreate": "forceRejectAchDepositRequestCreate",
  });
});

/** @internal */
export type AchDepositsForceRejectAchDepositRequest$Outbound = {
  account_id: string;
  achDeposit_id: string;
  ForceRejectAchDepositRequestCreate:
    components.ForceRejectAchDepositRequestCreate$Outbound;
};

/** @internal */
export const AchDepositsForceRejectAchDepositRequest$outboundSchema: z.ZodType<
  AchDepositsForceRejectAchDepositRequest$Outbound,
  z.ZodTypeDef,
  AchDepositsForceRejectAchDepositRequest
> = z.object({
  accountId: z.string(),
  achDepositId: z.string(),
  forceRejectAchDepositRequestCreate:
    components.ForceRejectAchDepositRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    achDepositId: "achDeposit_id",
    forceRejectAchDepositRequestCreate: "ForceRejectAchDepositRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositsForceRejectAchDepositRequest$ {
  /** @deprecated use `AchDepositsForceRejectAchDepositRequest$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositsForceRejectAchDepositRequest$inboundSchema;
  /** @deprecated use `AchDepositsForceRejectAchDepositRequest$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositsForceRejectAchDepositRequest$outboundSchema;
  /** @deprecated use `AchDepositsForceRejectAchDepositRequest$Outbound` instead. */
  export type Outbound = AchDepositsForceRejectAchDepositRequest$Outbound;
}

/** @internal */
export const AchDepositsForceRejectAchDepositResponse$inboundSchema: z.ZodType<
  AchDepositsForceRejectAchDepositResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  AchDeposit: components.AchDeposit$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "AchDeposit": "achDeposit",
    "Status": "status",
  });
});

/** @internal */
export type AchDepositsForceRejectAchDepositResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  AchDeposit?: components.AchDeposit$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AchDepositsForceRejectAchDepositResponse$outboundSchema: z.ZodType<
  AchDepositsForceRejectAchDepositResponse$Outbound,
  z.ZodTypeDef,
  AchDepositsForceRejectAchDepositResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  achDeposit: components.AchDeposit$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    achDeposit: "AchDeposit",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositsForceRejectAchDepositResponse$ {
  /** @deprecated use `AchDepositsForceRejectAchDepositResponse$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositsForceRejectAchDepositResponse$inboundSchema;
  /** @deprecated use `AchDepositsForceRejectAchDepositResponse$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositsForceRejectAchDepositResponse$outboundSchema;
  /** @deprecated use `AchDepositsForceRejectAchDepositResponse$Outbound` instead. */
  export type Outbound = AchDepositsForceRejectAchDepositResponse$Outbound;
}
