// @apexfintechsolutions/ascend-sdk/src/models/operations/achdepositschedulescancelachdepositschedule.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AchDepositSchedulesCancelAchDepositScheduleRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The achDepositSchedule id.
   */
  achDepositScheduleId: string;
  cancelAchDepositScheduleRequestCreate:
    components.CancelAchDepositScheduleRequestCreate;
};

export type AchDepositSchedulesCancelAchDepositScheduleResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  achDepositSchedule?: components.AchDepositSchedule | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AchDepositSchedulesCancelAchDepositScheduleRequest$inboundSchema:
  z.ZodType<
    AchDepositSchedulesCancelAchDepositScheduleRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    achDepositSchedule_id: z.string(),
    CancelAchDepositScheduleRequestCreate:
      components.CancelAchDepositScheduleRequestCreate$inboundSchema,
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "achDepositSchedule_id": "achDepositScheduleId",
      "CancelAchDepositScheduleRequestCreate":
        "cancelAchDepositScheduleRequestCreate",
    });
  });

/** @internal */
export type AchDepositSchedulesCancelAchDepositScheduleRequest$Outbound = {
  account_id: string;
  achDepositSchedule_id: string;
  CancelAchDepositScheduleRequestCreate:
    components.CancelAchDepositScheduleRequestCreate$Outbound;
};

/** @internal */
export const AchDepositSchedulesCancelAchDepositScheduleRequest$outboundSchema:
  z.ZodType<
    AchDepositSchedulesCancelAchDepositScheduleRequest$Outbound,
    z.ZodTypeDef,
    AchDepositSchedulesCancelAchDepositScheduleRequest
  > = z.object({
    accountId: z.string(),
    achDepositScheduleId: z.string(),
    cancelAchDepositScheduleRequestCreate:
      components.CancelAchDepositScheduleRequestCreate$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      achDepositScheduleId: "achDepositSchedule_id",
      cancelAchDepositScheduleRequestCreate:
        "CancelAchDepositScheduleRequestCreate",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositSchedulesCancelAchDepositScheduleRequest$ {
  /** @deprecated use `AchDepositSchedulesCancelAchDepositScheduleRequest$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositSchedulesCancelAchDepositScheduleRequest$inboundSchema;
  /** @deprecated use `AchDepositSchedulesCancelAchDepositScheduleRequest$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositSchedulesCancelAchDepositScheduleRequest$outboundSchema;
  /** @deprecated use `AchDepositSchedulesCancelAchDepositScheduleRequest$Outbound` instead. */
  export type Outbound =
    AchDepositSchedulesCancelAchDepositScheduleRequest$Outbound;
}

/** @internal */
export const AchDepositSchedulesCancelAchDepositScheduleResponse$inboundSchema:
  z.ZodType<
    AchDepositSchedulesCancelAchDepositScheduleResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchDepositSchedule: components.AchDepositSchedule$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "AchDepositSchedule": "achDepositSchedule",
      "Status": "status",
    });
  });

/** @internal */
export type AchDepositSchedulesCancelAchDepositScheduleResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  AchDepositSchedule?: components.AchDepositSchedule$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AchDepositSchedulesCancelAchDepositScheduleResponse$outboundSchema:
  z.ZodType<
    AchDepositSchedulesCancelAchDepositScheduleResponse$Outbound,
    z.ZodTypeDef,
    AchDepositSchedulesCancelAchDepositScheduleResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achDepositSchedule: components.AchDepositSchedule$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      achDepositSchedule: "AchDepositSchedule",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositSchedulesCancelAchDepositScheduleResponse$ {
  /** @deprecated use `AchDepositSchedulesCancelAchDepositScheduleResponse$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositSchedulesCancelAchDepositScheduleResponse$inboundSchema;
  /** @deprecated use `AchDepositSchedulesCancelAchDepositScheduleResponse$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositSchedulesCancelAchDepositScheduleResponse$outboundSchema;
  /** @deprecated use `AchDepositSchedulesCancelAchDepositScheduleResponse$Outbound` instead. */
  export type Outbound =
    AchDepositSchedulesCancelAchDepositScheduleResponse$Outbound;
}
