// @apexfintechsolutions/ascend-sdk/src/models/operations/achdepositscreateachdeposit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AchDepositsCreateAchDepositRequest = {
  /**
   * The account id.
   */
  accountId: string;
  achDepositCreate: components.AchDepositCreate;
};

export type AchDepositsCreateAchDepositResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  achDeposit?: components.AchDeposit | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AchDepositsCreateAchDepositRequest$inboundSchema: z.ZodType<
  AchDepositsCreateAchDepositRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  AchDepositCreate: components.AchDepositCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "AchDepositCreate": "achDepositCreate",
  });
});

/** @internal */
export type AchDepositsCreateAchDepositRequest$Outbound = {
  account_id: string;
  AchDepositCreate: components.AchDepositCreate$Outbound;
};

/** @internal */
export const AchDepositsCreateAchDepositRequest$outboundSchema: z.ZodType<
  AchDepositsCreateAchDepositRequest$Outbound,
  z.ZodTypeDef,
  AchDepositsCreateAchDepositRequest
> = z.object({
  accountId: z.string(),
  achDepositCreate: components.AchDepositCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    achDepositCreate: "AchDepositCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositsCreateAchDepositRequest$ {
  /** @deprecated use `AchDepositsCreateAchDepositRequest$inboundSchema` instead. */
  export const inboundSchema = AchDepositsCreateAchDepositRequest$inboundSchema;
  /** @deprecated use `AchDepositsCreateAchDepositRequest$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositsCreateAchDepositRequest$outboundSchema;
  /** @deprecated use `AchDepositsCreateAchDepositRequest$Outbound` instead. */
  export type Outbound = AchDepositsCreateAchDepositRequest$Outbound;
}

/** @internal */
export const AchDepositsCreateAchDepositResponse$inboundSchema: z.ZodType<
  AchDepositsCreateAchDepositResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  AchDeposit: components.AchDeposit$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "AchDeposit": "achDeposit",
    "Status": "status",
  });
});

/** @internal */
export type AchDepositsCreateAchDepositResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  AchDeposit?: components.AchDeposit$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AchDepositsCreateAchDepositResponse$outboundSchema: z.ZodType<
  AchDepositsCreateAchDepositResponse$Outbound,
  z.ZodTypeDef,
  AchDepositsCreateAchDepositResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  achDeposit: components.AchDeposit$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    achDeposit: "AchDeposit",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositsCreateAchDepositResponse$ {
  /** @deprecated use `AchDepositsCreateAchDepositResponse$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositsCreateAchDepositResponse$inboundSchema;
  /** @deprecated use `AchDepositsCreateAchDepositResponse$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositsCreateAchDepositResponse$outboundSchema;
  /** @deprecated use `AchDepositsCreateAchDepositResponse$Outbound` instead. */
  export type Outbound = AchDepositsCreateAchDepositResponse$Outbound;
}
