// @apexfintechsolutions/ascend-sdk/src/models/operations/achdepositschedulescreateachdepositschedule.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AchDepositSchedulesCreateAchDepositScheduleRequest = {
  /**
   * The account id.
   */
  accountId: string;
  achDepositScheduleCreate: components.AchDepositScheduleCreate;
};

export type AchDepositSchedulesCreateAchDepositScheduleResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  achDepositSchedule?: components.AchDepositSchedule | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AchDepositSchedulesCreateAchDepositScheduleRequest$inboundSchema:
  z.ZodType<
    AchDepositSchedulesCreateAchDepositScheduleRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    AchDepositScheduleCreate: components.AchDepositScheduleCreate$inboundSchema,
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "AchDepositScheduleCreate": "achDepositScheduleCreate",
    });
  });

/** @internal */
export type AchDepositSchedulesCreateAchDepositScheduleRequest$Outbound = {
  account_id: string;
  AchDepositScheduleCreate: components.AchDepositScheduleCreate$Outbound;
};

/** @internal */
export const AchDepositSchedulesCreateAchDepositScheduleRequest$outboundSchema:
  z.ZodType<
    AchDepositSchedulesCreateAchDepositScheduleRequest$Outbound,
    z.ZodTypeDef,
    AchDepositSchedulesCreateAchDepositScheduleRequest
  > = z.object({
    accountId: z.string(),
    achDepositScheduleCreate:
      components.AchDepositScheduleCreate$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      achDepositScheduleCreate: "AchDepositScheduleCreate",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositSchedulesCreateAchDepositScheduleRequest$ {
  /** @deprecated use `AchDepositSchedulesCreateAchDepositScheduleRequest$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositSchedulesCreateAchDepositScheduleRequest$inboundSchema;
  /** @deprecated use `AchDepositSchedulesCreateAchDepositScheduleRequest$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositSchedulesCreateAchDepositScheduleRequest$outboundSchema;
  /** @deprecated use `AchDepositSchedulesCreateAchDepositScheduleRequest$Outbound` instead. */
  export type Outbound =
    AchDepositSchedulesCreateAchDepositScheduleRequest$Outbound;
}

/** @internal */
export const AchDepositSchedulesCreateAchDepositScheduleResponse$inboundSchema:
  z.ZodType<
    AchDepositSchedulesCreateAchDepositScheduleResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchDepositSchedule: components.AchDepositSchedule$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "AchDepositSchedule": "achDepositSchedule",
      "Status": "status",
    });
  });

/** @internal */
export type AchDepositSchedulesCreateAchDepositScheduleResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  AchDepositSchedule?: components.AchDepositSchedule$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AchDepositSchedulesCreateAchDepositScheduleResponse$outboundSchema:
  z.ZodType<
    AchDepositSchedulesCreateAchDepositScheduleResponse$Outbound,
    z.ZodTypeDef,
    AchDepositSchedulesCreateAchDepositScheduleResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achDepositSchedule: components.AchDepositSchedule$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      achDepositSchedule: "AchDepositSchedule",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositSchedulesCreateAchDepositScheduleResponse$ {
  /** @deprecated use `AchDepositSchedulesCreateAchDepositScheduleResponse$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositSchedulesCreateAchDepositScheduleResponse$inboundSchema;
  /** @deprecated use `AchDepositSchedulesCreateAchDepositScheduleResponse$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositSchedulesCreateAchDepositScheduleResponse$outboundSchema;
  /** @deprecated use `AchDepositSchedulesCreateAchDepositScheduleResponse$Outbound` instead. */
  export type Outbound =
    AchDepositSchedulesCreateAchDepositScheduleResponse$Outbound;
}
