// @apexfintechsolutions/ascend-sdk/src/models/operations/achdepositschedulesgetachdepositschedule.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AchDepositSchedulesGetAchDepositScheduleRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The achDepositSchedule id.
   */
  achDepositScheduleId: string;
};

export type AchDepositSchedulesGetAchDepositScheduleResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  achDepositSchedule?: components.AchDepositSchedule | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AchDepositSchedulesGetAchDepositScheduleRequest$inboundSchema:
  z.ZodType<
    AchDepositSchedulesGetAchDepositScheduleRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    achDepositSchedule_id: z.string(),
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "achDepositSchedule_id": "achDepositScheduleId",
    });
  });

/** @internal */
export type AchDepositSchedulesGetAchDepositScheduleRequest$Outbound = {
  account_id: string;
  achDepositSchedule_id: string;
};

/** @internal */
export const AchDepositSchedulesGetAchDepositScheduleRequest$outboundSchema:
  z.ZodType<
    AchDepositSchedulesGetAchDepositScheduleRequest$Outbound,
    z.ZodTypeDef,
    AchDepositSchedulesGetAchDepositScheduleRequest
  > = z.object({
    accountId: z.string(),
    achDepositScheduleId: z.string(),
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      achDepositScheduleId: "achDepositSchedule_id",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositSchedulesGetAchDepositScheduleRequest$ {
  /** @deprecated use `AchDepositSchedulesGetAchDepositScheduleRequest$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositSchedulesGetAchDepositScheduleRequest$inboundSchema;
  /** @deprecated use `AchDepositSchedulesGetAchDepositScheduleRequest$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositSchedulesGetAchDepositScheduleRequest$outboundSchema;
  /** @deprecated use `AchDepositSchedulesGetAchDepositScheduleRequest$Outbound` instead. */
  export type Outbound =
    AchDepositSchedulesGetAchDepositScheduleRequest$Outbound;
}

/** @internal */
export const AchDepositSchedulesGetAchDepositScheduleResponse$inboundSchema:
  z.ZodType<
    AchDepositSchedulesGetAchDepositScheduleResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchDepositSchedule: components.AchDepositSchedule$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "AchDepositSchedule": "achDepositSchedule",
      "Status": "status",
    });
  });

/** @internal */
export type AchDepositSchedulesGetAchDepositScheduleResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  AchDepositSchedule?: components.AchDepositSchedule$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AchDepositSchedulesGetAchDepositScheduleResponse$outboundSchema:
  z.ZodType<
    AchDepositSchedulesGetAchDepositScheduleResponse$Outbound,
    z.ZodTypeDef,
    AchDepositSchedulesGetAchDepositScheduleResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achDepositSchedule: components.AchDepositSchedule$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      achDepositSchedule: "AchDepositSchedule",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositSchedulesGetAchDepositScheduleResponse$ {
  /** @deprecated use `AchDepositSchedulesGetAchDepositScheduleResponse$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositSchedulesGetAchDepositScheduleResponse$inboundSchema;
  /** @deprecated use `AchDepositSchedulesGetAchDepositScheduleResponse$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositSchedulesGetAchDepositScheduleResponse$outboundSchema;
  /** @deprecated use `AchDepositSchedulesGetAchDepositScheduleResponse$Outbound` instead. */
  export type Outbound =
    AchDepositSchedulesGetAchDepositScheduleResponse$Outbound;
}
