// @apexfintechsolutions/ascend-sdk/src/models/operations/achdepositscheduleslistachdepositschedules.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AchDepositSchedulesListAchDepositSchedulesRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * A CEL string to filter results; See the [CEL Search](https://developer.apexclearing.com/apex-fintech-solutions/docs/cel-search) page in Guides for more information; Filter options include:
   *
   * @remarks
   *  `state`
   *  `start_date`
   *  `end_date`
   */
  filter?: string | undefined;
  /**
   * The maximum number of schedules to return. The service may return fewer than this value. If unspecified, at most 25 schedules will be returned. The maximum value is 1000; values above 1000 will be coerced to 1000.
   */
  pageSize?: number | undefined;
  /**
   * The page token to request
   */
  pageToken?: string | undefined;
};

export type AchDepositSchedulesListAchDepositSchedulesResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  listAchDepositSchedulesResponse?:
    | components.ListAchDepositSchedulesResponse
    | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AchDepositSchedulesListAchDepositSchedulesRequest$inboundSchema:
  z.ZodType<
    AchDepositSchedulesListAchDepositSchedulesRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    filter: z.string().optional(),
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "page_size": "pageSize",
      "page_token": "pageToken",
    });
  });

/** @internal */
export type AchDepositSchedulesListAchDepositSchedulesRequest$Outbound = {
  account_id: string;
  filter?: string | undefined;
  page_size?: number | undefined;
  page_token?: string | undefined;
};

/** @internal */
export const AchDepositSchedulesListAchDepositSchedulesRequest$outboundSchema:
  z.ZodType<
    AchDepositSchedulesListAchDepositSchedulesRequest$Outbound,
    z.ZodTypeDef,
    AchDepositSchedulesListAchDepositSchedulesRequest
  > = z.object({
    accountId: z.string(),
    filter: z.string().optional(),
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      pageSize: "page_size",
      pageToken: "page_token",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositSchedulesListAchDepositSchedulesRequest$ {
  /** @deprecated use `AchDepositSchedulesListAchDepositSchedulesRequest$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositSchedulesListAchDepositSchedulesRequest$inboundSchema;
  /** @deprecated use `AchDepositSchedulesListAchDepositSchedulesRequest$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositSchedulesListAchDepositSchedulesRequest$outboundSchema;
  /** @deprecated use `AchDepositSchedulesListAchDepositSchedulesRequest$Outbound` instead. */
  export type Outbound =
    AchDepositSchedulesListAchDepositSchedulesRequest$Outbound;
}

/** @internal */
export const AchDepositSchedulesListAchDepositSchedulesResponse$inboundSchema:
  z.ZodType<
    AchDepositSchedulesListAchDepositSchedulesResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListAchDepositSchedulesResponse: components
      .ListAchDepositSchedulesResponse$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "ListAchDepositSchedulesResponse": "listAchDepositSchedulesResponse",
      "Status": "status",
    });
  });

/** @internal */
export type AchDepositSchedulesListAchDepositSchedulesResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  ListAchDepositSchedulesResponse?:
    | components.ListAchDepositSchedulesResponse$Outbound
    | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AchDepositSchedulesListAchDepositSchedulesResponse$outboundSchema:
  z.ZodType<
    AchDepositSchedulesListAchDepositSchedulesResponse$Outbound,
    z.ZodTypeDef,
    AchDepositSchedulesListAchDepositSchedulesResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listAchDepositSchedulesResponse: components
      .ListAchDepositSchedulesResponse$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      listAchDepositSchedulesResponse: "ListAchDepositSchedulesResponse",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositSchedulesListAchDepositSchedulesResponse$ {
  /** @deprecated use `AchDepositSchedulesListAchDepositSchedulesResponse$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositSchedulesListAchDepositSchedulesResponse$inboundSchema;
  /** @deprecated use `AchDepositSchedulesListAchDepositSchedulesResponse$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositSchedulesListAchDepositSchedulesResponse$outboundSchema;
  /** @deprecated use `AchDepositSchedulesListAchDepositSchedulesResponse$Outbound` instead. */
  export type Outbound =
    AchDepositSchedulesListAchDepositSchedulesResponse$Outbound;
}
