// @apexfintechsolutions/ascend-sdk/src/models/operations/achdepositsforceapproveachdeposit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AchDepositsForceApproveAchDepositRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The achDeposit id.
   */
  achDepositId: string;
  forceApproveAchDepositRequestCreate:
    components.ForceApproveAchDepositRequestCreate;
};

export type AchDepositsForceApproveAchDepositResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  achDeposit?: components.AchDeposit | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AchDepositsForceApproveAchDepositRequest$inboundSchema: z.ZodType<
  AchDepositsForceApproveAchDepositRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  achDeposit_id: z.string(),
  ForceApproveAchDepositRequestCreate:
    components.ForceApproveAchDepositRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "achDeposit_id": "achDepositId",
    "ForceApproveAchDepositRequestCreate":
      "forceApproveAchDepositRequestCreate",
  });
});

/** @internal */
export type AchDepositsForceApproveAchDepositRequest$Outbound = {
  account_id: string;
  achDeposit_id: string;
  ForceApproveAchDepositRequestCreate:
    components.ForceApproveAchDepositRequestCreate$Outbound;
};

/** @internal */
export const AchDepositsForceApproveAchDepositRequest$outboundSchema: z.ZodType<
  AchDepositsForceApproveAchDepositRequest$Outbound,
  z.ZodTypeDef,
  AchDepositsForceApproveAchDepositRequest
> = z.object({
  accountId: z.string(),
  achDepositId: z.string(),
  forceApproveAchDepositRequestCreate:
    components.ForceApproveAchDepositRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    achDepositId: "achDeposit_id",
    forceApproveAchDepositRequestCreate: "ForceApproveAchDepositRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositsForceApproveAchDepositRequest$ {
  /** @deprecated use `AchDepositsForceApproveAchDepositRequest$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositsForceApproveAchDepositRequest$inboundSchema;
  /** @deprecated use `AchDepositsForceApproveAchDepositRequest$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositsForceApproveAchDepositRequest$outboundSchema;
  /** @deprecated use `AchDepositsForceApproveAchDepositRequest$Outbound` instead. */
  export type Outbound = AchDepositsForceApproveAchDepositRequest$Outbound;
}

/** @internal */
export const AchDepositsForceApproveAchDepositResponse$inboundSchema: z.ZodType<
  AchDepositsForceApproveAchDepositResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  AchDeposit: components.AchDeposit$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "AchDeposit": "achDeposit",
    "Status": "status",
  });
});

/** @internal */
export type AchDepositsForceApproveAchDepositResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  AchDeposit?: components.AchDeposit$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AchDepositsForceApproveAchDepositResponse$outboundSchema:
  z.ZodType<
    AchDepositsForceApproveAchDepositResponse$Outbound,
    z.ZodTypeDef,
    AchDepositsForceApproveAchDepositResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achDeposit: components.AchDeposit$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      achDeposit: "AchDeposit",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchDepositsForceApproveAchDepositResponse$ {
  /** @deprecated use `AchDepositsForceApproveAchDepositResponse$inboundSchema` instead. */
  export const inboundSchema =
    AchDepositsForceApproveAchDepositResponse$inboundSchema;
  /** @deprecated use `AchDepositsForceApproveAchDepositResponse$outboundSchema` instead. */
  export const outboundSchema =
    AchDepositsForceApproveAchDepositResponse$outboundSchema;
  /** @deprecated use `AchDepositsForceApproveAchDepositResponse$Outbound` instead. */
  export type Outbound = AchDepositsForceApproveAchDepositResponse$Outbound;
}
