// @apexfintechsolutions/ascend-sdk/src/models/operations/achwithdrawalsforcereturnachwithdrawal.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type AchWithdrawalsForceReturnAchWithdrawalRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The achWithdrawal id.
   */
  achWithdrawalId: string;
  forceReturnAchWithdrawalRequestCreate:
    components.ForceReturnAchWithdrawalRequestCreate;
};

export type AchWithdrawalsForceReturnAchWithdrawalResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  achWithdrawal?: components.AchWithdrawal | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const AchWithdrawalsForceReturnAchWithdrawalRequest$inboundSchema:
  z.ZodType<
    AchWithdrawalsForceReturnAchWithdrawalRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    achWithdrawal_id: z.string(),
    ForceReturnAchWithdrawalRequestCreate:
      components.ForceReturnAchWithdrawalRequestCreate$inboundSchema,
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "achWithdrawal_id": "achWithdrawalId",
      "ForceReturnAchWithdrawalRequestCreate":
        "forceReturnAchWithdrawalRequestCreate",
    });
  });

/** @internal */
export type AchWithdrawalsForceReturnAchWithdrawalRequest$Outbound = {
  account_id: string;
  achWithdrawal_id: string;
  ForceReturnAchWithdrawalRequestCreate:
    components.ForceReturnAchWithdrawalRequestCreate$Outbound;
};

/** @internal */
export const AchWithdrawalsForceReturnAchWithdrawalRequest$outboundSchema:
  z.ZodType<
    AchWithdrawalsForceReturnAchWithdrawalRequest$Outbound,
    z.ZodTypeDef,
    AchWithdrawalsForceReturnAchWithdrawalRequest
  > = z.object({
    accountId: z.string(),
    achWithdrawalId: z.string(),
    forceReturnAchWithdrawalRequestCreate:
      components.ForceReturnAchWithdrawalRequestCreate$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      achWithdrawalId: "achWithdrawal_id",
      forceReturnAchWithdrawalRequestCreate:
        "ForceReturnAchWithdrawalRequestCreate",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchWithdrawalsForceReturnAchWithdrawalRequest$ {
  /** @deprecated use `AchWithdrawalsForceReturnAchWithdrawalRequest$inboundSchema` instead. */
  export const inboundSchema =
    AchWithdrawalsForceReturnAchWithdrawalRequest$inboundSchema;
  /** @deprecated use `AchWithdrawalsForceReturnAchWithdrawalRequest$outboundSchema` instead. */
  export const outboundSchema =
    AchWithdrawalsForceReturnAchWithdrawalRequest$outboundSchema;
  /** @deprecated use `AchWithdrawalsForceReturnAchWithdrawalRequest$Outbound` instead. */
  export type Outbound = AchWithdrawalsForceReturnAchWithdrawalRequest$Outbound;
}

/** @internal */
export const AchWithdrawalsForceReturnAchWithdrawalResponse$inboundSchema:
  z.ZodType<
    AchWithdrawalsForceReturnAchWithdrawalResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchWithdrawal: components.AchWithdrawal$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "AchWithdrawal": "achWithdrawal",
      "Status": "status",
    });
  });

/** @internal */
export type AchWithdrawalsForceReturnAchWithdrawalResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  AchWithdrawal?: components.AchWithdrawal$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const AchWithdrawalsForceReturnAchWithdrawalResponse$outboundSchema:
  z.ZodType<
    AchWithdrawalsForceReturnAchWithdrawalResponse$Outbound,
    z.ZodTypeDef,
    AchWithdrawalsForceReturnAchWithdrawalResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achWithdrawal: components.AchWithdrawal$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      achWithdrawal: "AchWithdrawal",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace AchWithdrawalsForceReturnAchWithdrawalResponse$ {
  /** @deprecated use `AchWithdrawalsForceReturnAchWithdrawalResponse$inboundSchema` instead. */
  export const inboundSchema =
    AchWithdrawalsForceReturnAchWithdrawalResponse$inboundSchema;
  /** @deprecated use `AchWithdrawalsForceReturnAchWithdrawalResponse$outboundSchema` instead. */
  export const outboundSchema =
    AchWithdrawalsForceReturnAchWithdrawalResponse$outboundSchema;
  /** @deprecated use `AchWithdrawalsForceReturnAchWithdrawalResponse$Outbound` instead. */
  export type Outbound =
    AchWithdrawalsForceReturnAchWithdrawalResponse$Outbound;
}
