// @apexfintechsolutions/ascend-sdk/src/models/operations/bankrelationshipsverifymicrodeposits.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type BankRelationshipsVerifyMicroDepositsRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The bankRelationship id.
   */
  bankRelationshipId: string;
  verifyMicroDepositsRequestCreate: components.VerifyMicroDepositsRequestCreate;
};

export type BankRelationshipsVerifyMicroDepositsResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  bankRelationship?: components.BankRelationship | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const BankRelationshipsVerifyMicroDepositsRequest$inboundSchema:
  z.ZodType<
    BankRelationshipsVerifyMicroDepositsRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    bankRelationship_id: z.string(),
    VerifyMicroDepositsRequestCreate:
      components.VerifyMicroDepositsRequestCreate$inboundSchema,
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "bankRelationship_id": "bankRelationshipId",
      "VerifyMicroDepositsRequestCreate": "verifyMicroDepositsRequestCreate",
    });
  });

/** @internal */
export type BankRelationshipsVerifyMicroDepositsRequest$Outbound = {
  account_id: string;
  bankRelationship_id: string;
  VerifyMicroDepositsRequestCreate:
    components.VerifyMicroDepositsRequestCreate$Outbound;
};

/** @internal */
export const BankRelationshipsVerifyMicroDepositsRequest$outboundSchema:
  z.ZodType<
    BankRelationshipsVerifyMicroDepositsRequest$Outbound,
    z.ZodTypeDef,
    BankRelationshipsVerifyMicroDepositsRequest
  > = z.object({
    accountId: z.string(),
    bankRelationshipId: z.string(),
    verifyMicroDepositsRequestCreate:
      components.VerifyMicroDepositsRequestCreate$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      bankRelationshipId: "bankRelationship_id",
      verifyMicroDepositsRequestCreate: "VerifyMicroDepositsRequestCreate",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BankRelationshipsVerifyMicroDepositsRequest$ {
  /** @deprecated use `BankRelationshipsVerifyMicroDepositsRequest$inboundSchema` instead. */
  export const inboundSchema =
    BankRelationshipsVerifyMicroDepositsRequest$inboundSchema;
  /** @deprecated use `BankRelationshipsVerifyMicroDepositsRequest$outboundSchema` instead. */
  export const outboundSchema =
    BankRelationshipsVerifyMicroDepositsRequest$outboundSchema;
  /** @deprecated use `BankRelationshipsVerifyMicroDepositsRequest$Outbound` instead. */
  export type Outbound = BankRelationshipsVerifyMicroDepositsRequest$Outbound;
}

/** @internal */
export const BankRelationshipsVerifyMicroDepositsResponse$inboundSchema:
  z.ZodType<
    BankRelationshipsVerifyMicroDepositsResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    BankRelationship: components.BankRelationship$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "BankRelationship": "bankRelationship",
      "Status": "status",
    });
  });

/** @internal */
export type BankRelationshipsVerifyMicroDepositsResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  BankRelationship?: components.BankRelationship$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const BankRelationshipsVerifyMicroDepositsResponse$outboundSchema:
  z.ZodType<
    BankRelationshipsVerifyMicroDepositsResponse$Outbound,
    z.ZodTypeDef,
    BankRelationshipsVerifyMicroDepositsResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    bankRelationship: components.BankRelationship$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      bankRelationship: "BankRelationship",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BankRelationshipsVerifyMicroDepositsResponse$ {
  /** @deprecated use `BankRelationshipsVerifyMicroDepositsResponse$inboundSchema` instead. */
  export const inboundSchema =
    BankRelationshipsVerifyMicroDepositsResponse$inboundSchema;
  /** @deprecated use `BankRelationshipsVerifyMicroDepositsResponse$outboundSchema` instead. */
  export const outboundSchema =
    BankRelationshipsVerifyMicroDepositsResponse$outboundSchema;
  /** @deprecated use `BankRelationshipsVerifyMicroDepositsResponse$Outbound` instead. */
  export type Outbound = BankRelationshipsVerifyMicroDepositsResponse$Outbound;
}
