// @apexfintechsolutions/ascend-sdk/src/models/operations/basketordersserviceaddorders.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type BasketOrdersServiceAddOrdersRequest = {
  /**
   * The correspondent id.
   */
  correspondentId: string;
  /**
   * The basket id.
   */
  basketId: string;
  addOrdersRequestCreate: components.AddOrdersRequestCreate;
};

export type BasketOrdersServiceAddOrdersResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  basket?: components.Basket | undefined;
  /**
   * INVALID_ARGUMENT: There was an issue with one or more fields in the request.  The message field will contain details about which field failed validation and why.
   *
   * @remarks
   * FAILED_PRECONDITION: The requested basket has already been submitted.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const BasketOrdersServiceAddOrdersRequest$inboundSchema: z.ZodType<
  BasketOrdersServiceAddOrdersRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  correspondent_id: z.string(),
  basket_id: z.string(),
  AddOrdersRequestCreate: components.AddOrdersRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "correspondent_id": "correspondentId",
    "basket_id": "basketId",
    "AddOrdersRequestCreate": "addOrdersRequestCreate",
  });
});

/** @internal */
export type BasketOrdersServiceAddOrdersRequest$Outbound = {
  correspondent_id: string;
  basket_id: string;
  AddOrdersRequestCreate: components.AddOrdersRequestCreate$Outbound;
};

/** @internal */
export const BasketOrdersServiceAddOrdersRequest$outboundSchema: z.ZodType<
  BasketOrdersServiceAddOrdersRequest$Outbound,
  z.ZodTypeDef,
  BasketOrdersServiceAddOrdersRequest
> = z.object({
  correspondentId: z.string(),
  basketId: z.string(),
  addOrdersRequestCreate: components.AddOrdersRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    correspondentId: "correspondent_id",
    basketId: "basket_id",
    addOrdersRequestCreate: "AddOrdersRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrdersServiceAddOrdersRequest$ {
  /** @deprecated use `BasketOrdersServiceAddOrdersRequest$inboundSchema` instead. */
  export const inboundSchema =
    BasketOrdersServiceAddOrdersRequest$inboundSchema;
  /** @deprecated use `BasketOrdersServiceAddOrdersRequest$outboundSchema` instead. */
  export const outboundSchema =
    BasketOrdersServiceAddOrdersRequest$outboundSchema;
  /** @deprecated use `BasketOrdersServiceAddOrdersRequest$Outbound` instead. */
  export type Outbound = BasketOrdersServiceAddOrdersRequest$Outbound;
}

/** @internal */
export const BasketOrdersServiceAddOrdersResponse$inboundSchema: z.ZodType<
  BasketOrdersServiceAddOrdersResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  Basket: components.Basket$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "Basket": "basket",
    "Status": "status",
  });
});

/** @internal */
export type BasketOrdersServiceAddOrdersResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  Basket?: components.Basket$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const BasketOrdersServiceAddOrdersResponse$outboundSchema: z.ZodType<
  BasketOrdersServiceAddOrdersResponse$Outbound,
  z.ZodTypeDef,
  BasketOrdersServiceAddOrdersResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  basket: components.Basket$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    basket: "Basket",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrdersServiceAddOrdersResponse$ {
  /** @deprecated use `BasketOrdersServiceAddOrdersResponse$inboundSchema` instead. */
  export const inboundSchema =
    BasketOrdersServiceAddOrdersResponse$inboundSchema;
  /** @deprecated use `BasketOrdersServiceAddOrdersResponse$outboundSchema` instead. */
  export const outboundSchema =
    BasketOrdersServiceAddOrdersResponse$outboundSchema;
  /** @deprecated use `BasketOrdersServiceAddOrdersResponse$Outbound` instead. */
  export type Outbound = BasketOrdersServiceAddOrdersResponse$Outbound;
}
