// @apexfintechsolutions/ascend-sdk/src/models/operations/basketordersservicelistbasketorders.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type BasketOrdersServiceListBasketOrdersRequest = {
  /**
   * The correspondent id.
   */
  correspondentId: string;
  /**
   * The basket id.
   */
  basketId: string;
  /**
   * The maximum number of basket orders to return. The service may return fewer than this value. If unspecified, at most 1000 basket orders will be returned. The maximum value is 1000; values above 1000 will be coerced to 1000.
   */
  pageSize?: number | undefined;
  /**
   * A page token, received from a previous `ListBasketOrders` call. Provide this to retrieve the subsequent page. When paginating, all other parameters provided to `ListBasketOrders` must match the call that provided the page token.
   */
  pageToken?: string | undefined;
};

export type BasketOrdersServiceListBasketOrdersResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  listBasketOrdersResponse?: components.ListBasketOrdersResponse | undefined;
  /**
   * INVALID_ARGUMENT: The correspondent_id or the basket_id could not be determined for the request.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const BasketOrdersServiceListBasketOrdersRequest$inboundSchema:
  z.ZodType<BasketOrdersServiceListBasketOrdersRequest, z.ZodTypeDef, unknown> =
    z.object({
      correspondent_id: z.string(),
      basket_id: z.string(),
      page_size: z.number().int().optional(),
      page_token: z.string().optional(),
    }).transform((v) => {
      return remap$(v, {
        "correspondent_id": "correspondentId",
        "basket_id": "basketId",
        "page_size": "pageSize",
        "page_token": "pageToken",
      });
    });

/** @internal */
export type BasketOrdersServiceListBasketOrdersRequest$Outbound = {
  correspondent_id: string;
  basket_id: string;
  page_size?: number | undefined;
  page_token?: string | undefined;
};

/** @internal */
export const BasketOrdersServiceListBasketOrdersRequest$outboundSchema:
  z.ZodType<
    BasketOrdersServiceListBasketOrdersRequest$Outbound,
    z.ZodTypeDef,
    BasketOrdersServiceListBasketOrdersRequest
  > = z.object({
    correspondentId: z.string(),
    basketId: z.string(),
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
  }).transform((v) => {
    return remap$(v, {
      correspondentId: "correspondent_id",
      basketId: "basket_id",
      pageSize: "page_size",
      pageToken: "page_token",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrdersServiceListBasketOrdersRequest$ {
  /** @deprecated use `BasketOrdersServiceListBasketOrdersRequest$inboundSchema` instead. */
  export const inboundSchema =
    BasketOrdersServiceListBasketOrdersRequest$inboundSchema;
  /** @deprecated use `BasketOrdersServiceListBasketOrdersRequest$outboundSchema` instead. */
  export const outboundSchema =
    BasketOrdersServiceListBasketOrdersRequest$outboundSchema;
  /** @deprecated use `BasketOrdersServiceListBasketOrdersRequest$Outbound` instead. */
  export type Outbound = BasketOrdersServiceListBasketOrdersRequest$Outbound;
}

/** @internal */
export const BasketOrdersServiceListBasketOrdersResponse$inboundSchema:
  z.ZodType<
    BasketOrdersServiceListBasketOrdersResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListBasketOrdersResponse: components.ListBasketOrdersResponse$inboundSchema
      .optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "ListBasketOrdersResponse": "listBasketOrdersResponse",
      "Status": "status",
    });
  });

/** @internal */
export type BasketOrdersServiceListBasketOrdersResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  ListBasketOrdersResponse?:
    | components.ListBasketOrdersResponse$Outbound
    | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const BasketOrdersServiceListBasketOrdersResponse$outboundSchema:
  z.ZodType<
    BasketOrdersServiceListBasketOrdersResponse$Outbound,
    z.ZodTypeDef,
    BasketOrdersServiceListBasketOrdersResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listBasketOrdersResponse: components.ListBasketOrdersResponse$outboundSchema
      .optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      listBasketOrdersResponse: "ListBasketOrdersResponse",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrdersServiceListBasketOrdersResponse$ {
  /** @deprecated use `BasketOrdersServiceListBasketOrdersResponse$inboundSchema` instead. */
  export const inboundSchema =
    BasketOrdersServiceListBasketOrdersResponse$inboundSchema;
  /** @deprecated use `BasketOrdersServiceListBasketOrdersResponse$outboundSchema` instead. */
  export const outboundSchema =
    BasketOrdersServiceListBasketOrdersResponse$outboundSchema;
  /** @deprecated use `BasketOrdersServiceListBasketOrdersResponse$Outbound` instead. */
  export type Outbound = BasketOrdersServiceListBasketOrdersResponse$Outbound;
}
