// @apexfintechsolutions/ascend-sdk/src/models/operations/basketordersservicegetbasket.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type BasketOrdersServiceGetBasketRequest = {
  /**
   * The correspondent id.
   */
  correspondentId: string;
  /**
   * The basket id.
   */
  basketId: string;
};

export type BasketOrdersServiceGetBasketResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  basket?: components.Basket | undefined;
  /**
   * INVALID_ARGUMENT: The correspondent_id or the basket_id could not be determined for the request.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const BasketOrdersServiceGetBasketRequest$inboundSchema: z.ZodType<
  BasketOrdersServiceGetBasketRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  correspondent_id: z.string(),
  basket_id: z.string(),
}).transform((v) => {
  return remap$(v, {
    "correspondent_id": "correspondentId",
    "basket_id": "basketId",
  });
});

/** @internal */
export type BasketOrdersServiceGetBasketRequest$Outbound = {
  correspondent_id: string;
  basket_id: string;
};

/** @internal */
export const BasketOrdersServiceGetBasketRequest$outboundSchema: z.ZodType<
  BasketOrdersServiceGetBasketRequest$Outbound,
  z.ZodTypeDef,
  BasketOrdersServiceGetBasketRequest
> = z.object({
  correspondentId: z.string(),
  basketId: z.string(),
}).transform((v) => {
  return remap$(v, {
    correspondentId: "correspondent_id",
    basketId: "basket_id",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrdersServiceGetBasketRequest$ {
  /** @deprecated use `BasketOrdersServiceGetBasketRequest$inboundSchema` instead. */
  export const inboundSchema =
    BasketOrdersServiceGetBasketRequest$inboundSchema;
  /** @deprecated use `BasketOrdersServiceGetBasketRequest$outboundSchema` instead. */
  export const outboundSchema =
    BasketOrdersServiceGetBasketRequest$outboundSchema;
  /** @deprecated use `BasketOrdersServiceGetBasketRequest$Outbound` instead. */
  export type Outbound = BasketOrdersServiceGetBasketRequest$Outbound;
}

/** @internal */
export const BasketOrdersServiceGetBasketResponse$inboundSchema: z.ZodType<
  BasketOrdersServiceGetBasketResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  Basket: components.Basket$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "Basket": "basket",
    "Status": "status",
  });
});

/** @internal */
export type BasketOrdersServiceGetBasketResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  Basket?: components.Basket$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const BasketOrdersServiceGetBasketResponse$outboundSchema: z.ZodType<
  BasketOrdersServiceGetBasketResponse$Outbound,
  z.ZodTypeDef,
  BasketOrdersServiceGetBasketResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  basket: components.Basket$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    basket: "Basket",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace BasketOrdersServiceGetBasketResponse$ {
  /** @deprecated use `BasketOrdersServiceGetBasketResponse$inboundSchema` instead. */
  export const inboundSchema =
    BasketOrdersServiceGetBasketResponse$inboundSchema;
  /** @deprecated use `BasketOrdersServiceGetBasketResponse$outboundSchema` instead. */
  export const outboundSchema =
    BasketOrdersServiceGetBasketResponse$outboundSchema;
  /** @deprecated use `BasketOrdersServiceGetBasketResponse$Outbound` instead. */
  export type Outbound = BasketOrdersServiceGetBasketResponse$Outbound;
}
