// @apexfintechsolutions/ascend-sdk/src/models/operations/cashjournalscancelcashjournal.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type CashJournalsCancelCashJournalRequest = {
  /**
   * The cashJournal id.
   */
  cashJournalId: string;
  cancelCashJournalRequestCreate: components.CancelCashJournalRequestCreate;
};

export type CashJournalsCancelCashJournalResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  cashJournal?: components.CashJournal | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const CashJournalsCancelCashJournalRequest$inboundSchema: z.ZodType<
  CashJournalsCancelCashJournalRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  cashJournal_id: z.string(),
  CancelCashJournalRequestCreate:
    components.CancelCashJournalRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "cashJournal_id": "cashJournalId",
    "CancelCashJournalRequestCreate": "cancelCashJournalRequestCreate",
  });
});

/** @internal */
export type CashJournalsCancelCashJournalRequest$Outbound = {
  cashJournal_id: string;
  CancelCashJournalRequestCreate:
    components.CancelCashJournalRequestCreate$Outbound;
};

/** @internal */
export const CashJournalsCancelCashJournalRequest$outboundSchema: z.ZodType<
  CashJournalsCancelCashJournalRequest$Outbound,
  z.ZodTypeDef,
  CashJournalsCancelCashJournalRequest
> = z.object({
  cashJournalId: z.string(),
  cancelCashJournalRequestCreate:
    components.CancelCashJournalRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    cashJournalId: "cashJournal_id",
    cancelCashJournalRequestCreate: "CancelCashJournalRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CashJournalsCancelCashJournalRequest$ {
  /** @deprecated use `CashJournalsCancelCashJournalRequest$inboundSchema` instead. */
  export const inboundSchema =
    CashJournalsCancelCashJournalRequest$inboundSchema;
  /** @deprecated use `CashJournalsCancelCashJournalRequest$outboundSchema` instead. */
  export const outboundSchema =
    CashJournalsCancelCashJournalRequest$outboundSchema;
  /** @deprecated use `CashJournalsCancelCashJournalRequest$Outbound` instead. */
  export type Outbound = CashJournalsCancelCashJournalRequest$Outbound;
}

/** @internal */
export const CashJournalsCancelCashJournalResponse$inboundSchema: z.ZodType<
  CashJournalsCancelCashJournalResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  CashJournal: components.CashJournal$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "CashJournal": "cashJournal",
    "Status": "status",
  });
});

/** @internal */
export type CashJournalsCancelCashJournalResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  CashJournal?: components.CashJournal$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const CashJournalsCancelCashJournalResponse$outboundSchema: z.ZodType<
  CashJournalsCancelCashJournalResponse$Outbound,
  z.ZodTypeDef,
  CashJournalsCancelCashJournalResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  cashJournal: components.CashJournal$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    cashJournal: "CashJournal",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CashJournalsCancelCashJournalResponse$ {
  /** @deprecated use `CashJournalsCancelCashJournalResponse$inboundSchema` instead. */
  export const inboundSchema =
    CashJournalsCancelCashJournalResponse$inboundSchema;
  /** @deprecated use `CashJournalsCancelCashJournalResponse$outboundSchema` instead. */
  export const outboundSchema =
    CashJournalsCancelCashJournalResponse$outboundSchema;
  /** @deprecated use `CashJournalsCancelCashJournalResponse$Outbound` instead. */
  export type Outbound = CashJournalsCancelCashJournalResponse$Outbound;
}
