// @apexfintechsolutions/ascend-sdk/src/models/operations/cashjournalscreatecashjournal.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type CashJournalsCreateCashJournalResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  cashJournal?: components.CashJournal | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const CashJournalsCreateCashJournalResponse$inboundSchema: z.ZodType<
  CashJournalsCreateCashJournalResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  CashJournal: components.CashJournal$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "CashJournal": "cashJournal",
    "Status": "status",
  });
});

/** @internal */
export type CashJournalsCreateCashJournalResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  CashJournal?: components.CashJournal$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const CashJournalsCreateCashJournalResponse$outboundSchema: z.ZodType<
  CashJournalsCreateCashJournalResponse$Outbound,
  z.ZodTypeDef,
  CashJournalsCreateCashJournalResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  cashJournal: components.CashJournal$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    cashJournal: "CashJournal",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CashJournalsCreateCashJournalResponse$ {
  /** @deprecated use `CashJournalsCreateCashJournalResponse$inboundSchema` instead. */
  export const inboundSchema =
    CashJournalsCreateCashJournalResponse$inboundSchema;
  /** @deprecated use `CashJournalsCreateCashJournalResponse$outboundSchema` instead. */
  export const outboundSchema =
    CashJournalsCreateCashJournalResponse$outboundSchema;
  /** @deprecated use `CashJournalsCreateCashJournalResponse$Outbound` instead. */
  export type Outbound = CashJournalsCreateCashJournalResponse$Outbound;
}
