// @apexfintechsolutions/ascend-sdk/src/models/operations/cashjournalsforcerejectcashjournal.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type CashJournalsForceRejectCashJournalRequest = {
  /**
   * The cashJournal id.
   */
  cashJournalId: string;
  forceRejectCashJournalRequestCreate:
    components.ForceRejectCashJournalRequestCreate;
};

export type CashJournalsForceRejectCashJournalResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  cashJournal?: components.CashJournal | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const CashJournalsForceRejectCashJournalRequest$inboundSchema: z.ZodType<
  CashJournalsForceRejectCashJournalRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  cashJournal_id: z.string(),
  ForceRejectCashJournalRequestCreate:
    components.ForceRejectCashJournalRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "cashJournal_id": "cashJournalId",
    "ForceRejectCashJournalRequestCreate":
      "forceRejectCashJournalRequestCreate",
  });
});

/** @internal */
export type CashJournalsForceRejectCashJournalRequest$Outbound = {
  cashJournal_id: string;
  ForceRejectCashJournalRequestCreate:
    components.ForceRejectCashJournalRequestCreate$Outbound;
};

/** @internal */
export const CashJournalsForceRejectCashJournalRequest$outboundSchema:
  z.ZodType<
    CashJournalsForceRejectCashJournalRequest$Outbound,
    z.ZodTypeDef,
    CashJournalsForceRejectCashJournalRequest
  > = z.object({
    cashJournalId: z.string(),
    forceRejectCashJournalRequestCreate:
      components.ForceRejectCashJournalRequestCreate$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      cashJournalId: "cashJournal_id",
      forceRejectCashJournalRequestCreate:
        "ForceRejectCashJournalRequestCreate",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CashJournalsForceRejectCashJournalRequest$ {
  /** @deprecated use `CashJournalsForceRejectCashJournalRequest$inboundSchema` instead. */
  export const inboundSchema =
    CashJournalsForceRejectCashJournalRequest$inboundSchema;
  /** @deprecated use `CashJournalsForceRejectCashJournalRequest$outboundSchema` instead. */
  export const outboundSchema =
    CashJournalsForceRejectCashJournalRequest$outboundSchema;
  /** @deprecated use `CashJournalsForceRejectCashJournalRequest$Outbound` instead. */
  export type Outbound = CashJournalsForceRejectCashJournalRequest$Outbound;
}

/** @internal */
export const CashJournalsForceRejectCashJournalResponse$inboundSchema:
  z.ZodType<CashJournalsForceRejectCashJournalResponse, z.ZodTypeDef, unknown> =
    z.object({
      HttpMeta: components.HTTPMetadata$inboundSchema,
      CashJournal: components.CashJournal$inboundSchema.optional(),
      Status: components.Status$inboundSchema.optional(),
    }).transform((v) => {
      return remap$(v, {
        "HttpMeta": "httpMeta",
        "CashJournal": "cashJournal",
        "Status": "status",
      });
    });

/** @internal */
export type CashJournalsForceRejectCashJournalResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  CashJournal?: components.CashJournal$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const CashJournalsForceRejectCashJournalResponse$outboundSchema:
  z.ZodType<
    CashJournalsForceRejectCashJournalResponse$Outbound,
    z.ZodTypeDef,
    CashJournalsForceRejectCashJournalResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    cashJournal: components.CashJournal$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      cashJournal: "CashJournal",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CashJournalsForceRejectCashJournalResponse$ {
  /** @deprecated use `CashJournalsForceRejectCashJournalResponse$inboundSchema` instead. */
  export const inboundSchema =
    CashJournalsForceRejectCashJournalResponse$inboundSchema;
  /** @deprecated use `CashJournalsForceRejectCashJournalResponse$outboundSchema` instead. */
  export const outboundSchema =
    CashJournalsForceRejectCashJournalResponse$outboundSchema;
  /** @deprecated use `CashJournalsForceRejectCashJournalResponse$Outbound` instead. */
  export type Outbound = CashJournalsForceRejectCashJournalResponse$Outbound;
}
