// @apexfintechsolutions/ascend-sdk/src/models/operations/cashjournalsforceapprovecashjournal.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type CashJournalsForceApproveCashJournalRequest = {
  /**
   * The cashJournal id.
   */
  cashJournalId: string;
  forceApproveCashJournalRequestCreate:
    components.ForceApproveCashJournalRequestCreate;
};

export type CashJournalsForceApproveCashJournalResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  cashJournal?: components.CashJournal | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const CashJournalsForceApproveCashJournalRequest$inboundSchema:
  z.ZodType<CashJournalsForceApproveCashJournalRequest, z.ZodTypeDef, unknown> =
    z.object({
      cashJournal_id: z.string(),
      ForceApproveCashJournalRequestCreate:
        components.ForceApproveCashJournalRequestCreate$inboundSchema,
    }).transform((v) => {
      return remap$(v, {
        "cashJournal_id": "cashJournalId",
        "ForceApproveCashJournalRequestCreate":
          "forceApproveCashJournalRequestCreate",
      });
    });

/** @internal */
export type CashJournalsForceApproveCashJournalRequest$Outbound = {
  cashJournal_id: string;
  ForceApproveCashJournalRequestCreate:
    components.ForceApproveCashJournalRequestCreate$Outbound;
};

/** @internal */
export const CashJournalsForceApproveCashJournalRequest$outboundSchema:
  z.ZodType<
    CashJournalsForceApproveCashJournalRequest$Outbound,
    z.ZodTypeDef,
    CashJournalsForceApproveCashJournalRequest
  > = z.object({
    cashJournalId: z.string(),
    forceApproveCashJournalRequestCreate:
      components.ForceApproveCashJournalRequestCreate$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      cashJournalId: "cashJournal_id",
      forceApproveCashJournalRequestCreate:
        "ForceApproveCashJournalRequestCreate",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CashJournalsForceApproveCashJournalRequest$ {
  /** @deprecated use `CashJournalsForceApproveCashJournalRequest$inboundSchema` instead. */
  export const inboundSchema =
    CashJournalsForceApproveCashJournalRequest$inboundSchema;
  /** @deprecated use `CashJournalsForceApproveCashJournalRequest$outboundSchema` instead. */
  export const outboundSchema =
    CashJournalsForceApproveCashJournalRequest$outboundSchema;
  /** @deprecated use `CashJournalsForceApproveCashJournalRequest$Outbound` instead. */
  export type Outbound = CashJournalsForceApproveCashJournalRequest$Outbound;
}

/** @internal */
export const CashJournalsForceApproveCashJournalResponse$inboundSchema:
  z.ZodType<
    CashJournalsForceApproveCashJournalResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    CashJournal: components.CashJournal$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "CashJournal": "cashJournal",
      "Status": "status",
    });
  });

/** @internal */
export type CashJournalsForceApproveCashJournalResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  CashJournal?: components.CashJournal$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const CashJournalsForceApproveCashJournalResponse$outboundSchema:
  z.ZodType<
    CashJournalsForceApproveCashJournalResponse$Outbound,
    z.ZodTypeDef,
    CashJournalsForceApproveCashJournalResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    cashJournal: components.CashJournal$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      cashJournal: "CashJournal",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CashJournalsForceApproveCashJournalResponse$ {
  /** @deprecated use `CashJournalsForceApproveCashJournalResponse$inboundSchema` instead. */
  export const inboundSchema =
    CashJournalsForceApproveCashJournalResponse$inboundSchema;
  /** @deprecated use `CashJournalsForceApproveCashJournalResponse$outboundSchema` instead. */
  export const outboundSchema =
    CashJournalsForceApproveCashJournalResponse$outboundSchema;
  /** @deprecated use `CashJournalsForceApproveCashJournalResponse$Outbound` instead. */
  export type Outbound = CashJournalsForceApproveCashJournalResponse$Outbound;
}
