// @apexfintechsolutions/ascend-sdk/src/models/operations/creditscancelcredit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type CreditsCancelCreditRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The credit id.
   */
  creditId: string;
  cancelCreditRequestCreate: components.CancelCreditRequestCreate;
};

export type CreditsCancelCreditResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  transfersCredit?: components.TransfersCredit | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const CreditsCancelCreditRequest$inboundSchema: z.ZodType<
  CreditsCancelCreditRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  credit_id: z.string(),
  CancelCreditRequestCreate: components.CancelCreditRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "credit_id": "creditId",
    "CancelCreditRequestCreate": "cancelCreditRequestCreate",
  });
});

/** @internal */
export type CreditsCancelCreditRequest$Outbound = {
  account_id: string;
  credit_id: string;
  CancelCreditRequestCreate: components.CancelCreditRequestCreate$Outbound;
};

/** @internal */
export const CreditsCancelCreditRequest$outboundSchema: z.ZodType<
  CreditsCancelCreditRequest$Outbound,
  z.ZodTypeDef,
  CreditsCancelCreditRequest
> = z.object({
  accountId: z.string(),
  creditId: z.string(),
  cancelCreditRequestCreate:
    components.CancelCreditRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    creditId: "credit_id",
    cancelCreditRequestCreate: "CancelCreditRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CreditsCancelCreditRequest$ {
  /** @deprecated use `CreditsCancelCreditRequest$inboundSchema` instead. */
  export const inboundSchema = CreditsCancelCreditRequest$inboundSchema;
  /** @deprecated use `CreditsCancelCreditRequest$outboundSchema` instead. */
  export const outboundSchema = CreditsCancelCreditRequest$outboundSchema;
  /** @deprecated use `CreditsCancelCreditRequest$Outbound` instead. */
  export type Outbound = CreditsCancelCreditRequest$Outbound;
}

/** @internal */
export const CreditsCancelCreditResponse$inboundSchema: z.ZodType<
  CreditsCancelCreditResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  TransfersCredit: components.TransfersCredit$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "TransfersCredit": "transfersCredit",
    "Status": "status",
  });
});

/** @internal */
export type CreditsCancelCreditResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  TransfersCredit?: components.TransfersCredit$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const CreditsCancelCreditResponse$outboundSchema: z.ZodType<
  CreditsCancelCreditResponse$Outbound,
  z.ZodTypeDef,
  CreditsCancelCreditResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  transfersCredit: components.TransfersCredit$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    transfersCredit: "TransfersCredit",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace CreditsCancelCreditResponse$ {
  /** @deprecated use `CreditsCancelCreditResponse$inboundSchema` instead. */
  export const inboundSchema = CreditsCancelCreditResponse$inboundSchema;
  /** @deprecated use `CreditsCancelCreditResponse$outboundSchema` instead. */
  export const outboundSchema = CreditsCancelCreditResponse$outboundSchema;
  /** @deprecated use `CreditsCancelCreditResponse$Outbound` instead. */
  export type Outbound = CreditsCancelCreditResponse$Outbound;
}
