// @apexfintechsolutions/ascend-sdk/src/models/operations/ictdepositscreateictdeposit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type IctDepositsCreateIctDepositRequest = {
  /**
   * The account id.
   */
  accountId: string;
  ictDepositCreate: components.IctDepositCreate;
};

export type IctDepositsCreateIctDepositResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  ictDeposit?: components.IctDeposit | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const IctDepositsCreateIctDepositRequest$inboundSchema: z.ZodType<
  IctDepositsCreateIctDepositRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  IctDepositCreate: components.IctDepositCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "IctDepositCreate": "ictDepositCreate",
  });
});

/** @internal */
export type IctDepositsCreateIctDepositRequest$Outbound = {
  account_id: string;
  IctDepositCreate: components.IctDepositCreate$Outbound;
};

/** @internal */
export const IctDepositsCreateIctDepositRequest$outboundSchema: z.ZodType<
  IctDepositsCreateIctDepositRequest$Outbound,
  z.ZodTypeDef,
  IctDepositsCreateIctDepositRequest
> = z.object({
  accountId: z.string(),
  ictDepositCreate: components.IctDepositCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    ictDepositCreate: "IctDepositCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctDepositsCreateIctDepositRequest$ {
  /** @deprecated use `IctDepositsCreateIctDepositRequest$inboundSchema` instead. */
  export const inboundSchema = IctDepositsCreateIctDepositRequest$inboundSchema;
  /** @deprecated use `IctDepositsCreateIctDepositRequest$outboundSchema` instead. */
  export const outboundSchema =
    IctDepositsCreateIctDepositRequest$outboundSchema;
  /** @deprecated use `IctDepositsCreateIctDepositRequest$Outbound` instead. */
  export type Outbound = IctDepositsCreateIctDepositRequest$Outbound;
}

/** @internal */
export const IctDepositsCreateIctDepositResponse$inboundSchema: z.ZodType<
  IctDepositsCreateIctDepositResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  IctDeposit: components.IctDeposit$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "IctDeposit": "ictDeposit",
    "Status": "status",
  });
});

/** @internal */
export type IctDepositsCreateIctDepositResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  IctDeposit?: components.IctDeposit$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const IctDepositsCreateIctDepositResponse$outboundSchema: z.ZodType<
  IctDepositsCreateIctDepositResponse$Outbound,
  z.ZodTypeDef,
  IctDepositsCreateIctDepositResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  ictDeposit: components.IctDeposit$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    ictDeposit: "IctDeposit",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctDepositsCreateIctDepositResponse$ {
  /** @deprecated use `IctDepositsCreateIctDepositResponse$inboundSchema` instead. */
  export const inboundSchema =
    IctDepositsCreateIctDepositResponse$inboundSchema;
  /** @deprecated use `IctDepositsCreateIctDepositResponse$outboundSchema` instead. */
  export const outboundSchema =
    IctDepositsCreateIctDepositResponse$outboundSchema;
  /** @deprecated use `IctDepositsCreateIctDepositResponse$Outbound` instead. */
  export type Outbound = IctDepositsCreateIctDepositResponse$Outbound;
}
