// @apexfintechsolutions/ascend-sdk/src/models/operations/ictdepositscancelictdeposit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type IctDepositsCancelIctDepositRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The ictDeposit id.
   */
  ictDepositId: string;
  cancelIctDepositRequestCreate: components.CancelIctDepositRequestCreate;
};

export type IctDepositsCancelIctDepositResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  ictDeposit?: components.IctDeposit | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const IctDepositsCancelIctDepositRequest$inboundSchema: z.ZodType<
  IctDepositsCancelIctDepositRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  ictDeposit_id: z.string(),
  CancelIctDepositRequestCreate:
    components.CancelIctDepositRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "ictDeposit_id": "ictDepositId",
    "CancelIctDepositRequestCreate": "cancelIctDepositRequestCreate",
  });
});

/** @internal */
export type IctDepositsCancelIctDepositRequest$Outbound = {
  account_id: string;
  ictDeposit_id: string;
  CancelIctDepositRequestCreate:
    components.CancelIctDepositRequestCreate$Outbound;
};

/** @internal */
export const IctDepositsCancelIctDepositRequest$outboundSchema: z.ZodType<
  IctDepositsCancelIctDepositRequest$Outbound,
  z.ZodTypeDef,
  IctDepositsCancelIctDepositRequest
> = z.object({
  accountId: z.string(),
  ictDepositId: z.string(),
  cancelIctDepositRequestCreate:
    components.CancelIctDepositRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    ictDepositId: "ictDeposit_id",
    cancelIctDepositRequestCreate: "CancelIctDepositRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctDepositsCancelIctDepositRequest$ {
  /** @deprecated use `IctDepositsCancelIctDepositRequest$inboundSchema` instead. */
  export const inboundSchema = IctDepositsCancelIctDepositRequest$inboundSchema;
  /** @deprecated use `IctDepositsCancelIctDepositRequest$outboundSchema` instead. */
  export const outboundSchema =
    IctDepositsCancelIctDepositRequest$outboundSchema;
  /** @deprecated use `IctDepositsCancelIctDepositRequest$Outbound` instead. */
  export type Outbound = IctDepositsCancelIctDepositRequest$Outbound;
}

/** @internal */
export const IctDepositsCancelIctDepositResponse$inboundSchema: z.ZodType<
  IctDepositsCancelIctDepositResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  IctDeposit: components.IctDeposit$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "IctDeposit": "ictDeposit",
    "Status": "status",
  });
});

/** @internal */
export type IctDepositsCancelIctDepositResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  IctDeposit?: components.IctDeposit$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const IctDepositsCancelIctDepositResponse$outboundSchema: z.ZodType<
  IctDepositsCancelIctDepositResponse$Outbound,
  z.ZodTypeDef,
  IctDepositsCancelIctDepositResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  ictDeposit: components.IctDeposit$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    ictDeposit: "IctDeposit",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctDepositsCancelIctDepositResponse$ {
  /** @deprecated use `IctDepositsCancelIctDepositResponse$inboundSchema` instead. */
  export const inboundSchema =
    IctDepositsCancelIctDepositResponse$inboundSchema;
  /** @deprecated use `IctDepositsCancelIctDepositResponse$outboundSchema` instead. */
  export const outboundSchema =
    IctDepositsCancelIctDepositResponse$outboundSchema;
  /** @deprecated use `IctDepositsCancelIctDepositResponse$Outbound` instead. */
  export type Outbound = IctDepositsCancelIctDepositResponse$Outbound;
}
