// @apexfintechsolutions/ascend-sdk/src/models/operations/ictdepositsforceapproveictdeposit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type IctDepositsForceApproveIctDepositRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The ictDeposit id.
   */
  ictDepositId: string;
  forceApproveIctDepositRequestCreate:
    components.ForceApproveIctDepositRequestCreate;
};

export type IctDepositsForceApproveIctDepositResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  ictDeposit?: components.IctDeposit | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const IctDepositsForceApproveIctDepositRequest$inboundSchema: z.ZodType<
  IctDepositsForceApproveIctDepositRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  ictDeposit_id: z.string(),
  ForceApproveIctDepositRequestCreate:
    components.ForceApproveIctDepositRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "ictDeposit_id": "ictDepositId",
    "ForceApproveIctDepositRequestCreate":
      "forceApproveIctDepositRequestCreate",
  });
});

/** @internal */
export type IctDepositsForceApproveIctDepositRequest$Outbound = {
  account_id: string;
  ictDeposit_id: string;
  ForceApproveIctDepositRequestCreate:
    components.ForceApproveIctDepositRequestCreate$Outbound;
};

/** @internal */
export const IctDepositsForceApproveIctDepositRequest$outboundSchema: z.ZodType<
  IctDepositsForceApproveIctDepositRequest$Outbound,
  z.ZodTypeDef,
  IctDepositsForceApproveIctDepositRequest
> = z.object({
  accountId: z.string(),
  ictDepositId: z.string(),
  forceApproveIctDepositRequestCreate:
    components.ForceApproveIctDepositRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    ictDepositId: "ictDeposit_id",
    forceApproveIctDepositRequestCreate: "ForceApproveIctDepositRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctDepositsForceApproveIctDepositRequest$ {
  /** @deprecated use `IctDepositsForceApproveIctDepositRequest$inboundSchema` instead. */
  export const inboundSchema =
    IctDepositsForceApproveIctDepositRequest$inboundSchema;
  /** @deprecated use `IctDepositsForceApproveIctDepositRequest$outboundSchema` instead. */
  export const outboundSchema =
    IctDepositsForceApproveIctDepositRequest$outboundSchema;
  /** @deprecated use `IctDepositsForceApproveIctDepositRequest$Outbound` instead. */
  export type Outbound = IctDepositsForceApproveIctDepositRequest$Outbound;
}

/** @internal */
export const IctDepositsForceApproveIctDepositResponse$inboundSchema: z.ZodType<
  IctDepositsForceApproveIctDepositResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  IctDeposit: components.IctDeposit$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "IctDeposit": "ictDeposit",
    "Status": "status",
  });
});

/** @internal */
export type IctDepositsForceApproveIctDepositResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  IctDeposit?: components.IctDeposit$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const IctDepositsForceApproveIctDepositResponse$outboundSchema:
  z.ZodType<
    IctDepositsForceApproveIctDepositResponse$Outbound,
    z.ZodTypeDef,
    IctDepositsForceApproveIctDepositResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    ictDeposit: components.IctDeposit$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      ictDeposit: "IctDeposit",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctDepositsForceApproveIctDepositResponse$ {
  /** @deprecated use `IctDepositsForceApproveIctDepositResponse$inboundSchema` instead. */
  export const inboundSchema =
    IctDepositsForceApproveIctDepositResponse$inboundSchema;
  /** @deprecated use `IctDepositsForceApproveIctDepositResponse$outboundSchema` instead. */
  export const outboundSchema =
    IctDepositsForceApproveIctDepositResponse$outboundSchema;
  /** @deprecated use `IctDepositsForceApproveIctDepositResponse$Outbound` instead. */
  export type Outbound = IctDepositsForceApproveIctDepositResponse$Outbound;
}
