// @apexfintechsolutions/ascend-sdk/src/models/operations/ictdepositsgetictdeposit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type IctDepositsGetIctDepositRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The ictDeposit id.
   */
  ictDepositId: string;
};

export type IctDepositsGetIctDepositResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  ictDeposit?: components.IctDeposit | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const IctDepositsGetIctDepositRequest$inboundSchema: z.ZodType<
  IctDepositsGetIctDepositRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  ictDeposit_id: z.string(),
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "ictDeposit_id": "ictDepositId",
  });
});

/** @internal */
export type IctDepositsGetIctDepositRequest$Outbound = {
  account_id: string;
  ictDeposit_id: string;
};

/** @internal */
export const IctDepositsGetIctDepositRequest$outboundSchema: z.ZodType<
  IctDepositsGetIctDepositRequest$Outbound,
  z.ZodTypeDef,
  IctDepositsGetIctDepositRequest
> = z.object({
  accountId: z.string(),
  ictDepositId: z.string(),
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    ictDepositId: "ictDeposit_id",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctDepositsGetIctDepositRequest$ {
  /** @deprecated use `IctDepositsGetIctDepositRequest$inboundSchema` instead. */
  export const inboundSchema = IctDepositsGetIctDepositRequest$inboundSchema;
  /** @deprecated use `IctDepositsGetIctDepositRequest$outboundSchema` instead. */
  export const outboundSchema = IctDepositsGetIctDepositRequest$outboundSchema;
  /** @deprecated use `IctDepositsGetIctDepositRequest$Outbound` instead. */
  export type Outbound = IctDepositsGetIctDepositRequest$Outbound;
}

/** @internal */
export const IctDepositsGetIctDepositResponse$inboundSchema: z.ZodType<
  IctDepositsGetIctDepositResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  IctDeposit: components.IctDeposit$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "IctDeposit": "ictDeposit",
    "Status": "status",
  });
});

/** @internal */
export type IctDepositsGetIctDepositResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  IctDeposit?: components.IctDeposit$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const IctDepositsGetIctDepositResponse$outboundSchema: z.ZodType<
  IctDepositsGetIctDepositResponse$Outbound,
  z.ZodTypeDef,
  IctDepositsGetIctDepositResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  ictDeposit: components.IctDeposit$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    ictDeposit: "IctDeposit",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctDepositsGetIctDepositResponse$ {
  /** @deprecated use `IctDepositsGetIctDepositResponse$inboundSchema` instead. */
  export const inboundSchema = IctDepositsGetIctDepositResponse$inboundSchema;
  /** @deprecated use `IctDepositsGetIctDepositResponse$outboundSchema` instead. */
  export const outboundSchema = IctDepositsGetIctDepositResponse$outboundSchema;
  /** @deprecated use `IctDepositsGetIctDepositResponse$Outbound` instead. */
  export type Outbound = IctDepositsGetIctDepositResponse$Outbound;
}
