// @apexfintechsolutions/ascend-sdk/src/models/operations/ictdepositsforcerejectictdeposit.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type IctDepositsForceRejectIctDepositRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The ictDeposit id.
   */
  ictDepositId: string;
  forceRejectIctDepositRequestCreate:
    components.ForceRejectIctDepositRequestCreate;
};

export type IctDepositsForceRejectIctDepositResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  ictDeposit?: components.IctDeposit | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const IctDepositsForceRejectIctDepositRequest$inboundSchema: z.ZodType<
  IctDepositsForceRejectIctDepositRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  ictDeposit_id: z.string(),
  ForceRejectIctDepositRequestCreate:
    components.ForceRejectIctDepositRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "ictDeposit_id": "ictDepositId",
    "ForceRejectIctDepositRequestCreate": "forceRejectIctDepositRequestCreate",
  });
});

/** @internal */
export type IctDepositsForceRejectIctDepositRequest$Outbound = {
  account_id: string;
  ictDeposit_id: string;
  ForceRejectIctDepositRequestCreate:
    components.ForceRejectIctDepositRequestCreate$Outbound;
};

/** @internal */
export const IctDepositsForceRejectIctDepositRequest$outboundSchema: z.ZodType<
  IctDepositsForceRejectIctDepositRequest$Outbound,
  z.ZodTypeDef,
  IctDepositsForceRejectIctDepositRequest
> = z.object({
  accountId: z.string(),
  ictDepositId: z.string(),
  forceRejectIctDepositRequestCreate:
    components.ForceRejectIctDepositRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    ictDepositId: "ictDeposit_id",
    forceRejectIctDepositRequestCreate: "ForceRejectIctDepositRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctDepositsForceRejectIctDepositRequest$ {
  /** @deprecated use `IctDepositsForceRejectIctDepositRequest$inboundSchema` instead. */
  export const inboundSchema =
    IctDepositsForceRejectIctDepositRequest$inboundSchema;
  /** @deprecated use `IctDepositsForceRejectIctDepositRequest$outboundSchema` instead. */
  export const outboundSchema =
    IctDepositsForceRejectIctDepositRequest$outboundSchema;
  /** @deprecated use `IctDepositsForceRejectIctDepositRequest$Outbound` instead. */
  export type Outbound = IctDepositsForceRejectIctDepositRequest$Outbound;
}

/** @internal */
export const IctDepositsForceRejectIctDepositResponse$inboundSchema: z.ZodType<
  IctDepositsForceRejectIctDepositResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  IctDeposit: components.IctDeposit$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "IctDeposit": "ictDeposit",
    "Status": "status",
  });
});

/** @internal */
export type IctDepositsForceRejectIctDepositResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  IctDeposit?: components.IctDeposit$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const IctDepositsForceRejectIctDepositResponse$outboundSchema: z.ZodType<
  IctDepositsForceRejectIctDepositResponse$Outbound,
  z.ZodTypeDef,
  IctDepositsForceRejectIctDepositResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  ictDeposit: components.IctDeposit$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    ictDeposit: "IctDeposit",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctDepositsForceRejectIctDepositResponse$ {
  /** @deprecated use `IctDepositsForceRejectIctDepositResponse$inboundSchema` instead. */
  export const inboundSchema =
    IctDepositsForceRejectIctDepositResponse$inboundSchema;
  /** @deprecated use `IctDepositsForceRejectIctDepositResponse$outboundSchema` instead. */
  export const outboundSchema =
    IctDepositsForceRejectIctDepositResponse$outboundSchema;
  /** @deprecated use `IctDepositsForceRejectIctDepositResponse$Outbound` instead. */
  export type Outbound = IctDepositsForceRejectIctDepositResponse$Outbound;
}
