// @apexfintechsolutions/ascend-sdk/src/models/operations/ictwithdrawalsforceapproveictwithdrawal.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type IctWithdrawalsForceApproveIctWithdrawalRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The ictWithdrawal id.
   */
  ictWithdrawalId: string;
  forceApproveIctWithdrawalRequestCreate:
    components.ForceApproveIctWithdrawalRequestCreate;
};

export type IctWithdrawalsForceApproveIctWithdrawalResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  ictWithdrawal?: components.IctWithdrawal | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const IctWithdrawalsForceApproveIctWithdrawalRequest$inboundSchema:
  z.ZodType<
    IctWithdrawalsForceApproveIctWithdrawalRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    ictWithdrawal_id: z.string(),
    ForceApproveIctWithdrawalRequestCreate:
      components.ForceApproveIctWithdrawalRequestCreate$inboundSchema,
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "ictWithdrawal_id": "ictWithdrawalId",
      "ForceApproveIctWithdrawalRequestCreate":
        "forceApproveIctWithdrawalRequestCreate",
    });
  });

/** @internal */
export type IctWithdrawalsForceApproveIctWithdrawalRequest$Outbound = {
  account_id: string;
  ictWithdrawal_id: string;
  ForceApproveIctWithdrawalRequestCreate:
    components.ForceApproveIctWithdrawalRequestCreate$Outbound;
};

/** @internal */
export const IctWithdrawalsForceApproveIctWithdrawalRequest$outboundSchema:
  z.ZodType<
    IctWithdrawalsForceApproveIctWithdrawalRequest$Outbound,
    z.ZodTypeDef,
    IctWithdrawalsForceApproveIctWithdrawalRequest
  > = z.object({
    accountId: z.string(),
    ictWithdrawalId: z.string(),
    forceApproveIctWithdrawalRequestCreate:
      components.ForceApproveIctWithdrawalRequestCreate$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      ictWithdrawalId: "ictWithdrawal_id",
      forceApproveIctWithdrawalRequestCreate:
        "ForceApproveIctWithdrawalRequestCreate",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctWithdrawalsForceApproveIctWithdrawalRequest$ {
  /** @deprecated use `IctWithdrawalsForceApproveIctWithdrawalRequest$inboundSchema` instead. */
  export const inboundSchema =
    IctWithdrawalsForceApproveIctWithdrawalRequest$inboundSchema;
  /** @deprecated use `IctWithdrawalsForceApproveIctWithdrawalRequest$outboundSchema` instead. */
  export const outboundSchema =
    IctWithdrawalsForceApproveIctWithdrawalRequest$outboundSchema;
  /** @deprecated use `IctWithdrawalsForceApproveIctWithdrawalRequest$Outbound` instead. */
  export type Outbound =
    IctWithdrawalsForceApproveIctWithdrawalRequest$Outbound;
}

/** @internal */
export const IctWithdrawalsForceApproveIctWithdrawalResponse$inboundSchema:
  z.ZodType<
    IctWithdrawalsForceApproveIctWithdrawalResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    IctWithdrawal: components.IctWithdrawal$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "IctWithdrawal": "ictWithdrawal",
      "Status": "status",
    });
  });

/** @internal */
export type IctWithdrawalsForceApproveIctWithdrawalResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  IctWithdrawal?: components.IctWithdrawal$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const IctWithdrawalsForceApproveIctWithdrawalResponse$outboundSchema:
  z.ZodType<
    IctWithdrawalsForceApproveIctWithdrawalResponse$Outbound,
    z.ZodTypeDef,
    IctWithdrawalsForceApproveIctWithdrawalResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    ictWithdrawal: components.IctWithdrawal$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      ictWithdrawal: "IctWithdrawal",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace IctWithdrawalsForceApproveIctWithdrawalResponse$ {
  /** @deprecated use `IctWithdrawalsForceApproveIctWithdrawalResponse$inboundSchema` instead. */
  export const inboundSchema =
    IctWithdrawalsForceApproveIctWithdrawalResponse$inboundSchema;
  /** @deprecated use `IctWithdrawalsForceApproveIctWithdrawalResponse$outboundSchema` instead. */
  export const outboundSchema =
    IctWithdrawalsForceApproveIctWithdrawalResponse$outboundSchema;
  /** @deprecated use `IctWithdrawalsForceApproveIctWithdrawalResponse$Outbound` instead. */
  export type Outbound =
    IctWithdrawalsForceApproveIctWithdrawalResponse$Outbound;
}
