// @apexfintechsolutions/ascend-sdk/src/models/operations/orderservicecancelorder.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type OrderServiceCancelOrderRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The order id.
   */
  orderId: string;
  cancelOrderRequestCreate: components.CancelOrderRequestCreate;
};

export type OrderServiceCancelOrderResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  order?: components.Order | undefined;
  /**
   * INVALID_ARGUMENT: The account_id or the order_id could not be determined for the request.
   *
   * @remarks
   * FAILED_PRECONDITION: The order is not in a cancelable state
   */
  status?: components.Status | undefined;
};

/** @internal */
export const OrderServiceCancelOrderRequest$inboundSchema: z.ZodType<
  OrderServiceCancelOrderRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  account_id: z.string(),
  order_id: z.string(),
  CancelOrderRequestCreate: components.CancelOrderRequestCreate$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "account_id": "accountId",
    "order_id": "orderId",
    "CancelOrderRequestCreate": "cancelOrderRequestCreate",
  });
});

/** @internal */
export type OrderServiceCancelOrderRequest$Outbound = {
  account_id: string;
  order_id: string;
  CancelOrderRequestCreate: components.CancelOrderRequestCreate$Outbound;
};

/** @internal */
export const OrderServiceCancelOrderRequest$outboundSchema: z.ZodType<
  OrderServiceCancelOrderRequest$Outbound,
  z.ZodTypeDef,
  OrderServiceCancelOrderRequest
> = z.object({
  accountId: z.string(),
  orderId: z.string(),
  cancelOrderRequestCreate: components.CancelOrderRequestCreate$outboundSchema,
}).transform((v) => {
  return remap$(v, {
    accountId: "account_id",
    orderId: "order_id",
    cancelOrderRequestCreate: "CancelOrderRequestCreate",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace OrderServiceCancelOrderRequest$ {
  /** @deprecated use `OrderServiceCancelOrderRequest$inboundSchema` instead. */
  export const inboundSchema = OrderServiceCancelOrderRequest$inboundSchema;
  /** @deprecated use `OrderServiceCancelOrderRequest$outboundSchema` instead. */
  export const outboundSchema = OrderServiceCancelOrderRequest$outboundSchema;
  /** @deprecated use `OrderServiceCancelOrderRequest$Outbound` instead. */
  export type Outbound = OrderServiceCancelOrderRequest$Outbound;
}

/** @internal */
export const OrderServiceCancelOrderResponse$inboundSchema: z.ZodType<
  OrderServiceCancelOrderResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  Order: components.Order$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "Order": "order",
    "Status": "status",
  });
});

/** @internal */
export type OrderServiceCancelOrderResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  Order?: components.Order$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const OrderServiceCancelOrderResponse$outboundSchema: z.ZodType<
  OrderServiceCancelOrderResponse$Outbound,
  z.ZodTypeDef,
  OrderServiceCancelOrderResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  order: components.Order$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    order: "Order",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace OrderServiceCancelOrderResponse$ {
  /** @deprecated use `OrderServiceCancelOrderResponse$inboundSchema` instead. */
  export const inboundSchema = OrderServiceCancelOrderResponse$inboundSchema;
  /** @deprecated use `OrderServiceCancelOrderResponse$outboundSchema` instead. */
  export const outboundSchema = OrderServiceCancelOrderResponse$outboundSchema;
  /** @deprecated use `OrderServiceCancelOrderResponse$Outbound` instead. */
  export type Outbound = OrderServiceCancelOrderResponse$Outbound;
}
