// @apexfintechsolutions/ascend-sdk/src/models/operations/retirementconstraintsretrievecashjournalconstraints.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type RetirementConstraintsRetrieveCashJournalConstraintsResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  cashJournalConstraints?: components.CashJournalConstraints | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const RetirementConstraintsRetrieveCashJournalConstraintsResponse$inboundSchema:
  z.ZodType<
    RetirementConstraintsRetrieveCashJournalConstraintsResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    CashJournalConstraints: components.CashJournalConstraints$inboundSchema
      .optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "CashJournalConstraints": "cashJournalConstraints",
      "Status": "status",
    });
  });

/** @internal */
export type RetirementConstraintsRetrieveCashJournalConstraintsResponse$Outbound =
  {
    HttpMeta: components.HTTPMetadata$Outbound;
    CashJournalConstraints?:
      | components.CashJournalConstraints$Outbound
      | undefined;
    Status?: components.Status$Outbound | undefined;
  };

/** @internal */
export const RetirementConstraintsRetrieveCashJournalConstraintsResponse$outboundSchema:
  z.ZodType<
    RetirementConstraintsRetrieveCashJournalConstraintsResponse$Outbound,
    z.ZodTypeDef,
    RetirementConstraintsRetrieveCashJournalConstraintsResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    cashJournalConstraints: components.CashJournalConstraints$outboundSchema
      .optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      cashJournalConstraints: "CashJournalConstraints",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetirementConstraintsRetrieveCashJournalConstraintsResponse$ {
  /** @deprecated use `RetirementConstraintsRetrieveCashJournalConstraintsResponse$inboundSchema` instead. */
  export const inboundSchema =
    RetirementConstraintsRetrieveCashJournalConstraintsResponse$inboundSchema;
  /** @deprecated use `RetirementConstraintsRetrieveCashJournalConstraintsResponse$outboundSchema` instead. */
  export const outboundSchema =
    RetirementConstraintsRetrieveCashJournalConstraintsResponse$outboundSchema;
  /** @deprecated use `RetirementConstraintsRetrieveCashJournalConstraintsResponse$Outbound` instead. */
  export type Outbound =
    RetirementConstraintsRetrieveCashJournalConstraintsResponse$Outbound;
}
