// @apexfintechsolutions/ascend-sdk/src/models/operations/retirementconstraintslistcontributionsummaries.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type RetirementConstraintsListContributionSummariesRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * Number of contribution summaries to get (partitioned by tax year) Default = 2 (current year and prior year), maximum = 10
   */
  pageSize?: number | undefined;
  /**
   * When paginating, this is used to retrieve a specific page from the overall response
   */
  pageToken?: string | undefined;
};

export type RetirementConstraintsListContributionSummariesResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  listContributionSummariesResponse?:
    | components.ListContributionSummariesResponse
    | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const RetirementConstraintsListContributionSummariesRequest$inboundSchema:
  z.ZodType<
    RetirementConstraintsListContributionSummariesRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "page_size": "pageSize",
      "page_token": "pageToken",
    });
  });

/** @internal */
export type RetirementConstraintsListContributionSummariesRequest$Outbound = {
  account_id: string;
  page_size?: number | undefined;
  page_token?: string | undefined;
};

/** @internal */
export const RetirementConstraintsListContributionSummariesRequest$outboundSchema:
  z.ZodType<
    RetirementConstraintsListContributionSummariesRequest$Outbound,
    z.ZodTypeDef,
    RetirementConstraintsListContributionSummariesRequest
  > = z.object({
    accountId: z.string(),
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      pageSize: "page_size",
      pageToken: "page_token",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetirementConstraintsListContributionSummariesRequest$ {
  /** @deprecated use `RetirementConstraintsListContributionSummariesRequest$inboundSchema` instead. */
  export const inboundSchema =
    RetirementConstraintsListContributionSummariesRequest$inboundSchema;
  /** @deprecated use `RetirementConstraintsListContributionSummariesRequest$outboundSchema` instead. */
  export const outboundSchema =
    RetirementConstraintsListContributionSummariesRequest$outboundSchema;
  /** @deprecated use `RetirementConstraintsListContributionSummariesRequest$Outbound` instead. */
  export type Outbound =
    RetirementConstraintsListContributionSummariesRequest$Outbound;
}

/** @internal */
export const RetirementConstraintsListContributionSummariesResponse$inboundSchema:
  z.ZodType<
    RetirementConstraintsListContributionSummariesResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListContributionSummariesResponse: components
      .ListContributionSummariesResponse$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "ListContributionSummariesResponse": "listContributionSummariesResponse",
      "Status": "status",
    });
  });

/** @internal */
export type RetirementConstraintsListContributionSummariesResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  ListContributionSummariesResponse?:
    | components.ListContributionSummariesResponse$Outbound
    | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const RetirementConstraintsListContributionSummariesResponse$outboundSchema:
  z.ZodType<
    RetirementConstraintsListContributionSummariesResponse$Outbound,
    z.ZodTypeDef,
    RetirementConstraintsListContributionSummariesResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listContributionSummariesResponse: components
      .ListContributionSummariesResponse$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      listContributionSummariesResponse: "ListContributionSummariesResponse",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetirementConstraintsListContributionSummariesResponse$ {
  /** @deprecated use `RetirementConstraintsListContributionSummariesResponse$inboundSchema` instead. */
  export const inboundSchema =
    RetirementConstraintsListContributionSummariesResponse$inboundSchema;
  /** @deprecated use `RetirementConstraintsListContributionSummariesResponse$outboundSchema` instead. */
  export const outboundSchema =
    RetirementConstraintsListContributionSummariesResponse$outboundSchema;
  /** @deprecated use `RetirementConstraintsListContributionSummariesResponse$Outbound` instead. */
  export type Outbound =
    RetirementConstraintsListContributionSummariesResponse$Outbound;
}
