// @apexfintechsolutions/ascend-sdk/src/models/operations/retirementconstraintslistdistributionsummaries.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type RetirementConstraintsListDistributionSummariesRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * Number of distribution summaries to get (partitioned by tax year) Default = 2 (current year and prior year), maximum = 10
   */
  pageSize?: number | undefined;
  /**
   * When paginating, this is used to retrieve a specific page from the overall response
   */
  pageToken?: string | undefined;
};

export type RetirementConstraintsListDistributionSummariesResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  listDistributionSummariesResponse?:
    | components.ListDistributionSummariesResponse
    | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const RetirementConstraintsListDistributionSummariesRequest$inboundSchema:
  z.ZodType<
    RetirementConstraintsListDistributionSummariesRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "page_size": "pageSize",
      "page_token": "pageToken",
    });
  });

/** @internal */
export type RetirementConstraintsListDistributionSummariesRequest$Outbound = {
  account_id: string;
  page_size?: number | undefined;
  page_token?: string | undefined;
};

/** @internal */
export const RetirementConstraintsListDistributionSummariesRequest$outboundSchema:
  z.ZodType<
    RetirementConstraintsListDistributionSummariesRequest$Outbound,
    z.ZodTypeDef,
    RetirementConstraintsListDistributionSummariesRequest
  > = z.object({
    accountId: z.string(),
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      pageSize: "page_size",
      pageToken: "page_token",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetirementConstraintsListDistributionSummariesRequest$ {
  /** @deprecated use `RetirementConstraintsListDistributionSummariesRequest$inboundSchema` instead. */
  export const inboundSchema =
    RetirementConstraintsListDistributionSummariesRequest$inboundSchema;
  /** @deprecated use `RetirementConstraintsListDistributionSummariesRequest$outboundSchema` instead. */
  export const outboundSchema =
    RetirementConstraintsListDistributionSummariesRequest$outboundSchema;
  /** @deprecated use `RetirementConstraintsListDistributionSummariesRequest$Outbound` instead. */
  export type Outbound =
    RetirementConstraintsListDistributionSummariesRequest$Outbound;
}

/** @internal */
export const RetirementConstraintsListDistributionSummariesResponse$inboundSchema:
  z.ZodType<
    RetirementConstraintsListDistributionSummariesResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListDistributionSummariesResponse: components
      .ListDistributionSummariesResponse$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "ListDistributionSummariesResponse": "listDistributionSummariesResponse",
      "Status": "status",
    });
  });

/** @internal */
export type RetirementConstraintsListDistributionSummariesResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  ListDistributionSummariesResponse?:
    | components.ListDistributionSummariesResponse$Outbound
    | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const RetirementConstraintsListDistributionSummariesResponse$outboundSchema:
  z.ZodType<
    RetirementConstraintsListDistributionSummariesResponse$Outbound,
    z.ZodTypeDef,
    RetirementConstraintsListDistributionSummariesResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listDistributionSummariesResponse: components
      .ListDistributionSummariesResponse$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      listDistributionSummariesResponse: "ListDistributionSummariesResponse",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace RetirementConstraintsListDistributionSummariesResponse$ {
  /** @deprecated use `RetirementConstraintsListDistributionSummariesResponse$inboundSchema` instead. */
  export const inboundSchema =
    RetirementConstraintsListDistributionSummariesResponse$inboundSchema;
  /** @deprecated use `RetirementConstraintsListDistributionSummariesResponse$outboundSchema` instead. */
  export const outboundSchema =
    RetirementConstraintsListDistributionSummariesResponse$outboundSchema;
  /** @deprecated use `RetirementConstraintsListDistributionSummariesResponse$Outbound` instead. */
  export type Outbound =
    RetirementConstraintsListDistributionSummariesResponse$Outbound;
}
