// @apexfintechsolutions/ascend-sdk/src/models/operations/subscribercreatepushsubscription.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type SubscriberCreatePushSubscriptionResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  pushSubscription?: components.PushSubscription | undefined;
  /**
   * INVALID_ARGUMENT: The request was not well formed.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const SubscriberCreatePushSubscriptionResponse$inboundSchema: z.ZodType<
  SubscriberCreatePushSubscriptionResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  PushSubscription: components.PushSubscription$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "PushSubscription": "pushSubscription",
    "Status": "status",
  });
});

/** @internal */
export type SubscriberCreatePushSubscriptionResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  PushSubscription?: components.PushSubscription$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const SubscriberCreatePushSubscriptionResponse$outboundSchema: z.ZodType<
  SubscriberCreatePushSubscriptionResponse$Outbound,
  z.ZodTypeDef,
  SubscriberCreatePushSubscriptionResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  pushSubscription: components.PushSubscription$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    pushSubscription: "PushSubscription",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace SubscriberCreatePushSubscriptionResponse$ {
  /** @deprecated use `SubscriberCreatePushSubscriptionResponse$inboundSchema` instead. */
  export const inboundSchema =
    SubscriberCreatePushSubscriptionResponse$inboundSchema;
  /** @deprecated use `SubscriberCreatePushSubscriptionResponse$outboundSchema` instead. */
  export const outboundSchema =
    SubscriberCreatePushSubscriptionResponse$outboundSchema;
  /** @deprecated use `SubscriberCreatePushSubscriptionResponse$Outbound` instead. */
  export type Outbound = SubscriberCreatePushSubscriptionResponse$Outbound;
}
