// @apexfintechsolutions/ascend-sdk/src/models/operations/subscriberlistpushsubscriptions.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type SubscriberListPushSubscriptionsRequest = {
  /**
   * A CEL string to filter results; See the [CEL Search](https://developer.apexclearing.com/apex-fintech-solutions/docs/cel-search) page in Guides for more information; If empty, all subscriptions the user has permission to view will be returned; Filter options include:
   *
   * @remarks
   *  `name`
   *  `subscription_id`
   *  `client_id`
   *  `correspondent_id`
   *  `display_name`
   *  `event_types`
   *  `state`
   *  `http_callback.url`
   *  `http_callback.timeout_seconds`
   */
  filter?: string | undefined;
  /**
   * The number of entries to return in a single page; Default = 100; Maximum = 1000
   */
  pageSize?: number | undefined;
  /**
   * Page token used for pagination; Supplying a page token returns the next page of results
   */
  pageToken?: string | undefined;
};

export type SubscriberListPushSubscriptionsResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  listPushSubscriptionsResponse?:
    | components.ListPushSubscriptionsResponse
    | undefined;
  /**
   * INVALID_ARGUMENT: The request was not well formed.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const SubscriberListPushSubscriptionsRequest$inboundSchema: z.ZodType<
  SubscriberListPushSubscriptionsRequest,
  z.ZodTypeDef,
  unknown
> = z.object({
  filter: z.string().optional(),
  page_size: z.number().int().optional(),
  page_token: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    "page_size": "pageSize",
    "page_token": "pageToken",
  });
});

/** @internal */
export type SubscriberListPushSubscriptionsRequest$Outbound = {
  filter?: string | undefined;
  page_size?: number | undefined;
  page_token?: string | undefined;
};

/** @internal */
export const SubscriberListPushSubscriptionsRequest$outboundSchema: z.ZodType<
  SubscriberListPushSubscriptionsRequest$Outbound,
  z.ZodTypeDef,
  SubscriberListPushSubscriptionsRequest
> = z.object({
  filter: z.string().optional(),
  pageSize: z.number().int().optional(),
  pageToken: z.string().optional(),
}).transform((v) => {
  return remap$(v, {
    pageSize: "page_size",
    pageToken: "page_token",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace SubscriberListPushSubscriptionsRequest$ {
  /** @deprecated use `SubscriberListPushSubscriptionsRequest$inboundSchema` instead. */
  export const inboundSchema =
    SubscriberListPushSubscriptionsRequest$inboundSchema;
  /** @deprecated use `SubscriberListPushSubscriptionsRequest$outboundSchema` instead. */
  export const outboundSchema =
    SubscriberListPushSubscriptionsRequest$outboundSchema;
  /** @deprecated use `SubscriberListPushSubscriptionsRequest$Outbound` instead. */
  export type Outbound = SubscriberListPushSubscriptionsRequest$Outbound;
}

/** @internal */
export const SubscriberListPushSubscriptionsResponse$inboundSchema: z.ZodType<
  SubscriberListPushSubscriptionsResponse,
  z.ZodTypeDef,
  unknown
> = z.object({
  HttpMeta: components.HTTPMetadata$inboundSchema,
  ListPushSubscriptionsResponse: components
    .ListPushSubscriptionsResponse$inboundSchema.optional(),
  Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    "HttpMeta": "httpMeta",
    "ListPushSubscriptionsResponse": "listPushSubscriptionsResponse",
    "Status": "status",
  });
});

/** @internal */
export type SubscriberListPushSubscriptionsResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  ListPushSubscriptionsResponse?:
    | components.ListPushSubscriptionsResponse$Outbound
    | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const SubscriberListPushSubscriptionsResponse$outboundSchema: z.ZodType<
  SubscriberListPushSubscriptionsResponse$Outbound,
  z.ZodTypeDef,
  SubscriberListPushSubscriptionsResponse
> = z.object({
  httpMeta: components.HTTPMetadata$outboundSchema,
  listPushSubscriptionsResponse: components
    .ListPushSubscriptionsResponse$outboundSchema.optional(),
  status: components.Status$outboundSchema.optional(),
}).transform((v) => {
  return remap$(v, {
    httpMeta: "HttpMeta",
    listPushSubscriptionsResponse: "ListPushSubscriptionsResponse",
    status: "Status",
  });
});

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace SubscriberListPushSubscriptionsResponse$ {
  /** @deprecated use `SubscriberListPushSubscriptionsResponse$inboundSchema` instead. */
  export const inboundSchema =
    SubscriberListPushSubscriptionsResponse$inboundSchema;
  /** @deprecated use `SubscriberListPushSubscriptionsResponse$outboundSchema` instead. */
  export const outboundSchema =
    SubscriberListPushSubscriptionsResponse$outboundSchema;
  /** @deprecated use `SubscriberListPushSubscriptionsResponse$Outbound` instead. */
  export type Outbound = SubscriberListPushSubscriptionsResponse$Outbound;
}
