// @apexfintechsolutions/ascend-sdk/src/models/operations/transferschedulesummarieslistschedulesummaries.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type TransferScheduleSummariesListScheduleSummariesRequest = {
  /**
   * A CEL string to filter results; See the [CEL Search](https://developer.apexclearing.com/apex-fintech-solutions/docs/cel-search) page in Guides for more information; Filter options include:
   *
   * @remarks
   *  `account`
   *  `mechanism`
   *  `direction`
   *  `state`
   *  `start_date`
   *  `end_date`
   */
  filter?: string | undefined;
  /**
   * The maximum number of schedules to return. The service may return fewer than this value. If unspecified, at most 25 schedules will be returned. The maximum value is 1000; values above 1000 will be coerced to 1000.
   */
  pageSize?: number | undefined;
  /**
   * The page token to request
   */
  pageToken?: string | undefined;
};

export type TransferScheduleSummariesListScheduleSummariesResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  listScheduleSummariesResponse?:
    | components.ListScheduleSummariesResponse
    | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const TransferScheduleSummariesListScheduleSummariesRequest$inboundSchema:
  z.ZodType<
    TransferScheduleSummariesListScheduleSummariesRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    filter: z.string().optional(),
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
  }).transform((v) => {
    return remap$(v, {
      "page_size": "pageSize",
      "page_token": "pageToken",
    });
  });

/** @internal */
export type TransferScheduleSummariesListScheduleSummariesRequest$Outbound = {
  filter?: string | undefined;
  page_size?: number | undefined;
  page_token?: string | undefined;
};

/** @internal */
export const TransferScheduleSummariesListScheduleSummariesRequest$outboundSchema:
  z.ZodType<
    TransferScheduleSummariesListScheduleSummariesRequest$Outbound,
    z.ZodTypeDef,
    TransferScheduleSummariesListScheduleSummariesRequest
  > = z.object({
    filter: z.string().optional(),
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
  }).transform((v) => {
    return remap$(v, {
      pageSize: "page_size",
      pageToken: "page_token",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TransferScheduleSummariesListScheduleSummariesRequest$ {
  /** @deprecated use `TransferScheduleSummariesListScheduleSummariesRequest$inboundSchema` instead. */
  export const inboundSchema =
    TransferScheduleSummariesListScheduleSummariesRequest$inboundSchema;
  /** @deprecated use `TransferScheduleSummariesListScheduleSummariesRequest$outboundSchema` instead. */
  export const outboundSchema =
    TransferScheduleSummariesListScheduleSummariesRequest$outboundSchema;
  /** @deprecated use `TransferScheduleSummariesListScheduleSummariesRequest$Outbound` instead. */
  export type Outbound =
    TransferScheduleSummariesListScheduleSummariesRequest$Outbound;
}

/** @internal */
export const TransferScheduleSummariesListScheduleSummariesResponse$inboundSchema:
  z.ZodType<
    TransferScheduleSummariesListScheduleSummariesResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListScheduleSummariesResponse: components
      .ListScheduleSummariesResponse$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "ListScheduleSummariesResponse": "listScheduleSummariesResponse",
      "Status": "status",
    });
  });

/** @internal */
export type TransferScheduleSummariesListScheduleSummariesResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  ListScheduleSummariesResponse?:
    | components.ListScheduleSummariesResponse$Outbound
    | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const TransferScheduleSummariesListScheduleSummariesResponse$outboundSchema:
  z.ZodType<
    TransferScheduleSummariesListScheduleSummariesResponse$Outbound,
    z.ZodTypeDef,
    TransferScheduleSummariesListScheduleSummariesResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listScheduleSummariesResponse: components
      .ListScheduleSummariesResponse$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      listScheduleSummariesResponse: "ListScheduleSummariesResponse",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace TransferScheduleSummariesListScheduleSummariesResponse$ {
  /** @deprecated use `TransferScheduleSummariesListScheduleSummariesResponse$inboundSchema` instead. */
  export const inboundSchema =
    TransferScheduleSummariesListScheduleSummariesResponse$inboundSchema;
  /** @deprecated use `TransferScheduleSummariesListScheduleSummariesResponse$outboundSchema` instead. */
  export const outboundSchema =
    TransferScheduleSummariesListScheduleSummariesResponse$outboundSchema;
  /** @deprecated use `TransferScheduleSummariesListScheduleSummariesResponse$Outbound` instead. */
  export type Outbound =
    TransferScheduleSummariesListScheduleSummariesResponse$Outbound;
}
