// @apexfintechsolutions/ascend-sdk/src/models/operations/wirewithdrawalscheduleslistwirewithdrawalschedules.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * A CEL string to filter results; See the [CEL Search](https://developer.apexclearing.com/apex-fintech-solutions/docs/cel-search) page in Guides for more information; Filter options include:
   *
   * @remarks
   *  `state`
   *  `start_date`
   *  `end_date`
   */
  filter?: string | undefined;
  /**
   * The maximum number of schedules to return. The service may return fewer than this value. If unspecified, at most 25 schedules will be returned. The maximum value is 1000; values above 1000 will be coerced to 1000.
   */
  pageSize?: number | undefined;
  /**
   * The page token to request
   */
  pageToken?: string | undefined;
};

export type WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  listWireWithdrawalSchedulesResponse?:
    | components.ListWireWithdrawalSchedulesResponse
    | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest$inboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    filter: z.string().optional(),
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "page_size": "pageSize",
      "page_token": "pageToken",
    });
  });

/** @internal */
export type WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest$Outbound =
  {
    account_id: string;
    filter?: string | undefined;
    page_size?: number | undefined;
    page_token?: string | undefined;
  };

/** @internal */
export const WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest$outboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest$Outbound,
    z.ZodTypeDef,
    WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest
  > = z.object({
    accountId: z.string(),
    filter: z.string().optional(),
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      pageSize: "page_size",
      pageToken: "page_token",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest$ {
  /** @deprecated use `WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest$inboundSchema` instead. */
  export const inboundSchema =
    WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest$inboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest$outboundSchema` instead. */
  export const outboundSchema =
    WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest$outboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest$Outbound` instead. */
  export type Outbound =
    WireWithdrawalSchedulesListWireWithdrawalSchedulesRequest$Outbound;
}

/** @internal */
export const WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse$inboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListWireWithdrawalSchedulesResponse: components
      .ListWireWithdrawalSchedulesResponse$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "ListWireWithdrawalSchedulesResponse":
        "listWireWithdrawalSchedulesResponse",
      "Status": "status",
    });
  });

/** @internal */
export type WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse$Outbound =
  {
    HttpMeta: components.HTTPMetadata$Outbound;
    ListWireWithdrawalSchedulesResponse?:
      | components.ListWireWithdrawalSchedulesResponse$Outbound
      | undefined;
    Status?: components.Status$Outbound | undefined;
  };

/** @internal */
export const WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse$outboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse$Outbound,
    z.ZodTypeDef,
    WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listWireWithdrawalSchedulesResponse: components
      .ListWireWithdrawalSchedulesResponse$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      listWireWithdrawalSchedulesResponse:
        "ListWireWithdrawalSchedulesResponse",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse$ {
  /** @deprecated use `WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse$inboundSchema` instead. */
  export const inboundSchema =
    WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse$inboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse$outboundSchema` instead. */
  export const outboundSchema =
    WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse$outboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse$Outbound` instead. */
  export type Outbound =
    WireWithdrawalSchedulesListWireWithdrawalSchedulesResponse$Outbound;
}
