// @apexfintechsolutions/ascend-sdk/src/models/operations/wirewithdrawalschedulescreatewirewithdrawalschedule.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest = {
  /**
   * The account id.
   */
  accountId: string;
  wireWithdrawalScheduleCreate: components.WireWithdrawalScheduleCreate;
};

export type WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  wireWithdrawalSchedule?: components.WireWithdrawalSchedule | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest$inboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    WireWithdrawalScheduleCreate:
      components.WireWithdrawalScheduleCreate$inboundSchema,
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "WireWithdrawalScheduleCreate": "wireWithdrawalScheduleCreate",
    });
  });

/** @internal */
export type WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest$Outbound =
  {
    account_id: string;
    WireWithdrawalScheduleCreate:
      components.WireWithdrawalScheduleCreate$Outbound;
  };

/** @internal */
export const WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest$outboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest$Outbound,
    z.ZodTypeDef,
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest
  > = z.object({
    accountId: z.string(),
    wireWithdrawalScheduleCreate:
      components.WireWithdrawalScheduleCreate$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      wireWithdrawalScheduleCreate: "WireWithdrawalScheduleCreate",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest$ {
  /** @deprecated use `WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest$inboundSchema` instead. */
  export const inboundSchema =
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest$inboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest$outboundSchema` instead. */
  export const outboundSchema =
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest$outboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest$Outbound` instead. */
  export type Outbound =
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleRequest$Outbound;
}

/** @internal */
export const WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse$inboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    WireWithdrawalSchedule: components.WireWithdrawalSchedule$inboundSchema
      .optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "WireWithdrawalSchedule": "wireWithdrawalSchedule",
      "Status": "status",
    });
  });

/** @internal */
export type WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse$Outbound =
  {
    HttpMeta: components.HTTPMetadata$Outbound;
    WireWithdrawalSchedule?:
      | components.WireWithdrawalSchedule$Outbound
      | undefined;
    Status?: components.Status$Outbound | undefined;
  };

/** @internal */
export const WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse$outboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse$Outbound,
    z.ZodTypeDef,
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    wireWithdrawalSchedule: components.WireWithdrawalSchedule$outboundSchema
      .optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      wireWithdrawalSchedule: "WireWithdrawalSchedule",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse$ {
  /** @deprecated use `WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse$inboundSchema` instead. */
  export const inboundSchema =
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse$inboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse$outboundSchema` instead. */
  export const outboundSchema =
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse$outboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse$Outbound` instead. */
  export type Outbound =
    WireWithdrawalSchedulesCreateWireWithdrawalScheduleResponse$Outbound;
}
