// @apexfintechsolutions/ascend-sdk/src/models/operations/wirewithdrawalsforceapprovewirewithdrawal.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type WireWithdrawalsForceApproveWireWithdrawalRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The wireWithdrawal id.
   */
  wireWithdrawalId: string;
  forceApproveWireWithdrawalRequestCreate:
    components.ForceApproveWireWithdrawalRequestCreate;
};

export type WireWithdrawalsForceApproveWireWithdrawalResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  wireWithdrawal?: components.WireWithdrawal | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const WireWithdrawalsForceApproveWireWithdrawalRequest$inboundSchema:
  z.ZodType<
    WireWithdrawalsForceApproveWireWithdrawalRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    wireWithdrawal_id: z.string(),
    ForceApproveWireWithdrawalRequestCreate:
      components.ForceApproveWireWithdrawalRequestCreate$inboundSchema,
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "wireWithdrawal_id": "wireWithdrawalId",
      "ForceApproveWireWithdrawalRequestCreate":
        "forceApproveWireWithdrawalRequestCreate",
    });
  });

/** @internal */
export type WireWithdrawalsForceApproveWireWithdrawalRequest$Outbound = {
  account_id: string;
  wireWithdrawal_id: string;
  ForceApproveWireWithdrawalRequestCreate:
    components.ForceApproveWireWithdrawalRequestCreate$Outbound;
};

/** @internal */
export const WireWithdrawalsForceApproveWireWithdrawalRequest$outboundSchema:
  z.ZodType<
    WireWithdrawalsForceApproveWireWithdrawalRequest$Outbound,
    z.ZodTypeDef,
    WireWithdrawalsForceApproveWireWithdrawalRequest
  > = z.object({
    accountId: z.string(),
    wireWithdrawalId: z.string(),
    forceApproveWireWithdrawalRequestCreate:
      components.ForceApproveWireWithdrawalRequestCreate$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      wireWithdrawalId: "wireWithdrawal_id",
      forceApproveWireWithdrawalRequestCreate:
        "ForceApproveWireWithdrawalRequestCreate",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace WireWithdrawalsForceApproveWireWithdrawalRequest$ {
  /** @deprecated use `WireWithdrawalsForceApproveWireWithdrawalRequest$inboundSchema` instead. */
  export const inboundSchema =
    WireWithdrawalsForceApproveWireWithdrawalRequest$inboundSchema;
  /** @deprecated use `WireWithdrawalsForceApproveWireWithdrawalRequest$outboundSchema` instead. */
  export const outboundSchema =
    WireWithdrawalsForceApproveWireWithdrawalRequest$outboundSchema;
  /** @deprecated use `WireWithdrawalsForceApproveWireWithdrawalRequest$Outbound` instead. */
  export type Outbound =
    WireWithdrawalsForceApproveWireWithdrawalRequest$Outbound;
}

/** @internal */
export const WireWithdrawalsForceApproveWireWithdrawalResponse$inboundSchema:
  z.ZodType<
    WireWithdrawalsForceApproveWireWithdrawalResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    WireWithdrawal: components.WireWithdrawal$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "WireWithdrawal": "wireWithdrawal",
      "Status": "status",
    });
  });

/** @internal */
export type WireWithdrawalsForceApproveWireWithdrawalResponse$Outbound = {
  HttpMeta: components.HTTPMetadata$Outbound;
  WireWithdrawal?: components.WireWithdrawal$Outbound | undefined;
  Status?: components.Status$Outbound | undefined;
};

/** @internal */
export const WireWithdrawalsForceApproveWireWithdrawalResponse$outboundSchema:
  z.ZodType<
    WireWithdrawalsForceApproveWireWithdrawalResponse$Outbound,
    z.ZodTypeDef,
    WireWithdrawalsForceApproveWireWithdrawalResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    wireWithdrawal: components.WireWithdrawal$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      wireWithdrawal: "WireWithdrawal",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace WireWithdrawalsForceApproveWireWithdrawalResponse$ {
  /** @deprecated use `WireWithdrawalsForceApproveWireWithdrawalResponse$inboundSchema` instead. */
  export const inboundSchema =
    WireWithdrawalsForceApproveWireWithdrawalResponse$inboundSchema;
  /** @deprecated use `WireWithdrawalsForceApproveWireWithdrawalResponse$outboundSchema` instead. */
  export const outboundSchema =
    WireWithdrawalsForceApproveWireWithdrawalResponse$outboundSchema;
  /** @deprecated use `WireWithdrawalsForceApproveWireWithdrawalResponse$Outbound` instead. */
  export type Outbound =
    WireWithdrawalsForceApproveWireWithdrawalResponse$Outbound;
}
