// @apexfintechsolutions/ascend-sdk/src/models/operations/wirewithdrawalschedulesgetwirewithdrawalschedule.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */

import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest = {
  /**
   * The account id.
   */
  accountId: string;
  /**
   * The wireWithdrawalSchedule id.
   */
  wireWithdrawalScheduleId: string;
};

export type WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse = {
  httpMeta: components.HTTPMetadata;
  /**
   * OK
   */
  wireWithdrawalSchedule?: components.WireWithdrawalSchedule | undefined;
  /**
   * INVALID_ARGUMENT: The request has an invalid argument.
   */
  status?: components.Status | undefined;
};

/** @internal */
export const WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$inboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest,
    z.ZodTypeDef,
    unknown
  > = z.object({
    account_id: z.string(),
    wireWithdrawalSchedule_id: z.string(),
  }).transform((v) => {
    return remap$(v, {
      "account_id": "accountId",
      "wireWithdrawalSchedule_id": "wireWithdrawalScheduleId",
    });
  });

/** @internal */
export type WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$Outbound = {
  account_id: string;
  wireWithdrawalSchedule_id: string;
};

/** @internal */
export const WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$outboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$Outbound,
    z.ZodTypeDef,
    WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest
  > = z.object({
    accountId: z.string(),
    wireWithdrawalScheduleId: z.string(),
  }).transform((v) => {
    return remap$(v, {
      accountId: "account_id",
      wireWithdrawalScheduleId: "wireWithdrawalSchedule_id",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$ {
  /** @deprecated use `WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$inboundSchema` instead. */
  export const inboundSchema =
    WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$inboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$outboundSchema` instead. */
  export const outboundSchema =
    WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$outboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$Outbound` instead. */
  export type Outbound =
    WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$Outbound;
}

/** @internal */
export const WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$inboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse,
    z.ZodTypeDef,
    unknown
  > = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    WireWithdrawalSchedule: components.WireWithdrawalSchedule$inboundSchema
      .optional(),
    Status: components.Status$inboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      "HttpMeta": "httpMeta",
      "WireWithdrawalSchedule": "wireWithdrawalSchedule",
      "Status": "status",
    });
  });

/** @internal */
export type WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$Outbound =
  {
    HttpMeta: components.HTTPMetadata$Outbound;
    WireWithdrawalSchedule?:
      | components.WireWithdrawalSchedule$Outbound
      | undefined;
    Status?: components.Status$Outbound | undefined;
  };

/** @internal */
export const WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$outboundSchema:
  z.ZodType<
    WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$Outbound,
    z.ZodTypeDef,
    WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse
  > = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    wireWithdrawalSchedule: components.WireWithdrawalSchedule$outboundSchema
      .optional(),
    status: components.Status$outboundSchema.optional(),
  }).transform((v) => {
    return remap$(v, {
      httpMeta: "HttpMeta",
      wireWithdrawalSchedule: "WireWithdrawalSchedule",
      status: "Status",
    });
  });

/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export namespace WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$ {
  /** @deprecated use `WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$inboundSchema` instead. */
  export const inboundSchema =
    WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$inboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$outboundSchema` instead. */
  export const outboundSchema =
    WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$outboundSchema;
  /** @deprecated use `WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$Outbound` instead. */
  export type Outbound =
    WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$Outbound;
}
