// @apexfintechsolutions/ascend-sdk/lib/config.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SDK_METADATA = exports.ServerList = exports.ServerSbx = exports.ServerProd = exports.ServerUat = void 0;
exports.serverURLFromOptions = serverURLFromOptions;
const url_js_1 = require("./url.js");
/**
 * our uat environment
 */
exports.ServerUat = "uat";
/**
 * our production environment
 */
exports.ServerProd = "prod";
/**
 * our sandbox environment
 */
exports.ServerSbx = "sbx";
/**
 * Contains the list of servers available to the SDK
 */
exports.ServerList = {
    [exports.ServerUat]: "https://uat.apexapis.com",
    [exports.ServerProd]: "https://api.apexapis.com",
    [exports.ServerSbx]: "https://sbx.apexapis.com",
};
function serverURLFromOptions(options) {
    var _a;
    let serverURL = options.serverURL;
    const params = {};
    if (!serverURL) {
        const server = (_a = options.server) !== null && _a !== void 0 ? _a : exports.ServerUat;
        serverURL = exports.ServerList[server] || "";
    }
    const u = (0, url_js_1.pathToFunc)(serverURL)(params);
    return new URL(u);
}
exports.SDK_METADATA = {
    language: "typescript",
    openapiDocVersion: "v1:20250425:uat:4e9cbfb617ca",
    sdkVersion: "1.5.5",
    genVersion: "2.437.1",
    userAgent: "speakeasy-sdk/typescript 1.5.5 2.437.1 v1:20250425:uat:4e9cbfb617ca @apexfintechsolutions/ascend-sdk",
};
