// @apexfintechsolutions/ascend-sdk/models/operations/accountsupdatetrustedcontact.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsUpdateTrustedContactResponse$ = exports.AccountsUpdateTrustedContactResponse$outboundSchema = exports.AccountsUpdateTrustedContactResponse$inboundSchema = exports.AccountsUpdateTrustedContactRequest$ = exports.AccountsUpdateTrustedContactRequest$outboundSchema = exports.AccountsUpdateTrustedContactRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsUpdateTrustedContactRequest$inboundSchema = z.object({
    account_id: z.string(),
    trustedContact_id: z.string(),
    update_mask: z.string().optional(),
    TrustedContactUpdate: components.TrustedContactUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "trustedContact_id": "trustedContactId",
        "update_mask": "updateMask",
        "TrustedContactUpdate": "trustedContactUpdate",
    });
});
/** @internal */
exports.AccountsUpdateTrustedContactRequest$outboundSchema = z.object({
    accountId: z.string(),
    trustedContactId: z.string(),
    updateMask: z.string().optional(),
    trustedContactUpdate: components.TrustedContactUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        trustedContactId: "trustedContact_id",
        updateMask: "update_mask",
        trustedContactUpdate: "TrustedContactUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsUpdateTrustedContactRequest$;
(function (AccountsUpdateTrustedContactRequest$) {
    /** @deprecated use `AccountsUpdateTrustedContactRequest$inboundSchema` instead. */
    AccountsUpdateTrustedContactRequest$.inboundSchema = exports.AccountsUpdateTrustedContactRequest$inboundSchema;
    /** @deprecated use `AccountsUpdateTrustedContactRequest$outboundSchema` instead. */
    AccountsUpdateTrustedContactRequest$.outboundSchema = exports.AccountsUpdateTrustedContactRequest$outboundSchema;
})(AccountsUpdateTrustedContactRequest$ || (exports.AccountsUpdateTrustedContactRequest$ = AccountsUpdateTrustedContactRequest$ = {}));
/** @internal */
exports.AccountsUpdateTrustedContactResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    TrustedContact: components.TrustedContact$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "TrustedContact": "trustedContact",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsUpdateTrustedContactResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    trustedContact: components.TrustedContact$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        trustedContact: "TrustedContact",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsUpdateTrustedContactResponse$;
(function (AccountsUpdateTrustedContactResponse$) {
    /** @deprecated use `AccountsUpdateTrustedContactResponse$inboundSchema` instead. */
    AccountsUpdateTrustedContactResponse$.inboundSchema = exports.AccountsUpdateTrustedContactResponse$inboundSchema;
    /** @deprecated use `AccountsUpdateTrustedContactResponse$outboundSchema` instead. */
    AccountsUpdateTrustedContactResponse$.outboundSchema = exports.AccountsUpdateTrustedContactResponse$outboundSchema;
})(AccountsUpdateTrustedContactResponse$ || (exports.AccountsUpdateTrustedContactResponse$ = AccountsUpdateTrustedContactResponse$ = {}));
