// @apexfintechsolutions/ascend-sdk/models/operations/accountsaffirmagreements.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsAffirmAgreementsResponse$ = exports.AccountsAffirmAgreementsResponse$outboundSchema = exports.AccountsAffirmAgreementsResponse$inboundSchema = exports.AccountsAffirmAgreementsRequest$ = exports.AccountsAffirmAgreementsRequest$outboundSchema = exports.AccountsAffirmAgreementsRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsAffirmAgreementsRequest$inboundSchema = z.object({
    account_id: z.string(),
    AffirmAgreementsRequestCreate: components.AffirmAgreementsRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "AffirmAgreementsRequestCreate": "affirmAgreementsRequestCreate",
    });
});
/** @internal */
exports.AccountsAffirmAgreementsRequest$outboundSchema = z.object({
    accountId: z.string(),
    affirmAgreementsRequestCreate: components.AffirmAgreementsRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        affirmAgreementsRequestCreate: "AffirmAgreementsRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsAffirmAgreementsRequest$;
(function (AccountsAffirmAgreementsRequest$) {
    /** @deprecated use `AccountsAffirmAgreementsRequest$inboundSchema` instead. */
    AccountsAffirmAgreementsRequest$.inboundSchema = exports.AccountsAffirmAgreementsRequest$inboundSchema;
    /** @deprecated use `AccountsAffirmAgreementsRequest$outboundSchema` instead. */
    AccountsAffirmAgreementsRequest$.outboundSchema = exports.AccountsAffirmAgreementsRequest$outboundSchema;
})(AccountsAffirmAgreementsRequest$ || (exports.AccountsAffirmAgreementsRequest$ = AccountsAffirmAgreementsRequest$ = {}));
/** @internal */
exports.AccountsAffirmAgreementsResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AffirmAgreementsResponse: components.AffirmAgreementsResponse$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "AffirmAgreementsResponse": "affirmAgreementsResponse",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsAffirmAgreementsResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    affirmAgreementsResponse: components.AffirmAgreementsResponse$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        affirmAgreementsResponse: "AffirmAgreementsResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsAffirmAgreementsResponse$;
(function (AccountsAffirmAgreementsResponse$) {
    /** @deprecated use `AccountsAffirmAgreementsResponse$inboundSchema` instead. */
    AccountsAffirmAgreementsResponse$.inboundSchema = exports.AccountsAffirmAgreementsResponse$inboundSchema;
    /** @deprecated use `AccountsAffirmAgreementsResponse$outboundSchema` instead. */
    AccountsAffirmAgreementsResponse$.outboundSchema = exports.AccountsAffirmAgreementsResponse$outboundSchema;
})(AccountsAffirmAgreementsResponse$ || (exports.AccountsAffirmAgreementsResponse$ = AccountsAffirmAgreementsResponse$ = {}));
