// @apexfintechsolutions/ascend-sdk/models/operations/accounttransferscreatetransfer.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountTransfersCreateTransferResponse$ = exports.AccountTransfersCreateTransferResponse$outboundSchema = exports.AccountTransfersCreateTransferResponse$inboundSchema = exports.AccountTransfersCreateTransferRequest$ = exports.AccountTransfersCreateTransferRequest$outboundSchema = exports.AccountTransfersCreateTransferRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountTransfersCreateTransferRequest$inboundSchema = z.object({
    correspondent_id: z.string(),
    account_id: z.string(),
    request_id: z.string().optional(),
    TransferCreate: components.TransferCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "correspondent_id": "correspondentId",
        "account_id": "accountId",
        "request_id": "requestId",
        "TransferCreate": "transferCreate",
    });
});
/** @internal */
exports.AccountTransfersCreateTransferRequest$outboundSchema = z.object({
    correspondentId: z.string(),
    accountId: z.string(),
    requestId: z.string().optional(),
    transferCreate: components.TransferCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        correspondentId: "correspondent_id",
        accountId: "account_id",
        requestId: "request_id",
        transferCreate: "TransferCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountTransfersCreateTransferRequest$;
(function (AccountTransfersCreateTransferRequest$) {
    /** @deprecated use `AccountTransfersCreateTransferRequest$inboundSchema` instead. */
    AccountTransfersCreateTransferRequest$.inboundSchema = exports.AccountTransfersCreateTransferRequest$inboundSchema;
    /** @deprecated use `AccountTransfersCreateTransferRequest$outboundSchema` instead. */
    AccountTransfersCreateTransferRequest$.outboundSchema = exports.AccountTransfersCreateTransferRequest$outboundSchema;
})(AccountTransfersCreateTransferRequest$ || (exports.AccountTransfersCreateTransferRequest$ = AccountTransfersCreateTransferRequest$ = {}));
/** @internal */
exports.AccountTransfersCreateTransferResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AcatsTransfer: components.AcatsTransfer$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "AcatsTransfer": "acatsTransfer",
        "Status": "status",
    });
});
/** @internal */
exports.AccountTransfersCreateTransferResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    acatsTransfer: components.AcatsTransfer$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        acatsTransfer: "AcatsTransfer",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountTransfersCreateTransferResponse$;
(function (AccountTransfersCreateTransferResponse$) {
    /** @deprecated use `AccountTransfersCreateTransferResponse$inboundSchema` instead. */
    AccountTransfersCreateTransferResponse$.inboundSchema = exports.AccountTransfersCreateTransferResponse$inboundSchema;
    /** @deprecated use `AccountTransfersCreateTransferResponse$outboundSchema` instead. */
    AccountTransfersCreateTransferResponse$.outboundSchema = exports.AccountTransfersCreateTransferResponse$outboundSchema;
})(AccountTransfersCreateTransferResponse$ || (exports.AccountTransfersCreateTransferResponse$ = AccountTransfersCreateTransferResponse$ = {}));
