// @apexfintechsolutions/ascend-sdk/models/operations/accountsgetlegalnaturalperson.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsGetLegalNaturalPersonResponse$ = exports.AccountsGetLegalNaturalPersonResponse$outboundSchema = exports.AccountsGetLegalNaturalPersonResponse$inboundSchema = exports.AccountsGetLegalNaturalPersonRequest$ = exports.AccountsGetLegalNaturalPersonRequest$outboundSchema = exports.AccountsGetLegalNaturalPersonRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsGetLegalNaturalPersonRequest$inboundSchema = z.object({
    legalNaturalPerson_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "legalNaturalPerson_id": "legalNaturalPersonId",
    });
});
/** @internal */
exports.AccountsGetLegalNaturalPersonRequest$outboundSchema = z.object({
    legalNaturalPersonId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        legalNaturalPersonId: "legalNaturalPerson_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsGetLegalNaturalPersonRequest$;
(function (AccountsGetLegalNaturalPersonRequest$) {
    /** @deprecated use `AccountsGetLegalNaturalPersonRequest$inboundSchema` instead. */
    AccountsGetLegalNaturalPersonRequest$.inboundSchema = exports.AccountsGetLegalNaturalPersonRequest$inboundSchema;
    /** @deprecated use `AccountsGetLegalNaturalPersonRequest$outboundSchema` instead. */
    AccountsGetLegalNaturalPersonRequest$.outboundSchema = exports.AccountsGetLegalNaturalPersonRequest$outboundSchema;
})(AccountsGetLegalNaturalPersonRequest$ || (exports.AccountsGetLegalNaturalPersonRequest$ = AccountsGetLegalNaturalPersonRequest$ = {}));
/** @internal */
exports.AccountsGetLegalNaturalPersonResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    LegalNaturalPerson: components.LegalNaturalPerson$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "LegalNaturalPerson": "legalNaturalPerson",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsGetLegalNaturalPersonResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    legalNaturalPerson: components.LegalNaturalPerson$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        legalNaturalPerson: "LegalNaturalPerson",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsGetLegalNaturalPersonResponse$;
(function (AccountsGetLegalNaturalPersonResponse$) {
    /** @deprecated use `AccountsGetLegalNaturalPersonResponse$inboundSchema` instead. */
    AccountsGetLegalNaturalPersonResponse$.inboundSchema = exports.AccountsGetLegalNaturalPersonResponse$inboundSchema;
    /** @deprecated use `AccountsGetLegalNaturalPersonResponse$outboundSchema` instead. */
    AccountsGetLegalNaturalPersonResponse$.outboundSchema = exports.AccountsGetLegalNaturalPersonResponse$outboundSchema;
})(AccountsGetLegalNaturalPersonResponse$ || (exports.AccountsGetLegalNaturalPersonResponse$ = AccountsGetLegalNaturalPersonResponse$ = {}));
