// @apexfintechsolutions/ascend-sdk/models/operations/accountsendrestriction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsEndRestrictionResponse$ = exports.AccountsEndRestrictionResponse$outboundSchema = exports.AccountsEndRestrictionResponse$inboundSchema = exports.AccountsEndRestrictionRequest$ = exports.AccountsEndRestrictionRequest$outboundSchema = exports.AccountsEndRestrictionRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsEndRestrictionRequest$inboundSchema = z.object({
    account_id: z.string(),
    restriction_id: z.string(),
    EndRestrictionRequestCreate: components.EndRestrictionRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "restriction_id": "restrictionId",
        "EndRestrictionRequestCreate": "endRestrictionRequestCreate",
    });
});
/** @internal */
exports.AccountsEndRestrictionRequest$outboundSchema = z.object({
    accountId: z.string(),
    restrictionId: z.string(),
    endRestrictionRequestCreate: components.EndRestrictionRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        restrictionId: "restriction_id",
        endRestrictionRequestCreate: "EndRestrictionRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsEndRestrictionRequest$;
(function (AccountsEndRestrictionRequest$) {
    /** @deprecated use `AccountsEndRestrictionRequest$inboundSchema` instead. */
    AccountsEndRestrictionRequest$.inboundSchema = exports.AccountsEndRestrictionRequest$inboundSchema;
    /** @deprecated use `AccountsEndRestrictionRequest$outboundSchema` instead. */
    AccountsEndRestrictionRequest$.outboundSchema = exports.AccountsEndRestrictionRequest$outboundSchema;
})(AccountsEndRestrictionRequest$ || (exports.AccountsEndRestrictionRequest$ = AccountsEndRestrictionRequest$ = {}));
/** @internal */
exports.AccountsEndRestrictionResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsEndRestrictionResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsEndRestrictionResponse$;
(function (AccountsEndRestrictionResponse$) {
    /** @deprecated use `AccountsEndRestrictionResponse$inboundSchema` instead. */
    AccountsEndRestrictionResponse$.inboundSchema = exports.AccountsEndRestrictionResponse$inboundSchema;
    /** @deprecated use `AccountsEndRestrictionResponse$outboundSchema` instead. */
    AccountsEndRestrictionResponse$.outboundSchema = exports.AccountsEndRestrictionResponse$outboundSchema;
})(AccountsEndRestrictionResponse$ || (exports.AccountsEndRestrictionResponse$ = AccountsEndRestrictionResponse$ = {}));
