// @apexfintechsolutions/ascend-sdk/models/operations/accountscloseaccount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsCloseAccountResponse$ = exports.AccountsCloseAccountResponse$outboundSchema = exports.AccountsCloseAccountResponse$inboundSchema = exports.AccountsCloseAccountRequest$ = exports.AccountsCloseAccountRequest$outboundSchema = exports.AccountsCloseAccountRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsCloseAccountRequest$inboundSchema = z.object({
    account_id: z.string(),
    CloseAccountRequestCreate: components.CloseAccountRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "CloseAccountRequestCreate": "closeAccountRequestCreate",
    });
});
/** @internal */
exports.AccountsCloseAccountRequest$outboundSchema = z.object({
    accountId: z.string(),
    closeAccountRequestCreate: components.CloseAccountRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        closeAccountRequestCreate: "CloseAccountRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsCloseAccountRequest$;
(function (AccountsCloseAccountRequest$) {
    /** @deprecated use `AccountsCloseAccountRequest$inboundSchema` instead. */
    AccountsCloseAccountRequest$.inboundSchema = exports.AccountsCloseAccountRequest$inboundSchema;
    /** @deprecated use `AccountsCloseAccountRequest$outboundSchema` instead. */
    AccountsCloseAccountRequest$.outboundSchema = exports.AccountsCloseAccountRequest$outboundSchema;
})(AccountsCloseAccountRequest$ || (exports.AccountsCloseAccountRequest$ = AccountsCloseAccountRequest$ = {}));
/** @internal */
exports.AccountsCloseAccountResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Account: components.Account$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Account": "account",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsCloseAccountResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    account: components.Account$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        account: "Account",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsCloseAccountResponse$;
(function (AccountsCloseAccountResponse$) {
    /** @deprecated use `AccountsCloseAccountResponse$inboundSchema` instead. */
    AccountsCloseAccountResponse$.inboundSchema = exports.AccountsCloseAccountResponse$inboundSchema;
    /** @deprecated use `AccountsCloseAccountResponse$outboundSchema` instead. */
    AccountsCloseAccountResponse$.outboundSchema = exports.AccountsCloseAccountResponse$outboundSchema;
})(AccountsCloseAccountResponse$ || (exports.AccountsCloseAccountResponse$ = AccountsCloseAccountResponse$ = {}));
