// @apexfintechsolutions/ascend-sdk/models/operations/accountsupdatelegalnaturalperson.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsUpdateLegalNaturalPersonResponse$ = exports.AccountsUpdateLegalNaturalPersonResponse$outboundSchema = exports.AccountsUpdateLegalNaturalPersonResponse$inboundSchema = exports.AccountsUpdateLegalNaturalPersonRequest$ = exports.AccountsUpdateLegalNaturalPersonRequest$outboundSchema = exports.AccountsUpdateLegalNaturalPersonRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsUpdateLegalNaturalPersonRequest$inboundSchema = z.object({
    legalNaturalPerson_id: z.string(),
    update_mask: z.string().optional(),
    LegalNaturalPersonUpdate: components.LegalNaturalPersonUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "legalNaturalPerson_id": "legalNaturalPersonId",
        "update_mask": "updateMask",
        "LegalNaturalPersonUpdate": "legalNaturalPersonUpdate",
    });
});
/** @internal */
exports.AccountsUpdateLegalNaturalPersonRequest$outboundSchema = z.object({
    legalNaturalPersonId: z.string(),
    updateMask: z.string().optional(),
    legalNaturalPersonUpdate: components.LegalNaturalPersonUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        legalNaturalPersonId: "legalNaturalPerson_id",
        updateMask: "update_mask",
        legalNaturalPersonUpdate: "LegalNaturalPersonUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsUpdateLegalNaturalPersonRequest$;
(function (AccountsUpdateLegalNaturalPersonRequest$) {
    /** @deprecated use `AccountsUpdateLegalNaturalPersonRequest$inboundSchema` instead. */
    AccountsUpdateLegalNaturalPersonRequest$.inboundSchema = exports.AccountsUpdateLegalNaturalPersonRequest$inboundSchema;
    /** @deprecated use `AccountsUpdateLegalNaturalPersonRequest$outboundSchema` instead. */
    AccountsUpdateLegalNaturalPersonRequest$.outboundSchema = exports.AccountsUpdateLegalNaturalPersonRequest$outboundSchema;
})(AccountsUpdateLegalNaturalPersonRequest$ || (exports.AccountsUpdateLegalNaturalPersonRequest$ = AccountsUpdateLegalNaturalPersonRequest$ = {}));
/** @internal */
exports.AccountsUpdateLegalNaturalPersonResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    LegalNaturalPerson: components.LegalNaturalPerson$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "LegalNaturalPerson": "legalNaturalPerson",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsUpdateLegalNaturalPersonResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    legalNaturalPerson: components.LegalNaturalPerson$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        legalNaturalPerson: "LegalNaturalPerson",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsUpdateLegalNaturalPersonResponse$;
(function (AccountsUpdateLegalNaturalPersonResponse$) {
    /** @deprecated use `AccountsUpdateLegalNaturalPersonResponse$inboundSchema` instead. */
    AccountsUpdateLegalNaturalPersonResponse$.inboundSchema = exports.AccountsUpdateLegalNaturalPersonResponse$inboundSchema;
    /** @deprecated use `AccountsUpdateLegalNaturalPersonResponse$outboundSchema` instead. */
    AccountsUpdateLegalNaturalPersonResponse$.outboundSchema = exports.AccountsUpdateLegalNaturalPersonResponse$outboundSchema;
})(AccountsUpdateLegalNaturalPersonResponse$ || (exports.AccountsUpdateLegalNaturalPersonResponse$ = AccountsUpdateLegalNaturalPersonResponse$ = {}));
