// @apexfintechsolutions/ascend-sdk/models/operations/accountsaddparty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsAddPartyResponse$ = exports.AccountsAddPartyResponse$outboundSchema = exports.AccountsAddPartyResponse$inboundSchema = exports.AccountsAddPartyRequest$ = exports.AccountsAddPartyRequest$outboundSchema = exports.AccountsAddPartyRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsAddPartyRequest$inboundSchema = z.object({
    account_id: z.string(),
    AddPartyRequestCreate: components.AddPartyRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "AddPartyRequestCreate": "addPartyRequestCreate",
    });
});
/** @internal */
exports.AccountsAddPartyRequest$outboundSchema = z.object({
    accountId: z.string(),
    addPartyRequestCreate: components.AddPartyRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        addPartyRequestCreate: "AddPartyRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsAddPartyRequest$;
(function (AccountsAddPartyRequest$) {
    /** @deprecated use `AccountsAddPartyRequest$inboundSchema` instead. */
    AccountsAddPartyRequest$.inboundSchema = exports.AccountsAddPartyRequest$inboundSchema;
    /** @deprecated use `AccountsAddPartyRequest$outboundSchema` instead. */
    AccountsAddPartyRequest$.outboundSchema = exports.AccountsAddPartyRequest$outboundSchema;
})(AccountsAddPartyRequest$ || (exports.AccountsAddPartyRequest$ = AccountsAddPartyRequest$ = {}));
/** @internal */
exports.AccountsAddPartyResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Party: components.Party$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Party": "party",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsAddPartyResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    party: components.Party$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        party: "Party",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsAddPartyResponse$;
(function (AccountsAddPartyResponse$) {
    /** @deprecated use `AccountsAddPartyResponse$inboundSchema` instead. */
    AccountsAddPartyResponse$.inboundSchema = exports.AccountsAddPartyResponse$inboundSchema;
    /** @deprecated use `AccountsAddPartyResponse$outboundSchema` instead. */
    AccountsAddPartyResponse$.outboundSchema = exports.AccountsAddPartyResponse$outboundSchema;
})(AccountsAddPartyResponse$ || (exports.AccountsAddPartyResponse$ = AccountsAddPartyResponse$ = {}));
