// @apexfintechsolutions/ascend-sdk/models/operations/accountsupdateaccount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsUpdateAccountResponse$ = exports.AccountsUpdateAccountResponse$outboundSchema = exports.AccountsUpdateAccountResponse$inboundSchema = exports.AccountsUpdateAccountRequest$ = exports.AccountsUpdateAccountRequest$outboundSchema = exports.AccountsUpdateAccountRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsUpdateAccountRequest$inboundSchema = z.object({
    account_id: z.string(),
    update_mask: z.string().optional(),
    AccountRequestUpdate: components.AccountRequestUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "update_mask": "updateMask",
        "AccountRequestUpdate": "accountRequestUpdate",
    });
});
/** @internal */
exports.AccountsUpdateAccountRequest$outboundSchema = z.object({
    accountId: z.string(),
    updateMask: z.string().optional(),
    accountRequestUpdate: components.AccountRequestUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        updateMask: "update_mask",
        accountRequestUpdate: "AccountRequestUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsUpdateAccountRequest$;
(function (AccountsUpdateAccountRequest$) {
    /** @deprecated use `AccountsUpdateAccountRequest$inboundSchema` instead. */
    AccountsUpdateAccountRequest$.inboundSchema = exports.AccountsUpdateAccountRequest$inboundSchema;
    /** @deprecated use `AccountsUpdateAccountRequest$outboundSchema` instead. */
    AccountsUpdateAccountRequest$.outboundSchema = exports.AccountsUpdateAccountRequest$outboundSchema;
})(AccountsUpdateAccountRequest$ || (exports.AccountsUpdateAccountRequest$ = AccountsUpdateAccountRequest$ = {}));
/** @internal */
exports.AccountsUpdateAccountResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Account: components.Account$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Account": "account",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsUpdateAccountResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    account: components.Account$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        account: "Account",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsUpdateAccountResponse$;
(function (AccountsUpdateAccountResponse$) {
    /** @deprecated use `AccountsUpdateAccountResponse$inboundSchema` instead. */
    AccountsUpdateAccountResponse$.inboundSchema = exports.AccountsUpdateAccountResponse$inboundSchema;
    /** @deprecated use `AccountsUpdateAccountResponse$outboundSchema` instead. */
    AccountsUpdateAccountResponse$.outboundSchema = exports.AccountsUpdateAccountResponse$outboundSchema;
})(AccountsUpdateAccountResponse$ || (exports.AccountsUpdateAccountResponse$ = AccountsUpdateAccountResponse$ = {}));
