// @apexfintechsolutions/ascend-sdk/models/operations/accountslistaccounts.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsListAccountsResponse$ = exports.AccountsListAccountsResponse$outboundSchema = exports.AccountsListAccountsResponse$inboundSchema = exports.AccountsListAccountsRequest$ = exports.AccountsListAccountsRequest$outboundSchema = exports.AccountsListAccountsRequest$inboundSchema = exports.View$ = exports.View$outboundSchema = exports.View$inboundSchema = exports.View = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/**
 * The view to return. Defaults to `BASIC`.
 */
var View;
(function (View) {
    View["AccountViewUnspecified"] = "ACCOUNT_VIEW_UNSPECIFIED";
    View["Basic"] = "BASIC";
    View["Full"] = "FULL";
})(View || (exports.View = View = {}));
/** @internal */
exports.View$inboundSchema = z.nativeEnum(View);
/** @internal */
exports.View$outboundSchema = exports.View$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var View$;
(function (View$) {
    /** @deprecated use `View$inboundSchema` instead. */
    View$.inboundSchema = exports.View$inboundSchema;
    /** @deprecated use `View$outboundSchema` instead. */
    View$.outboundSchema = exports.View$outboundSchema;
})(View$ || (exports.View$ = View$ = {}));
/** @internal */
exports.AccountsListAccountsRequest$inboundSchema = z.object({
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
    order_by: z.string().optional(),
    filter: z.string().optional(),
    view: exports.View$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "page_size": "pageSize",
        "page_token": "pageToken",
        "order_by": "orderBy",
    });
});
/** @internal */
exports.AccountsListAccountsRequest$outboundSchema = z.object({
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
    orderBy: z.string().optional(),
    filter: z.string().optional(),
    view: exports.View$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        pageSize: "page_size",
        pageToken: "page_token",
        orderBy: "order_by",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsListAccountsRequest$;
(function (AccountsListAccountsRequest$) {
    /** @deprecated use `AccountsListAccountsRequest$inboundSchema` instead. */
    AccountsListAccountsRequest$.inboundSchema = exports.AccountsListAccountsRequest$inboundSchema;
    /** @deprecated use `AccountsListAccountsRequest$outboundSchema` instead. */
    AccountsListAccountsRequest$.outboundSchema = exports.AccountsListAccountsRequest$outboundSchema;
})(AccountsListAccountsRequest$ || (exports.AccountsListAccountsRequest$ = AccountsListAccountsRequest$ = {}));
/** @internal */
exports.AccountsListAccountsResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListAccountsResponse: components.ListAccountsResponse$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "ListAccountsResponse": "listAccountsResponse",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsListAccountsResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listAccountsResponse: components.ListAccountsResponse$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        listAccountsResponse: "ListAccountsResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsListAccountsResponse$;
(function (AccountsListAccountsResponse$) {
    /** @deprecated use `AccountsListAccountsResponse$inboundSchema` instead. */
    AccountsListAccountsResponse$.inboundSchema = exports.AccountsListAccountsResponse$inboundSchema;
    /** @deprecated use `AccountsListAccountsResponse$outboundSchema` instead. */
    AccountsListAccountsResponse$.outboundSchema = exports.AccountsListAccountsResponse$outboundSchema;
})(AccountsListAccountsResponse$ || (exports.AccountsListAccountsResponse$ = AccountsListAccountsResponse$ = {}));
