// @apexfintechsolutions/ascend-sdk/models/operations/accountsgetaccount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsGetAccountResponse$ = exports.AccountsGetAccountResponse$outboundSchema = exports.AccountsGetAccountResponse$inboundSchema = exports.AccountsGetAccountRequest$ = exports.AccountsGetAccountRequest$outboundSchema = exports.AccountsGetAccountRequest$inboundSchema = exports.QueryParamView$ = exports.QueryParamView$outboundSchema = exports.QueryParamView$inboundSchema = exports.QueryParamView = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/**
 * The view to return. Defaults to `FULL`.
 */
var QueryParamView;
(function (QueryParamView) {
    QueryParamView["AccountViewUnspecified"] = "ACCOUNT_VIEW_UNSPECIFIED";
    QueryParamView["Basic"] = "BASIC";
    QueryParamView["Full"] = "FULL";
})(QueryParamView || (exports.QueryParamView = QueryParamView = {}));
/** @internal */
exports.QueryParamView$inboundSchema = z.nativeEnum(QueryParamView);
/** @internal */
exports.QueryParamView$outboundSchema = exports.QueryParamView$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var QueryParamView$;
(function (QueryParamView$) {
    /** @deprecated use `QueryParamView$inboundSchema` instead. */
    QueryParamView$.inboundSchema = exports.QueryParamView$inboundSchema;
    /** @deprecated use `QueryParamView$outboundSchema` instead. */
    QueryParamView$.outboundSchema = exports.QueryParamView$outboundSchema;
})(QueryParamView$ || (exports.QueryParamView$ = QueryParamView$ = {}));
/** @internal */
exports.AccountsGetAccountRequest$inboundSchema = z.object({
    account_id: z.string(),
    view: exports.QueryParamView$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
    });
});
/** @internal */
exports.AccountsGetAccountRequest$outboundSchema = z.object({
    accountId: z.string(),
    view: exports.QueryParamView$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsGetAccountRequest$;
(function (AccountsGetAccountRequest$) {
    /** @deprecated use `AccountsGetAccountRequest$inboundSchema` instead. */
    AccountsGetAccountRequest$.inboundSchema = exports.AccountsGetAccountRequest$inboundSchema;
    /** @deprecated use `AccountsGetAccountRequest$outboundSchema` instead. */
    AccountsGetAccountRequest$.outboundSchema = exports.AccountsGetAccountRequest$outboundSchema;
})(AccountsGetAccountRequest$ || (exports.AccountsGetAccountRequest$ = AccountsGetAccountRequest$ = {}));
/** @internal */
exports.AccountsGetAccountResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Account: components.Account$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Account": "account",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsGetAccountResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    account: components.Account$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        account: "Account",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsGetAccountResponse$;
(function (AccountsGetAccountResponse$) {
    /** @deprecated use `AccountsGetAccountResponse$inboundSchema` instead. */
    AccountsGetAccountResponse$.inboundSchema = exports.AccountsGetAccountResponse$inboundSchema;
    /** @deprecated use `AccountsGetAccountResponse$outboundSchema` instead. */
    AccountsGetAccountResponse$.outboundSchema = exports.AccountsGetAccountResponse$outboundSchema;
})(AccountsGetAccountResponse$ || (exports.AccountsGetAccountResponse$ = AccountsGetAccountResponse$ = {}));
