// @apexfintechsolutions/ascend-sdk/models/operations/accountscreateinterestedparty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsCreateInterestedPartyResponse$ = exports.AccountsCreateInterestedPartyResponse$outboundSchema = exports.AccountsCreateInterestedPartyResponse$inboundSchema = exports.AccountsCreateInterestedPartyRequest$ = exports.AccountsCreateInterestedPartyRequest$outboundSchema = exports.AccountsCreateInterestedPartyRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsCreateInterestedPartyRequest$inboundSchema = z.object({
    account_id: z.string(),
    InterestedPartyCreate: components.InterestedPartyCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "InterestedPartyCreate": "interestedPartyCreate",
    });
});
/** @internal */
exports.AccountsCreateInterestedPartyRequest$outboundSchema = z.object({
    accountId: z.string(),
    interestedPartyCreate: components.InterestedPartyCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        interestedPartyCreate: "InterestedPartyCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsCreateInterestedPartyRequest$;
(function (AccountsCreateInterestedPartyRequest$) {
    /** @deprecated use `AccountsCreateInterestedPartyRequest$inboundSchema` instead. */
    AccountsCreateInterestedPartyRequest$.inboundSchema = exports.AccountsCreateInterestedPartyRequest$inboundSchema;
    /** @deprecated use `AccountsCreateInterestedPartyRequest$outboundSchema` instead. */
    AccountsCreateInterestedPartyRequest$.outboundSchema = exports.AccountsCreateInterestedPartyRequest$outboundSchema;
})(AccountsCreateInterestedPartyRequest$ || (exports.AccountsCreateInterestedPartyRequest$ = AccountsCreateInterestedPartyRequest$ = {}));
/** @internal */
exports.AccountsCreateInterestedPartyResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    InterestedParty: components.InterestedParty$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "InterestedParty": "interestedParty",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsCreateInterestedPartyResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    interestedParty: components.InterestedParty$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        interestedParty: "InterestedParty",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsCreateInterestedPartyResponse$;
(function (AccountsCreateInterestedPartyResponse$) {
    /** @deprecated use `AccountsCreateInterestedPartyResponse$inboundSchema` instead. */
    AccountsCreateInterestedPartyResponse$.inboundSchema = exports.AccountsCreateInterestedPartyResponse$inboundSchema;
    /** @deprecated use `AccountsCreateInterestedPartyResponse$outboundSchema` instead. */
    AccountsCreateInterestedPartyResponse$.outboundSchema = exports.AccountsCreateInterestedPartyResponse$outboundSchema;
})(AccountsCreateInterestedPartyResponse$ || (exports.AccountsCreateInterestedPartyResponse$ = AccountsCreateInterestedPartyResponse$ = {}));
