// @apexfintechsolutions/ascend-sdk/models/operations/accountsremoveparty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsRemovePartyResponse$ = exports.AccountsRemovePartyResponse$outboundSchema = exports.AccountsRemovePartyResponse$inboundSchema = exports.AccountsRemovePartyRequest$ = exports.AccountsRemovePartyRequest$outboundSchema = exports.AccountsRemovePartyRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsRemovePartyRequest$inboundSchema = z.object({
    account_id: z.string(),
    party_id: z.string(),
    RemovePartyRequestCreate: components.RemovePartyRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "party_id": "partyId",
        "RemovePartyRequestCreate": "removePartyRequestCreate",
    });
});
/** @internal */
exports.AccountsRemovePartyRequest$outboundSchema = z.object({
    accountId: z.string(),
    partyId: z.string(),
    removePartyRequestCreate: components.RemovePartyRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        partyId: "party_id",
        removePartyRequestCreate: "RemovePartyRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsRemovePartyRequest$;
(function (AccountsRemovePartyRequest$) {
    /** @deprecated use `AccountsRemovePartyRequest$inboundSchema` instead. */
    AccountsRemovePartyRequest$.inboundSchema = exports.AccountsRemovePartyRequest$inboundSchema;
    /** @deprecated use `AccountsRemovePartyRequest$outboundSchema` instead. */
    AccountsRemovePartyRequest$.outboundSchema = exports.AccountsRemovePartyRequest$outboundSchema;
})(AccountsRemovePartyRequest$ || (exports.AccountsRemovePartyRequest$ = AccountsRemovePartyRequest$ = {}));
/** @internal */
exports.AccountsRemovePartyResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsRemovePartyResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsRemovePartyResponse$;
(function (AccountsRemovePartyResponse$) {
    /** @deprecated use `AccountsRemovePartyResponse$inboundSchema` instead. */
    AccountsRemovePartyResponse$.inboundSchema = exports.AccountsRemovePartyResponse$inboundSchema;
    /** @deprecated use `AccountsRemovePartyResponse$outboundSchema` instead. */
    AccountsRemovePartyResponse$.outboundSchema = exports.AccountsRemovePartyResponse$outboundSchema;
})(AccountsRemovePartyResponse$ || (exports.AccountsRemovePartyResponse$ = AccountsRemovePartyResponse$ = {}));
