// @apexfintechsolutions/ascend-sdk/models/operations/accountslistenrollments.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsListEnrollmentsResponse$ = exports.AccountsListEnrollmentsResponse$outboundSchema = exports.AccountsListEnrollmentsResponse$inboundSchema = exports.AccountsListEnrollmentsRequest$ = exports.AccountsListEnrollmentsRequest$outboundSchema = exports.AccountsListEnrollmentsRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsListEnrollmentsRequest$inboundSchema = z.object({
    account_id: z.string(),
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "page_size": "pageSize",
        "page_token": "pageToken",
    });
});
/** @internal */
exports.AccountsListEnrollmentsRequest$outboundSchema = z.object({
    accountId: z.string(),
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        pageSize: "page_size",
        pageToken: "page_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsListEnrollmentsRequest$;
(function (AccountsListEnrollmentsRequest$) {
    /** @deprecated use `AccountsListEnrollmentsRequest$inboundSchema` instead. */
    AccountsListEnrollmentsRequest$.inboundSchema = exports.AccountsListEnrollmentsRequest$inboundSchema;
    /** @deprecated use `AccountsListEnrollmentsRequest$outboundSchema` instead. */
    AccountsListEnrollmentsRequest$.outboundSchema = exports.AccountsListEnrollmentsRequest$outboundSchema;
})(AccountsListEnrollmentsRequest$ || (exports.AccountsListEnrollmentsRequest$ = AccountsListEnrollmentsRequest$ = {}));
/** @internal */
exports.AccountsListEnrollmentsResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListEnrollmentsResponse: components.ListEnrollmentsResponse$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "ListEnrollmentsResponse": "listEnrollmentsResponse",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsListEnrollmentsResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listEnrollmentsResponse: components.ListEnrollmentsResponse$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        listEnrollmentsResponse: "ListEnrollmentsResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsListEnrollmentsResponse$;
(function (AccountsListEnrollmentsResponse$) {
    /** @deprecated use `AccountsListEnrollmentsResponse$inboundSchema` instead. */
    AccountsListEnrollmentsResponse$.inboundSchema = exports.AccountsListEnrollmentsResponse$inboundSchema;
    /** @deprecated use `AccountsListEnrollmentsResponse$outboundSchema` instead. */
    AccountsListEnrollmentsResponse$.outboundSchema = exports.AccountsListEnrollmentsResponse$outboundSchema;
})(AccountsListEnrollmentsResponse$ || (exports.AccountsListEnrollmentsResponse$ = AccountsListEnrollmentsResponse$ = {}));
