// @apexfintechsolutions/ascend-sdk/models/operations/accountsendlargetrader1.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsEndLargeTrader1Response$ = exports.AccountsEndLargeTrader1Response$outboundSchema = exports.AccountsEndLargeTrader1Response$inboundSchema = exports.AccountsEndLargeTrader1Request$ = exports.AccountsEndLargeTrader1Request$outboundSchema = exports.AccountsEndLargeTrader1Request$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsEndLargeTrader1Request$inboundSchema = z.object({
    legalEntity_id: z.string(),
    EndLargeTraderRequestCreate: components.EndLargeTraderRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "legalEntity_id": "legalEntityId",
        "EndLargeTraderRequestCreate": "endLargeTraderRequestCreate",
    });
});
/** @internal */
exports.AccountsEndLargeTrader1Request$outboundSchema = z.object({
    legalEntityId: z.string(),
    endLargeTraderRequestCreate: components.EndLargeTraderRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        legalEntityId: "legalEntity_id",
        endLargeTraderRequestCreate: "EndLargeTraderRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsEndLargeTrader1Request$;
(function (AccountsEndLargeTrader1Request$) {
    /** @deprecated use `AccountsEndLargeTrader1Request$inboundSchema` instead. */
    AccountsEndLargeTrader1Request$.inboundSchema = exports.AccountsEndLargeTrader1Request$inboundSchema;
    /** @deprecated use `AccountsEndLargeTrader1Request$outboundSchema` instead. */
    AccountsEndLargeTrader1Request$.outboundSchema = exports.AccountsEndLargeTrader1Request$outboundSchema;
})(AccountsEndLargeTrader1Request$ || (exports.AccountsEndLargeTrader1Request$ = AccountsEndLargeTrader1Request$ = {}));
/** @internal */
exports.AccountsEndLargeTrader1Response$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsEndLargeTrader1Response$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsEndLargeTrader1Response$;
(function (AccountsEndLargeTrader1Response$) {
    /** @deprecated use `AccountsEndLargeTrader1Response$inboundSchema` instead. */
    AccountsEndLargeTrader1Response$.inboundSchema = exports.AccountsEndLargeTrader1Response$inboundSchema;
    /** @deprecated use `AccountsEndLargeTrader1Response$outboundSchema` instead. */
    AccountsEndLargeTrader1Response$.outboundSchema = exports.AccountsEndLargeTrader1Response$outboundSchema;
})(AccountsEndLargeTrader1Response$ || (exports.AccountsEndLargeTrader1Response$ = AccountsEndLargeTrader1Response$ = {}));
