// @apexfintechsolutions/ascend-sdk/models/operations/accountsupdatelegalentity.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsUpdateLegalEntityResponse$ = exports.AccountsUpdateLegalEntityResponse$outboundSchema = exports.AccountsUpdateLegalEntityResponse$inboundSchema = exports.AccountsUpdateLegalEntityRequest$ = exports.AccountsUpdateLegalEntityRequest$outboundSchema = exports.AccountsUpdateLegalEntityRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsUpdateLegalEntityRequest$inboundSchema = z.object({
    legalEntity_id: z.string(),
    update_mask: z.string().optional(),
    LegalEntityUpdate: components.LegalEntityUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "legalEntity_id": "legalEntityId",
        "update_mask": "updateMask",
        "LegalEntityUpdate": "legalEntityUpdate",
    });
});
/** @internal */
exports.AccountsUpdateLegalEntityRequest$outboundSchema = z.object({
    legalEntityId: z.string(),
    updateMask: z.string().optional(),
    legalEntityUpdate: components.LegalEntityUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        legalEntityId: "legalEntity_id",
        updateMask: "update_mask",
        legalEntityUpdate: "LegalEntityUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsUpdateLegalEntityRequest$;
(function (AccountsUpdateLegalEntityRequest$) {
    /** @deprecated use `AccountsUpdateLegalEntityRequest$inboundSchema` instead. */
    AccountsUpdateLegalEntityRequest$.inboundSchema = exports.AccountsUpdateLegalEntityRequest$inboundSchema;
    /** @deprecated use `AccountsUpdateLegalEntityRequest$outboundSchema` instead. */
    AccountsUpdateLegalEntityRequest$.outboundSchema = exports.AccountsUpdateLegalEntityRequest$outboundSchema;
})(AccountsUpdateLegalEntityRequest$ || (exports.AccountsUpdateLegalEntityRequest$ = AccountsUpdateLegalEntityRequest$ = {}));
/** @internal */
exports.AccountsUpdateLegalEntityResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    LegalEntity: components.LegalEntity$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "LegalEntity": "legalEntity",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsUpdateLegalEntityResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    legalEntity: components.LegalEntity$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        legalEntity: "LegalEntity",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsUpdateLegalEntityResponse$;
(function (AccountsUpdateLegalEntityResponse$) {
    /** @deprecated use `AccountsUpdateLegalEntityResponse$inboundSchema` instead. */
    AccountsUpdateLegalEntityResponse$.inboundSchema = exports.AccountsUpdateLegalEntityResponse$inboundSchema;
    /** @deprecated use `AccountsUpdateLegalEntityResponse$outboundSchema` instead. */
    AccountsUpdateLegalEntityResponse$.outboundSchema = exports.AccountsUpdateLegalEntityResponse$outboundSchema;
})(AccountsUpdateLegalEntityResponse$ || (exports.AccountsUpdateLegalEntityResponse$ = AccountsUpdateLegalEntityResponse$ = {}));
