// @apexfintechsolutions/ascend-sdk/models/operations/accountsupdateparty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsUpdatePartyResponse$ = exports.AccountsUpdatePartyResponse$outboundSchema = exports.AccountsUpdatePartyResponse$inboundSchema = exports.AccountsUpdatePartyRequest$ = exports.AccountsUpdatePartyRequest$outboundSchema = exports.AccountsUpdatePartyRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsUpdatePartyRequest$inboundSchema = z.object({
    account_id: z.string(),
    party_id: z.string(),
    update_mask: z.string().optional(),
    PartyRequestUpdate: components.PartyRequestUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "party_id": "partyId",
        "update_mask": "updateMask",
        "PartyRequestUpdate": "partyRequestUpdate",
    });
});
/** @internal */
exports.AccountsUpdatePartyRequest$outboundSchema = z.object({
    accountId: z.string(),
    partyId: z.string(),
    updateMask: z.string().optional(),
    partyRequestUpdate: components.PartyRequestUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        partyId: "party_id",
        updateMask: "update_mask",
        partyRequestUpdate: "PartyRequestUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsUpdatePartyRequest$;
(function (AccountsUpdatePartyRequest$) {
    /** @deprecated use `AccountsUpdatePartyRequest$inboundSchema` instead. */
    AccountsUpdatePartyRequest$.inboundSchema = exports.AccountsUpdatePartyRequest$inboundSchema;
    /** @deprecated use `AccountsUpdatePartyRequest$outboundSchema` instead. */
    AccountsUpdatePartyRequest$.outboundSchema = exports.AccountsUpdatePartyRequest$outboundSchema;
})(AccountsUpdatePartyRequest$ || (exports.AccountsUpdatePartyRequest$ = AccountsUpdatePartyRequest$ = {}));
/** @internal */
exports.AccountsUpdatePartyResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Party: components.Party$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Party": "party",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsUpdatePartyResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    party: components.Party$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        party: "Party",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsUpdatePartyResponse$;
(function (AccountsUpdatePartyResponse$) {
    /** @deprecated use `AccountsUpdatePartyResponse$inboundSchema` instead. */
    AccountsUpdatePartyResponse$.inboundSchema = exports.AccountsUpdatePartyResponse$inboundSchema;
    /** @deprecated use `AccountsUpdatePartyResponse$outboundSchema` instead. */
    AccountsUpdatePartyResponse$.outboundSchema = exports.AccountsUpdatePartyResponse$outboundSchema;
})(AccountsUpdatePartyResponse$ || (exports.AccountsUpdatePartyResponse$ = AccountsUpdatePartyResponse$ = {}));
