// @apexfintechsolutions/ascend-sdk/models/operations/accountsdeletetrustedcontact.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsDeleteTrustedContactResponse$ = exports.AccountsDeleteTrustedContactResponse$outboundSchema = exports.AccountsDeleteTrustedContactResponse$inboundSchema = exports.AccountsDeleteTrustedContactRequest$ = exports.AccountsDeleteTrustedContactRequest$outboundSchema = exports.AccountsDeleteTrustedContactRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsDeleteTrustedContactRequest$inboundSchema = z.object({
    account_id: z.string(),
    trustedContact_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "trustedContact_id": "trustedContactId",
    });
});
/** @internal */
exports.AccountsDeleteTrustedContactRequest$outboundSchema = z.object({
    accountId: z.string(),
    trustedContactId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        trustedContactId: "trustedContact_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsDeleteTrustedContactRequest$;
(function (AccountsDeleteTrustedContactRequest$) {
    /** @deprecated use `AccountsDeleteTrustedContactRequest$inboundSchema` instead. */
    AccountsDeleteTrustedContactRequest$.inboundSchema = exports.AccountsDeleteTrustedContactRequest$inboundSchema;
    /** @deprecated use `AccountsDeleteTrustedContactRequest$outboundSchema` instead. */
    AccountsDeleteTrustedContactRequest$.outboundSchema = exports.AccountsDeleteTrustedContactRequest$outboundSchema;
})(AccountsDeleteTrustedContactRequest$ || (exports.AccountsDeleteTrustedContactRequest$ = AccountsDeleteTrustedContactRequest$ = {}));
/** @internal */
exports.AccountsDeleteTrustedContactResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsDeleteTrustedContactResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsDeleteTrustedContactResponse$;
(function (AccountsDeleteTrustedContactResponse$) {
    /** @deprecated use `AccountsDeleteTrustedContactResponse$inboundSchema` instead. */
    AccountsDeleteTrustedContactResponse$.inboundSchema = exports.AccountsDeleteTrustedContactResponse$inboundSchema;
    /** @deprecated use `AccountsDeleteTrustedContactResponse$outboundSchema` instead. */
    AccountsDeleteTrustedContactResponse$.outboundSchema = exports.AccountsDeleteTrustedContactResponse$outboundSchema;
})(AccountsDeleteTrustedContactResponse$ || (exports.AccountsDeleteTrustedContactResponse$ = AccountsDeleteTrustedContactResponse$ = {}));
