// @apexfintechsolutions/ascend-sdk/models/operations/accountsgetlegalentity.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsGetLegalEntityResponse$ = exports.AccountsGetLegalEntityResponse$outboundSchema = exports.AccountsGetLegalEntityResponse$inboundSchema = exports.AccountsGetLegalEntityRequest$ = exports.AccountsGetLegalEntityRequest$outboundSchema = exports.AccountsGetLegalEntityRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsGetLegalEntityRequest$inboundSchema = z.object({
    legalEntity_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "legalEntity_id": "legalEntityId",
    });
});
/** @internal */
exports.AccountsGetLegalEntityRequest$outboundSchema = z.object({
    legalEntityId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        legalEntityId: "legalEntity_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsGetLegalEntityRequest$;
(function (AccountsGetLegalEntityRequest$) {
    /** @deprecated use `AccountsGetLegalEntityRequest$inboundSchema` instead. */
    AccountsGetLegalEntityRequest$.inboundSchema = exports.AccountsGetLegalEntityRequest$inboundSchema;
    /** @deprecated use `AccountsGetLegalEntityRequest$outboundSchema` instead. */
    AccountsGetLegalEntityRequest$.outboundSchema = exports.AccountsGetLegalEntityRequest$outboundSchema;
})(AccountsGetLegalEntityRequest$ || (exports.AccountsGetLegalEntityRequest$ = AccountsGetLegalEntityRequest$ = {}));
/** @internal */
exports.AccountsGetLegalEntityResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    LegalEntity: components.LegalEntity$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "LegalEntity": "legalEntity",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsGetLegalEntityResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    legalEntity: components.LegalEntity$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        legalEntity: "LegalEntity",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsGetLegalEntityResponse$;
(function (AccountsGetLegalEntityResponse$) {
    /** @deprecated use `AccountsGetLegalEntityResponse$inboundSchema` instead. */
    AccountsGetLegalEntityResponse$.inboundSchema = exports.AccountsGetLegalEntityResponse$inboundSchema;
    /** @deprecated use `AccountsGetLegalEntityResponse$outboundSchema` instead. */
    AccountsGetLegalEntityResponse$.outboundSchema = exports.AccountsGetLegalEntityResponse$outboundSchema;
})(AccountsGetLegalEntityResponse$ || (exports.AccountsGetLegalEntityResponse$ = AccountsGetLegalEntityResponse$ = {}));
