// @apexfintechsolutions/ascend-sdk/models/operations/accountsdeleteinterestedparty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsDeleteInterestedPartyResponse$ = exports.AccountsDeleteInterestedPartyResponse$outboundSchema = exports.AccountsDeleteInterestedPartyResponse$inboundSchema = exports.AccountsDeleteInterestedPartyRequest$ = exports.AccountsDeleteInterestedPartyRequest$outboundSchema = exports.AccountsDeleteInterestedPartyRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsDeleteInterestedPartyRequest$inboundSchema = z.object({
    account_id: z.string(),
    interestedParty_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "interestedParty_id": "interestedPartyId",
    });
});
/** @internal */
exports.AccountsDeleteInterestedPartyRequest$outboundSchema = z.object({
    accountId: z.string(),
    interestedPartyId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        interestedPartyId: "interestedParty_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsDeleteInterestedPartyRequest$;
(function (AccountsDeleteInterestedPartyRequest$) {
    /** @deprecated use `AccountsDeleteInterestedPartyRequest$inboundSchema` instead. */
    AccountsDeleteInterestedPartyRequest$.inboundSchema = exports.AccountsDeleteInterestedPartyRequest$inboundSchema;
    /** @deprecated use `AccountsDeleteInterestedPartyRequest$outboundSchema` instead. */
    AccountsDeleteInterestedPartyRequest$.outboundSchema = exports.AccountsDeleteInterestedPartyRequest$outboundSchema;
})(AccountsDeleteInterestedPartyRequest$ || (exports.AccountsDeleteInterestedPartyRequest$ = AccountsDeleteInterestedPartyRequest$ = {}));
/** @internal */
exports.AccountsDeleteInterestedPartyResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsDeleteInterestedPartyResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsDeleteInterestedPartyResponse$;
(function (AccountsDeleteInterestedPartyResponse$) {
    /** @deprecated use `AccountsDeleteInterestedPartyResponse$inboundSchema` instead. */
    AccountsDeleteInterestedPartyResponse$.inboundSchema = exports.AccountsDeleteInterestedPartyResponse$inboundSchema;
    /** @deprecated use `AccountsDeleteInterestedPartyResponse$outboundSchema` instead. */
    AccountsDeleteInterestedPartyResponse$.outboundSchema = exports.AccountsDeleteInterestedPartyResponse$outboundSchema;
})(AccountsDeleteInterestedPartyResponse$ || (exports.AccountsDeleteInterestedPartyResponse$ = AccountsDeleteInterestedPartyResponse$ = {}));
