// @apexfintechsolutions/ascend-sdk/models/operations/accountsreplaceparty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountsReplacePartyResponse$ = exports.AccountsReplacePartyResponse$outboundSchema = exports.AccountsReplacePartyResponse$inboundSchema = exports.AccountsReplacePartyRequest$ = exports.AccountsReplacePartyRequest$outboundSchema = exports.AccountsReplacePartyRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AccountsReplacePartyRequest$inboundSchema = z.object({
    account_id: z.string(),
    party_id: z.string(),
    ReplacePartyRequestCreate: components.ReplacePartyRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "party_id": "partyId",
        "ReplacePartyRequestCreate": "replacePartyRequestCreate",
    });
});
/** @internal */
exports.AccountsReplacePartyRequest$outboundSchema = z.object({
    accountId: z.string(),
    partyId: z.string(),
    replacePartyRequestCreate: components.ReplacePartyRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        partyId: "party_id",
        replacePartyRequestCreate: "ReplacePartyRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsReplacePartyRequest$;
(function (AccountsReplacePartyRequest$) {
    /** @deprecated use `AccountsReplacePartyRequest$inboundSchema` instead. */
    AccountsReplacePartyRequest$.inboundSchema = exports.AccountsReplacePartyRequest$inboundSchema;
    /** @deprecated use `AccountsReplacePartyRequest$outboundSchema` instead. */
    AccountsReplacePartyRequest$.outboundSchema = exports.AccountsReplacePartyRequest$outboundSchema;
})(AccountsReplacePartyRequest$ || (exports.AccountsReplacePartyRequest$ = AccountsReplacePartyRequest$ = {}));
/** @internal */
exports.AccountsReplacePartyResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Party: components.Party$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Party": "party",
        "Status": "status",
    });
});
/** @internal */
exports.AccountsReplacePartyResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    party: components.Party$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        party: "Party",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AccountsReplacePartyResponse$;
(function (AccountsReplacePartyResponse$) {
    /** @deprecated use `AccountsReplacePartyResponse$inboundSchema` instead. */
    AccountsReplacePartyResponse$.inboundSchema = exports.AccountsReplacePartyResponse$inboundSchema;
    /** @deprecated use `AccountsReplacePartyResponse$outboundSchema` instead. */
    AccountsReplacePartyResponse$.outboundSchema = exports.AccountsReplacePartyResponse$outboundSchema;
})(AccountsReplacePartyResponse$ || (exports.AccountsReplacePartyResponse$ = AccountsReplacePartyResponse$ = {}));
