// @apexfintechsolutions/ascend-sdk/models/operations/achdepositsforceapproveachdeposit.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchDepositsForceApproveAchDepositResponse$ = exports.AchDepositsForceApproveAchDepositResponse$outboundSchema = exports.AchDepositsForceApproveAchDepositResponse$inboundSchema = exports.AchDepositsForceApproveAchDepositRequest$ = exports.AchDepositsForceApproveAchDepositRequest$outboundSchema = exports.AchDepositsForceApproveAchDepositRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AchDepositsForceApproveAchDepositRequest$inboundSchema = z.object({
    account_id: z.string(),
    achDeposit_id: z.string(),
    ForceApproveAchDepositRequestCreate: components.ForceApproveAchDepositRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "achDeposit_id": "achDepositId",
        "ForceApproveAchDepositRequestCreate": "forceApproveAchDepositRequestCreate",
    });
});
/** @internal */
exports.AchDepositsForceApproveAchDepositRequest$outboundSchema = z.object({
    accountId: z.string(),
    achDepositId: z.string(),
    forceApproveAchDepositRequestCreate: components.ForceApproveAchDepositRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        achDepositId: "achDeposit_id",
        forceApproveAchDepositRequestCreate: "ForceApproveAchDepositRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositsForceApproveAchDepositRequest$;
(function (AchDepositsForceApproveAchDepositRequest$) {
    /** @deprecated use `AchDepositsForceApproveAchDepositRequest$inboundSchema` instead. */
    AchDepositsForceApproveAchDepositRequest$.inboundSchema = exports.AchDepositsForceApproveAchDepositRequest$inboundSchema;
    /** @deprecated use `AchDepositsForceApproveAchDepositRequest$outboundSchema` instead. */
    AchDepositsForceApproveAchDepositRequest$.outboundSchema = exports.AchDepositsForceApproveAchDepositRequest$outboundSchema;
})(AchDepositsForceApproveAchDepositRequest$ || (exports.AchDepositsForceApproveAchDepositRequest$ = AchDepositsForceApproveAchDepositRequest$ = {}));
/** @internal */
exports.AchDepositsForceApproveAchDepositResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchDeposit: components.AchDeposit$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "AchDeposit": "achDeposit",
        "Status": "status",
    });
});
/** @internal */
exports.AchDepositsForceApproveAchDepositResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achDeposit: components.AchDeposit$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        achDeposit: "AchDeposit",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositsForceApproveAchDepositResponse$;
(function (AchDepositsForceApproveAchDepositResponse$) {
    /** @deprecated use `AchDepositsForceApproveAchDepositResponse$inboundSchema` instead. */
    AchDepositsForceApproveAchDepositResponse$.inboundSchema = exports.AchDepositsForceApproveAchDepositResponse$inboundSchema;
    /** @deprecated use `AchDepositsForceApproveAchDepositResponse$outboundSchema` instead. */
    AchDepositsForceApproveAchDepositResponse$.outboundSchema = exports.AchDepositsForceApproveAchDepositResponse$outboundSchema;
})(AchDepositsForceApproveAchDepositResponse$ || (exports.AchDepositsForceApproveAchDepositResponse$ = AchDepositsForceApproveAchDepositResponse$ = {}));
