// @apexfintechsolutions/ascend-sdk/models/operations/achdepositscancelachdeposit.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchDepositsCancelAchDepositResponse$ = exports.AchDepositsCancelAchDepositResponse$outboundSchema = exports.AchDepositsCancelAchDepositResponse$inboundSchema = exports.AchDepositsCancelAchDepositRequest$ = exports.AchDepositsCancelAchDepositRequest$outboundSchema = exports.AchDepositsCancelAchDepositRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AchDepositsCancelAchDepositRequest$inboundSchema = z.object({
    account_id: z.string(),
    achDeposit_id: z.string(),
    CancelAchDepositRequestCreate: components.CancelAchDepositRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "achDeposit_id": "achDepositId",
        "CancelAchDepositRequestCreate": "cancelAchDepositRequestCreate",
    });
});
/** @internal */
exports.AchDepositsCancelAchDepositRequest$outboundSchema = z.object({
    accountId: z.string(),
    achDepositId: z.string(),
    cancelAchDepositRequestCreate: components.CancelAchDepositRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        achDepositId: "achDeposit_id",
        cancelAchDepositRequestCreate: "CancelAchDepositRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositsCancelAchDepositRequest$;
(function (AchDepositsCancelAchDepositRequest$) {
    /** @deprecated use `AchDepositsCancelAchDepositRequest$inboundSchema` instead. */
    AchDepositsCancelAchDepositRequest$.inboundSchema = exports.AchDepositsCancelAchDepositRequest$inboundSchema;
    /** @deprecated use `AchDepositsCancelAchDepositRequest$outboundSchema` instead. */
    AchDepositsCancelAchDepositRequest$.outboundSchema = exports.AchDepositsCancelAchDepositRequest$outboundSchema;
})(AchDepositsCancelAchDepositRequest$ || (exports.AchDepositsCancelAchDepositRequest$ = AchDepositsCancelAchDepositRequest$ = {}));
/** @internal */
exports.AchDepositsCancelAchDepositResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchDeposit: components.AchDeposit$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "AchDeposit": "achDeposit",
        "Status": "status",
    });
});
/** @internal */
exports.AchDepositsCancelAchDepositResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achDeposit: components.AchDeposit$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        achDeposit: "AchDeposit",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositsCancelAchDepositResponse$;
(function (AchDepositsCancelAchDepositResponse$) {
    /** @deprecated use `AchDepositsCancelAchDepositResponse$inboundSchema` instead. */
    AchDepositsCancelAchDepositResponse$.inboundSchema = exports.AchDepositsCancelAchDepositResponse$inboundSchema;
    /** @deprecated use `AchDepositsCancelAchDepositResponse$outboundSchema` instead. */
    AchDepositsCancelAchDepositResponse$.outboundSchema = exports.AchDepositsCancelAchDepositResponse$outboundSchema;
})(AchDepositsCancelAchDepositResponse$ || (exports.AchDepositsCancelAchDepositResponse$ = AchDepositsCancelAchDepositResponse$ = {}));
