// @apexfintechsolutions/ascend-sdk/models/operations/achdepositsgetachdeposit.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchDepositsGetAchDepositResponse$ = exports.AchDepositsGetAchDepositResponse$outboundSchema = exports.AchDepositsGetAchDepositResponse$inboundSchema = exports.AchDepositsGetAchDepositRequest$ = exports.AchDepositsGetAchDepositRequest$outboundSchema = exports.AchDepositsGetAchDepositRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AchDepositsGetAchDepositRequest$inboundSchema = z.object({
    account_id: z.string(),
    achDeposit_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "achDeposit_id": "achDepositId",
    });
});
/** @internal */
exports.AchDepositsGetAchDepositRequest$outboundSchema = z.object({
    accountId: z.string(),
    achDepositId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        achDepositId: "achDeposit_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositsGetAchDepositRequest$;
(function (AchDepositsGetAchDepositRequest$) {
    /** @deprecated use `AchDepositsGetAchDepositRequest$inboundSchema` instead. */
    AchDepositsGetAchDepositRequest$.inboundSchema = exports.AchDepositsGetAchDepositRequest$inboundSchema;
    /** @deprecated use `AchDepositsGetAchDepositRequest$outboundSchema` instead. */
    AchDepositsGetAchDepositRequest$.outboundSchema = exports.AchDepositsGetAchDepositRequest$outboundSchema;
})(AchDepositsGetAchDepositRequest$ || (exports.AchDepositsGetAchDepositRequest$ = AchDepositsGetAchDepositRequest$ = {}));
/** @internal */
exports.AchDepositsGetAchDepositResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchDeposit: components.AchDeposit$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "AchDeposit": "achDeposit",
        "Status": "status",
    });
});
/** @internal */
exports.AchDepositsGetAchDepositResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achDeposit: components.AchDeposit$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        achDeposit: "AchDeposit",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositsGetAchDepositResponse$;
(function (AchDepositsGetAchDepositResponse$) {
    /** @deprecated use `AchDepositsGetAchDepositResponse$inboundSchema` instead. */
    AchDepositsGetAchDepositResponse$.inboundSchema = exports.AchDepositsGetAchDepositResponse$inboundSchema;
    /** @deprecated use `AchDepositsGetAchDepositResponse$outboundSchema` instead. */
    AchDepositsGetAchDepositResponse$.outboundSchema = exports.AchDepositsGetAchDepositResponse$outboundSchema;
})(AchDepositsGetAchDepositResponse$ || (exports.AchDepositsGetAchDepositResponse$ = AchDepositsGetAchDepositResponse$ = {}));
