// @apexfintechsolutions/ascend-sdk/models/operations/achdepositschedulescreateachdepositschedule.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchDepositSchedulesCreateAchDepositScheduleResponse$ = exports.AchDepositSchedulesCreateAchDepositScheduleResponse$outboundSchema = exports.AchDepositSchedulesCreateAchDepositScheduleResponse$inboundSchema = exports.AchDepositSchedulesCreateAchDepositScheduleRequest$ = exports.AchDepositSchedulesCreateAchDepositScheduleRequest$outboundSchema = exports.AchDepositSchedulesCreateAchDepositScheduleRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AchDepositSchedulesCreateAchDepositScheduleRequest$inboundSchema = z.object({
    account_id: z.string(),
    AchDepositScheduleCreate: components.AchDepositScheduleCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "AchDepositScheduleCreate": "achDepositScheduleCreate",
    });
});
/** @internal */
exports.AchDepositSchedulesCreateAchDepositScheduleRequest$outboundSchema = z.object({
    accountId: z.string(),
    achDepositScheduleCreate: components.AchDepositScheduleCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        achDepositScheduleCreate: "AchDepositScheduleCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositSchedulesCreateAchDepositScheduleRequest$;
(function (AchDepositSchedulesCreateAchDepositScheduleRequest$) {
    /** @deprecated use `AchDepositSchedulesCreateAchDepositScheduleRequest$inboundSchema` instead. */
    AchDepositSchedulesCreateAchDepositScheduleRequest$.inboundSchema = exports.AchDepositSchedulesCreateAchDepositScheduleRequest$inboundSchema;
    /** @deprecated use `AchDepositSchedulesCreateAchDepositScheduleRequest$outboundSchema` instead. */
    AchDepositSchedulesCreateAchDepositScheduleRequest$.outboundSchema = exports.AchDepositSchedulesCreateAchDepositScheduleRequest$outboundSchema;
})(AchDepositSchedulesCreateAchDepositScheduleRequest$ || (exports.AchDepositSchedulesCreateAchDepositScheduleRequest$ = AchDepositSchedulesCreateAchDepositScheduleRequest$ = {}));
/** @internal */
exports.AchDepositSchedulesCreateAchDepositScheduleResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchDepositSchedule: components.AchDepositSchedule$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "AchDepositSchedule": "achDepositSchedule",
        "Status": "status",
    });
});
/** @internal */
exports.AchDepositSchedulesCreateAchDepositScheduleResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achDepositSchedule: components.AchDepositSchedule$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        achDepositSchedule: "AchDepositSchedule",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositSchedulesCreateAchDepositScheduleResponse$;
(function (AchDepositSchedulesCreateAchDepositScheduleResponse$) {
    /** @deprecated use `AchDepositSchedulesCreateAchDepositScheduleResponse$inboundSchema` instead. */
    AchDepositSchedulesCreateAchDepositScheduleResponse$.inboundSchema = exports.AchDepositSchedulesCreateAchDepositScheduleResponse$inboundSchema;
    /** @deprecated use `AchDepositSchedulesCreateAchDepositScheduleResponse$outboundSchema` instead. */
    AchDepositSchedulesCreateAchDepositScheduleResponse$.outboundSchema = exports.AchDepositSchedulesCreateAchDepositScheduleResponse$outboundSchema;
})(AchDepositSchedulesCreateAchDepositScheduleResponse$ || (exports.AchDepositSchedulesCreateAchDepositScheduleResponse$ = AchDepositSchedulesCreateAchDepositScheduleResponse$ = {}));
