// @apexfintechsolutions/ascend-sdk/models/operations/achdepositscreateachdeposit.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchDepositsCreateAchDepositResponse$ = exports.AchDepositsCreateAchDepositResponse$outboundSchema = exports.AchDepositsCreateAchDepositResponse$inboundSchema = exports.AchDepositsCreateAchDepositRequest$ = exports.AchDepositsCreateAchDepositRequest$outboundSchema = exports.AchDepositsCreateAchDepositRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AchDepositsCreateAchDepositRequest$inboundSchema = z.object({
    account_id: z.string(),
    AchDepositCreate: components.AchDepositCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "AchDepositCreate": "achDepositCreate",
    });
});
/** @internal */
exports.AchDepositsCreateAchDepositRequest$outboundSchema = z.object({
    accountId: z.string(),
    achDepositCreate: components.AchDepositCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        achDepositCreate: "AchDepositCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositsCreateAchDepositRequest$;
(function (AchDepositsCreateAchDepositRequest$) {
    /** @deprecated use `AchDepositsCreateAchDepositRequest$inboundSchema` instead. */
    AchDepositsCreateAchDepositRequest$.inboundSchema = exports.AchDepositsCreateAchDepositRequest$inboundSchema;
    /** @deprecated use `AchDepositsCreateAchDepositRequest$outboundSchema` instead. */
    AchDepositsCreateAchDepositRequest$.outboundSchema = exports.AchDepositsCreateAchDepositRequest$outboundSchema;
})(AchDepositsCreateAchDepositRequest$ || (exports.AchDepositsCreateAchDepositRequest$ = AchDepositsCreateAchDepositRequest$ = {}));
/** @internal */
exports.AchDepositsCreateAchDepositResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchDeposit: components.AchDeposit$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "AchDeposit": "achDeposit",
        "Status": "status",
    });
});
/** @internal */
exports.AchDepositsCreateAchDepositResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achDeposit: components.AchDeposit$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        achDeposit: "AchDeposit",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchDepositsCreateAchDepositResponse$;
(function (AchDepositsCreateAchDepositResponse$) {
    /** @deprecated use `AchDepositsCreateAchDepositResponse$inboundSchema` instead. */
    AchDepositsCreateAchDepositResponse$.inboundSchema = exports.AchDepositsCreateAchDepositResponse$inboundSchema;
    /** @deprecated use `AchDepositsCreateAchDepositResponse$outboundSchema` instead. */
    AchDepositsCreateAchDepositResponse$.outboundSchema = exports.AchDepositsCreateAchDepositResponse$outboundSchema;
})(AchDepositsCreateAchDepositResponse$ || (exports.AchDepositsCreateAchDepositResponse$ = AchDepositsCreateAchDepositResponse$ = {}));
