// @apexfintechsolutions/ascend-sdk/models/operations/achwithdrawalschedulesgetachwithdrawalschedule.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$ = exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$outboundSchema = exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$inboundSchema = exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$ = exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$outboundSchema = exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$inboundSchema = z.object({
    account_id: z.string(),
    achWithdrawalSchedule_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "achWithdrawalSchedule_id": "achWithdrawalScheduleId",
    });
});
/** @internal */
exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$outboundSchema = z.object({
    accountId: z.string(),
    achWithdrawalScheduleId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        achWithdrawalScheduleId: "achWithdrawalSchedule_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$;
(function (AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$) {
    /** @deprecated use `AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$inboundSchema` instead. */
    AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$.inboundSchema = exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$inboundSchema;
    /** @deprecated use `AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$outboundSchema` instead. */
    AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$.outboundSchema = exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$outboundSchema;
})(AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$ || (exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$ = AchWithdrawalSchedulesGetAchWithdrawalScheduleRequest$ = {}));
/** @internal */
exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchWithdrawalSchedule: components.AchWithdrawalSchedule$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "AchWithdrawalSchedule": "achWithdrawalSchedule",
        "Status": "status",
    });
});
/** @internal */
exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achWithdrawalSchedule: components.AchWithdrawalSchedule$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        achWithdrawalSchedule: "AchWithdrawalSchedule",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$;
(function (AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$) {
    /** @deprecated use `AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$inboundSchema` instead. */
    AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$.inboundSchema = exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$inboundSchema;
    /** @deprecated use `AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$outboundSchema` instead. */
    AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$.outboundSchema = exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$outboundSchema;
})(AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$ || (exports.AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$ = AchWithdrawalSchedulesGetAchWithdrawalScheduleResponse$ = {}));
