// @apexfintechsolutions/ascend-sdk/models/operations/achwithdrawalschedulesupdateachwithdrawalschedule.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$ = exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$outboundSchema = exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$inboundSchema = exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$ = exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$outboundSchema = exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$inboundSchema = z.object({
    account_id: z.string(),
    achWithdrawalSchedule_id: z.string(),
    update_mask: z.string().optional(),
    AchWithdrawalScheduleUpdate: components.AchWithdrawalScheduleUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "achWithdrawalSchedule_id": "achWithdrawalScheduleId",
        "update_mask": "updateMask",
        "AchWithdrawalScheduleUpdate": "achWithdrawalScheduleUpdate",
    });
});
/** @internal */
exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$outboundSchema = z.object({
    accountId: z.string(),
    achWithdrawalScheduleId: z.string(),
    updateMask: z.string().optional(),
    achWithdrawalScheduleUpdate: components.AchWithdrawalScheduleUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        achWithdrawalScheduleId: "achWithdrawalSchedule_id",
        updateMask: "update_mask",
        achWithdrawalScheduleUpdate: "AchWithdrawalScheduleUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$;
(function (AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$) {
    /** @deprecated use `AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$inboundSchema` instead. */
    AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$.inboundSchema = exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$inboundSchema;
    /** @deprecated use `AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$outboundSchema` instead. */
    AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$.outboundSchema = exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$outboundSchema;
})(AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$ || (exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$ = AchWithdrawalSchedulesUpdateAchWithdrawalScheduleRequest$ = {}));
/** @internal */
exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    AchWithdrawalSchedule: components.AchWithdrawalSchedule$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "AchWithdrawalSchedule": "achWithdrawalSchedule",
        "Status": "status",
    });
});
/** @internal */
exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    achWithdrawalSchedule: components.AchWithdrawalSchedule$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        achWithdrawalSchedule: "AchWithdrawalSchedule",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$;
(function (AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$) {
    /** @deprecated use `AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$inboundSchema` instead. */
    AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$.inboundSchema = exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$inboundSchema;
    /** @deprecated use `AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$outboundSchema` instead. */
    AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$.outboundSchema = exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$outboundSchema;
})(AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$ || (exports.AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$ = AchWithdrawalSchedulesUpdateAchWithdrawalScheduleResponse$ = {}));
