// @apexfintechsolutions/ascend-sdk/models/operations/authenticationlistsigningkeys.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticationListSigningKeysResponse$ = exports.AuthenticationListSigningKeysResponse$outboundSchema = exports.AuthenticationListSigningKeysResponse$inboundSchema = exports.AuthenticationListSigningKeysRequest$ = exports.AuthenticationListSigningKeysRequest$outboundSchema = exports.AuthenticationListSigningKeysRequest$inboundSchema = exports.AuthenticationListSigningKeysSecurity$ = exports.AuthenticationListSigningKeysSecurity$outboundSchema = exports.AuthenticationListSigningKeysSecurity$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AuthenticationListSigningKeysSecurity$inboundSchema = z.object({
    ApiKeyAuth: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ApiKeyAuth": "apiKeyAuth",
    });
});
/** @internal */
exports.AuthenticationListSigningKeysSecurity$outboundSchema = z.object({
    apiKeyAuth: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        apiKeyAuth: "ApiKeyAuth",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AuthenticationListSigningKeysSecurity$;
(function (AuthenticationListSigningKeysSecurity$) {
    /** @deprecated use `AuthenticationListSigningKeysSecurity$inboundSchema` instead. */
    AuthenticationListSigningKeysSecurity$.inboundSchema = exports.AuthenticationListSigningKeysSecurity$inboundSchema;
    /** @deprecated use `AuthenticationListSigningKeysSecurity$outboundSchema` instead. */
    AuthenticationListSigningKeysSecurity$.outboundSchema = exports.AuthenticationListSigningKeysSecurity$outboundSchema;
})(AuthenticationListSigningKeysSecurity$ || (exports.AuthenticationListSigningKeysSecurity$ = AuthenticationListSigningKeysSecurity$ = {}));
/** @internal */
exports.AuthenticationListSigningKeysRequest$inboundSchema = z.object({
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "page_size": "pageSize",
        "page_token": "pageToken",
    });
});
/** @internal */
exports.AuthenticationListSigningKeysRequest$outboundSchema = z.object({
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        pageSize: "page_size",
        pageToken: "page_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AuthenticationListSigningKeysRequest$;
(function (AuthenticationListSigningKeysRequest$) {
    /** @deprecated use `AuthenticationListSigningKeysRequest$inboundSchema` instead. */
    AuthenticationListSigningKeysRequest$.inboundSchema = exports.AuthenticationListSigningKeysRequest$inboundSchema;
    /** @deprecated use `AuthenticationListSigningKeysRequest$outboundSchema` instead. */
    AuthenticationListSigningKeysRequest$.outboundSchema = exports.AuthenticationListSigningKeysRequest$outboundSchema;
})(AuthenticationListSigningKeysRequest$ || (exports.AuthenticationListSigningKeysRequest$ = AuthenticationListSigningKeysRequest$ = {}));
/** @internal */
exports.AuthenticationListSigningKeysResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListSigningKeysResponse: components.ListSigningKeysResponse$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "ListSigningKeysResponse": "listSigningKeysResponse",
        "Status": "status",
    });
});
/** @internal */
exports.AuthenticationListSigningKeysResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listSigningKeysResponse: components.ListSigningKeysResponse$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        listSigningKeysResponse: "ListSigningKeysResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AuthenticationListSigningKeysResponse$;
(function (AuthenticationListSigningKeysResponse$) {
    /** @deprecated use `AuthenticationListSigningKeysResponse$inboundSchema` instead. */
    AuthenticationListSigningKeysResponse$.inboundSchema = exports.AuthenticationListSigningKeysResponse$inboundSchema;
    /** @deprecated use `AuthenticationListSigningKeysResponse$outboundSchema` instead. */
    AuthenticationListSigningKeysResponse$.outboundSchema = exports.AuthenticationListSigningKeysResponse$outboundSchema;
})(AuthenticationListSigningKeysResponse$ || (exports.AuthenticationListSigningKeysResponse$ = AuthenticationListSigningKeysResponse$ = {}));
