// @apexfintechsolutions/ascend-sdk/models/operations/bankrelationshipslistbankrelationships.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankRelationshipsListBankRelationshipsResponse$ = exports.BankRelationshipsListBankRelationshipsResponse$outboundSchema = exports.BankRelationshipsListBankRelationshipsResponse$inboundSchema = exports.BankRelationshipsListBankRelationshipsRequest$ = exports.BankRelationshipsListBankRelationshipsRequest$outboundSchema = exports.BankRelationshipsListBankRelationshipsRequest$inboundSchema = exports.State$ = exports.State$outboundSchema = exports.State$inboundSchema = exports.State = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/**
 * The state of bank relationships to filter by. Unspecified returns relationships of all states.
 */
var State;
(function (State) {
    State["StateUnspecified"] = "STATE_UNSPECIFIED";
    State["Pending"] = "PENDING";
    State["Approved"] = "APPROVED";
    State["Canceled"] = "CANCELED";
    State["Rejected"] = "REJECTED";
})(State || (exports.State = State = {}));
/** @internal */
exports.State$inboundSchema = z.nativeEnum(State);
/** @internal */
exports.State$outboundSchema = exports.State$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var State$;
(function (State$) {
    /** @deprecated use `State$inboundSchema` instead. */
    State$.inboundSchema = exports.State$inboundSchema;
    /** @deprecated use `State$outboundSchema` instead. */
    State$.outboundSchema = exports.State$outboundSchema;
})(State$ || (exports.State$ = State$ = {}));
/** @internal */
exports.BankRelationshipsListBankRelationshipsRequest$inboundSchema = z.object({
    account_id: z.string(),
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
    state: exports.State$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "page_size": "pageSize",
        "page_token": "pageToken",
    });
});
/** @internal */
exports.BankRelationshipsListBankRelationshipsRequest$outboundSchema = z.object({
    accountId: z.string(),
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
    state: exports.State$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        pageSize: "page_size",
        pageToken: "page_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BankRelationshipsListBankRelationshipsRequest$;
(function (BankRelationshipsListBankRelationshipsRequest$) {
    /** @deprecated use `BankRelationshipsListBankRelationshipsRequest$inboundSchema` instead. */
    BankRelationshipsListBankRelationshipsRequest$.inboundSchema = exports.BankRelationshipsListBankRelationshipsRequest$inboundSchema;
    /** @deprecated use `BankRelationshipsListBankRelationshipsRequest$outboundSchema` instead. */
    BankRelationshipsListBankRelationshipsRequest$.outboundSchema = exports.BankRelationshipsListBankRelationshipsRequest$outboundSchema;
})(BankRelationshipsListBankRelationshipsRequest$ || (exports.BankRelationshipsListBankRelationshipsRequest$ = BankRelationshipsListBankRelationshipsRequest$ = {}));
/** @internal */
exports.BankRelationshipsListBankRelationshipsResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListBankRelationshipsResponse: components
        .ListBankRelationshipsResponse$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "ListBankRelationshipsResponse": "listBankRelationshipsResponse",
        "Status": "status",
    });
});
/** @internal */
exports.BankRelationshipsListBankRelationshipsResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listBankRelationshipsResponse: components
        .ListBankRelationshipsResponse$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        listBankRelationshipsResponse: "ListBankRelationshipsResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BankRelationshipsListBankRelationshipsResponse$;
(function (BankRelationshipsListBankRelationshipsResponse$) {
    /** @deprecated use `BankRelationshipsListBankRelationshipsResponse$inboundSchema` instead. */
    BankRelationshipsListBankRelationshipsResponse$.inboundSchema = exports.BankRelationshipsListBankRelationshipsResponse$inboundSchema;
    /** @deprecated use `BankRelationshipsListBankRelationshipsResponse$outboundSchema` instead. */
    BankRelationshipsListBankRelationshipsResponse$.outboundSchema = exports.BankRelationshipsListBankRelationshipsResponse$outboundSchema;
})(BankRelationshipsListBankRelationshipsResponse$ || (exports.BankRelationshipsListBankRelationshipsResponse$ = BankRelationshipsListBankRelationshipsResponse$ = {}));
