// @apexfintechsolutions/ascend-sdk/models/operations/bankrelationshipsgetbankrelationship.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankRelationshipsGetBankRelationshipResponse$ = exports.BankRelationshipsGetBankRelationshipResponse$outboundSchema = exports.BankRelationshipsGetBankRelationshipResponse$inboundSchema = exports.BankRelationshipsGetBankRelationshipRequest$ = exports.BankRelationshipsGetBankRelationshipRequest$outboundSchema = exports.BankRelationshipsGetBankRelationshipRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BankRelationshipsGetBankRelationshipRequest$inboundSchema = z.object({
    account_id: z.string(),
    bankRelationship_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "bankRelationship_id": "bankRelationshipId",
    });
});
/** @internal */
exports.BankRelationshipsGetBankRelationshipRequest$outboundSchema = z.object({
    accountId: z.string(),
    bankRelationshipId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        bankRelationshipId: "bankRelationship_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BankRelationshipsGetBankRelationshipRequest$;
(function (BankRelationshipsGetBankRelationshipRequest$) {
    /** @deprecated use `BankRelationshipsGetBankRelationshipRequest$inboundSchema` instead. */
    BankRelationshipsGetBankRelationshipRequest$.inboundSchema = exports.BankRelationshipsGetBankRelationshipRequest$inboundSchema;
    /** @deprecated use `BankRelationshipsGetBankRelationshipRequest$outboundSchema` instead. */
    BankRelationshipsGetBankRelationshipRequest$.outboundSchema = exports.BankRelationshipsGetBankRelationshipRequest$outboundSchema;
})(BankRelationshipsGetBankRelationshipRequest$ || (exports.BankRelationshipsGetBankRelationshipRequest$ = BankRelationshipsGetBankRelationshipRequest$ = {}));
/** @internal */
exports.BankRelationshipsGetBankRelationshipResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    BankRelationship: components.BankRelationship$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "BankRelationship": "bankRelationship",
        "Status": "status",
    });
});
/** @internal */
exports.BankRelationshipsGetBankRelationshipResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    bankRelationship: components.BankRelationship$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        bankRelationship: "BankRelationship",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BankRelationshipsGetBankRelationshipResponse$;
(function (BankRelationshipsGetBankRelationshipResponse$) {
    /** @deprecated use `BankRelationshipsGetBankRelationshipResponse$inboundSchema` instead. */
    BankRelationshipsGetBankRelationshipResponse$.inboundSchema = exports.BankRelationshipsGetBankRelationshipResponse$inboundSchema;
    /** @deprecated use `BankRelationshipsGetBankRelationshipResponse$outboundSchema` instead. */
    BankRelationshipsGetBankRelationshipResponse$.outboundSchema = exports.BankRelationshipsGetBankRelationshipResponse$outboundSchema;
})(BankRelationshipsGetBankRelationshipResponse$ || (exports.BankRelationshipsGetBankRelationshipResponse$ = BankRelationshipsGetBankRelationshipResponse$ = {}));
