// @apexfintechsolutions/ascend-sdk/models/operations/basketordersservicelistcompressedorders.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasketOrdersServiceListCompressedOrdersResponse$ = exports.BasketOrdersServiceListCompressedOrdersResponse$outboundSchema = exports.BasketOrdersServiceListCompressedOrdersResponse$inboundSchema = exports.BasketOrdersServiceListCompressedOrdersRequest$ = exports.BasketOrdersServiceListCompressedOrdersRequest$outboundSchema = exports.BasketOrdersServiceListCompressedOrdersRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BasketOrdersServiceListCompressedOrdersRequest$inboundSchema = z.object({
    correspondent_id: z.string(),
    basket_id: z.string(),
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "correspondent_id": "correspondentId",
        "basket_id": "basketId",
        "page_size": "pageSize",
        "page_token": "pageToken",
    });
});
/** @internal */
exports.BasketOrdersServiceListCompressedOrdersRequest$outboundSchema = z.object({
    correspondentId: z.string(),
    basketId: z.string(),
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        correspondentId: "correspondent_id",
        basketId: "basket_id",
        pageSize: "page_size",
        pageToken: "page_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BasketOrdersServiceListCompressedOrdersRequest$;
(function (BasketOrdersServiceListCompressedOrdersRequest$) {
    /** @deprecated use `BasketOrdersServiceListCompressedOrdersRequest$inboundSchema` instead. */
    BasketOrdersServiceListCompressedOrdersRequest$.inboundSchema = exports.BasketOrdersServiceListCompressedOrdersRequest$inboundSchema;
    /** @deprecated use `BasketOrdersServiceListCompressedOrdersRequest$outboundSchema` instead. */
    BasketOrdersServiceListCompressedOrdersRequest$.outboundSchema = exports.BasketOrdersServiceListCompressedOrdersRequest$outboundSchema;
})(BasketOrdersServiceListCompressedOrdersRequest$ || (exports.BasketOrdersServiceListCompressedOrdersRequest$ = BasketOrdersServiceListCompressedOrdersRequest$ = {}));
/** @internal */
exports.BasketOrdersServiceListCompressedOrdersResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListCompressedOrdersResponse: components
        .ListCompressedOrdersResponse$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "ListCompressedOrdersResponse": "listCompressedOrdersResponse",
        "Status": "status",
    });
});
/** @internal */
exports.BasketOrdersServiceListCompressedOrdersResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listCompressedOrdersResponse: components
        .ListCompressedOrdersResponse$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        listCompressedOrdersResponse: "ListCompressedOrdersResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BasketOrdersServiceListCompressedOrdersResponse$;
(function (BasketOrdersServiceListCompressedOrdersResponse$) {
    /** @deprecated use `BasketOrdersServiceListCompressedOrdersResponse$inboundSchema` instead. */
    BasketOrdersServiceListCompressedOrdersResponse$.inboundSchema = exports.BasketOrdersServiceListCompressedOrdersResponse$inboundSchema;
    /** @deprecated use `BasketOrdersServiceListCompressedOrdersResponse$outboundSchema` instead. */
    BasketOrdersServiceListCompressedOrdersResponse$.outboundSchema = exports.BasketOrdersServiceListCompressedOrdersResponse$outboundSchema;
})(BasketOrdersServiceListCompressedOrdersResponse$ || (exports.BasketOrdersServiceListCompressedOrdersResponse$ = BasketOrdersServiceListCompressedOrdersResponse$ = {}));
