// @apexfintechsolutions/ascend-sdk/models/operations/basketordersservicegetbasket.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasketOrdersServiceGetBasketResponse$ = exports.BasketOrdersServiceGetBasketResponse$outboundSchema = exports.BasketOrdersServiceGetBasketResponse$inboundSchema = exports.BasketOrdersServiceGetBasketRequest$ = exports.BasketOrdersServiceGetBasketRequest$outboundSchema = exports.BasketOrdersServiceGetBasketRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BasketOrdersServiceGetBasketRequest$inboundSchema = z.object({
    correspondent_id: z.string(),
    basket_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "correspondent_id": "correspondentId",
        "basket_id": "basketId",
    });
});
/** @internal */
exports.BasketOrdersServiceGetBasketRequest$outboundSchema = z.object({
    correspondentId: z.string(),
    basketId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        correspondentId: "correspondent_id",
        basketId: "basket_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BasketOrdersServiceGetBasketRequest$;
(function (BasketOrdersServiceGetBasketRequest$) {
    /** @deprecated use `BasketOrdersServiceGetBasketRequest$inboundSchema` instead. */
    BasketOrdersServiceGetBasketRequest$.inboundSchema = exports.BasketOrdersServiceGetBasketRequest$inboundSchema;
    /** @deprecated use `BasketOrdersServiceGetBasketRequest$outboundSchema` instead. */
    BasketOrdersServiceGetBasketRequest$.outboundSchema = exports.BasketOrdersServiceGetBasketRequest$outboundSchema;
})(BasketOrdersServiceGetBasketRequest$ || (exports.BasketOrdersServiceGetBasketRequest$ = BasketOrdersServiceGetBasketRequest$ = {}));
/** @internal */
exports.BasketOrdersServiceGetBasketResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Basket: components.Basket$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Basket": "basket",
        "Status": "status",
    });
});
/** @internal */
exports.BasketOrdersServiceGetBasketResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    basket: components.Basket$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        basket: "Basket",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BasketOrdersServiceGetBasketResponse$;
(function (BasketOrdersServiceGetBasketResponse$) {
    /** @deprecated use `BasketOrdersServiceGetBasketResponse$inboundSchema` instead. */
    BasketOrdersServiceGetBasketResponse$.inboundSchema = exports.BasketOrdersServiceGetBasketResponse$inboundSchema;
    /** @deprecated use `BasketOrdersServiceGetBasketResponse$outboundSchema` instead. */
    BasketOrdersServiceGetBasketResponse$.outboundSchema = exports.BasketOrdersServiceGetBasketResponse$outboundSchema;
})(BasketOrdersServiceGetBasketResponse$ || (exports.BasketOrdersServiceGetBasketResponse$ = BasketOrdersServiceGetBasketResponse$ = {}));
