// @apexfintechsolutions/ascend-sdk/models/operations/basketordersservicecreatebasket.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasketOrdersServiceCreateBasketResponse$ = exports.BasketOrdersServiceCreateBasketResponse$outboundSchema = exports.BasketOrdersServiceCreateBasketResponse$inboundSchema = exports.BasketOrdersServiceCreateBasketRequest$ = exports.BasketOrdersServiceCreateBasketRequest$outboundSchema = exports.BasketOrdersServiceCreateBasketRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BasketOrdersServiceCreateBasketRequest$inboundSchema = z.object({
    correspondent_id: z.string(),
    BasketCreate: components.BasketCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "correspondent_id": "correspondentId",
        "BasketCreate": "basketCreate",
    });
});
/** @internal */
exports.BasketOrdersServiceCreateBasketRequest$outboundSchema = z.object({
    correspondentId: z.string(),
    basketCreate: components.BasketCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        correspondentId: "correspondent_id",
        basketCreate: "BasketCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BasketOrdersServiceCreateBasketRequest$;
(function (BasketOrdersServiceCreateBasketRequest$) {
    /** @deprecated use `BasketOrdersServiceCreateBasketRequest$inboundSchema` instead. */
    BasketOrdersServiceCreateBasketRequest$.inboundSchema = exports.BasketOrdersServiceCreateBasketRequest$inboundSchema;
    /** @deprecated use `BasketOrdersServiceCreateBasketRequest$outboundSchema` instead. */
    BasketOrdersServiceCreateBasketRequest$.outboundSchema = exports.BasketOrdersServiceCreateBasketRequest$outboundSchema;
})(BasketOrdersServiceCreateBasketRequest$ || (exports.BasketOrdersServiceCreateBasketRequest$ = BasketOrdersServiceCreateBasketRequest$ = {}));
/** @internal */
exports.BasketOrdersServiceCreateBasketResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Basket: components.Basket$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Basket": "basket",
        "Status": "status",
    });
});
/** @internal */
exports.BasketOrdersServiceCreateBasketResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    basket: components.Basket$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        basket: "Basket",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BasketOrdersServiceCreateBasketResponse$;
(function (BasketOrdersServiceCreateBasketResponse$) {
    /** @deprecated use `BasketOrdersServiceCreateBasketResponse$inboundSchema` instead. */
    BasketOrdersServiceCreateBasketResponse$.inboundSchema = exports.BasketOrdersServiceCreateBasketResponse$inboundSchema;
    /** @deprecated use `BasketOrdersServiceCreateBasketResponse$outboundSchema` instead. */
    BasketOrdersServiceCreateBasketResponse$.outboundSchema = exports.BasketOrdersServiceCreateBasketResponse$outboundSchema;
})(BasketOrdersServiceCreateBasketResponse$ || (exports.BasketOrdersServiceCreateBasketResponse$ = BasketOrdersServiceCreateBasketResponse$ = {}));
