// @apexfintechsolutions/ascend-sdk/models/operations/basketordersserviceaddorders.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasketOrdersServiceAddOrdersResponse$ = exports.BasketOrdersServiceAddOrdersResponse$outboundSchema = exports.BasketOrdersServiceAddOrdersResponse$inboundSchema = exports.BasketOrdersServiceAddOrdersRequest$ = exports.BasketOrdersServiceAddOrdersRequest$outboundSchema = exports.BasketOrdersServiceAddOrdersRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BasketOrdersServiceAddOrdersRequest$inboundSchema = z.object({
    correspondent_id: z.string(),
    basket_id: z.string(),
    AddOrdersRequestCreate: components.AddOrdersRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "correspondent_id": "correspondentId",
        "basket_id": "basketId",
        "AddOrdersRequestCreate": "addOrdersRequestCreate",
    });
});
/** @internal */
exports.BasketOrdersServiceAddOrdersRequest$outboundSchema = z.object({
    correspondentId: z.string(),
    basketId: z.string(),
    addOrdersRequestCreate: components.AddOrdersRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        correspondentId: "correspondent_id",
        basketId: "basket_id",
        addOrdersRequestCreate: "AddOrdersRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BasketOrdersServiceAddOrdersRequest$;
(function (BasketOrdersServiceAddOrdersRequest$) {
    /** @deprecated use `BasketOrdersServiceAddOrdersRequest$inboundSchema` instead. */
    BasketOrdersServiceAddOrdersRequest$.inboundSchema = exports.BasketOrdersServiceAddOrdersRequest$inboundSchema;
    /** @deprecated use `BasketOrdersServiceAddOrdersRequest$outboundSchema` instead. */
    BasketOrdersServiceAddOrdersRequest$.outboundSchema = exports.BasketOrdersServiceAddOrdersRequest$outboundSchema;
})(BasketOrdersServiceAddOrdersRequest$ || (exports.BasketOrdersServiceAddOrdersRequest$ = BasketOrdersServiceAddOrdersRequest$ = {}));
/** @internal */
exports.BasketOrdersServiceAddOrdersResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Basket: components.Basket$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Basket": "basket",
        "Status": "status",
    });
});
/** @internal */
exports.BasketOrdersServiceAddOrdersResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    basket: components.Basket$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        basket: "Basket",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BasketOrdersServiceAddOrdersResponse$;
(function (BasketOrdersServiceAddOrdersResponse$) {
    /** @deprecated use `BasketOrdersServiceAddOrdersResponse$inboundSchema` instead. */
    BasketOrdersServiceAddOrdersResponse$.inboundSchema = exports.BasketOrdersServiceAddOrdersResponse$inboundSchema;
    /** @deprecated use `BasketOrdersServiceAddOrdersResponse$outboundSchema` instead. */
    BasketOrdersServiceAddOrdersResponse$.outboundSchema = exports.BasketOrdersServiceAddOrdersResponse$outboundSchema;
})(BasketOrdersServiceAddOrdersResponse$ || (exports.BasketOrdersServiceAddOrdersResponse$ = BasketOrdersServiceAddOrdersResponse$ = {}));
