// @apexfintechsolutions/ascend-sdk/models/operations/bookingcanceltradeallocation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookingCancelTradeAllocationResponse$ = exports.BookingCancelTradeAllocationResponse$outboundSchema = exports.BookingCancelTradeAllocationResponse$inboundSchema = exports.BookingCancelTradeAllocationRequest$ = exports.BookingCancelTradeAllocationRequest$outboundSchema = exports.BookingCancelTradeAllocationRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BookingCancelTradeAllocationRequest$inboundSchema = z.object({
    account_id: z.string(),
    tradeAllocation_id: z.string(),
    CancelTradeAllocationRequestCreate: components.CancelTradeAllocationRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "tradeAllocation_id": "tradeAllocationId",
        "CancelTradeAllocationRequestCreate": "cancelTradeAllocationRequestCreate",
    });
});
/** @internal */
exports.BookingCancelTradeAllocationRequest$outboundSchema = z.object({
    accountId: z.string(),
    tradeAllocationId: z.string(),
    cancelTradeAllocationRequestCreate: components.CancelTradeAllocationRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        tradeAllocationId: "tradeAllocation_id",
        cancelTradeAllocationRequestCreate: "CancelTradeAllocationRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BookingCancelTradeAllocationRequest$;
(function (BookingCancelTradeAllocationRequest$) {
    /** @deprecated use `BookingCancelTradeAllocationRequest$inboundSchema` instead. */
    BookingCancelTradeAllocationRequest$.inboundSchema = exports.BookingCancelTradeAllocationRequest$inboundSchema;
    /** @deprecated use `BookingCancelTradeAllocationRequest$outboundSchema` instead. */
    BookingCancelTradeAllocationRequest$.outboundSchema = exports.BookingCancelTradeAllocationRequest$outboundSchema;
})(BookingCancelTradeAllocationRequest$ || (exports.BookingCancelTradeAllocationRequest$ = BookingCancelTradeAllocationRequest$ = {}));
/** @internal */
exports.BookingCancelTradeAllocationResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    CancelTradeAllocationResponse: components
        .CancelTradeAllocationResponse$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "CancelTradeAllocationResponse": "cancelTradeAllocationResponse",
        "Status": "status",
    });
});
/** @internal */
exports.BookingCancelTradeAllocationResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    cancelTradeAllocationResponse: components
        .CancelTradeAllocationResponse$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        cancelTradeAllocationResponse: "CancelTradeAllocationResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BookingCancelTradeAllocationResponse$;
(function (BookingCancelTradeAllocationResponse$) {
    /** @deprecated use `BookingCancelTradeAllocationResponse$inboundSchema` instead. */
    BookingCancelTradeAllocationResponse$.inboundSchema = exports.BookingCancelTradeAllocationResponse$inboundSchema;
    /** @deprecated use `BookingCancelTradeAllocationResponse$outboundSchema` instead. */
    BookingCancelTradeAllocationResponse$.outboundSchema = exports.BookingCancelTradeAllocationResponse$outboundSchema;
})(BookingCancelTradeAllocationResponse$ || (exports.BookingCancelTradeAllocationResponse$ = BookingCancelTradeAllocationResponse$ = {}));
