// @apexfintechsolutions/ascend-sdk/models/operations/bookinggettrade.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookingGetTradeResponse$ = exports.BookingGetTradeResponse$outboundSchema = exports.BookingGetTradeResponse$inboundSchema = exports.BookingGetTradeRequest$ = exports.BookingGetTradeRequest$outboundSchema = exports.BookingGetTradeRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BookingGetTradeRequest$inboundSchema = z.object({
    account_id: z.string(),
    trade_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "trade_id": "tradeId",
    });
});
/** @internal */
exports.BookingGetTradeRequest$outboundSchema = z.object({
    accountId: z.string(),
    tradeId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        tradeId: "trade_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BookingGetTradeRequest$;
(function (BookingGetTradeRequest$) {
    /** @deprecated use `BookingGetTradeRequest$inboundSchema` instead. */
    BookingGetTradeRequest$.inboundSchema = exports.BookingGetTradeRequest$inboundSchema;
    /** @deprecated use `BookingGetTradeRequest$outboundSchema` instead. */
    BookingGetTradeRequest$.outboundSchema = exports.BookingGetTradeRequest$outboundSchema;
})(BookingGetTradeRequest$ || (exports.BookingGetTradeRequest$ = BookingGetTradeRequest$ = {}));
/** @internal */
exports.BookingGetTradeResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Trade: components.Trade$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Trade": "trade",
        "Status": "status",
    });
});
/** @internal */
exports.BookingGetTradeResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    trade: components.Trade$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        trade: "Trade",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BookingGetTradeResponse$;
(function (BookingGetTradeResponse$) {
    /** @deprecated use `BookingGetTradeResponse$inboundSchema` instead. */
    BookingGetTradeResponse$.inboundSchema = exports.BookingGetTradeResponse$inboundSchema;
    /** @deprecated use `BookingGetTradeResponse$outboundSchema` instead. */
    BookingGetTradeResponse$.outboundSchema = exports.BookingGetTradeResponse$outboundSchema;
})(BookingGetTradeResponse$ || (exports.BookingGetTradeResponse$ = BookingGetTradeResponse$ = {}));
