// @apexfintechsolutions/ascend-sdk/models/operations/bookingcreatetrade.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookingCreateTradeResponse$ = exports.BookingCreateTradeResponse$outboundSchema = exports.BookingCreateTradeResponse$inboundSchema = exports.BookingCreateTradeRequest$ = exports.BookingCreateTradeRequest$outboundSchema = exports.BookingCreateTradeRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BookingCreateTradeRequest$inboundSchema = z.object({
    account_id: z.string(),
    TradeCreate: components.TradeCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "TradeCreate": "tradeCreate",
    });
});
/** @internal */
exports.BookingCreateTradeRequest$outboundSchema = z.object({
    accountId: z.string(),
    tradeCreate: components.TradeCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        tradeCreate: "TradeCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BookingCreateTradeRequest$;
(function (BookingCreateTradeRequest$) {
    /** @deprecated use `BookingCreateTradeRequest$inboundSchema` instead. */
    BookingCreateTradeRequest$.inboundSchema = exports.BookingCreateTradeRequest$inboundSchema;
    /** @deprecated use `BookingCreateTradeRequest$outboundSchema` instead. */
    BookingCreateTradeRequest$.outboundSchema = exports.BookingCreateTradeRequest$outboundSchema;
})(BookingCreateTradeRequest$ || (exports.BookingCreateTradeRequest$ = BookingCreateTradeRequest$ = {}));
/** @internal */
exports.BookingCreateTradeResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Trade: components.Trade$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Trade": "trade",
        "Status": "status",
    });
});
/** @internal */
exports.BookingCreateTradeResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    trade: components.Trade$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        trade: "Trade",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BookingCreateTradeResponse$;
(function (BookingCreateTradeResponse$) {
    /** @deprecated use `BookingCreateTradeResponse$inboundSchema` instead. */
    BookingCreateTradeResponse$.inboundSchema = exports.BookingCreateTradeResponse$inboundSchema;
    /** @deprecated use `BookingCreateTradeResponse$outboundSchema` instead. */
    BookingCreateTradeResponse$.outboundSchema = exports.BookingCreateTradeResponse$outboundSchema;
})(BookingCreateTradeResponse$ || (exports.BookingCreateTradeResponse$ = BookingCreateTradeResponse$ = {}));
