// @apexfintechsolutions/ascend-sdk/models/operations/bookingcreatetradeallocation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookingCreateTradeAllocationResponse$ = exports.BookingCreateTradeAllocationResponse$outboundSchema = exports.BookingCreateTradeAllocationResponse$inboundSchema = exports.BookingCreateTradeAllocationRequest$ = exports.BookingCreateTradeAllocationRequest$outboundSchema = exports.BookingCreateTradeAllocationRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BookingCreateTradeAllocationRequest$inboundSchema = z.object({
    account_id: z.string(),
    request_id: z.string().optional(),
    TradeAllocationCreate: components.TradeAllocationCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "request_id": "requestId",
        "TradeAllocationCreate": "tradeAllocationCreate",
    });
});
/** @internal */
exports.BookingCreateTradeAllocationRequest$outboundSchema = z.object({
    accountId: z.string(),
    requestId: z.string().optional(),
    tradeAllocationCreate: components.TradeAllocationCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        requestId: "request_id",
        tradeAllocationCreate: "TradeAllocationCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BookingCreateTradeAllocationRequest$;
(function (BookingCreateTradeAllocationRequest$) {
    /** @deprecated use `BookingCreateTradeAllocationRequest$inboundSchema` instead. */
    BookingCreateTradeAllocationRequest$.inboundSchema = exports.BookingCreateTradeAllocationRequest$inboundSchema;
    /** @deprecated use `BookingCreateTradeAllocationRequest$outboundSchema` instead. */
    BookingCreateTradeAllocationRequest$.outboundSchema = exports.BookingCreateTradeAllocationRequest$outboundSchema;
})(BookingCreateTradeAllocationRequest$ || (exports.BookingCreateTradeAllocationRequest$ = BookingCreateTradeAllocationRequest$ = {}));
/** @internal */
exports.BookingCreateTradeAllocationResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    TradeAllocation: components.TradeAllocation$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "TradeAllocation": "tradeAllocation",
        "Status": "status",
    });
});
/** @internal */
exports.BookingCreateTradeAllocationResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    tradeAllocation: components.TradeAllocation$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        tradeAllocation: "TradeAllocation",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BookingCreateTradeAllocationResponse$;
(function (BookingCreateTradeAllocationResponse$) {
    /** @deprecated use `BookingCreateTradeAllocationResponse$inboundSchema` instead. */
    BookingCreateTradeAllocationResponse$.inboundSchema = exports.BookingCreateTradeAllocationResponse$inboundSchema;
    /** @deprecated use `BookingCreateTradeAllocationResponse$outboundSchema` instead. */
    BookingCreateTradeAllocationResponse$.outboundSchema = exports.BookingCreateTradeAllocationResponse$outboundSchema;
})(BookingCreateTradeAllocationResponse$ || (exports.BookingCreateTradeAllocationResponse$ = BookingCreateTradeAllocationResponse$ = {}));
