// @apexfintechsolutions/ascend-sdk/models/operations/bookingcanceltrade.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookingCancelTradeResponse$ = exports.BookingCancelTradeResponse$outboundSchema = exports.BookingCancelTradeResponse$inboundSchema = exports.BookingCancelTradeRequest$ = exports.BookingCancelTradeRequest$outboundSchema = exports.BookingCancelTradeRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BookingCancelTradeRequest$inboundSchema = z.object({
    account_id: z.string(),
    trade_id: z.string(),
    CancelTradeRequestCreate: components.CancelTradeRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "trade_id": "tradeId",
        "CancelTradeRequestCreate": "cancelTradeRequestCreate",
    });
});
/** @internal */
exports.BookingCancelTradeRequest$outboundSchema = z.object({
    accountId: z.string(),
    tradeId: z.string(),
    cancelTradeRequestCreate: components.CancelTradeRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        tradeId: "trade_id",
        cancelTradeRequestCreate: "CancelTradeRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BookingCancelTradeRequest$;
(function (BookingCancelTradeRequest$) {
    /** @deprecated use `BookingCancelTradeRequest$inboundSchema` instead. */
    BookingCancelTradeRequest$.inboundSchema = exports.BookingCancelTradeRequest$inboundSchema;
    /** @deprecated use `BookingCancelTradeRequest$outboundSchema` instead. */
    BookingCancelTradeRequest$.outboundSchema = exports.BookingCancelTradeRequest$outboundSchema;
})(BookingCancelTradeRequest$ || (exports.BookingCancelTradeRequest$ = BookingCancelTradeRequest$ = {}));
/** @internal */
exports.BookingCancelTradeResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    CancelTradeResponse: components.CancelTradeResponse$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "CancelTradeResponse": "cancelTradeResponse",
        "Status": "status",
    });
});
/** @internal */
exports.BookingCancelTradeResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    cancelTradeResponse: components.CancelTradeResponse$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        cancelTradeResponse: "CancelTradeResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BookingCancelTradeResponse$;
(function (BookingCancelTradeResponse$) {
    /** @deprecated use `BookingCancelTradeResponse$inboundSchema` instead. */
    BookingCancelTradeResponse$.inboundSchema = exports.BookingCancelTradeResponse$inboundSchema;
    /** @deprecated use `BookingCancelTradeResponse$outboundSchema` instead. */
    BookingCancelTradeResponse$.outboundSchema = exports.BookingCancelTradeResponse$outboundSchema;
})(BookingCancelTradeResponse$ || (exports.BookingCancelTradeResponse$ = BookingCancelTradeResponse$ = {}));
