// @apexfintechsolutions/ascend-sdk/models/operations/cashbalancescalculatecashbalance.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CashBalancesCalculateCashBalanceResponse$ = exports.CashBalancesCalculateCashBalanceResponse$outboundSchema = exports.CashBalancesCalculateCashBalanceResponse$inboundSchema = exports.CashBalancesCalculateCashBalanceRequest$ = exports.CashBalancesCalculateCashBalanceRequest$outboundSchema = exports.CashBalancesCalculateCashBalanceRequest$inboundSchema = exports.Mechanism$ = exports.Mechanism$outboundSchema = exports.Mechanism$inboundSchema = exports.Mechanism = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/**
 * The withdraw mechanism to calculate the balance for. The mechanism determines what account activity will affect the balance.
 */
var Mechanism;
(function (Mechanism) {
    Mechanism["MechanismUnspecified"] = "MECHANISM_UNSPECIFIED";
    Mechanism["Acat"] = "ACAT";
    Mechanism["Ach"] = "ACH";
    Mechanism["CashJournal"] = "CASH_JOURNAL";
    Mechanism["Check"] = "CHECK";
    Mechanism["Credit"] = "CREDIT";
    Mechanism["Fee"] = "FEE";
    Mechanism["Ict"] = "ICT";
    Mechanism["Paypal"] = "PAYPAL";
    Mechanism["Rtp"] = "RTP";
    Mechanism["Tpj"] = "TPJ";
    Mechanism["Wire"] = "WIRE";
    Mechanism["ExternalAch"] = "EXTERNAL_ACH";
})(Mechanism || (exports.Mechanism = Mechanism = {}));
/** @internal */
exports.Mechanism$inboundSchema = z
    .nativeEnum(Mechanism);
/** @internal */
exports.Mechanism$outboundSchema = exports.Mechanism$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Mechanism$;
(function (Mechanism$) {
    /** @deprecated use `Mechanism$inboundSchema` instead. */
    Mechanism$.inboundSchema = exports.Mechanism$inboundSchema;
    /** @deprecated use `Mechanism$outboundSchema` instead. */
    Mechanism$.outboundSchema = exports.Mechanism$outboundSchema;
})(Mechanism$ || (exports.Mechanism$ = Mechanism$ = {}));
/** @internal */
exports.CashBalancesCalculateCashBalanceRequest$inboundSchema = z.object({
    account_id: z.string(),
    mechanism: exports.Mechanism$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
    });
});
/** @internal */
exports.CashBalancesCalculateCashBalanceRequest$outboundSchema = z.object({
    accountId: z.string(),
    mechanism: exports.Mechanism$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashBalancesCalculateCashBalanceRequest$;
(function (CashBalancesCalculateCashBalanceRequest$) {
    /** @deprecated use `CashBalancesCalculateCashBalanceRequest$inboundSchema` instead. */
    CashBalancesCalculateCashBalanceRequest$.inboundSchema = exports.CashBalancesCalculateCashBalanceRequest$inboundSchema;
    /** @deprecated use `CashBalancesCalculateCashBalanceRequest$outboundSchema` instead. */
    CashBalancesCalculateCashBalanceRequest$.outboundSchema = exports.CashBalancesCalculateCashBalanceRequest$outboundSchema;
})(CashBalancesCalculateCashBalanceRequest$ || (exports.CashBalancesCalculateCashBalanceRequest$ = CashBalancesCalculateCashBalanceRequest$ = {}));
/** @internal */
exports.CashBalancesCalculateCashBalanceResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    CalculateCashBalanceResponse: components
        .CalculateCashBalanceResponse$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "CalculateCashBalanceResponse": "calculateCashBalanceResponse",
        "Status": "status",
    });
});
/** @internal */
exports.CashBalancesCalculateCashBalanceResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    calculateCashBalanceResponse: components
        .CalculateCashBalanceResponse$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        calculateCashBalanceResponse: "CalculateCashBalanceResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashBalancesCalculateCashBalanceResponse$;
(function (CashBalancesCalculateCashBalanceResponse$) {
    /** @deprecated use `CashBalancesCalculateCashBalanceResponse$inboundSchema` instead. */
    CashBalancesCalculateCashBalanceResponse$.inboundSchema = exports.CashBalancesCalculateCashBalanceResponse$inboundSchema;
    /** @deprecated use `CashBalancesCalculateCashBalanceResponse$outboundSchema` instead. */
    CashBalancesCalculateCashBalanceResponse$.outboundSchema = exports.CashBalancesCalculateCashBalanceResponse$outboundSchema;
})(CashBalancesCalculateCashBalanceResponse$ || (exports.CashBalancesCalculateCashBalanceResponse$ = CashBalancesCalculateCashBalanceResponse$ = {}));
