// @apexfintechsolutions/ascend-sdk/models/operations/cashjournalsforcerejectcashjournal.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CashJournalsForceRejectCashJournalResponse$ = exports.CashJournalsForceRejectCashJournalResponse$outboundSchema = exports.CashJournalsForceRejectCashJournalResponse$inboundSchema = exports.CashJournalsForceRejectCashJournalRequest$ = exports.CashJournalsForceRejectCashJournalRequest$outboundSchema = exports.CashJournalsForceRejectCashJournalRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.CashJournalsForceRejectCashJournalRequest$inboundSchema = z.object({
    cashJournal_id: z.string(),
    ForceRejectCashJournalRequestCreate: components.ForceRejectCashJournalRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "cashJournal_id": "cashJournalId",
        "ForceRejectCashJournalRequestCreate": "forceRejectCashJournalRequestCreate",
    });
});
/** @internal */
exports.CashJournalsForceRejectCashJournalRequest$outboundSchema = z.object({
    cashJournalId: z.string(),
    forceRejectCashJournalRequestCreate: components.ForceRejectCashJournalRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        cashJournalId: "cashJournal_id",
        forceRejectCashJournalRequestCreate: "ForceRejectCashJournalRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalsForceRejectCashJournalRequest$;
(function (CashJournalsForceRejectCashJournalRequest$) {
    /** @deprecated use `CashJournalsForceRejectCashJournalRequest$inboundSchema` instead. */
    CashJournalsForceRejectCashJournalRequest$.inboundSchema = exports.CashJournalsForceRejectCashJournalRequest$inboundSchema;
    /** @deprecated use `CashJournalsForceRejectCashJournalRequest$outboundSchema` instead. */
    CashJournalsForceRejectCashJournalRequest$.outboundSchema = exports.CashJournalsForceRejectCashJournalRequest$outboundSchema;
})(CashJournalsForceRejectCashJournalRequest$ || (exports.CashJournalsForceRejectCashJournalRequest$ = CashJournalsForceRejectCashJournalRequest$ = {}));
/** @internal */
exports.CashJournalsForceRejectCashJournalResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    CashJournal: components.CashJournal$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "CashJournal": "cashJournal",
        "Status": "status",
    });
});
/** @internal */
exports.CashJournalsForceRejectCashJournalResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    cashJournal: components.CashJournal$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        cashJournal: "CashJournal",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CashJournalsForceRejectCashJournalResponse$;
(function (CashJournalsForceRejectCashJournalResponse$) {
    /** @deprecated use `CashJournalsForceRejectCashJournalResponse$inboundSchema` instead. */
    CashJournalsForceRejectCashJournalResponse$.inboundSchema = exports.CashJournalsForceRejectCashJournalResponse$inboundSchema;
    /** @deprecated use `CashJournalsForceRejectCashJournalResponse$outboundSchema` instead. */
    CashJournalsForceRejectCashJournalResponse$.outboundSchema = exports.CashJournalsForceRejectCashJournalResponse$outboundSchema;
})(CashJournalsForceRejectCashJournalResponse$ || (exports.CashJournalsForceRejectCashJournalResponse$ = CashJournalsForceRejectCashJournalResponse$ = {}));
