// @apexfintechsolutions/ascend-sdk/models/operations/customeridentificationresultservicegetcustomeridentification.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerIdentificationResultServiceGetCustomerIdentificationResponse$ = exports.CustomerIdentificationResultServiceGetCustomerIdentificationResponse$outboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationResponse$inboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationRequest$ = exports.CustomerIdentificationResultServiceGetCustomerIdentificationRequest$outboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationRequest$inboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$ = exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$outboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$inboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/**
 * Optional. The view to return. Defaults to BASIC.
 */
var CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView;
(function (CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView) {
    CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView["ViewUnspecified"] = "VIEW_UNSPECIFIED";
    CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView["Basic"] = "BASIC";
    CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView["Full"] = "FULL";
})(CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView || (exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView = CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView = {}));
/** @internal */
exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$inboundSchema = z.nativeEnum(CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView);
/** @internal */
exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$outboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$;
(function (CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$) {
    /** @deprecated use `CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$inboundSchema` instead. */
    CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$.inboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$inboundSchema;
    /** @deprecated use `CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$outboundSchema` instead. */
    CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$.outboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$outboundSchema;
})(CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$ || (exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$ = CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$ = {}));
/** @internal */
exports.CustomerIdentificationResultServiceGetCustomerIdentificationRequest$inboundSchema = z.object({
    correspondent_id: z.string(),
    customerIdentification_id: z.string(),
    view: exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$inboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "correspondent_id": "correspondentId",
        "customerIdentification_id": "customerIdentificationId",
    });
});
/** @internal */
exports.CustomerIdentificationResultServiceGetCustomerIdentificationRequest$outboundSchema = z.object({
    correspondentId: z.string(),
    customerIdentificationId: z.string(),
    view: exports.CustomerIdentificationResultServiceGetCustomerIdentificationQueryParamView$outboundSchema
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        correspondentId: "correspondent_id",
        customerIdentificationId: "customerIdentification_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerIdentificationResultServiceGetCustomerIdentificationRequest$;
(function (CustomerIdentificationResultServiceGetCustomerIdentificationRequest$) {
    /** @deprecated use `CustomerIdentificationResultServiceGetCustomerIdentificationRequest$inboundSchema` instead. */
    CustomerIdentificationResultServiceGetCustomerIdentificationRequest$.inboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationRequest$inboundSchema;
    /** @deprecated use `CustomerIdentificationResultServiceGetCustomerIdentificationRequest$outboundSchema` instead. */
    CustomerIdentificationResultServiceGetCustomerIdentificationRequest$.outboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationRequest$outboundSchema;
})(CustomerIdentificationResultServiceGetCustomerIdentificationRequest$ || (exports.CustomerIdentificationResultServiceGetCustomerIdentificationRequest$ = CustomerIdentificationResultServiceGetCustomerIdentificationRequest$ = {}));
/** @internal */
exports.CustomerIdentificationResultServiceGetCustomerIdentificationResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    CustomerIdentification: components.CustomerIdentification$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "CustomerIdentification": "customerIdentification",
        "Status": "status",
    });
});
/** @internal */
exports.CustomerIdentificationResultServiceGetCustomerIdentificationResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    customerIdentification: components.CustomerIdentification$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        customerIdentification: "CustomerIdentification",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerIdentificationResultServiceGetCustomerIdentificationResponse$;
(function (CustomerIdentificationResultServiceGetCustomerIdentificationResponse$) {
    /** @deprecated use `CustomerIdentificationResultServiceGetCustomerIdentificationResponse$inboundSchema` instead. */
    CustomerIdentificationResultServiceGetCustomerIdentificationResponse$.inboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationResponse$inboundSchema;
    /** @deprecated use `CustomerIdentificationResultServiceGetCustomerIdentificationResponse$outboundSchema` instead. */
    CustomerIdentificationResultServiceGetCustomerIdentificationResponse$.outboundSchema = exports.CustomerIdentificationResultServiceGetCustomerIdentificationResponse$outboundSchema;
})(CustomerIdentificationResultServiceGetCustomerIdentificationResponse$ || (exports.CustomerIdentificationResultServiceGetCustomerIdentificationResponse$ = CustomerIdentificationResultServiceGetCustomerIdentificationResponse$ = {}));
