// @apexfintechsolutions/ascend-sdk/models/operations/ictdepositscancelictdeposit.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IctDepositsCancelIctDepositResponse$ = exports.IctDepositsCancelIctDepositResponse$outboundSchema = exports.IctDepositsCancelIctDepositResponse$inboundSchema = exports.IctDepositsCancelIctDepositRequest$ = exports.IctDepositsCancelIctDepositRequest$outboundSchema = exports.IctDepositsCancelIctDepositRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.IctDepositsCancelIctDepositRequest$inboundSchema = z.object({
    account_id: z.string(),
    ictDeposit_id: z.string(),
    CancelIctDepositRequestCreate: components.CancelIctDepositRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "ictDeposit_id": "ictDepositId",
        "CancelIctDepositRequestCreate": "cancelIctDepositRequestCreate",
    });
});
/** @internal */
exports.IctDepositsCancelIctDepositRequest$outboundSchema = z.object({
    accountId: z.string(),
    ictDepositId: z.string(),
    cancelIctDepositRequestCreate: components.CancelIctDepositRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        ictDepositId: "ictDeposit_id",
        cancelIctDepositRequestCreate: "CancelIctDepositRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IctDepositsCancelIctDepositRequest$;
(function (IctDepositsCancelIctDepositRequest$) {
    /** @deprecated use `IctDepositsCancelIctDepositRequest$inboundSchema` instead. */
    IctDepositsCancelIctDepositRequest$.inboundSchema = exports.IctDepositsCancelIctDepositRequest$inboundSchema;
    /** @deprecated use `IctDepositsCancelIctDepositRequest$outboundSchema` instead. */
    IctDepositsCancelIctDepositRequest$.outboundSchema = exports.IctDepositsCancelIctDepositRequest$outboundSchema;
})(IctDepositsCancelIctDepositRequest$ || (exports.IctDepositsCancelIctDepositRequest$ = IctDepositsCancelIctDepositRequest$ = {}));
/** @internal */
exports.IctDepositsCancelIctDepositResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    IctDeposit: components.IctDeposit$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "IctDeposit": "ictDeposit",
        "Status": "status",
    });
});
/** @internal */
exports.IctDepositsCancelIctDepositResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    ictDeposit: components.IctDeposit$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        ictDeposit: "IctDeposit",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IctDepositsCancelIctDepositResponse$;
(function (IctDepositsCancelIctDepositResponse$) {
    /** @deprecated use `IctDepositsCancelIctDepositResponse$inboundSchema` instead. */
    IctDepositsCancelIctDepositResponse$.inboundSchema = exports.IctDepositsCancelIctDepositResponse$inboundSchema;
    /** @deprecated use `IctDepositsCancelIctDepositResponse$outboundSchema` instead. */
    IctDepositsCancelIctDepositResponse$.outboundSchema = exports.IctDepositsCancelIctDepositResponse$outboundSchema;
})(IctDepositsCancelIctDepositResponse$ || (exports.IctDepositsCancelIctDepositResponse$ = IctDepositsCancelIctDepositResponse$ = {}));
