// @apexfintechsolutions/ascend-sdk/models/operations/ictwithdrawalscreateictwithdrawal.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IctWithdrawalsCreateIctWithdrawalResponse$ = exports.IctWithdrawalsCreateIctWithdrawalResponse$outboundSchema = exports.IctWithdrawalsCreateIctWithdrawalResponse$inboundSchema = exports.IctWithdrawalsCreateIctWithdrawalRequest$ = exports.IctWithdrawalsCreateIctWithdrawalRequest$outboundSchema = exports.IctWithdrawalsCreateIctWithdrawalRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.IctWithdrawalsCreateIctWithdrawalRequest$inboundSchema = z.object({
    account_id: z.string(),
    IctWithdrawalCreate: components.IctWithdrawalCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "IctWithdrawalCreate": "ictWithdrawalCreate",
    });
});
/** @internal */
exports.IctWithdrawalsCreateIctWithdrawalRequest$outboundSchema = z.object({
    accountId: z.string(),
    ictWithdrawalCreate: components.IctWithdrawalCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        ictWithdrawalCreate: "IctWithdrawalCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IctWithdrawalsCreateIctWithdrawalRequest$;
(function (IctWithdrawalsCreateIctWithdrawalRequest$) {
    /** @deprecated use `IctWithdrawalsCreateIctWithdrawalRequest$inboundSchema` instead. */
    IctWithdrawalsCreateIctWithdrawalRequest$.inboundSchema = exports.IctWithdrawalsCreateIctWithdrawalRequest$inboundSchema;
    /** @deprecated use `IctWithdrawalsCreateIctWithdrawalRequest$outboundSchema` instead. */
    IctWithdrawalsCreateIctWithdrawalRequest$.outboundSchema = exports.IctWithdrawalsCreateIctWithdrawalRequest$outboundSchema;
})(IctWithdrawalsCreateIctWithdrawalRequest$ || (exports.IctWithdrawalsCreateIctWithdrawalRequest$ = IctWithdrawalsCreateIctWithdrawalRequest$ = {}));
/** @internal */
exports.IctWithdrawalsCreateIctWithdrawalResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    IctWithdrawal: components.IctWithdrawal$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "IctWithdrawal": "ictWithdrawal",
        "Status": "status",
    });
});
/** @internal */
exports.IctWithdrawalsCreateIctWithdrawalResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    ictWithdrawal: components.IctWithdrawal$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        ictWithdrawal: "IctWithdrawal",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var IctWithdrawalsCreateIctWithdrawalResponse$;
(function (IctWithdrawalsCreateIctWithdrawalResponse$) {
    /** @deprecated use `IctWithdrawalsCreateIctWithdrawalResponse$inboundSchema` instead. */
    IctWithdrawalsCreateIctWithdrawalResponse$.inboundSchema = exports.IctWithdrawalsCreateIctWithdrawalResponse$inboundSchema;
    /** @deprecated use `IctWithdrawalsCreateIctWithdrawalResponse$outboundSchema` instead. */
    IctWithdrawalsCreateIctWithdrawalResponse$.outboundSchema = exports.IctWithdrawalsCreateIctWithdrawalResponse$outboundSchema;
})(IctWithdrawalsCreateIctWithdrawalResponse$ || (exports.IctWithdrawalsCreateIctWithdrawalResponse$ = IctWithdrawalsCreateIctWithdrawalResponse$ = {}));
