// @apexfintechsolutions/ascend-sdk/models/operations/investigationservicelinkdocuments.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvestigationServiceLinkDocumentsResponse$ = exports.InvestigationServiceLinkDocumentsResponse$outboundSchema = exports.InvestigationServiceLinkDocumentsResponse$inboundSchema = exports.InvestigationServiceLinkDocumentsRequest$ = exports.InvestigationServiceLinkDocumentsRequest$outboundSchema = exports.InvestigationServiceLinkDocumentsRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.InvestigationServiceLinkDocumentsRequest$inboundSchema = z.object({
    investigation_id: z.string(),
    LinkDocumentsRequestCreate: components.LinkDocumentsRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "investigation_id": "investigationId",
        "LinkDocumentsRequestCreate": "linkDocumentsRequestCreate",
    });
});
/** @internal */
exports.InvestigationServiceLinkDocumentsRequest$outboundSchema = z.object({
    investigationId: z.string(),
    linkDocumentsRequestCreate: components.LinkDocumentsRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        investigationId: "investigation_id",
        linkDocumentsRequestCreate: "LinkDocumentsRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InvestigationServiceLinkDocumentsRequest$;
(function (InvestigationServiceLinkDocumentsRequest$) {
    /** @deprecated use `InvestigationServiceLinkDocumentsRequest$inboundSchema` instead. */
    InvestigationServiceLinkDocumentsRequest$.inboundSchema = exports.InvestigationServiceLinkDocumentsRequest$inboundSchema;
    /** @deprecated use `InvestigationServiceLinkDocumentsRequest$outboundSchema` instead. */
    InvestigationServiceLinkDocumentsRequest$.outboundSchema = exports.InvestigationServiceLinkDocumentsRequest$outboundSchema;
})(InvestigationServiceLinkDocumentsRequest$ || (exports.InvestigationServiceLinkDocumentsRequest$ = InvestigationServiceLinkDocumentsRequest$ = {}));
/** @internal */
exports.InvestigationServiceLinkDocumentsResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    LinkDocumentsResponse: components.LinkDocumentsResponse$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "LinkDocumentsResponse": "linkDocumentsResponse",
        "Status": "status",
    });
});
/** @internal */
exports.InvestigationServiceLinkDocumentsResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    linkDocumentsResponse: components.LinkDocumentsResponse$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        linkDocumentsResponse: "LinkDocumentsResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InvestigationServiceLinkDocumentsResponse$;
(function (InvestigationServiceLinkDocumentsResponse$) {
    /** @deprecated use `InvestigationServiceLinkDocumentsResponse$inboundSchema` instead. */
    InvestigationServiceLinkDocumentsResponse$.inboundSchema = exports.InvestigationServiceLinkDocumentsResponse$inboundSchema;
    /** @deprecated use `InvestigationServiceLinkDocumentsResponse$outboundSchema` instead. */
    InvestigationServiceLinkDocumentsResponse$.outboundSchema = exports.InvestigationServiceLinkDocumentsResponse$outboundSchema;
})(InvestigationServiceLinkDocumentsResponse$ || (exports.InvestigationServiceLinkDocumentsResponse$ = InvestigationServiceLinkDocumentsResponse$ = {}));
