// @apexfintechsolutions/ascend-sdk/models/operations/investigationserviceupdateinvestigation.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvestigationServiceUpdateInvestigationResponse$ = exports.InvestigationServiceUpdateInvestigationResponse$outboundSchema = exports.InvestigationServiceUpdateInvestigationResponse$inboundSchema = exports.InvestigationServiceUpdateInvestigationRequest$ = exports.InvestigationServiceUpdateInvestigationRequest$outboundSchema = exports.InvestigationServiceUpdateInvestigationRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.InvestigationServiceUpdateInvestigationRequest$inboundSchema = z.object({
    investigation_id: z.string(),
    update_mask: z.string().optional(),
    InvestigationUpdate: components.InvestigationUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "investigation_id": "investigationId",
        "update_mask": "updateMask",
        "InvestigationUpdate": "investigationUpdate",
    });
});
/** @internal */
exports.InvestigationServiceUpdateInvestigationRequest$outboundSchema = z.object({
    investigationId: z.string(),
    updateMask: z.string().optional(),
    investigationUpdate: components.InvestigationUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        investigationId: "investigation_id",
        updateMask: "update_mask",
        investigationUpdate: "InvestigationUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InvestigationServiceUpdateInvestigationRequest$;
(function (InvestigationServiceUpdateInvestigationRequest$) {
    /** @deprecated use `InvestigationServiceUpdateInvestigationRequest$inboundSchema` instead. */
    InvestigationServiceUpdateInvestigationRequest$.inboundSchema = exports.InvestigationServiceUpdateInvestigationRequest$inboundSchema;
    /** @deprecated use `InvestigationServiceUpdateInvestigationRequest$outboundSchema` instead. */
    InvestigationServiceUpdateInvestigationRequest$.outboundSchema = exports.InvestigationServiceUpdateInvestigationRequest$outboundSchema;
})(InvestigationServiceUpdateInvestigationRequest$ || (exports.InvestigationServiceUpdateInvestigationRequest$ = InvestigationServiceUpdateInvestigationRequest$ = {}));
/** @internal */
exports.InvestigationServiceUpdateInvestigationResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Investigation: components.Investigation$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Investigation": "investigation",
        "Status": "status",
    });
});
/** @internal */
exports.InvestigationServiceUpdateInvestigationResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    investigation: components.Investigation$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        investigation: "Investigation",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InvestigationServiceUpdateInvestigationResponse$;
(function (InvestigationServiceUpdateInvestigationResponse$) {
    /** @deprecated use `InvestigationServiceUpdateInvestigationResponse$inboundSchema` instead. */
    InvestigationServiceUpdateInvestigationResponse$.inboundSchema = exports.InvestigationServiceUpdateInvestigationResponse$inboundSchema;
    /** @deprecated use `InvestigationServiceUpdateInvestigationResponse$outboundSchema` instead. */
    InvestigationServiceUpdateInvestigationResponse$.outboundSchema = exports.InvestigationServiceUpdateInvestigationResponse$outboundSchema;
})(InvestigationServiceUpdateInvestigationResponse$ || (exports.InvestigationServiceUpdateInvestigationResponse$ = InvestigationServiceUpdateInvestigationResponse$ = {}));
