// @apexfintechsolutions/ascend-sdk/models/operations/investorcommunicationservicelistdocuments.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvestorCommunicationServiceListDocumentsResponse$ = exports.InvestorCommunicationServiceListDocumentsResponse$outboundSchema = exports.InvestorCommunicationServiceListDocumentsResponse$inboundSchema = exports.InvestorCommunicationServiceListDocumentsRequest$ = exports.InvestorCommunicationServiceListDocumentsRequest$outboundSchema = exports.InvestorCommunicationServiceListDocumentsRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.InvestorCommunicationServiceListDocumentsRequest$inboundSchema = z.object({
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
    filter: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "page_size": "pageSize",
        "page_token": "pageToken",
    });
});
/** @internal */
exports.InvestorCommunicationServiceListDocumentsRequest$outboundSchema = z.object({
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
    filter: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        pageSize: "page_size",
        pageToken: "page_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InvestorCommunicationServiceListDocumentsRequest$;
(function (InvestorCommunicationServiceListDocumentsRequest$) {
    /** @deprecated use `InvestorCommunicationServiceListDocumentsRequest$inboundSchema` instead. */
    InvestorCommunicationServiceListDocumentsRequest$.inboundSchema = exports.InvestorCommunicationServiceListDocumentsRequest$inboundSchema;
    /** @deprecated use `InvestorCommunicationServiceListDocumentsRequest$outboundSchema` instead. */
    InvestorCommunicationServiceListDocumentsRequest$.outboundSchema = exports.InvestorCommunicationServiceListDocumentsRequest$outboundSchema;
})(InvestorCommunicationServiceListDocumentsRequest$ || (exports.InvestorCommunicationServiceListDocumentsRequest$ = InvestorCommunicationServiceListDocumentsRequest$ = {}));
/** @internal */
exports.InvestorCommunicationServiceListDocumentsResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListDocumentsResponse: components.ListDocumentsResponse$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "ListDocumentsResponse": "listDocumentsResponse",
        "Status": "status",
    });
});
/** @internal */
exports.InvestorCommunicationServiceListDocumentsResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listDocumentsResponse: components.ListDocumentsResponse$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        listDocumentsResponse: "ListDocumentsResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InvestorCommunicationServiceListDocumentsResponse$;
(function (InvestorCommunicationServiceListDocumentsResponse$) {
    /** @deprecated use `InvestorCommunicationServiceListDocumentsResponse$inboundSchema` instead. */
    InvestorCommunicationServiceListDocumentsResponse$.inboundSchema = exports.InvestorCommunicationServiceListDocumentsResponse$inboundSchema;
    /** @deprecated use `InvestorCommunicationServiceListDocumentsResponse$outboundSchema` instead. */
    InvestorCommunicationServiceListDocumentsResponse$.outboundSchema = exports.InvestorCommunicationServiceListDocumentsResponse$outboundSchema;
})(InvestorCommunicationServiceListDocumentsResponse$ || (exports.InvestorCommunicationServiceListDocumentsResponse$ = InvestorCommunicationServiceListDocumentsResponse$ = {}));
