// @apexfintechsolutions/ascend-sdk/models/operations/orderpriceserviceretrievequote.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderPriceServiceRetrieveQuoteResponse$ = exports.OrderPriceServiceRetrieveQuoteResponse$outboundSchema = exports.OrderPriceServiceRetrieveQuoteResponse$inboundSchema = exports.OrderPriceServiceRetrieveQuoteRequest$ = exports.OrderPriceServiceRetrieveQuoteRequest$outboundSchema = exports.OrderPriceServiceRetrieveQuoteRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.OrderPriceServiceRetrieveQuoteRequest$inboundSchema = z.object({
    account_id: z.string(),
    RetrieveQuoteRequestCreate: components.RetrieveQuoteRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "RetrieveQuoteRequestCreate": "retrieveQuoteRequestCreate",
    });
});
/** @internal */
exports.OrderPriceServiceRetrieveQuoteRequest$outboundSchema = z.object({
    accountId: z.string(),
    retrieveQuoteRequestCreate: components.RetrieveQuoteRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        retrieveQuoteRequestCreate: "RetrieveQuoteRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderPriceServiceRetrieveQuoteRequest$;
(function (OrderPriceServiceRetrieveQuoteRequest$) {
    /** @deprecated use `OrderPriceServiceRetrieveQuoteRequest$inboundSchema` instead. */
    OrderPriceServiceRetrieveQuoteRequest$.inboundSchema = exports.OrderPriceServiceRetrieveQuoteRequest$inboundSchema;
    /** @deprecated use `OrderPriceServiceRetrieveQuoteRequest$outboundSchema` instead. */
    OrderPriceServiceRetrieveQuoteRequest$.outboundSchema = exports.OrderPriceServiceRetrieveQuoteRequest$outboundSchema;
})(OrderPriceServiceRetrieveQuoteRequest$ || (exports.OrderPriceServiceRetrieveQuoteRequest$ = OrderPriceServiceRetrieveQuoteRequest$ = {}));
/** @internal */
exports.OrderPriceServiceRetrieveQuoteResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    RetrieveQuoteResponse: components.RetrieveQuoteResponse$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "RetrieveQuoteResponse": "retrieveQuoteResponse",
        "Status": "status",
    });
});
/** @internal */
exports.OrderPriceServiceRetrieveQuoteResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    retrieveQuoteResponse: components.RetrieveQuoteResponse$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        retrieveQuoteResponse: "RetrieveQuoteResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderPriceServiceRetrieveQuoteResponse$;
(function (OrderPriceServiceRetrieveQuoteResponse$) {
    /** @deprecated use `OrderPriceServiceRetrieveQuoteResponse$inboundSchema` instead. */
    OrderPriceServiceRetrieveQuoteResponse$.inboundSchema = exports.OrderPriceServiceRetrieveQuoteResponse$inboundSchema;
    /** @deprecated use `OrderPriceServiceRetrieveQuoteResponse$outboundSchema` instead. */
    OrderPriceServiceRetrieveQuoteResponse$.outboundSchema = exports.OrderPriceServiceRetrieveQuoteResponse$outboundSchema;
})(OrderPriceServiceRetrieveQuoteResponse$ || (exports.OrderPriceServiceRetrieveQuoteResponse$ = OrderPriceServiceRetrieveQuoteResponse$ = {}));
