// @apexfintechsolutions/ascend-sdk/models/operations/orderservicecreateorder.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderServiceCreateOrderResponse$ = exports.OrderServiceCreateOrderResponse$outboundSchema = exports.OrderServiceCreateOrderResponse$inboundSchema = exports.OrderServiceCreateOrderRequest$ = exports.OrderServiceCreateOrderRequest$outboundSchema = exports.OrderServiceCreateOrderRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.OrderServiceCreateOrderRequest$inboundSchema = z.object({
    account_id: z.string(),
    OrderCreate: components.OrderCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "OrderCreate": "orderCreate",
    });
});
/** @internal */
exports.OrderServiceCreateOrderRequest$outboundSchema = z.object({
    accountId: z.string(),
    orderCreate: components.OrderCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        orderCreate: "OrderCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderServiceCreateOrderRequest$;
(function (OrderServiceCreateOrderRequest$) {
    /** @deprecated use `OrderServiceCreateOrderRequest$inboundSchema` instead. */
    OrderServiceCreateOrderRequest$.inboundSchema = exports.OrderServiceCreateOrderRequest$inboundSchema;
    /** @deprecated use `OrderServiceCreateOrderRequest$outboundSchema` instead. */
    OrderServiceCreateOrderRequest$.outboundSchema = exports.OrderServiceCreateOrderRequest$outboundSchema;
})(OrderServiceCreateOrderRequest$ || (exports.OrderServiceCreateOrderRequest$ = OrderServiceCreateOrderRequest$ = {}));
/** @internal */
exports.OrderServiceCreateOrderResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    Order: components.Order$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "Order": "order",
        "Status": "status",
    });
});
/** @internal */
exports.OrderServiceCreateOrderResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    order: components.Order$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        order: "Order",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderServiceCreateOrderResponse$;
(function (OrderServiceCreateOrderResponse$) {
    /** @deprecated use `OrderServiceCreateOrderResponse$inboundSchema` instead. */
    OrderServiceCreateOrderResponse$.inboundSchema = exports.OrderServiceCreateOrderResponse$inboundSchema;
    /** @deprecated use `OrderServiceCreateOrderResponse$outboundSchema` instead. */
    OrderServiceCreateOrderResponse$.outboundSchema = exports.OrderServiceCreateOrderResponse$outboundSchema;
})(OrderServiceCreateOrderResponse$ || (exports.OrderServiceCreateOrderResponse$ = OrderServiceCreateOrderResponse$ = {}));
