// @apexfintechsolutions/ascend-sdk/models/operations/readergeteventmessage.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReaderGetEventMessageResponse$ = exports.ReaderGetEventMessageResponse$outboundSchema = exports.ReaderGetEventMessageResponse$inboundSchema = exports.ReaderGetEventMessageRequest$ = exports.ReaderGetEventMessageRequest$outboundSchema = exports.ReaderGetEventMessageRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.ReaderGetEventMessageRequest$inboundSchema = z.object({
    message_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "message_id": "messageId",
    });
});
/** @internal */
exports.ReaderGetEventMessageRequest$outboundSchema = z.object({
    messageId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        messageId: "message_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ReaderGetEventMessageRequest$;
(function (ReaderGetEventMessageRequest$) {
    /** @deprecated use `ReaderGetEventMessageRequest$inboundSchema` instead. */
    ReaderGetEventMessageRequest$.inboundSchema = exports.ReaderGetEventMessageRequest$inboundSchema;
    /** @deprecated use `ReaderGetEventMessageRequest$outboundSchema` instead. */
    ReaderGetEventMessageRequest$.outboundSchema = exports.ReaderGetEventMessageRequest$outboundSchema;
})(ReaderGetEventMessageRequest$ || (exports.ReaderGetEventMessageRequest$ = ReaderGetEventMessageRequest$ = {}));
/** @internal */
exports.ReaderGetEventMessageResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    EventMessage: components.EventMessage$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "EventMessage": "eventMessage",
        "Status": "status",
    });
});
/** @internal */
exports.ReaderGetEventMessageResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    eventMessage: components.EventMessage$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        eventMessage: "EventMessage",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ReaderGetEventMessageResponse$;
(function (ReaderGetEventMessageResponse$) {
    /** @deprecated use `ReaderGetEventMessageResponse$inboundSchema` instead. */
    ReaderGetEventMessageResponse$.inboundSchema = exports.ReaderGetEventMessageResponse$inboundSchema;
    /** @deprecated use `ReaderGetEventMessageResponse$outboundSchema` instead. */
    ReaderGetEventMessageResponse$.outboundSchema = exports.ReaderGetEventMessageResponse$outboundSchema;
})(ReaderGetEventMessageResponse$ || (exports.ReaderGetEventMessageResponse$ = ReaderGetEventMessageResponse$ = {}));
