// @apexfintechsolutions/ascend-sdk/models/operations/retirementconstraintslistcontributionsummaries.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetirementConstraintsListContributionSummariesResponse$ = exports.RetirementConstraintsListContributionSummariesResponse$outboundSchema = exports.RetirementConstraintsListContributionSummariesResponse$inboundSchema = exports.RetirementConstraintsListContributionSummariesRequest$ = exports.RetirementConstraintsListContributionSummariesRequest$outboundSchema = exports.RetirementConstraintsListContributionSummariesRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.RetirementConstraintsListContributionSummariesRequest$inboundSchema = z.object({
    account_id: z.string(),
    page_size: z.number().int().optional(),
    page_token: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "page_size": "pageSize",
        "page_token": "pageToken",
    });
});
/** @internal */
exports.RetirementConstraintsListContributionSummariesRequest$outboundSchema = z.object({
    accountId: z.string(),
    pageSize: z.number().int().optional(),
    pageToken: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        pageSize: "page_size",
        pageToken: "page_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetirementConstraintsListContributionSummariesRequest$;
(function (RetirementConstraintsListContributionSummariesRequest$) {
    /** @deprecated use `RetirementConstraintsListContributionSummariesRequest$inboundSchema` instead. */
    RetirementConstraintsListContributionSummariesRequest$.inboundSchema = exports.RetirementConstraintsListContributionSummariesRequest$inboundSchema;
    /** @deprecated use `RetirementConstraintsListContributionSummariesRequest$outboundSchema` instead. */
    RetirementConstraintsListContributionSummariesRequest$.outboundSchema = exports.RetirementConstraintsListContributionSummariesRequest$outboundSchema;
})(RetirementConstraintsListContributionSummariesRequest$ || (exports.RetirementConstraintsListContributionSummariesRequest$ = RetirementConstraintsListContributionSummariesRequest$ = {}));
/** @internal */
exports.RetirementConstraintsListContributionSummariesResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    ListContributionSummariesResponse: components
        .ListContributionSummariesResponse$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "ListContributionSummariesResponse": "listContributionSummariesResponse",
        "Status": "status",
    });
});
/** @internal */
exports.RetirementConstraintsListContributionSummariesResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    listContributionSummariesResponse: components
        .ListContributionSummariesResponse$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        listContributionSummariesResponse: "ListContributionSummariesResponse",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetirementConstraintsListContributionSummariesResponse$;
(function (RetirementConstraintsListContributionSummariesResponse$) {
    /** @deprecated use `RetirementConstraintsListContributionSummariesResponse$inboundSchema` instead. */
    RetirementConstraintsListContributionSummariesResponse$.inboundSchema = exports.RetirementConstraintsListContributionSummariesResponse$inboundSchema;
    /** @deprecated use `RetirementConstraintsListContributionSummariesResponse$outboundSchema` instead. */
    RetirementConstraintsListContributionSummariesResponse$.outboundSchema = exports.RetirementConstraintsListContributionSummariesResponse$outboundSchema;
})(RetirementConstraintsListContributionSummariesResponse$ || (exports.RetirementConstraintsListContributionSummariesResponse$ = RetirementConstraintsListContributionSummariesResponse$ = {}));
