// @apexfintechsolutions/ascend-sdk/models/operations/retirementconstraintsretrievedistributionconstraints.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetirementConstraintsRetrieveDistributionConstraintsResponse$ = exports.RetirementConstraintsRetrieveDistributionConstraintsResponse$outboundSchema = exports.RetirementConstraintsRetrieveDistributionConstraintsResponse$inboundSchema = exports.RetirementConstraintsRetrieveDistributionConstraintsRequest$ = exports.RetirementConstraintsRetrieveDistributionConstraintsRequest$outboundSchema = exports.RetirementConstraintsRetrieveDistributionConstraintsRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.RetirementConstraintsRetrieveDistributionConstraintsRequest$inboundSchema = z.object({
    account_id: z.string(),
    RetrieveDistributionConstraintsRequestCreate: components.RetrieveDistributionConstraintsRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "RetrieveDistributionConstraintsRequestCreate": "retrieveDistributionConstraintsRequestCreate",
    });
});
/** @internal */
exports.RetirementConstraintsRetrieveDistributionConstraintsRequest$outboundSchema = z.object({
    accountId: z.string(),
    retrieveDistributionConstraintsRequestCreate: components.RetrieveDistributionConstraintsRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        retrieveDistributionConstraintsRequestCreate: "RetrieveDistributionConstraintsRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetirementConstraintsRetrieveDistributionConstraintsRequest$;
(function (RetirementConstraintsRetrieveDistributionConstraintsRequest$) {
    /** @deprecated use `RetirementConstraintsRetrieveDistributionConstraintsRequest$inboundSchema` instead. */
    RetirementConstraintsRetrieveDistributionConstraintsRequest$.inboundSchema = exports.RetirementConstraintsRetrieveDistributionConstraintsRequest$inboundSchema;
    /** @deprecated use `RetirementConstraintsRetrieveDistributionConstraintsRequest$outboundSchema` instead. */
    RetirementConstraintsRetrieveDistributionConstraintsRequest$.outboundSchema = exports.RetirementConstraintsRetrieveDistributionConstraintsRequest$outboundSchema;
})(RetirementConstraintsRetrieveDistributionConstraintsRequest$ || (exports.RetirementConstraintsRetrieveDistributionConstraintsRequest$ = RetirementConstraintsRetrieveDistributionConstraintsRequest$ = {}));
/** @internal */
exports.RetirementConstraintsRetrieveDistributionConstraintsResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    DistributionConstraints: components.DistributionConstraints$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "DistributionConstraints": "distributionConstraints",
        "Status": "status",
    });
});
/** @internal */
exports.RetirementConstraintsRetrieveDistributionConstraintsResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    distributionConstraints: components.DistributionConstraints$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        distributionConstraints: "DistributionConstraints",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RetirementConstraintsRetrieveDistributionConstraintsResponse$;
(function (RetirementConstraintsRetrieveDistributionConstraintsResponse$) {
    /** @deprecated use `RetirementConstraintsRetrieveDistributionConstraintsResponse$inboundSchema` instead. */
    RetirementConstraintsRetrieveDistributionConstraintsResponse$.inboundSchema = exports.RetirementConstraintsRetrieveDistributionConstraintsResponse$inboundSchema;
    /** @deprecated use `RetirementConstraintsRetrieveDistributionConstraintsResponse$outboundSchema` instead. */
    RetirementConstraintsRetrieveDistributionConstraintsResponse$.outboundSchema = exports.RetirementConstraintsRetrieveDistributionConstraintsResponse$outboundSchema;
})(RetirementConstraintsRetrieveDistributionConstraintsResponse$ || (exports.RetirementConstraintsRetrieveDistributionConstraintsResponse$ = RetirementConstraintsRetrieveDistributionConstraintsResponse$ = {}));
