// @apexfintechsolutions/ascend-sdk/models/operations/subscribergetpushsubscriptiondelivery.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriberGetPushSubscriptionDeliveryResponse$ = exports.SubscriberGetPushSubscriptionDeliveryResponse$outboundSchema = exports.SubscriberGetPushSubscriptionDeliveryResponse$inboundSchema = exports.SubscriberGetPushSubscriptionDeliveryRequest$ = exports.SubscriberGetPushSubscriptionDeliveryRequest$outboundSchema = exports.SubscriberGetPushSubscriptionDeliveryRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.SubscriberGetPushSubscriptionDeliveryRequest$inboundSchema = z.object({
    subscription_id: z.string(),
    delivery_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "subscription_id": "subscriptionId",
        "delivery_id": "deliveryId",
    });
});
/** @internal */
exports.SubscriberGetPushSubscriptionDeliveryRequest$outboundSchema = z.object({
    subscriptionId: z.string(),
    deliveryId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        subscriptionId: "subscription_id",
        deliveryId: "delivery_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriberGetPushSubscriptionDeliveryRequest$;
(function (SubscriberGetPushSubscriptionDeliveryRequest$) {
    /** @deprecated use `SubscriberGetPushSubscriptionDeliveryRequest$inboundSchema` instead. */
    SubscriberGetPushSubscriptionDeliveryRequest$.inboundSchema = exports.SubscriberGetPushSubscriptionDeliveryRequest$inboundSchema;
    /** @deprecated use `SubscriberGetPushSubscriptionDeliveryRequest$outboundSchema` instead. */
    SubscriberGetPushSubscriptionDeliveryRequest$.outboundSchema = exports.SubscriberGetPushSubscriptionDeliveryRequest$outboundSchema;
})(SubscriberGetPushSubscriptionDeliveryRequest$ || (exports.SubscriberGetPushSubscriptionDeliveryRequest$ = SubscriberGetPushSubscriptionDeliveryRequest$ = {}));
/** @internal */
exports.SubscriberGetPushSubscriptionDeliveryResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    PushSubscriptionDelivery: components.PushSubscriptionDelivery$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "PushSubscriptionDelivery": "pushSubscriptionDelivery",
        "Status": "status",
    });
});
/** @internal */
exports.SubscriberGetPushSubscriptionDeliveryResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    pushSubscriptionDelivery: components.PushSubscriptionDelivery$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        pushSubscriptionDelivery: "PushSubscriptionDelivery",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriberGetPushSubscriptionDeliveryResponse$;
(function (SubscriberGetPushSubscriptionDeliveryResponse$) {
    /** @deprecated use `SubscriberGetPushSubscriptionDeliveryResponse$inboundSchema` instead. */
    SubscriberGetPushSubscriptionDeliveryResponse$.inboundSchema = exports.SubscriberGetPushSubscriptionDeliveryResponse$inboundSchema;
    /** @deprecated use `SubscriberGetPushSubscriptionDeliveryResponse$outboundSchema` instead. */
    SubscriberGetPushSubscriptionDeliveryResponse$.outboundSchema = exports.SubscriberGetPushSubscriptionDeliveryResponse$outboundSchema;
})(SubscriberGetPushSubscriptionDeliveryResponse$ || (exports.SubscriberGetPushSubscriptionDeliveryResponse$ = SubscriberGetPushSubscriptionDeliveryResponse$ = {}));
