// @apexfintechsolutions/ascend-sdk/models/operations/subscriberupdatepushsubscription.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriberUpdatePushSubscriptionResponse$ = exports.SubscriberUpdatePushSubscriptionResponse$outboundSchema = exports.SubscriberUpdatePushSubscriptionResponse$inboundSchema = exports.SubscriberUpdatePushSubscriptionRequest$ = exports.SubscriberUpdatePushSubscriptionRequest$outboundSchema = exports.SubscriberUpdatePushSubscriptionRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.SubscriberUpdatePushSubscriptionRequest$inboundSchema = z.object({
    subscription_id: z.string(),
    update_mask: z.string().optional(),
    PushSubscriptionUpdate: components.PushSubscriptionUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "subscription_id": "subscriptionId",
        "update_mask": "updateMask",
        "PushSubscriptionUpdate": "pushSubscriptionUpdate",
    });
});
/** @internal */
exports.SubscriberUpdatePushSubscriptionRequest$outboundSchema = z.object({
    subscriptionId: z.string(),
    updateMask: z.string().optional(),
    pushSubscriptionUpdate: components.PushSubscriptionUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        subscriptionId: "subscription_id",
        updateMask: "update_mask",
        pushSubscriptionUpdate: "PushSubscriptionUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriberUpdatePushSubscriptionRequest$;
(function (SubscriberUpdatePushSubscriptionRequest$) {
    /** @deprecated use `SubscriberUpdatePushSubscriptionRequest$inboundSchema` instead. */
    SubscriberUpdatePushSubscriptionRequest$.inboundSchema = exports.SubscriberUpdatePushSubscriptionRequest$inboundSchema;
    /** @deprecated use `SubscriberUpdatePushSubscriptionRequest$outboundSchema` instead. */
    SubscriberUpdatePushSubscriptionRequest$.outboundSchema = exports.SubscriberUpdatePushSubscriptionRequest$outboundSchema;
})(SubscriberUpdatePushSubscriptionRequest$ || (exports.SubscriberUpdatePushSubscriptionRequest$ = SubscriberUpdatePushSubscriptionRequest$ = {}));
/** @internal */
exports.SubscriberUpdatePushSubscriptionResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    PushSubscription: components.PushSubscription$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "PushSubscription": "pushSubscription",
        "Status": "status",
    });
});
/** @internal */
exports.SubscriberUpdatePushSubscriptionResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    pushSubscription: components.PushSubscription$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        pushSubscription: "PushSubscription",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriberUpdatePushSubscriptionResponse$;
(function (SubscriberUpdatePushSubscriptionResponse$) {
    /** @deprecated use `SubscriberUpdatePushSubscriptionResponse$inboundSchema` instead. */
    SubscriberUpdatePushSubscriptionResponse$.inboundSchema = exports.SubscriberUpdatePushSubscriptionResponse$inboundSchema;
    /** @deprecated use `SubscriberUpdatePushSubscriptionResponse$outboundSchema` instead. */
    SubscriberUpdatePushSubscriptionResponse$.outboundSchema = exports.SubscriberUpdatePushSubscriptionResponse$outboundSchema;
})(SubscriberUpdatePushSubscriptionResponse$ || (exports.SubscriberUpdatePushSubscriptionResponse$ = SubscriberUpdatePushSubscriptionResponse$ = {}));
