// @apexfintechsolutions/ascend-sdk/models/operations/wirewithdrawalschedulesgetwirewithdrawalschedule.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$ = exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$outboundSchema = exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$inboundSchema = exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$ = exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$outboundSchema = exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$inboundSchema = z.object({
    account_id: z.string(),
    wireWithdrawalSchedule_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "wireWithdrawalSchedule_id": "wireWithdrawalScheduleId",
    });
});
/** @internal */
exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$outboundSchema = z.object({
    accountId: z.string(),
    wireWithdrawalScheduleId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        wireWithdrawalScheduleId: "wireWithdrawalSchedule_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$;
(function (WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$) {
    /** @deprecated use `WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$inboundSchema` instead. */
    WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$.inboundSchema = exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$inboundSchema;
    /** @deprecated use `WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$outboundSchema` instead. */
    WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$.outboundSchema = exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$outboundSchema;
})(WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$ || (exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$ = WireWithdrawalSchedulesGetWireWithdrawalScheduleRequest$ = {}));
/** @internal */
exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    WireWithdrawalSchedule: components.WireWithdrawalSchedule$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "WireWithdrawalSchedule": "wireWithdrawalSchedule",
        "Status": "status",
    });
});
/** @internal */
exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    wireWithdrawalSchedule: components.WireWithdrawalSchedule$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        wireWithdrawalSchedule: "WireWithdrawalSchedule",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$;
(function (WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$) {
    /** @deprecated use `WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$inboundSchema` instead. */
    WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$.inboundSchema = exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$inboundSchema;
    /** @deprecated use `WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$outboundSchema` instead. */
    WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$.outboundSchema = exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$outboundSchema;
})(WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$ || (exports.WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$ = WireWithdrawalSchedulesGetWireWithdrawalScheduleResponse$ = {}));
