// @apexfintechsolutions/ascend-sdk/models/operations/wirewithdrawalschedulescancelwirewithdrawalschedule.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$ = exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$outboundSchema = exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$inboundSchema = exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$ = exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$outboundSchema = exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$inboundSchema = z.object({
    account_id: z.string(),
    wireWithdrawalSchedule_id: z.string(),
    CancelWireWithdrawalScheduleRequestCreate: components.CancelWireWithdrawalScheduleRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "wireWithdrawalSchedule_id": "wireWithdrawalScheduleId",
        "CancelWireWithdrawalScheduleRequestCreate": "cancelWireWithdrawalScheduleRequestCreate",
    });
});
/** @internal */
exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$outboundSchema = z.object({
    accountId: z.string(),
    wireWithdrawalScheduleId: z.string(),
    cancelWireWithdrawalScheduleRequestCreate: components.CancelWireWithdrawalScheduleRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        wireWithdrawalScheduleId: "wireWithdrawalSchedule_id",
        cancelWireWithdrawalScheduleRequestCreate: "CancelWireWithdrawalScheduleRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$;
(function (WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$) {
    /** @deprecated use `WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$inboundSchema` instead. */
    WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$.inboundSchema = exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$inboundSchema;
    /** @deprecated use `WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$outboundSchema` instead. */
    WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$.outboundSchema = exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$outboundSchema;
})(WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$ || (exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$ = WireWithdrawalSchedulesCancelWireWithdrawalScheduleRequest$ = {}));
/** @internal */
exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    WireWithdrawalSchedule: components.WireWithdrawalSchedule$inboundSchema
        .optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "WireWithdrawalSchedule": "wireWithdrawalSchedule",
        "Status": "status",
    });
});
/** @internal */
exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    wireWithdrawalSchedule: components.WireWithdrawalSchedule$outboundSchema
        .optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        wireWithdrawalSchedule: "WireWithdrawalSchedule",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$;
(function (WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$) {
    /** @deprecated use `WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$inboundSchema` instead. */
    WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$.inboundSchema = exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$inboundSchema;
    /** @deprecated use `WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$outboundSchema` instead. */
    WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$.outboundSchema = exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$outboundSchema;
})(WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$ || (exports.WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$ = WireWithdrawalSchedulesCancelWireWithdrawalScheduleResponse$ = {}));
