// @apexfintechsolutions/ascend-sdk/models/operations/wirewithdrawalscancelwirewithdrawal.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.WireWithdrawalsCancelWireWithdrawalResponse$ = exports.WireWithdrawalsCancelWireWithdrawalResponse$outboundSchema = exports.WireWithdrawalsCancelWireWithdrawalResponse$inboundSchema = exports.WireWithdrawalsCancelWireWithdrawalRequest$ = exports.WireWithdrawalsCancelWireWithdrawalRequest$outboundSchema = exports.WireWithdrawalsCancelWireWithdrawalRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.WireWithdrawalsCancelWireWithdrawalRequest$inboundSchema = z.object({
    account_id: z.string(),
    wireWithdrawal_id: z.string(),
    CancelWireWithdrawalRequestCreate: components.CancelWireWithdrawalRequestCreate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_id": "accountId",
        "wireWithdrawal_id": "wireWithdrawalId",
        "CancelWireWithdrawalRequestCreate": "cancelWireWithdrawalRequestCreate",
    });
});
/** @internal */
exports.WireWithdrawalsCancelWireWithdrawalRequest$outboundSchema = z.object({
    accountId: z.string(),
    wireWithdrawalId: z.string(),
    cancelWireWithdrawalRequestCreate: components.CancelWireWithdrawalRequestCreate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountId: "account_id",
        wireWithdrawalId: "wireWithdrawal_id",
        cancelWireWithdrawalRequestCreate: "CancelWireWithdrawalRequestCreate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalsCancelWireWithdrawalRequest$;
(function (WireWithdrawalsCancelWireWithdrawalRequest$) {
    /** @deprecated use `WireWithdrawalsCancelWireWithdrawalRequest$inboundSchema` instead. */
    WireWithdrawalsCancelWireWithdrawalRequest$.inboundSchema = exports.WireWithdrawalsCancelWireWithdrawalRequest$inboundSchema;
    /** @deprecated use `WireWithdrawalsCancelWireWithdrawalRequest$outboundSchema` instead. */
    WireWithdrawalsCancelWireWithdrawalRequest$.outboundSchema = exports.WireWithdrawalsCancelWireWithdrawalRequest$outboundSchema;
})(WireWithdrawalsCancelWireWithdrawalRequest$ || (exports.WireWithdrawalsCancelWireWithdrawalRequest$ = WireWithdrawalsCancelWireWithdrawalRequest$ = {}));
/** @internal */
exports.WireWithdrawalsCancelWireWithdrawalResponse$inboundSchema = z.object({
    HttpMeta: components.HTTPMetadata$inboundSchema,
    WireWithdrawal: components.WireWithdrawal$inboundSchema.optional(),
    Status: components.Status$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "HttpMeta": "httpMeta",
        "WireWithdrawal": "wireWithdrawal",
        "Status": "status",
    });
});
/** @internal */
exports.WireWithdrawalsCancelWireWithdrawalResponse$outboundSchema = z.object({
    httpMeta: components.HTTPMetadata$outboundSchema,
    wireWithdrawal: components.WireWithdrawal$outboundSchema.optional(),
    status: components.Status$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        httpMeta: "HttpMeta",
        wireWithdrawal: "WireWithdrawal",
        status: "Status",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WireWithdrawalsCancelWireWithdrawalResponse$;
(function (WireWithdrawalsCancelWireWithdrawalResponse$) {
    /** @deprecated use `WireWithdrawalsCancelWireWithdrawalResponse$inboundSchema` instead. */
    WireWithdrawalsCancelWireWithdrawalResponse$.inboundSchema = exports.WireWithdrawalsCancelWireWithdrawalResponse$inboundSchema;
    /** @deprecated use `WireWithdrawalsCancelWireWithdrawalResponse$outboundSchema` instead. */
    WireWithdrawalsCancelWireWithdrawalResponse$.outboundSchema = exports.WireWithdrawalsCancelWireWithdrawalResponse$outboundSchema;
})(WireWithdrawalsCancelWireWithdrawalResponse$ || (exports.WireWithdrawalsCancelWireWithdrawalResponse$ = WireWithdrawalsCancelWireWithdrawalResponse$ = {}));
